--- /etc/profile.orig	Sat Dec  5 20:46:25 2009
+++ /etc/profile	Sat Dec  5 20:25:30 2009
@@ -3,7 +3,7 @@
 
 PATH="/usr/local/bin:/usr/bin:/bin"
 EDITOR="/bin/vi"	# needed for packages like cron
-TERM="vt100"		# Basic terminal capab. For screen etc.
+: ${TERM:=vt100}	# Basic terminal capab. For screen etc.
 DBUS_SESSION_BUS_ADDRESS="unix:path=/tmp/session_bus_socket"
 DBUS_SYSTEM_BUS_ADDRESS="unix:path=/var/run/dbus/system_bus_socket"
 
