diff --git a/etc/init.d/swaphack.sh b/etc/init.d/swaphack.sh
index e5226b4..237db85 100755
--- a/etc/init.d/swaphack.sh
+++ b/etc/init.d/swaphack.sh
@@ -43,7 +43,7 @@ else
 fi
 
 echo "Configuring Block Device and VM systems"
-echo cfq > /sys/block/mmcblk0/queue/scheduler
+#echo cfq > /sys/block/mmcblk0/queue/scheduler 
 echo 6000 > /proc/sys/vm/dirty_expire_centisecs
 #echo 3071 > /proc/sys/vm/min_free_kbytes
 #echo '8 3 3 1' > /proc/sys/vm/wmark_highlow_factors
