diff --git a/media/cryptofs/apps/usr/palm/applications/com.palm.app.email/mail/source/DarkSwipeable.js b/media/cryptofs/apps/usr/palm/applications/com.palm.app.email/mail/source/DarkSwipeable.js
index dfc95c3..093c34f 100644
--- a/media/cryptofs/apps/usr/palm/applications/com.palm.app.email/mail/source/DarkSwipeable.js
+++ b/media/cryptofs/apps/usr/palm/applications/com.palm.app.email/mail/source/DarkSwipeable.js
@@ -10,7 +10,7 @@ enyo.kind({
 	],
 	
 	dragHandler: function(inSender, inEvent) {
-		var dx = this.getDx(inEvent);
+		var dx = this.getDx(inEvent) - this.dragOffset;
 
 		if (this.handlingDrag) {
 			if (this.hasNode()) {
diff --git a/usr/palm/frameworks/enyo/0.10/framework/build/enyo-build.js b/usr/palm/frameworks/enyo/0.10/framework/build/enyo-build.js
index dcb8329..45e2b21 100644
--- a/usr/palm/frameworks/enyo/0.10/framework/build/enyo-build.js
+++ b/usr/palm/frameworks/enyo/0.10/framework/build/enyo-build.js
@@ -8467,6 +8467,7 @@ confirmWhenAutoHidden: !1,
 allowLeft: !0
 },
 triggerRatio: .35,
+dragOffset: 0,
 className: "enyo-item enyo-swipeableitem",
 lastConfirmIndex: null,
 events: {
@@ -8504,10 +8505,10 @@ flickHandler: function(a, b) {
 return this.handlingDrag;
 },
 dragstartHandler: function(a, b) {
-return this.resetPosition(), this.swipeable && b.horizontal && !this.confirmShowing && this.hasNode() ? (this.triggerDistance = this.fetchTriggerDistance(), this.index = b.rowIndex, this.handlingDrag = !0, !0) : this.fire("ondragstart", b);
+return this.resetPosition(), this.swipeable && b.horizontal && !this.confirmShowing && this.hasNode() ? (this.dragOffset = this.getDx(b), this.triggerDistance = this.fetchTriggerDistance(), this.index = b.rowIndex, this.handlingDrag = !0, !0) : this.fire("ondragstart", b);
 },
 dragHandler: function(a, b) {
-var c = this.getDx(b);
+var c = this.getDx(b) - this.dragOffset;
 if (this.handlingDrag) return this.hasNode() ? (this.node.style.webkitTransform = "translate3d(" + c + "px, 0, 0)", this.doDrag(c)) : enyo.log("drag with no node!"), !0;
 },
 dragfinishHandler: function(a, b) {
@@ -10485,6 +10486,7 @@ className: "enyo-toaster",
 published: {
 flyInFrom: "bottom"
 },
+dragOffset: 0,
 chrome: [ {
 name: "animator",
 kind: enyo.Animator,
@@ -10524,11 +10526,11 @@ isHorizontal: function() {
 return this.flyInFrom == "right" || this.flyInFrom == "left";
 },
 dragstartHandler: function(a, b) {
-this.isDraggableEvent(b) && (this.dragging = !0, this.dragD0 = 0);
+this.isDraggableEvent(b) && (this.dragging = !0, this.dragD0 = 0, this.dragOffset = this.isHorizontal() ? b.dx : b.dy);
 },
 dragHandler: function(a, b) {
 if (this.dragging) {
-var c = this.isHorizontal() ? b.dx : b.dy;
+var c = this.isHorizontal() ? b.dx - this.dragOffset : b.dy - this.dragOffset;
 this.dragD = this.dragD0 - c, this.dragD0 = c;
 if (this.dragD0 * (this.flyInFrom == "right" || this.flyInFrom == "bottom" ? 1 : -1) > 0) {
 var d = "translate3d(" + (this.isHorizontal() ? this.dragD0 + "px,0,0)" : "0," + this.dragD0 + "px,0)");
@@ -10815,10 +10817,13 @@ this.shouldDragSelect = b < this.slidePosition;
 if (b < this.dragMin || b > this.dragMax && !this.overSliding || b < this.dragMax && this.overSliding) return {
 select: this.getDragSelect()
 };
-if (this.overSliding && !this.dismissible) {
 var c = this.lastDragDx || 0;
+if (this.overSliding && !this.dismissible) {
 b = (a.dx - c) / 4 + this.slidePosition;
 }
+else {
+b = (a.dx - c) + this.slidePosition;
+}
 this.lastDragDx = a.dx;
 var d = Math.max(this.dragMin, Math.min(b, this.overSliding ? 1e9 : this.dragMax));
 this.pendingDragMove = this._drag(d);
