diff --git a/etc/palm/luna.conf b/etc/palm/luna.conf
index 73841a8..cacd1a4 100644
--- a/etc/palm/luna.conf
+++ b/etc/palm/luna.conf
@@ -13,11 +13,11 @@ LaunchPointsPath=/var/luna/launchpoints
 PreferencesPath=/var/luna/preferences
 DisplayWidth=1024
 DisplayHeight=768
-DisplayNumBuffers=3
+DisplayNumBuffers=16
 MaxPenMoveFreq=30
 MinPenMoveFreq=10
-MoveMinX=16
-MoveMinY=12
+MoveMinX=4
+MoveMinY=3
 MaxPaintLoad=6
 MaxGestureChangeFreq=30
 MinGestureChangeFreq=20
@@ -48,16 +48,16 @@ CardLimit=-1
 AppsToAllowInLowMemory=com.palm.app.phone;com.palm.app.contacts;com.palm.app.messaging
 
 [TouchEvents]
-TapRadiusMax=25
-TapRadiusMin=5
+TapRadiusMax=8
+TapRadiusMin=2
 TapRadiusShrinkPerc=10
 TapRadiusShrinkGranMs=200
 DoubleClickDuration=300
 EnableForWebApps=true
 
 [VTrackBall]
-PixelsPerMoveH=14
-PixelsPerMoveV=26
+PixelsPerMoveH=4
+PixelsPerMoveV=6
 AccelRateH1=350
 AccelRateV1=200
 AccelConstH1=1
