diff --git a/etc/palm/luna.conf b/etc/palm/luna.conf
index 73841a8..953c698 100644
--- a/etc/palm/luna.conf
+++ b/etc/palm/luna.conf
@@ -13,11 +13,11 @@ LaunchPointsPath=/var/luna/launchpoints
 PreferencesPath=/var/luna/preferences
 DisplayWidth=1024
 DisplayHeight=768
-DisplayNumBuffers=3
+DisplayNumBuffers=16
 MaxPenMoveFreq=30
 MinPenMoveFreq=10
-MoveMinX=16
-MoveMinY=12
+MoveMinX=4
+MoveMinY=3
 MaxPaintLoad=6
 MaxGestureChangeFreq=30
 MinGestureChangeFreq=20
@@ -48,8 +48,8 @@ CardLimit=-1
 AppsToAllowInLowMemory=com.palm.app.phone;com.palm.app.contacts;com.palm.app.messaging
 
 [TouchEvents]
-TapRadiusMax=25
-TapRadiusMin=5
+TapRadiusMax=10
+TapRadiusMin=2
 TapRadiusShrinkPerc=10
 TapRadiusShrinkGranMs=200
 DoubleClickDuration=300
@@ -74,7 +74,7 @@ PiranhaDrawColoredOutlines=false
 PiranhaDisplayFps=false
 ShowGestures=false
 FailAllMigration=false
-PerformanceLogs=true
+PerformanceLogs=false
 LauncherAtlasStatistics=false
 DumpLauncherAtlas=false
 
@@ -85,7 +85,7 @@ Applications=com.palm.app.phone;com.palm.app.email;com.palm.app.calendar;com.pal
 Applications=com.palm.app.phone
 
 [KeepAliveUntilMemPressure]
-Applications=com.palm.app.browser
+Applications=
 
 [AccelCompositingDisabled]
 Applications=com.palm.app.phone
@@ -125,10 +125,10 @@ GestureRepeaterPort=5001
 DemoMode=false
 
 [CpuShare]
-UiMainLow=512
-UiOtherLow=128
-JavaLow=128
-WebLow=64
+UiMainLow=1024
+UiOtherLow=64
+JavaLow=64
+WebLow=32
 Default=1024
 
 [Launcher]
diff --git a/etc/palm/lunaAnimations.conf b/etc/palm/lunaAnimations.conf
index 1fd9ba2..687128c 100644
--- a/etc/palm/lunaAnimations.conf
+++ b/etc/palm/lunaAnimations.conf
@@ -1,6 +1,6 @@
 [FPS]
-normalFPS=35
-slowFPS=20
+normalFPS=100
+slowFPS=80
 
 [Cards]
 cardLaunchDuration=400
