diff --git a/usr/sbin/suspend_action b/usr/sbin/suspend_action
index e6f16aa..5def92e 100755
--- a/usr/sbin/suspend_action
+++ b/usr/sbin/suspend_action
@@ -1,17 +1,17 @@
 #!/bin/sh
 
-echo -e "powerd: System going to sleep now\n" | wall 
+# echo -e "powerd: System going to sleep now\n" | wall 
 
 # Raise frequency to max so we'll resume at max
 CPUFREQDIR=/sys/devices/system/cpu/cpu0/cpufreq
 min=`cat $CPUFREQDIR/scaling_min_freq`
 max=`cat $CPUFREQDIR/scaling_max_freq`
 echo $max >  $CPUFREQDIR/scaling_min_freq 
 
 # Initiate suspend
 echo mem > /sys/power/state
 
 echo $min >  $CPUFREQDIR/scaling_min_freq
 
-echo -e "powerd: System woke up.\n" | wall 
+# echo -e "powerd: System woke up.\n" | wall 
 
