diff --git a/etc/palm/downloadManager.conf b/etc/palm/downloadManager.conf
index 9581289..613f5f1 100644
--- a/etc/palm/downloadManager.conf
+++ b/etc/palm/downloadManager.conf
@@ -8,3 +8,11 @@ AutoResume=true
 BeAggressive__Bee_Eee_AGGRESSIVE=true
 AppCompatibilityMode=false
 
+#Tested to work with WebOS 1.3.5
+#Mark R (Diomark) 01/04/10
+#unthrottles the download manager from 64kb/sec to unlimited
+
+[DownloadManager]
+MaxQueueLength=128
+MaxConcurrent=2
+MaxRecvSpeed=-1
