diff --git a/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js b/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
index 2888b7c..dfeb3ba 100644
--- a/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
+++ b/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
@@ -180,8 +180,8 @@ handleSystemTimeUpdate: function(payload) {
 		return;
 	this.startClock();	
 	
-	if(payload.NITZValid != undefined && payload.NITZValid === false)
-		this.showTimezoneErrorAlert();
+	//if(payload.NITZValid != undefined && payload.NITZValid === false)
+	//	this.showTimezoneErrorAlert();
 },
 
 registerTimeFormat:function(){
