diff --git a/etc/profile b/etc/profile
index 3bb165e..bb41e1c 100644
--- a/etc/profile
+++ b/etc/profile
@@ -3,7 +3,7 @@
 
 PATH="/usr/local/bin:/usr/bin:/bin"
 EDITOR="/bin/vi"	# needed for packages like cron
-TERM="vt100"		# Basic terminal capab. For screen etc.
+: ${TERM:=vt100}	# Basic terminal capab. For screen etc.
 DBUS_SESSION_BUS_ADDRESS="unix:path=/tmp/session_bus_socket"
 DBUS_SYSTEM_BUS_ADDRESS="unix:path=/var/run/dbus/system_bus_socket"
 
