diff --git a/etc/palm/luna-platform.conf b/etc/palm/luna-platform.conf
index c9cca90..a320615 100644
--- a/etc/palm/luna-platform.conf
+++ b/etc/palm/luna-platform.conf
@@ -12,10 +12,10 @@ TabletUi=true
 DisplayUiRotates=true
 HomeButtonOrientationAngle=270
 EmuModeOrientationAngle=0
-ActiveCardWindowRatio=0.55
+ActiveCardWindowRatio=0.7
 NonActiveCardWindowRatio=0.50
 CardGroupRotFactor=90
-GapBetweenCardGroups=30
+GapBetweenCardGroups=5
 CardGroupingXDistanceFactor=0.35
 ModalWindowWidth=320
 ModalWindowHeight=480
