diff --git a/usr/palm/applications/com.palm.app.photos/app/models/dock-slideshow-model.js b/usr/palm/applications/com.palm.app.photos/app/models/dock-slideshow-model.js
index ae2a470..cf641b9 100644
--- a/usr/palm/applications/com.palm.app.photos/app/models/dock-slideshow-model.js
+++ b/usr/palm/applications/com.palm.app.photos/app/models/dock-slideshow-model.js
@@ -35,6 +35,8 @@ function DockSlideshowModel(settings) {
 
 	var timeCommandToMS = {
 		"menu-time-5s": 5 * 1000,
+		"menu-time-30s": 30 * 1000,
+		"menu-time-1m": 1 * 60 * 1000,
 		"menu-time-5m": 5 * 60 * 1000,
 		"menu-time-30m": 30 * 60 * 1000,
 		"menu-time-1h": 60 * 60 * 1000
@@ -55,6 +57,12 @@ function DockSlideshowModel(settings) {
 			label: $L("5 seconds"),
 			command: "menu-time-5s"
 		},{
+			label: $L("30 seconds"),
+			command: "menu-time-30s"
+		},{	
+			label: $L("1 minute"),
+			command: "menu-time-1m"
+		},{		
 			label: $L("5 minutes"),
 			command: "menu-time-5m"
 		},{
@@ -144,6 +152,8 @@ function DockSlideshowModel(settings) {
 			this.saveSettings();
 			return true;
 		case "menu-time-5s": // falls through
+		case "menu-time-30s": // falls through
+		case "menu-time-1m": // falls through
 		case "menu-time-5m": // falls through
 		case "menu-time-30m": // falls through
 		case "menu-time-1h": 
