diff --git a/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js b/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js
index 484cd45..da7e786 100644
--- a/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js
+++ b/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js
@@ -110,7 +110,7 @@ var ChatviewAssistant = Class.create({
 		
 		this.handleWindowResize = this.handleWindowResize.bindAsEventListener(this);
 		this.handleMessageTap = this.handleMessageTap.bindAsEventListener(this);
-		this.handleMessageDelete = this.handleMessageDelete.bindAsEventListener(this);
+		/*this.handleMessageDelete = this.handleMessageDelete.bindAsEventListener(this);*/
 		this.renderTransportPickerPopup = this.renderTransportPickerPopup.bindAsEventListener(this);
 		this.considerForSend = this.considerForSend.bind(this);
 		this.attachFilePicker = this.attachFilePicker.bind(this);
@@ -149,7 +149,7 @@ var ChatviewAssistant = Class.create({
 			dividerTemplate: 'chatview/separator',
 			dividerFunction: this.chatListDividerFunction.bind(this),
 			onItemRendered: this.itemRendered.bind(this),
-			swipeToDelete: true,
+			swipeToDelete: false,
 			uniquenessProperty: 'messageId'
 		};
 		this.chatListModel = {}; // This is used to reset the list by calling modelChanged
@@ -175,7 +175,7 @@ var ChatviewAssistant = Class.create({
 		//this.chatListWidget.observe("click", this.onClickHandler);
 		
 		this.chatListWidget.observe(Mojo.Event.listTap, this.handleMessageTap);
-		this.chatListWidget.observe(Mojo.Event.listDelete, this.handleMessageDelete);
+		/*this.chatListWidget.observe(Mojo.Event.listDelete, this.handleMessageDelete);*/
 		this.sendButton.observe(Mojo.Event.tap, this.considerForSend);
 		this.attachButton.observe(Mojo.Event.tap, this.attachFilePicker);
 		
@@ -1953,7 +1953,6 @@ var ChatviewAssistant = Class.create({
 						{label: $L('Forward'), command: 'forward-cmd'},
 						{label: $L('Forward Via Email'), command: 'forward-as-email-cmd'},
 						{label: $L('Copy Text'), command: 'copy-cmd'},
-						{label: $L('Delete'), command: 'delete-cmd'}
 					]
 				});
 
diff --git a/usr/palm/applications/com.palm.app.messaging/app/controllers/listview-assistant.js b/usr/palm/applications/com.palm.app.messaging/app/controllers/listview-assistant.js
index c301f25..9bc7bc9 100644
--- a/usr/palm/applications/com.palm.app.messaging/app/controllers/listview-assistant.js
+++ b/usr/palm/applications/com.palm.app.messaging/app/controllers/listview-assistant.js
@@ -139,7 +139,7 @@ var ListviewAssistant = Class.create(App.Scene, {
 		this.filterImmediate = this.filterImmediate.bind(this);
 		this.filter = this.filter.bind(this);
 		this.handleHistoryListTap = this.handleHistoryListTap.bindAsEventListener(this);
-		this.handleHistoryListDelete = this.handleHistoryListDelete.bindAsEventListener(this);
+		/* this.handleHistoryListDelete = this.handleHistoryListDelete.bindAsEventListener(this);*/
 		this.handleBuddyListTap = this.handleBuddyListTap.bindAsEventListener(this);
 		this.handleAvailabilityPickerTap = this.handleAvailabilityPickerTap.bindAsEventListener(this);
 		this.customMessageUpdate = this.customMessageUpdate.bind(this);
@@ -254,7 +254,7 @@ var ListviewAssistant = Class.create(App.Scene, {
 				}
 				return item.dividerText;
 			}.bind(this),
-			swipeToDelete: true,
+			swipeToDelete: false,
 			uniquenessProperty: 'id',
 			delay: 1,
 			optimizedOptIn: true,
@@ -271,8 +271,8 @@ var ListviewAssistant = Class.create(App.Scene, {
 		// observe clicks on the buddy list & history list
 		this.historyListWidget.observe(Mojo.Event.listTap, this.handleHistoryListTap);
 		// observe swipe to delete on the history list
-		this.historyListWidget.observe(Mojo.Event.listDelete, this.handleHistoryListDelete);
-		this.buddyListWidget.observe(Mojo.Event.listTap, this.handleBuddyListTap);
+		/*this.historyListWidget.observe(Mojo.Event.listDelete, this.handleHistoryListDelete);*/
+		/*this.buddyListWidget.observe(Mojo.Event.listTap, this.handleBuddyListTap);*/
 	
 		//set hint text later
 		this.customMessageTextWidgetAttributes = {
diff --git a/usr/palm/applications/com.palm.app.phone/app/controllers/calllog-assistant.js b/usr/palm/applications/com.palm.app.phone/app/controllers/calllog-assistant.js
index 2cb234a..56b8a3c 100644
--- a/usr/palm/applications/com.palm.app.phone/app/controllers/calllog-assistant.js
+++ b/usr/palm/applications/com.palm.app.phone/app/controllers/calllog-assistant.js
@@ -60,7 +60,7 @@ var CalllogAssistant = Class.create({
 			itemsCallback: this.calllogQuery.bind(this),
 	        itemTemplate: 'calllog/listitem',
 			dividerTemplate: 'calllog/listdivider', 
-			swipeToDelete: true,
+			swipeToDelete: false,
 			dividerFunction: this.dividerFunc,
 			emptyTemplate: 'calllog/nil',                                     
 			lookahead: 25,
@@ -90,7 +90,7 @@ var CalllogAssistant = Class.create({
 		var appMenuModel = {
 			visible: true,
 			items: [
-				{label: $L('Clear Call History'), command: this.kClearAllCallsMenuCommand},
+				/*{label: $L('Clear Call History'), command: this.kClearAllCallsMenuCommand},*/
 				{label: $L('Sounds & Ringtones'), command: this.kRingtonesMenuCommand}
 			]
 		};
@@ -136,7 +136,7 @@ var CalllogAssistant = Class.create({
 				itemsCallback: this.calllogQueryMissed.bind(this),
 	            itemTemplate: 'calllog/listitem',
 				dividerTemplate: 'calllog/listdivider',
-				swipeToDelete: true,
+				swipeToDelete: false,
 				dividerFunction: this.dividerFunc,
 				emptyTemplate: 'calllog/nil-missed',
 				lookahead: 25,
