diff --git a/etc/palm/luna.conf b/etc/palm/luna.conf
index 573777f..7b65444 100644
--- a/etc/palm/luna.conf
+++ b/etc/palm/luna.conf
@@ -71,7 +71,7 @@ ShowGestures=false
 FailAllMigration=false
 
 [LaunchAtBoot]
-Applications=com.palm.app.phone;com.palm.app.email;com.palm.app.calendar;com.palm.app.messaging;com.palm.app.contacts;com.palm.app.camera
+Applications=com.palm.app.phone;com.palm.app.email;com.palm.app.calendar;com.palm.app.messaging;com.palm.app.contacts;com.palm.app.camera;com.palm.org.e-lnx.wee.apps.modeswitcher
 
 [KeepAlive]
 Applications=com.palm.app.phone
