diff --git a/etc/palm/luna.patch.rej b/etc/palm/luna.patch.rej
new file mode 100644
index 0000000..80c8851
--- /dev/null
+++ b/etc/palm/luna.patch.rej
@@ -0,0 +1,59 @@
+--- etc/palm/luna.patch
++++ etc/palm/luna.patch
+@@ -13,7 +13,7 @@
+ PreferencesPath=/var/luna/preferences
+ DisplayWidth=320
+ DisplayHeight=480
+-DisplayNumBuffers=3
++DisplayNumBuffers=8
+ MaxPenMoveFreq=30
+ MinPenMoveFreq=10
+ MoveMinX=16
+@@ -16,8 +16,8 @@
+ DisplayNumBuffers=3
+ MaxPenMoveFreq=30
+ MinPenMoveFreq=10
+-MoveMinX=16
+-MoveMinY=12
++MoveMinX=1
++MoveMinY=1
+ MaxPaintLoad=6
+ MaxGestureChangeFreq=30
+ MinGestureChangeFreq=20
+@@ -77,7 +77,7 @@
+ Applications=com.palm.app.phone
+ 
+ [KeepAliveUntilMemPressure]
+-Applications=com.palm.app.browser
++Applications=
+ 
+ [Fonts]
+ Banner=/usr/share/fonts/Coconut-BoldD.ttf
+@@ -101,10 +101,10 @@
+ GestureRepeaterPort=5001
+ 
+ [CpuShare]
+-UiMainLow=512
+-UiOtherLow=128
+-JavaLow=128
+-WebLow=64
+-Default=1024
++UiMainLow=2048
++UiOtherLow=1024
++JavaLow=64
++WebLow=32
++Default=256
+ 
+--- .orig/etc/palm/lunaAnimations.conf
++++ /etc/palm/lunaAnimations.conf
+@@ -4,8 +4,8 @@
+ easeInOutStrength=3
+ 
+ [FPS]
+-normalFPS=35
+-slowFPS=20
++normalFPS=60
++slowFPS=60
+ 
+ [Cards]
+ cardLaunchDuration=400
diff --git a/etc/palm/luna.patch.~1~ b/etc/palm/luna.patch.~1~
new file mode 100644
index 0000000..e69de29
