diff --git a/media/cryptofs/apps/user/palm/applications/com.drewksparks.pandora-tablet/app/core.rej b/media/cryptofs/apps/user/palm/applications/com.drewksparks.pandora-tablet/app/core.rej
new file mode 100644
index 0000000..c48c746
--- /dev/null
+++ b/media/cryptofs/apps/user/palm/applications/com.drewksparks.pandora-tablet/app/core.rej
@@ -0,0 +1,9 @@
+--- media/cryptofs/apps/user/palm/applications/com.drewksparks.pandora-tablet/app/core
++++ media/cryptofs/apps/user/palm/applications/com.drewksparks.pandora-tablet/app/core
+@@ -37,5 +37,5 @@
+ 
+ });
+ 
+-SkipLimitService.STATION_SKIP_WINDOW = 60 * 60 * 1000;
++SkipLimitService.STATION_SKIP_WINDOW = 60 * 0 * 1000;
+ // 60 minutes
\ No newline at end of file
diff --git a/media/cryptofs/apps/user/palm/applications/com.drewksparks.pandora-tablet/app/core.~1~ b/media/cryptofs/apps/user/palm/applications/com.drewksparks.pandora-tablet/app/core.~1~
new file mode 100644
index 0000000..e69de29
