diff --git a/usr/palm/applications/com.palm.app.firstuse/app/controllers/app-assistant.js b/usr/palm/applications/com.palm.app.firstuse/app/controllers/app-assistant.js
index 8adacb2..16982ab 100644
--- a/usr/palm/applications/com.palm.app.firstuse/app/controllers/app-assistant.js
+++ b/usr/palm/applications/com.palm.app.firstuse/app/controllers/app-assistant.js
@@ -31,18 +31,8 @@ function setupSimulator() {
 }
 
 function setUpFirstUse() {
-Mojo.Log.info("---------------- setupFirstUse() --------------------");
-	this.firstUseDepot = new Mojo.Depot(stateEngine, dbSetUpSuccess, dbSetupFailure);	
-	isInitialized = true;
-	attemptCount = 0;
-	dbVersion = -1;
-	sessionStatus = 'none';
-	
-	FirstUseUtil.loadValues();
-	RoamUtil.checkNetworkStatus();
-	isConnected = false;
-	// Subscribe to connection manager
-	conStatusRequest = AppAssistant.accountService.isConManUp(readConManStatus, errorConManStatus);
+setupSimulator();
+isDevice = true;
 }
 
 function readConManStatus(response) {
diff --git a/usr/palm/applications/com.palm.app.firstuse/app/controllers/provision-assistant.js b/usr/palm/applications/com.palm.app.firstuse/app/controllers/provision-assistant.js
index 711ab01..317ad16 100644
--- a/usr/palm/applications/com.palm.app.firstuse/app/controllers/provision-assistant.js
+++ b/usr/palm/applications/com.palm.app.firstuse/app/controllers/provision-assistant.js
@@ -57,6 +57,11 @@ var ProvisionAssistant = Class.create({
 			
 				
 		this.setupProgressBar();
+		
+		this.provisionComplete();
+		return;
+		
+		
 		ProvisionUtil.setScene(this);
 				
 		if((!isActivated) && (activationFailed)) {
