diff --git a/etc/palm/luna.conf b/etc/palm/luna.conf
index d6a04ed..d8c972f 100644
--- a/etc/palm/luna.conf
+++ b/etc/palm/luna.conf
@@ -77,7 +77,7 @@ LauncherAtlasStatistics=false
 DumpLauncherAtlas=false
 
 [LaunchAtBoot]
-Applications=com.palm.app.phone;com.palm.app.email;com.palm.app.calendar;com.palm.app.messaging;com.palm.app.contacts;com.palm.app.camera
+Applications=com.palm.app.phone;com.palm.app.email;com.palm.app.calendar;com.palm.app.messaging;com.palm.app.contacts;com.palm.app.camera;com.palm.org.e-lnx.wee.apps.modeswitcher
 
 [KeepAlive]
 Applications=com.palm.app.phone
