diff --git a/usr/sbin/suspend_action b/usr/sbin/suspend_action
index bf2dad3..8473747 100755
--- a/usr/sbin/suspend_action
+++ b/usr/sbin/suspend_action
@@ -1,6 +1,6 @@
 #!/bin/sh
 
-echo -e "powerd: System going to sleep now\n" | wall 
+# echo -e "powerd: System going to sleep now\n" | wall 
 
 # Raise frequency to max so we'll resume at max
 CPUFREQDIR=/sys/devices/system/cpu/cpu0/cpufreq
@@ -13,4 +13,4 @@ echo mem > /sys/power/state
 
 echo $min >  $CPUFREQDIR/scaling_min_freq
 
-echo -e "powerd: System woke up.\n" | wall
\ No newline at end of file
+# echo -e "powerd: System woke up.\n" | wall
\ No newline at end of file
