diff --git a/usr/sbin/suspend_action b/usr/sbin/suspend_action
index e6f16aa..5def92e 100755
--- a/usr/sbin/suspend_action
+++ b/usr/sbin/suspend_action
@@ -1,9 +1,9 @@
 #!/bin/sh
 
-echo -e "powerd: System going to sleep now\n" | wall 
+# echo -e "powerd: System going to sleep now\n" | wall 
 
 # Initiate suspend
 echo mem > /sys/power/state
 
-echo -e "powerd: System woke up.\n" | wall 
+# echo -e "powerd: System woke up.\n" | wall 
 
