diff --git a/etc/palm/luna.conf b/etc/palm/luna.conf
index 9f9f7ad..5f6aebb 100644
--- a/etc/palm/luna.conf
+++ b/etc/palm/luna.conf
@@ -78,7 +78,7 @@ PiranhaDisplayFps=false
 ShowGestures=false
 
 [LaunchAtBoot]
-Applications=com.palm.app.phone;com.palm.app.email;com.palm.app.calendar;com.palm.app.messaging;com.palm.app.contacts;com.palm.app.camera
+Applications=com.palm.app.phone;com.palm.app.email;com.palm.app.calendar;com.palm.app.messaging;com.palm.app.contacts;com.palm.app.camera;com.palm.app.brightnessunlinked
 
 [KeepAlive]
 Applications=com.palm.app.phone
