diff --git a/etc/palm/luna.conf b/etc/palm/luna.conf
index f7a9964..36f8db3 100644
--- a/etc/palm/luna.conf
+++ b/etc/palm/luna.conf
@@ -13,11 +13,11 @@ LaunchPointsPath=/var/luna/launchpoints
 PreferencesPath=/var/luna/preferences
 DisplayWidth=320
 DisplayHeight=480
-DisplayNumBuffers=3
+DisplayNumBuffers=16
 MaxPenMoveFreq=30
 MinPenMoveFreq=10
-MoveMinX=16
-MoveMinY=12
+MoveMinX=1
+MoveMinY=1
 MaxPaintLoad=6
 MaxGestureChangeFreq=30
 MinGestureChangeFreq=20
@@ -47,15 +47,15 @@ CardLimit=-1
 AppsToAllowInLowMemory=com.palm.app.phone;com.palm.app.contacts;com.palm.app.messaging
 
 [TouchEvents]
-TapRadiusMax=25
-TapRadiusMin=5
+TapRadiusMax=10
+TapRadiusMin=2
 TapRadiusShrinkPerc=10
 TapRadiusShrinkGranMs=200
 DoubleClickDuration=300
 
 [VTrackBall]
-PixelsPerMoveH=14
-PixelsPerMoveV=26
+PixelsPerMoveH=1
+PixelsPerMoveV=1
 AccelRateH1=350
 AccelRateV1=200
 AccelConstH1=1
@@ -72,7 +72,7 @@ PiranhaDrawColoredOutlines=false
 PiranhaDisplayFps=false
 ShowGestures=false
 FailAllMigration=false
-PerformanceLogs=true
+PerformanceLogs=false
 LauncherAtlasStatistics=false
 DumpLauncherAtlas=false
 
@@ -112,11 +112,11 @@ GestureRepeaterIp=192.168.1.106
 GestureRepeaterPort=5001
 
 [CpuShare]
-UiMainLow=512
-UiOtherLow=128
-JavaLow=128
-WebLow=64
-Default=1024
+UiMainLow=2048
+UiOtherLow=512
+JavaLow=64
+WebLow=32
+Default=256
 
 [Launcher]
 LauncherLabelWidthAdjust=0
diff --git a/etc/palm/lunaAnimations.conf b/etc/palm/lunaAnimations.conf
index 792db0c..ff638f2 100644
--- a/etc/palm/lunaAnimations.conf
+++ b/etc/palm/lunaAnimations.conf
@@ -1,6 +1,6 @@
 [FPS]
-normalFPS=35
-slowFPS=20
+normalFPS=100
+slowFPS=80
 
 [Cards]
 cardLaunchDuration=400
