diff --git a/usr/palm/frameworks/enyo/0.10/framework/build/enyo-build.js b/usr/palm/frameworks/enyo/0.10/framework/build/enyo-build.js
index dcb8329..a9aec42 100644
--- a/usr/palm/frameworks/enyo/0.10/framework/build/enyo-build.js
+++ b/usr/palm/frameworks/enyo/0.10/framework/build/enyo-build.js
@@ -10815,10 +10815,13 @@ this.shouldDragSelect = b < this.slidePosition;
 if (b < this.dragMin || b > this.dragMax && !this.overSliding || b < this.dragMax && this.overSliding) return {
 select: this.getDragSelect()
 };
-if (this.overSliding && !this.dismissible) {
 var c = this.lastDragDx || 0;
+if (this.overSliding && !this.dismissible) {
 b = (a.dx - c) / 4 + this.slidePosition;
 }
+else {
+b = (a.dx - c) + this.slidePosition;
+}
 this.lastDragDx = a.dx;
 var d = Math.max(this.dragMin, Math.min(b, this.overSliding ? 1e9 : this.dragMax));
 this.pendingDragMove = this._drag(d);
