diff --git a/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js b/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
index 21f3714..fbe5a8b 100644
--- a/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
+++ b/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
@@ -3109,7 +3109,7 @@ showPowerOffAlert: function() {
 		    var f = function(stageController){
 		      stageController.pushScene('poweroffalert',this);
 		    }.bind(this);
-		    var params = {name: 'PowerOffAlert', height: 175,lightweight:true};		    	    
+		    var params = {name: 'PowerOffAlert', height: 290,lightweight:true};
 		    appController.createStageWithCallback(params,f,'popupalert');
         }
 	//Close the Storage Alert and create a USB dashboard.
