diff --git a/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js b/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
index 21f3714..c60e2b2 100644
--- a/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
+++ b/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
@@ -2263,7 +2263,7 @@ handlePowerNotifications: function(payload) {
 					this.controller.showBanner({
 						messageText: batteryalert,
 						icon: "/usr/lib/luna/system/luna-systemui/images/battery-"+i+'.png',
-						soundClass: "notifications"
+						soundFile: "/media/internal/.sounds/tenPercent.mp3"
 					},{}, 'batteryAlert');
 				}				
 				else if(this.batteryLevel > 10 && this.batteryLevel <= 20 && !this.batteryLevel20Shown) {
@@ -2272,7 +2272,7 @@ handlePowerNotifications: function(payload) {
 					this.controller.showBanner({
 						messageText: batteryalert,
 						icon: "/usr/lib/luna/system/luna-systemui/images/battery-"+i+'.png',
-						soundClass: "notifications"
+						soundFile: "/media/internal/.sounds/twentyPercent.mp3"
 					},{}, 'batteryAlert');	
 				}
 				
