diff --git a/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js b/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
index 21f3714..0b12003 100644
--- a/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
+++ b/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
@@ -40,7 +40,7 @@ var BarAssistant = Foundations.Class.create({
 		Mojo.Dom.get('vpn').hide();
 
 		
-		this.delayedOpenSearchEngineAvailable = Mojo.Function.debounce(undefined, this.handleOpenSearchEngineAvailable.bind(this), 10, this.controller.window);
+		//this.delayedOpenSearchEngineAvailable = Mojo.Function.debounce(undefined, this.handleOpenSearchEngineAvailable.bind(this), 10, this.controller.window);
 		this.showDeviceMenuFn = Mojo.Function.debounce(undefined, this.showDeviceMenu.bind(this), .10, this.controller.window);
 		
 		this.uiAlertsMethods["subscribeToBackupStatus"] = this.handleBackupStatusNotifications.bind(this);
@@ -49,7 +49,7 @@ var BarAssistant = Foundations.Class.create({
 		this.uiAlertsMethods["getDataTransferStatus"] = this.handleDataSyncNotifications.bind(this);
 		this.uiAlertsMethods["showTokenError"] = this.handleAccountServiceNotifications.bind(this);
 		this.uiAlertsMethods["VPNStatusNotification"] = this.handleVPNStatusNotification.bind(this);
-		this.uiAlertsMethods["openSearchEngineAvailable"] = this.delayedOpenSearchEngineAvailable;
+		//this.uiAlertsMethods["openSearchEngineAvailable"] = this.delayedOpenSearchEngineAvailable;
 		this.uiAlertsMethods["textAssistNotification"] = this.handleTextAssistNotification.bind(this);
 		this.uiAlertsMethods["RecordingDashboard"] = this.handleRecordNotification.bind(this);
 		this.uiAlertsMethods["dbError"] = this.handledbCriticalErrorNotifications.bind(this);
