diff --git a/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js b/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
index d4ae043..a388036 100644
--- a/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
+++ b/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
@@ -1995,12 +1995,54 @@ powerDeInit: function() {
 	this.powerSource['usb'] = false;
 	this.powerSource['inductive'] = false;
 	this.powerdServiceUp = false;
+	this.batteryStatusTimer.cancel();
+	this.batteryStatusTimer = undefined;
 	return;
 },
 
 powerInitialize: function() {
 	Mojo.Log.info("SystemUI - Power service online!");
 	this.powerdServiceUp = true;
+	if(!this.batteryStatusTimer) {
+		this.batteryStatusTimer = window.setTimeout(this.getBatteryDetails.bind(this), 300000);
+	}
+},
+
+getBatteryDetails: function() {
+	this.batteryStatusTimer = window.setTimeout(this.getBatteryDetails.bind(this), 300000);
+	this.controller.serviceRequest('palm://com.palm.bus/signal/', {
+		method: 'addmatch',
+		parameters: {
+			"category":"/com/palm/power",
+			"method":"batteryStatus"
+		},
+		onSuccess : this.handleBatteryStatus.bind(this)
+	});
+	this.controller.serviceRequest('palm://com.palm.power/com/palm/power/', {
+		method: 'batteryStatusQuery'
+	});
+},
+
+handleBatteryStatus: function(payload) {
+	if(payload.temperature_C) {
+		if(payload.temperature_C >= 50) {
+			var bannerText = "Critical! Device temperature too high";
+			var soundClassName = this.onActiveCall ? "none" : "notifications";			
+			this.controller.showBanner({
+				messageText: bannerText,
+				icon: "/usr/lib/luna/system/luna-systemui/images/notification-small-error.png",
+				soundClass: soundClassName
+			},""); 
+		} else if(payload.temperature_C >= 45) {
+			var bannerText = "Warning: Device temperature is high";
+			var soundClassName = this.onActiveCall ? "none" : "notifications";			
+			this.controller.showBanner({
+				messageText: bannerText,
+				icon: "/usr/lib/luna/system/luna-systemui/images/notification-small-error.png",
+				soundClass: soundClassName
+			},""); 
+		}
+	}
 },
 
 /*
