diff --git a/usr/lib/luna/system/luna-systemui/app/StoragedAlerts/StoragedAlerts.js b/usr/lib/luna/system/luna-systemui/app/StoragedAlerts/StoragedAlerts.js
index 0f7420e..3b8d4a0 100644
--- a/usr/lib/luna/system/luna-systemui/app/StoragedAlerts/StoragedAlerts.js
+++ b/usr/lib/luna/system/luna-systemui/app/StoragedAlerts/StoragedAlerts.js
@@ -8,6 +8,7 @@ enyo.kind({
 	name: "StorageAlert",
 	kind: "VFlexBox",
 	showMSMWarning: false,
+	style: "display:none;",
 	components: [
 		{
 			kind: enyo.Control,
@@ -57,6 +58,7 @@ enyo.kind({
 		this.inherited(arguments);
 		this.showMSMWarning = enyo.application.getSystemPreferences().showMSMWarning;
 		this.tapOnButton = false;
+		this.charge();
 	},
 	
 	enterMSM: function(inSender) {
