diff --git a/usr/lib/luna/system/luna-systemui/app/StoragedAlerts/StoragedAlerts.js b/usr/lib/luna/system/luna-systemui/app/StoragedAlerts/StoragedAlerts.js
index ffcae40..37845a8 100644
--- a/usr/lib/luna/system/luna-systemui/app/StoragedAlerts/StoragedAlerts.js
+++ b/usr/lib/luna/system/luna-systemui/app/StoragedAlerts/StoragedAlerts.js
@@ -41,7 +41,7 @@ enyo.kind({
 		                   {name:"infoIcon", className:"info-icon", onclick: "showInfo"}
 		                ]
 		 },
-		 {kind: "NotificationButton", className: "enyo-notification-button", layoutKind:"HFlexLayout", pack:"center", onclick:"charge", components:[{content: $L("Close")}]},
+		 {kind: "NotificationButton", className: "enyo-notification-button", layoutKind:"HFlexLayout", pack:"center", onclick:"charge", components:[{content: "Nur aufladen"}]},
 		 {
 		 	kind:enyo.PalmService, name:"enterMSMMode", service:"palm://com.palm.storage/diskmode/", method:"enterMSM"
 		 },
