diff --git a/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js b/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
index a4a150e..082d8bf 100644
--- a/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
+++ b/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
@@ -2428,7 +2428,7 @@ createUSBDashboard: function(showChargingBanner) {
 			icon: '/usr/lib/luna/system/luna-systemui/images/notification-small-usb.png',
 			messageText: $L("Charging battery"),			
 			}	
-			Mojo.Controller.getAppController().showBanner(bannerParams, {},'chargingAlert');
+			//Mojo.Controller.getAppController().showBanner(bannerParams, {},'chargingAlert');
 		}
 		
 		var stageController = Mojo.Controller.getAppController().getStageProxy("usb-dashboard");
