diff --git a/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js b/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
index 786f7ae..1c41342 100644
--- a/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
+++ b/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
@@ -1881,7 +1881,7 @@ handlePowerNotifications: function(payload) {
 				stageController.delegateToSceneAssistant("close");
 			}
 			
-			if(!this.chargingBannerShown && payload.name && (payload.name == "wall" || payload.name == "puck")) {
+			if(!this.chargingBannerShown && payload.name && (payload.name == "wall")) {
 				var batteryalert = this.$L("Charging Battery");
 				var soundClassName = this.onActiveCall ? "none" : "notifications";			
 				this.controller.showBanner({
