diff --git a/usr/palm/applications/com.palm.app.phone/app/models/TelephonyEventListener.js b/usr/palm/applications/com.palm.app.phone/app/models/TelephonyEventListener.js
index 3d17f47..fdced43 100644
--- a/usr/palm/applications/com.palm.app.phone/app/models/TelephonyEventListener.js
+++ b/usr/palm/applications/com.palm.app.phone/app/models/TelephonyEventListener.js
@@ -727,6 +727,32 @@ var TelephonyEventListener = Class.create({
 			this.announcer.announceIncoming(callState.affectedCallId, future.result, this.isConnected());
 			return true; // continue
 		});
+
+		//Patch to report call to BT-watch SE MBW-150
+		this.getTweaksPrefs = new Mojo.Service.Request("palm://org.webosinternals.tweaks.prefs/", {
+			method: 'get', parameters: {'owner': "bluetooth-mbw150", 
+			keys: ["mbwCall"]},
+			onSuccess: function(response) {
+				if(response) {
+					if(response.mbwCall == true) {
+ 						if(contact.name){
+							var MBWName = contact.name;
+						} else {
+							var MBWName = callState.number;
+                                                }
+						//Report Mail to ST-Watch MBW150
+						var request = new Mojo.Service.Request('palm://com.palm.applicationManager', {
+					        method: 'open',
+					        parameters: {
+					            id: "de.metaviewsoft.mwatch",
+					            params: {command: "RING", caller: MBWName} 
+				   	    	 },
+				     	   	onSuccess: function() {},
+				       	  	onFailure: function() {}
+				          	});
+					}
+				}
+			}.bind(this)});
 		
 		this.clearDialPending();
 	},
