diff --git a/usr/lib/luna/system/luna-systemui/app/PowerdAlerts/PowerdAlerts.js b/usr/lib/luna/system/luna-systemui/app/PowerdAlerts/PowerdAlerts.js
index 07a5f24..13120b3 100644
--- a/usr/lib/luna/system/luna-systemui/app/PowerdAlerts/PowerdAlerts.js
+++ b/usr/lib/luna/system/luna-systemui/app/PowerdAlerts/PowerdAlerts.js
@@ -62,9 +62,17 @@ enyo.kind({
 		                   {name:"infoIcon", className:"info-icon", onclick: "showInfo"}
 		                ]
 		 },
-		 {kind: "NotificationButton", className: "enyo-notification-button-negative", layoutKind:"HFlexLayout", pack:"center",  onclick:"poweroff", components:[{content: $L("Shut Down")}]},
-		 {kind: "NotificationButton", className: "enyo-notification-button", layoutKind:"HFlexLayout", pack:"center", onclick:"closeAlert", components:[{content: $L("Cancel")}]},
+		 {kind: "HFlexBox", components: [
+		 	{kind: "NotificationButton", className: "enyo-notification-button-alternate", flex: 1, onclick:"lunarestart", components:[{content: "Luna Neustart"}]},
+		 	{kind: "NotificationButton", className: "enyo-notification-button-alternate", flex: 1, onclick:"devicerestart", components:[{content: "Gerät Neustart"}]},
+		 ]},
+		 {kind: "HFlexBox", components: [
+		 	{kind: "NotificationButton", className: "enyo-notification-button-negative", flex: 1,  onclick:"poweroff", components:[{content: $L("Shut Down")}]},
+		 	{kind: "NotificationButton", className: "enyo-notification-button", flex: 1, onclick:"closeAlert", components:[{content: $L("Cancel")}]},
+		 ]},
 		 {kind:"PalmService", name:"launchHelp", service:"palm://com.palm.applicationManager/", method:"open"},
+		 {kind: "PalmService", name:"sysToolsMgr", service:"palm://ca.canucksoftware.systoolsmgr"},
+		 {kind: "PalmService", name:"reboot", service:"palm://com.palm.power/shutdown/", method:"machineReboot"},
 		 {kind: "PalmService", name:"shutdown", service:"palm://com.palm.power/shutdown/", method:"machineOff"}
 	],
 	
@@ -90,6 +98,18 @@ enyo.kind({
 		close();
 	},
 	
+	lunarestart: function(inSender) { 
+		this.$.sysToolsMgr.call({}, {method: "lunaRestart"});
+		close();
+	},
+	
+	devicerestart: function(inSender) { 
+		this.$.reboot.call({
+			reason: "Restart request by User"
+		});
+		close();
+	},
+	
 	poweroff: function(inSender) { 
 		this.$.shutdown.call({
 			reason: "Shutdown request by User"
