diff --git a/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js b/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
index 6fdbb39..61a8e7c 100644
--- a/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
+++ b/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
@@ -2174,7 +2174,7 @@ storageHandleNotifications: function(payload) {
 		this.isUSBConnected = true;
 	 	if (!this.onActiveCall) {
 			if (this.canUSBAlertDisplayed()) 
-				this.showStorageModeAlert();
+				this.createUSBDashboard(true);
 			else 
 				this.createUSBDashboard();
 		}
@@ -2308,7 +2308,7 @@ isUSBCableConnected: function() {
 					if (this.updateAvailable) 
 						this.createUSBDashboard();
 					else 
-						this.showStorageModeAlert();
+						this.createUSBDashboard(true);
 				}							
 			}
 		}.bind(this),	
