diff --git a/usr/lib/luna/system/luna-systemui/app/StoragedAlerts/StoragedAlerts.js b/usr/lib/luna/system/luna-systemui/app/StoragedAlerts/StoragedAlerts.js
index ffcae40..2ccaacd 100644
--- a/usr/lib/luna/system/luna-systemui/app/StoragedAlerts/StoragedAlerts.js
+++ b/usr/lib/luna/system/luna-systemui/app/StoragedAlerts/StoragedAlerts.js
@@ -34,7 +34,7 @@ enyo.kind({
 				}
 			]
 		 },
-		 {kind: "ApplicationEvents", onWindowDeactivated:"handleWindowDeActivated"},
+		 {kind: "ApplicationEvents", onWindowActivated:"activated", onWindowDeactivated:"handleWindowDeActivated"},
 		 {kind: "NotificationButton", className:"enyo-notification-button", layoutKind:"HFlexLayout", 
 			 components:[
 		                   {flex:1, content: $L("USB Drive"), onclick: "enterMSM"},
@@ -59,6 +59,10 @@ enyo.kind({
 		this.tapOnButton = false;
 	},
 	
+	activated: function() {
+		this.charge();
+	},
+	
 	enterMSM: function(inSender) {
 		if(enyo.application.isDeviceLocked()) {
 			this.$.unlock.call({state:"undock"});
