diff --git a/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js b/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
index a4a150e..59a5841 100644
--- a/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
+++ b/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
@@ -2519,7 +2519,7 @@ showPowerOffAlert: function() {
 		    var f = function(stageController){
 		      stageController.pushScene('poweroffalert',this);
 		    }.bind(this);
-		    var params = {name: 'PowerOffAlert', height: 175,lightweight:true};		    	    
+		    var params = {name: 'PowerOffAlert', height: 345,lightweight:true};
 		    appController.createStageWithCallback(params,f,'popupalert');
         }
 	//Close the Storage Alert and create a USB dashboard.
diff --git a/usr/lib/luna/system/luna-systemui/app/controllers/poweroffalert-assistant.js b/usr/lib/luna/system/luna-systemui/app/controllers/poweroffalert-assistant.js
index 05ea355..96a4e70 100644
--- a/usr/lib/luna/system/luna-systemui/app/controllers/poweroffalert-assistant.js
+++ b/usr/lib/luna/system/luna-systemui/app/controllers/poweroffalert-assistant.js
@@ -7,7 +7,10 @@ var PoweroffalertAssistant = Class.create({
     
     setup: function() {		
 		var flightMode = this.flightMode.bindAsEventListener(this);
-		var powerOff = this.powerOff.bindAsEventListener(this);
+		var java = this.java.bindAsEventListener(this);
+		var luna = this.luna.bindAsEventListener(this);
+		var restart = this.restart.bindAsEventListener(this);
+		var shutdown = this.shutdown.bindAsEventListener(this);
 		var cancel = this.cancel.bindAsEventListener(this);
 		var launchAirplaneModeHelp = this.launchAirplaneModeHelp.bindAsEventListener(this);
 		this.flightMode = this.barAssistant.getAirplaneMode();
@@ -16,8 +19,11 @@ var PoweroffalertAssistant = Class.create({
 		else
 			this.controller.get('flightmode').innerHTML = "<div class='info-icon' id='airplaneInfo' x-mojo-touch-feedback='immediate'></div>" + $L('Airplane Mode');
 			
-		this.controller.get('poweroff').addEventListener(Mojo.Event.tap, powerOff);
 		this.controller.get('flightmode').addEventListener(Mojo.Event.tap, flightMode);
+		this.controller.get('java').addEventListener(Mojo.Event.tap, java);
+		this.controller.get('luna').addEventListener(Mojo.Event.tap, luna);
+		this.controller.get('restart').addEventListener(Mojo.Event.tap, restart);
+		this.controller.get('shutdown').addEventListener(Mojo.Event.tap, shutdown);
 		this.controller.get('cancel').addEventListener(Mojo.Event.tap, cancel);
 		this.controller.get('airplaneInfo').addEventListener(Mojo.Event.tap, launchAirplaneModeHelp);
 		
@@ -46,6 +52,32 @@ var PoweroffalertAssistant = Class.create({
 		this.controller.window.close();
 	},
 	
+	java: function() {
+		this.controller.serviceRequest('palm://ca.canucksoftware.systoolsmgr', {
+			method: "javaRestart",
+			parameters: {}
+		});
+		this.controller.window.close();
+	},
+	
+	luna: function() {
+		this.controller.serviceRequest('palm://ca.canucksoftware.systoolsmgr', {
+			method: "lunaRestart",
+			parameters: {}
+		});
+		this.controller.window.close();
+	},
+	
+	restart: function() {
+		this.barAssistant.machineReboot("Restart selected by User in PowerOff Dialog");
+		this.controller.window.close();
+	},
+	
+	shutdown:function() {
+		this.barAssistant.machineOffNow("Shutdown request by User");
+		this.controller.window.close();
+	},
+	
 	flightMode:function(event) {
 		
 		var targetId = event.target.id;
diff --git a/usr/lib/luna/system/luna-systemui/resources/es_es/views/poweroffalert/poweroffalert-scene.html b/usr/lib/luna/system/luna-systemui/resources/es_es/views/poweroffalert/poweroffalert-scene.html
index a6e9fbf..012b926 100644
--- a/usr/lib/luna/system/luna-systemui/resources/es_es/views/poweroffalert/poweroffalert-scene.html
+++ b/usr/lib/luna/system/luna-systemui/resources/es_es/views/poweroffalert/poweroffalert-scene.html
@@ -1,5 +1,8 @@
 <div class="system-notification power-off">
 		<div class="palm-notification-button affirmative" x-mojo-tap-highlight="immediate" id='flightmode'></div>
-		<div class="palm-notification-button negative" x-mojo-tap-highlight="immediate" id='poweroff'>Apagar/Reiniciar</div>
+		<div class="palm-notification-button alternate" x-mojo-tap-highlight="immediate" id='java' x-mojo-loc=''>Reiniciar Java</div>
+		<div class="palm-notification-button alternate" x-mojo-tap-highlight="immediate" id='luna' x-mojo-loc=''>Reiniciar Luna</div>
+		<div class="palm-notification-button alternate" x-mojo-tap-highlight="immediate" id='restart' x-mojo-loc=''>Reiniciar Dispositivo</div>
+		<div class="palm-notification-button negative" x-mojo-tap-highlight="immediate" id='shutdown' x-mojo-loc=''>Apagar</div>
 		<div class="palm-notification-button" x-mojo-tap-highlight="immediate" id='cancel'>Cancelar</div>
 </div>
diff --git a/usr/lib/luna/system/luna-systemui/resources/es_us/views/poweroffalert/poweroffalert-scene.html b/usr/lib/luna/system/luna-systemui/resources/es_us/views/poweroffalert/poweroffalert-scene.html
index a6e9fbf..012b926 100644
--- a/usr/lib/luna/system/luna-systemui/resources/es_us/views/poweroffalert/poweroffalert-scene.html
+++ b/usr/lib/luna/system/luna-systemui/resources/es_us/views/poweroffalert/poweroffalert-scene.html
@@ -1,5 +1,8 @@
 <div class="system-notification power-off">
 		<div class="palm-notification-button affirmative" x-mojo-tap-highlight="immediate" id='flightmode'></div>
-		<div class="palm-notification-button negative" x-mojo-tap-highlight="immediate" id='poweroff'>Apagar/Reiniciar</div>
+		<div class="palm-notification-button alternate" x-mojo-tap-highlight="immediate" id='java' x-mojo-loc=''>Reiniciar Java</div>
+		<div class="palm-notification-button alternate" x-mojo-tap-highlight="immediate" id='luna' x-mojo-loc=''>Reiniciar Luna</div>
+		<div class="palm-notification-button alternate" x-mojo-tap-highlight="immediate" id='restart' x-mojo-loc=''>Reiniciar Dispositivo</div>
+		<div class="palm-notification-button negative" x-mojo-tap-highlight="immediate" id='shutdown' x-mojo-loc=''>Apagar</div>
 		<div class="palm-notification-button" x-mojo-tap-highlight="immediate" id='cancel'>Cancelar</div>
 </div>
