diff --git a/usr/palm/applications/com.palm.app.messaging/app/controllers/notification-assistant.js b/usr/palm/applications/com.palm.app.messaging/app/controllers/notification-assistant.js
index ffab634..3411cf2 100644
--- a/usr/palm/applications/com.palm.app.messaging/app/controllers/notification-assistant.js
+++ b/usr/palm/applications/com.palm.app.messaging/app/controllers/notification-assistant.js
@@ -402,9 +402,10 @@ NotificationAssistant.prototype.sendNewMessageNotificationAndUpdateDashboard = f
 	if (message === "") {
 		notificationText = name;
 	}else{
+		
 		notificationText = $L("#{name}: #{message}").interpolate({
 			name: name,
-			message: message
+			message: '[New Message]'
 		});
 	}
 	
