diff --git a/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js b/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
index 21f3714..10307a5 100644
--- a/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
+++ b/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
@@ -3018,7 +3018,7 @@ createUSBDashboard: function(showChargingBanner) {
 			icon: '/usr/lib/luna/system/luna-systemui/images/notification-small-usb.png',
 			messageText: $L("Charging battery"),			
 			}	
-			Mojo.Controller.getAppController().showBanner(bannerParams, {},'chargingAlert');
+			//Mojo.Controller.getAppController().showBanner(bannerParams, {},'chargingAlert');
 		}
 		
 		var stageController = Mojo.Controller.getAppController().getStageController("usb-dashboard");
