diff --git a/usr/palm/applications/com.palm.app.network/app/assistants/app-assistant.js b/usr/palm/applications/com.palm.app.network/app/assistants/app-assistant.js
index 1560d07..ec426c9 100644
--- a/usr/palm/applications/com.palm.app.network/app/assistants/app-assistant.js
+++ b/usr/palm/applications/com.palm.app.network/app/assistants/app-assistant.js
@@ -162,6 +162,11 @@ AppAssistant.prototype.showDashboard = function(params) {
 	    var stageController = appController.getStageController("captiveportal-dashboard");
 	    if (undefined===stageController) {
 	          Mojo.Log.info("Bringing up the dashboard");
+	
+			 /* DR NELD Tue Mar 15th 2011
+			 This comments out the constant notifications forced upon us when a Captive Portal
+			 is detected.
+			
 	          var dashboardBannerText = $L("Captive Portal Detected");
 	          appController.showBanner(
 	          {
@@ -175,6 +180,7 @@ AppAssistant.prototype.showDashboard = function(params) {
 	          appController.createStageWithCallback({name: "captiveportal-dashboard", lightweight: true}, 
 	                  this.notificationCreatedHandler, "dashboard");
 	          appController.playSoundNotification("notification");
+	*/
 	    } else {
 	        Mojo.Log.info("Updating the dashboard");
 	         stageController.delegateToSceneAssistant("updateDashboard", paramsToSendDashboard, new Date());
