Index: /usr/palm/applications/com.palm.app.amazonstore/app/models/download-manager.js
===================================================================
--- .orig/usr/palm/applications/com.palm.app.amazonstore/app/models/download-manager.js
+++ /usr/palm/applications/com.palm.app.amazonstore/app/models/download-manager.js
@@ -437,9 +437,9 @@ var DownloadManager = Class.create({	
 						}).bind(this));
 				}
 				
-				if (userInitiated && AppAssistant.connectionResponse.wifi.state == "disconnected") {
-					this.showWifiRequiredDialog();
-				}
+				//if (userInitiated && AppAssistant.connectionResponse.wifi.state == "disconnected") {
+					//this.showWifiRequiredDialog();
+				//}
 			}
 			else if (userInitiated) {
 				Mojo.Log.info("Amazon App: DOWNLOAD MANGER purchaseCB: purchase FAILED");
@@ -462,8 +462,8 @@ var DownloadManager = Class.create({	
 		// download only if wifi is on		
 		Mojo.Log.info("Amazon App: DOWNLOAD MANGER download: wifi: %s, userInitiated: %s", AppAssistant.connectionResponse.wifi.state, userInitiated);
 		
-		if (AppAssistant.connectionResponse.wifi.state == "connected") {
-			if ( (this.IpInterfaceSet != "" && this.IpInterfaceSet == AppAssistant.connectionResponse.wifi.ipAddress) 
+		if (true || AppAssistant.connectionResponse.wifi.state == "connected") {
+			if (true || (this.IpInterfaceSet != "" && this.IpInterfaceSet == AppAssistant.connectionResponse.wifi.ipAddress) 
 				|| PalmSystem.version.indexOf("desktop") >= 0) 
 			{
 				Mojo.Log.info("Amazon App: DOWNLOAD MANGER download: track: %s, serviceRequestCounter: %d", track.title, this.serviceRequestCounter);
