diff --git a/usr/palm/applications/com.palm.app.photos/css/AlbumMode.css b/usr/palm/applications/com.palm.app.photos/css/AlbumMode.css
index bd642ed..02ab125 100644
--- a/usr/palm/applications/com.palm.app.photos/css/AlbumMode.css
+++ b/usr/palm/applications/com.palm.app.photos/css/AlbumMode.css
@@ -55,13 +55,18 @@
 	-webkit-border-image: url(../images/photo-9tile.png) 6 7 7 6 repeat repeat;
 	border-width: 6px 7px 7px 6px;
 	margin: auto;
-	position: absolute; 
+	position: static; 
 	top: 0; 
 	bottom: 0; 
 	left: 0; 
 	right: 0;
 }
 
+.AlbumGridFileName {
+        position: static;
+        text-align: center;
+        font-size: 14px;
+}
 /*
 DEFINED IN NonCssParams.js
 .AlbumGridThumb img {
diff --git a/usr/palm/applications/com.palm.app.photos/source/AlbumGridView.js b/usr/palm/applications/com.palm.app.photos/source/AlbumGridView.js
index ad52729..9f8d2bc 100644
--- a/usr/palm/applications/com.palm.app.photos/source/AlbumGridView.js
+++ b/usr/palm/applications/com.palm.app.photos/source/AlbumGridView.js
@@ -22,6 +22,10 @@ enyo.kind({
 			ondragfinish: "doCellDragFinish",
 			onerror: "handleError"
 		},
+                { kind: 'Control', 
+                        name: 'filename', className: 'AlbumGridFileName', 
+                        content: ''
+                },
 		{ kind: 'VideoSymbolBar',
 			onBarClick: "doCellClick",  // just use the event that Simon made
 			onmousehold: "doCellMouseHold",
@@ -60,6 +64,7 @@ enyo.kind({
 	published: {
 		debugging: true,
 		selecting: false,
+		showFilenames: false,
 	},
 	photoCount: 0,
 	videoCount: 0,
@@ -271,9 +276,12 @@ enyo.kind({
 
 		var img = inFlyweight.$.image;
 		img.setSrc(path);
-//		var fname = inRecord.path.split("/");
-//		fname = fname[fname.length-1];
+                if( this.showFilenames ) {
+		  var fname = inRecord.path.split("/");
+		  fname = fname[fname.length-1];
+                }
 //		console.log("- @ - @ -  setup cell (" + inColumn + "," + inRow + "):   " + fname);
+                inFlyweight.$.filename.setContent(fname);
 
 		// If this cell contains a video-thumbnail, we need to show some extra controls.
 		var vidBar = inFlyweight.$.videoSymbolBar;
diff --git a/usr/palm/applications/com.palm.app.photos/source/modes/AlbumMode.js b/usr/palm/applications/com.palm.app.photos/source/modes/AlbumMode.js
index 311d112..8afca8c 100644
--- a/usr/palm/applications/com.palm.app.photos/source/modes/AlbumMode.js
+++ b/usr/palm/applications/com.palm.app.photos/source/modes/AlbumMode.js
@@ -11,6 +11,7 @@ enyo.kind({
 	},
 	showingMultiselect: false,
 	components: [
+		{kind:"PalmService", name:"getTweaks", service:"palm://org.webosinternals.tweaks.prefs/", method:"get", onSuccess:"tweakSuccess", onFailure:"tweakFailure"},
 		{ kind: 'PanelHeader', label: $L('Album'), controlAreaStyle: 'gridview-header-controls' },	
 		{ kind: 'VFlexBox', name: 'gridSpace', style: 'position: relative;', flex: 1, components: [
 			{ name: 'emptyAlbum', kind: 'VFlexBox', style: 'position: absolute; top: 0; left: 0; width: 100%; height: 100%; z-index: 1; text-align: center', components: [
@@ -110,17 +111,44 @@ enyo.kind({
 			this.handleAlbumPictureCountChanged(null, newAlbum.photoCount, newAlbum.videoCount);
 			this.handleAlbumDateChanged(null, newAlbum.modifiedTime);
 					
+			// make the call to the Tweaks service
+			this.$.getTweaks.call({owner: "org.webosinternals.patches.photos-show-filenames", keys: ["showFn"]});
+			/*
 			this.grid = this.createGrid();
 			
 			// The render ensures that a DOM-node is created so that resize() knows
 			// its bounds so that it can eg: compute the number of columns.
 			this.grid.render();
 			this.grid.resize();
+			*/
 		}
 		else {
 			this.$.panelHeader.setIconStyle('library-navigation-icon-40x40-local');
 		}
 	},
+	tweakSuccess: function(inSender, inResponse) {
+		this.grid = this.createGrid();
+		if((inResponse) && (inResponse.showFn)) {
+			// Do something with the config value
+			this.grid.setShowFilenames(inResponse.showFn);
+		} else {
+			// Deal with failure
+		}
+
+		// The render ensures that a DOM-node is created so that resize() knows
+		// its bounds so that it can eg: compute the number of columns.
+		this.grid.render();
+		this.grid.resize();
+	},
+	tweakFailure: function() {
+		this.grid = this.createGrid();
+		this.grid.setShowFilenames(true);
+
+		// The render ensures that a DOM-node is created so that resize() knows
+		// its bounds so that it can eg: compute the number of columns.
+		this.grid.render();
+		this.grid.resize();
+	},
 	createGrid: function() {
 		var gridViewKind = 'AlbumGridView';
 		
