diff --git a/usr/palm/services/com.palm.service.photos/photos-src/commands/AlbumManageAssistant.js b/usr/palm/services/com.palm.service.photos/photos-src/commands/AlbumManageAssistant.js
index bf87a95..b15fa0b 100644
--- a/usr/palm/services/com.palm.service.photos/photos-src/commands/AlbumManageAssistant.js
+++ b/usr/palm/services/com.palm.service.photos/photos-src/commands/AlbumManageAssistant.js
@@ -752,6 +752,36 @@ function copyOnlyImage(photo,album)
 	return copyPromise;
 }
 
+function moveOnlyImage(photo,album)
+{
+	var movePromise =Promise();
+	var path =(photo.appOriginal)?photo.appOriginal:photo.path;
+	var file=getFileName(path);
+	var dstFile = album.path+"/"+file;
+	
+	//first check if this file already exists, if it does then
+	//possibly we are making a second copy of the same file into this
+	//album. so make another copy of it.
+	node_path.exists(dstFile, function(exists) {
+		
+		if(exists){
+			console.info("*** File already exists, probably making another copy");
+			dstFile=album.path+"/"+(new Date().getTime())+"-"+file;
+		}
+		var move =moveFile(path,dstFile);
+		move.whenFulfilled(function(){
+			movePromise.fulfill(true);
+		});
+		move.whenBroken(function(errMsg){
+			movePromise.breakk(errMsg);
+		});
+		
+	});
+		
+	
+	return movePromise;
+}
+
 //helper functions for photos add assistant
 function preProcessCopy(photo,album)
 {
@@ -768,7 +798,7 @@ function preProcessCopy(photo,album)
 		//we over ride fileindexer and copy the cached images if available ourselves
 		//var copyCache =copyCachedImagesAndUpdateDb(photo,album);
 		//we are not able to override fileindexer as fileindexer as fileindexer is overriding anyway
-		var copyCache =copyOnlyImage(photo,album);
+		var copyCache =moveOnlyImage(photo,album);
 		promise1.fulfill(copyCache);
 		copyCache.whenFulfilled(function(){
 			console.info("*********Successfully copied "+photo.path+" to album:"+album.name);
