diff --git a/usr/palm/applications/com.palm.app.photos/app/controllers/thumbs-assistant.js b/usr/palm/applications/com.palm.app.photos/app/controllers/thumbs-assistant.js
index 8e9edc3..e418e14 100644
--- a/usr/palm/applications/com.palm.app.photos/app/controllers/thumbs-assistant.js
+++ b/usr/palm/applications/com.palm.app.photos/app/controllers/thumbs-assistant.js
@@ -5,8 +5,8 @@ Photos.Scene.Thumbs = Class.create(Photos.Scene.Scroll, Photos.Common.Opacity, {
 		$super(passedArg);
 		this.albumCount = 0;
 		// optimized for 99 photos.
-		this.firstScreenThumbCount = 15;
-		this.listRenderWindow = 30;
+		this.firstScreenThumbCount = 24;
+		this.listRenderWindow = 999;
 		this.deferActivate = passedArg.deferActivate;
 		this.rowsPerScreen = 4;
 		this.screensRendered =
