diff --git a/usr/palm/frameworks/mojo/builtins/palmInitFramework347.js b/usr/palm/frameworks/mojo/builtins/palmInitFramework347.js
index aed4b02..27df88b 100644
--- a/usr/palm/frameworks/mojo/builtins/palmInitFramework347.js
+++ b/usr/palm/frameworks/mojo/builtins/palmInitFramework347.js
@@ -18273,10 +18273,10 @@ Mojo.Widget.ImageView=Class.create({
 
 defaultExtractFSParams:"800:800:3",
 defaultLowResExtractFSParams:"160:160:3",
-defaultGutterWidth:25,
+defaultGutterWidth:16,
 defaultZoomThreshold:25,
 dragSnapThreshold:0.4,
-highResolutionLoadTimeout:1.2,
+highResolutionLoadTimeout:0.1,
 extractFSPath:"/var/luna/data/extractfs",
 corruptImage:Mojo.Config.IMAGES_HOME+'/corrupt-image.png',
 
@@ -18292,8 +18292,8 @@ Mojo.assert(this.controller.model,
 this.controller.widgetName+"?");
 
 
-this.zoomTargetTime=0.5;
-this.flickTargetTime=0.3;
+this.zoomTargetTime=0.25;
+this.flickTargetTime=0.2;
 this.sameTargetTime=0.5;
 this.dragTargetTime=0.4;
 this.flickScale=0.075;
@@ -18731,7 +18731,7 @@ if(this._userBusy())
 {
 this.highResolutionTimer=
 this._applyHighResExtractFSParams.bind(this)
-.delay(0.5,image,url);
+.delay(0.1,image,url);
 return;
 }
 
@@ -18752,7 +18752,7 @@ if(url!==this.originalLeftUrl)
 {
 this.leftImageExists=false;
 this.originalLeftUrl=url;
-this.imageLeft.src=this._getLowResUrl(thumbUrl||url);
+this.imageLeft.src=this._getMediumResUrl(thumbUrl||url);
 }
 },
 
@@ -18763,7 +18763,7 @@ if(url!==this.originalCenterUrl)
 {
 this.centerImageExists=false;
 this.originalCenterUrl=url;
-this.imageCenter.src=this._getLowResUrl(thumbUrl||url);
+this.imageCenter.src=this._getMediumResUrl(thumbUrl||url);
 }
 },
 
@@ -18774,7 +18774,7 @@ if(url!==this.originalRightUrl)
 {
 this.rightImageExists=false;
 this.originalRightUrl=url;
-this.imageRight.src=this._getLowResUrl(thumbUrl||url);
+this.imageRight.src=this._getMediumResUrl(thumbUrl||url);
 }
 },
 
