diff --git a/usr/palm/applications/com.palm.app.photos/css/AlbumMode.css b/usr/palm/applications/com.palm.app.photos/css/AlbumMode.css
index bd642ed..02ab125 100644
--- a/usr/palm/applications/com.palm.app.photos/css/AlbumMode.css
+++ b/usr/palm/applications/com.palm.app.photos/css/AlbumMode.css
@@ -55,13 +55,18 @@
 	-webkit-border-image: url(../images/photo-9tile.png) 6 7 7 6 repeat repeat;
 	border-width: 6px 7px 7px 6px;
 	margin: auto;
-	position: absolute; 
+	position: static; 
 	top: 0; 
 	bottom: 0; 
 	left: 0; 
 	right: 0;
 }
 
+.AlbumGridFileName {
+        position: static;
+        text-align: center;
+        font-size: 14px;
+}
 /*
 DEFINED IN NonCssParams.js
 .AlbumGridThumb img {
diff --git a/usr/palm/applications/com.palm.app.photos/source/AlbumGridView.js b/usr/palm/applications/com.palm.app.photos/source/AlbumGridView.js
index ad52729..5843a64 100644
--- a/usr/palm/applications/com.palm.app.photos/source/AlbumGridView.js
+++ b/usr/palm/applications/com.palm.app.photos/source/AlbumGridView.js
@@ -22,6 +22,10 @@ enyo.kind({
 			ondragfinish: "doCellDragFinish",
 			onerror: "handleError"
 		},
+                { kind: 'Control', 
+                        name: 'filename', className: 'AlbumGridFileName', 
+                        content: ''
+                },
 		{ kind: 'VideoSymbolBar',
 			onBarClick: "doCellClick",  // just use the event that Simon made
 			onmousehold: "doCellMouseHold",
@@ -271,9 +275,10 @@ enyo.kind({
 
 		var img = inFlyweight.$.image;
 		img.setSrc(path);
-//		var fname = inRecord.path.split("/");
-//		fname = fname[fname.length-1];
+		var fname = inRecord.path.split("/");
+		fname = fname[fname.length-1];
 //		console.log("- @ - @ -  setup cell (" + inColumn + "," + inRow + "):   " + fname);
+                inFlyweight.$.filename.setContent(fname);
 
 		// If this cell contains a video-thumbnail, we need to show some extra controls.
 		var vidBar = inFlyweight.$.videoSymbolBar;
