diff --git a/usr/palm/frameworks/metascene.videos/submission/105.7/javascript/assistants/details-assistant.js b/usr/palm/frameworks/metascene.videos/submission/105.7/javascript/assistants/details-assistant.js
index adb9dd4..805f129 100644
--- a/usr/palm/frameworks/metascene.videos/submission/105.7/javascript/assistants/details-assistant.js
+++ b/usr/palm/frameworks/metascene.videos/submission/105.7/javascript/assistants/details-assistant.js
@@ -72,14 +72,14 @@ DetailsAssistant.prototype.setupTextFields = function(){
 		textFieldAttrs,
 		this.titleModel = {
 			value: this.video.title || "",
-			disabled: !this.video.capturedOnDevice
+			disabled: false
 	});
 	
 	this.controller.setupWidget("desc_preview",
 	   textFieldAttrs,
 	   this.descModel = {
 		   value: this.video.description || "",
-		   disabled: !this.video.capturedOnDevice
+		   disabled: false
 	});
 
 	//slight workaround, since it seems that these won't render correctly as regular textfield labels...
