diff --git a/usr/palm/frameworks/private/metascene.videos/submission/75/javascript/assistants/details-assistant.js b/usr/palm/frameworks/private/metascene.videos/submission/75/javascript/assistants/details-assistant.js
index a197d1a..75eccd2 100644
--- a/usr/palm/frameworks/private/metascene.videos/submission/75/javascript/assistants/details-assistant.js
+++ b/usr/palm/frameworks/private/metascene.videos/submission/75/javascript/assistants/details-assistant.js
@@ -41,7 +41,7 @@ DetailsAssistant.prototype.setup = function(){
        },
        this.titleModel = {
            value: this.videoTitle? this.videoTitle : "",
-           disabled: !this.captured
+           disabled: false
 	});
 	var titleLabel = new Element('div', {'class':'label'}).update($LF("Title"));
 	this.controller.get("title_preview").insert({ before:titleLabel });
@@ -54,7 +54,7 @@ DetailsAssistant.prototype.setup = function(){
        },
        this.descModel = {
            value: this.videoDescription ? this.videoDescription : "",
-           disabled: !this.captured
+           disabled: false
 	});
 	var descriptionLabel = new Element('div', {'class':'label'}).update($LF("Description"));
 	this.controller.get("desc_preview").insert({ before:descriptionLabel });
