diff --git a/usr/palm/frameworks/metascene.videos/submission/105.7/concatenated.js b/usr/palm/frameworks/metascene.videos/submission/105.7/concatenated.js
index 18f2c50..a767969 100755
--- a/usr/palm/frameworks/metascene.videos/submission/105.7/concatenated.js
+++ b/usr/palm/frameworks/metascene.videos/submission/105.7/concatenated.js
@@ -1364,13 +1364,13 @@ var NowplayingAssistant = Class.create(metascene.Base, {
 	
 	_onBlurHandle: function(){
 		Mojo.Log.info ("blur handler");
-		this.playEngine.setBlockPlayEvents(true);
-		
 		this._blockDim(false);
 
 		if (this.mediaController.isStaticStream){
-			this.pause();		
+			//this.pause();		
 		} else {
+			this.playEngine.setBlockPlayEvents(true);
+		
 			this.mediaController.stop();
 		}
 		
@@ -1461,9 +1461,9 @@ var NowplayingAssistant = Class.create(metascene.Base, {
 	_powerServiceCallback: function(event){
 		if (!this.playEngine){ return; }
 		if (event.event === "displayOff"){
-			this.playEngine.setBlockPlayEvents(true);
+			//this.playEngine.setBlockPlayEvents(true);
 			Mojo.Log.info ("display is off, sending pause");
-			this.pause();
+			//this.pause();
 		} else if (event.event === "displayOn" && this.controller.stageController.active){
 			//this tells the videoplayer to continue blocking play events if autoplay is set to false
 			this.playEngine.setBlockPlayEvents(!this.autoplay);
diff --git a/usr/palm/frameworks/metascene.videos/submission/105.7/javascript/assistants/nowplaying-assistant.js b/usr/palm/frameworks/metascene.videos/submission/105.7/javascript/assistants/nowplaying-assistant.js
index afd4524..fb44296 100644
--- a/usr/palm/frameworks/metascene.videos/submission/105.7/javascript/assistants/nowplaying-assistant.js
+++ b/usr/palm/frameworks/metascene.videos/submission/105.7/javascript/assistants/nowplaying-assistant.js
@@ -277,13 +277,13 @@ var NowplayingAssistant = Class.create(metascene.Base, {
 	
 	_onBlurHandle: function(){
 		Mojo.Log.info ("blur handler");
-		this.playEngine.setBlockPlayEvents(true);
-		
 		this._blockDim(false);
 
 		if (this.mediaController.isStaticStream){
-			this.pause();		
+			//this.pause();		
 		} else {
+			this.playEngine.setBlockPlayEvents(true);
+		
 			this.mediaController.stop();
 		}
 		
@@ -374,9 +374,9 @@ var NowplayingAssistant = Class.create(metascene.Base, {
 	_powerServiceCallback: function(event){
 		if (!this.playEngine){ return; }
 		if (event.event === "displayOff"){
-			this.playEngine.setBlockPlayEvents(true);
+			//this.playEngine.setBlockPlayEvents(true);
 			Mojo.Log.info ("display is off, sending pause");
-			this.pause();
+			//this.pause();
 		} else if (event.event === "displayOn" && this.controller.stageController.active){
 			//this tells the videoplayer to continue blocking play events if autoplay is set to false
 			this.playEngine.setBlockPlayEvents(!this.autoplay);
diff --git a/usr/palm/frameworks/metascene.videos/submission/105.7metascene_videos.js b/usr/palm/frameworks/metascene.videos/submission/105.7metascene_videos.js
index ddeb588..cc5fe94 100644
--- a/usr/palm/frameworks/metascene.videos/submission/105.7metascene_videos.js
+++ b/usr/palm/frameworks/metascene.videos/submission/105.7metascene_videos.js
@@ -54,10 +54,10 @@ Mojo.Event.flick,this.onVideoFlick.bind(this),false)}else this.autoListen(this.v
 "mouseup",this.onControllerDragEnd.bind(this),false);this.autoListen(videoControls,Mojo.Event.tap,this.onControllerTap.bind(this),false);this.autoListen(this.videoPlayer,"loadstart",this.onLoadStart.bind(this),false);this.autoListen(this.videoPlayer,"ended",this._ended.bind(this),false);this.autoListen(this.videoPlayer,"progress",this._progress.bind(this),false)},onVideoFlick:function(event){event.stop();if(!this.mediaController.isStaticStream||!this.enableControls)return;if(this.controller.getMenuVisible(Mojo.Menu.commandMenu)){Mojo.Log.info("controls are up");
 this.showControls();this._doVideoFlick(event)}else{Mojo.Log.info("doing delayed show");this.showControls();this._doVideoFlick.bind(this).delay(this.FLICK_DELAY_SECS,event)}},_doVideoFlick:function(event){if(event.velocity.x>0){if(this.videoPlayer.currentTime+this.FLICK_FORWARD_SKIP_AMOUNT<this.videoPlayer.duration)this.videoPlayer.currentTime+=this.FLICK_FORWARD_SKIP_AMOUNT}else if(this.videoPlayer.currentTime-this.FLICK_BACK_SKIP_AMOUNT>0)this.videoPlayer.currentTime-=this.FLICK_BACK_SKIP_AMOUNT;
 else this.videoPlayer.currentTime=0},_overridingMenuSetVisible:function(which,visible){var controlsDiv=this.controller.get("video-controls");if(visible){controlsDiv.removeClassName("faded");this.controller.sceneElement.removeClassName("faded")}else{controlsDiv.addClassName("faded");this.controller.sceneElement.addClassName("faded")}this._originalSetMenuVisible(which,visible);this.updateTime(this.videoPlayer.currentTime,true)},_onBlurHandle:function(){Mojo.Log.info("blur handler");this.playEngine.setBlockPlayEvents(true);
-this._blockDim(false);if(this.mediaController.isStaticStream)this.pause();else this.mediaController.stop();this.showControls();Util.markForeground(this.controller,false,Util.powerService)},_clearUnloadTimeout:function(){if(this.unloadTimeoutId!==0){this.controller.window.clearTimeout(this.unloadTimeoutId);this.unloadTimeoutId=0}},_onFocusHandle:function(){this._clearUnloadTimeout();this.playEngine.setBlockPlayEvents(!this.autoplay);Util.markForeground(this.controller,true,Util.powerService);var currentState=
+this._blockDim(false);if(!this.mediaController.isStaticStream)this.playEngine.setBlockPlayEvents(true);this.mediaController.stop();this.showControls();Util.markForeground(this.controller,false,Util.powerService)},_clearUnloadTimeout:function(){if(this.unloadTimeoutId!==0){this.controller.window.clearTimeout(this.unloadTimeoutId);this.unloadTimeoutId=0}},_onFocusHandle:function(){this._clearUnloadTimeout();this.playEngine.setBlockPlayEvents(!this.autoplay);Util.markForeground(this.controller,true,Util.powerService);var currentState=
 this.playEngine.getCurrentState();if(this.shouldPreventDim(currentState))this._blockDim(true);if(currentState==this.playEngine.STATE_DISCONNECTED){Mojo.Log.info("was disconnected, reload the file");this.playEngine.load(this.getCurrentPos())}},getCurrentPos:function(){var pos=this.sliderModel.value;var secs=Math.round(pos/100*this.videoPlayer.duration);Mojo.Log.info("video player getCurrentPos returning "+secs);return secs},progressBarDragEnd:function(){this.sliderIsDragging=false;var pos=this.sliderModel.value;
 var secs=Math.round(pos/100*this.videoPlayer.duration);this.videoPlayer.currentTime=secs;if(this.pausedFromDrag){this.videoPlayer.play();this.pausedFromDrag=false}},progressBarDragStart:function(){if(this.sliderIsDragging)return;this.sliderIsDragging=true;Mojo.Log.info("pausing on drag");if(!this.videoPlayer.paused){this.eatPause=true;this.videoPlayer.pause();this.pausedFromDrag=true}else this.pausedFromDrag=false},progressBarSeek:function(event){var pos=event.value;var secs=Math.round(pos/100*this.videoPlayer.duration);
-this.updateCounters(secs)},_registerForDisplayChanges:function(){this.powerService=Util.powerService.registerForEvents(this.controller,this._powerServiceCallback.bind(this))},_powerServiceCallback:function(event){if(!this.playEngine)return;if(event.event==="displayOff"){this.playEngine.setBlockPlayEvents(true);Mojo.Log.info("display is off, sending pause");this.pause()}else if(event.event==="displayOn"&&this.controller.stageController.active)this.playEngine.setBlockPlayEvents(!this.autoplay)},_isLocal:function(url){var localProtocol=
+this.updateCounters(secs)},_registerForDisplayChanges:function(){this.powerService=Util.powerService.registerForEvents(this.controller,this._powerServiceCallback.bind(this))},_powerServiceCallback:function(event){if(!this.playEngine)return;if(event.event==="displayOff"){Mojo.Log.info("display is off, sending pause");}else if(event.event==="displayOn"&&this.controller.stageController.active)this.playEngine.setBlockPlayEvents(!this.autoplay)},_isLocal:function(url){var localProtocol=
 "file:";return url[0]=="/"||url.substring(0,localProtocol.length).toLowerCase()===localProtocol},aboutToActivate:function(callback){this.controller.stageController.setWindowProperties({suppressBannerMessages:true});this.controller.stageController.setWindowOrientation("left");if(this.video.path)callback.defer();else this.aboutToActivateCallback=callback},activate:function(){if(!Util.transcoding&&this.video.path)this.loadVideo();else this._transcodingInterval=this.controller.window.setInterval(function(){if(Util.transcoding&&
 this.video.path);else{this.controller.window.clearInterval(this._transcodingInterval);this.loadVideo()}}.bind(this),1E3)},loadVideo:function(){if(!this.playEngine){this.showProgressInfo();this.playEngine=new StreamingPlayEngine(this.videoPlayer,this.videoExt,this,this.video.path,this.video.playbackPosition,!this.autoplay);this.mediaController=new MediaController(this.videoPlayer,this.playEngine,this);Util.markForeground(this.controller,true)}},savePlaybackProgress:function(){if(this.video._id!==undefined)if(this.videoPlayer.duration>
 this.MIN_DURATION_FOR_SAVE&&Math.abs(this.videoPlayer.duration-this.videoPlayer.currentTime)>this.MIN_TIME_DELTA_FOR_SAVE)this._doWritePlaybackProgress(this.video._id,this.videoPlayer.currentTime);else if(this.video.playbackPosition!==undefined&&this.video.playbackPosition>0)this._doWritePlaybackProgress(this.video._id,0)},deactivate:function(){this._blockDim(false);this.controller.document.removeEventListener(Mojo.Event.stageActivate,this.focusHandler,false);this.controller.document.removeEventListener(Mojo.Event.stageDeactivate,
