diff --git a/usr/palm/frameworks/metascene.videos.share/submission/66/concatenated.js.rej b/usr/palm/frameworks/metascene.videos.share/submission/66/concatenated.js.rej
new file mode 100644
index 0000000..0ec8840
--- /dev/null
+++ b/usr/palm/frameworks/metascene.videos.share/submission/66/concatenated.js.rej
@@ -0,0 +1,37 @@
+--- usr/palm/frameworks/metascene.videos.share/submission/66/concatenated.js
++++ usr/palm/frameworks/metascene.videos.share/submission/66/concatenated.js
+@@ -72,7 +72,7 @@
+ };
+ 
+ ShareService.Email = {
+-	link : function(sceneController, videoUrl){
++	link : function(sceneController, videoUrl, videoTitle){
+ 		sceneController.serviceRequest(
+ 			'palm://com.palm.applicationManager', {
+ 			method: 'open',
+@@ -80,6 +80,7 @@
+ 				id: 'com.palm.app.email',
+ 				params: {
+ 				recipients: [],
++				summary : videoTitle,
+ 				text : videoUrl
+ 			}}
+ 		});
+@@ -681,7 +682,7 @@
+ 	},
+ 	
+ 	handleEmailIcon : function(){
+-		ShareService.Email.link(this.controller, this.link + " - " + this.video.title);
++		ShareService.Email.link(this.controller, this.link + " - " + this.video.title, this.video.title);
+ 		this.closeDashboard();
+ 	},
+ 	
+@@ -1568,7 +1569,7 @@
+ 			case "link":
+ 				var startIndex = 11; // "email_link_".length === 11		
+ 				ShareService.Email.link(sceneController, 
+-					cmd.substr(startIndex) + " - " + vid.details.title);
++					cmd.substr(startIndex) + " - " + vid.details.title, vid.details.title);
+ 			break;
+ 		}
+ 	}
\ No newline at end of file
diff --git a/usr/palm/frameworks/metascene.videos.share/submission/66/concatenated.js.~1~ b/usr/palm/frameworks/metascene.videos.share/submission/66/concatenated.js.~1~
new file mode 100644
index 0000000..e69de29
