diff --git a/usr/palm/frameworks/private/metascene.videos.share/submission/48/javascript/share-master.js b/usr/palm/frameworks/private/metascene.videos.share/submission/48/javascript/share-master.js
index f2ee560..4131972 100644
--- a/usr/palm/frameworks/private/metascene.videos.share/submission/48/javascript/share-master.js
+++ b/usr/palm/frameworks/private/metascene.videos.share/submission/48/javascript/share-master.js
@@ -47,7 +47,7 @@ var ShareMaster = function(){
 	var limits = {
 		size : {
 			youtube : 2147483648, //2GB
-			email : 7340032, //7MB
+			email : 26214400, //25MB which is gmail's limit
 			facebook : 1073741824 //1GB default
 		},
 		duration : {
