diff --git a/usr/palm/frameworks/private/metascene.videos/submission/76.5/javascript/assistants/nowplaying-assistant.js b/usr/palm/frameworks/private/metascene.videos/submission/76.5/javascript/assistants/nowplaying-assistant.js
index fe70ac7..58354ea 100644
--- a/usr/palm/frameworks/private/metascene.videos/submission/76.5/javascript/assistants/nowplaying-assistant.js
+++ b/usr/palm/frameworks/private/metascene.videos/submission/76.5/javascript/assistants/nowplaying-assistant.js
@@ -258,7 +258,7 @@ var NowplayingAssistant = Class.create(metascene.Base, {
 		this._blockDim(false);
 
 		if (this.mediaController.isStaticStream){
-			this.pause();		
+			//this.pause();		
 		} else {
 			this.mediaController.stop();
 		}
@@ -359,7 +359,7 @@ var NowplayingAssistant = Class.create(metascene.Base, {
 		if (event.event === "displayOff"){
 			this.playEngine.setBlockPlayEvents(true);
 			Mojo.Log.info ("display is off, sending pause");
-			this.pause();
+			//this.pause();
 		} else if (event.event === "displayOn" && this.controller.stageController.active){
 			//this tells the videoplayer to continue blocking play events if autoplay is set to false
 			this.playEngine.setBlockPlayEvents(!this.autoplay);
