diff --git a/usr/palm/frameworks/private/metascene.videos/submission/76/javascript/assistants/details-assistant.js b/usr/palm/frameworks/private/metascene.videos/submission/76/javascript/assistants/details-assistant.js
index a197d1a..9c2f44e 100644
--- a/usr/palm/frameworks/private/metascene.videos/submission/76/javascript/assistants/details-assistant.js
+++ b/usr/palm/frameworks/private/metascene.videos/submission/76/javascript/assistants/details-assistant.js
@@ -28,6 +28,7 @@ DetailsAssistant.prototype.setup = function(){
             visible: true,
             items: [
                 { icon: "video-edit", command: "edit", disabled: !this.captured },
+                { icon: "video-delete", command: "delete", disabled: false },
                 { icon: "video-share", command: "share", disabled: !this.captured }
             ]
         }
@@ -95,6 +96,9 @@ DetailsAssistant.prototype.handleCommand = function(e) {
             case "share":
                 this.shareVideo(e);
                 break;
+            case "delete":
+                this.handleDelete();
+                break;
             case "edit":
                 this.editVideo();
                 break;
diff --git a/usr/palm/frameworks/private/metascene.videos/submission/76/javascript/assistants/nowplaying-assistant.js b/usr/palm/frameworks/private/metascene.videos/submission/76/javascript/assistants/nowplaying-assistant.js
index b242152..1c8f07e 100644
--- a/usr/palm/frameworks/private/metascene.videos/submission/76/javascript/assistants/nowplaying-assistant.js
+++ b/usr/palm/frameworks/private/metascene.videos/submission/76/javascript/assistants/nowplaying-assistant.js
@@ -73,6 +73,12 @@ var NowplayingAssistant = Class.create(metascene.Base, {
 		}
 
 		this.setVideoTitle();
+		this.controller.get("video_delete_item").show();
+		this.autoListen(
+			this.controller.get("video_delete_item"), 
+			Mojo.Event.tap, 
+			this.handleDelete.bind(this), 
+			false);
 		
 		// setup the spinning progress widget		
 		var spinnerAttrs = {
@@ -804,6 +810,30 @@ var NowplayingAssistant = Class.create(metascene.Base, {
 		}		
 	},
 
+	handleDelete: function(){ //function adapted from the details assistant
+		this.mediaController.pause();
+		this.controller.showAlertDialog({
+			onChoose: function(value){
+					if(value === "confirmed"){
+						Mojo.Log.info("Deleting the following file: "+this.videoUrl);
+						this.deleteFileRequest = VideoService.deleteFile(this.controller, this.videoUrl);
+						this.controller.stageController.popScenesTo("browser");
+					}
+				}.bind(this),
+			title: $LF("Delete video?"),
+			message: $LF("This action cannot be undone."),
+			choices: [{
+				label: $LF("Delete"),
+				value: "confirmed",
+				type: 'negative'
+			},{
+				label: $LF("Cancel"),
+				value: "cancelled",
+				type: 'dismiss'
+			}]
+		});
+	},
+
 	/*
 	 * Called by the play engine if we cannot support pausing this stream 
 		 */
diff --git a/usr/palm/frameworks/private/metascene.videos/submission/76/styles/media-video.css b/usr/palm/frameworks/private/metascene.videos/submission/76/styles/media-video.css
index ceecba9..9ce24c9 100644
--- a/usr/palm/frameworks/private/metascene.videos/submission/76/styles/media-video.css
+++ b/usr/palm/frameworks/private/metascene.videos/submission/76/styles/media-video.css
@@ -327,6 +327,16 @@ body.palm-default,
     margin-right: -25px;
 }
 
+.video-delete-button {
+    position: static;
+    float: right;
+    margin-right: -25px;
+}
+
+.palm-menu-icon.video-delete{
+	background-image: url(/usr/palm/frameworks/mojo/submissions/338/images/menu-icon-delete.png); 
+}
+
 .palm-menu-icon { background: url(../images/menu-icons.png) 0px 0px no-repeat; }
 
 .palm-menu-icon.video-play { background-position: 0px -64px; }
diff --git a/usr/palm/frameworks/private/metascene.videos/submission/76/views/nowplaying-scene.html b/usr/palm/frameworks/private/metascene.videos/submission/76/views/nowplaying-scene.html
index 865f1e2..09f877c 100644
--- a/usr/palm/frameworks/private/metascene.videos/submission/76/views/nowplaying-scene.html
+++ b/usr/palm/frameworks/private/metascene.videos/submission/76/views/nowplaying-scene.html
@@ -10,6 +10,9 @@
         <div id="video_share_item" class="palm-menu-button video-share-button" style="display:none" x-mojo-touch-feedback="immediate">
             <div class="palm-menu-icon video-share"></div>
         </div>
+        <div id="video_delete_item" class="palm-menu-button video-delete-button" style="display:none" x-mojo-touch-feedback="immediate">
+            <div class="palm-menu-icon video-delete"></div>
+        </div>
         <h1 id="video_title" class="truncating-text"></h1>
         <div id="video_limit" class="video-limit"></div>
     </div>
