diff --git a/usr/palm/frameworks/metascene.videos/submission/105.7/javascript/assistants/videoeditor-assistant.js b/usr/palm/frameworks/metascene.videos/submission/105.7/javascript/assistants/videoeditor-assistant.js
index 807a7b9..63f9abc 100644
--- a/usr/palm/frameworks/metascene.videos/submission/105.7/javascript/assistants/videoeditor-assistant.js
+++ b/usr/palm/frameworks/metascene.videos/submission/105.7/javascript/assistants/videoeditor-assistant.js
@@ -13,12 +13,23 @@ var VideoeditorAssistant = Class.create(NowplayingAssistant, {
 		$super(params);
 
 		//extra editing params
+		var date = new Date();
+		var YYYY = date.getFullYear();
+		var MM = (1+date.getMonth()); if (MM<10) { MM = "0"+MM; }
+		var DD = (0+date.getDate()); if (DD<10) { DD = "0"+DD; }
+		var hh = date.getHours(); if (hh < 10) { hh = "0"+hh; }
+		var mm = date.getMinutes(); if (mm < 10) { mm = "0"+mm; }
+		var ss = date.getSeconds(); if (ss < 10) { ss = "0"+ss; }
+		var path = params.video;
+		if(path) { path = path.path; } else { path = params.target; }
+		var fileBase = "_" + YYYY + "-" +  MM + "-" + DD + "-" + hh + mm + ss;
+		var destination = path.substring(0, path.lastIndexOf(".")) + fileBase + path.substring(path.lastIndexOf("."));
 		this.editingOptions = {
 			popAfterEdit : params.popAfterEdit,
 			openInEditMode : params.openInEditMode || params.popAfterEdit,
 			callback : params.callback,
 			timeLimit : params.timeLimit,
-			destination : params.destination,
+			destination : destination,
 			sizeLimit : params.sizeLimit,
 			restoreDefaults : function(){
 				this.popAfterEdit = false;
