diff --git a/usr/palm/frameworks/mediastream/submission/24/concatenated.js b/usr/palm/frameworks/mediastream/submission/24/concatenated.js
index d804c05..d0c4aa9 100755
--- a/usr/palm/frameworks/mediastream/submission/24/concatenated.js
+++ b/usr/palm/frameworks/mediastream/submission/24/concatenated.js
@@ -882,7 +882,11 @@ PausedState.prototype = {
 		//this just double checks to make sure we still have the scene(controller)
 		if(this.app.controller){
 			this.app.showPlayButton(true);
+			if (this.app.pcw_mediastream_donotshowcontrols !== true) {
 			this.app.showControls();
+			} else {
+				delete this.app.pcw_mediastream_donotshowcontrols;
+			}
 			this.app.hideStatus();
 		}
 	},
diff --git a/usr/palm/frameworks/metascene.videos/submission/103/concatenated.js b/usr/palm/frameworks/metascene.videos/submission/103/concatenated.js
index 9f3bb1d..2c266b5 100755
--- a/usr/palm/frameworks/metascene.videos/submission/103/concatenated.js
+++ b/usr/palm/frameworks/metascene.videos/submission/103/concatenated.js
@@ -1579,10 +1579,21 @@ var NowplayingAssistant = Class.create(metascene.Base, {
 		this.showControls();
 	},
 
-	onVideoTap: function()
+	onVideoTap: function(event)
 	{
 		Mojo.Log.info ("on video tap");
 		
+		if (event.down.screenX > (this.videoPlayer.offsetWidth / 2) /*&& this.videoPlayer.networkState !== 0*/) {
+			if (this.videoPlayer.paused === false) {
+				//this.pause();
+				this.videoPlayer.pause();
+				this.pcw_mediastream_donotshowcontrols = true;
+			} else {
+				this.videoPlayer.play();
+			}
+			return;
+		}
+		
 		if (this.controller.getMenuVisible(Mojo.Menu.commandMenu))
 		{
 			this.hideControls();
