diff --git a/usr/palm/frameworks/private/metascene.videos/submission/76/javascript/assistants/details-assistant.js b/usr/palm/frameworks/private/metascene.videos/submission/76/javascript/assistants/details-assistant.js
index a197d1a..495bf87 100644
--- a/usr/palm/frameworks/private/metascene.videos/submission/76/javascript/assistants/details-assistant.js
+++ b/usr/palm/frameworks/private/metascene.videos/submission/76/javascript/assistants/details-assistant.js
@@ -27,8 +27,8 @@ DetailsAssistant.prototype.setup = function(){
         {
             visible: true,
             items: [
-                { icon: "video-edit", command: "edit", disabled: !this.captured },
-                { icon: "video-share", command: "share", disabled: !this.captured }
+                { icon: "video-edit", command: "edit", disabled: false },
+                { icon: "video-share", command: "share", disabled: false }
             ]
         }
     );
