diff --git a/usr/palm/frameworks/private/metascene.videos/submission/75/javascript/assistants/browser-assistant.js b/usr/palm/frameworks/private/metascene.videos/submission/75/javascript/assistants/browser-assistant.js
index 5d05297..6ab6114 100644
--- a/usr/palm/frameworks/private/metascene.videos/submission/75/javascript/assistants/browser-assistant.js
+++ b/usr/palm/frameworks/private/metascene.videos/submission/75/javascript/assistants/browser-assistant.js
@@ -185,6 +185,9 @@ var BrowserAssistant = Class.create(metascene.Base, {
 		} else {
 			// show details scene
 			params.captured = this.query.captured;
+			params.destination = params.target.substring(0,
+					params.target.lastIndexOf(".")) + "_edit" +
+					params.target.substring(params.target.lastIndexOf("."));
 			metascene.Push(this.controller.stageController, exports.Details, params);
 		}
 	}
