diff --git a/usr/palm/applications/com.palm.app.browser/css/browser.css b/usr/palm/applications/com.palm.app.browser/css/browser.css
index b2734ba..50aff08 100644
--- a/usr/palm/applications/com.palm.app.browser/css/browser.css
+++ b/usr/palm/applications/com.palm.app.browser/css/browser.css
@@ -1,3 +1,32 @@
+.gestureBarRight {
+	width: 7px;
+	height: 1024px;
+	background: none;
+	position: absolute;
+	top: 0px;
+	right: 0px;
+	z-index: 120;
+}
+.gestureBarLeft {
+	width: 7px;
+	height: 1024px;
+	background: none;
+	position: absolute;
+	top: 0px;
+	left: 0px;
+	z-index: 120;
+}
+.navBar {
+	width: 50px;
+	background: rgba(0,0,0,.7);
+	padding: 5px 5px 5px 5px;
+	margin: 0px;
+	min-width: 10px;
+	-webkit-box-sizing: border-box;
+	border-width: 0px;
+	-webkit-border-image: url(../images/popup.png) 0 0 0 0 stretch stretch;
+}
+
 /* Start Page */
 
 .startpage-list .enyo-datalist-client {
diff --git a/usr/palm/applications/com.palm.app.browser/source/Browser.js b/usr/palm/applications/com.palm.app.browser/source/Browser.js
index f2d5a46..a823400 100644
--- a/usr/palm/applications/com.palm.app.browser/source/Browser.js
+++ b/usr/palm/applications/com.palm.app.browser/source/Browser.js
@@ -56,6 +56,8 @@ enyo.kind({
 			onPrint: "doPrint",
 			minFontSize: 2,
 		},
+		{kind: "Control", className: "gestureBarLeft", onmousedown: "openNavBar"},
+		{kind: "Control", className: "gestureBarRight", onmousedown: "openNavBar"},
 		{kind: "FindBar", showing: false, onFind: "find", onGoToPrevious: "goToPrevious", onGoToNext: "goToNext"},
 		{name: "context", kind: "BrowserContextMenu", onItemClick: "contextItemClick"},
 		{name: "dialog", kind: "VerticalAcceptCancelPopup", cancelCaption: "", components: [
@@ -84,7 +86,15 @@ enyo.kind({
 				{kind: "Button", flex: 1, caption: $L("Trust Once"), response: "2", onclick: "closeSSLConfirmBox"},
 				{kind: "Button", flex: 1, caption: $L("Don't Trust"), response: "0", onclick: "closeSSLConfirmBox"}
 			]}
-		]}
+		]},
+		{name: "navBar", kind: "Popup", className: "navBar", components: [
+			{kind: "ToolButton", name: "back", className: "actionbar-tool-button", icon: "images/chrome/menu-icon-back.png", onclick: "nbBack"},
+			{kind: "ToolButton", name: "forward", className: "actionbar-tool-button", icon: "images/chrome/menu-icon-forward.png", onclick: "nbForward"},
+			{kind: "ToolButton", name: "refresh", className: "actionbar-tool-button", icon: "images/chrome/menu-icon-refresh.png", onclick: "nbRefresh"},
+			//{kind: "ToolButton", name: "share", className: "actionbar-tool-button", icon: "images/chrome/menu-icon-share.png", onclick: "nbShare"},
+			{kind: "ToolButton", className: "actionbar-tool-button", icon: "images/chrome/menu-icon-newcard.png", onclick: "nbNewCard"},
+			{kind: "ToolButton", className: "actionbar-tool-button", icon: "images/chrome/menu-icon-bookmark.png", onclick: "nbOpenBookmarks"},
+		]},
 	],
 	WebKitErrors: {
 		ERR_SYS_FILE_DOESNT_EXIST: 14,
@@ -104,6 +114,32 @@ enyo.kind({
 			this.$.view.setIdentifier(enyo.windowParams.webviewId);
 		}
 	},
+	openNavBar: function(inSender, inEvent) {
+		this.$.navBar.openAtEvent(inEvent, {top: -100, left: -7});
+	},
+	nbBack: function() {
+		this.goBack();
+		this.$.navBar.close();
+	},
+	nbForward: function() {
+		this.goForward();
+		this.$.navBar.close();
+	},
+	nbRefresh: function() {
+		this.reloadClick();
+		this.$.navBar.close();
+	},
+	/*nbShare: function() {
+		
+	},*/
+	nbNewCard: function() {
+		this.openNewCard();
+		this.$.navBar.close();
+	},
+	nbOpenBookmarks: function() {
+		this.doOpenBookmarks();
+		this.$.navBar.close();
+	},
 	resize: function() {
 		this.$.actionbar.resize();
 		this.$.view.resize();
