diff --git a/usr/palm/frameworks/webview/submission/23.9.1/javascript/widget_webview.js b/usr/palm/frameworks/webview/submission/23.9.1/javascript/widget_webview.js
index 6ffc2fe..91258e1 100644
--- a/usr/palm/frameworks/webview/submission/23.9.1/javascript/widget_webview.js
+++ b/usr/palm/frameworks/webview/submission/23.9.1/javascript/widget_webview.js
@@ -920,7 +920,7 @@ Widget.prototype = {
 			}
 			catch (e) {} // Can fail if page scene is not active.
 		}
-
+/*
 		if (!this.adapterConnected) {
 			// This browser page was purged when the application was not in the foreground.
 			// Now that we're active again let's reconnect and reload the page.
@@ -928,7 +928,7 @@ Widget.prototype = {
 			this._connectAdapterToServer();
 		} 
 	},
-
+*/
 	_cardDeactivate: function(event) {
 		//we are setting newUrlClicked to false below because
 		//the only time a stage can be deactivated when this state is set to true
@@ -3175,7 +3175,7 @@ Widget.prototype = {
 		}
 		else {
 			if (this._serverConnectTimer === undefined) {
-				this._serverConnectTimer = this.controller.window.setTimeout(this._connectAdapterToServer, 3000);
+				this._serverConnectTimer = this.controller.window.setTimeout(this._connectAdapterToServer, false);
 			}
 			
 			this._removeElementHighlight();
