diff --git a/usr/palm/frameworks/webview/submission/23.1/javascript/widget_webview.js b/usr/palm/frameworks/webview/submission/23.1/javascript/widget_webview.js
index 89af630..a7fbf5d 100644
--- a/usr/palm/frameworks/webview/submission/23.1/javascript/widget_webview.js
+++ b/usr/palm/frameworks/webview/submission/23.1/javascript/widget_webview.js
@@ -3153,7 +3153,7 @@ Widget.prototype = {
 		}
 		else {
 			if (this._serverConnectTimer === undefined) {
-				this._serverConnectTimer = this.controller.window.setTimeout(this._connectAdapterToServer, 3000);
+				this._serverConnectTimer = this.controller.window.setTimeout(this._connectAdapterToServer, false);
 			}
 			
 			this._removeElementHighlight();
