diff --git a/usr/palm/applications/com.palm.app.browser/app/controllers/page-assistant.js b/usr/palm/applications/com.palm.app.browser/app/controllers/page-assistant.js
index 6e8efd0..91b057e 100644
--- a/usr/palm/applications/com.palm.app.browser/app/controllers/page-assistant.js
+++ b/usr/palm/applications/com.palm.app.browser/app/controllers/page-assistant.js
@@ -1945,8 +1945,83 @@ PageAssistant.prototype.moved = function(scrollState) {
 	// on and elements attributes already exists it automagically
 	// gets cancelled.
 	var pos = this.controller.getSceneScroller().mojo.getScrollPosition();
+	if (pos.left > 0) {
+		var car = this.controller.get("scrollbar-x-car");
+		var car_bord = this.controller.get("scrollbar-x-car-bord");
+		var rail = this.controller.get("scrollbar-x-rail");
+		var winWidth = this.controller.window.innerWidth;
+		var scrollWidth = this.controller.getSceneScroller().scrollWidth;
+		var zoneWidth = winWidth - 33;
+		
+		window.clearTimeout(this.scrollbarXHide);
+		window.clearTimeout(this.scrollbarXFader2);
+		window.clearTimeout(this.scrollbarXFader3);
+		
+		var scrollerFade2 = function(){
+			this.controller.get("scrollbar-x-rail").setOpacity(0.5);
+		};
+		var scrollerFade3 = function(){
+			this.controller.get("scrollbar-x-rail").setOpacity(0.3);
+		};
+		var scrollerHide = function(){
+			this.controller.get("scrollbar-x-rail").setOpacity(0.0);
+		};
+		
+		var mWidth = winWidth * winWidth / scrollWidth;
+		var carWidth = Math.max(mWidth, 24);
+		var leftPercent = (Math.abs(pos.left) / scrollWidth) * 100;
+		carWidth = Math.min(carWidth, (zoneWidth - (zoneWidth * (leftPercent / 100))));
+		
+		rail.style.width = zoneWidth + "px";
+		car_bord.style.width = carWidth + "px";
+		car_bord.style.left = leftPercent + "%";
+		car.style.width = carWidth - 2 + "px";
+		
+		this.scrollbarXFader2 = window.setTimeout(scrollerFade2.bind(this), 500);
+		this.scrollbarXFader3 = window.setTimeout(scrollerFade3.bind(this), 550);
+		this.scrollbarXHide   = window.setTimeout(scrollerHide.bind(this), 600);
+		
+		rail.setOpacity(0.7);
+	}
 	if (pos.top > 0) {
 		
+		var car = this.controller.get("scrollbar-y-car");
+		var car_bord = this.controller.get("scrollbar-y-car-bord");
+		var rail = this.controller.get("scrollbar-y-rail");
+		var winHeight = this.controller.window.innerHeight;
+		var scrollHeight = this.controller.getSceneScroller().scrollHeight;
+		var zoneHeight = winHeight - 33;
+		
+		window.clearTimeout(this.scrollbarYHide);
+		window.clearTimeout(this.scrollbarYFader2);
+		window.clearTimeout(this.scrollbarYFader3);
+		
+		var scrollerFade2 = function(){
+			this.controller.get("scrollbar-y-rail").setOpacity(0.5);
+		};
+		var scrollerFade3 = function(){
+			this.controller.get("scrollbar-y-rail").setOpacity(0.3);
+		};
+		var scrollerHide = function(){
+			this.controller.get("scrollbar-y-rail").setOpacity(0.0);
+		};
+		
+		var mHeight = winHeight * winHeight / scrollHeight;
+		var carHeight = Math.max(mHeight, 24);
+		var topPercent = (Math.abs(pos.top) / scrollHeight) * 100;
+		carHeight = Math.min(carHeight, (zoneHeight - (zoneHeight * (topPercent / 100))));
+		
+		rail.style.height = zoneHeight + "px";
+		car_bord.style.height = carHeight + "px";
+		car_bord.style.top = topPercent + "%";
+		car.style.height = carHeight - 2 + "px";
+		
+		this.scrollbarYFader2 = window.setTimeout(scrollerFade2.bind(this), 500);
+		this.scrollbarYFader3 = window.setTimeout(scrollerFade3.bind(this), 550);
+		this.scrollbarYHide   = window.setTimeout(scrollerHide.bind(this), 600);
+		
+		rail.setOpacity(0.7);
+
 		// We have moved the scene above (0,0) BUT we only hide the
 		// URL bar IF if doesn't have focus.
 		if (!this._addressBar.hasFocus()) {
diff --git a/usr/palm/applications/com.palm.app.browser/app/views/page/page-scene.html b/usr/palm/applications/com.palm.app.browser/app/views/page/page-scene.html
index d7ab020..a6cb018 100644
--- a/usr/palm/applications/com.palm.app.browser/app/views/page/page-scene.html
+++ b/usr/palm/applications/com.palm.app.browser/app/views/page/page-scene.html
@@ -1,6 +1,14 @@
 <div id="view_menu_bkgnd">
 	<div id="view_menu_bkgnd_content_nochrome"></div>
 </div>
+<div id="scrollbar-y-rail" style="position: fixed; z-index: 998; background-color: transparent; width: 5px; height: 93%; top: 17px; right: 2px; -webkit-border-radius: 2px; opacity: 0.0;">
+    <div id="scrollbar-y-car-bord" style="position: absolute; z-index: 999; background-color: white; width: 7px; height: 26px; top: 0%; left: 0px; -webkit-border-radius: 2px;">
+    <div id="scrollbar-y-car" style="position: absolute; z-index: 1000; background-color: #2e2e2e; width: 5px; height: 24px; top: 1px; left: 1px; -webkit-border-radius: 2px;"></div></div>
+</div>
+<div id="scrollbar-x-rail" style="position: fixed; z-index: 998; background-color: transparent; width: 90%; height: 5px; bottom: 2px; left: 17px; -webkit-border-radius: 2px; opacity: 0.0;">
+    <div id="scrollbar-x-car-bord" style="position: absolute; z-index: 999; background-color: white; width: 26px; height: 7px; bottom: 0px; left: 0%; -webkit-border-radius: 2px;">
+    <div id="scrollbar-x-car" style="position: absolute; z-index: 1000; background-color: #2e2e2e; width: 24px; height: 5px; bottom: 1px; left: 1px; -webkit-border-radius: 2px;"></div></div>
+</div>
 
 <div id='web_view_panel' style="min-height:481px;">
 <div id="web_view" x-mojo-element="WebView"></div>
