diff --git a/usr/palm/applications/com.palm.app.browser/app/controllers/page-assistant.js b/usr/palm/applications/com.palm.app.browser/app/controllers/page-assistant.js
index 6e8efd0..c90474d 100644
--- a/usr/palm/applications/com.palm.app.browser/app/controllers/page-assistant.js
+++ b/usr/palm/applications/com.palm.app.browser/app/controllers/page-assistant.js
@@ -232,7 +232,7 @@ PageAssistant.prototype.orientationChanged = function(event) {
 	// Switch the orientation of the menu spacer or other components..
 	//Mojo.Log.info("ORIENTATIONCHANGED: old: %s, new: %s", this._orientation, orientation);
 	var targetWindow = this.controller.window;
-	if (this._orientation === event.orientation) {
+	if (this._orientation) {
 		return;
 	}
 
@@ -463,7 +463,7 @@ PageAssistant.prototype.ready = function() {
 
 		// Update the web preferences.
 		AppAssistant.WebPreferences.activate(this._webView.mojo);
-
+		this._orientation = "free";
 		// Set the initial orientation.
 		this._setOrientation(this._orientation);
 
@@ -637,6 +637,7 @@ PageAssistant.prototype._onWebViewSingleTap = function(event) {
 		if (event.altKey) {
 
 			var popupItems = [
+			    {label: $L('Lock Screen'), command:'lockScreen'},
 				{label: $L('Open In New Card'), command:'openNew'},
 				{label: $L('Share Link'), command:'shareUrl'},
 				{label: $L('Copy URL'), command:'copyUrl'},
@@ -682,6 +683,9 @@ PageAssistant.prototype._onWebViewSingleTap = function(event) {
 			var popupSelectFunc = function(value) {
 				selectedCommand = value;
 				switch (value) {
+					case 'lockScreen':
+						this.lockScreen(urlInfo.url);
+						break;				
 					case 'openNew':
 						this._newBrowserPage(urlInfo.url);
 						break;
@@ -708,6 +712,7 @@ PageAssistant.prototype._onWebViewSingleTap = function(event) {
 				imageInfo = response;
 				var usedItems = [];
 				if (urlInfo.url) {
+				    usedItems.push( findItem('lockScreen') );
 					usedItems.push( findItem('openNew') );
 					usedItems.push( findItem('shareUrl') );
 					usedItems.push( findItem('copyUrl') );
@@ -1854,6 +1859,16 @@ PageAssistant.prototype._shareUrl = function(url, title, withCapture) {
 		method: 'open',
 		parameters: parameters
 	});
+};PageAssistant.prototype.lockScreen = function(link) {
+           if (this.controller.stageController.setWindowOrientation) {	
+								if(this.isWindowLocked){
+									this.controller.stageController.setWindowOrientation(Mojo.Controller.appController.getScreenOrientation());	
+									this.isWindowLocked = false;
+								}else{
+									this.controller.stageController.setWindowOrientation("free");	
+ 								this.isWindowLocked = true;
+								}								
+						}
 };
 
 PageAssistant.prototype._setWallpaper = function(pathToImage) {
