diff --git a/usr/palm/applications/com.palm.app.browser/app/controllers/page-applicationmenu.js b/usr/palm/applications/com.palm.app.browser/app/controllers/page-applicationmenu.js
index 6bf5870..a583cec 100644
--- a/usr/palm/applications/com.palm.app.browser/app/controllers/page-applicationmenu.js
+++ b/usr/palm/applications/com.palm.app.browser/app/controllers/page-applicationmenu.js
@@ -41,6 +41,12 @@ MenuData.ApplicationMenu = {
 		label: $L('History'),
 		command: 'show-history-cmd',
 		checkEnabled: true
+       },                                                               
+
+       MobilizePage: {
+               label: $L('Mobilize'),
+               command: 'mobilize-page-cmd',
+               checkEnabled: true
 	}
 };
 
diff --git a/usr/palm/applications/com.palm.app.browser/app/controllers/page-assistant.js b/usr/palm/applications/com.palm.app.browser/app/controllers/page-assistant.js
index 6e8efd0..4e2dbb2 100644
--- a/usr/palm/applications/com.palm.app.browser/app/controllers/page-assistant.js
+++ b/usr/palm/applications/com.palm.app.browser/app/controllers/page-assistant.js
@@ -804,6 +804,7 @@ PageAssistant.prototype._setupMenus = function() {
 				{
 					label: $L("Page"),
 					items: [
+						MenuData.ApplicationMenu.MobilizePage,
 						MenuData.ApplicationMenu.AddToLauncher,
 						MenuData.ApplicationMenu.SharePage]
 				},
@@ -2184,6 +2185,10 @@ PageAssistant.prototype.handleCommand = function(event) {
 				case MenuData.ApplicationMenu.SharePage.command:
 					this._shareUrl(this.currentUrl, this._getDisplayTitle(), true /*capture*/);
 					break;
+					
+                               case MenuData.ApplicationMenu.MobilizePage.command:
+                                       this.openUrl("http://instapaper.com/m?u=" + this.currentUrl);
+                                       break;
 
 				case MenuData.NavigationMenu.Back.command:
 					this._goBack();
@@ -2245,7 +2250,8 @@ PageAssistant.prototype.handleCommand = function(event) {
 				}
 
 				// Application specific Applicaiton Menu commands.
-				if (event.command === MenuData.ApplicationMenu.SharePage.command) {
+				if (event.command === MenuData.ApplicationMenu.SharePage.command
+					|| event.command === MenuData.ApplicationMenu.MobilizePage.command) {
 					// Only enable the menu items if we have a valid
 					// pages... (define a valid page)
 					if (!this.currentUrl) {
