diff --git a/usr/palm/applications/com.palm.app.browser/app/controllers/page-assistant.js b/usr/palm/applications/com.palm.app.browser/app/controllers/page-assistant.js
index 09c898f..1dc6465 100644
--- a/usr/palm/applications/com.palm.app.browser/app/controllers/page-assistant.js
+++ b/usr/palm/applications/com.palm.app.browser/app/controllers/page-assistant.js
@@ -621,7 +621,7 @@ PageAssistant.prototype._onWebViewSingleTap = function(event) {
 		tapPt.left = event.centerX - tapPt.left;
 		tapPt.top  = event.centerY - tapPt.top;
 
-		if (event.altKey) {
+		if (event.altKey || event.metaKey) {
 
 			var popupItems = [
 				{label: $L('Open In New Card'), command:'openNew'},
diff --git a/usr/palm/frameworks/mojo/builtins/palmInitFramework330.js b/usr/palm/frameworks/mojo/builtins/palmInitFramework330.js
index 177ea93..41b577d 100644
--- a/usr/palm/frameworks/mojo/builtins/palmInitFramework330.js
+++ b/usr/palm/frameworks/mojo/builtins/palmInitFramework330.js
@@ -28075,7 +28075,7 @@ if(!this.isElementHighlighted&&this.trackballMode===false&&this.selectionMode===
 this._addElementHighlight(event.centerX,event.centerY);
 }
 
-if(event.altKey){
+if(event.altKey || event.metaKey){
 
 
 
