diff --git a/usr/palm/applications/com.palm.app.browser/app/controllers/page-assistant.js b/usr/palm/applications/com.palm.app.browser/app/controllers/page-assistant.js
index 6e8efd0..ffef235 100644
--- a/usr/palm/applications/com.palm.app.browser/app/controllers/page-assistant.js
+++ b/usr/palm/applications/com.palm.app.browser/app/controllers/page-assistant.js
@@ -1814,6 +1814,7 @@ PageAssistant.prototype.deleteImage = function(filename) {
 /**
  * Share the url and optional image capture with an email recipient.
  */
+// bbito: Make bettter for use with plain text email
 PageAssistant.prototype._shareUrl = function(url, title, withCapture) {
 
 	if (url === undefined) {
@@ -1828,28 +1829,33 @@ PageAssistant.prototype._shareUrl = function(url, title, withCapture) {
 			title = url;
 		}
 	}
-
+	// bbito: Skip the png, I consider it useless
+/*
 	var captureFile;
 	if (withCapture) {
 		captureFile = '/tmp/captures/browser_page.png';
 
 		this._webView.mojo.saveViewToFile(captureFile);
 	}
-
-	var msg = $L("Here's a website I think you'll like: <a href='#{src}'>#{title}</a>").interpolate(
-			{src: url, title: title});
+*/
+	// bbito: Make the message just the title of the page, a linebreak, and URL.
+	// bbito: Original was 'Here's a website I think you'll like: ' plus the href-ed page title
+	var msg = title + "<br>" + url;
+	// bbito: Make the email subject the page title
+	// bbito: Original was 'Check out this web page...'
 	var parameters = {
 		id: 'com.palm.app.email',
 		params: {
-			summary: $L('Check out this web page...'),
+			summary: title,
 			text: msg
 		}
 	};
-
+	// bbito: just ignore the PNG
+/*
 	if (withCapture) {
 		parameters.params.attachments = [{fullPath: captureFile}];
 	}
-
+*/
 	this.controller.serviceRequest('palm://com.palm.applicationManager', {
 		method: 'open',
 		parameters: parameters
