diff --git a/usr/palm/applications/com.palm.app.browser/app/controllers/page-assistant.js b/usr/palm/applications/com.palm.app.browser/app/controllers/page-assistant.js
index 6e8efd0..b5961e5 100644
--- a/usr/palm/applications/com.palm.app.browser/app/controllers/page-assistant.js
+++ b/usr/palm/applications/com.palm.app.browser/app/controllers/page-assistant.js
@@ -93,7 +93,7 @@ PageAssistant.prototype.setup = function() {
 		this._updateOrientation();
 
 		// Allow override of 'back' gesture in landscape mode.
-		this.controller.useLandscapePageUpDown(true);
+		this.controller.useLandscapePageUpDown(false);
 
 		// Create the browser server status control.
 		this._browserServerStatus = new BrowserServerStatus(this.controller);
diff --git a/usr/palm/applications/com.palm.app.browser/app/controllers/startpage-assistant.js b/usr/palm/applications/com.palm.app.browser/app/controllers/startpage-assistant.js
index 0fbc276..aff5057 100644
--- a/usr/palm/applications/com.palm.app.browser/app/controllers/startpage-assistant.js
+++ b/usr/palm/applications/com.palm.app.browser/app/controllers/startpage-assistant.js
@@ -23,7 +23,7 @@ StartpageAssistant.prototype.setup = function() {
 	this._updateOrientation();
 
 	// Allow override of 'back' gesture in landscape mode.
-	this.controller.useLandscapePageUpDown(true);
+	this.controller.useLandscapePageUpDown(false);
 
 	// Listen for scene scope keyboard events.
 	this.controller.listen(this.controller.sceneElement, Mojo.Event.keydown, this._onKeyDownEventHandler);
