diff --git a/usr/palm/applications/com.palm.app.browser/source/ActionBar.js b/usr/palm/applications/com.palm.app.browser/source/ActionBar.js
index b680425..68bcbb0 100644
--- a/usr/palm/applications/com.palm.app.browser/source/ActionBar.js
+++ b/usr/palm/applications/com.palm.app.browser/source/ActionBar.js
@@ -106,4 +106,11 @@ enyo.kind({
 			}
 		}
 	},
+	privateChanged: function() {
+		if (this.private) {
+			this.applyStyle('background-color', '#8B0000');
+		} else {
+			this.applyStyle('background-color', null);
+		}
+	},
 });
diff --git a/usr/palm/applications/com.palm.app.browser/source/Browser.js b/usr/palm/applications/com.palm.app.browser/source/Browser.js
index 6094e1c..3faa841 100644
--- a/usr/palm/applications/com.palm.app.browser/source/Browser.js
+++ b/usr/palm/applications/com.palm.app.browser/source/Browser.js
@@ -104,7 +104,10 @@ enyo.kind({
 		this.defaultSearchChanged();
 		if (window.PalmSystem) {
 			this.$.view.setIdentifier(enyo.windowParams.webviewId);
+			this.private = enyo.windowParams.private || false;
 		}
+		this.privateChanged();
+		this.doPrivateChanged(this.private);
 	},
 	resize: function() {
 		this.$.actionbar.resize();
@@ -175,6 +178,9 @@ enyo.kind({
 	defaultSearchChanged: function() {
 		this.$.actionbar.setDefaultSearch(this.defaultSearch);
 	},
+	privateChanged: function() {
+		this.$.actionbar.setPrivate(this.private);
+	},
 	pageTitleChanged: function(inSender, inTitle, inUrl, inBack, inForward) {
 		this.log(inUrl, inTitle, inBack, inForward);
 		this.url = enyo.string.escapeHtml(inUrl);
diff --git a/usr/palm/applications/com.palm.app.browser/source/BrowserApp.js b/usr/palm/applications/com.palm.app.browser/source/BrowserApp.js
index 6d8659b..a6ed8bd 100644
--- a/usr/palm/applications/com.palm.app.browser/source/BrowserApp.js
+++ b/usr/palm/applications/com.palm.app.browser/source/BrowserApp.js
@@ -114,6 +114,9 @@ enyo.kind({
 		this.inherited(arguments);
 		this.fetchInitialPreferences();
 		this.log(enyo.json.stringify(enyo.windowParams));
+		
+		this.log("PRIVATE:", enyo.windowParams.private);
+		this.privateBrowsing = enyo.windowParams.private;
 	},
 	applicationRelaunchHandler: function(inSender) {
 		var c = enyo.windows.getActiveWindow();
@@ -147,6 +150,16 @@ enyo.kind({
 		this.showBookmarks();
 		//this.showAppMenu();
 		//this.resize();
+
+this.log("PRIVATE:", this.privateBrowsing);
+		if (this.privateBrowsing !== undefined) {
+			enyo.nextTick(this, this.privateSet(this.privateBrowsing));
+		} else {
+			this.$.getTweaks.call({
+				owner:  "org.webosinternals.patches.browser-private-browsing",
+				keys:   ["privateByDefault"]
+			});
+		}
 	},
 	processQueryString: function() {
 		var q = location.search.slice(1), queryArgs = {};
@@ -238,6 +251,13 @@ enyo.kind({
 		var browser = this.isBrowserShowing();
 		this.$.printMenuItem.setDisabled(!browser || this.isBrowserLoading());
 		this.$.preferencesItem.setDisabled(this.isPreferencesShowing());
+		this.$.privateMenuItem.setDisabled(this.privateBrowsing === undefined);
+
+		if (this.privateBrowsing) {
+			this.$.privateMenuItem.setCaption($L("Disable Private Browsing"));
+		} else {
+			this.$.privateMenuItem.setCaption($L("Enable Private Browsing"));
+		}
 	},
 	applyPreference: function(inPreference, inValue) {
 		this.log(inPreference, inValue);
@@ -495,6 +515,10 @@ enyo.kind({
 		this.$.launchApplicationService.call({id: inAppId, params: {target: inUrl, mime: inMime}});
 	},
 	updateHistory: function(inTitle, inUrl) {
+		if (this.privateBrowsing) {
+			return;
+		}
+
 		this.$.historyService.call(
 			{query: {where: [{prop:"url",op:"=",val:inUrl}]}},
 			{method: "delByQuery"});
@@ -604,6 +628,26 @@ enyo.kind({
 	preferencesClick: function() {
 		this.gotoView("preferences");
 	},
+	privateSet: function(private) {
+		this.privateBrowsing = private;
+
+		if (this.$.browser) {
+			this.$.browser.setPrivate(this.privateBrowsing);
+		}
+		if (this.$.startPage) {
+			this.$.startPage.setPrivate(this.privateBrowsing);
+		}
+	},
+	privateChanged: function() {
+		this.privateSet(this.$.browser.getPrivate());
+	},
+	privateClick: function() {
+		this.privateSet(!this.privateBrowsing);
+	},
+	gotTweaks: function(inSender, inResponse) {
+		this.log("PRIVATE:", inResponse);
+		enyo.nextTick(this, this.privateSet(inResponse.privateByDefault));
+	},
 	helpClick: function() {
 		this.$.launchApplicationService.call({params: {target: "http://help.palm.com/web/index.html"}, id: "com.palm.app.help"});
 	},
diff --git a/usr/palm/applications/com.palm.app.browser/source/StartPage.js b/usr/palm/applications/com.palm.app.browser/source/StartPage.js
index 8252498..3a3d8da 100644
--- a/usr/palm/applications/com.palm.app.browser/source/StartPage.js
+++ b/usr/palm/applications/com.palm.app.browser/source/StartPage.js
@@ -45,6 +45,9 @@ enyo.kind({
 	defaultSearchChanged: function() {
 		this.$.actionbar.setDefaultSearch(this.defaultSearch);
 	},
+	privateChanged: function() {
+		this.$.actionbar.setPrivate(this.private);
+	},
 	//* @public
 	resize: function() {
 		this.$.actionbar.resize();
diff --git a/usr/palm/applications/com.palm.app.browser/source/URLSearch.js b/usr/palm/applications/com.palm.app.browser/source/URLSearch.js
index af5952e..c6ef6bc 100644
--- a/usr/palm/applications/com.palm.app.browser/source/URLSearch.js
+++ b/usr/palm/applications/com.palm.app.browser/source/URLSearch.js
@@ -46,7 +46,11 @@ enyo.kind({
 	defaultSearchPreferences: [{
 		title: "Google",
 		//url: "http://www.google.com/m/search?client=ms-palm-webOS&channel=bm&q={$query}",
-		url: "http://www.google.com/search?q={$query}",
+		url: "https://encrypted.google.com/search?q={$query}&safe=off&pws=0",
+		icon: "list-icon-google.png"
+	}, {
+		title: "Google Secure",
+		url: "https://encrypted.google.com/search?q={$query}&safe=off&pws=0",	
 		icon: "list-icon-google.png"
 	}, {
 		title: "Wikipedia",
