diff --git a/usr/palm/applications/com.palm.app.browser/app/controllers/page-assistant.js b/usr/palm/applications/com.palm.app.browser/app/controllers/page-assistant.js
index 09c898f..12b7ab5 100644
--- a/usr/palm/applications/com.palm.app.browser/app/controllers/page-assistant.js
+++ b/usr/palm/applications/com.palm.app.browser/app/controllers/page-assistant.js
@@ -74,6 +74,7 @@ function PageAssistant(params){
 	this._onUrlRedirectHandler = this._onUrlRedirect.bind(this);
 	this._onWebViewResourceHandoffHandler = this._onWebViewResourceHandoff.bind(this);
 	this._onWebViewSingleTap = this._onWebViewSingleTap.bind(this);
+	this._onWebViewHoldTap = this._onWebViewHoldTap.bind(this);
 
 	// Spotlight handlers (flash support)
 	this._onSpotlightStartHandler = this._onSpotlightStart.bind(this);
@@ -325,6 +326,7 @@ PageAssistant.prototype.activate = function(message) {
 		webView.addEventListener(Mojo.Event.webViewServerConnect, this._onAdapterConnectHandler, false);
 		webView.addEventListener(Mojo.Event.webViewServerDisconnect, this._onAdapterDisconnectHandler, false);
 		webView.addEventListener('singletap', this._onWebViewSingleTap, true);
+		webView.addEventListener(Mojo.Event.hold, this._onWebViewHoldTap, true);
 
 		this.controller.getSceneScroller().addEventListener(Mojo.Event.scrollStarting,
 			this._addAsScrollListenerHandler, false);
@@ -429,6 +431,7 @@ PageAssistant.prototype.deactivate = function() {
 		webView.removeEventListener(Mojo.Event.webViewServerConnect, this._onAdapterConnectHandler, false);
 		webView.removeEventListener(Mojo.Event.webViewServerDisconnect, this._onAdapterDisconnectHandler, false);
 		webView.removeEventListener('singletap', this._onWebViewSingleTap, true);
+		webView.removeEventListener(Mojo.Event.hold, this._onWebViewHoldTap, true);
 
 		this.controller.getSceneScroller().removeEventListener(Mojo.Event.scrollStarting,
 			this._addAsScrollListenerHandler, false);
@@ -618,8 +621,16 @@ PageAssistant.prototype._onWebViewSingleTap = function(event) {
 
 	try {
 		var tapPt = Element.viewportOffset(this._webView);
-		tapPt.left = event.centerX - tapPt.left;
-		tapPt.top  = event.centerY - tapPt.top;
+		
+		if(event.isHoldTap) {
+			event.altKey = true;
+			tapPt.left = event.down.offsetX;
+			tapPt.top = event.down.offsetY;
+		}
+		else {
+			tapPt.left = event.centerX - tapPt.left;
+			tapPt.top  = event.centerY - tapPt.top;		
+		}
 
 		if (event.altKey) {
 
@@ -729,6 +740,11 @@ PageAssistant.prototype._onWebViewSingleTap = function(event) {
 	}
 };
 
+PageAssistant.prototype._onWebViewHoldTap = function(event) {	
+	event.isHoldTap = true;
+	this._onWebViewSingleTap(event);	
+};
+
 PageAssistant.prototype._onCardActivate = function(event) {
 
 	this.hasFocus = true;
