diff --git a/usr/palm/applications/com.palm.app.browser/app/controllers/page-assistant.js b/usr/palm/applications/com.palm.app.browser/app/controllers/page-assistant.js
index 09c898f..525086c 100644
--- a/usr/palm/applications/com.palm.app.browser/app/controllers/page-assistant.js
+++ b/usr/palm/applications/com.palm.app.browser/app/controllers/page-assistant.js
@@ -628,7 +628,8 @@ PageAssistant.prototype._onWebViewSingleTap = function(event) {
 				{label: $L('Share Link'), command:'shareUrl'},
 				{label: $L('Copy URL'), command:'copyUrl'},
 				{label: $L('Copy to Photos'), command:'copyToPhotos'},
-				{label: $L('Share Image'), command:'shareImage'},
+				{label: $L('Share Image Email'), command:'shareImage'},
+				{label: $L('Share Image Message '), command:'shareImageMms'},
 				{label: $L('Set Wallpaper'), command:'setWallpaper'}
 			];
 
@@ -650,6 +651,9 @@ PageAssistant.prototype._onWebViewSingleTap = function(event) {
 						case 'shareImage':
 							this._shareImage(imageInfo, path);
 							break;
+						case 'shareImageMms':
+							this._shareImageMms(imageInfo, path);
+							break;
 						case 'setWallpaper':
 							this._setWallpaper(path);
 							break;
@@ -684,6 +688,9 @@ PageAssistant.prototype._onWebViewSingleTap = function(event) {
 					case 'shareImage':
 						this._webView.mojo.saveImageAtPoint(tapPt.left, tapPt.top, "/tmp", saveImageCallback);
 						break;
+					case 'shareImageMms':
+						this._webView.mojo.saveImageAtPoint(tapPt.left, tapPt.top, "/tmp", saveImageCallback);
+						break;
 					case 'setWallpaper':
 						this._webView.mojo.saveImageAtPoint(tapPt.left, tapPt.top, "/media/internal", saveImageCallback);
 						break;
@@ -702,6 +709,7 @@ PageAssistant.prototype._onWebViewSingleTap = function(event) {
 
 				if (response.src) {
 					usedItems.push( findItem('shareImage') );
+					usedItems.push( findItem('shareImageMms') );
 				}
 
 				if (this._supportedImageType(response.src, response.mimeType)) {
@@ -1873,6 +1881,25 @@ PageAssistant.prototype._shareImage = function(imageInfoObj, pathToImage) {
 };
 
 /**
+ * Send text message with the supplied image attached.
+ */
+PageAssistant.prototype._shareImageMms = function(imageInfoObj, pathToImage) {
+
+	var parameters = {
+		id: 'com.palm.app.messaging',
+		params: {
+			messageText: $L('Check out this picture...'),
+			"attachment": pathToImage
+		}
+	};
+
+	this.controller.serviceRequest('palm://com.palm.applicationManager', {
+		method: 'open',
+		parameters: parameters
+	});
+};
+
+/**
  * Called whenever the scroller is moved.
  */
 PageAssistant.prototype.moved = function() {
