diff --git a/usr/palm/applications/com.palm.app.browser/app/controllers/page-assistant.js b/usr/palm/applications/com.palm.app.browser/app/controllers/page-assistant.js
index 09c898f..ddfa6f0 100644
--- a/usr/palm/applications/com.palm.app.browser/app/controllers/page-assistant.js
+++ b/usr/palm/applications/com.palm.app.browser/app/controllers/page-assistant.js
@@ -74,6 +74,7 @@ function PageAssistant(params){
 	this._onUrlRedirectHandler = this._onUrlRedirect.bind(this);
 	this._onWebViewResourceHandoffHandler = this._onWebViewResourceHandoff.bind(this);
 	this._onWebViewSingleTap = this._onWebViewSingleTap.bind(this);
+	this._shakeHandler = this._handleShake.bindAsEventListener(this);
 
 	// Spotlight handlers (flash support)
 	this._onSpotlightStartHandler = this._onSpotlightStart.bind(this);
@@ -308,6 +309,7 @@ PageAssistant.prototype.activate = function(message) {
 
 		this.controller.document.addEventListener(Mojo.Event.activate, this._onCardActivateHandler, false);
 		this.controller.document.addEventListener(Mojo.Event.deactivate, this._onCardDeactivateHandler, false);
+		this.controller.document.addEventListener('shakeend', this._shakeHandler, false);
 
 		var webView = this.controller.get('web_view');
 		webView.addEventListener(Mojo.Event.webViewLinkClicked, this._onLinkClickedHandler, true);
@@ -412,6 +414,7 @@ PageAssistant.prototype.deactivate = function() {
 		// Cleanup focus handlers.
 		this.controller.document.removeEventListener(Mojo.Event.activate, this._onCardActivateHandler, false);
 		this.controller.document.removeEventListener(Mojo.Event.deactivate, this._onCardDeactivateHandler, false);
+		this.controller.document.removeEventListener('shakeend', this._shakeHandler, false);
 
 		var webView = this.controller.get('web_view');
 		webView.removeEventListener(Mojo.Event.webViewTitleUrlChanged, this._onTitleUrlChangeHandler, true);
@@ -729,6 +732,11 @@ PageAssistant.prototype._onWebViewSingleTap = function(event) {
 	}
 };
 
+PageAssistant.prototype._handleShake = function(event) {
+
+	this._reload();
+};
+
 PageAssistant.prototype._onCardActivate = function(event) {
 
 	this.hasFocus = true;
