diff --git a/usr/palm/applications/com.palm.app.browser/source/BrowserApp.js b/usr/palm/applications/com.palm.app.browser/source/BrowserApp.js
index 1afbc52..5aa5d91 100644
--- a/usr/palm/applications/com.palm.app.browser/source/BrowserApp.js
+++ b/usr/palm/applications/com.palm.app.browser/source/BrowserApp.js
@@ -99,6 +99,7 @@ enyo.kind({
 		{kind: "AppMenu", onBeforeOpen: "toggleAppMenuItems", components: [
 			//{caption: $L("Find on Page"), onclick: "showFindOnPage"},
 			{name: "preferencesItem", caption: $L("Preferences"), onclick: "preferencesClick"},
+			{name: "privateMenuItem", caption: $L("Enable Private Browsing"), onclick: "privateClick"},
 			{name: "printMenuItem", caption: $L("Print"), onclick: "printClick"},
 			{caption: $L("Help"), onclick: "helpClick"}
 		]}
@@ -221,6 +222,11 @@ enyo.kind({
 		var browser = this.isBrowserShowing();
 		this.$.printMenuItem.setDisabled(!browser || this.isBrowserLoading());
 		this.$.preferencesItem.setDisabled(this.isPreferencesShowing());
+		if (this.privateBrowsing) {
+			this.$.privateMenuItem.setCaption($L("Disable Private Browsing"));
+		} else {
+			this.$.privateMenuItem.setCaption($L("Enable Private Browsing"));
+		}
 	},
 	applyPreference: function(inPreference, inValue) {
 		this.log(inPreference, inValue);
@@ -477,6 +483,10 @@ enyo.kind({
 		this.$.launchApplicationService.call({id: inAppId, params: {target: inUrl, mime: inMime}});
 	},
 	updateHistory: function(inTitle, inUrl) {
+		if (this.privateBrowsing) {
+			return;
+		}
+		
 		this.$.historyService.call(
 			{query: {where: [{prop:"url",op:"=",val:inUrl}]}},
 			{method: "delByQuery"});
@@ -586,6 +596,9 @@ enyo.kind({
 	preferencesClick: function() {
 		this.gotoView("preferences");
 	},
+	privateClick: function() {
+		this.privateBrowsing = !this.privateBrowsing;
+	},
 	helpClick: function() {
 		this.$.launchApplicationService.call({params: {target: "http://help.palm.com/web/index.html"}, id: "com.palm.app.help"});
 	},
