diff --git a/etc/palm/browser-app.conf b/etc/palm/browser-app.conf
index f952e33..dd3214e 100755
--- a/etc/palm/browser-app.conf
+++ b/etc/palm/browser-app.conf
@@ -43,7 +43,7 @@ MemCacheTotalBytes=8MB
 MemCachePruneUnusedTimeout=0
 
 # The page cache is how many previous pages to keep in memory.
-PageCacheCapacity=0
+PageCacheCapacity=1
 
 DiskCacheEnabled=true
 DiskCacheTotalBytes=8MB
@@ -51,7 +51,7 @@ DiskCacheDebug=false
 DiskCacheLargestObjectSize=1MB
 
 # disable gif animations till display list is optimized
-DisableGifAnimations=false
+DisableGifAnimations=true
 
 [JavaScript]
 Enabled=true
@@ -64,10 +64,10 @@ StackLimit=1M
 
 [Fonts]
 
-MinFontSize=8
+MinFontSize=6
 MinLogicalFontSize=5
-DefaultFontSize=14
-DefaultFixedFontSize=14
+DefaultFontSize=12
+DefaultFixedFontSize=12
 
 MinTextAutoSizeFontSize=18
 
