diff --git a/usr/palm/applications/com.palm.app.browser/app/controllers/history-assistant.js b/usr/palm/applications/com.palm.app.browser/app/controllers/history-assistant.js
index 8871138..cceb594 100644
--- a/usr/palm/applications/com.palm.app.browser/app/controllers/history-assistant.js
+++ b/usr/palm/applications/com.palm.app.browser/app/controllers/history-assistant.js
@@ -34,10 +34,13 @@ HistoryAssistant.prototype.setup = function() {
 		this.controller.setupWidget('historyList', {
 			itemTemplate:'history/history-entry',
 			listTemplate:'history/history-container',
-			itemsCallback:this._itemsCallback.bind(this)
+			itemsCallback:this._itemsCallback.bind(this),
+			swipeToDelete:true,
+			autoconfirmDelete:true
 		});
 
 		this._onListSelectionHandler = this._onListSelection.bindAsEventListener(this);
+		this._onListDeleteHandler = this._onListDelete.bindAsEventListener(this);
 		this._historyListWidget = this.controller.get('historyList');
 
 		var appMenuModel = {
@@ -63,6 +66,7 @@ HistoryAssistant.prototype.cleanup = function() {
 HistoryAssistant.prototype.activate = function() {
 
 	this._historyListWidget.addEventListener(Mojo.Event.listTap, this._onListSelectionHandler);
+	this._historyListWidget.addEventListener(Mojo.Event.listDelete, this._onListDeleteHandler);
 
 	// On an activation for a re-render of the contents.
 	var len = this._historyListWidget.mojo.getLength();	
@@ -72,6 +76,7 @@ HistoryAssistant.prototype.activate = function() {
 HistoryAssistant.prototype.deactivate = function() {
 
 	Mojo.Event.stopListening(this._historyListWidget, Mojo.Event.listTap, this._onListSelectionHandler);
+	Mojo.Event.stopListening(this._historyListWidget, Mojo.Event.listDelete, this._onListDeleteHandler);
 };
 
 /**
@@ -151,6 +156,10 @@ HistoryAssistant.prototype._onListSelection = function(event) {
 	});
 };
 
+HistoryAssistant.prototype._onListDelete = function(event) {
+	this.historyStore.deleteHistoryEntry(event.item.url, function() {}, function() {});
+};
+
 /**
  * handle a menu command.
  */
diff --git a/usr/palm/applications/com.palm.app.browser/app/models/history-store.js b/usr/palm/applications/com.palm.app.browser/app/models/history-store.js
index 360c746..51fb245 100644
--- a/usr/palm/applications/com.palm.app.browser/app/models/history-store.js
+++ b/usr/palm/applications/com.palm.app.browser/app/models/history-store.js
@@ -24,6 +24,11 @@ function HistoryStore(options, onSuccess, onFailure) {
 }
 
 /** @private */
+HistoryStore.prototype._deleteHistoryEntry = function(url, onSuccess, onFailure, transaction) {
+	this._executeSql(transaction, onSuccess, onFailure,
+		"DELETE FROM 'history' WHERE url = ?", [url] );
+};
+
 HistoryStore.prototype._addHistoryEntry = function(url, title, date, onSuccess, onFailure, transaction) {
 
 	var utcNow = date.getUTC().getTime();
@@ -41,6 +46,10 @@ HistoryStore.prototype._addHistoryEntry = function(url, title, date, onSuccess,
 		"INSERT INTO 'history' (date, url, title) VALUES(?, ?, ?)", [utcNow, url, title]);
 };
 
+HistoryStore.prototype.deleteHistoryEntry = function(url, onSuccess, onFailure) {
+	this.database.transaction(this._deleteHistoryEntry.bind(this, url, onSuccess, onFailure));
+};
+
 /**
  * Add a history entry.
  *
