--- .orig/usr/palm/applications/com.palm.app.browser/app/controllers/widget_webview.js
+++ /usr/palm/applications/com.palm.app.browser/app/controllers/widget_webview.js
@@ -465,7 +465,7 @@
 
 	reloadPage: function() {
 		if (this.useAdapter) {
-			this.adapter.reloadPage();
+			this.adapter.reloadPage(false);
 		}
 	},
 
@@ -1542,7 +1542,7 @@
 		}
 		else {
 			if (this._serverConnectTimer === undefined) {
-				this._serverConnectTimer = this.controller.window.setTimeout(this._connectAdapterToServer, 3000);
+				this._serverConnectTimer = this.controller.window.setTimeout(this._connectAdapterToServer, false);
 			}
 		}
 	},
