diff --git a/usr/palm/applications/com.palm.app.browser/app/controllers/page-assistant.js b/usr/palm/applications/com.palm.app.browser/app/controllers/page-assistant.js
index b2d6de4..e50e863 100644
--- a/usr/palm/applications/com.palm.app.browser/app/controllers/page-assistant.js
+++ b/usr/palm/applications/com.palm.app.browser/app/controllers/page-assistant.js
@@ -621,7 +621,7 @@ PageAssistant.prototype._onWebViewSingleTap = function(event) {
 		tapPt.left = event.centerX - tapPt.left;
 		tapPt.top  = event.centerY - tapPt.top;
 
-		if (event.altKey) {
+		if (event.altKey || event.metaKey) {
 
 			var popupItems = [
 				{label: $L('Open In New Card'), command:'openNew'},
diff --git a/usr/palm/frameworks/mojo/builtins/palmInitFramework200_72.js b/usr/palm/frameworks/mojo/builtins/palmInitFramework200_72.js
index ac3129f..4deb358 100644
--- a/usr/palm/frameworks/mojo/builtins/palmInitFramework200_72.js
+++ b/usr/palm/frameworks/mojo/builtins/palmInitFramework200_72.js
@@ -27025,7 +27025,7 @@ if(!this.isElementHighlighted&&this.trackballMode===false&&this.selectionMode===
 this._addElementHighlight(event.centerX,event.centerY);
 }
 
-if(event.altKey){
+if(event.altKey || event.metaKey){
 
 
 
