diff --git a/usr/palm/applications/com.palm.app.browser/app/controllers/page-assistant.js b/usr/palm/applications/com.palm.app.browser/app/controllers/page-assistant.js
index 09c898f..1dc6465 100644
--- a/usr/palm/applications/com.palm.app.browser/app/controllers/page-assistant.js
+++ b/usr/palm/applications/com.palm.app.browser/app/controllers/page-assistant.js
@@ -621,7 +621,7 @@ PageAssistant.prototype._onWebViewSingleTap = function(event) {
 		tapPt.left = event.centerX - tapPt.left;
 		tapPt.top  = event.centerY - tapPt.top;
 
-		if (event.altKey) {
+		if (event.altKey || event.metaKey) {
 
 			var popupItems = [
 				{label: $L('Open In New Card'), command:'openNew'},
