diff --git a/usr/palm/applications/com.palm.app.browser/app/controllers/page-assistant.js b/usr/palm/applications/com.palm.app.browser/app/controllers/page-assistant.js
index 6e8efd0..b4c9cd4 100644
--- a/usr/palm/applications/com.palm.app.browser/app/controllers/page-assistant.js
+++ b/usr/palm/applications/com.palm.app.browser/app/controllers/page-assistant.js
@@ -1947,6 +1947,33 @@ PageAssistant.prototype.moved = function(scrollState) {
 	var pos = this.controller.getSceneScroller().mojo.getScrollPosition();
 	if (pos.top > 0) {
 		
+		var mHeight = this.controller.getSceneScroller().scrollHeight - this.controller.window.innerHeight;
+		var curHeight = pos.top;
+		window.clearTimeout(this.scrollbarTimer);
+		window.clearTimeout(this.scrollbarFader2);
+		window.clearTimeout(this.scrollbarFader3);
+		var scrollerFade2 = function(){
+			this.controller.get('scrollbarbg').setOpacity(0.5);
+		};
+		var scrollerFade3 = function(){
+			this.controller.get('scrollbarbg').setOpacity(0.3);
+		};
+		var scrollerHide = function(){
+			this.controller.get('scrollbarbg').setOpacity(0.0);
+		};
+		this.scrollbarFader2 = window.setTimeout(scrollerFade2.bind(this),500);
+		this.scrollbarFader3 = window.setTimeout(scrollerFade3.bind(this),550);
+		this.scrollbarTimer  = window.setTimeout(scrollerHide.bind(this),600);
+		this.controller.get('scrollbarbg').setOpacity(0.7);
+		var maxHeight = this.controller.window.innerHeight / 2.5;
+		var maxScrubberLength=this.controller.window.innerHeight*0.45;
+		var hHeight = maxScrubberLength - Math.round(mHeight/20);
+		hHeight = Math.max(hHeight,24);
+		hHeight = Math.min(hHeight,maxScrubberLength);
+		this.controller.get('scrollbar').style.height = hHeight + 'px';
+		var topPercent = ((curHeight) / (mHeight)) * 100 * (720-hHeight)/720;
+		this.controller.get('scrollbar').style.top = topPercent + '%';
+
 		// We have moved the scene above (0,0) BUT we only hide the
 		// URL bar IF if doesn't have focus.
 		if (!this._addressBar.hasFocus()) {
diff --git a/usr/palm/applications/com.palm.app.browser/app/views/page/page-scene.html b/usr/palm/applications/com.palm.app.browser/app/views/page/page-scene.html
index d7ab020..b88c252 100644
--- a/usr/palm/applications/com.palm.app.browser/app/views/page/page-scene.html
+++ b/usr/palm/applications/com.palm.app.browser/app/views/page/page-scene.html
@@ -1,6 +1,11 @@
 <div id="view_menu_bkgnd">
 	<div id="view_menu_bkgnd_content_nochrome"></div>
 </div>
+<div id='scrollbarbg' style='position: fixed; z-index: 999; background-color: #2e2e2e; width: 4px; height:  87%; top: 6%; right: 2px; opacity: 0.0; -webkit-border-radius: 2px;'>
+<div id='scrollbar' style='position: absolute; z-index: 1000; background-color: white; width: 4px; height: 24px; top: 0%; left:  0px; -webkit-border-radius: 2px;'>
+ 
+</div>
+</div>
 
 <div id='web_view_panel' style="min-height:481px;">
 <div id="web_view" x-mojo-element="WebView"></div>
