diff --git a/usr/palm/applications/com.palm.app.browser/app/controllers/page-assistant.js b/usr/palm/applications/com.palm.app.browser/app/controllers/page-assistant.js
index 6e8efd0..5837d14 100644
--- a/usr/palm/applications/com.palm.app.browser/app/controllers/page-assistant.js
+++ b/usr/palm/applications/com.palm.app.browser/app/controllers/page-assistant.js
@@ -1836,12 +1836,12 @@ PageAssistant.prototype._shareUrl = function(url, title, withCapture) {
 		this._webView.mojo.saveViewToFile(captureFile);
 	}
 
-	var msg = $L("Here's a website I think you'll like: <a href='#{src}'>#{title}</a>").interpolate(
+	var msg = $L("<br><br><br> <a href='#{src}'>#{title}</a>").interpolate(
 			{src: url, title: title});
 	var parameters = {
 		id: 'com.palm.app.email',
 		params: {
-			summary: $L('Check out this web page...'),
+			summary: $L(''),
 			text: msg
 		}
 	};
