diff --git a/etc/palm/browser-app.conf b/etc/palm/browser-app.conf
index f0285ae..0477475 100755
--- a/etc/palm/browser-app.conf
+++ b/etc/palm/browser-app.conf
@@ -38,6 +38,7 @@ EnableAppCache=true
 AppCacheDbPath=/media/cryptofs/dbdata/appcache
 AppCacheDbMaxSize=1073741824
 EnableNoSteCommonTagsCheck=true
+UserAgentOverride=Mozilla/6.0 (Windows NT 6.2; WOW64; rv:16.0.1) Gecko/20121011 Firefox/16.0.1
 
 # The 'EnableFitWidth' parameter lets us control the default behavior of ScrollView.
 EnableFitWidth=false
