diff --git a/usr/lib/luna/system/luna-systemui/data/SystemManagerService.js b/usr/lib/luna/system/luna-systemui/data/SystemManagerService.js
index 9a29a1f..b3c7c71 100644
--- a/usr/lib/luna/system/luna-systemui/data/SystemManagerService.js
+++ b/usr/lib/luna/system/luna-systemui/data/SystemManagerService.js
@@ -47,8 +47,8 @@ enyo.kind({
 		this.uiAlertsMethods["tetheringError"] = enyo.bind(this, "handleCMBTPANNotifications");
 		this.uiAlertsMethods["mediaFilePermissionRequest"] = enyo.bind(this, "handleMediaFilePermissionRequest");
 		
-		this.delayedOpenSearchEngineAvailable = Utils.debounce(undefined, enyo.bind(this, "handleOpenSearchEngineAvailable"), 10);
-		this.uiAlertsMethods["openSearchEngineAvailable"] = this.delayedOpenSearchEngineAvailable;
+		//this.delayedOpenSearchEngineAvailable = Utils.debounce(undefined, enyo.bind(this, "handleOpenSearchEngineAvailable"), 10);
+		//this.uiAlertsMethods["openSearchEngineAvailable"] = this.delayedOpenSearchEngineAvailable;
 		
 		this.uiAlertsMethods["macroCallEvents"] = enyo.bind(this, "handleMacroCallEvents");
 		
