diff --git a/usr/palm/applications/com.palm.app.browser/app/controllers/history-assistant.js b/usr/palm/applications/com.palm.app.browser/app/controllers/history-assistant.js
index 8871138..2eea00b 100644
--- a/usr/palm/applications/com.palm.app.browser/app/controllers/history-assistant.js
+++ b/usr/palm/applications/com.palm.app.browser/app/controllers/history-assistant.js
@@ -20,13 +20,13 @@ function HistoryAssistant(webPreferences) {
 	this._chunkSize = 50;
 }
 
-HistoryAssistant.Menu = {
+//HistoryAssistant.Menu = {
 
-	Clear: {
-		label: $L('Clear History'),
-		command: 'clear-history-command'
-	}
-};
+//	Clear: {
+	//	label: $L('Clear History'),
+	//	command: 'clear-history-command'
+	//}
+//};
 
 HistoryAssistant.prototype.setup = function() {
 
@@ -45,7 +45,7 @@ HistoryAssistant.prototype.setup = function() {
 			items: [
 				MenuData.ApplicationMenu.ShowBookmarks,
 				MenuData.ApplicationMenu.ShowHistory,
-				HistoryAssistant.Menu.Clear
+				//HistoryAssistant.Menu.Clear
 			]
 		};
 
@@ -167,9 +167,9 @@ HistoryAssistant.prototype.handleCommand = function(event) {
 						this._openBookmarkView();
 						break;
 
-					case HistoryAssistant.Menu.Clear.command:
-						this._clearHistory();
-						break;
+					//case HistoryAssistant.Menu.Clear.command:
+						//this._clearHistory();
+					//	break;
 
 					case Mojo.Menu.prefsCmd:
 						this._openPreferencesView();
@@ -254,4 +254,3 @@ HistoryAssistant.prototype._showHelpCommand = function() {
 	// Launch the help system.
 	AppAssistant.launchHelp();
 };
-
diff --git a/usr/palm/applications/com.palm.app.browser/app/views/preferences/preferences-scene.html b/usr/palm/applications/com.palm.app.browser/app/views/preferences/preferences-scene.html
index 6daacae..a4f9682 100644
--- a/usr/palm/applications/com.palm.app.browser/app/views/preferences/preferences-scene.html
+++ b/usr/palm/applications/com.palm.app.browser/app/views/preferences/preferences-scene.html
@@ -38,7 +38,6 @@
             </div>
         </div>
     </div>
-    <div class="palm-button" id="clear_history_button" x-mojo-tap-highlight="momentary" x-mojo-loc=''>Clear History</div>
     <div class="palm-button" id="clear_cookies_button" x-mojo-tap-highlight="momentary" x-mojo-loc=''>Clear Cookies</div>
     <div class="palm-button" id="clear_cache_button" x-mojo-tap-highlight="momentary" x-mojo-loc=''>Clear Cache</div>
 	<div class="scroll-fade-bottom" x-mojo-scroll-fade="bottom"></div>
