diff --git a/usr/palm/applications/com.palm.app.browser/source/StartPage.js b/usr/palm/applications/com.palm.app.browser/source/StartPage.js
index 8252498..8dcae79 100644
--- a/usr/palm/applications/com.palm.app.browser/source/StartPage.js
+++ b/usr/palm/applications/com.palm.app.browser/source/StartPage.js
@@ -31,7 +31,7 @@ enyo.kind({
 		this.inherited(arguments);
 		// Always focus the action bar when start page is shown.
 		if (this.showing) {
-			this.$.actionbar.forceFocus();
+			//this.$.actionbar.forceFocus();
 		} else {
 			this.$.actionbar.forceBlur();
 		}
