diff --git a/usr/palm/applications/com.palm.app.browser/source/ActionBar.js b/usr/palm/applications/com.palm.app.browser/source/ActionBar.js
index b680425..bd431a7 100644
--- a/usr/palm/applications/com.palm.app.browser/source/ActionBar.js
+++ b/usr/palm/applications/com.palm.app.browser/source/ActionBar.js
@@ -36,6 +36,7 @@ enyo.kind({
 			{kind: "ToolButton", name: "share", className: "actionbar-tool-button", icon: "images/chrome/menu-icon-share.png", onclick: "showSharePopup"},
 			{kind: "ToolButton", className: "actionbar-tool-button", icon: "images/chrome/menu-icon-newcard.png", onclick: "doNewCard"},
 			{kind: "ToolButton", className: "actionbar-tool-button", icon: "images/chrome/menu-icon-bookmark.png", onclick: "doOpenBookmarks"},
+			{kind: "Image", className: "actionbar-tool-button", src: "../../sysmgr/images/keyboard-tablet/icon-hide-keyboard.png", onclick: "changeKB"},
 		]},	
 		{name: "sharePopup", className: "launch-popup",  kind: "Menu", components: [
 			{caption: $L("Add Bookmark"), onclick: "doAddBookmark"},
@@ -94,6 +95,15 @@ enyo.kind({
 	progressChanged: function() {
 		this.$.progressBar.setPosition(this.progress);
 	},
+	changeKB: function(){
+			if (enyo.keyboard.isShowing()){
+				enyo.keyboard.forceHide();
+				enyo.keyboard.setManualMode(false);
+			}
+			else{
+				enyo.keyboard.forceShow(0);
+			}
+		},
 	loadingChanged: function() {
 		this.$.search.setLoading(this.loading);
 		if (this.loading) {
