diff --git a/usr/palm/applications/com.palm.app.browser/source/Preferences.js b/usr/palm/applications/com.palm.app.browser/source/Preferences.js
index 2d67f4b..21b4423 100644
--- a/usr/palm/applications/com.palm.app.browser/source/Preferences.js
+++ b/usr/palm/applications/com.palm.app.browser/source/Preferences.js
@@ -39,7 +39,6 @@ enyo.kind({
 						{kind: "ToggleButton", name: "click2play", onChange: "togglePreferenceClick", preference: "click2play", type: "System", inverted: true}
 					]}
 				]},
-				{kind: "Button", caption: $L("Clear Bookmarks"), onclick: "promptButtonClick", dialog: "clearBookmarksPrompt"},
 				{kind: "Button", caption: $L("Clear History"), onclick: "promptButtonClick", dialog: "clearHistoryPrompt"},
 				{kind: "Button", caption: $L("Clear Cookies"), onclick: "promptButtonClick", dialog: "clearCookiesPrompt"},
 				{kind: "Button", caption: $L("Clear Cache"), onclick: "promptButtonClick", dialog: "clearCachePrompt"},
