diff --git a/usr/palm/frameworks/webview/submission/23.11/javascript/widget_webview.js b/usr/palm/frameworks/webview/submission/23.11/javascript/widget_webview.js
index 8f5d4be..9a53654 100644
--- a/usr/palm/frameworks/webview/submission/23.11/javascript/widget_webview.js
+++ b/usr/palm/frameworks/webview/submission/23.11/javascript/widget_webview.js
@@ -931,13 +931,14 @@ Widget.prototype = {
 			}
 			catch (e) {} // Can fail if page scene is not active.
 		}
-
+/*
 		if (!this.adapterConnected) {
 			// This browser page was purged when the application was not in the foreground.
 			// Now that we're active again let's reconnect and reload the page.
 			Mojo.Log.info("Reconnecting adapter to server.");
 			this._connectAdapterToServer();
 		} 
+*/
 	},
 
 	_cardDeactivate: function(event) {
@@ -3189,7 +3190,7 @@ Widget.prototype = {
                 if (this._serverConnectRetryTime > this._serverConnectRetryTimeMax) {
                     this._serverConnectRetryTime = this._serverConnectRetryTimeMax;
                 }
-                var retryTime = this._serverConnectRetryTime;
+                var retryTime = false;
                 this._serverConnectTimer = this.controller.window.setTimeout(this._connectAdapterToServer, retryTime);
                 this._serverConnectRetryTime = retryTime * 2;
 			}
