diff --git a/etc/hosts b/etc/hosts
index 6d85b45..8c76832 100644
--- a/etc/hosts
+++ b/etc/hosts
@@ -1,2 +1,12 @@
 127.0.0.1	localhost.localdomain		localhost
+127.0.0.1	rrmprod.amobee.com
+127.0.0.1	vfdeprod.amobee.com
+127.0.0.1	akamai.smartadserver.com
+127.0.0.1	www4.smartadserver.com
+127.0.0.1	img.ads.mocean.mobi
+127.0.0.1	img.ads1.mocean.mobi
+127.0.0.1	img.ads2.mocean.mobi
+127.0.0.1	im.banner.t-online.de
+127.0.0.1	dynimages.handy.de
+127.0.0.1	adserver.yoc.mobi
 127.0.0.1	palm-webos
\ No newline at end of file
