diff --git a/usr/palm/applications/com.palm.app.browser/app/controllers/page-controls.js b/usr/palm/applications/com.palm.app.browser/app/controllers/page-controls.js
index 6ad7ebd..ef60d09 100644
--- a/usr/palm/applications/com.palm.app.browser/app/controllers/page-controls.js
+++ b/usr/palm/applications/com.palm.app.browser/app/controllers/page-controls.js
@@ -26,7 +26,7 @@ function PageControls (controller){
 		};
 
 		this._model = {
-			visible: true,
+			visible: false,
 			items: [this._backModel, {}, {}]
 		};
 
@@ -64,7 +64,7 @@ PageControls.prototype={
 
 		var items = _.toArray();
 		this._backModel.disabled = !back;
-		items.push(this._backModel);
+		items.push(false);
 
 		if (forward) {
 			this._forwardModel.disabled = !forward;
@@ -74,7 +74,7 @@ PageControls.prototype={
 		}
 
 		// The spacer... right aligns the following button.
-		items.push({});
+		items.push(false);
 
 		this._back = back;
 		this._forward = forward;
@@ -85,7 +85,7 @@ PageControls.prototype={
 	updateBackForward: function(back, forward) {
 
 		// Add the back/forward buttons
-		this._model.items = this._setBackForward(back, forward);
+		this._model.items = this._setBackForward(false);
 		// The spacer... right aligns the following button.
 		this._model.items.push(this._currentPageModel);
 
@@ -108,11 +108,12 @@ PageControls.prototype={
 		}
 
 		// Add the back/forward buttons
-		this._model.items = this._setBackForward(this._back, this._forward);
+		this._model.items = this._setBackForward(false);
 		// The spacer... right aligns the following button.
 		this._model.items.push(idleModel);
 		this._controller.modelChanged(this._model);
 
+		this._controller.setMenuVisible(Mojo.Menu.commandMenu, false);
 		this._currentPageModel = idleModel;
 	},
 
@@ -138,11 +139,12 @@ PageControls.prototype={
 			});
 
 			// Add the back/forward buttons
-			this._model.items = this._setBackForward(this._back, this._forward);
+			this._model.items = this._setBackForward(false);
 			this._model.items.push(progressModel);
 			this._controller.modelChanged(this._model);
 			this._animation.start(0);
-
+			this._controller.setMenuVisible(Mojo.Menu.commandMenu, true);
+			
 			this._currentPageModel = progressModel;
 
 		} catch (e) {
@@ -183,11 +185,12 @@ PageControls.prototype={
 			});
 
 			// Add the back/forward buttons
-			this._model.items = this._setBackForward(this._back, this._forward);
+			this._model.items = this._setBackForward(false);
 			this._model.items.push(searchModel);
 			this._controller.modelChanged(this._model);
 			this._animation.start();
-
+			this._controller.setMenuVisible(Mojo.Menu.commandMenu, true);
+			
 			this._currentPageModel = searchModel;
 
 		} catch (e) {
@@ -204,6 +207,7 @@ PageControls.prototype={
 					this._model.items[i].type='flash';
 					this._model.items[i].icon= MenuData.NavigationMenu.Cancel.icon;
 					this._model.items[i].command=MenuData.NavigationMenu.Cancel.command;
+					this._controller.setMenuVisible(Mojo.Menu.commandMenu, true);
 					break;
 				}
 				
@@ -219,6 +223,7 @@ PageControls.prototype={
 					this._model.items[i].type='idle';
 					this._model.items[i].icon= MenuData.NavigationMenu.Reload.icon;
 					this._model.items[i].command=MenuData.NavigationMenu.Reload.command;
+					this._controller.setMenuVisible(Mojo.Menu.commandMenu, false);
 					break;
 				}
 				
