diff --git a/usr/palm/applications/com.palm.app.browser/source/Browser.js b/usr/palm/applications/com.palm.app.browser/source/Browser.js
index f2d5a46..28674a8 100644
--- a/usr/palm/applications/com.palm.app.browser/source/Browser.js
+++ b/usr/palm/applications/com.palm.app.browser/source/Browser.js
@@ -36,6 +36,7 @@ enyo.kind({
 			onOpenBookmarks: "doOpenBookmarks",
 			onNewCard: "openNewCard"
 		},
+		{kind: "FindBar", showing: false, onFind: "find", onGoToPrevious: "goToPrevious", onGoToNext: "goToNext"},
 		{name: "view", kind: window.PalmSystem ? "WebView" : "Iframe", flex: 1,
 			onMousehold: "openContextMenu",
 			onPageTitleChanged: "pageTitleChanged",
@@ -56,7 +57,6 @@ enyo.kind({
 			onPrint: "doPrint",
 			minFontSize: 2,
 		},
-		{kind: "FindBar", showing: false, onFind: "find", onGoToPrevious: "goToPrevious", onGoToNext: "goToNext"},
 		{name: "context", kind: "BrowserContextMenu", onItemClick: "contextItemClick"},
 		{name: "dialog", kind: "VerticalAcceptCancelPopup", cancelCaption: "", components: [
 			{name: "dialogTitle", classNAme: "enyo-dialog-prompt-title"},
@@ -119,16 +119,23 @@ enyo.kind({
 		this.viewCall("printFrame", ["", inJobID, inPrintParams.width, inPrintParams.height, inPrintParams.pixelUnits, false, inPrintParams.renderInReverseOrder]);
 	},
 	showFind: function() {
+		this.findStr = null;
 		this.$.findBar.show();
 	},
 	//* @protected
 	find: function(inSender, inString) {
 		this.log(inString);
-		this.$.view.callBrowserAdapter("findInPage", [inString]);
+		if (this.findStr != inString) {
+			/* Reset */
+			this.$.view.findInPage("");
+			this.findStr = inString;
+		}
+		this.$.view.findInPage(this.findStr);
 	},
 	goToPrevious: function() {
 	},
 	goToNext: function() {
+		this.$.view.findInPage(this.findStr);
 	},
 	setEnableJavascript: function(inEnable) {
 		this.viewCall("setEnableJavascript", [inEnable]);
diff --git a/usr/palm/applications/com.palm.app.browser/source/BrowserApp.js b/usr/palm/applications/com.palm.app.browser/source/BrowserApp.js
index 9a4bf6f..8ed437c 100644
--- a/usr/palm/applications/com.palm.app.browser/source/BrowserApp.js
+++ b/usr/palm/applications/com.palm.app.browser/source/BrowserApp.js
@@ -98,7 +98,7 @@ enyo.kind({
 		},
 		{kind: "PrintDialog", lazy: false, duplexOption: true, colorOption: true, onRenderDocument: "renderDocument", appName: "Browser"},
 		{kind: "AppMenu", onBeforeOpen: "toggleAppMenuItems", components: [
-			//{caption: $L("Find on Page"), onclick: "showFindOnPage"},
+			{name: "findMenuItem", caption: $L("Find on Page"), onclick: "showFindOnPage"},
 			{name: "preferencesItem", caption: $L("Preferences"), onclick: "preferencesClick"},
 			{name: "printMenuItem", caption: $L("Print"), onclick: "printClick"},
 			{caption: $L("Help"), onclick: "helpClick"}
@@ -221,6 +221,7 @@ enyo.kind({
 	toggleAppMenuItems: function() {
 		var browser = this.isBrowserShowing();
 		this.$.printMenuItem.setDisabled(!browser || this.isBrowserLoading());
+		this.$.findMenuItem.setDisabled(!browser);
 		this.$.preferencesItem.setDisabled(this.isPreferencesShowing());
 	},
 	applyPreference: function(inPreference, inValue) {
diff --git a/usr/palm/applications/com.palm.app.browser/source/FindBar.js b/usr/palm/applications/com.palm.app.browser/source/FindBar.js
index 90e8375..98ca277 100644
--- a/usr/palm/applications/com.palm.app.browser/source/FindBar.js
+++ b/usr/palm/applications/com.palm.app.browser/source/FindBar.js
@@ -9,7 +9,7 @@ enyo.kind({
 	components: [
 		{kind: "Input", name: "input", flex: 2, autoCapitalize: "lowercase", changeOnKeypress: true, onchange: "inputChange"},
 		{flex: 1},
-		{kind: "NoFocusButton", name: "prev", caption: $L("prev"), disabled: true, onclick: "findPrevious"},
+		{kind: "NoFocusButton", name: "prev", caption: $L("prev"), disabled: true, onclick: "findPrevious", showing: false},
 		{kind: "NoFocusButton", name: "next", caption: $L("next"), disabled: true, onclick: "findNext"},
 		{kind: "NoFocusButton", caption: "done", onclick: "close"}
 	],
