diff --git a/usr/lib/luna/system/luna-systemui/data/SystemManagerService.js b/usr/lib/luna/system/luna-systemui/data/SystemManagerService.js
index 9564d0d..5ec4fc9 100644
--- a/usr/lib/luna/system/luna-systemui/data/SystemManagerService.js
+++ b/usr/lib/luna/system/luna-systemui/data/SystemManagerService.js
@@ -44,8 +44,8 @@ enyo.kind({
 		this.uiAlertsMethods["tetheringError"] = enyo.bind(this, "handleCMBTPANNotifications");
 		this.uiAlertsMethods["mediaFilePermissionRequest"] = enyo.bind(this, "handleMediaFilePermissionRequest");
 		
-		this.delayedOpenSearchEngineAvailable = Utils.debounce(undefined, enyo.bind(this, "handleOpenSearchEngineAvailable"), 10);
-		this.uiAlertsMethods["openSearchEngineAvailable"] = this.delayedOpenSearchEngineAvailable;
+		//this.delayedOpenSearchEngineAvailable = Utils.debounce(undefined, enyo.bind(this, "handleOpenSearchEngineAvailable"), 10);
+		//this.uiAlertsMethods["openSearchEngineAvailable"] = this.delayedOpenSearchEngineAvailable;
 		
 		this.uiAlertsMethods["macroCallEvents"] = enyo.bind(this, "handleMacroCallEvents");
 		
