diff --git a/usr/palm/applications/com.palm.app.browser/app/controllers/page-applicationmenu.js b/usr/palm/applications/com.palm.app.browser/app/controllers/page-applicationmenu.js
index 6bf5870..0e1a092 100644
--- a/usr/palm/applications/com.palm.app.browser/app/controllers/page-applicationmenu.js
+++ b/usr/palm/applications/com.palm.app.browser/app/controllers/page-applicationmenu.js
@@ -19,6 +19,12 @@ MenuData.ApplicationMenu = {
 		checkEnabled: true
 	},
 	
+	NeatoThatUrl: {
+		label: $L('Neato! it'),
+		command: 'add-neato-cmd',
+		checkEnabled: true
+	},
+	
 	AddToLauncher: {
 		label: $L('Add to Launcher'),
 		command: 'add-launch-icon-cmd',
diff --git a/usr/palm/applications/com.palm.app.browser/app/controllers/page-assistant.js b/usr/palm/applications/com.palm.app.browser/app/controllers/page-assistant.js
index 388cffc..a89597d 100644
--- a/usr/palm/applications/com.palm.app.browser/app/controllers/page-assistant.js
+++ b/usr/palm/applications/com.palm.app.browser/app/controllers/page-assistant.js
@@ -796,6 +796,7 @@ PageAssistant.prototype._setupMenus = function() {
 				{
 					label: $L("Page"),
 					items: [
+						MenuData.ApplicationMenu.NeatoThatUrl,
 						MenuData.ApplicationMenu.AddToLauncher,
 						MenuData.ApplicationMenu.SharePage]
 				},
@@ -2172,6 +2173,45 @@ PageAssistant.prototype.handleCommand = function(event) {
 					this._createDefaultBookmarkImages(urlReference);
 					this.showBookmarkDialog(BookmarkDialogAssistant.createLaunchpointTask, urlReference);
 					break;
+				
+				case MenuData.ApplicationMenu.NeatoThatUrl.command:
+					// Lets Neato! that URL
+					try{
+						this.neatoappid = "com.zhephree.neato";
+						this.controller.serviceRequest("palm://com.palm.applicationManager", {
+							method: 'open',
+							parameters: {
+								id: this.neatoappid,
+								params: {send: '{"a":"url","c":"'+this.currentUrl+'"}'}
+							},
+							onFailure:function(){
+								//--> Promt to download if Neato! is not installed
+								this.controller.showAlertDialog({
+									onChoose: function(value){
+										if (value=="yes"){
+											//--> open app catalog at Neato! page
+											this.controller.serviceRequest('palm://com.palm.applicationManager', {
+												method:'open',
+												parameters:{
+													target: "http://developer.palm.com/appredirect/?packageid=" + this.neatoappid
+												}
+											});
+										}
+									},
+									preventCancel: false,
+									title: $L("Neato! Not Installed"),
+									message: $L("Neato! is not installed. Would you like to download it?"),
+									choices:[
+										{label:$L('Yes'), value:"yes", type:'affirmative'},
+										{label:$L('No'), value:"no", type:'dismissal'}
+									]
+								});
+							}.bind(this)
+						})
+					}catch(e){
+						Mojo.Log.error("Neato Patch Error: " + e);
+					}
+					break;
 
 				case MenuData.ApplicationMenu.SharePage.command:
 					this._shareUrl(this.currentUrl, this._getDisplayTitle(), true /*capture*/);
