diff --git a/usr/palm/applications/com.palm.app.browser/app/controllers/page-applicationmenu.js b/usr/palm/applications/com.palm.app.browser/app/controllers/page-applicationmenu.js
index 9030f4d..0776df6 100644
--- a/usr/palm/applications/com.palm.app.browser/app/controllers/page-applicationmenu.js
+++ b/usr/palm/applications/com.palm.app.browser/app/controllers/page-applicationmenu.js
@@ -17,6 +17,30 @@ MenuData.ApplicationMenu = {
 		checkEnabled: true
 	},
 	
+	ShortenURL: {
+		label: $L('Shorten URL'),
+		command: 'shorten-url-cmd',
+		checkEnabled: true
+	},
+	
+	TweetViaBadKitty: {
+		label: $L('Tweet Link'),
+		command: 'add-badkitty-cmd',
+		checkEnabled: true
+	},
+	
+	AddToRelego: {
+		label: $L('Add To Relego'),
+		command: 'add-relego-cmd',
+		checkEnabled: true
+	},
+
+	AddToSpareTime: {
+		label: $L('Add To Spare Time'),
+		command: 'add-sparetime-cmd',
+		checkEnabled: true
+	},
+			
 	AddToLauncher: {
 		label: $L('Add to Launcher'),
 		command: 'add-launch-icon-cmd',
diff --git a/usr/palm/applications/com.palm.app.browser/app/controllers/page-assistant.js b/usr/palm/applications/com.palm.app.browser/app/controllers/page-assistant.js
index 09c898f..df10bc7 100644
--- a/usr/palm/applications/com.palm.app.browser/app/controllers/page-assistant.js
+++ b/usr/palm/applications/com.palm.app.browser/app/controllers/page-assistant.js
@@ -786,16 +786,23 @@ PageAssistant.prototype._setupMenus = function() {
 		this.appMenuModel = {
 			visible: true,
 			items: [
-				MenuData.ApplicationMenu.NewCard,
-				MenuData.ApplicationMenu.AddBookmark,
+				{label:$L('New Card'),     shortcut:'n', command:'new-page-cmd'},
+				{label:$L('Add Bookmark'),     shortcut:'m', command:'add-bookmark-cmd'},
 				{
 					label: $L("Page"),
 					items: [
-						MenuData.ApplicationMenu.AddToLauncher,
-						MenuData.ApplicationMenu.SharePage]
+						MenuData.ApplicationMenu.ShortenURL,
+						MenuData.ApplicationMenu.TweetViaBadKitty,
+						MenuData.ApplicationMenu.AddToRelego,
+						MenuData.ApplicationMenu.AddToSpareTime,
+						{label:$L('Add To Launcher'),     shortcut:'l', command:'add-launch-icon-cmd'},
+						{label:$L('Share Page'),     shortcut:'s', command:'share-page-cmd'},
+                        {label:$L('Refresh Page'),     shortcut:'r', command:'refresh'},
+                           ]
 				},
-				MenuData.ApplicationMenu.ShowBookmarks,
-				MenuData.ApplicationMenu.ShowHistory]
+				{label:$L('Bookmarks'),     shortcut:'b', command:'show-bookmarks-cmd'},
+				{label:$L('History'),     shortcut:'h', command:'show-history-cmd'},
+                           ]
 		};
 
 		this.controller.setupWidget(Mojo.Menu.appMenu, undefined, this.appMenuModel);
@@ -2119,6 +2126,148 @@ PageAssistant.prototype.handleCommand = function(event) {
 					this._createDefaultBookmarkImages(urlReference);
 					this.showBookmarkDialog(BookmarkDialogAssistant.createLaunchpointTask, urlReference);
 					break;
+					
+				case MenuData.ApplicationMenu.ShortenURL.command:
+					// Set up the url
+						var url = "http://is.gd/api.php?longurl=" + this.currentUrl;
+						var target = this;
+	
+					// Attempt to connect to api
+						var request = new Ajax.Request(url, {
+							method: 'get',
+							evalJSON: 'false',
+							onSuccess: function(response) {
+								// Retrieve the short url and copy it
+  								var short_url = response.responseText;
+								target.controller.stageController.setClipboard(short_url, false);
+		
+								// Set up and show the banner message for success
+								var message = "URL copied to clipboard.";
+								var appController = Mojo.Controller.getAppController();
+								var bannerParams = {soundClass: '', soundFile: '', icon: '', messageText: message};
+								appController.showBanner(bannerParams, {banner: message});
+							},
+							onFailure: function(response) {
+							// What happens when the api call fails
+							}
+						});
+				break;
+				
+				case MenuData.ApplicationMenu.TweetViaBadKitty.command:
+					// Send the URL to Bad Kitty
+					try{
+						this.badkittyappid = "com.superinhuman.badkitty";
+						this.controller.serviceRequest("palm://com.palm.applicationManager", {
+							method: 'open',
+							parameters: {
+								id: this.badkittyappid,
+								params: {action: 'tweet', tweet: this.currentTitle + ' ' + this.currentUrl}
+							},
+							onFailure:function(){
+								//--> How do they not have it installed?
+								this.controller.showAlertDialog({
+									onChoose: function(value){
+										if (value=="yes"){
+											//--> And weee'rrre.. of to see the wizard
+											this.controller.serviceRequest('palm://com.palm.applicationManager', { 
+												method:'open',
+												parameters:{
+													target: "http://developer.palm.com/appredirect/?packageid=" + this.badkittyappid
+												}
+											});
+										}
+									},
+									preventCancel: false,
+									title: $L("Bad Kitty Not Installed"),
+									message: $L("Bad Kitty is not installed. Would you like to download it?"),
+									choices:[
+										{label:$L('Yes'), value:"yes", type:'affirmative'},
+										{label:$L('No'), value:"no", type:'dismissal'}
+									]
+								});
+							}.bind(this)
+						})
+					}catch(e){
+						Mojo.Log.error("Bad Kitty Patch Error: " + e);
+					}
+					break;
+				
+				case MenuData.ApplicationMenu.AddToRelego.command:
+					// Send the URL to Spare Time
+					try{
+						this.relegoappid = "com.webosroundup.relego";
+						this.controller.serviceRequest("palm://com.palm.applicationManager", {
+							method: 'open',
+							parameters: {
+								id: this.relegoappid,
+								params: {action: 'addtorelego', url: this.currentUrl, title: this.currentTitle}
+							},
+							onFailure:function(){
+								//--> How do they not have it installed?
+								this.controller.showAlertDialog({
+									onChoose: function(value){
+										if (value=="yes"){
+											//--> And weee'rrre.. of to see the wizard
+											this.controller.serviceRequest('palm://com.palm.applicationManager', {
+												method:'open',
+												parameters:{
+													target: "http://developer.palm.com/appredirect/?packageid=" + this.relegoappid
+												}
+											});
+										}
+									},
+									preventCancel: false,
+									title: $L("Relego Not Installed"),
+									message: $L("Relego is not installed. Would you like to download it? It's Free!"),
+									choices:[
+										{label:$L('Yes'), value:"yes", type:'affirmative'},
+										{label:$L('No'), value:"no", type:'dismissal'}
+									]
+								});
+							}.bind(this)
+						})
+					}catch(e){
+						Mojo.Log.error("Relego Patch Error: " + e);
+					}
+					break;
+				case MenuData.ApplicationMenu.AddToSpareTime.command:
+					// Send the URL to Spare Time
+					try{
+						this.sparetimeappid = "com.semicolonapps.sparetime";
+						this.controller.serviceRequest("palm://com.palm.applicationManager", {
+							method: 'open',
+							parameters: {
+								id: this.sparetimeappid,
+								params: {action: "add_url", url: this.currentUrl, title: this.currentTitle}
+							},
+							onFailure:function(){
+								//--> How do they not have it installed?
+								this.controller.showAlertDialog({
+									onChoose: function(value){
+										if (value=="yes"){
+											//--> And weee'rrre.. of to see the wizard
+											this.controller.serviceRequest('palm://com.palm.applicationManager', {
+												method:'open',
+												parameters:{
+													target: "http://developer.palm.com/appredirect/?packageid=" + this.sparetimeappid
+												}
+											});
+										}
+									},
+									preventCancel: false,
+									title: $L("Spare Time Not Installed"),
+									message: $L("Spare Time is not installed. Would you like to download it?"),
+									choices:[
+										{label:$L('Yes'), value:"yes", type:'affirmative'},
+										{label:$L('No'), value:"no", type:'dismissal'}
+									]
+								});
+							}.bind(this)
+						})
+					}catch(e){
+						Mojo.Log.error("Spare Time Patch Error: " + e);
+					}
+					break;
 
 				case MenuData.ApplicationMenu.SharePage.command:
 					this._shareUrl(this.currentUrl, this._getDisplayTitle(), true /*capture*/);
diff --git a/usr/palm/applications/com.palm.app.browser/app/controllers/widget_webview.js b/usr/palm/applications/com.palm.app.browser/app/controllers/widget_webview.js
index 6f1811c..301d890 100644
--- a/usr/palm/applications/com.palm.app.browser/app/controllers/widget_webview.js
+++ b/usr/palm/applications/com.palm.app.browser/app/controllers/widget_webview.js
@@ -537,7 +537,7 @@ Mojo.Widget.WebView = Class.create({
 			// This browser page was purged when the application was not in the foreground.
 			// Now that we're active again let's reconnect and reload the page.
 			Mojo.Log.info("Reconnecting adapter to server.");
-			this._connectAdapterToServer();
+			this._connectAdapterToServer(false);
 		}
 	},
 
