diff --git a/usr/palm/applications/com.palm.app.browser/app/controllers/page-applicationmenu.js b/usr/palm/applications/com.palm.app.browser/app/controllers/page-applicationmenu.js
index 9030f4d..7e4be8d 100644
--- a/usr/palm/applications/com.palm.app.browser/app/controllers/page-applicationmenu.js
+++ b/usr/palm/applications/com.palm.app.browser/app/controllers/page-applicationmenu.js
@@ -17,6 +17,12 @@ MenuData.ApplicationMenu = {
 		checkEnabled: true
 	},
 	
+	ShortenURL: {
+		label: $L('Shorten URL'),
+		command: 'shorten-url-cmd',
+		checkEnabled: true
+	},
+	
 	AddToLauncher: {
 		label: $L('Add to Launcher'),
 		command: 'add-launch-icon-cmd',
diff --git a/usr/palm/applications/com.palm.app.browser/app/controllers/page-assistant.js b/usr/palm/applications/com.palm.app.browser/app/controllers/page-assistant.js
index 09c898f..ac568cb 100644
--- a/usr/palm/applications/com.palm.app.browser/app/controllers/page-assistant.js
+++ b/usr/palm/applications/com.palm.app.browser/app/controllers/page-assistant.js
@@ -791,6 +791,7 @@ PageAssistant.prototype._setupMenus = function() {
 				{
 					label: $L("Page"),
 					items: [
+						MenuData.ApplicationMenu.ShortenURL,
 						MenuData.ApplicationMenu.AddToLauncher,
 						MenuData.ApplicationMenu.SharePage]
 				},
@@ -2119,6 +2120,32 @@ PageAssistant.prototype.handleCommand = function(event) {
 					this._createDefaultBookmarkImages(urlReference);
 					this.showBookmarkDialog(BookmarkDialogAssistant.createLaunchpointTask, urlReference);
 					break;
+					
+				case MenuData.ApplicationMenu.ShortenURL.command:
+					// Set up the url
+						var url = "http://is.gd/api.php?longurl=" + this.currentUrl;
+						var target = this;
+	
+					// Attempt to connect to api
+						var request = new Ajax.Request(url, {
+							method: 'get',
+							evalJSON: 'false',
+							onSuccess: function(response) {
+								// Retrieve the short url and copy it
+  								var short_url = response.responseText;
+								target.controller.stageController.setClipboard(short_url, false);
+		
+								// Set up and show the banner message for success
+								var message = "URL copied to clipboard.";
+								var appController = Mojo.Controller.getAppController();
+								var bannerParams = {soundClass: '', soundFile: '', icon: '', messageText: message};
+								appController.showBanner(bannerParams, {banner: message});
+							},
+							onFailure: function(response) {
+							// What happens when the api call fails
+							}
+						});
+				break;
 
 				case MenuData.ApplicationMenu.SharePage.command:
 					this._shareUrl(this.currentUrl, this._getDisplayTitle(), true /*capture*/);
