diff --git a/usr/palm/frameworks/webview/submission/23.1/concatenated.js b/usr/palm/frameworks/webview/submission/23.1/concatenated.js
index 6cff4d9..9441f3b 100755
--- a/usr/palm/frameworks/webview/submission/23.1/concatenated.js
+++ b/usr/palm/frameworks/webview/submission/23.1/concatenated.js
@@ -615,8 +615,8 @@ Widget.prototype = {
 			Mojo.Event.stopListening(this.adapter, 'keyup', this._keyUpHandlerPluginSpotlight);
 		}
 
-		this._removeElementHighlight();
-		this._disableSelectionMode();
+		//this._removeElementHighlight();
+		//this._disableSelectionMode();
 
 		Mojo.Event.stopListening(this.controller.document, 'gesturestart', this._gestureStartHandler, false);
 		Mojo.Event.stopListening(this.controller.document, 'gesturechange', this._gestureChangeHandler, false);
@@ -631,7 +631,7 @@ Widget.prototype = {
 				// Setting focus to false sends a message to the BrowserServer which in turn disables
 				// active content (like Flash & audio) on the page so that they don't consume resources
 				// when the scene is not active.
-				this.adapter.pageFocused(false);
+				this.adapter.pageFocused(this.hasFocus);
 			}
 			catch (e) {
 				Mojo.Log.logException(e, '_sceneDeactivate');
@@ -948,7 +948,7 @@ Widget.prototype = {
 		{
 			this.newUrlClicked=false;
 		}
-		this.hasFocus = false;
+		this.hasFocus = true;
 
 		if (this.useAdapter) {
 			try {
