diff --git a/etc/palm/luna.conf b/etc/palm/luna.conf
index 573777f..c49095a 100644
--- a/etc/palm/luna.conf
+++ b/etc/palm/luna.conf
@@ -44,7 +44,7 @@ ThrobberBrightnessInDark=50
 
 [Memory]
 CardLimit=-1
-AppsToAllowInLowMemory=com.palm.app.phone;com.palm.app.contacts;com.palm.app.messaging
+AppsToAllowInLowMemory=com.palm.app.phone;com.palm.app.contacts;com.palm.app.messaging,com.palm.app.browser
 
 [TouchEvents]
 TapRadiusSquared=625
