diff --git a/usr/palm/applications/com.palm.app.browser/app/controllers/page-assistant.js b/usr/palm/applications/com.palm.app.browser/app/controllers/page-assistant.js
index 6e8efd0..ffe3b22 100644
--- a/usr/palm/applications/com.palm.app.browser/app/controllers/page-assistant.js
+++ b/usr/palm/applications/com.palm.app.browser/app/controllers/page-assistant.js
@@ -1947,6 +1947,23 @@ PageAssistant.prototype.moved = function(scrollState) {
 	var pos = this.controller.getSceneScroller().mojo.getScrollPosition();
 	if (pos.top > 0) {
 		
+		var mHeight = this.controller.getSceneScroller().scrollHeight - this.controller.window.innerHeight;
+		var mWidth = this.controller.getSceneScroller().scrollWidth - this.controller.window.innerWidth;
+		var curHeight = pos.top;
+		var curWidth = pos.left;
+		window.clearTimeout(this.scrollbarTimer);
+		var scrollerHide = function(){
+			this.controller.get('scrollbarbg').setOpacity(0.0);
+			this.controller.get('scrollbarhbg').setOpacity(0.0);
+		};
+		this.scrollbarTimer = window.setTimeout(scrollerHide.bind(this),2000);
+		this.controller.get('scrollbarbg').setOpacity(1);
+		this.controller.get('scrollbarhbg').setOpacity(1);
+		var percent = Math.round((curHeight / mHeight) * 100);
+		var percenth = Math.round((curWidth / mWidth) * 100);
+		this.controller.get('scrollbar').style.top = percent + '%';
+		this.controller.get('scrollbarh').style.left = percenth + '%';
+		
 		// We have moved the scene above (0,0) BUT we only hide the
 		// URL bar IF if doesn't have focus.
 		if (!this._addressBar.hasFocus()) {
diff --git a/usr/palm/applications/com.palm.app.browser/app/views/page/page-scene.html b/usr/palm/applications/com.palm.app.browser/app/views/page/page-scene.html
index d7ab020..41c83e0 100644
--- a/usr/palm/applications/com.palm.app.browser/app/views/page/page-scene.html
+++ b/usr/palm/applications/com.palm.app.browser/app/views/page/page-scene.html
@@ -2,6 +2,15 @@
 	<div id="view_menu_bkgnd_content_nochrome"></div>
 </div>
 
+<div id='scrollbarbg' style='position: fixed; z-index: 999; background-color: #000000; width: 2px; height: 90%; top: 5%; right: 0px; opacity: 0;'>
+	<div id='scrollbar' style='position: absolute; z-index: 1000; background-color: white; width: 2px; height: 8px; top: 0%; left: 0px;'>
+	</div>
+</div>
+<div id='scrollbarhbg' style='position: fixed; z-index: 999; background-color: #000000; width: 90%; height: 2px; bottom: 0px; left: 5%; opacity: 0;'>
+	<div id='scrollbarh' style='position: absolute; z-index: 1000; background-color: white; width: 8px; height: 2px; left: 0%;'>
+	</div>
+</div>
+
 <div id='web_view_panel' style="min-height:481px;">
 <div id="web_view" x-mojo-element="WebView"></div>
 </div>
