diff --git a/usr/palm/applications/com.palm.app.browser/app/controllers/page-assistant.js b/usr/palm/applications/com.palm.app.browser/app/controllers/page-assistant.js
index 09c898f..78a45bd 100644
--- a/usr/palm/applications/com.palm.app.browser/app/controllers/page-assistant.js
+++ b/usr/palm/applications/com.palm.app.browser/app/controllers/page-assistant.js
@@ -624,6 +624,7 @@ PageAssistant.prototype._onWebViewSingleTap = function(event) {
 		if (event.altKey) {
 
 			var popupItems = [
+			    {label: $L('Open URL'), command:'openUrl'},
 				{label: $L('Open In New Card'), command:'openNew'},
 				{label: $L('Share Link'), command:'shareUrl'},
 				{label: $L('Copy URL'), command:'copyUrl'},
@@ -669,6 +670,9 @@ PageAssistant.prototype._onWebViewSingleTap = function(event) {
 			var popupSelectFunc = function(value) {
 				selectedCommand = value;
 				switch (value) {
+					case 'openUrl':
+						this.openUrl(urlInfo.url);
+						break;				
 					case 'openNew':
 						this._newBrowserPage(urlInfo.url);
 						break;
@@ -695,6 +699,7 @@ PageAssistant.prototype._onWebViewSingleTap = function(event) {
 				imageInfo = response;
 				var usedItems = [];
 				if (urlInfo.url) {
+				    usedItems.push( findItem('openUrl') );
 					usedItems.push( findItem('openNew') );
 					usedItems.push( findItem('shareUrl') );
 					usedItems.push( findItem('copyUrl') );
