diff --git a/usr/palm/applications/com.palm.app.browser/app/controllers/page-applicationmenu.js b/usr/palm/applications/com.palm.app.browser/app/controllers/page-applicationmenu.js
index 9030f4d..e0b6c0a 100644
--- a/usr/palm/applications/com.palm.app.browser/app/controllers/page-applicationmenu.js
+++ b/usr/palm/applications/com.palm.app.browser/app/controllers/page-applicationmenu.js
@@ -17,6 +17,12 @@ MenuData.ApplicationMenu = {
 		checkEnabled: true
 	},
 	
+	AddToSpareTime: {
+	  label: $L('Add to Spare Time'),
+	  command: 'add-to-spare-time-cmd',
+	  checkEnabled: true
+	},
+
 	AddToLauncher: {
 		label: $L('Add to Launcher'),
 		command: 'add-launch-icon-cmd',
diff --git a/usr/palm/applications/com.palm.app.browser/app/controllers/page-assistant.js b/usr/palm/applications/com.palm.app.browser/app/controllers/page-assistant.js
index 09c898f..4b708fe 100644
--- a/usr/palm/applications/com.palm.app.browser/app/controllers/page-assistant.js
+++ b/usr/palm/applications/com.palm.app.browser/app/controllers/page-assistant.js
@@ -788,6 +788,7 @@ PageAssistant.prototype._setupMenus = function() {
 			items: [
 				MenuData.ApplicationMenu.NewCard,
 				MenuData.ApplicationMenu.AddBookmark,
+				MenuData.ApplicationMenu.AddToSpareTime,
 				{
 					label: $L("Page"),
 					items: [
@@ -2113,6 +2114,39 @@ PageAssistant.prototype.handleCommand = function(event) {
 					this.showBookmarkDialog(BookmarkDialogAssistant.createBookmarkTask, urlReference);
 					break;
 
+        case MenuData.ApplicationMenu.AddToSpareTime.command:
+          this.controller.serviceRequest("palm://com.palm.applicationManager", {
+            method: "open",
+
+            parameters: {
+              id: "com.semicolonapps.sparetime",
+              params: {action: "add_url", url: this.currentUrl, title: this.currentTitle}
+            },
+
+            onFailure: function() {
+              this.controller.showAlertDialog({
+                title: $L("Spare Time Not Installed"),
+                message: $L("Spare Time is not installed. Would you like to purchase it?"),
+
+                choices:[
+                  {label:$L("Yes"), value:"yes", type:"affirmative"},
+                  {label:$L("No"), value:"no", type:"dismissal"}
+                ],
+
+                onChoose: function(value){
+                  if("yes" == value){
+                    this.controller.serviceRequest("palm://com.palm.applicationManager", {
+                      method:"open",
+                      parameters:{target: "http://developer.palm.com/appredirect/?packageid=com.semicolonapps.sparetime"}
+                    });
+                  }
+                }
+              });
+            }.bind(this)
+          });
+
+          break;
+
 				case MenuData.ApplicationMenu.AddToLauncher.command:
 					// Bookmarks can deal with one that has no title, but not the launcher
 					urlReference = new UrlReference(this.currentUrl, this._getDisplayTitle(), new Date());
