diff --git a/usr/palm/applications/com.palm.app.browser/source/Preferences.js b/usr/palm/applications/com.palm.app.browser/source/Preferences.js
index 5e59b86..868dfc2 100644
--- a/usr/palm/applications/com.palm.app.browser/source/Preferences.js
+++ b/usr/palm/applications/com.palm.app.browser/source/Preferences.js
@@ -36,7 +36,6 @@ enyo.kind({
 						{kind: "ToggleButton", name: "flashplugins", onChange: "togglePreferenceClick", preference: "flashplugins", type: "System"}
 					]},
 				]},
-				{kind: "Button", caption: $L("Clear Bookmarks"), onclick: "promptButtonClick", dialog: "clearBookmarksPrompt"},
 				{kind: "Button", caption: $L("Clear History"), onclick: "promptButtonClick", dialog: "clearHistoryPrompt"},
 				{kind: "Button", caption: $L("Clear Cookies"), onclick: "promptButtonClick", dialog: "clearCookiesPrompt"},
 				{kind: "Button", caption: $L("Clear Cache"), onclick: "promptButtonClick", dialog: "clearCachePrompt"},
