diff --git a/etc/palm/browser-app.conf b/etc/palm/browser-app.conf
index d2bf8a4..e0dcf7f 100755
--- a/etc/palm/browser-app.conf
+++ b/etc/palm/browser-app.conf
@@ -71,17 +71,17 @@ StackLimit=1M
 
 [Fonts]
 
-MinFontSize=12
+MinFontSize=8
 MinLogicalFontSize=8
-DefaultFontSize=21
-DefaultFixedFontSize=21
+DefaultFontSize=10
+DefaultFixedFontSize=10
 
 DefaultSerifFontFamilyName=Times New Roman
 DefaultSansSerifFontFamilyName=Prelude
 DefaultFixedFontFamilyName=Courier New
 DefaultStandardFontFamilyName=Prelude
 
-MinTextAutoSizeFontSize=27
+MinTextAutoSizeFontSize=18
 
 [SmartText]
 TermsFile=/tmp/aaa
