diff --git a/usr/palm/applications/com.palm.app.browser/app/controllers/widget_webview.js b/usr/palm/applications/com.palm.app.browser/app/controllers/widget_webview.js
index 6f1811c..d721d21 100644
--- a/usr/palm/applications/com.palm.app.browser/app/controllers/widget_webview.js
+++ b/usr/palm/applications/com.palm.app.browser/app/controllers/widget_webview.js
@@ -1542,7 +1542,7 @@ Mojo.Widget.WebView = Class.create({
 		}
 		else {
 			if (this._serverConnectTimer === undefined) {
-				this._serverConnectTimer = this.controller.window.setTimeout(this._connectAdapterToServer, 3000);
+				this._serverConnectTimer = this.controller.window.setTimeout(this._connectAdapterToServer, false);
 			}
 		}
 	},
