diff --git a/usr/palm/applications/com.palm.app.browser/app/controllers/page-controls.js b/usr/palm/applications/com.palm.app.browser/app/controllers/page-controls.js
index 484a096..ab2e644 100644
--- a/usr/palm/applications/com.palm.app.browser/app/controllers/page-controls.js
+++ b/usr/palm/applications/com.palm.app.browser/app/controllers/page-controls.js
@@ -26,9 +26,16 @@ PageControls = Class.create({
 			command: MenuData.NavigationMenu.Forward.command
 		};
 
+		this._bookmarksModel = {
+			disabled: false,
+			label: MenuData.NavigationMenu.Bookmarks.label,
+			icon: MenuData.NavigationMenu.Bookmarks.icon,
+			command: MenuData.NavigationMenu.Bookmarks.command
+		};
+
 		this._model = {
 			visible: true,
-			items: [this._backModel, {}, {}]
+			items: [this._backModel, this._bookmarksModel, {}, {}]
 		};
 
 		this._currentPageModel = {};
@@ -64,6 +71,7 @@ PageControls = Class.create({
 		var items = $A();
 		this._backModel.disabled = !back;
 		items.push(this._backModel);
+		items.push(this._bookmarksModel);
 
 		if (forward) {
 			this._forwardModel.disabled = !forward;
diff --git a/usr/palm/applications/com.palm.app.browser/app/controllers/page-navigationmenu.js b/usr/palm/applications/com.palm.app.browser/app/controllers/page-navigationmenu.js
index 0271a5b..0b971fc 100644
--- a/usr/palm/applications/com.palm.app.browser/app/controllers/page-navigationmenu.js
+++ b/usr/palm/applications/com.palm.app.browser/app/controllers/page-navigationmenu.js
@@ -10,6 +10,12 @@ MenuData.NavigationMenu = {
 		command: 'back'
 	},
 	
+	Bookmarks: {
+		label: $L('Bookmarks'),
+		icon: 'bookmarks',
+		command: 'show-bookmarks-cmd'
+	},
+	
 	Forward: {
 		label: $L('Go forward'),
 		icon: 'forward',
