diff --git a/usr/palm/applications/com.palm.app.browser/app/controllers/page-applicationmenu.js b/usr/palm/applications/com.palm.app.browser/app/controllers/page-applicationmenu.js
index 6bf5870..9bf0329 100644
--- a/usr/palm/applications/com.palm.app.browser/app/controllers/page-applicationmenu.js
+++ b/usr/palm/applications/com.palm.app.browser/app/controllers/page-applicationmenu.js
@@ -41,6 +41,11 @@ MenuData.ApplicationMenu = {
 		label: $L('History'),
 		command: 'show-history-cmd',
 		checkEnabled: true
+	},
+	
+	TogglePrivate: {
+		label: $L('Enable Private Browsing'),
+		command: 'enable-private-cmd'
 	}
 };
 
diff --git a/usr/palm/applications/com.palm.app.browser/app/controllers/page-assistant.js b/usr/palm/applications/com.palm.app.browser/app/controllers/page-assistant.js
index 388cffc..8d31617 100644
--- a/usr/palm/applications/com.palm.app.browser/app/controllers/page-assistant.js
+++ b/usr/palm/applications/com.palm.app.browser/app/controllers/page-assistant.js
@@ -799,6 +799,7 @@ PageAssistant.prototype._setupMenus = function() {
 						MenuData.ApplicationMenu.AddToLauncher,
 						MenuData.ApplicationMenu.SharePage]
 				},
+				this.togglePrivate = MenuData.ApplicationMenu.TogglePrivate,
 				MenuData.ApplicationMenu.ShowBookmarks,
 				MenuData.ApplicationMenu.ShowHistory]
 		};
@@ -1066,6 +1067,10 @@ PageAssistant.prototype._updateHistory = function() {
 	if (!this.currentUrl || (this.currentUrl.length === 0)) {
 		return;
 	}
+	
+	if (this.togglePrivate.enabled) {
+		return;
+	}
 
 	var title = this.currentTitle || null;
 	this.historyStore.addHistory(this.currentUrl, title, new Date(),
@@ -2151,6 +2156,24 @@ PageAssistant.prototype.handleCommand = function(event) {
 				case MenuData.ApplicationMenu.ShowHistory.command:
 					this._openHistoryView();
 					break;
+					
+				case 'enable-private-cmd':
+					this.togglePrivate.command = 'disable-private-cmd';
+					this.togglePrivate.label = $L('Disable Private Browsing');
+					
+					this.togglePrivate.enabled = true;
+					
+					this.controller.modelChanged(this.appMenuModel);
+					break;
+					
+				case 'disable-private-cmd':
+					this.togglePrivate.command = 'enable-private-cmd';
+					this.togglePrivate.label = $L('Enable Private Browsing');
+					
+					this.togglePrivate.enabled = false;
+					
+					this.controller.modelChanged(this.appMenuModel);
+					break;
 
 				case MenuData.ApplicationMenu.ShowBookmarks.command:
 					this._openBookmarkView();
diff --git a/usr/palm/applications/com.palm.app.browser/app/controllers/startpage-assistant.js b/usr/palm/applications/com.palm.app.browser/app/controllers/startpage-assistant.js
index f024692..831fa29 100644
--- a/usr/palm/applications/com.palm.app.browser/app/controllers/startpage-assistant.js
+++ b/usr/palm/applications/com.palm.app.browser/app/controllers/startpage-assistant.js
@@ -50,6 +50,7 @@ StartpageAssistant.prototype.setup = function() {
 					MenuData.ApplicationMenu.AddToLauncher,
 					MenuData.ApplicationMenu.SharePage]
 			},
+			this.togglePrivate = MenuData.ApplicationMenu.TogglePrivate,
 			MenuData.ApplicationMenu.ShowBookmarks,
 			MenuData.ApplicationMenu.ShowHistory]
 	};
@@ -367,6 +368,24 @@ StartpageAssistant.prototype.handleCommand = function(event) {
 				case MenuData.ApplicationMenu.ShowHistory.command:
 					this._openHistoryView();
 					break;
+					
+				case 'enable-private-cmd':
+					this.togglePrivate.command = 'disable-private-cmd';
+				        this.togglePrivate.label = $L('Disable Private Browsing');
+				        
+				        this.togglePrivate.enabled = true;
+				        
+				        this.controller.modelChanged(this.appMenuModel);
+				        break;
+				        
+			        case 'disable-private-cmd':
+				        this.togglePrivate.command = 'enable-private-cmd';
+				        this.togglePrivate.label = $L('Enable Private Browsing');
+				        
+				        this.togglePrivate.enabled = false;
+				        
+				        this.controller.modelChanged(this.appMenuModel);
+				        break;
 
 				case MenuData.ApplicationMenu.ShowBookmarks.command:
 					this._openBookmarkView();
