diff --git a/usr/palm/applications/com.palm.app.browser/source/Browser.js b/usr/palm/applications/com.palm.app.browser/source/Browser.js
index 6094e1c..29734b8 100644
--- a/usr/palm/applications/com.palm.app.browser/source/Browser.js
+++ b/usr/palm/applications/com.palm.app.browser/source/Browser.js
@@ -339,6 +339,42 @@ enyo.kind({
 		this.viewCall("saveImageAtPoint", [inPosition.left, inPosition.top, "/tmp",
 			enyo.hitch(this, "finishShareImage", inTapInfo)]);
 	},
+	setPinClick: function(inTapInfo, inPosition){
+		// Basic Setup
+		
+		var PinImageUrl = inTapInfo.imageUrl ;
+		
+		// The original way.
+		// var PinlinkUrl = inTapInfo.linkUrl || this.owner.$.browser.url ;
+		
+		// This seems to mimic the way the pin this bookmark works
+		var PinlinkUrl = this.owner.$.browser.url ;
+		var PinTitle = inTapInfo.title || this.owner.$.browser.title  ;
+		
+		//PinImageUrl = PinImageUrl.replace(/&amp;/g,"&");
+		//PinlinkUrl = PinlinkUrl.replace(/&amp;/g,"&");
+		//PinTitle = PinTitle.replace(/&amp;/g,"&");
+		
+		PinImageUrl = PinImageUrl.replace(/&amp;/g,"%26");
+		PinlinkUrl = PinlinkUrl.replace(/&amp;/g,"%26");
+		PinTitle = PinTitle.replace(/&amp;/g,"%26");
+		
+		// this is the original way
+		// var PinMarklet = $L("http://pinterest.com/pin/create/bookmarklet/?media=")+PinImageUrl+$L("&url=")+PinlinkUrl+$L("&alt=alt&title=")+PinTitle+$L("&is_video=false&");
+		
+		// not sure if this is better for cleaner URL
+		var PinMarklet = $L('http://huaban.com/bookmarklet/?media={$srcImageUrl}&url={$srclinkUrl}&alt={$alt}&title=title&is_video=false&');
+		PinMarklet = enyo.macroize(PinMarklet, {srcImageUrl: PinImageUrl, srclinkUrl: PinlinkUrl, alt: PinTitle});
+		PinMarklet = PinMarklet.replace(/&amp;/g,"&");
+		// enyo.windows.openWindow("index.html", null, {url: PinMarklet.replace(/&amp;/g,"&")});
+		enyo.windows.openWindow("index.html", null, {url: PinMarklet});
+		// enyo.dom.setClipboard(PinMarklet);
+		//var params = {
+		//	url: PinMarklet
+		//	};
+		//	this.$.launchApplicationService.call({id: "com.palm.app.browser", params: params});
+		
+	},
 	setWallpaperClick: function(inTapInfo, inPosition) {
 		this.viewCall("saveImageAtPoint", [inPosition.left, inPosition.top, "/media/internal",
 			enyo.hitch(this, "finishSetWallpaper", inTapInfo)]);
diff --git a/usr/palm/applications/com.palm.app.browser/source/BrowserContextMenu.js b/usr/palm/applications/com.palm.app.browser/source/BrowserContextMenu.js
index b029eba..e0cf69f 100644
--- a/usr/palm/applications/com.palm.app.browser/source/BrowserContextMenu.js
+++ b/usr/palm/applications/com.palm.app.browser/source/BrowserContextMenu.js
@@ -16,7 +16,8 @@ enyo.kind({
 	imageItems: [
 		{caption: $L("Copy To Photos"), value: "copyToPhotosClick"},
 		{caption: $L("Share Image"), value: "shareImageClick"},
-		{caption: $L("Set Wallpaper"), value: "setWallpaperClick"}
+		{caption: $L("Set Wallpaper"), value: "setWallpaperClick"},
+		{caption: $L("采集到花瓣"), value: "setPinClick"}
 	],
 	openAtTap: function(inEvent, inTapInfo) {
 		this.tapPosition = {left: inEvent.pageX, top: inEvent.pageY};
