diff --git a/usr/palm/applications/com.palm.app.browser/app/controllers/page-assistant.js b/usr/palm/applications/com.palm.app.browser/app/controllers/page-assistant.js
index 388cffc..df4d251 100644
--- a/usr/palm/applications/com.palm.app.browser/app/controllers/page-assistant.js
+++ b/usr/palm/applications/com.palm.app.browser/app/controllers/page-assistant.js
@@ -791,16 +791,19 @@ PageAssistant.prototype._setupMenus = function() {
 		this.appMenuModel = {
 			visible: true,
 			items: [
-				MenuData.ApplicationMenu.NewCard,
-				MenuData.ApplicationMenu.AddBookmark,
+				{label:$L('New Card'), shortcut:'n', command:'new-page-cmd'},
+				{label:$L('Add Bookmark'), shortcut:'m', command:'add-bookmark-cmd'},
 				{
 					label: $L("Page"),
 					items: [
-						MenuData.ApplicationMenu.AddToLauncher,
-						MenuData.ApplicationMenu.SharePage]
+						{label:$L('Add To Launcher'), shortcut:'l', command:'add-launch-icon-cmd'},
+						{label:$L('Share Page'), shortcut:'s', command:'share-page-cmd'},
+						{label:$L('Refresh Page'), shortcut:'r', command:'refresh'},
+                    ]
 				},
-				MenuData.ApplicationMenu.ShowBookmarks,
-				MenuData.ApplicationMenu.ShowHistory]
+				{label:$L('Bookmarks'),     shortcut:'b', command:'show-bookmarks-cmd'},
+				{label:$L('History'),     shortcut:'h', command:'show-history-cmd'},
+                           ]
 		};
 
 		this.controller.setupWidget(Mojo.Menu.appMenu, undefined, this.appMenuModel);
