diff --git a/usr/palm/applications/com.palm.app.browser/app/controllers/history-assistant.js b/usr/palm/applications/com.palm.app.browser/app/controllers/history-assistant.js
index 7f6d335..a344d59 100644
--- a/usr/palm/applications/com.palm.app.browser/app/controllers/history-assistant.js
+++ b/usr/palm/applications/com.palm.app.browser/app/controllers/history-assistant.js
@@ -19,13 +19,13 @@ function HistoryAssistant(webPreferences) {
 	this.historyStore = new HistoryStore({});
 }
 
-HistoryAssistant.Menu = {
+//HistoryAssistant.Menu = {
 
-	Clear: {
-		label: $L('Clear History'),
-		command: 'clear-history-command'
-	}
-};
+	//Clear: {
+	//	label: $L('Clear History'),
+	//	command: 'clear-history-command'
+	//}
+//};
 
 HistoryAssistant.prototype.setup = function() {
 	this.historyModel=[];
@@ -37,7 +37,7 @@ HistoryAssistant.prototype.setup = function() {
 			items: [
 				MenuData.ApplicationMenu.ShowBookmarks,
 				MenuData.ApplicationMenu.ShowHistory,
-				HistoryAssistant.Menu.Clear
+				//HistoryAssistant.Menu.Clear
 			]
 		};
 
@@ -105,9 +105,9 @@ HistoryAssistant.prototype.handleCommand = function(event) {
 						this._openBookmarkView();
 						break;
 
-					case HistoryAssistant.Menu.Clear.command:
-						this._clearHistory();
-						break;
+					//case HistoryAssistant.Menu.Clear.command:
+					//	this._clearHistory();
+					//	break;
 
 					case Mojo.Menu.prefsCmd:
 						this._openPreferencesView();
diff --git a/usr/palm/applications/com.palm.app.browser/app/views/preferences/preferences-scene.html b/usr/palm/applications/com.palm.app.browser/app/views/preferences/preferences-scene.html
index ede7991..dbb067e 100644
--- a/usr/palm/applications/com.palm.app.browser/app/views/preferences/preferences-scene.html
+++ b/usr/palm/applications/com.palm.app.browser/app/views/preferences/preferences-scene.html
@@ -50,7 +50,6 @@
             </div>
         </div>
     </div>
-    <div class="palm-button" id="clear_history_button" x-mojo-tap-highlight="momentary" x-mojo-loc=''>Clear History</div>
     <div class="palm-button" id="clear_cookies_button" x-mojo-tap-highlight="momentary" x-mojo-loc=''>Clear Cookies</div>
     <div class="palm-button" id="clear_cache_button" x-mojo-tap-highlight="momentary" x-mojo-loc=''>Clear Cache</div>
 	<div class="scroll-fade-bottom" x-mojo-scroll-fade="bottom"></div>
