diff --git a/usr/palm/applications/com.palm.app.browser/app/controllers/startpage-assistant.js b/usr/palm/applications/com.palm.app.browser/app/controllers/startpage-assistant.js
index 1d890db..b463ac8 100644
--- a/usr/palm/applications/com.palm.app.browser/app/controllers/startpage-assistant.js
+++ b/usr/palm/applications/com.palm.app.browser/app/controllers/startpage-assistant.js
@@ -7,7 +7,7 @@ function StartpageAssistant(params) {
 	this._destinationUrl = params.lastUrl;
 	this._orientation = params.orientation || 'up';
 	this._bookmarks = [];
-	this._maxShowCount = 12;
+	this._maxShowCount = 100;
 	this.store = new BookmarkStore({
 		database: Mojo.Controller.appController.assistant.getDatabase()
 	});
