diff --git a/usr/palm/applications/com.palm.app.browser/app/controllers/page-assistant.js b/usr/palm/applications/com.palm.app.browser/app/controllers/page-assistant.js
index 6e8efd0..420c710 100644
--- a/usr/palm/applications/com.palm.app.browser/app/controllers/page-assistant.js
+++ b/usr/palm/applications/com.palm.app.browser/app/controllers/page-assistant.js
@@ -311,6 +311,10 @@ PageAssistant.prototype.activate = function(message) {
 	try {
 		
 		this._updateOrientation();
+
+		// FullScreen Browser
+        this.controller.enableFullScreenMode(true);		
+		
 		
 		// Update any pending web preferences changes
 		AppAssistant.WebPreferences.activate(this._webView.mojo);
@@ -1954,6 +1958,7 @@ PageAssistant.prototype.moved = function(scrollState) {
 			this._addressBar.hide();
 			this.chrome.hide();
 
+		    this.controller.enableFullScreenMode(true);
 			// Only focus this webview widget if this assistant has
 			// focus. All page-assisants can receive 'moved' events
 			// so we have to be careful when explicitly setting the
@@ -1978,6 +1983,7 @@ PageAssistant.prototype.moved = function(scrollState) {
 		this._addressBar.show();
 		this.chrome.show();
 
+		this.controller.enableFullScreenMode(false);
 	} else {
 		//Mojo.Log.info("-> TOP == 0 Do Nothing.");
 		// Do nothing.
