diff --git a/etc/palm/luna.conf b/etc/palm/luna.conf
index 1ecc945..cdcd55d 100644
--- a/etc/palm/luna.conf
+++ b/etc/palm/luna.conf
@@ -48,7 +48,7 @@ ThrobberHoldDownTimeInMs=2000
 
 [Memory]
 CardLimit=-1
-AppsToAllowInLowMemory=com.palm.app.phone;com.palm.app.contacts;com.palm.app.messaging
+AppsToAllowInLowMemory=com.palm.app.phone;com.palm.app.contacts;com.palm.app.messaging,com.palm.app.browser
 
 [TouchEvents]
 TapRadiusSquared=576
