diff --git a/usr/palm/applications/com.palm.app.browser/app/controllers/preferences-assistant.js b/usr/palm/applications/com.palm.app.browser/app/controllers/preferences-assistant.js
index 866f941..dae7481 100644
--- a/usr/palm/applications/com.palm.app.browser/app/controllers/preferences-assistant.js
+++ b/usr/palm/applications/com.palm.app.browser/app/controllers/preferences-assistant.js
@@ -43,8 +43,6 @@ PreferencesAssistant.prototype.setup = function() {
 			this.setupJavaScriptWidget();
 			this.setupSearchEngineWidget();
 			
-			this.controller.get("flashPref").hide();
-			this.controller.get("autoLoadPref").hide();
 			this.setUpFlashPrefWidget();
 			this.setUpAutoloadPrefWidget();
 
