diff --git a/usr/palm/applications/com.palm.app.browser/source/ActionBar.js b/usr/palm/applications/com.palm.app.browser/source/ActionBar.js
index b680425..4fb2518 100644
--- a/usr/palm/applications/com.palm.app.browser/source/ActionBar.js
+++ b/usr/palm/applications/com.palm.app.browser/source/ActionBar.js
@@ -13,6 +13,7 @@ enyo.kind({
 		searchPreferences: {},
 		defaultSearch: "",
 		progress: 0,
+		private: false,
 		loading: false
 	},
 	events: {
@@ -63,6 +64,7 @@ enyo.kind({
 		this.canGoForwardChanged();
 		this.canShareChanged();
 		this.loadingChanged();
+		this.privateChanged();
 	},
 	urlChanged: function() {
 		this.$.search.setUrl(this.url);
@@ -106,4 +108,11 @@ enyo.kind({
 			}
 		}
 	},
+	privateChanged: function() {
+		if (this.private) {
+			this.applyStyle('background-color', '#8B0000');
+		} else {
+			this.applyStyle('background-color', null);
+		}
+	}
 });
diff --git a/usr/palm/applications/com.palm.app.browser/source/Browser.js b/usr/palm/applications/com.palm.app.browser/source/Browser.js
index f2d5a46..fe17961 100644
--- a/usr/palm/applications/com.palm.app.browser/source/Browser.js
+++ b/usr/palm/applications/com.palm.app.browser/source/Browser.js
@@ -5,10 +5,12 @@ enyo.kind({
 	published: {
 		url: "",
 		searchPreferences: {},
+		private: false,
 		defaultSearch: ""
 	},
 	events: {
 		onPageTitleChanged: "",
+		privateChanged: "",
 		onPageLoadStopped: "",
 		onFileLoad: "",
 		onAddBookmark: "",
@@ -102,7 +104,11 @@ enyo.kind({
 		this.defaultSearchChanged();
 		if (window.PalmSystem) {
 			this.$.view.setIdentifier(enyo.windowParams.webviewId);
+
+			this.private = enyo.windowParams.private || false;
 		}
+		this.privateChanged();
+		this.doPrivateChanged(this.private);
 	},
 	resize: function() {
 		this.$.actionbar.resize();
@@ -162,7 +168,6 @@ enyo.kind({
 		}
 	},
 	urlChanged: function() {
-		this.log(this.url);
 		this.$.view.setUrl(this.url);
 		this.$.actionbar.setLoading(true);
 		this.$.actionbar.setUrl(this.url);
@@ -173,6 +178,9 @@ enyo.kind({
 	defaultSearchChanged: function() {
 		this.$.actionbar.setDefaultSearch(this.defaultSearch);
 	},
+	privateChanged: function() {
+		this.$.actionbar.setPrivate(this.private);
+	},
 	pageTitleChanged: function(inSender, inTitle, inUrl, inBack, inForward) {
 		this.log(inUrl, inTitle, inBack, inForward);
 		this.url = enyo.string.escapeHtml(inUrl);
@@ -291,13 +299,13 @@ enyo.kind({
 		}
 	},
 	newCardClick: function(inTapInfo) {
-		enyo.windows.openWindow("index.html", null, {url: inTapInfo.linkUrl});
+		enyo.windows.openWindow("index.html", null, {url: inTapInfo.linkUrl, private: this.private});
 	},
 	openNewCard: function() {
-		enyo.windows.openWindow("index.html", null, null);
+		enyo.windows.openWindow("index.html", null, {private: this.private});
 	},
 	openNewCardWithIdentifier: function(inSender, inIdentifier) {
-		enyo.windows.openWindow("index.html", null, {webviewId: inIdentifier});
+		enyo.windows.openWindow("index.html", null, {webviewId: inIdentifier, private: this.private});
 	},
 	copyLinkClick: function(inTapInfo) {
 		enyo.dom.setClipboard(inTapInfo.linkUrl);
diff --git a/usr/palm/applications/com.palm.app.browser/source/BrowserApp.js b/usr/palm/applications/com.palm.app.browser/source/BrowserApp.js
index 9a4bf6f..c1a4c6d 100644
--- a/usr/palm/applications/com.palm.app.browser/source/BrowserApp.js
+++ b/usr/palm/applications/com.palm.app.browser/source/BrowserApp.js
@@ -26,6 +26,7 @@ enyo.kind({
 		{name: "launchApplicationService", kind: "PalmService", service: enyo.palmServices.application, method: "open", onFailure: "gotResourceError"},
 		{name: "addToLauncherService", kind: "PalmService", service: enyo.palmServices.application, method: "addLaunchPoint"},
 		{name: "resourceInfoService", kind: "PalmService", service: enyo.palmServices.application, method: "getResourceInfo", onSuccess: "gotResourceInfo", onFailure: "gotResourceError"},
+		{kind: "PalmService", name:"getTweaks", service:"palm://org.webosinternals.tweaks.prefs/", method:"get", onSuccess:"gotTweaks" },
 		{kind: enyo.Pane, flex: 1, lazyViews: [
 			{name: "startPage", kind: "StartPage", 
 				onUrlChange: "processUrlChange",
@@ -35,6 +36,7 @@ enyo.kind({
 			},
 			{name: "browser", kind: "Browser",
 				onPageTitleChanged: "pageTitleChanged",
+				onPrivateChanged: "privateChanged",
 				onPageLoadStopped: "pageLoadStopped",
 				onFileLoad: "handleResource",
 				onAddBookmark: "addBookmark",
@@ -100,6 +102,7 @@ enyo.kind({
 		{kind: "AppMenu", onBeforeOpen: "toggleAppMenuItems", components: [
 			//{caption: $L("Find on Page"), onclick: "showFindOnPage"},
 			{name: "preferencesItem", caption: $L("Preferences"), onclick: "preferencesClick"},
+			{name: "privateMenuItem", caption: $L("Enable Private Browsing"), onclick: "privateClick"},
 			{name: "printMenuItem", caption: $L("Print"), onclick: "printClick"},
 			{caption: $L("Help"), onclick: "helpClick"}
 		]}
@@ -114,6 +117,11 @@ enyo.kind({
 		this.inherited(arguments);
 		this.fetchInitialPreferences();
 		this.log(enyo.json.stringify(enyo.windowParams));
+
+this.log("PRIVATE:", enyo.windowParams.private);
+		if (enyo.windowParams.private !== undefined) {
+			this.privateBrowsing = enyo.windowParams.private;
+		}
 	},
 	applicationRelaunchHandler: function(inSender) {
 		var c = enyo.windows.getActiveWindow();
@@ -147,6 +155,16 @@ enyo.kind({
 		this.showBookmarks();
 		//this.showAppMenu();
 		//this.resize();
+
+this.log("PRIVATE:", this.privateBrowsing);
+		if (this.privateBrowsing !== undefined) {
+			enyo.nextTick(this, this.privateSet(this.privateBrowsing));
+		} else {
+			this.$.getTweaks.call({
+				owner:	"org.webosinternals.patches.browser-private-browsing",
+				keys:	["privateByDefault"]
+			});
+		}
 	},
 	isBrowserShowing: function() {
 		return this.$.pane.getViewName() === "browser";
@@ -222,6 +240,12 @@ enyo.kind({
 		var browser = this.isBrowserShowing();
 		this.$.printMenuItem.setDisabled(!browser || this.isBrowserLoading());
 		this.$.preferencesItem.setDisabled(this.isPreferencesShowing());
+
+		if (this.privateBrowsing) {
+			this.$.privateMenuItem.setCaption($L("Disable Private Browsing"));
+		} else {
+			this.$.privateMenuItem.setCaption($L("Enable Private Browsing"));
+		}
 	},
 	applyPreference: function(inPreference, inValue) {
 		this.log(inPreference, inValue);
@@ -479,6 +503,10 @@ enyo.kind({
 		this.$.launchApplicationService.call({id: inAppId, params: {target: inUrl, mime: inMime}});
 	},
 	updateHistory: function(inTitle, inUrl) {
+		if (this.privateBrowsing) {
+			return;
+		}
+
 		this.$.historyService.call(
 			{query: {where: [{prop:"url",op:"=",val:inUrl}]}},
 			{method: "delByQuery"});
@@ -576,7 +604,7 @@ enyo.kind({
 		this.$.bookmarksService.call(undefined,{method:"delByQuery"});
 	},
 	newCardClick: function() {
-		enyo.windows.openWindow("index.html");
+		enyo.windows.openWindow("index.html", null, {private: this.privateBrowsing});
 	},
 	shareClick: function() {
 		this.$.browser.shareLink(this.url, this.title);
@@ -588,6 +616,26 @@ enyo.kind({
 	preferencesClick: function() {
 		this.gotoView("preferences");
 	},
+	privateSet: function(private) {
+		this.privateBrowsing = private;
+
+		if (this.$.browser) {
+			this.$.browser.setPrivate(this.privateBrowsing);
+		}
+		if (this.$.startPage) {
+			this.$.startPage.setPrivate(this.privateBrowsing);
+		}
+	},
+	privateChanged: function() {
+		this.privateSet(this.$.browser.getPrivate());
+	},
+	privateClick: function() {
+		this.privateSet(!this.privateBrowsing);
+	},
+	gotTweaks: function(inSender, inResponse) {
+this.log("PRIVATE:", inResponse);
+		enyo.nextTick(this, this.privateSet(inResponse.privateByDefault));
+	},
 	helpClick: function() {
 		this.$.launchApplicationService.call({params: {target: "http://help.palm.com/web/index.html"}, id: "com.palm.app.help"});
 	},
diff --git a/usr/palm/applications/com.palm.app.browser/source/StartPage.js b/usr/palm/applications/com.palm.app.browser/source/StartPage.js
index 8252498..8127c50 100644
--- a/usr/palm/applications/com.palm.app.browser/source/StartPage.js
+++ b/usr/palm/applications/com.palm.app.browser/source/StartPage.js
@@ -5,6 +5,7 @@ enyo.kind({
 	published: {
 		url: "",
 		searchPreferences: {},
+		private: false,
 		defaultSearch: ""
 	},
 	events: {
@@ -23,6 +24,7 @@ enyo.kind({
 		this.inherited(arguments);
 		this.searchPreferencesChanged();
 		this.defaultSearchChanged();
+		this.privateChanged();
 	},
 	addressSelect: function(inSender, inUrl) {
 		this.doUrlChange(inUrl);
@@ -45,6 +47,9 @@ enyo.kind({
 	defaultSearchChanged: function() {
 		this.$.actionbar.setDefaultSearch(this.defaultSearch);
 	},
+	privateChanged: function() {
+		this.$.actionbar.setPrivate(this.private);
+	},
 	//* @public
 	resize: function() {
 		this.$.actionbar.resize();
diff --git a/usr/palm/applications/com.palm.app.browser/source/URLSearch.js b/usr/palm/applications/com.palm.app.browser/source/URLSearch.js
index af5952e..eb9e385 100644
--- a/usr/palm/applications/com.palm.app.browser/source/URLSearch.js
+++ b/usr/palm/applications/com.palm.app.browser/source/URLSearch.js
@@ -46,7 +46,11 @@ enyo.kind({
 	defaultSearchPreferences: [{
 		title: "Google",
 		//url: "http://www.google.com/m/search?client=ms-palm-webOS&channel=bm&q={$query}",
-		url: "http://www.google.com/search?q={$query}",
+		url: "https://encrypted.google.com/search?q={$query}&safe=off&pws=0",
+		icon: "list-icon-google.png"
+	}, {
+		title: "Google Secure",
+		url: "https://encrypted.google.com/search?q={$query}&safe=off&pws=0",
 		icon: "list-icon-google.png"
 	}, {
 		title: "Wikipedia",
