diff --git a/usr/palm/applications/com.palm.app.browser/app/controllers/page-assistant.js b/usr/palm/applications/com.palm.app.browser/app/controllers/page-assistant.js
index 388cffc..3cb6850 100644
--- a/usr/palm/applications/com.palm.app.browser/app/controllers/page-assistant.js
+++ b/usr/palm/applications/com.palm.app.browser/app/controllers/page-assistant.js
@@ -1938,6 +1938,22 @@ PageAssistant.prototype.moved = function(scrollState) {
 	// gets cancelled.
 	var pos = this.controller.getSceneScroller().mojo.getScrollPosition();
 	if (pos.top > 0) {
+		var mHeight = this.controller.getSceneScroller().scrollHeight - this.controller.window.innerHeight;
+		var mWidth = this.controller.getSceneScroller().scrollWidth - this.controller.window.innerWidth;
+		var curHeight = pos.top;
+		var curWidth = pos.left;
+		window.clearTimeout(this.scrollbarTimer);
+		var scrollerHide = function(){
+			this.controller.get('scrollbarbg').setOpacity(0.0);
+			this.controller.get('scrollbarhbg').setOpacity(0.0);
+		};
+		this.scrollbarTimer = window.setTimeout(scrollerHide.bind(this),2000);
+		this.controller.get('scrollbarbg').setOpacity(0.7);
+		this.controller.get('scrollbarhbg').setOpacity(0.7);
+		var percent = Math.round((curHeight / mHeight) * 100);
+		var percenth = Math.round((curWidth / mWidth) * 100);
+		this.controller.get('scrollbar').style.top = percent + '%';
+		this.controller.get('scrollbarh').style.left = percenth + '%';		
 		
 		// We have moved the scene above (0,0) BUT we only hide the
 		// URL bar IF if doesn't have focus.
diff --git a/usr/palm/applications/com.palm.app.browser/app/views/page/page-scene.html b/usr/palm/applications/com.palm.app.browser/app/views/page/page-scene.html
index d7ab020..3ae3875 100644
--- a/usr/palm/applications/com.palm.app.browser/app/views/page/page-scene.html
+++ b/usr/palm/applications/com.palm.app.browser/app/views/page/page-scene.html
@@ -2,6 +2,17 @@
 	<div id="view_menu_bkgnd_content_nochrome"></div>
 </div>
 
+<div id='scrollbarbg' style='position: fixed; z-index: 999; background-color: #2e2e2e; width: 5px; height: 80%; top: 5%; right: 6px; opacity: 0.0; -webkit-border-radius: 2px;'>
+<div id='scrollbar' style='position: absolute; z-index: 1000; background-color: #FFCC66; width: 5px; height: 8px; top: 0%; left: 0px; -webkit-border-radius: 2px;'>
+ 
+</div>
+</div>
+<div id='scrollbarhbg' style='position: fixed; z-index: 999; background-color: #2e2e2e; width: 60%; height: 5px; bottom: 6px; left: 20%; opacity: 0.0; -webkit-border-radius: 2px;'>
+<div id='scrollbarh' style='position: absolute; z-index: 1000; background-color: #FFCC66; width: 8px; height: 5px; left: 0%; -webkit-border-radius: 2px;'>
+ 
+</div>
+</div>
+
 <div id='web_view_panel' style="min-height:481px;">
 <div id="web_view" x-mojo-element="WebView"></div>
 </div>
