diff --git a/usr/palm/applications/com.palm.app.browser/app/controllers/page-assistant.js b/usr/palm/applications/com.palm.app.browser/app/controllers/page-assistant.js
index 6e8efd0..921480a 100644
--- a/usr/palm/applications/com.palm.app.browser/app/controllers/page-assistant.js
+++ b/usr/palm/applications/com.palm.app.browser/app/controllers/page-assistant.js
@@ -805,6 +805,7 @@ PageAssistant.prototype._setupMenus = function() {
 					label: $L("Page"),
 					items: [
 						MenuData.ApplicationMenu.AddToLauncher,
+						{label:"View Source", command:"view-source-cmd"},
 						MenuData.ApplicationMenu.SharePage]
 				},
 				MenuData.ApplicationMenu.ShowBookmarks,
@@ -2181,6 +2182,18 @@ PageAssistant.prototype.handleCommand = function(event) {
 					this.showBookmarkDialog(BookmarkDialogAssistant.createLaunchpointTask, urlReference);
 					break;
 
+				case "view-source-cmd":
+					this.controller.serviceRequest("palm://com.palm.applicationManager", {
+						method: "open",
+						parameters: {
+							id: "ca.canucksoftware.internalz",
+							params: {
+								target: "view-source://" + this.currentUrl
+							}
+						}
+					});
+					break;
+					
 				case MenuData.ApplicationMenu.SharePage.command:
 					this._shareUrl(this.currentUrl, this._getDisplayTitle(), true /*capture*/);
 					break;
