diff --git a/usr/palm/applications/com.palm.app.browser/source/HistoryList.js b/usr/palm/applications/com.palm.app.browser/source/HistoryList.js
index 1284171..2dbdb97 100644
--- a/usr/palm/applications/com.palm.app.browser/source/HistoryList.js
+++ b/usr/palm/applications/com.palm.app.browser/source/HistoryList.js
@@ -12,7 +12,7 @@ enyo.kind({
 		{name: "historyService", kind: "DbService", dbKind: "com.palm.browserhistory:1", reCallWatches: true, method: "find", onSuccess: "gotHistoryData", subscribe: true, onWatch:"refreshList"},
 		{name:"list", kind:"DbList", flex:1, onQuery:"historyQuery", onSetupRow: "listSetupRow", desc:true, components: [
 			{name: "divider", kind: "Divider", showing: false},
-			{name: "item", kind: "SwipeableItem", layoutKind: "HFlexLayout", tapHighlight: true, onclick: "itemClick", onConfirm: "deleteItem", components: [
+			{name: "item", layoutKind: "HFlexLayout", tapHighlight: true, onclick: "itemClick", onConfirm: "deleteItem", components: [
 				{name: "icon", kind: "Image", src: "images/header-icon-history.png"},
 				{flex: 1, components: [
 					{name: "title", className: "url-item-title enyo-text-ellipsis"},
diff --git a/usr/palm/applications/com.palm.app.browser/source/Preferences.js b/usr/palm/applications/com.palm.app.browser/source/Preferences.js
index 5e59b86..ac042c3 100644
--- a/usr/palm/applications/com.palm.app.browser/source/Preferences.js
+++ b/usr/palm/applications/com.palm.app.browser/source/Preferences.js
@@ -37,7 +37,7 @@ enyo.kind({
 					]},
 				]},
 				{kind: "Button", caption: $L("Clear Bookmarks"), onclick: "promptButtonClick", dialog: "clearBookmarksPrompt"},
-				{kind: "Button", caption: $L("Clear History"), onclick: "promptButtonClick", dialog: "clearHistoryPrompt"},
+		
 				{kind: "Button", caption: $L("Clear Cookies"), onclick: "promptButtonClick", dialog: "clearCookiesPrompt"},
 				{kind: "Button", caption: $L("Clear Cache"), onclick: "promptButtonClick", dialog: "clearCachePrompt"},
 				/*
