diff --git a/usr/palm/applications/com.palm.app.browser/app/controllers/bookmarks-assistant.js b/usr/palm/applications/com.palm.app.browser/app/controllers/bookmarks-assistant.js
index 99a2286..06481b3 100644
--- a/usr/palm/applications/com.palm.app.browser/app/controllers/bookmarks-assistant.js
+++ b/usr/palm/applications/com.palm.app.browser/app/controllers/bookmarks-assistant.js
@@ -31,7 +31,7 @@ BookmarksAssistant.prototype.setup = function() {
 						 listTemplate:'bookmarks/bookmarks-container',
 						 itemsCallback:this._retrieveItemsCallback.bind(this),
 						 swipeToDelete:true,
-						 reorderable:true});
+						 reorderable:true}, this.bookmarksModel);
 		this.bookmarksListWidget = this.controller.get('bookmarksList');
 
 		this._onListSelectionHandler = this._onListSelection.bind(this);
@@ -179,6 +179,8 @@ BookmarksAssistant.prototype._onBookmarksReadSuccess = function(bookmarkFolder)
 		}
 
 		this.bookmarksListWidget.mojo.setLength(this.bookmarksModel.length);
+		this.controller.setWidgetModel('bookmarksList', this.bookmarksModel)
+		this.controller.modelChanged(this.bookmarksModel, this);
 	}
 	finally {
 		this.usingBookmarksDb = false;
