diff --git a/etc/palm/browser-app.conf b/etc/palm/browser-app.conf
index 7da4e1e..b37c3ec 100755
--- a/etc/palm/browser-app.conf
+++ b/etc/palm/browser-app.conf
@@ -27,6 +27,7 @@ UserAgentIndex=0
 FastLocalLoad=true
 DatabaseQuota=5242880
 EnableFileAccessWhitelist=true
+MaxImagePixelCount=10485760
 
 # The 'EnableFitWidth' parameter lets us control the default behavior of ScrollView.
 EnableFitWidth=false
