diff --git a/usr/palm/applications/com.palm.app.browser/source/BrowserApp.js b/usr/palm/applications/com.palm.app.browser/source/BrowserApp.js
index 9a4bf6f..1668be4 100644
--- a/usr/palm/applications/com.palm.app.browser/source/BrowserApp.js
+++ b/usr/palm/applications/com.palm.app.browser/source/BrowserApp.js
@@ -143,6 +143,8 @@ enyo.kind({
 			this.$.pane.selectViewByName("browser");
 		} else {
 			this.$.pane.selectViewByName("startPage");
+                        this.$.toaster.open();
+                        this.$.drawerPane.selectView(this.$.drawerPane.getView());
 		}
 		this.showBookmarks();
 		//this.showAppMenu();
