diff --git a/usr/palm/applications/com.palm.app.browser/app/controllers/global_code.js b/usr/palm/applications/com.palm.app.browser/app/controllers/global_code.js
index b2c8848..08d56ac 100644
--- a/usr/palm/applications/com.palm.app.browser/app/controllers/global_code.js
+++ b/usr/palm/applications/com.palm.app.browser/app/controllers/global_code.js
@@ -1101,10 +1101,13 @@ HistoryAssistant.prototype.setup = function() {
 		this.controller.setupWidget('historyList', {
 			itemTemplate:'history/history-entry',
 			listTemplate:'history/history-container',
-			itemsCallback:this._itemsCallback.bind(this)
+			itemsCallback:this._itemsCallback.bind(this),
+      swipeToDelete:true,
+      autoconfirmDelete:true
 		});
 
 		this._onListSelectionHandler = this._onListSelection.bindAsEventListener(this);
+    this._onListDeleteHandler = this._onListDelete.bindAsEventListener(this);
 		this._historyListWidget = this.controller.get('historyList');
 
 		var appMenuModel = {
@@ -1130,6 +1133,7 @@ HistoryAssistant.prototype.cleanup = function() {
 HistoryAssistant.prototype.activate = function() {
 
 	this._historyListWidget.addEventListener(Mojo.Event.listTap, this._onListSelectionHandler);
+	this._historyListWidget.addEventListener(Mojo.Event.listDelete, this._onListDeleteHandler);
 
 	// On an activation for a re-render of the contents.
 	var len = this._historyListWidget.mojo.getLength();	
@@ -1139,6 +1143,7 @@ HistoryAssistant.prototype.activate = function() {
 HistoryAssistant.prototype.deactivate = function() {
 
 	Mojo.Event.stopListening(this._historyListWidget, Mojo.Event.listTap, this._onListSelectionHandler);
+	Mojo.Event.stopListening(this._historyListWidget, Mojo.Event.listDelete, this._onListDeleteHandler);
 };
 
 /**
@@ -1218,6 +1223,10 @@ HistoryAssistant.prototype._onListSelection = function(event) {
 	});
 };
 
+HistoryAssistant.prototype._onListDelete = function(event) {
+	this.historyStore.deleteHistoryEntry(event.item.url, function() {}, function() {});
+};
+
 /**
  * handle a menu command.
  */
@@ -7349,6 +7358,15 @@ function HistoryStore(options, onSuccess, onFailure) {
 	}
 }
 
+HistoryStore.prototype._deleteHistoryEntry = function(url, onSuccess, onFailure, transaction) {
+	this._executeSql(transaction, onSuccess, onFailure,
+		"DELETE FROM 'history' WHERE url = ?", [url] );
+};
+
+HistoryStore.prototype.deleteHistoryEntry = function(url, onSuccess, onFailure) {
+        this.database.transaction(this._deleteHistoryEntry.bind(this, url, onSuccess, onFailure));
+};
+
 /** @private */
 HistoryStore.prototype._addHistoryEntry = function(url, title, date, onSuccess, onFailure, transaction) {
 
