Index: /usr/palm/applications/com.palm.app.browser/app/controllers/app-assistant.js
===================================================================
--- /usr/palm/applications/com.palm.app.browser/app/controllers/app-assistant.js
+++ /usr/palm/applications/com.palm.app.browser/app/controllers/app-assistant.js
@@ -51,8 +51,50 @@
 AppAssistant.prototype.setup = function() {
 
+//BEGIN Slider Orientation Patch block 1 of 3
+	AppAssistant.sliderPos='up';
+	AppAssistant.KTorient='up';
+	AppAssistant.KTstagecount=0;
+	this._initSlider();
+//END Slider Orientation Patch block 1 of 3
 	AppAssistant.Network.observe(function(response){Mojo.Log.info("@@@ Listening to network events: online=", response.isInternetConnectionAvailable);});
 };
 
+//BEGIN Slider Orientation Patch block 2 of 3
+AppAssistant.prototype._initSlider = function() {
+  	var request = new Mojo.Service.Request('palm://com.palm.keys/switches', {
+            method: 'status',
+            parameters: {"get" : "slider"},
+			onFailure: function() {
+				},
+			onSuccess: this._onSliderEvent.bind(this)
+    		});
+	this.sliderSubscriptionx = this.sliderSubscribe(this._onSliderEvent.bind(this));
+};
+
+AppAssistant.prototype._onSliderEvent = function(response) {
+	if (!(response && response.key && response.state)) return;
+	
+	if (response.key == "slider") {
+		AppAssistant.sliderPos = response.state;
+		if (AppAssistant.KTstagecount==0) AppAssistant.KTorient = 'up';
+		else AppAssistant.KTorient = Mojo.Controller.appController.getScreenOrientation();
+	};
+};
+
+AppAssistant.prototype.sliderSubscribe = function(callback){
+    var request = new Mojo.Service.Request("palm://com.palm.keys/switches", {
+        method: 'status',
+		parameters: {
+			'subscribe': true
+		},
+		onSuccess: callback,
+		onFailure: function() {
+		}
+    });
+    return request;
+};
+//END Slider Orientation Patch block 2 of 3
+
 AppAssistant.prototype.getDatabase = function() {
 
 	return this.database;
@@ -127,5 +169,8 @@
 AppAssistant.prototype.cleanup = function() {
 
+//BEGIN Slider Orientation Patch block 3 of 3
+		if (this.sliderSubscriptionx) this.sliderSubscriptionx.cancel();
+//END Slider Orientation Patch block 3 of 3
 };
 
 AppAssistant.prototype.goToNotifyScene = function(params) {
Index: /usr/palm/applications/com.palm.app.browser/app/controllers/page-assistant.js
===================================================================
--- /usr/palm/applications/com.palm.app.browser/app/controllers/page-assistant.js
+++ /usr/palm/applications/com.palm.app.browser/app/controllers/page-assistant.js
@@ -16,5 +16,9 @@
 var BrowserServerStatus;
 
 function PageAssistant(params){
+//BEGIN Slider Orientation Patch block 1 of 5
+	this._initSlider();
+	AppAssistant.KTstagecount++;
+//END Slider Orientation Patch block 1 of 5
 
 	// Record our launch parameters for later use in adapter setup.
@@ -155,3 +159,9 @@
 		this.controller.stopListening(this.controller.sceneElement, Mojo.Event.keypress, this._onKeyPressEventHandler);
 
+//BEGIN Slider Orientation Patch block 2 of 5
+		if (this.sliderSubscriptionx) this.sliderSubscriptionx.cancel();
+		AppAssistant.KTstagecount--;
+		if (AppAssistant.KTstagecount==0) AppAssistant.KTorient='up';
+//END Slider Orientation Patch block 2 of 5
+
 		// Delete old history
@@ -196,10 +206,32 @@
 	if (targetWindow.PalmSystem && targetWindow.PalmSystem.setWindowOrientation) {
-		this._orientation = Mojo.Controller.appController.getScreenOrientation();
+//BEGIN Slider Orientation Patch block 3 of 5
+		if (AppAssistant.sliderPos=='up') {
+			this._orientation = AppAssistant.KTorient;
+		} else {
+			this._orientation = Mojo.Controller.appController.getScreenOrientation();
+		}
+//BEGIN Slider Orientation Patch block 3 of 5
 		targetWindow.PalmSystem.setWindowOrientation(this._orientation);
 	}
 };
 
+//BEGIN Slider Orientation Patch block 4 of 5
+PageAssistant.prototype._initSlider = function() {
+	this.sliderSubscriptionx = AppAssistant.prototype.sliderSubscribe(this._onSliderEvent.bind(this));
+};
+
+PageAssistant.prototype._onSliderEvent = function(response) {
+	if (response && response.key && response.state && response.key == "slider" && response.state=='down') {
+		this.orientationChanged(Mojo.Controller.appController.getScreenOrientation());
+	}
+};
+
+//END Slider Orientation Patch block 4 of 5
+
 PageAssistant.prototype.orientationChanged = function(orientation) {
+//BEGIN Slider Orientation Patch block 5 of 5
+	if (AppAssistant.sliderPos=='up') return;
+//END Slider Orientation Patch block 5 of 5
 
 	// Switch the orientation of the menu spacer or other components..
