diff --git a/usr/palm/applications/com.palm.app.browser/source/Browser.js b/usr/palm/applications/com.palm.app.browser/source/Browser.js
index f2d5a46..75cb1f0 100644
--- a/usr/palm/applications/com.palm.app.browser/source/Browser.js
+++ b/usr/palm/applications/com.palm.app.browser/source/Browser.js
@@ -318,6 +318,9 @@ enyo.kind({
 		this.log(params.text);
 		this.$.launchApplicationService.call({id: "com.palm.app.email", params: params});
 	},
+	downloadLinkClick: function(inTapInfo) {
+		this.doDownloadLink(inTapInfo.linkUrl);
+	},
 	copyToPhotosClick: function(inTapInfo, inPosition) {
 		this.viewCall("saveImageAtPoint", [inPosition.left, inPosition.top, "/media/internal",
 			enyo.hitch(this, "finishCopyToPhotos", inTapInfo)]);
diff --git a/usr/palm/applications/com.palm.app.browser/source/BrowserApp.js b/usr/palm/applications/com.palm.app.browser/source/BrowserApp.js
index 9a4bf6f..da601b0 100644
--- a/usr/palm/applications/com.palm.app.browser/source/BrowserApp.js
+++ b/usr/palm/applications/com.palm.app.browser/source/BrowserApp.js
@@ -581,6 +581,9 @@ enyo.kind({
 	shareClick: function() {
 		this.$.browser.shareLink(this.url, this.title);
 	},
+	downloadClick: function(inSender, inLink) {
+		this.downloadResource(inLink);
+	},
 	bookmarksClick: function() {
 		this.$.toaster.open();
 		this.$.drawerPane.selectView(this.$.drawerPane.getView());
diff --git a/usr/palm/applications/com.palm.app.browser/source/BrowserContextMenu.js b/usr/palm/applications/com.palm.app.browser/source/BrowserContextMenu.js
index b029eba..d527f03 100644
--- a/usr/palm/applications/com.palm.app.browser/source/BrowserContextMenu.js
+++ b/usr/palm/applications/com.palm.app.browser/source/BrowserContextMenu.js
@@ -11,7 +11,8 @@ enyo.kind({
 	linkItems: [
 		{caption: $L("Open In New Card"), value:"newCardClick"},
 		{caption: $L("Share Link"), value:"shareLinkClick"},
-		{caption: $L("Copy URL"), value:"copyLinkClick"}
+		{caption: $L("Copy URL"), value:"copyLinkClick"},
+		{caption: $L("Download Link"), value:"downloadLinkClick"}
 	],
 	imageItems: [
 		{caption: $L("Copy To Photos"), value: "copyToPhotosClick"},
