diff --git a/usr/palm/applications/com.palm.app.browser/source/util.js b/usr/palm/applications/com.palm.app.browser/source/util.js
index e4fcba9..9fae574 100644
--- a/usr/palm/applications/com.palm.app.browser/source/util.js
+++ b/usr/palm/applications/com.palm.app.browser/source/util.js
@@ -33,7 +33,7 @@ enyo.uri = {
 	},
 	isValidScheme: function(inUri) {
 		var scheme = inUri.scheme ? inUri.scheme.toLowerCase() : "";
-		if (!scheme || scheme === "http" || scheme === "https" || scheme === "data" || scheme === "ftp" || scheme === "about") {
+		if (!scheme || scheme === "http" || scheme === "https" || scheme === "data" || scheme === "ftp" || scheme === "about" || scheme === "file") {
 			return true;
 		}
 	}
