diff --git a/usr/palm/applications/com.palm.app.browser/source/BrowserApp.js b/usr/palm/applications/com.palm.app.browser/source/BrowserApp.js
index 9a4bf6f..1f1acd5 100644
--- a/usr/palm/applications/com.palm.app.browser/source/BrowserApp.js
+++ b/usr/palm/applications/com.palm.app.browser/source/BrowserApp.js
@@ -114,6 +114,7 @@ enyo.kind({
 		this.inherited(arguments);
 		this.fetchInitialPreferences();
 		this.log(enyo.json.stringify(enyo.windowParams));
+               
 	},
 	applicationRelaunchHandler: function(inSender) {
 		var c = enyo.windows.getActiveWindow();
@@ -143,6 +144,8 @@ enyo.kind({
 			this.$.pane.selectViewByName("browser");
 		} else {
 			this.$.pane.selectViewByName("startPage");
+                        this.$.toaster.open();
+		        this.$.drawerPane.selectView(this.$.drawerPane.getView());
 		}
 		this.showBookmarks();
 		//this.showAppMenu();
