diff --git a/usr/palm/applications/com.palm.app.browser/app/controllers/downloaddialog-assistant.js b/usr/palm/applications/com.palm.app.browser/app/controllers/downloaddialog-assistant.js
new file mode 100644
index 0000000..4c07986
--- /dev/null
+++ b/usr/palm/applications/com.palm.app.browser/app/controllers/downloaddialog-assistant.js
@@ -0,0 +1,46 @@
+/**
+ * A dialog assistant for display of yes/no box.
+ */
+DownloadDialogAssistant = Class.create({
+
+        initialize: function(params) {
+                this.onDismiss = params.onDismiss;
+                this.onAccept = params.onAccept;
+                this.controller= params.sceneAssistant.controller;
+
+                // Button handlers.
+                this.onDismissHandler = this.handleDismiss.bindAsEventListener(this);
+                this.onAcceptHandler = this.handleAccept.bindAsEventListener(this);
+        },
+
+        setup: function(widget) {
+                this.widget = widget;
+                this.controller.get('acceptButton').addEventListener(Mojo.Event.tap, this.onAcceptHandler);
+                this.controller.get('acceptButton').focus();
+                this.controller.get('dismissButton').addEventListener(Mojo.Event.tap, this.onDismissHandler);
+                this.controller.get('dismissButton').focus();
+        },
+
+        handleDismiss: function() {
+                this.onDismiss();
+                delete this.onDismiss;
+                this.widget.mojo.close();
+        },
+        handleAccept: function() {
+                this.onAccept();
+                delete this.onAccept;
+				 delete this.onDismiss;
+                this.widget.mojo.close();
+        },
+
+        cleanup: function() {
+                Mojo.Log.info("NetworkDialogAssistant#cleanup()");
+                Mojo.Event.stopListening(this.controller.get('dismissButton'), Mojo.Event.tap, this.onDismissHandler);
+                Mojo.Event.stopListening(this.controller.get('acceptButton'), Mojo.Event.tap, this.onAcceptHandler);
+
+                // Send a dismiss if NOT already sent a response
+                if (this.onDismiss) {
+                        this.onDismiss();
+                }
+        }
+});
diff --git a/usr/palm/applications/com.palm.app.browser/app/controllers/page-assistant.js b/usr/palm/applications/com.palm.app.browser/app/controllers/page-assistant.js
index 388cffc..88b9c52 100644
--- a/usr/palm/applications/com.palm.app.browser/app/controllers/page-assistant.js
+++ b/usr/palm/applications/com.palm.app.browser/app/controllers/page-assistant.js
@@ -632,6 +632,7 @@ PageAssistant.prototype._onWebViewSingleTap = function(event) {
 				{label: $L('Open In New Card'), command:'openNew'},
 				{label: $L('Share Link'), command:'shareUrl'},
 				{label: $L('Copy URL'), command:'copyUrl'},
+				{label: $L('Download URL'), command:'downloadUrl'},
 				{label: $L('Copy to Photos'), command:'copyToPhotos'},
 				{label: $L('Share Image'), command:'shareImage'},
 				{label: $L('Set Wallpaper'), command:'setWallpaper'}
@@ -683,6 +684,9 @@ PageAssistant.prototype._onWebViewSingleTap = function(event) {
 					case 'copyUrl':
 						this.controller.stageController.setClipboard(urlInfo.url);
 						break;
+					case 'downloadUrl':
+						this._downloadResource(urlInfo.url);
+						break;
 					case 'copyToPhotos':
 						this._webView.mojo.saveImageAtPoint(tapPt.left, tapPt.top, "/media/internal", saveImageCallback);
 						break;
@@ -703,6 +707,7 @@ PageAssistant.prototype._onWebViewSingleTap = function(event) {
 					usedItems.push( findItem('openNew') );
 					usedItems.push( findItem('shareUrl') );
 					usedItems.push( findItem('copyUrl') );
+					usedItems.push( findItem('downloadUrl') );
 				}
 
 				if (response.src) {
@@ -1224,6 +1229,30 @@ PageAssistant.prototype._newBrowserPage = function(url, pageIdentifier){
  */
 PageAssistant.prototype._streamResource = function(uri, appid, mimeType){
 
+	if(appid === 'com.palm.app.email' || appid === 'com.palm.app.docviewer' || appid === 'com.palm.app.pdfviewer' || appid === 'com.palm.app.maps' || appid === 'com.palm.app.phone' || appid === 'com.palm.app.youtube')
+	//a phone #, Google Map, docviewer, email, pdfviewer, or YouTube link was clicked, so we bypass the download dialog and open the app.
+	{
+		var params = {target: uri, mimeType: mimeType};
+		this.controller.serviceRequest('palm://com.palm.applicationManager',{
+			method: 'open',
+			parameters: {
+				'id': appid,
+				'params': params
+			}}
+		);
+		return;
+	}
+	    this._downloadWidgetElement = this.controller.showDialog({
+	        uri: uri,
+	        mimeType: mimeType,
+	        appid: appid,
+	        template: 'download/download-stream-popup',
+	        assistant: new DownloadDialogAssistant({
+	        sceneAssistant: this,
+	        onDismiss: function(cParams) { // DOWNLOAD
+	                this._downloadResource(uri);
+	        }.bind(this),
+	        onAccept: function(cParams) { // STREAM
 	// Only a few select applications can be
 	var crossAppScene = {
 		'com.palm.app.videoplayer': 'nowplaying',
@@ -1244,6 +1273,9 @@ PageAssistant.prototype._streamResource = function(uri, appid, mimeType){
 			}
 		});
 	}
+	        }.bind(this)})
+	});
+
 };
 
 /**
diff --git a/usr/palm/applications/com.palm.app.browser/app/views/download/download-stream-popup.html b/usr/palm/applications/com.palm.app.browser/app/views/download/download-stream-popup.html
new file mode 100644
index 0000000..b705ece
--- /dev/null
+++ b/usr/palm/applications/com.palm.app.browser/app/views/download/download-stream-popup.html
@@ -0,0 +1,8 @@
+<div id="palm-dialog-content" class="palm-dialog-content">
+        <div class="dialog-message" x-mojo-loc="">This file type has been registered as a streaming media file. Would you like to download it to /media/internal/downloads instead?</div>
+</div>
+
+<div class="palm-dialog-buttons">
+        <div class="dismiss palm-button" id="acceptButton" x-mojo-loc="" x-mojo-tap-highlight="momentary">Stream</div>
+        <div class="dismiss palm-button" id="dismissButton" x-mojo-loc="" x-mojo-tap-highlight="momentary">Download</div>
+</div>
diff --git a/usr/palm/applications/com.palm.app.browser/sources.json b/usr/palm/applications/com.palm.app.browser/sources.json
index 69e8f0b..8514ad2 100644
--- a/usr/palm/applications/com.palm.app.browser/sources.json
+++ b/usr/palm/applications/com.palm.app.browser/sources.json
@@ -47,7 +47,10 @@
   {
 	"source": "app\/controllers\/download-controller.js"
   },
-  {  
+  {
+  	"source":"app\/controllers\/downloaddialog-assistant.js"
+  },
+  {    
 	"source": "app\/controllers\/preferences-assistant.js"
   },
   {
