diff --git a/usr/palm/applications/com.palm.app.browser/app/controllers/page-applicationmenu.js b/usr/palm/applications/com.palm.app.browser/app/controllers/page-applicationmenu.js
index 6bf5870..d1d1e7b 100644
--- a/usr/palm/applications/com.palm.app.browser/app/controllers/page-applicationmenu.js
+++ b/usr/palm/applications/com.palm.app.browser/app/controllers/page-applicationmenu.js
@@ -37,6 +37,11 @@ MenuData.ApplicationMenu = {
 		checkEnabled: true
 	},
 	
+	TogglePrivate: {
+		label: $L('Enable Private Browsing'),
+		command: 'enable-private-cmd'
+	},
+	
 	ShowHistory: {
 		label: $L('History'),
 		command: 'show-history-cmd',
diff --git a/usr/palm/applications/com.palm.app.browser/app/controllers/page-assistant.js b/usr/palm/applications/com.palm.app.browser/app/controllers/page-assistant.js
index 388cffc..7764edf 100644
--- a/usr/palm/applications/com.palm.app.browser/app/controllers/page-assistant.js
+++ b/usr/palm/applications/com.palm.app.browser/app/controllers/page-assistant.js
@@ -93,8 +93,26 @@ PageAssistant.prototype.setup = function() {
 		this._updateOrientation();
 
 		// Allow override of 'back' gesture in landscape mode.
-		this.controller.useLandscapePageUpDown(true);
 
+		this.getTweaksOptions = new Mojo.Service.Request("palm://org.webosinternals.tweaks.prefs/", {
+				method: 'get', parameters: {'owner': "browser-behavior-options-tweaks", 
+				keys: ["gestureScroll", "goFullscreen"]},
+				onSuccess: function(response) {
+					if(response) {
+						if(response.gestureScroll == false) {
+							this.gestureScroll = false;
+						} else {
+							this.gestureScroll = true;
+						}
+						if(response.goFullscreen == true) {
+							this.goFullscreen = true;
+						} else {
+							this.goFullscreen = false;
+						}
+					}
+		}.bind(this)});
+		
+		this.controller.useLandscapePageUpDown(this.gestureScroll);
 		// Create the browser server status control.
 		this._browserServerStatus = new BrowserServerStatus(this.controller);
 		this._browserServerStatus.setup({
@@ -304,6 +322,11 @@ PageAssistant.prototype.activate = function(message) {
 		
 		this._updateOrientation();
 		
+		if (this.goFullscreen) {
+			this.controller.enableFullScreenMode(true);
+		}
+
+		
 		// Update any pending web preferences changes
 		AppAssistant.WebPreferences.activate(this._webView.mojo);
 
@@ -804,6 +827,20 @@ PageAssistant.prototype._setupMenus = function() {
 		};
 
 		this.controller.setupWidget(Mojo.Menu.appMenu, undefined, this.appMenuModel);
+		
+		this.getTweaksMainMenu = new Mojo.Service.Request("palm://org.webosinternals.tweaks.prefs/", {
+			method: 'get', parameters: {'owner': "browser-behavior-options-tweaks", 
+			keys: ["privateBrowsing", "viewSource"]},
+			onSuccess: function(response) {
+				if(response) {
+					if(response.privateBrowsing == true)
+						this.appMenuModel.items.push(this.togglePrivate = MenuData.ApplicationMenu.TogglePrivate);
+					if(response.viewSource == true)
+						this.appMenuModel.items.push({label:"View Source", command:"view-source-cmd"});
+				}
+		}.bind(this)});
+		
+		this.controller.modelChanged(this.appMenuModel);
 
 		// Create the URL Bar
 		this._addressBar = new AddressBar(this.controller);
@@ -1066,6 +1103,10 @@ PageAssistant.prototype._updateHistory = function() {
 	if (!this.currentUrl || (this.currentUrl.length === 0)) {
 		return;
 	}
+	
+	if (this.togglePrivate.enabled) {
+		return;
+	}
 
 	var title = this.currentTitle || null;
 	this.historyStore.addHistory(this.currentUrl, title, new Date(),
@@ -1946,6 +1987,10 @@ PageAssistant.prototype.moved = function(scrollState) {
 			this._addressBar.hide();
 			this.chrome.hide();
 
+			if(this.goFullscreen) {
+				this.controller.enableFullScreenMode(true);
+			}
+
 			// Only focus this webview widget if this assistant has
 			// focus. All page-assisants can receive 'moved' events
 			// so we have to be careful when explicitly setting the
@@ -1969,6 +2014,10 @@ PageAssistant.prototype.moved = function(scrollState) {
 		//Mojo.Log.info("-> TOP > 0 SO SHOWING");
 		this._addressBar.show();
 		this.chrome.show();
+		
+		if(this.goFullscreen) {
+			this.controller.enableFullScreenMode(false);
+		}
 
 	} else {
 		//Mojo.Log.info("-> TOP == 0 Do Nothing.");
@@ -2180,6 +2229,41 @@ PageAssistant.prototype.handleCommand = function(event) {
 				case MenuData.NavigationMenu.Back.command:
 					this._goBack();
 					break;
+				
+				//PRIVATE BROWSING
+					
+				case 'enable-private-cmd':
+					this.togglePrivate.command = 'disable-private-cmd';
+					this.togglePrivate.label = $L('Disable Private Browsing');
+					
+					this.togglePrivate.enabled = true;
+					
+					this.controller.modelChanged(this.appMenuModel);
+					Mojo.Controller.getAppController().showBanner("Private Browsing: Enabled!",  {source: 'notification'});
+					break;
+					
+				case 'disable-private-cmd':
+					this.togglePrivate.command = 'enable-private-cmd';
+					this.togglePrivate.label = $L('Enable Private Browsing');
+					
+					this.togglePrivate.enabled = false;
+					
+					this.controller.modelChanged(this.appMenuModel);
+					Mojo.Controller.getAppController().showBanner("Private Browsing: Disabled!",  {source: 'notification'});
+					break;		
+
+				//VIEW SOURCE
+				case "view-source-cmd":
+					this.controller.serviceRequest("palm://com.palm.applicationManager", {
+						method: "open",
+						parameters: {
+							id: "ca.canucksoftware.internalz",
+							params: {
+								target: "view-source://" + this.currentUrl
+							}
+						}
+					});
+					break;
 
 				case MenuData.NavigationMenu.Forward.command:
 					this._goForward();
diff --git a/usr/palm/applications/com.palm.app.browser/app/controllers/page-controls.js b/usr/palm/applications/com.palm.app.browser/app/controllers/page-controls.js
index 6ad7ebd..c03a552 100644
--- a/usr/palm/applications/com.palm.app.browser/app/controllers/page-controls.js
+++ b/usr/palm/applications/com.palm.app.browser/app/controllers/page-controls.js
@@ -6,6 +6,17 @@
 */
 function PageControls (controller){
 	this._controller = controller;
+	
+		this.hideIcons = false;
+		this.getTweaksOptions = new Mojo.Service.Request("palm://org.webosinternals.tweaks.prefs/", {
+			method: 'get', parameters: {'owner': "browser-behavior-options-tweaks", 
+			keys: ["hideIcons"]},
+			onSuccess: function(response) {
+				if(response) {
+					if(response.hideIcons == true)
+						this.hideIcons = true;
+				}
+		}.bind(this)});
 
 		// We can always go back.
 		this._back = true;
@@ -26,11 +37,12 @@ function PageControls (controller){
 		};
 
 		this._model = {
-			visible: true,
+			visible: !this.hideIcons,
 			items: [this._backModel, {}, {}]
 		};
 
 		this._currentPageModel = {};
+		
 }
 
 PageControls.prototype={
@@ -64,7 +76,11 @@ PageControls.prototype={
 
 		var items = _.toArray();
 		this._backModel.disabled = !back;
-		items.push(this._backModel);
+		if (!this.hideIcons) {
+			items.push(this._backModel);
+		} else {
+			items.push(false);
+		}
 
 		if (forward) {
 			this._forwardModel.disabled = !forward;
@@ -74,7 +90,11 @@ PageControls.prototype={
 		}
 
 		// The spacer... right aligns the following button.
-		items.push({});
+		if (!this.hideIcons) {
+			items.push({});
+		} else {
+			items.push(false);
+		}
 
 		this._back = back;
 		this._forward = forward;
@@ -85,10 +105,14 @@ PageControls.prototype={
 	updateBackForward: function(back, forward) {
 
 		// Add the back/forward buttons
-		this._model.items = this._setBackForward(back, forward);
+		if(!this.hideIcons) {
+			this._model.items = this._setBackForward(back, forward);
+		} else {
+			this._model.items = this._setBackForward(false);
+		}
+		
 		// The spacer... right aligns the following button.
 		this._model.items.push(this._currentPageModel);
-
 		this._controller.modelChanged(this._model);
 	},
 
@@ -106,13 +130,21 @@ PageControls.prototype={
 			this._animation.stop();
 			delete this._animation;
 		}
+		
+		
 
 		// Add the back/forward buttons
-		this._model.items = this._setBackForward(this._back, this._forward);
+		if(!this.hideIcons) {
+			this._model.items = this._setBackForward(this._back, this._forward);
+		} else {
+			this._model.items = this._setBackForward(false);
+		}
 		// The spacer... right aligns the following button.
 		this._model.items.push(idleModel);
 		this._controller.modelChanged(this._model);
-
+		if(this.hideIcons) {
+			this._controller.setMenuVisible(Mojo.Menu.commandMenu, false);
+		}
 		this._currentPageModel = idleModel;
 	},
 
@@ -138,11 +170,17 @@ PageControls.prototype={
 			});
 
 			// Add the back/forward buttons
-			this._model.items = this._setBackForward(this._back, this._forward);
+			if(!this.hideIcons) {
+				this._model.items = this._setBackForward(this._back, this._forward);
+			} else {
+				this._model.items = this._setBackForward(false);
+			}
 			this._model.items.push(progressModel);
 			this._controller.modelChanged(this._model);
 			this._animation.start(0);
-
+			if(this.hideIcons) {
+				this._controller.setMenuVisible(Mojo.Menu.commandMenu, true);
+			}
 			this._currentPageModel = progressModel;
 
 		} catch (e) {
@@ -183,11 +221,17 @@ PageControls.prototype={
 			});
 
 			// Add the back/forward buttons
-			this._model.items = this._setBackForward(this._back, this._forward);
+			if(!this.hideIcons) {
+				this._model.items = this._setBackForward(this._back, this._forward);
+			} else {
+				this._model.items = this._setBackForward(false);
+			}
 			this._model.items.push(searchModel);
 			this._controller.modelChanged(this._model);
 			this._animation.start();
-
+			if(this.hideIcons) {
+				this._controller.setMenuVisible(Mojo.Menu.commandMenu, true);
+			}
 			this._currentPageModel = searchModel;
 
 		} catch (e) {
@@ -204,6 +248,9 @@ PageControls.prototype={
 					this._model.items[i].type='flash';
 					this._model.items[i].icon= MenuData.NavigationMenu.Cancel.icon;
 					this._model.items[i].command=MenuData.NavigationMenu.Cancel.command;
+					if(this.hideIcons) {
+						this._controller.setMenuVisible(Mojo.Menu.commandMenu, true);
+					}
 					break;
 				}
 				
@@ -219,6 +266,9 @@ PageControls.prototype={
 					this._model.items[i].type='idle';
 					this._model.items[i].icon= MenuData.NavigationMenu.Reload.icon;
 					this._model.items[i].command=MenuData.NavigationMenu.Reload.command;
+					if(this.hideIcons) {
+						this._controller.setMenuVisible(Mojo.Menu.commandMenu, false);
+					}
 					break;
 				}
 				
diff --git a/usr/palm/applications/com.palm.app.browser/app/controllers/startpage-assistant.js b/usr/palm/applications/com.palm.app.browser/app/controllers/startpage-assistant.js
index 0fbc276..8d767c9 100644
--- a/usr/palm/applications/com.palm.app.browser/app/controllers/startpage-assistant.js
+++ b/usr/palm/applications/com.palm.app.browser/app/controllers/startpage-assistant.js
@@ -24,6 +24,16 @@ StartpageAssistant.prototype.setup = function() {
 
 	// Allow override of 'back' gesture in landscape mode.
 	this.controller.useLandscapePageUpDown(true);
+	this.goFullscreen = false;
+	this.getTweaksOptions = new Mojo.Service.Request("palm://org.webosinternals.tweaks.prefs/", {
+			method: 'get', parameters: {'owner': "browser-behavior-options-tweaks", 
+			keys: ["gestureScroll"]},
+			onSuccess: function(response) {
+				if(response) {
+					if(response.gestureScroll == false)
+						this.controller.useLandscapePageUpDown(false);
+				}
+		}.bind(this)});
 
 	// Listen for scene scope keyboard events.
 	this.controller.listen(this.controller.sceneElement, Mojo.Event.keydown, this._onKeyDownEventHandler);
@@ -53,6 +63,16 @@ StartpageAssistant.prototype.setup = function() {
 			MenuData.ApplicationMenu.ShowBookmarks,
 			MenuData.ApplicationMenu.ShowHistory]
 	};
+	
+	this.getTweaksPrefs = new Mojo.Service.Request("palm://org.webosinternals.tweaks.prefs/", {
+			method: 'get', parameters: {'owner': "browser-behavior-options-tweaks", 
+			keys: ["privateBrowsing" ]},
+			onSuccess: function(response) {
+				if(response) {
+					if(response.privateBrowsing == true)
+						this.appMenuModel.items.splice(3,0,this.togglePrivate = MenuData.ApplicationMenu.TogglePrivate);
+				}
+		}.bind(this)});
 
 	this.controller.setupWidget(Mojo.Menu.appMenu, undefined, this.appMenuModel);
 
@@ -367,7 +387,27 @@ StartpageAssistant.prototype.handleCommand = function(event) {
 				case MenuData.ApplicationMenu.ShowHistory.command:
 					this._openHistoryView();
 					break;
-
+				
+				case 'enable-private-cmd':
+					this.togglePrivate.command = 'disable-private-cmd';
+					this.togglePrivate.label = $L('Disable Private Browsing');
+					
+					this.togglePrivate.enabled = true;
+					
+					this.controller.modelChanged(this.appMenuModel);
+					Mojo.Controller.getAppController().showBanner("Private Browsing: Enabled!",  {source: 'notification'});
+					break;
+					
+				case 'disable-private-cmd':
+					this.togglePrivate.command = 'enable-private-cmd';
+					this.togglePrivate.label = $L('Enable Private Browsing');
+					
+					this.togglePrivate.enabled = false;
+					
+					this.controller.modelChanged(this.appMenuModel);
+					Mojo.Controller.getAppController().showBanner("Private Browsing: Disabled!",  {source: 'notification'});
+					break;
+				
 				case MenuData.ApplicationMenu.ShowBookmarks.command:
 					this._openBookmarkView();
 					break;
