diff --git a/usr/palm/applications/com.palm.app.browser/app/controllers/startpage-assistant.js b/usr/palm/applications/com.palm.app.browser/app/controllers/startpage-assistant.js
index f024692..c0056f0 100644
--- a/usr/palm/applications/com.palm.app.browser/app/controllers/startpage-assistant.js
+++ b/usr/palm/applications/com.palm.app.browser/app/controllers/startpage-assistant.js
@@ -9,7 +9,7 @@ function StartpageAssistant(params) {
 	this._destinationUrl = params.lastUrl;
 	this._orientation = params.orientation || 'up';
 	this._bookmarks = [];
-	this._maxShowCount = 12;
+	this._maxShowCount = 99;
 	this.store = new BookmarkStore();
 
 	this._onStartPageTapHandler = this._onStartPageTap.bind(this);
