diff --git a/usr/palm/applications/com.palm.app.browser/app/controllers/page-controls.js b/usr/palm/applications/com.palm.app.browser/app/controllers/page-controls.js
index 6ad7ebd..c6d86da 100644
--- a/usr/palm/applications/com.palm.app.browser/app/controllers/page-controls.js
+++ b/usr/palm/applications/com.palm.app.browser/app/controllers/page-controls.js
@@ -24,6 +24,13 @@ function PageControls (controller){
 			icon: MenuData.NavigationMenu.Forward.icon,
 			command: MenuData.NavigationMenu.Forward.command
 		};
+		
+		this._newcardModel = {
+			disabled: false,
+			label: MenuData.NavigationMenu.NewCard.label,
+			icon: MenuData.NavigationMenu.NewCard.icon,
+			command: MenuData.NavigationMenu.NewCard.command
+		};
 
 		this._model = {
 			visible: true,
@@ -49,6 +56,7 @@ PageControls.prototype={
 			this._model.items.push(idleModel);
 		}
 
+		this._model.items.push(this._newcardModel);
 		this._controller.setupWidget(Mojo.Menu.commandMenu, {menuClass: 'no-fade'}, this._model);
 	},
 
@@ -75,6 +83,7 @@ PageControls.prototype={
 
 		// The spacer... right aligns the following button.
 		items.push({});
+		items.push(this._newcardModel);
 
 		this._back = back;
 		this._forward = forward;
diff --git a/usr/palm/applications/com.palm.app.browser/app/controllers/page-navigationmenu.js b/usr/palm/applications/com.palm.app.browser/app/controllers/page-navigationmenu.js
index 007600b..13315dc 100644
--- a/usr/palm/applications/com.palm.app.browser/app/controllers/page-navigationmenu.js
+++ b/usr/palm/applications/com.palm.app.browser/app/controllers/page-navigationmenu.js
@@ -23,6 +23,12 @@ MenuData.NavigationMenu = {
 		icon: 'load-progress',
 		command: 'stop'
 	},
+
+	NewCard: {
+		label: $L('New Card'),
+		icon: 'new',
+		command: 'new-page-cmd'
+	},
 	
 	Reload: {
 		label: $L('Reload'),
