diff --git a/usr/palm/applications/com.palm.app.browser/app/controllers/page-controls.js b/usr/palm/applications/com.palm.app.browser/app/controllers/page-controls.js
index 965e167..6c99ea2 100644
--- a/usr/palm/applications/com.palm.app.browser/app/controllers/page-controls.js
+++ b/usr/palm/applications/com.palm.app.browser/app/controllers/page-controls.js
@@ -26,6 +26,13 @@ PageControls = Class.create({
 			command: MenuData.NavigationMenu.Forward.command
 		};
 
+		this._newcardModel = {
+			disabled: false,
+			label: MenuData.NavigationMenu.NewCard.label,
+			icon: MenuData.NavigationMenu.NewCard.icon,
+			command: MenuData.NavigationMenu.NewCard.command
+		};
+
 		this._model = {
 			visible: true,
 			items: [this._backModel, {}, {}]
@@ -48,6 +55,7 @@ PageControls = Class.create({
 			this._model.items.push(idleModel);
 		}
 
+		this._model.items.push(this._newcardModel);
 		this._controller.setupWidget(Mojo.Menu.commandMenu, {menuClass: 'no-fade'}, this._model);
 	},
 
@@ -74,6 +82,7 @@ PageControls = Class.create({
 
 		// The spacer... right aligns the following button.
 		items.push({});
+		items.push(this._newcardModel);
 
 		this._back = back;
 		this._forward = forward;
diff --git a/usr/palm/applications/com.palm.app.browser/app/controllers/page-navigationmenu.js b/usr/palm/applications/com.palm.app.browser/app/controllers/page-navigationmenu.js
index 0271a5b..e82d1b1 100644
--- a/usr/palm/applications/com.palm.app.browser/app/controllers/page-navigationmenu.js
+++ b/usr/palm/applications/com.palm.app.browser/app/controllers/page-navigationmenu.js
@@ -22,6 +22,12 @@ MenuData.NavigationMenu = {
 		command: 'stop'
 	},
 	
+	NewCard: {
+		label: $L('New Card'),
+		icon: 'new',
+		command: 'new-page-cmd'
+	},
+	
 	Reload: {
 		label: $L('Reload'),
 		icon: 'refresh',
