diff --git a/etc/palm/browser-app.conf b/etc/palm/browser-app.conf
index c8f1607..e591b6a 100755
--- a/etc/palm/browser-app.conf
+++ b/etc/palm/browser-app.conf
@@ -35,6 +35,7 @@ EnableAppCache=true
 AppCacheDbPath=/media/internal/dbdata/appcache
 AppCacheDbMaxSize=1073741824
 EnableNoSteCommonTagsCheck=true
+UserAgentOverride=Mozilla/6.0 (Windows NT 6.2; WOW64; rv:16.0.1) Gecko/20121011 Firefox/16.0.1
 
 # The 'EnableFitWidth' parameter lets us control the default behavior of ScrollView.
 EnableFitWidth=false
