diff --git a/usr/palm/applications/com.palm.app.browser/app/controllers/page-assistant.js b/usr/palm/applications/com.palm.app.browser/app/controllers/page-assistant.js
index 09c898f..2fda813 100644
--- a/usr/palm/applications/com.palm.app.browser/app/controllers/page-assistant.js
+++ b/usr/palm/applications/com.palm.app.browser/app/controllers/page-assistant.js
@@ -621,7 +621,15 @@ PageAssistant.prototype._onWebViewSingleTap = function(event) {
 		tapPt.left = event.centerX - tapPt.left;
 		tapPt.top  = event.centerY - tapPt.top;
 
-		if (event.altKey) {
+        if (event.metaKey) {
+            var urlInfo = {};
+            var urlInspectResponse = function(response) {
+				urlInfo = response || {};
+				this._newBrowserPage(urlInfo.url);
+			}.bind(this);
+			this._webView.mojo.inspectUrlAtPoint(tapPt.left, tapPt.top, urlInspectResponse);
+        }
+		else if (event.altKey) {
 
 			var popupItems = [
 				{label: $L('Open In New Card'), command:'openNew'},
diff --git a/usr/palm/frameworks/mojo/builtins/palmInitFramework338.js b/usr/palm/frameworks/mojo/builtins/palmInitFramework338.js
index 2db731c..e36f50c 100644
--- a/usr/palm/frameworks/mojo/builtins/palmInitFramework338.js
+++ b/usr/palm/frameworks/mojo/builtins/palmInitFramework338.js
@@ -28127,7 +28127,7 @@ if(!this.isElementHighlighted&&this.trackballMode===false&&this.selectionMode===
 this._addElementHighlight(event.centerX,event.centerY);
 }
 
-if(event.altKey){
+if(event.altKey || event.metaKey){
 
 
 
