diff --git a/usr/palm/frameworks/webview/submission/23.1/javascript/widget_webview.js b/usr/palm/frameworks/webview/submission/23.1/javascript/widget_webview.js
index 89af630..97187d4 100644
--- a/usr/palm/frameworks/webview/submission/23.1/javascript/widget_webview.js
+++ b/usr/palm/frameworks/webview/submission/23.1/javascript/widget_webview.js
@@ -911,13 +911,14 @@ Widget.prototype = {
 			}
 			catch (e) {} // Can fail if page scene is not active.
 		}
-
+/*
 		if (!this.adapterConnected) {
 			// This browser page was purged when the application was not in the foreground.
 			// Now that we're active again let's reconnect and reload the page.
 			Mojo.Log.info("Reconnecting adapter to server.");
 			this._connectAdapterToServer();
 		} 
+*/
 	},
 
 	_cardDeactivate: function(event) {
@@ -3153,7 +3154,7 @@ Widget.prototype = {
 		}
 		else {
 			if (this._serverConnectTimer === undefined) {
-				this._serverConnectTimer = this.controller.window.setTimeout(this._connectAdapterToServer, 3000);
+				this._serverConnectTimer = this.controller.window.setTimeout(this._connectAdapterToServer, false);
 			}
 			
 			this._removeElementHighlight();
