diff --git a/usr/palm/applications/com.palm.app.browser/app/controllers/page-assistant.js b/usr/palm/applications/com.palm.app.browser/app/controllers/page-assistant.js
index 388cffc..fd84037 100644
--- a/usr/palm/applications/com.palm.app.browser/app/controllers/page-assistant.js
+++ b/usr/palm/applications/com.palm.app.browser/app/controllers/page-assistant.js
@@ -632,6 +632,7 @@ PageAssistant.prototype._onWebViewSingleTap = function(event) {
 				{label: $L('Open In New Card'), command:'openNew'},
 				{label: $L('Share Link'), command:'shareUrl'},
 				{label: $L('Copy URL'), command:'copyUrl'},
+			    {label: $L('Add to Spare Time'), command:'add-to-spare-time-cmd'},
 				{label: $L('Copy to Photos'), command:'copyToPhotos'},
 				{label: $L('Share Image'), command:'shareImage'},
 				{label: $L('Set Wallpaper'), command:'setWallpaper'}
@@ -674,6 +675,9 @@ PageAssistant.prototype._onWebViewSingleTap = function(event) {
 			var popupSelectFunc = function(value) {
 				selectedCommand = value;
 				switch (value) {
+					case 'add-to-spare-time-cmd':
+						this._addST(urlInfo.url);
+						break;
 					case 'openNew':
 						this._newBrowserPage(urlInfo.url);
 						break;
@@ -703,6 +707,7 @@ PageAssistant.prototype._onWebViewSingleTap = function(event) {
 					usedItems.push( findItem('openNew') );
 					usedItems.push( findItem('shareUrl') );
 					usedItems.push( findItem('copyUrl') );
+					usedItems.push( findItem('add-to-spare-time-cmd') );
 				}
 
 				if (response.src) {
@@ -1846,7 +1851,37 @@ PageAssistant.prototype._shareUrl = function(url, title, withCapture) {
 		method: 'open',
 		parameters: parameters
 	});
-};
+};PageAssistant.prototype._addST = function(link) {
+          this.controller.serviceRequest("palm://com.palm.applicationManager", {
+            method: "launch",
+
+            parameters: {
+              id: "com.semicolonapps.sparetime",
+              params: {action: "add_url", url: link}
+            },
+
+            onFailure: function() {
+              this.controller.showAlertDialog({
+                title: $L("Spare Time Not Installed"),
+                message: $L("Spare Time is not installed. Would you like to purchase it?"),
+
+                choices:[
+                  {label:$L("Yes"), value:"yes", type:"affirmative"},
+                  {label:$L("No"), value:"no", type:"dismissal"}
+                ],
+
+                onChoose: function(value){
+                  if("yes" == value){
+                    this.controller.serviceRequest("palm://com.palm.applicationManager", {
+                      method:"open",
+                      parameters:{target: "http://developer.palm.com/appredirect/?packageid=com.semicolonapps.sparetime"}
+                    });
+                  }
+                }
+              });
+            }.bind(this)
+          });
+ };
 
 PageAssistant.prototype._setWallpaper = function(pathToImage) {
 	var errorTitle = $L("Error Setting Wallpaper");
