diff --git a/usr/palm/applications/com.palm.app.browser/app/controllers/page-assistant.js b/usr/palm/applications/com.palm.app.browser/app/controllers/page-assistant.js
index 388cffc..0d0c992 100644
--- a/usr/palm/applications/com.palm.app.browser/app/controllers/page-assistant.js
+++ b/usr/palm/applications/com.palm.app.browser/app/controllers/page-assistant.js
@@ -2180,6 +2180,10 @@ PageAssistant.prototype.handleCommand = function(event) {
 				case MenuData.NavigationMenu.Back.command:
 					this._goBack();
 					break;
+					
+				case MenuData.NavigationMenu.Bookmarks.command:
+					this._showStartPage();
+					break;
 
 				case MenuData.NavigationMenu.Forward.command:
 					this._goForward();
diff --git a/usr/palm/applications/com.palm.app.browser/app/controllers/page-controls.js b/usr/palm/applications/com.palm.app.browser/app/controllers/page-controls.js
index 6ad7ebd..eac15f3 100644
--- a/usr/palm/applications/com.palm.app.browser/app/controllers/page-controls.js
+++ b/usr/palm/applications/com.palm.app.browser/app/controllers/page-controls.js
@@ -24,10 +24,17 @@ function PageControls (controller){
 			icon: MenuData.NavigationMenu.Forward.icon,
 			command: MenuData.NavigationMenu.Forward.command
 		};
+		
+		this._bookmarksModel = {
+			disabled: false,
+			label: MenuData.NavigationMenu.Bookmarks.label,
+			icon: MenuData.NavigationMenu.Bookmarks.icon,
+			command: MenuData.NavigationMenu.Bookmarks.command
+		};
 
 		this._model = {
 			visible: true,
-			items: [this._backModel, {}, {}]
+			items: [this._backModel, this._bookmarksModel, {}, {}]
 		};
 
 		this._currentPageModel = {};
@@ -65,6 +72,7 @@ PageControls.prototype={
 		var items = _.toArray();
 		this._backModel.disabled = !back;
 		items.push(this._backModel);
+		items.push(this._bookmarksModel);
 
 		if (forward) {
 			this._forwardModel.disabled = !forward;
diff --git a/usr/palm/applications/com.palm.app.browser/app/controllers/page-navigationmenu.js b/usr/palm/applications/com.palm.app.browser/app/controllers/page-navigationmenu.js
index 007600b..acd8963 100644
--- a/usr/palm/applications/com.palm.app.browser/app/controllers/page-navigationmenu.js
+++ b/usr/palm/applications/com.palm.app.browser/app/controllers/page-navigationmenu.js
@@ -11,7 +11,13 @@ MenuData.NavigationMenu = {
 		icon: 'back',
 		command: 'back'
 	},
-	
+
+	Bookmarks: {
+		label: $L('Bookmarks'),
+		icon: 'bookmarks',
+		command: 'bookmarks-page'
+	},
+		
 	Forward: {
 		label: $L('Go forward'),
 		icon: 'forward',
