diff --git a/usr/palm/applications/com.palm.app.browser/source/Browser.js b/usr/palm/applications/com.palm.app.browser/source/Browser.js
index 6094e1c..4bfb1c3 100644
--- a/usr/palm/applications/com.palm.app.browser/source/Browser.js
+++ b/usr/palm/applications/com.palm.app.browser/source/Browser.js
@@ -320,12 +320,16 @@ enyo.kind({
 	shareLinkClick: function(inTapInfo) {
 		this.shareLink(inTapInfo.linkUrl, inTapInfo.linkText || inTapInfo.linkUrl);
 	},
+	// bbito: Make bettter for use with plain text email
 	shareLink: function(inUrl, inTitle) {
 		this.log(inUrl, inTitle);
-		var msg = $L("Here's a website I think you'll like: <a href=\"{$src}\">{$title}</a>");
-		msg = enyo.macroize(msg, {src: inUrl, title: inTitle || inUrl});
+		// bbito: Make the message just the title of the page, a linebreak, and URL.
+		// bbito: Original was 'Here's a website I think you'll like: ' plus the href-ed page title
+		var msg = inTitle + "<br>" + inUrl;
+		// bbito: Make the email subject the page title
+		// bbito: Original was 'Check out this web page...'
 		var params = {
-			summary: $L("Check out this web page..."),
+			summary: inTitle,
 			text: msg
 		};
 		this.log(params.text);
