--- .orig/usr/palm/applications/com.palm.app.browser/app/controllers/page-assistant.js
+++ /usr/palm/applications/com.palm.app.browser/app/controllers/page-assistant.js
@@ -608,7 +608,7 @@
 		tapPt.left = event.centerX - tapPt.left;
 		tapPt.top  = event.centerY - tapPt.top;
 
-		if (event.altKey) {
+		if (event.altKey || event.metaKey) {
 
 			var popupItems = [
 				{label: $L('Open In New Card'), command:'openNew'},
--- .orig/usr/palm/frameworks/mojo/builtins/palmInitFramework200_47.js
+++ /usr/palm/frameworks/mojo/builtins/palmInitFramework200_47.js
@@ -26666,7 +26666,7 @@
 this._addElementHighlight(event.centerX,event.centerY);
 }
 
-if(event.altKey){
+if(event.altKey || event.metaKey){
 
 
 

