diff --git a/etc/palm/browser-app.conf b/etc/palm/browser-app.conf
index c7fd0de..ef2b0d7 100755
--- a/etc/palm/browser-app.conf
+++ b/etc/palm/browser-app.conf
@@ -38,6 +38,7 @@ EnableAppCache=true
 AppCacheDbPath=/media/internal/dbdata/appcache
 AppCacheDbMaxSize=1073741824
 EnableNoSteCommonTagsCheck=true
+UserAgentOverride=Mozilla/5.0 (iPad; CPU OS 6_0 like Mac OS X) AppleWebKit/536.26 (KHTML, like Gecko) Version/6.0 Mobile/10A5355d Safari/8536.25
 
 # The 'EnableFitWidth' parameter lets us control the default behavior of ScrollView.
 EnableFitWidth=false
