diff --git a/usr/palm/applications/com.palm.app.browser/css/browser.css b/usr/palm/applications/com.palm.app.browser/css/browser.css
index a2dd6b4..5cb1fdb 100644
--- a/usr/palm/applications/com.palm.app.browser/css/browser.css
+++ b/usr/palm/applications/com.palm.app.browser/css/browser.css
@@ -1,3 +1,77 @@
+.gestureBarSide {
+  position: absolute;
+  width: 7px;
+  height: 25%;
+  z-index: 120;
+}
+.gestureBarTop {
+  position: absolute;
+  width: 25%;
+  height: 7px;
+  z-index: 120;
+}
+.gestureTop1{
+  top: 0px;
+  left: 0px;
+}
+.gestureTop2 {
+  top: 0px;
+  left: 25%;
+}
+.gestureTop3 {
+  top: 0px;
+  right: 25%;
+}
+.gestureTop4 {
+  top: 0px;
+  right: 0px;
+}
+.gestureLeft1 {
+  top: 0px;
+  left: 0px;
+}
+.gestureLeft2 {
+  top: 25%;
+  left: 0px;
+}
+.gestureLeft3 {
+  top: 50%;
+  left: 0px;
+}
+.gestureLeft4 {
+  bottom: 0px;
+  left: 0px;
+}
+.gestureRight1 {
+  top: 0px;
+  right: 0px;
+}
+.gestureRight2 {
+  top: 25%;
+  right: 0px;
+}
+.gestureRight3 {
+  top: 50%;
+  right: 0px;
+}
+.gestureRight4 {
+  bottom: 0px;
+  right: 0px;
+}
+.gestureOverlay1 {
+  background: #000;
+  opacity: 0.3;
+}
+.gestureOverlay2 {
+  background: #fff;
+  opacity: 0.3;
+}
+.gBot { border-bottom: 1px solid #aaa; }
+.gRight { border-right: 1px solid #aaa; }
+.gLeft { border-left: 1px solid #aaa; }
+.actionbar { z-index: 121; }
+
+
 /* Start Page */
 
 .startpage-list .enyo-datalist-client {
diff --git a/usr/palm/applications/com.palm.app.browser/source/ActionBar.js b/usr/palm/applications/com.palm.app.browser/source/ActionBar.js
index b680425..cbb7c78 100644
--- a/usr/palm/applications/com.palm.app.browser/source/ActionBar.js
+++ b/usr/palm/applications/com.palm.app.browser/source/ActionBar.js
@@ -29,7 +29,7 @@ enyo.kind({
 	},
 	components: [
     	{kind: "Control", showing: false, name: "title", className: "page-title enyo-text-ellipsis", content: "Untitled"},
-		{kind: enyo.HFlexBox, className: "menu-container", align: "center", components: [
+		{kind: enyo.HFlexBox, name: "gActionBar", className: "menu-container", align: "center", components: [
 			{kind: "ToolButton", name: "back", className: "actionbar-tool-button", icon: "images/chrome/menu-icon-back.png", onclick: "doBack"},
 			{kind: "ToolButton", name: "forward", className: "actionbar-tool-button", icon: "images/chrome/menu-icon-forward.png", onclick: "doForward"},
 			{kind: "ToolButton", name: "search", className: "actionbar-tool-button", flex: 1, kind: "URLSearch", onLoad: "doLoad", onStopLoad: "doStopLoad", onRefresh: "doRefresh"},
diff --git a/usr/palm/applications/com.palm.app.browser/source/Browser.js b/usr/palm/applications/com.palm.app.browser/source/Browser.js
index 6094e1c..de0f822 100644
--- a/usr/palm/applications/com.palm.app.browser/source/Browser.js
+++ b/usr/palm/applications/com.palm.app.browser/source/Browser.js
@@ -15,6 +15,8 @@ enyo.kind({
 		onAddToLauncher: "",
 		onShareLink: "",
 		onOpenBookmarks: "",
+		onJumpHistory: "",
+		onJumpBookmarks: "",
 		onPrint: "",
 		onUrlRedirected: "",
 		// called when user wants to leave the browser
@@ -56,6 +58,18 @@ enyo.kind({
 			onPrint: "doPrint",
 			minFontSize: 2,
 		},
+    {name: "gl1", kind: "Control", className: "gestureBarSide gestureLeft1", onmousedown: "gestureLeft1"},
+    {name: "gl2", kind: "Control", className: "gestureBarSide gestureLeft2", onmousedown: "gestureLeft2"},
+    {name: "gl3", kind: "Control", className: "gestureBarSide gestureLeft3", onmousedown: "gestureLeft3"},
+    {name: "gl4", kind: "Control", className: "gestureBarSide gestureLeft4", onmousedown: "gestureLeft4"},
+    {name: "gr1", kind: "Control", className: "gestureBarSide gestureRight1", onmousedown: "gestureRight1"},
+    {name: "gr2", kind: "Control", className: "gestureBarSide gestureRight2", onmousedown: "gestureRight2"},
+    {name: "gr3", kind: "Control", className: "gestureBarSide gestureRight3", onmousedown: "gestureRight3"},
+    {name: "gr4", kind: "Control", className: "gestureBarSide gestureRight4", onmousedown: "gestureRight4"},
+    {name: "gt1", kind: "Control", className: "gestureBarTop gestureTop1", onmousedown: "gestureTop1"},
+    {name: "gt2", kind: "Control", className: "gestureBarTop gestureTop2", onmousedown: "gestureTop2"},
+    {name: "gt3", kind: "Control", className: "gestureBarTop gestureTop3", onmousedown: "gestureTop3"},
+    {name: "gt4", kind: "Control", className: "gestureBarTop gestureTop4", onmousedown: "gestureTop4"},
 		{kind: "FindBar", showing: false, onFind: "find", onGoToPrevious: "goToPrevious", onGoToNext: "goToNext"},
 		{name: "context", kind: "BrowserContextMenu", onItemClick: "contextItemClick"},
 		{name: "dialog", kind: "VerticalAcceptCancelPopup", cancelCaption: "", components: [
@@ -86,7 +100,8 @@ enyo.kind({
 				{kind: "Button", flex: 1, caption: $L("Don't Trust"), response: "0", className: "enyo-button-dark", onclick: "closeSSLConfirmBox"}
 			]}
 		]},
-        {name: "sslCertDialog", kind: "CertificateDialog", onCertLoad: "enableViewSSLCertificate", onClose: "closeSSLCertificate"}
+    {name: "sslCertDialog", kind: "CertificateDialog", onCertLoad: "enableViewSSLCertificate", onClose: "closeSSLCertificate"},
+    {kind:"PalmService", name:"getTweaks", service:"palm://org.webosinternals.tweaks.prefs/", method:"get", onSuccess:"setGestureTweaks", onFailure:"setDefaultGestures"}
     ],
 	WebKitErrors: {
 		ERR_SYS_FILE_DOESNT_EXIST: 14,
@@ -105,6 +120,194 @@ enyo.kind({
 		if (window.PalmSystem) {
 			this.$.view.setIdentifier(enyo.windowParams.webviewId);
 		}
+    this.$.getTweaks.call({owner: "customBrowserGestures", keys: ["browserGestureLeft1","browserGestureLeft2","browserGestureLeft3","browserGestureLeft4","browserGestureRight1","browserGestureRight2","browserGestureRight3","browserGestureRight4","browserGestureTop1","browserGestureTop2","browserGestureTop3","browserGestureTop4","browserGestureOverlay"]});
+  	
+	  this.$.gt1.hide();
+	  this.$.gt2.hide();
+	  this.$.gt3.hide();
+	  this.$.gt4.hide();
+	  
+	},
+	gestureAction: function(gestureArea) {
+		switch (gestureArea) {
+			case "back":
+			  if(this.$.actionbar.canGoBack) {
+    	    this.goBack();
+       }
+				break;
+			case "forward":
+    	  this.goForward();
+				break;
+			case "refresh":
+    	  this.reloadClick();
+				break;
+			case "card":
+    	  this.openNewCard();
+			break;
+			case "bookmarks":
+    	  this.doJumpBookmarks();
+				break;
+			case "history":
+    	  this.doJumpHistory();
+				break;
+			case "fullscreen":
+			  if (!this.fullscreen) {
+      	  enyo.setFullScreen(true); 
+      	  this.gestureHideActionBarKeepProgressBar();
+      	  this.resize();
+      	  this.fullscreen = true;
+      	  this.$.gt1.show();
+      	  this.$.gt2.show();
+      	  this.$.gt3.show();
+      	  this.$.gt4.show();
+    	  } else {
+      	  enyo.setFullScreen(false);
+          this.gestureShowActionBar();
+      	  this.resize();
+      	  this.fullscreen = false;
+      	  this.$.gt1.hide();
+      	  this.$.gt2.hide();
+      	  this.$.gt3.hide();
+      	  this.$.gt4.hide();
+    	  }
+    	  
+				break;
+			case "toggleActionBar":
+			  if (this.actionBarHidden) {
+          this.gestureShowActionBar();
+      	  this.resize();
+      	  this.actionBarHidden = false;
+    	  } else {
+      	  this.gestureHideActionBarKeepProgressBar();
+      	  this.resize();
+      	  this.actionBarHidden = true;
+    	  }
+				break;
+			case "disable":
+			  break;
+			default:
+				break;
+		}
+	},
+	gestureLeft1: function() {
+    this.gestureAction(this.gestureLeft1Action);
+	},
+	gestureLeft2: function() {
+  	this.gestureAction(this.gestureLeft2Action);
+	},
+	gestureLeft3: function() {
+  	this.gestureAction(this.gestureLeft3Action);
+	},
+	gestureLeft4: function() {
+  	this.gestureAction(this.gestureLeft4Action);
+	},
+	gestureRight1: function() {
+    this.gestureAction(this.gestureRight1Action);
+	},
+	gestureRight2: function() {
+  	this.gestureAction(this.gestureRight2Action);
+	},
+	gestureRight3: function() {
+  	this.gestureAction(this.gestureRight3Action);
+	},
+	gestureRight4: function() {
+  	this.gestureAction(this.gestureRight4Action);
+	},
+	gestureTop1: function() {
+    this.gestureAction(this.gestureTop1Action);
+	},
+	gestureTop2: function() {
+  	this.gestureAction(this.gestureTop2Action);
+	},
+	gestureTop3: function() {
+  	this.gestureAction(this.gestureTop3Action);
+	},
+	gestureTop4: function() {
+  	this.gestureAction(this.gestureTop4Action);
+	},
+	setGestureTweaks: function(inSender, inResponse) {
+  	if((inResponse) && (inResponse.browserGestureLeft1)) {
+      this.gestureLeft1Action = inResponse.browserGestureLeft1;
+    }
+  	if((inResponse) && (inResponse.browserGestureLeft2)) {
+      this.gestureLeft2Action = inResponse.browserGestureLeft2;
+    }
+  	if((inResponse) && (inResponse.browserGestureLeft3)) {
+      this.gestureLeft3Action = inResponse.browserGestureLeft3;
+    }
+  	if((inResponse) && (inResponse.browserGestureLeft4)) {
+      this.gestureLeft4Action = inResponse.browserGestureLeft4;
+    }
+  	if((inResponse) && (inResponse.browserGestureRight1)) {
+      this.gestureRight1Action = inResponse.browserGestureRight1;
+    }
+  	if((inResponse) && (inResponse.browserGestureRight2)) {
+      this.gestureRight2Action = inResponse.browserGestureRight2;
+    }
+  	if((inResponse) && (inResponse.browserGestureRight3)) {
+      this.gestureRight3Action = inResponse.browserGestureRight3;
+    }
+  	if((inResponse) && (inResponse.browserGestureRight4)) {
+      this.gestureRight4Action = inResponse.browserGestureRight4;
+    }
+  	if((inResponse) && (inResponse.browserGestureTop1)) {
+      this.gestureTop1Action = inResponse.browserGestureTop1;
+    }
+  	if((inResponse) && (inResponse.browserGestureTop2)) {
+      this.gestureTop2Action = inResponse.browserGestureTop2;
+    }
+  	if((inResponse) && (inResponse.browserGestureTop3)) {
+      this.gestureTop3Action = inResponse.browserGestureTop3;
+    }
+  	if((inResponse) && (inResponse.browserGestureTop4)) {
+      this.gestureTop4Action = inResponse.browserGestureTop4;
+    }
+  	if((inResponse) && (inResponse.browserGestureOverlay)) {
+  	  this.gestureOverlay();
+    }
+	},
+	setDefaultGestures: function(inSender, inResponse) {
+  	this.gestureLeft1Action = "fullscreen";
+  	this.gestureLeft2Action = "fullscreen";
+  	this.gestureLeft3Action = "back";
+  	this.gestureLeft4Action = "back";
+  	
+  	this.gestureRight1Action = "bookmarks";
+  	this.gestureRight2Action = "bookmarks";
+  	this.gestureRight3Action = "forward";
+  	this.gestureRight4Action = "forward";
+  	
+  	this.gestureTop1Action = "fullscreen";
+  	this.gestureTop2Action = "fullscreen";
+  	this.gestureTop3Action = "fullscreen";
+  	this.gestureTop4Action = "fullscreen";
+  	this.gestureOverlay();
+	},
+	gestureOverlay: function() {
+    this.$.gl1.addClass("gestureOverlay1");
+    this.$.gl2.addClass("gestureOverlay2");
+    this.$.gl3.addClass("gestureOverlay1");
+    this.$.gl4.addClass("gestureOverlay2");
+    this.$.gr1.addClass("gestureOverlay2");
+    this.$.gr2.addClass("gestureOverlay1");
+    this.$.gr3.addClass("gestureOverlay2");
+    this.$.gr4.addClass("gestureOverlay1");
+    this.$.gt1.addClass("gestureOverlay2");
+    this.$.gt2.addClass("gestureOverlay1");
+    this.$.gt3.addClass("gestureOverlay2");
+    this.$.gt4.addClass("gestureOverlay1");
+	},
+	gestureHideActionBarKeepProgressBar: function() {
+	  this.$.actionbar.removeClass("enyo-toolbar actionbar enyo-vflexbox");
+    this.$.actionbar.setStyle("height:0px;");
+    this.$.actionbar.$.progressBar.setStyle("position:absolute;top:0px;z-index:121");
+    this.$.actionbar.$.gActionBar.hide();
+	},
+	gestureShowActionBar: function() {
+    this.$.actionbar.setClassName("enyo-toolbar actionbar enyo-vflexbox");
+    this.$.actionbar.setStyle("");
+    this.$.actionbar.$.progressBar.setStyle("");
+    this.$.actionbar.$.gActionBar.show();
 	},
 	resize: function() {
 		this.$.actionbar.resize();
diff --git a/usr/palm/applications/com.palm.app.browser/source/BrowserApp.js b/usr/palm/applications/com.palm.app.browser/source/BrowserApp.js
index 6d8659b..ef7c1da 100644
--- a/usr/palm/applications/com.palm.app.browser/source/BrowserApp.js
+++ b/usr/palm/applications/com.palm.app.browser/source/BrowserApp.js
@@ -40,6 +40,8 @@ enyo.kind({
 				onAddBookmark: "addBookmark",
 				onAddToLauncher: "showAddtoLauncherDialog",
 				onShareLink: "shareClick",
+				onJumpHistory: "jumpToHistory",
+				onJumpBookmarks: "jumpToBookmarks",
 				onOpenBookmarks: "bookmarksClick",
 				onPrint: "printClick",
 				onUrlRedirected: "openResource",
@@ -601,6 +603,14 @@ enyo.kind({
 		this.$.toaster.open();
 		this.$.drawerPane.selectView(this.$.drawerPane.getView());
 	},
+	jumpToHistory: function() {
+		this.$.toaster.open();
+		this.showHistory();
+	},
+	jumpToBookmarks: function() {
+		this.$.toaster.open();
+		this.showBookmarks();
+	},
 	preferencesClick: function() {
 		this.gotoView("preferences");
 	},
