diff --git a/usr/palm/applications/com.palm.app.browser/app/controllers/page-assistant.js b/usr/palm/applications/com.palm.app.browser/app/controllers/page-assistant.js
index 09c898f..cd8d426 100644
--- a/usr/palm/applications/com.palm.app.browser/app/controllers/page-assistant.js
+++ b/usr/palm/applications/com.palm.app.browser/app/controllers/page-assistant.js
@@ -492,13 +492,13 @@ PageAssistant.prototype.ready = function() {
 			// input.
 			self._addressBar.blur();
 		};
-
+/*
 		AppAssistant.Network.addNetworkCheckedMethods({
 			target: this,
 			methods: ['openUrl', '_goBack', '_goForward', '_reload'],
 			onNetworkDown: netAlert
 		});
-
+*/
 		// Never display the search results on first launch.
 		this._addressBar.enableTitleView();
 		this._addressBar.closeSearchResults();
@@ -509,7 +509,7 @@ PageAssistant.prototype.ready = function() {
 			this._addressBar.hide();
 			this.chrome.hide();
 		}
-
+/*
 		// On first launch we alway attempt to connect to the network if we are not online.
 		if (!AppAssistant.Network.online) {
 
@@ -526,7 +526,7 @@ PageAssistant.prototype.ready = function() {
 				}
 			}, self.controller.stageController);
 		}
-
+*/
 		if (this.currPageBookmark) {
 			this.openBookmark(this.currPageBookmark);
 		}
