diff --git a/usr/palm/applications/com.palm.app.browser/source/ActionBar.js b/usr/palm/applications/com.palm.app.browser/source/ActionBar.js
index b680425..30d29b1 100644
--- a/usr/palm/applications/com.palm.app.browser/source/ActionBar.js
+++ b/usr/palm/applications/com.palm.app.browser/source/ActionBar.js
@@ -24,6 +24,7 @@ enyo.kind({
 		onAddBookmark: "",
 		onAddToLauncher: "",
 		onShareLink: "",
+		onAddToReadOnTouch: "",
 		onOpenBookmarks: "",
 		onNewCard: ""
 	},
@@ -40,7 +41,8 @@ enyo.kind({
 		{name: "sharePopup", className: "launch-popup",  kind: "Menu", components: [
 			{caption: $L("Add Bookmark"), onclick: "doAddBookmark"},
 			{caption: $L("Share Link"), onclick: "doShareLink"},
-			{caption: $L("Add to Launcher"), onclick: "doAddToLauncher"}
+			{caption: $L("Add to Launcher"), onclick: "doAddToLauncher"},
+			{caption: $L("Add to ReadOnTouch"), onclick: "doAddToReadOnTouch"}
 		]},
 		{name: "progressBar", kind: "ProgressBar", className: "url-progress invisible", animatePosition: false},
 	],
diff --git a/usr/palm/applications/com.palm.app.browser/source/Browser.js b/usr/palm/applications/com.palm.app.browser/source/Browser.js
index f2d5a46..7c910b0 100644
--- a/usr/palm/applications/com.palm.app.browser/source/Browser.js
+++ b/usr/palm/applications/com.palm.app.browser/source/Browser.js
@@ -14,6 +14,7 @@ enyo.kind({
 		onAddBookmark: "",
 		onAddToLauncher: "",
 		onShareLink: "",
+		onAddToReadOnTouch: "",
 		onOpenBookmarks: "",
 		onPrint: "",
 		onUrlRedirected: "",
@@ -33,6 +34,7 @@ enyo.kind({
 			onAddBookmark: "doAddBookmark",
 			onAddToLauncher: "doAddToLauncher",
 			onShareLink: "doShareLink",
+			onAddToReadOnTouch: "doAddToReadOnTouch",
 			onOpenBookmarks: "doOpenBookmarks",
 			onNewCard: "openNewCard"
 		},
@@ -318,6 +320,19 @@ enyo.kind({
 		this.log(params.text);
 		this.$.launchApplicationService.call({id: "com.palm.app.email", params: params});
 	},
+	addToReadOnTouch: function(inUrl, inTitle) {
+		this.log(inUrl, inTitle);
+		var params = {
+			action: "addLink",
+           title: inTitle,
+			url: inUrl,
+		};
+		this.log(params.text);
+		this.$.launchApplicationService.call({id: "com.sven-ziegler.readontouch", params: params});
+	},
+	addLinkToReadOnTouch: function(inTapInfo) {
+		this.addToReadOnTouch(inTapInfo.linkUrl);
+	},
 	copyToPhotosClick: function(inTapInfo, inPosition) {
 		this.viewCall("saveImageAtPoint", [inPosition.left, inPosition.top, "/media/internal",
 			enyo.hitch(this, "finishCopyToPhotos", inTapInfo)]);
diff --git a/usr/palm/applications/com.palm.app.browser/source/BrowserApp.js b/usr/palm/applications/com.palm.app.browser/source/BrowserApp.js
index 9a4bf6f..d8240f0 100644
--- a/usr/palm/applications/com.palm.app.browser/source/BrowserApp.js
+++ b/usr/palm/applications/com.palm.app.browser/source/BrowserApp.js
@@ -45,7 +45,8 @@ enyo.kind({
 				onUrlRedirected: "openResource",
 				onClose: "browserClosed",
 				onShow: "browserShown",
-				onHide: "browserHidden"
+				onHide: "browserHidden",
+				onAddToReadOnTouch: "addToReadOnTouch"
 			},
 			{name: "preferences", kind: "Preferences",
 				onPreferenceChanged: "preferenceChanged",
@@ -581,6 +582,9 @@ enyo.kind({
 	shareClick: function() {
 		this.$.browser.shareLink(this.url, this.title);
 	},
+	addToReadOnTouch: function() {
+		this.$.browser.addToReadOnTouch(this.url, this.title);
+	},
 	bookmarksClick: function() {
 		this.$.toaster.open();
 		this.$.drawerPane.selectView(this.$.drawerPane.getView());
diff --git a/usr/palm/applications/com.palm.app.browser/source/BrowserContextMenu.js b/usr/palm/applications/com.palm.app.browser/source/BrowserContextMenu.js
index b029eba..38ff9cc 100644
--- a/usr/palm/applications/com.palm.app.browser/source/BrowserContextMenu.js
+++ b/usr/palm/applications/com.palm.app.browser/source/BrowserContextMenu.js
@@ -11,7 +11,8 @@ enyo.kind({
 	linkItems: [
 		{caption: $L("Open In New Card"), value:"newCardClick"},
 		{caption: $L("Share Link"), value:"shareLinkClick"},
-		{caption: $L("Copy URL"), value:"copyLinkClick"}
+		{caption: $L("Copy URL"), value:"copyLinkClick"},
+		{caption: $L("Add To ReadOnTouch"), value:"addLinkToReadOnTouch"}
 	],
 	imageItems: [
 		{caption: $L("Copy To Photos"), value: "copyToPhotosClick"},
