diff --git a/etc/palm/lunaAnimations.conf b/etc/palm/lunaAnimations.conf
index 199b2c9..5f77b6c 100644
--- a/etc/palm/lunaAnimations.conf
+++ b/etc/palm/lunaAnimations.conf
@@ -3,18 +3,18 @@ normalFPS=35
 slowFPS=20
 
 [Cards]
-cardLaunchDuration=400
-cardLaunchCurve=40
+cardLaunchDuration=100
+cardLaunchCurve=50
 cardSlideDuration=300
 cardSlideCurve=10
-cardMaximizeDuration=300
-cardMaximizeCurve=10
-cardMinimizeDuration=350
-cardMinimizeCurve=6
-cardDeleteDuration=300
-cardDeleteCurve=6
-cardScootAwayOnLaunchDuration=400
-cardScootAwayOnLaunchCurve=30
+cardMaximizeDuration=100
+cardMaximizeCurve=40
+cardMinimizeDuration=100
+cardMinimizeCurve=50
+cardDeleteDuration=150
+cardDeleteCurve=40
+cardScootAwayOnLaunchDuration=100
+cardScootAwayOnLaunchCurve=40
 cardMoveNormalDuration=100
 cardMoveNormalCurve=30
 cardMoveOverviewDuration=100
@@ -25,19 +25,19 @@ cardGroupReorderDuration=500
 cardGroupReorderCurve=6
 cardSwitchReachEndMaximizedDuration=200
 cardSwitchReachEndMaximizedCurve=6
-cardSwitchMaximizedDuration=350
+cardSwitchMaximizedDuration=150
 cardSwitchMaximizedCurve=6
-cardPrepareAddDuration=150
+cardPrepareAddDuration=50
 cardBeforeAddDelay=0
-cardAddMaxDuration=750
+cardAddMaxDuration=250
 modalCardAddMaxDuration=10
-cardLoadingPulsePauseDuration=1000
-cardLoadingPulseDuration=1000
+cardLoadingPulsePauseDuration=100
+cardLoadingPulseDuration=400
 cardLoadingPulseCurve=1
 cardLoadingCrossFadeDuration=300
 cardLoadingCrossFadeCurve=0
-cardLoadingTimeBeforeShowingPulsing=900
-cardTransitionDuration=300
+cardLoadingTimeBeforeShowingPulsing=150
+cardTransitionDuration=100
 cardTransitionCurve=20
 cardGhostDuration=750
 cardGhostCurve=10
@@ -51,13 +51,13 @@ positiveSpaceChangeDuration=400
 positiveSpaceChangeCurve=6
 
 [Launcher]
-quickLaunchDuration=350
+quickLaunchDuration=150
 quickLaunchCurve=6
-quickLaunchSlideToStacheDuration=150
+quickLaunchSlideToStacheDuration=90
 quickLaunchSlideToStacheCurve=6
 quickLaunchFadeDuration=200
 quickLaunchFadeCurve=6
-launcherDuration=350
+launcherDuration=150
 launcherCurve=15
 launcherNormalModeArrowSlideDuration=350
 launcherNormalModeArrowSlideCurve=22
