diff --git a/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js b/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
index 0e239cc..c060b40 100644
--- a/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
+++ b/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
@@ -2314,7 +2314,7 @@ handleWANNotifications: function(payload) {
 				
 				//Icon custmization
 				if(payload.networktype === 'evdo' && this.show3GForEvdo)
-					payload.networktype = 'evdo3g';
+					payload.networktype = 'evdo';
 				
 				if(payload.connectedservices[i].connectstatus == 'active' && payload.dataaccess == "usable") {
 					Mojo.Dom.get('datanet').className = 'connected-'+payload.networktype;
