diff --git a/usr/lib/luna/system/luna-systemui/app/controllers/devicemenu-assistant.js b/usr/lib/luna/system/luna-systemui/app/controllers/devicemenu-assistant.js
index 5beb152..cd23ac4 100644
--- a/usr/lib/luna/system/luna-systemui/app/controllers/devicemenu-assistant.js
+++ b/usr/lib/luna/system/luna-systemui/app/controllers/devicemenu-assistant.js
@@ -129,6 +129,12 @@ var DevicemenuAssistant = Foundations.Class.create({
 			this.controller.get('dm_wifi').hide();
 			this.controller.hideWidgetContainer(this.controller.get('dm_wifi'));
 		}
+		
+		//setup temperature label
+		var phone = this.controller.get("dm_phone");
+		phone.id = "batteryRow";
+		phone.innerHTML = '<span id="dm_phone">' + phone.innerHTML + '</span>' +
+				'<div class="label right" id="tempmsg" style="font-size:0.9em;">&nbsp;</div>';
 
 		//set up the the menu div
 		this.scrim = this.controller.sceneElement.querySelector('div[x-mojo-scrim]');
@@ -1582,6 +1588,31 @@ var DevicemenuAssistant = Foundations.Class.create({
 	activateStage: function() {
 		if (!this._stageRef) {
 			this._stageRef = 1;
+			this.controller.serviceRequest('palm://ca.canucksoftware.systoolsmgr', {
+				method: 'cpuTemperature',
+				onSuccess: function(payload) {
+					if(payload.value) {
+						this.controller.get("tempmsg").innerHTML = payload.value + "&deg;C";
+					}
+				}.bind(this),
+				onFailure: function(err) {
+					this.controller.serviceRequest('palm://com.palm.bus/signal/', {
+						method: 'addmatch',
+						parameters: {
+							"category":"/com/palm/power",
+							"method":"batteryStatus"
+						},
+						onSuccess: function(payload2) {
+							if(payload2.temperature_C) {
+								this.controller.get("tempmsg").innerHTML = payload2.temperature_C + "&deg;C";
+							}
+						}.bind(this)
+					});
+					this.controller.serviceRequest('palm://com.palm.power/com/palm/power/', {
+						method: 'batteryStatusQuery'
+					});
+				}.bind(this)
+			});
 			this.controller.stageController.activate();
 		}
 	},
