diff --git a/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js b/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
index 21f3714..ea3cbfd 100644
--- a/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
+++ b/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
@@ -388,7 +388,7 @@ updateAppMenuTitle: function(payload) {
 			if (this.phoneAppName)
 				Mojo.Dom.get('appname').innerHTML = this.phoneAppName;
 			else
-				Mojo.Dom.get('appname').innerHTML = this.carrierText;
+				Mojo.Dom.get('appname').innerHTML = payload.appmenu || payload.title;
 			this.phoneIsForeground = true;
 		}
 		else if (this.isPreferences(payload))  {
@@ -492,7 +492,7 @@ handleStatusChange: function(payload) {
 	if (payload.locked != null) {
 		if (payload.locked) {
 			Mojo.Dom.get('clock').hide();
-			Mojo.Dom.get('date').show();
+			Mojo.Dom.get('date').hide();
 			this.deviceLocked = true;
 		}
 		else {
@@ -818,8 +818,8 @@ handlePowerQueryNotifications: function(payload){
 				this.updateRSSIIcon("flight-mode");
 			}
 			else {
-				this.carrierText = $L("Phone offline");
-				this.currentCarrierText = "Phone offline";
+				this.carrierText = "Palm";
+				this.currentCarrierText = this.carrierText;
 				Mojo.Dom.get('carrier').innerHTML = this.carrierText;
 			}							
 			Mojo.Dom.get('roaming').hide();
@@ -850,8 +850,8 @@ handleNetworkStatusQueryNotification: function(payload) {
 				this.currentCarrierText = "Palm";
 			}
 			else {
-				this.carrierText = $L(payload.extended.networkName.escapeHTML());
-				this.currentCarrierText = payload.extended.networkName.escapeHTML();
+				this.carrierText = "Palm";
+				this.currentCarrierText = this.carrierText;
 			}
 			Mojo.Dom.get('carrier').innerHTML = this.carrierText;	   	
 			if(payload.extended.registration == 'home') {			
@@ -1288,8 +1288,8 @@ handletelephonyNotifications: function(payload) {
 				this.updateRSSIIcon("flight-mode");
 			}
 			else {
-				this.carrierText = $L("Phone offline");
-				this.currentCarrierText = "Phone offline";
+				this.carrierText = "Palm";
+				this.currentCarrierText = this.carrierText;
 				Mojo.Dom.get('carrier').innerHTML = this.carrierText;
 			}
 			Mojo.Dom.get('datanet').className = 'off';
@@ -1314,8 +1314,8 @@ handletelephonyNotifications: function(payload) {
 					this.currentCarrierText = "Palm";
 				}
 				else {
-					this.carrierText = $L(networkMsg.networkName.escapeHTML());
-					this.currentCarrierText = networkMsg.networkName.escapeHTML();
+					this.carrierText = "Palm";
+					this.currentCarrierText = this.carrierText;
 				}
 				Mojo.Dom.get('carrier').innerHTML = this.carrierText;
 				this.simlocked = false;
@@ -3171,6 +3171,8 @@ updateClockEveryMinute: function() {
 	
 	//Mojo.Log.error("UpdateClockEveryMinute " + date);
 	
+	this.carrierText = "Palm";
+	Mojo.Dom.get('carrier').innerHTML = this.carrierText;
 	if(Mojo.Format.using12HrTime()) 
 		format = Mojo.Format.getFormatHash()['shortTime12'];
 	else
