diff --git a/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js b/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
index 0e239cc..1ce4dd4 100644
--- a/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
+++ b/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
@@ -2988,7 +2988,19 @@ updateClockEveryMinute: function() {
 	else
 		format = Mojo.Format.getFormatHash()['shortTime24'];
 	format = format.replace(/\s?a\s?/,"");	
-	Mojo.Dom.get('clock').innerHTML = Mojo.Format.formatDate(date, {time: format});				
+	var suffix = "";
+	if(Mojo.Format.using12HrTime()){
+		if(new Date().getHours() < 12) {
+			suffix = "AM";
+		} else {
+			suffix = "PM";
+		}
+	}
+   var prefixDate = Mojo.Format.formatDate(date, {date:'short'});
+   prefixDate = prefixDate.substr(0, prefixDate.lastIndexOf('/'));
+   clock = Mojo.Dom.get('clock');
+	clock.innerHTML = prefixDate+'&nbsp; '+Mojo.Format.formatDate(date, {time: format}) + ' ' + suffix;
+	clock.style.wordSpacing = "0em";
 	Mojo.Dom.get('today').innerHTML = Mojo.Format.formatDate(date,{date:'short'});
 },
 
