diff --git a/usr/lib/luna/system/luna-systemui/app/controllers/devicemenu-assistant.js b/usr/lib/luna/system/luna-systemui/app/controllers/devicemenu-assistant.js
index 2a42b1a..12d21e9 100644
--- a/usr/lib/luna/system/luna-systemui/app/controllers/devicemenu-assistant.js
+++ b/usr/lib/luna/system/luna-systemui/app/controllers/devicemenu-assistant.js
@@ -29,8 +29,6 @@ var DevicemenuAssistant = Class.create({
 			
 		var date = new Date();
 		this.controller.get('dm_date').innerHTML = Mojo.Format.formatDate(date,{date:'EEE '}) +  Mojo.Format.formatDate(date,{date:'long'});
-		var msg = $L("Battery: #{batlevel}");
-		this.controller.get('dm_battery').innerHTML = new Template(msg).evaluate({"batlevel": Mojo.Format.formatPercent(this.barAssistant.getBatteryLevel())});
 						
 		this.modesModel = {scrollbars: false, mode: "vertical"},
 		this.controller.setupWidget('devicescroller', {}, this.modesModel);
diff --git a/usr/lib/luna/system/luna-systemui/app/views/devicemenu/devicemenu-scene.html b/usr/lib/luna/system/luna-systemui/app/views/devicemenu/devicemenu-scene.html
index 933094e..f0fe511 100644
--- a/usr/lib/luna/system/luna-systemui/app/views/devicemenu/devicemenu-scene.html
+++ b/usr/lib/luna/system/luna-systemui/app/views/devicemenu/devicemenu-scene.html
@@ -21,13 +21,6 @@
 				<div class="palm-section-divider">
 				</div> 
 				
-				<div class="palm-row"><div class="palm-row-wrapper">
-					<div id="dm_battery" class="title dim"></div>
-				</div></div> 
-				
-				<div class="palm-section-divider">
-				</div>               
-				
 				<div class="palm-row" id="dm_wifi" x-mojo-tap-highlight='momentary'><div class="palm-row-wrapper">
 					<div class="title truncating-text">
 						<div class="label right" id="wifimsg">&nbsp;</div>                               
