diff --git a/usr/lib/luna/system/luna-systemui/app/controllers/devicemenu-assistant.js b/usr/lib/luna/system/luna-systemui/app/controllers/devicemenu-assistant.js
index eef1fb2..94761d4 100644
--- a/usr/lib/luna/system/luna-systemui/app/controllers/devicemenu-assistant.js
+++ b/usr/lib/luna/system/luna-systemui/app/controllers/devicemenu-assistant.js
@@ -130,6 +130,13 @@ var DevicemenuAssistant = Foundations.Class.create({
 			this.controller.hideWidgetContainer(this.controller.get('dm_wifi'));
 		}
 
+		//Hide Cycle WiFi Menu if WiFi is not present.
+		if(!Mojo.Environment.DeviceInfo.wifiAvailable) {
+			this.controller.get('dm_wifix').hide();
+			this.controller.hideWidgetContainer(this.controller.get('dm_wifix'));
+		}
+
+
 		//set up the the menu div
 		this.scrim = this.controller.sceneElement.querySelector('div[x-mojo-scrim]');
 		this.systemmenu = this.controller.sceneElement.querySelector('#palm-device-menu');
@@ -155,6 +162,7 @@ var DevicemenuAssistant = Foundations.Class.create({
 		this.controller.get('btlist').addEventListener(Mojo.Event.listTap,this.handleBTTap.bind(this));
 		this.controller.get('vpnlist').addEventListener(Mojo.Event.listTap,this.handleVPNTap.bind(this));		
 		this.controller.get('dm_wifi').addEventListener(Mojo.Event.tap, this.togglewifiList.bind(this));
+		this.controller.get('dm_wifix').addEventListener(Mojo.Event.tap, this.cycleWifi.bind(this));
 		this.controller.get('dm_vpn').addEventListener(Mojo.Event.tap, this.toggleVPNList.bind(this));
 		this.controller.get('dm_bluetooth').addEventListener(Mojo.Event.tap, this.togglebluetoothList.bind(this));
 		this.controller.get('wifi_radio').addEventListener(Mojo.Event.tap, this.toggleWifiRadio.bind(this));
@@ -780,6 +788,24 @@ var DevicemenuAssistant = Foundations.Class.create({
 		this.wifidrawer.mojo.setOpenState(!this.wifidrawer.mojo.getOpenState());						
 	},
 	
+	//Cycles the WiFi OFF and ON to refresh and improve signal.		
+	cycleWifi: function(event) {
+		if(RadioState.get('wifi')) {
+			this.spinOff();
+		
+			if(this.wifiFindNetworksReq) {
+				this.wifiFindNetworksReq.cancel();
+				this.wifiFindNetworksReq = undefined;
+			}
+			WiFiService.setState("disabled",null,null);	
+			WiFiService.setState("enabled",null,null);	
+		}			
+		else {
+			this.spinOn();
+			WiFiService.setState("enabled",null,null);	
+		}		
+	},
+	
 	/*
 	 * Handler for Wifi Preference launch
 	 */
diff --git a/usr/lib/luna/system/luna-systemui/app/views/devicemenu/devicemenu-scene.html b/usr/lib/luna/system/luna-systemui/app/views/devicemenu/devicemenu-scene.html
index be38f0d..75ccd5d 100644
--- a/usr/lib/luna/system/luna-systemui/app/views/devicemenu/devicemenu-scene.html
+++ b/usr/lib/luna/system/luna-systemui/app/views/devicemenu/devicemenu-scene.html
@@ -95,7 +95,16 @@
 						</div>
 					</div>
 				</div> 
+
+				<div class="palm-section-divider"></div>
 				
+				<div class="palm-row" id="dm_wifix" x-mojo-tap-highlight='momentary'><div class="palm-row-wrapper">
+					<div class="title truncating-text">
+						<div class="label right" id="cycleWiFi">&nbsp;</div>
+						<span x-mojo-loc=''>Cycle Wi-Fi</span>
+					</div>
+				</div></div>
+
 				<div class="palm-section-divider"></div>
 				
 				<div class="palm-row" id="dm_vpn" x-mojo-tap-highlight='momentary'><div class="palm-row-wrapper">
