diff --git a/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js b/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
index 786f7ae..618ac15 100644
--- a/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
+++ b/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
@@ -3,6 +3,7 @@
 var BarAssistant = Class.create({
 	
 	setup: function() {
+		this.modeSwitcherConfig = {};
 						
 		this.showMSMWarning = false;
 		this.showMediaWarning = false;	
@@ -527,6 +528,9 @@ sysServiceInit: function() {
 	
 	this.getRoamingPreference();
 		
+	//Subscribe to Mode Switcher configuration
+	this.initModeSwitcher();
+
 	//Get the Airplane Mode initial values
 	this.isAirplaneModeEnabled();
 						
@@ -4570,6 +4574,58 @@ isAppRestoreRequired: function() {
 	Mojo.Log.info("SystemUI-App Restore Reading from the Cookie " + this.appRestoreNeeded);
 },
 
+// Mode Switcher Support
+
+initModeSwitcher: function() {
+	this.modesConfigRequest = new Mojo.Service.Request("palm://com.palm.systemservice/", {
+		method: 'getPreferences', 
+		parameters: {"subscribe":true,"keys":["modeSwitcher", "modesConfig", 
+			"currentMode", "defaultMode", "modifierModes"]},
+		onSuccess: this.handleModeSwitcherUpdate.bind(this) });	
+},
+
+handleModeSwitcherUpdate: function(payload) {
+	if(payload.modeSwitcher != undefined) {
+		this.modeSwitcherConfig.modeSwitcher = payload.modeSwitcher;
+	}
+		
+	if(payload.modesConfig != undefined) {
+		this.modeSwitcherConfig.modesConfig = payload.modesConfig;
+	}
+
+	if(payload.currentMode != undefined) {
+	 	if(payload.currentMode != "none") {
+			this.modeSwitcherConfig.currentMode = payload.currentMode;
+			
+			if(payload.currentMode.type == "default") {
+				this.statusText = null;
+				$('carrier').innerHTML = this.carrierText;
+			}
+			else {
+				this.statusText = payload.currentMode.name;
+				$('carrier').innerHTML = this.statusText;
+			}
+		}
+		else {
+			this.statusText = null;
+			$('carrier').innerHTML = this.carrierText;
+		}
+	}
+
+	if(payload.defaultMode != undefined) {
+	 	if(payload.defaultMode != "none")
+			this.modeSwitcherConfig.defaultMode = payload.defaultMode;
+	}
+	
+	if(payload.modifierModes != undefined) {
+		this.modeSwitcherConfig.modifierModes = payload.modifierModes;
+	}
+},
+
+//Returns the MS config. Called by Mode Menu.
+getModeSwitcherConfig: function() {	
+	return this.modeSwitcherConfig;
+},
 
 //  Main Variables
 	timeAMPM: true,
diff --git a/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js.rej b/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js.rej
new file mode 100644
index 0000000..d0dc585
--- /dev/null
+++ b/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js.rej
@@ -0,0 +1,41 @@
+***************
+*** 515,533 ****
+  
+  showStatusMenu: function(edit) {
+  
+- //	var stageController = Mojo.Controller.getAppController().getStageProxy("StatusMenu");
+- //	if (stageController) {
+- //		this.hideSystemMenus("status");
+- //		stageController.delegateToSceneAssistant("toggleStatusMenu", edit);			
+- //	} else {
+- //		this.statusMenuCreatedHandler = this.statusMenuCreated.bind(this);
+- //		Mojo.Controller.getAppController().createStageWithCallback({name:"StatusMenu",lightweight: true}, 
+- //			this.statusMenuCreatedHandler, "menu");
+- //	}
+  },
+  
+  statusMenuCreated: function(stageController) {
+- //	stageController.pushScene('statusmenu',this);
+  },
+  
+  // Today menu
+--- 516,534 ----
+  
+  showStatusMenu: function(edit) {
+  
++ 	var stageController = Mojo.Controller.getAppController().getStageProxy("StatusMenu");
++ 	if (stageController) {
++ 		this.hideSystemMenus("status");
++ 		stageController.delegateToSceneAssistant("toggleStatusMenu", edit);			
++ 	} else {
++ 		this.statusMenuCreatedHandler = this.statusMenuCreated.bind(this);
++ 		Mojo.Controller.getAppController().createStageWithCallback({name:"StatusMenu",lightweight: true}, 
++ 			this.statusMenuCreatedHandler, "menu");
++ 	}
+  },
+  
+  statusMenuCreated: function(stageController) {
++ 	stageController.pushScene('statusmenu',this);
+  },
+  
+  // Today menu
diff --git a/usr/lib/luna/system/luna-systemui/sources.json b/usr/lib/luna/system/luna-systemui/sources.json
index 8290025..b6a018f 100644
--- a/usr/lib/luna/system/luna-systemui/sources.json
+++ b/usr/lib/luna/system/luna-systemui/sources.json
@@ -6,6 +6,9 @@
     "source": "app\/controllers\/bar-assistant.js"
   },  
   {
+    "source": "app\/controllers\/statusmenu-assistant.js"
+  },  
+  {
     "source": "app\/controllers\/devicemenu-assistant.js"
   },  
   {
