diff --git a/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js b/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
index d4ae043..8513245 100644
--- a/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
+++ b/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
@@ -13,7 +13,7 @@ var BarAssistant = Class.create({
 		this.airplaneModeInitialQueryDone = false;
 		this.carrierText = "Palm";
 		this.airplaneModeTriggered = false;
-		this.show3GForEvdo = false;
+		this.show3GForEvdo = true;
 		this.carrierTextEnglish = "Palm";
 		this.stringTable = null;
 		
