diff --git a/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js b/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
index d4ae043..b96ef27 100644
--- a/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
+++ b/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
@@ -2251,17 +2251,20 @@ handleWIFINotifications: function(payload) {
 			case 'associated': 
 				$('wifi').className = 'connecting';				
 				this.wifiState = payload.networkInfo.ssid;
+				$('carrier').innerHTML = this.wifiState;
 			 	break;
 			case 'ipFailed':			
 			case 'notAssociated':
 			case 'associationFailed':
 				this.wifiState = 'On';				
 				this.wifiConnected = false;				
+				$('carrier').innerHTML = this.carrierText;
 				$('wifi').className = 'on';			
 				break;
 			case 'ipConfigured':
 				this.wifiConnected = true;
 				this.wifiState = payload.networkInfo.ssid;
+				$('carrier').innerHTML = this.wifiState;
 				if(payload.networkInfo.signalBars != undefined){
 					$('wifi').className = 'bar-'+payload.networkInfo.signalBars;					
 				}
@@ -2281,6 +2284,7 @@ handleWIFINotifications: function(payload) {
 	}
 	else if (payload.status == 'serviceEnabled') {
 		this.wifiConnected = false;
+		$('carrier').innerHTML = this.carrierText;
 		$('wifi').className = 'on';
 		this.wifiState = 'On';
 		this.wifiRadio = true;
@@ -2293,6 +2297,7 @@ handleWIFINotifications: function(payload) {
 	}
 	else if (payload.status == 'serviceDisabled') {
 		this.wifiConnected = false;
+		$('carrier').innerHTML = this.carrierText;
 		$('wifi').className = 'off';
 		this.wifiState = 'Off';
 		this.wifiRadio = false;	
