diff --git a/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js b/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
index 5dc60f6..5a577e5 100644
--- a/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
+++ b/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
@@ -25,6 +25,7 @@ var BarAssistant = Class.create({
 		$('datanet').className = 'off';
 		$('wifi').className = 'off';
 		$('bluetooth').className = 'off';
+		$('gps').className = 'off';
 		$('roaming').hide();
 		$('tty').hide();
 		$('hac').hide();
@@ -100,6 +101,20 @@ var BarAssistant = Class.create({
 		this.currentLocale = Mojo.Locale.getCurrentLocale();				
 },
 
+updateGPSIcon: function() {
+	this.controller.serviceRequest('palm://com.palm.location', {
+		method: 'getUseGps',
+		parameters: {},
+		onSuccess: function(status) {
+			if (status.useGps) {
+				$('gps').className = 'on';
+			} else {
+				$('gps').className = 'off';
+			}
+		}.bind(this)
+	});
+},
+
 refreshCarrierText: function() {
 	$('carrier').innerHTML = this.$L(this.carrierTextEnglish);
 },
@@ -1776,6 +1791,7 @@ closePbapAlert: function() {
  */
 handlePowerNotifications: function(payload) {
 	
+	this.updateGPSIcon();
 	Mojo.Log.info("SystemUI Power Payload "+ Object.toJSON(payload));
 	if(!this.powerdServiceUp) {
 		Mojo.Log.error("SystemUI PowerD down -- ignoring the notification");
diff --git a/usr/lib/luna/system/luna-systemui/app/views/bar/bar-scene.html b/usr/lib/luna/system/luna-systemui/app/views/bar/bar-scene.html
index c1daf79..41b1245 100644
--- a/usr/lib/luna/system/luna-systemui/app/views/bar/bar-scene.html
+++ b/usr/lib/luna/system/luna-systemui/app/views/bar/bar-scene.html
@@ -29,6 +29,8 @@
 		</div>
 		<div id="bluetooth">
 		</div>
+		<div id="gps">
+		</div>
 		<div id="tty">
 		</div>
 		<div id="callforward">
diff --git a/usr/lib/luna/system/luna-systemui/stylesheets/systemui.css b/usr/lib/luna/system/luna-systemui/stylesheets/systemui.css
index a0dee3c..355dd9d 100644
--- a/usr/lib/luna/system/luna-systemui/stylesheets/systemui.css
+++ b/usr/lib/luna/system/luna-systemui/stylesheets/systemui.css
@@ -388,6 +388,18 @@ body.palm-default {
 		width: 17px;
 }
 
+#gps.off {
+		display: none;
+}
+
+#gps.on {
+		background: url(../images/gps-on.png) center center no-repeat;
+}
+
+#gps.active{
+		background: url(../images/gps-active.png) center center no-repeat;
+}
+
 .brique {
 		width: 320px;
 		height: 480px;
