diff --git a/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js b/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
index a4a150e..1a1faa3 100644
--- a/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
+++ b/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
@@ -1657,10 +1657,36 @@ closePbapAlert: function() {
 	}	
 },
 
+/* Format battery level percent */
+getBatteryLevelPercent: function(percent, color, image, charging) { 	
+	if (color) {
+		if (charging == 1)
+			bColor = '#00FFCC'; //teal
+		else if (percent >= 13) 
+			bColor = '#FFFFFF'; //white
+		else if(percent <= 12)
+			bColor = '#FF0000'; //red
+		else 
+			bColor = 'white';
+		
+		if (image != undefined)
+			var fPercent = '<span style="color: ' + bColor + ';">' + percent + unescape('%') + '</span>';
+		else 
+			var fPercent = '<span style="position: relative; bottom: 2px; color: ' + bColor + ';">' + percent + unescape('%') + '</span>';
+	}
+	else {
+		var fPercent = percent + unescape('%');
+	}
+	return fPercent;
+},
+
 /*
  * Handle power and charging notifications
  */
 handlePowerNotifications: function(payload) {
+	// user definable setting
+	var batteryLevelImage = 2; 	// 2 displays the image on left and numerical percentage on right
+	var batteryLevelColor = 1; // Set to 0 for white, 1 for color
 	
 	Mojo.Log.info("SystemUI Power Payload "+ Object.toJSON(payload));
 	if(!this.powerdServiceUp) {
@@ -1684,7 +1710,15 @@ handlePowerNotifications: function(payload) {
 			// Show the battery level if not charging
 			if (!this.isCharging) {
 				Mojo.Log.info("SystemUI- Is Charging "+ this.isCharging + " Battery level "+ i);
+				if (batteryLevelImage == 0) {
 				$('power').className = 'battery-' + i;
+				} else if(batteryLevelImage == 1) {
+					$('power').className = 'battery-percent';
+					$('power').innerHTML = this.getBatteryLevelPercent(payload.percent_ui, batteryLevelColor, true, 0);
+				} else if(batteryLevelImage == 2) {
+					$('power').className = 'battery-percentimage';
+					$('power').innerHTML = '<img src="images/battery-' + i + '.png" />' + this.getBatteryLevelPercent(payload.percent_ui, batteryLevelColor);
+				}
 				
 				//Show Banner Message if the Battery level is below 20%
 				var batteryalert = $L("#{percent_ui}% battery remaining").interpolate(payload);	
@@ -1716,10 +1750,23 @@ handlePowerNotifications: function(payload) {
 			}
 			else {
 				Mojo.Log.info("SystemUI- Is Charging "+ this.isCharging );
-				if(payload.percent_ui == 100)
+				if (batteryLevelImage == 0) {
+					if (payload.percent_ui == 100)				
 					$('power').className = 'battery-charged';
 				else
 					$('power').className = 'battery-charging-' + i;
+				} else if (batteryLevelImage == 1) {
+					$('power').className = 'battery-percent';
+					$('power').innerHTML = this.getBatteryLevelPercent(payload.percent_ui, batteryLevelColor, true, 1);
+				} else if(batteryLevelImage == 2) {
+					if (payload.percent_ui == 100) {
+						$('power').className = 'battery-percentimage-charged';
+						$('power').innerHTML = '<img src="images/battery-charged.png" />' + this.getBatteryLevelPercent(payload.percent_ui, batteryLevelColor, true, 1);
+					} else {
+						$('power').className = 'battery-percentimage';
+						$('power').innerHTML = '<img src="images/battery-charging-' + i + '.png" />' + this.getBatteryLevelPercent(payload.percent_ui, batteryLevelColor, true, 1);
+					}
+				}
 					
 			}
 			
@@ -1756,10 +1803,23 @@ handlePowerNotifications: function(payload) {
 						
 		if (this.isCharging) {
 			Mojo.Log.info("SystemUI- Charger Notification "+ this.isCharging + " Last battery level "+ this.lastBatteryLevel);
-			if(this.batteryLevel == 100)
+			if (batteryLevelImage == 0) {
+				if (this.batteryLevel == 100)
 				$('power').className = 'battery-charged';
 			else
 				$('power').className = 'battery-charging-' + this.lastBatteryLevel;
+			} else if(batteryLevelImage == 1) {
+				$('power').className = 'battery-percent';
+				$('power').innerHTML = this.getBatteryLevelPercent(this.batteryLevel, batteryLevelColor, true, 1);
+			} else if(batteryLevelImage == 2) {
+				if (this.batteryLevel == 100) {
+					$('power').className = 'battery-percentimage-charged';
+					$('power').innerHTML = '<img src="images/battery-charged.png" />' + this.getBatteryLevelPercent(this.batteryLevel, batteryLevelColor, true, 1);
+				} else {
+					$('power').className = 'battery-percentimage';
+					$('power').innerHTML = '<img src="images/battery-charging-' + this.lastBatteryLevel + '.png" />' + this.getBatteryLevelPercent(this.batteryLevel, batteryLevelColor, true, 1);
+				}
+			}				
 						
 			var stageController = Mojo.Controller.getAppController().getStageProxy("LowBatteryAlert");
 			if (stageController) {
@@ -1780,7 +1840,17 @@ handlePowerNotifications: function(payload) {
 		else {
 			this.chargingBannerShown = false;			
 			Mojo.Log.info("SystemUI- Charger Notification "+ this.isCharging + " Last battery level "+ this.lastBatteryLevel);
+			if (batteryLevelImage == 0) {
 			$('power').className = 'battery-' + this.lastBatteryLevel;	
+			}
+			else if (batteryLevelImage == 1) {
+				$('power').className = 'battery-percent';
+				$('power').innerHTML = this.getBatteryLevelPercent(this.batteryLevel, batteryLevelColor, true);
+			}
+			else if(batteryLevelImage == 2) {
+				$('power').className = 'battery-percentimage';
+				$('power').innerHTML = '<img src="images/battery-' + this.lastBatteryLevel + '.png" />' + this.getBatteryLevelPercent(this.batteryLevel, batteryLevelColor);
+			}				
 			Mojo.Controller.getAppController().removeBanner('chargingAlert');		
 		}	
 	}
diff --git a/usr/lib/luna/system/luna-systemui/stylesheets/systemui.css b/usr/lib/luna/system/luna-systemui/stylesheets/systemui.css
index a0dee3c..3f76828 100644
--- a/usr/lib/luna/system/luna-systemui/stylesheets/systemui.css
+++ b/usr/lib/luna/system/luna-systemui/stylesheets/systemui.css
@@ -136,6 +136,22 @@ body.palm-default {
 	margin-left: 2px;
 	width: 17px;		
 }
+
+/* Begin: Battery Percent / Image w/ Percent */
+#power.battery-percent {
+	width: 38px;
+	background-color: black;
+}
+#power.battery-percentimage {
+	width: 49px;
+	background-color: black;
+}
+#power.battery-percentimage-charged {
+	width: 49px;
+	background-color: black;
+}
+/* End: Battery Percent / Image w/ Percent */
+
 #power.error {
 	background: url(../images/battery-error.png) center center no-repeat;
 }
