diff --git a/usr/lib/luna/system/luna-systemui/app/controllers/devicemenu-assistant.js b/usr/lib/luna/system/luna-systemui/app/controllers/devicemenu-assistant.js
index 08f7f75..e761872 100644
--- a/usr/lib/luna/system/luna-systemui/app/controllers/devicemenu-assistant.js
+++ b/usr/lib/luna/system/luna-systemui/app/controllers/devicemenu-assistant.js
@@ -147,6 +147,12 @@ var DevicemenuAssistant = Foundations.Class.create({
 				setText: this.altCharSelected.bind(this)
 			};
 		}
+		this.controller.get("dm_vpn").style.display = "none";
+		var elem = this.controller.get("dm_vpn").previousSibling;
+		while(elem.innerHTML==null || elem.className.indexOf("palm-section-divider")<0)
+			elem = elem.previousSibling;
+		elem.style.display = "none";
+		this.controller.get("vpndetails").style.display = "none";
 	},
 	
 	activate: function() {
@@ -1062,7 +1068,7 @@ var DevicemenuAssistant = Foundations.Class.create({
 	},
 	
 	showVPN:function(available) {
-		if(available) {
+		if(false) {
 			this.controller.get('dm_vpn').show();
 			this.controller.showWidgetContainer(this.controller.get('dm_vpn'));
 		}
