--- .orig/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
+++ /usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
@@ -525,19 +525,19 @@
 },
 
 showTodayMenu: function() {
-//	var stageController = Mojo.Controller.getAppController().getStageProxy("TodayMenu");
-//	if (stageController) {
-//		this.hideSystemMenus("today");
-//		stageController.delegateToSceneAssistant("toggleTodayMenu");			
-//	} else {
-//		this.todayMenuCreatedHandler = this.todayMenuCreated.bind(this);
-//		Mojo.Controller.getAppController().createStageWithCallback({name:"TodayMenu",lightweight: true}, 
-//			this.todayMenuCreatedHandler, "menu");
-//	}
+	var stageController = Mojo.Controller.getAppController().getStageProxy("TodayMenu");
+	if (stageController) {
+		this.hideSystemMenus("today");
+		stageController.delegateToSceneAssistant("toggleTodayMenu");			
+	} else {
+		this.todayMenuCreatedHandler = this.todayMenuCreated.bind(this);
+		Mojo.Controller.getAppController().createStageWithCallback({name:"TodayMenu",lightweight: true}, 
+			this.todayMenuCreatedHandler, "menu");
+	}
 },
 
 todayMenuCreated: function(stageController) {
-//	stageController.pushScene('todaymenu',this);
+	stageController.pushScene('todaymenu',this);
 },
 
 // Device menu
--- .orig/usr/lib/luna/system/luna-systemui/sources.json
+++ /usr/lib/luna/system/luna-systemui/sources.json
@@ -9,6 +9,9 @@
     "source": "app\/controllers\/devicemenu-assistant.js"
   },  
   {
+    "source": "app\/controllers\/todaymenu-assistant.js"
+  },  
+  {
     "source": "app\/controllers\/pbapalert-assistant.js"
   },
   {
