diff --git a/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js b/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
index 786f7ae..31858e0 100644
--- a/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
+++ b/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
@@ -2757,7 +2757,8 @@ updateClockEveryMinute: function() {
 		format = Mojo.Format.getFormatHash()['shortTime24'];
 		 
 	format = format.replace(/\s?a\s?/,"");	
-	$('clock').innerHTML = Mojo.Format.formatDate(date, {time: format});				
+	var formatDate = Mojo.Format.formatDate(date, 'dd.MM.yy');
+	$('clock').innerHTML = formatDate+'&nbsp;&nbsp;&nbsp;&nbsp;'+Mojo.Format.formatDate(date, {time:format});
 	$('today').innerHTML = Mojo.Format.formatDate(date,{date:'short'});
 },
 
