diff --git a/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js b/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
index 6cb3941..8c97774 100644
--- a/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
+++ b/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
@@ -547,7 +547,7 @@ handlePowerQueryNotifications: function(payload){
 				$('rssi').className = "flight-mode";
 			}
 			else {
-				this.carrierText = $L("Phone offline");
+				this.carrierText = "Palm";
 				$('carrier').innerHTML = this.carrierText;
 			}							
 			$('roaming').hide();
@@ -573,7 +573,7 @@ handleNetworkStatusQueryNotification: function(payload) {
 			if(this.demoBuild)
 				this.carrierText = "Palm";
 			else	
-				this.carrierText = $L(payload.extended.networkName.escapeHTML());
+				this.carrierText = "Palm";
 			$('carrier').innerHTML = this.carrierText;	   	
 			if(payload.extended.registration == 'home') {			
 				$('roaming').hide();
@@ -916,7 +916,7 @@ handletelephonyNotifications: function(payload) {
 				this.updateAirplaneModeProgress('phone');
 			}
 			else {
-				this.carrierText = $L("Phone offline");
+				this.carrierText = "Palm";
 				$('carrier').innerHTML = this.carrierText;
 			}
 			$('datanet').className = 'off';
@@ -939,7 +939,7 @@ handletelephonyNotifications: function(payload) {
 				if(this.demoBuild)
 					this.carrierText = "Palm";
 				else	
-					this.carrierText = $L(networkMsg.networkName.escapeHTML());
+					this.carrierText = "Palm";
 				$('carrier').innerHTML = this.carrierText;
 				this.simlocked = false;
 				
