--- .orig/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
+++ /usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
@@ -556,19 +556,19 @@
 
 showTodayMenu: function(edit) {
 
-//	var stageController = Mojo.Controller.getAppController().getStageProxy("TodayMenu");
-//	if (stageController) {
-//		this.hideSystemMenus("today");
-//		stageController.delegateToSceneAssistant("toggleTodayMenu", edit);			
-//	} else {
-//		this.todayMenuCreatedHandler = this.todayMenuCreated.bind(this);
-//		Mojo.Controller.getAppController().createStageWithCallback({name:"TodayMenu",lightweight: true}, 
-//			this.todayMenuCreatedHandler, "menu");
-//	}
+	var stageController = Mojo.Controller.getAppController().getStageProxy("TodayMenu");
+	if (stageController) {
+		this.hideSystemMenus("today");
+		stageController.delegateToSceneAssistant("toggleTodayMenu", edit);			
+	} else {
+		this.todayMenuCreatedHandler = this.todayMenuCreated.bind(this);
+		Mojo.Controller.getAppController().createStageWithCallback({name:"TodayMenu",lightweight: true}, 
+			this.todayMenuCreatedHandler, "menu");
+	}
 },
 
 todayMenuCreated: function(stageController) {
-//	stageController.pushScene('todaymenu',this);
+	stageController.pushScene('todaymenu',this);
 },
 
 // Device menu
--- .orig/usr/lib/luna/system/luna-systemui/sources.json
+++ /usr/lib/luna/system/luna-systemui/sources.json
@@ -9,6 +9,9 @@
     "source": "app\/controllers\/devicemenu-assistant.js"
   },  
   {
+    "source": "app\/controllers\/todaymenu-assistant.js"
+  },  
+  {
     "source": "app\/controllers\/pbapalert-assistant.js"
   },
   {
--- .orig/usr/lib/luna/system/luna-systemui/stylesheets/systemui.css
+++ /usr/lib/luna/system/luna-systemui/stylesheets/systemui.css
@@ -417,3 +417,48 @@
 		padding: 0;
 }
 
+/* Make today menu items with smaller font and lower row height */
+
+#palm-today-menu #menu_items .palm-row,
+#palm-today-menu #edit_items .palm-row {
+	line-height: 42px;
+	min-height: 46px;
+	max-height: 46px;
+	margin:0px;
+	padding:0px;
+}
+
+#palm-today-menu #menu_items .palm-row .title,
+#palm-today-menu #edit_items .palm-row .title {
+	line-height: 42px;
+	min-height: 46px;
+	font-size: 18px;
+	margin: 0px 0px 0px 16px;
+	padding: 0px 0px 0px 0px;
+}
+
+#palm-today-menu #menu_items .palm-row .label,
+#palm-today-menu #edit_items .palm-row .label {
+	line-height: 42px;
+	min-height: 46px;
+	margin: 0px 16px 0px 0px;
+	padding: 0px 0px 0px 5px;
+}
+
+/* HACK: could not find a way to limit height of the selected row */
+
+#palm-today-menu #menu_items .palm-row.selected,
+#palm-today-menu #edit_items .palm-row.selected {
+	position: relative;
+	top:-2px;
+}
+#palm-today-menu #menu_items .palm-row.selected .title,
+#palm-today-menu #edit_items .palm-row.selected .title {
+	padding: 2px 0px 0px 0px;
+}
+
+#palm-today-menu #menu_items .palm-row.selected .label,
+#palm-today-menu #edit_items .palm-row.selected .label {
+	padding: 2px 0px 0px 5px;
+}
+
