diff --git a/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js b/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
index d4ae043..11b5b01 100644
--- a/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
+++ b/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
@@ -1798,8 +1798,9 @@ handlePowerNotifications: function(payload) {
 			// Show the battery level if not charging
 			if (!this.isCharging) {
 				Mojo.Log.info("SystemUI- Is Charging "+ this.isCharging + " Battery level "+ i);
-				$('power').className = 'battery-' + i;
-				
+				$('power').className = 'battery';
+				$('power').innerHTML = '<img src="images/battery-' + i + '.png" /><div style="-webkit-transform: rotate(-90deg); position: absolute; left: 162px; top: 1px; text-align: center; font-size: 10px">' + payload.percent_ui + '%</div>';
+
 				//Show Banner Message if the Battery level is below 20%
 				var batteryalert = this.$L("#{percent_ui}% battery remaining").interpolate(payload);	
 				if(this.batteryLevel <= 5 && !this.batteryLevel5Shown) {
@@ -1830,11 +1831,12 @@ handlePowerNotifications: function(payload) {
 			}
 			else {
 				Mojo.Log.info("SystemUI- Is Charging "+ this.isCharging );
-				if(payload.percent_ui == 100)
-					$('power').className = 'battery-charged';
-				else
-					$('power').className = 'battery-charging-' + i;
-					
+				$('power').className = 'battery';
+				if (payload.percent_ui == 100) {
+				    $('power').innerHTML = '<img src="images/battery-charged.png" /><div style="-webkit-transform: rotate(-90deg); position: absolute; left: 162px; top: 1px; text-align: center; font-size: 10px">' + payload.percent_ui + '%</div>';
+				} else {
+				    $('power').innerHTML = '<img src="images/battery-charging-' + i + '.png" /><div style="-webkit-transform: rotate(-90deg); position: absolute; left: 162px; top: 1px; text-align: center; font-size: 10px">' + payload.percent_ui + '%</div>';
+				}
 			}
 			
 			if(this.batteryLevel > 5 && this.batteryLevel <= 10) {				
@@ -1870,11 +1872,12 @@ handlePowerNotifications: function(payload) {
 						
 		if (this.isCharging) {
 			Mojo.Log.info("SystemUI- Charger Notification "+ this.isCharging + " Last battery level "+ this.lastBatteryLevel);
-			if(this.batteryLevel == 100)
-				$('power').className = 'battery-charged';
-			else
-				$('power').className = 'battery-charging-' + this.lastBatteryLevel;
-						
+			$('power').className = 'battery';
+			if (this.batteryLevel == 100) {
+			    $('power').innerHTML = '<img src="images/battery-charged.png" /><div style="-webkit-transform: rotate(-90deg); position: absolute; left: 162px; top: 1px; text-align: center; font-size: 10px">' + this.batteryLevel + '%</div>';
+			} else {
+			    $('power').innerHTML = '<img src="images/battery-charging-' + this.lastBatteryLevel + '.png" /><div style="-webkit-transform: rotate(-90deg); position: absolute; left: 162px; top: 1px; text-align: center; font-size: 10px">' + this.batteryLevel + '%</div>';
+			}
 			var stageController = Mojo.Controller.getAppController().getStageProxy("LowBatteryAlert");
 			if (stageController) {
 				stageController.delegateToSceneAssistant("close");
@@ -1894,7 +1897,9 @@ handlePowerNotifications: function(payload) {
 		else {
 			this.chargingBannerShown = false;			
 			Mojo.Log.info("SystemUI- Charger Notification "+ this.isCharging + " Last battery level "+ this.lastBatteryLevel);
-			$('power').className = 'battery-' + this.lastBatteryLevel;	
+			$('power').className = 'battery';
+			$('power').innerHTML = '<img src="images/battery-' + this.lastBatteryLevel + '.png" /><div style="-webkit-transform: rotate(-90deg); position: absolute; left: 162px; top: 1px; text-align: center; font-size: 10px">' + this.batteryLevel + '%</div>';
+
 			Mojo.Controller.getAppController().removeBanner('chargingAlert');		
 		}	
 	}
diff --git a/usr/lib/luna/system/luna-systemui/stylesheets/systemui.css b/usr/lib/luna/system/luna-systemui/stylesheets/systemui.css
index a0dee3c..e5aa76b 100644
--- a/usr/lib/luna/system/luna-systemui/stylesheets/systemui.css
+++ b/usr/lib/luna/system/luna-systemui/stylesheets/systemui.css
@@ -134,7 +134,10 @@ body.palm-default {
 
 #power {
 	margin-left: 2px;
-	width: 17px;		
+	width: 17px;
+}
+#power.battery {
+	width: 32px;
 }
 #power.error {
 	background: url(../images/battery-error.png) center center no-repeat;
