diff --git a/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js b/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
index 0e239cc..8144bfa 100644
--- a/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
+++ b/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
@@ -2033,6 +2033,23 @@ closePbapAlert: function() {
 	}	
 },
 
+
+getBatteryLevelColor: function(percent, charging) {
+		if (charging == 1)
+			batPercentColor = "charging";
+		else if (percent >= 70) 
+			batPercentColor = "full";
+		else if(percent <= 69 && percent >= 45) 
+			batPercentColor = "medium";
+		else if(percent <= 44 && percent >= 20)
+			batPercentColor = "low";
+		else if(percent <= 19)
+			batPercentColor = "critical";
+
+	return batPercentColor;
+},
+
+
 /*
  * Handle power and charging notifications
  */
@@ -2065,6 +2082,8 @@ handlePowerNotifications: function(payload) {
 			if (!this.isCharging) {
 				Mojo.Log.info("SystemUI- Is Charging "+ this.isCharging + " Battery level "+ i);
 				Mojo.Dom.get('power').className = 'battery-' + i;
+				Mojo.Dom.get('batpercent').className = this.getBatteryLevelColor(payload.percent_ui);
+				Mojo.Dom.get('batpercent').innerHTML = this.batteryLevel + unescape('%');
 				
 				//Show Banner Message if the Battery level is below 20%
 				var batteryalert = $L("#{percent_ui}% battery remaining").interpolate(payload);	
@@ -2103,6 +2122,10 @@ handlePowerNotifications: function(payload) {
 				else
 					Mojo.Dom.get('power').className = 'battery-charging-' + i;
 					
+				Mojo.Dom.get('batpercent').className = this.getBatteryLevelColor(payload.percent_ui, true);
+				Mojo.Dom.get('batpercent').innerHTML = this.batteryLevel + unescape('%');
+
+					
 			}
 			
 			if(this.batteryLevel > 5 && this.batteryLevel <= 10) {				
@@ -2151,6 +2174,9 @@ handlePowerNotifications: function(payload) {
 			else
 				Mojo.Dom.get('power').className = 'battery-charging-' + this.lastBatteryLevel;
 						
+			Mojo.Dom.get('batpercent').className = this.getBatteryLevelColor(this.batteryLevel, true);
+			Mojo.Dom.get('batpercent').innerHTML = this.batteryLevel + unescape('%');
+			
 			var stageController = Mojo.Controller.getAppController().getStageController("LowBatteryAlert");
 			if (stageController) {
 				stageController.delegateToSceneAssistant("close");
@@ -2174,6 +2200,8 @@ handlePowerNotifications: function(payload) {
 			this.chargingBannerShown = false;			
 			Mojo.Log.info("SystemUI- Charger Notification "+ this.isCharging + " Last battery level "+ this.lastBatteryLevel);
 			Mojo.Dom.get('power').className = 'battery-' + this.lastBatteryLevel;	
+			Mojo.Dom.get('batpercent').className = this.getBatteryLevelColor(this.batteryLevel);
+			Mojo.Dom.get('batpercent').innerHTML = this.batteryLevel + unescape('%');
 			Mojo.Controller.getAppController().removeBanner('chargingAlert');	
 			if (this.batteryLevel < this.minBatThresholdForUpdate) {
 				Mojo.Log.info("SystemUI - OTA - Closing All Update Alerts due to battery level dropped below threshold level and charger disconnected.");
diff --git a/usr/lib/luna/system/luna-systemui/app/views/bar/bar-scene.html b/usr/lib/luna/system/luna-systemui/app/views/bar/bar-scene.html
index d728ec9..bd088a3 100644
--- a/usr/lib/luna/system/luna-systemui/app/views/bar/bar-scene.html
+++ b/usr/lib/luna/system/luna-systemui/app/views/bar/bar-scene.html
@@ -13,6 +13,8 @@
 	</div>
 
 	<div id="devicemenu">
+		<span id="batpercent">
+		</span>
 		<div id="power"></div>
 		<div id="rssi"></div>
 		<div id="ev1x">
diff --git a/usr/lib/luna/system/luna-systemui/stylesheets/systemui.css b/usr/lib/luna/system/luna-systemui/stylesheets/systemui.css
index 349c68c..9acd29c 100644
--- a/usr/lib/luna/system/luna-systemui/stylesheets/systemui.css
+++ b/usr/lib/luna/system/luna-systemui/stylesheets/systemui.css
@@ -121,6 +121,7 @@ body.palm-default {
 /* device menu icons */
 
 #power,
+#batpercent,
 #rssi,
 #roaming,
 #datanet,
@@ -142,8 +143,37 @@ body.palm-default {
 }
 
 #power {
+	position:absolute;
+	right: 2px;
+	top: 9px;
+	margin-top: 0px;
+	height: 20px;
+	width: 20px;
+	-webkit-transform: rotate(270deg);
+}
+#batpercent { 
 	margin-left: 2px;
-	width: 17px;		
+	position: relative;
+	bottom: 11px;
+	font-size: 12px;
+	margin-right: 2px;
+	min-width: 22px;
+	text-align: center;
+}
+#batpercent.charging {
+	/*color: #33CCFF;*/
+}
+#batpercent.full {
+	/*color: #33FF33;*/
+}
+#batpercent.medium {
+	/*color: #FFFF33;*/
+}
+#batpercent.low {
+	/*color: #FFCC00;*/
+}
+#batpercent.critical {
+	/*color: #FF0000;*/
 }
 
 #power.error { background-image: url(../images/battery-error.png); }
