diff --git a/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js b/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
index a4a150e..582514b 100644
--- a/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
+++ b/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
@@ -65,6 +65,9 @@ var BarAssistant = Class.create({
 		//Initialize and Subscribe for Storage / Media Mode Notification.			
 		this.storageInit();
 		
+		//Subscribe for Ringer Switch Status
+		this.ringerStatus();
+
 		//Initialize and Subscribe for Power Off Notification.
 		this.powerOffInit();
 		
@@ -3099,6 +3102,30 @@ showAudioAlert: function(volume,showringer,audioTarget) {
 		}	
 },
 
+ringerStatus: function() {
+	this.controller.serviceRequest('palm://com.palm.keys/switches', {
+		method:'status',
+		parameters: {"subscribe":true},
+		onSuccess: function(payload) {
+			this.controller.serviceRequest('palm://com.palm.audio/media', {
+				method:'status',
+				onSuccess: function(response) {
+					this.displayMutedIcon(!response['ringer switch']);
+				}.bind(this)
+			});
+		}.bind(this)
+	});
+},
+
+displayMutedIcon: function(mute) {
+	if(mute) {
+		$('ringer').className = 'off';
+	}
+	else {
+		$('ringer').className = 'on';	
+	}
+},
+
 dataImportInit: function() {
 	//Subscribe to Data Srevice status Notifications.
 	this.dataImportNotificationSession = new Mojo.Service.Request('palm://com.palm.dataimport/', {
diff --git a/usr/lib/luna/system/luna-systemui/app/views/bar/bar-scene.html b/usr/lib/luna/system/luna-systemui/app/views/bar/bar-scene.html
index c1daf79..4b574ae 100644
--- a/usr/lib/luna/system/luna-systemui/app/views/bar/bar-scene.html
+++ b/usr/lib/luna/system/luna-systemui/app/views/bar/bar-scene.html
@@ -19,6 +19,8 @@
 	<div id="devicemenu">
 		<div id="power">
 		</div>
+		<div id="ringer">
+		</div>
 		<div id="rssi">
 		</div>
 		<div id="roaming">
diff --git a/usr/lib/luna/system/luna-systemui/stylesheets/systemui.css b/usr/lib/luna/system/luna-systemui/stylesheets/systemui.css
index a0dee3c..8f9e33c 100644
--- a/usr/lib/luna/system/luna-systemui/stylesheets/systemui.css
+++ b/usr/lib/luna/system/luna-systemui/stylesheets/systemui.css
@@ -122,6 +122,7 @@ body.palm-default {
 #tty,
 #hac,
 #callforward,
+#ringer,
 #gps {
 		float: right;
 		height: 20px;
@@ -217,6 +218,16 @@ body.palm-default {
 	background: url(../images/battery-charged.png) center center no-repeat;
 }
 
+#ringer {
+	width: 20px;
+	margin-left: 2px;
+}
+#ringer.off {
+	background: url(../../../../../palm/applications/com.palm.app.soundsandalerts/images/prefs-mute.png) center left no-repeat;
+}
+#ringer.on {
+	display: none;
+}
 
 #rssi {
 	width: 19px;
