diff --git a/usr/lib/luna/system/luna-systemui/app/controllers/devicemenu-assistant.js b/usr/lib/luna/system/luna-systemui/app/controllers/devicemenu-assistant.js
index eef1fb2..bcd880c 100644
--- a/usr/lib/luna/system/luna-systemui/app/controllers/devicemenu-assistant.js
+++ b/usr/lib/luna/system/luna-systemui/app/controllers/devicemenu-assistant.js
@@ -91,10 +91,11 @@ var DevicemenuAssistant = Foundations.Class.create({
 		//Spinner for BT
 		this.controller.setupWidget('btSpinner', this.spinnerAttrs, this.btspinnerModel);
 			
-		if(this.barAssistant.getAirplaneMode()) 			
-			this.controller.get('dm_airplanemode_status').innerHTML = $L('Turn off Airplane Mode');
-		else
-			this.controller.get('dm_airplanemode_status').innerHTML = $L('Turn on Airplane Mode');
+		this.controller.get('dm_airplanemode').style.display="none";
+		var elem = this.controller.get('dm_airplanemode').previousSibling;
+		while(elem.innerHTML==null || elem.className.indexOf("palm-section-divider")<0)
+			elem = elem.previousSibling;
+		elem.style.display = "none";
 		
 		var btstate = this.barAssistant.getCurrentBluetoothState();
 		if(btstate === 'Off') {
