diff --git a/usr/lib/luna/system/luna-systemui/app/controllers/devicemenu-assistant.js b/usr/lib/luna/system/luna-systemui/app/controllers/devicemenu-assistant.js
index 08f7f75..0adb5b0 100644
--- a/usr/lib/luna/system/luna-systemui/app/controllers/devicemenu-assistant.js
+++ b/usr/lib/luna/system/luna-systemui/app/controllers/devicemenu-assistant.js
@@ -163,6 +163,7 @@ var DevicemenuAssistant = Foundations.Class.create({
 		this.controller.get('wifi_pref').addEventListener(Mojo.Event.tap,this.handleWiFiLaunch.bind(this));
 		this.controller.get('vpn_pref').addEventListener(Mojo.Event.tap,this.handleVPNLaunch.bind(this));
 		this.controller.get('ime_OnOff').addEventListener(Mojo.Event.tap, this.toggleIMEOnOff.bind(this));
+		this.controller.get('dm_flash').addEventListener(Mojo.Event.tap, this.toggleFlash.bindAsEventListener(this));
 		this.controller.get('imelist').addEventListener(Mojo.Event.listTap,this.handleIMETap.bindAsEventListener(this));
 		this.controller.get('dm_ime').addEventListener(Mojo.Event.tap, this.toggleIMEList.bindAsEventListener(this));
 		this.controller.get('ime_pref').addEventListener(Mojo.Event.tap,this.handleIMELaunch.bindAsEventListener(this));
@@ -441,6 +442,31 @@ var DevicemenuAssistant = Foundations.Class.create({
 		}
 	},
 	
+	toggleFlash: function() {
+		if(this.flashVal==0) {
+			this.flashVal = 100;
+		} else {
+			this.flashVal = 0;
+		}
+		this.controller.serviceRequest('palm://ca.canucksoftware.systoolsmgr', {
+			method: 'flashOn',
+			parameters: {
+				value: this.flashVal
+			},
+			onSuccess: function() {
+				this.setFlashLabel();
+			}.bind(this)
+		});
+	},
+	
+	setFlashLabel: function() {
+		if(this.flashVal==0) {
+			this.controller.get('flashStatus').innerText = "Off";
+		} else {
+			this.controller.get('flashStatus').innerText = "On";
+		}
+	},
+	
 	/*
 	 * Disconnect All Bluetooth Profiles.
 	 */	
@@ -1420,6 +1446,18 @@ var DevicemenuAssistant = Foundations.Class.create({
 	
 	toggleDeviceMenu: function() {
 		this.clearAllAnimations();
+		this.flashVal = 0;
+		this.controller.serviceRequest('palm://ca.canucksoftware.systoolsmgr', {
+			method: 'flashState',
+			parameters: {},
+			onSuccess: function(response) {
+				this.flashVal = response.value;
+				this.setFlashLabel();
+			}.bind(this),
+			onFailure: function(err) {
+				this.controller.get('flashStatus').innerText = "Error";
+			}.bind(this)
+		});
 		var self = this;
 		if(this.isVisible) {
 			this.isVisible = false;
diff --git a/usr/lib/luna/system/luna-systemui/app/views/devicemenu/devicemenu-scene.html b/usr/lib/luna/system/luna-systemui/app/views/devicemenu/devicemenu-scene.html
index 0e43a60..1277b86 100644
--- a/usr/lib/luna/system/luna-systemui/app/views/devicemenu/devicemenu-scene.html
+++ b/usr/lib/luna/system/luna-systemui/app/views/devicemenu/devicemenu-scene.html
@@ -159,6 +159,15 @@
 				</div>	
 				
 				<div class="palm-section-divider"></div>
+
+				<div class="palm-row" id="dm_flash"><div class="palm-row-wrapper">
+					<div class="title truncating-text">
+						<div class="label right" id="flashStatus">&nbsp;</div>
+						<span x-mojo-loc=''>Flashlight</span>
+					</div>
+				</div></div>
+
+				<div class="palm-section-divider"></div>
 				
 				<div id="dm_airplanemode" class="palm-row last menu-end" x-mojo-tap-highlight='momentary'><div class="palm-row-wrapper">
 					<div id="dm_airplanemode_status" class="title truncating-text">
