diff --git a/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js b/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
index 0e239cc..73006ed 100644
--- a/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
+++ b/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
@@ -2546,17 +2546,20 @@ handleWIFINotifications: function(payload) {
 			case 'associated': 
 				Mojo.Dom.get('wifi').className = 'connecting';				
 				this.wifiState = payload.networkInfo.ssid;
+				Mojo.Dom.get('carrier').innerHTML = this.wifiState;
 			 	break;
 			case 'ipFailed':			
 			case 'notAssociated':
 			case 'associationFailed':
 				this.wifiState = 'On';				
 				this.wifiConnected = false;				
+				Mojo.Dom.get('carrier').innerHTML = this.carrierText;
 				Mojo.Dom.get('wifi').className = 'on';			
 				break;
 			case 'ipConfigured':
 				this.wifiConnected = true;
 				this.wifiState = payload.networkInfo.ssid;
+				Mojo.Dom.get('carrier').innerHTML = this.wifiState;
 				if(payload.networkInfo.signalBars != undefined){
 					Mojo.Dom.get('wifi').className = 'bar-'+payload.networkInfo.signalBars;					
 				}			
@@ -2575,6 +2578,7 @@ handleWIFINotifications: function(payload) {
 	}
 	else if (payload.status == 'serviceEnabled') {
 		this.wifiConnected = false;
+		Mojo.Dom.get('carrier').innerHTML = this.carrierText;
 		Mojo.Dom.get('wifi').className = 'on';
 		this.wifiState = 'On';
 		this.wifiRadio = true;
@@ -2587,6 +2591,7 @@ handleWIFINotifications: function(payload) {
 	}
 	else if (payload.status == 'serviceDisabled') {
 		this.wifiConnected = false;
+		Mojo.Dom.get('carrier').innerHTML = this.carrierText;
 		Mojo.Dom.get('wifi').className = 'off';
 		this.wifiState = 'Off';
 		this.wifiRadio = false;	
