diff --git a/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js b/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
index ee479ac..f02e1cd 100644
--- a/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
+++ b/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
@@ -2337,7 +2337,7 @@ updateClockEveryMinute: function() {
 		}
 	}
 	format = format.replace(/\s?a\s?/,"");	
-	$('clock').innerHTML = Mojo.Format.formatDate(date, {time: format});				
+	$('clock').innerHTML = Mojo.Format.formatDate(date, {date:'short',time: format});				
 	$('today').innerHTML = Mojo.Format.formatDate(date,{date:'short'});
 },
 
