diff --git a/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js b/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
index ee479ac..ae3563c 100644
--- a/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
+++ b/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
@@ -1435,6 +1435,10 @@ closePbapAlert: function() {
  */
 handlePowerNotifications: function(payload) {
 	
+	//Show Actual Percent
+	//Ugly, but it's the easiest way to keep this compatible with the Battery Icon patches.	
+	payload.percent_ui = payload.percent;
+
 	Mojo.Log.info("SystemUI Power Payload "+ Object.toJSON(payload));
 	if(!this.powerdServiceUp) {
 		Mojo.Log.error("SystemUI PowerD down -- ignoring the notification");
