diff --git a/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js b/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
index 786f7ae..d1ca2b9 100644
--- a/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
+++ b/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
@@ -297,7 +297,7 @@ updateAppMenuTitle: function(payload) {
 	if (payload.title) {
 		$('carrier').hide();
 		if (payload.id == "com.palm.app.phone" || payload.id == "com.palm.app.firstuse"){ 
-			$('appname').innerHTML = this.carrierText;
+			$('appname').innerHTML = payload.appmenu || payload.title;
 			this.phoneIsForeground = true;
 		}
 		else if (this.isPreferences(payload))  {
@@ -390,7 +390,7 @@ handleStatusChange: function(payload) {
 	if (payload.locked != null) {
 		if (payload.locked) {
 			$('clock').hide();
-			$('date').show();
+			$('date').hide();
 			this.deviceLocked = true;
 		}
 		else {
@@ -583,8 +583,8 @@ handlePowerQueryNotifications: function(payload){
 				$('rssi').className = "flight-mode";
 			}
 			else {
-				this.carrierText = this.$L("Phone offline");
-				this.carrierTextEnglish = "Phone offline";
+				this.carrierText = new Date().toDateString().substring(0, new Date().toDateString().lastIndexOf(" "));
+				this.carrierTextEnglish = this.carrierText;
 				$('carrier').innerHTML = this.carrierText;
 			}							
 			$('roaming').hide();
@@ -612,8 +612,8 @@ handleNetworkStatusQueryNotification: function(payload) {
 				this.carrierTextEnglish = "Palm";
 			}
 			else{
-				this.carrierText = this.$L(payload.extended.networkName.escapeHTML());
-				this.carrierTextEnglish = payload.extended.networkName.escapeHTML();
+				this.carrierText = new Date().toDateString().substring(0, new Date().toDateString().lastIndexOf(" "));
+				this.carrierTextEnglish = this.carrierText;
 			} 
 				
 			$('carrier').innerHTML = this.carrierText;	   	
@@ -989,8 +989,8 @@ handletelephonyNotifications: function(payload) {
 				this.updateAirplaneModeProgress('phone');
 			}
 			else {
-				this.carrierText = this.$L("Phone offline");
-				this.carrierTextEnglish = "Phone offline";
+				this.carrierText = new Date().toDateString().substring(0, new Date().toDateString().lastIndexOf(" "));
+				this.carrierTextEnglish = this.carrierText;
 				$('carrier').innerHTML = this.carrierText;
 			}
 			$('datanet').className = 'off';
@@ -1016,8 +1016,8 @@ handletelephonyNotifications: function(payload) {
 				}
 					
 				else{
-					this.carrierText = this.$L(networkMsg.networkName.escapeHTML());
-					this.carrierTextEnglish = networkMsg.networkName.escapeHTML();
+					this.carrierText = new Date().toDateString().substring(0, new Date().toDateString().lastIndexOf(" "));
+					this.carrierTextEnglish = this.carrierText;
 				}	
 					
 				$('carrier').innerHTML = this.carrierText;
@@ -2751,6 +2751,9 @@ updateClockEveryMinute: function() {
 	var date = new Date();
 	var format;
 	
+	this.carrierText = new Date().toDateString().substring(0, new Date().toDateString().lastIndexOf(" "));
+	$('carrier').innerHTML = this.carrierText;
+	
 	if(Mojo.Format.using12HrTime()) 
 		format = Mojo.Format.getFormatHash()['shortTime12'];
 	else
