diff --git a/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js b/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
index 786f7ae..3b9d9ac 100644
--- a/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
+++ b/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
@@ -279,6 +279,7 @@ handleAppName:function(payload) {
 		this.foregroundAppID = payload.id;
 		$('appname').hide();
 		$('carrier').show();
+		$('devicemenu').className = "";
 		return;
 	}
 	
diff --git a/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js.rej b/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js.rej
new file mode 100644
index 0000000..8b1d3bd
--- /dev/null
+++ b/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js.rej
@@ -0,0 +1,52 @@
+***************
+*** 309,319 ****
+  		else 
+  			$('appname').innerHTML = payload.appmenu || payload.title;
+  		$('appmenu').show();
+  		this.foregroundAppID = payload.id;
+  	}
+  	else{
+  		$('appmenu').hide();
+  		$('carrier').show();
+  		this.foregroundAppID=null;
+  	}
+  },
+--- 310,322 ----
+  		else 
+  			$('appname').innerHTML = payload.appmenu || payload.title;
+  		$('appmenu').show();
++ 		$('devicemenu').className = "devicemenu-appmenu";
+  		this.foregroundAppID = payload.id;
+  	}
+  	else{
+  		$('appmenu').hide();
+  		$('carrier').show();
++ 		$('devicemenu').className = "";
+  		this.foregroundAppID=null;
+  	}
+  },
+***************
+*** 401,410 ****
+  		if (payload.locked || this.foregroundAppID == null) {
+  			$('carrier').show();
+  			$('appmenu').hide();
+  		}
+  		else {
+  			$('carrier').hide();
+  			$('appmenu').show();
+  		}
+  	}	
+  	
+--- 404,415 ----
+  		if (payload.locked || this.foregroundAppID == null) {
+  			$('carrier').show();
+  			$('appmenu').hide();
++ 			$('devicemenu').className = "";
+  		}
+  		else {
+  			$('carrier').hide();
+  			$('appmenu').show();
++ 			$('devicemenu').className = "devicemenu-appmenu";
+  		}
+  	}	
+  	
diff --git a/usr/lib/luna/system/luna-systemui/stylesheets/systemui.css b/usr/lib/luna/system/luna-systemui/stylesheets/systemui.css
index a0dee3c..ca5d817 100644
--- a/usr/lib/luna/system/luna-systemui/stylesheets/systemui.css
+++ b/usr/lib/luna/system/luna-systemui/stylesheets/systemui.css
@@ -36,8 +36,9 @@ body.palm-default {
 		top: 0;
 		left: 0;
 		height: 28px;
-		width: 140px;
+		width: 100%;
 		padding: 0;
+		background: transparent;
 }
 
 #date-and-time {
@@ -68,6 +69,7 @@ body.palm-default {
 		max-height: 26px;
 		float: left;
 		border-width: 0 20px 0 13px;
+		min-width: 287px;
 		-webkit-border-image: url(../images/appname-background.png) 0 20 0 13 repeat repeat;
 }
 
diff --git a/usr/lib/luna/system/luna-systemui/stylesheets/systemui.css.rej b/usr/lib/luna/system/luna-systemui/stylesheets/systemui.css.rej
new file mode 100644
index 0000000..64f0e46
--- /dev/null
+++ b/usr/lib/luna/system/luna-systemui/stylesheets/systemui.css.rej
@@ -0,0 +1,24 @@
+***************
+*** 394,399 ****
+  		background-color: black;
+  }
+  
+  #statusmenu {
+  		color: #fff;
+  		position: fixed;
+--- 396,410 ----
+  		background-color: black;
+  }
+  
++ .devicemenu-appmenu {
++ 		position: fixed;
++ 		margin: 0;
++ 		top: 0;
++ 		right: 0;
++ 		padding-right: 8px;
++ 		width: 120px;
++ }
++ 
+  #statusmenu {
+  		color: #fff;
+  		position: fixed;
diff --git a/usr/palm/frameworks/mojo/submissions/347/stylesheets/global-menus.css.rej b/usr/palm/frameworks/mojo/submissions/347/stylesheets/global-menus.css.rej
new file mode 100644
index 0000000..1a7406c
--- /dev/null
+++ b/usr/palm/frameworks/mojo/submissions/347/stylesheets/global-menus.css.rej
@@ -0,0 +1,19 @@
+***************
+*** 659,666 ****
+  #palm-app-menu.palm-popup-container {
+    top: -2px;
+    left: -6px;
+-   min-width: 306px;
+-   max-width: 306px;
+    max-height: 413px;
+  }
+  
+--- 659,666 ----
+  #palm-app-menu.palm-popup-container {
+    top: -2px;
+    left: -6px;
++   min-width: 332px;
++   max-width: 332px;
+    max-height: 413px;
+  }
+  
