diff --git a/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js b/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
index 0e239cc..f7e8f80 100644
--- a/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
+++ b/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
@@ -99,6 +99,9 @@ var BarAssistant = Foundations.Class.create({
 		
 		//Initialize and Subscribe for Storage / Media Mode Notification.			
 		this.storageInit();
+
+                //Subscribe for Ringer Switch Status
+                this.ringerStatus();
 		
 		//Initialize and Subscribe for Power Off Notification.
 		this.powerOffInit();
@@ -3454,6 +3457,30 @@ showAudioAlert: function(volume,showringer,audioTarget) {
 		}	
 },
 
+ringerStatus: function() {
+       this.controller.serviceRequest('palm://com.palm.keys/switches', {
+               method:'status',
+               parameters: {"subscribe":true},
+               onSuccess: function(payload) {
+                       this.controller.serviceRequest('palm://com.palm.audio/media', {
+                               method:'status',
+                               onSuccess: function(response) {
+                                       this.displayMutedIcon(!response['ringer switch']);
+                               }.bind(this)
+                       });
+               }.bind(this)
+       });
+},
+
+displayMutedIcon: function(mute) {
+       if(mute) {
+               $('ringer').className = 'off';
+       }
+       else {
+               $('ringer').className = 'on';   
+       }
+},
+
 dataImportInit: function() {
 	//Subscribe to Data Srevice status Notifications.
 	this.dataImportNotificationSession = new Mojo.Service.Request('palm://com.palm.dataimport/', {
diff --git a/usr/lib/luna/system/luna-systemui/app/views/bar/bar-scene.html b/usr/lib/luna/system/luna-systemui/app/views/bar/bar-scene.html
index d728ec9..4c0952f 100644
--- a/usr/lib/luna/system/luna-systemui/app/views/bar/bar-scene.html
+++ b/usr/lib/luna/system/luna-systemui/app/views/bar/bar-scene.html
@@ -14,6 +14,7 @@
 
 	<div id="devicemenu">
 		<div id="power"></div>
+		<div id="ringer"></div>
 		<div id="rssi"></div>
 		<div id="ev1x">
 			<div id="rssi-ev"></div>
diff --git a/usr/lib/luna/system/luna-systemui/stylesheets/systemui.css b/usr/lib/luna/system/luna-systemui/stylesheets/systemui.css
index 349c68c..5707005 100644
--- a/usr/lib/luna/system/luna-systemui/stylesheets/systemui.css
+++ b/usr/lib/luna/system/luna-systemui/stylesheets/systemui.css
@@ -130,6 +130,7 @@ body.palm-default {
 #tty,
 #hac,
 #callforward,
+#ringer,
 #gps,
 #ev1x {
 	float: right;
@@ -174,6 +175,17 @@ body.palm-default {
 #power.battery-charging-11 { background-image: url(../images/battery-charging-11.png); }
 #power.battery-charged { background-image: url(../images/battery-charged.png); }
 
+#ringer {
+       width: 20px;
+       margin-left: 2px;
+}
+#ringer.off {
+       background: url(../../../../../palm/applications/com.palm.app.soundsandalerts/images/prefs-mute.png) center left no-repeat;
+}
+#ringer.on {
+       display: none;
+}
+
 #rssi {
 	width: 19px;
 }
