diff --git a/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js b/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
index 6d0a87a..9ac40ee 100644
--- a/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
+++ b/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
@@ -12,7 +12,7 @@ var BarAssistant = Foundations.Class.create({
 		this.carrierText = "Palm";
 		this.currentCarrierText = "Palm";
 		this.airplaneModeTriggered = false;
-		this.show3GForEvdo = false;
+		this.show3GForEvdo = true;
 		this.showBlankStatusOnLimited = false;
 		this.minBatThresholdForUpdate = 5;
 		
