--- /usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js.orig	Tue Oct 20 00:42:04 2009
+++ /usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js	Tue Oct 20 00:35:18 2009
@@ -1877,17 +1879,20 @@
 			case 'associated': 
 				$('wifi').className = 'connecting';				
 				this.wifiState = payload.networkInfo.ssid;
+				$('carrier').innerHTML = this.wifiState;
 			 	break;
 			case 'ipFailed':			
 			case 'notAssociated':
 			case 'associationFailed':
 				this.wifiState = 'On';				
 				this.wifiConnected = false;				
+				$('carrier').innerHTML = this.carrierText;
 				$('wifi').className = 'on';			
 				break;
 			case 'ipConfigured':
 				this.wifiConnected = true;
 				this.wifiState = payload.networkInfo.ssid;
+				$('carrier').innerHTML = this.wifiState;
 				if(payload.networkInfo.signalBars != undefined){
 					$('wifi').className = 'bar-'+payload.networkInfo.signalBars;					
 				}
@@ -1907,6 +1912,7 @@
 	}
 	else if (payload.status == 'serviceEnabled') {
 		this.wifiConnected = false;
+		$('carrier').innerHTML = this.carrierText;
 		$('wifi').className = 'on';
 		this.wifiState = 'On';
 		this.wifiRadio = true;
@@ -1919,6 +1925,7 @@
 	}
 	else if (payload.status == 'serviceDisabled') {
 		this.wifiConnected = false;
+		$('carrier').innerHTML = this.carrierText;
 		$('wifi').className = 'off';
 		this.wifiState = 'Off';
 		this.wifiRadio = false;	
