diff --git a/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js b/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
index 786f7ae..0a2b005 100644
--- a/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
+++ b/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
@@ -1772,6 +1772,23 @@ closePbapAlert: function() {
 	}	
 },
 
+
+getBatteryLevelColor: function(percent, charging) {
+		if (charging == 1)
+			batPercentColor = "charging";
+		else if (percent >= 70) 
+			batPercentColor = "full";
+		else if(percent <= 69 && percent >= 45) 
+			batPercentColor = "medium";
+		else if(percent <= 44 && percent >= 20)
+			batPercentColor = "low";
+		else if(percent <= 19)
+			batPercentColor = "critical";
+
+	return batPercentColor;
+},
+
+
 /*
  * Handle power and charging notifications
  */
@@ -1800,6 +1817,8 @@ handlePowerNotifications: function(payload) {
 			if (!this.isCharging) {
 				Mojo.Log.info("SystemUI- Is Charging "+ this.isCharging + " Battery level "+ i);
 				$('power').className = 'battery-' + i;
+				$('batpercent').className = this.getBatteryLevelColor(payload.percent_ui);
+				$('batpercent').innerHTML = this.batteryLevel + unescape('%');
 				
 				//Show Banner Message if the Battery level is below 20%
 				var batteryalert = this.$L("#{percent_ui}% battery remaining").interpolate(payload);	
@@ -1836,6 +1855,10 @@ handlePowerNotifications: function(payload) {
 				else
 					$('power').className = 'battery-charging-' + i;
 					
+				$('batpercent').className = this.getBatteryLevelColor(payload.percent_ui, true);
+				$('batpercent').innerHTML = this.batteryLevel + unescape('%');
+
+					
 			}
 			
 			if(this.batteryLevel > 5 && this.batteryLevel <= 10) {				
@@ -1876,6 +1899,9 @@ handlePowerNotifications: function(payload) {
 			else
 				$('power').className = 'battery-charging-' + this.lastBatteryLevel;
 						
+			$('batpercent').className = this.getBatteryLevelColor(this.batteryLevel, true);
+			$('batpercent').innerHTML = this.batteryLevel + unescape('%');
+			
 			var stageController = Mojo.Controller.getAppController().getStageProxy("LowBatteryAlert");
 			if (stageController) {
 				stageController.delegateToSceneAssistant("close");
@@ -1896,6 +1922,8 @@ handlePowerNotifications: function(payload) {
 			this.chargingBannerShown = false;			
 			Mojo.Log.info("SystemUI- Charger Notification "+ this.isCharging + " Last battery level "+ this.lastBatteryLevel);
 			$('power').className = 'battery-' + this.lastBatteryLevel;	
+			$('batpercent').className = this.getBatteryLevelColor(this.batteryLevel);
+			$('batpercent').innerHTML = this.batteryLevel + unescape('%');
 			Mojo.Controller.getAppController().removeBanner('chargingAlert');		
 		}	
 	}
diff --git a/usr/lib/luna/system/luna-systemui/app/views/bar/bar-scene.html b/usr/lib/luna/system/luna-systemui/app/views/bar/bar-scene.html
index c1daf79..6a551ef 100644
--- a/usr/lib/luna/system/luna-systemui/app/views/bar/bar-scene.html
+++ b/usr/lib/luna/system/luna-systemui/app/views/bar/bar-scene.html
@@ -17,6 +17,8 @@
 	</div>                   
 	
 	<div id="devicemenu">
+		<span id="batpercent">
+		</span>
 		<div id="power">
 		</div>
 		<div id="rssi">
diff --git a/usr/lib/luna/system/luna-systemui/stylesheets/systemui.css b/usr/lib/luna/system/luna-systemui/stylesheets/systemui.css
index a0dee3c..f0f9387 100644
--- a/usr/lib/luna/system/luna-systemui/stylesheets/systemui.css
+++ b/usr/lib/luna/system/luna-systemui/stylesheets/systemui.css
@@ -114,6 +114,7 @@ body.palm-default {
 /* device menu icons */
 
 #power,
+#batpercent,
 #rssi,
 #roaming,
 #datanet,
@@ -133,8 +134,35 @@ body.palm-default {
 }
 
 #power {
+	position:absolute;
+	right: 2px;
+	top: 9px;
+	margin-top: 0px;
+	height: 20px;
+	width: 20px;
+	-webkit-transform: rotate(270deg);
+}
+#batpercent { 
 	margin-left: 2px;
-	width: 17px;		
+	position: relative;
+	bottom: 11px;
+	font-size: 12px;
+	margin-right: 2px;
+}
+#batpercent.charging {
+	/*color: #33CCFF;*/
+}
+#batpercent.full {
+	/*color: #33FF33;*/
+}
+#batpercent.medium {
+	/*color: #FFFF33;*/
+}
+#batpercent.low {
+	/*color: #FFCC00;*/
+}
+#batpercent.critical {
+	/*color: #FF0000;*/
 }
 #power.error {
 	background: url(../images/battery-error.png) center center no-repeat;
