diff --git a/usr/lib/luna/system/luna-systemui/app/controllers/devicemenu-assistant.js b/usr/lib/luna/system/luna-systemui/app/controllers/devicemenu-assistant.js
index 335a73c..3e9b00c 100644
--- a/usr/lib/luna/system/luna-systemui/app/controllers/devicemenu-assistant.js
+++ b/usr/lib/luna/system/luna-systemui/app/controllers/devicemenu-assistant.js
@@ -72,10 +72,7 @@ var DevicemenuAssistant = Class.create({
 		//Spinner for Airplande Mode
 		//this.controller.setupWidget('apSpinner', this.spinnerAttrs, this.apSpinnerModel);	    	
 			
-		if(this.barAssistant.getAirplaneMode()) 			
-			this.controller.get('dm_airplanemode_status').innerHTML = $L('Turn off Airplane Mode');
-		else
-			this.controller.get('dm_airplanemode_status').innerHTML = $L('Turn on Airplane Mode');
+		this.controller.get('dm_airplanemode').style.display="none";
 			
 		var btstate = this.barAssistant.getCurrentBluetoothState();
 		if(btstate === 'Off') {
