diff --git a/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js b/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
index 21f3714..76b8e52 100644
--- a/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
+++ b/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
@@ -33,6 +33,7 @@ var BarAssistant = Foundations.Class.create({
 		Mojo.Dom.get('datanet').className = 'off';
 		Mojo.Dom.get('wifi').className = 'off';
 		Mojo.Dom.get('bluetooth').className = 'off';
+		Mojo.Dom.get('gps').className = 'off';
 		Mojo.Dom.get('roaming').hide();
 		Mojo.Dom.get('tty').hide();
 		Mojo.Dom.get('hac').hide();
@@ -150,6 +151,20 @@ loadNetworkAlertLibrary: function() {
 	}		
 },
 
+updateGPSIcon: function() {
+       this.controller.serviceRequest('palm://com.palm.location', {
+               method: 'getUseGps',
+               parameters: {},
+               onSuccess: function(status) {
+                       if (status.useGps) {
+                               $('gps').className = 'on';
+                       } else {
+                               $('gps').className = 'off';
+                       }
+               }.bind(this)
+       });
+},
+
 refreshCarrierText: function() {
 	Mojo.Dom.get('carrier').innerHTML = $L(this.currentCarrierText);
 },
@@ -2223,6 +2238,7 @@ closeProfAlert: function() {
  */
 handlePowerNotifications: function(payload) {
 	
+	this.updateGPSIcon();
 	Mojo.Log.info("SystemUI Power Payload "+ Object.toJSON(payload));
 	if(!this.powerdServiceUp) {
 		Mojo.Log.error("SystemUI PowerD down -- ignoring the notification");
diff --git a/usr/lib/luna/system/luna-systemui/app/views/bar/bar-scene.html b/usr/lib/luna/system/luna-systemui/app/views/bar/bar-scene.html
index d728ec9..749fcc5 100644
--- a/usr/lib/luna/system/luna-systemui/app/views/bar/bar-scene.html
+++ b/usr/lib/luna/system/luna-systemui/app/views/bar/bar-scene.html
@@ -24,6 +24,7 @@
 		<div id="wifi"></div>
 		<div id="vpn"></div>
 		<div id="bluetooth"></div>
+		<div id="gps"></div>
 		<div id="tty"></div>
 		<div id="callforward"></div>
 		<div id="hac"></div>
diff --git a/usr/lib/luna/system/luna-systemui/stylesheets/systemui.css b/usr/lib/luna/system/luna-systemui/stylesheets/systemui.css
index a1cafa2..3e559cc 100644
--- a/usr/lib/luna/system/luna-systemui/stylesheets/systemui.css
+++ b/usr/lib/luna/system/luna-systemui/stylesheets/systemui.css
@@ -294,6 +294,18 @@ body.palm-default {
 	width: 17px;
 }
 
+#gps.off {
+               display: none;
+}
+
+#gps.on {
+               background: url(../images/gps-on.png) center center no-repeat;
+}
+
+#gps.active{
+               background: url(../images/gps-active.png) center center no-repeat;
+}
+
 #vpn {
 	width: 22px;
 	background-image: url(../images/vpn-status-icon.png);
