diff --git a/usr/lib/luna/system/luna-systemui/app/controllers/devicemenu-assistant.js b/usr/lib/luna/system/luna-systemui/app/controllers/devicemenu-assistant.js
index 335a73c..8a321e5 100644
--- a/usr/lib/luna/system/luna-systemui/app/controllers/devicemenu-assistant.js
+++ b/usr/lib/luna/system/luna-systemui/app/controllers/devicemenu-assistant.js
@@ -36,12 +36,14 @@ var DevicemenuAssistant = Class.create({
 		this.controller.get('dm_date').innerHTML = Mojo.Format.formatDate(date,{date:'EEE '}) +  Mojo.Format.formatDate(date,{date:'long'});
 		var msg = $L("Battery: #{batlevel}");
 		this.controller.get('dm_battery').innerHTML = new Template(msg).evaluate({"batlevel": Mojo.Format.formatPercent(this.barAssistant.getBatteryLevel())});
+		this.controller.setupWidget('brightSlider', {minValue: 1, maxValue: 100}, this.brightModel = { value: 50});
 						
 		this.modesModel = {scrollbars: false, mode: "vertical"},
 		this.controller.setupWidget('devicescroller', {}, this.modesModel);
 		this.menuscroller = this.controller.get('devicescroller');
 		
 		this.drawerModel = {myOpenProperty:false};
+		this.controller.setupWidget('brightdetails', {modelProperty:'myOpenProperty'}, this.drawerModel);
 		this.controller.setupWidget('wifidetails', {modelProperty:'myOpenProperty'}, this.drawerModel);
 		this.controller.setupWidget('btdetails', {modelProperty:'myOpenProperty'}, this.drawerModel);
 		this.wifidrawer = this.controller.get('wifidetails');
@@ -121,6 +123,8 @@ var DevicemenuAssistant = Class.create({
 	},
 	
 	activate: function() {
+		this.controller.get('brightSlider').addEventListener(Mojo.Event.propertyChange, this.handleBrightSlider.bindAsEventListener(this));
+		this.controller.get('brightRow').addEventListener(Mojo.Event.tap, this.handleBrightTap.bindAsEventListener(this));
 		this.controller.get('dm_airplanemode').addEventListener(Mojo.Event.tap, this.toggleAirplaneMode.bindAsEventListener(this));
 		this.controller.get('wifilist').addEventListener(Mojo.Event.listTap,this.handleWiFiTap.bindAsEventListener(this));
 		this.controller.get('btlist').addEventListener(Mojo.Event.listTap,this.handleBTTap.bindAsEventListener(this));		
@@ -135,6 +139,18 @@ var DevicemenuAssistant = Class.create({
 		this.toggleDeviceMenu();						
 	},	
 		
+	handleBrightTap: function(event) {
+		this.controller.get('brightdetails').mojo.setOpenState(!this.controller.get('brightdetails').mojo.getOpenState());
+	},
+
+	handleBrightSlider: function(event) {
+		this.controller.serviceRequest('palm://com.palm.display/control', {
+			method: 'setProperty',
+			parameters:{maximumBrightness:Math.round(event.value)}
+		});
+		this.controller.get('dm_brightness').innerText = "Brightness: " + Math.round(event.value) + "%";
+	},
+		
 	cleanup: function() {
 		Mojo.Log.error("Device Menu - Clean up called");				
 	},	
@@ -1028,6 +1044,7 @@ var DevicemenuAssistant = Class.create({
 			this.controller.hideWidgetContainer(this.controller.get('wifidetails'));
 			this.wifidrawer.mojo.setOpenState(false);
 		}		
+		this.controller.get('brightdetails').mojo.setOpenState(false);
 		//this.isVisible = false;
 		if(this.isVisible){
 			this.toggleDeviceMenu();
@@ -1074,6 +1091,20 @@ var DevicemenuAssistant = Class.create({
 			var date = new Date();		
 			this.controller.get('dm_date').innerHTML = Mojo.Format.formatDate(date,{date:'EEE '})  +  Mojo.Format.formatDate(date,{date:'long'});
 			
+			this.controller.serviceRequest('palm://com.palm.display/control', {
+				method: 'getProperty',
+				parameters:{properties:['maximumBrightness']},
+				onSuccess: function(response) {
+					if(response.maximumBrightness != undefined) {
+						this.brightModel.value = parseInt(response.maximumBrightness);
+						this.controller.modelChanged(this.brightModel);
+						this.controller.get('dm_brightness').innerText = "Brightness: " + response.maximumBrightness + "%";
+					} else {
+						this.controller.get('dm_brightness').innerText = "Brightness: ??%";
+					}
+				}.bind(this)
+			});
+			
 			if(this.turningOnBtRadio) {
 				this.btspinnerModel.spinning = true;
 				this.controller.modelChanged(this.btspinnerModel);
diff --git a/usr/lib/luna/system/luna-systemui/app/views/devicemenu/devicemenu-scene.html b/usr/lib/luna/system/luna-systemui/app/views/devicemenu/devicemenu-scene.html
index d153a8d..adde623 100644
--- a/usr/lib/luna/system/luna-systemui/app/views/devicemenu/devicemenu-scene.html
+++ b/usr/lib/luna/system/luna-systemui/app/views/devicemenu/devicemenu-scene.html
@@ -28,6 +28,19 @@
 				<div class="palm-section-divider">
 				</div>               
 				
+				<div class="palm-row" id="brightRow" x-mojo-tap-highlight='momentary'><div class="palm-row-wrapper">
+					<div id="dm_brightness" class="title"></div>
+				</div></div>
+
+				<div id="brightdetails" x-mojo-element="Drawer">
+					<div class="palm-row first"><div class="palm-row-wrapper">
+						<div x-mojo-element="Slider" id="brightSlider" style="padding-top: 30px; height: 50px;"></div>
+					</div></div>
+				</div>
+
+				<div class="palm-section-divider">
+				</div>               
+				
 				<div class="palm-row" id="dm_wifi" x-mojo-tap-highlight='momentary'><div class="palm-row-wrapper">
 					<div class="title truncating-text">
 						<div class="label right" id="wifimsg">&nbsp;</div>                               
