diff --git a/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js b/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
index 0e239cc..955cc1e 100644
--- a/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
+++ b/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
@@ -2988,7 +2988,13 @@ updateClockEveryMinute: function() {
 	else
 		format = Mojo.Format.getFormatHash()['shortTime24'];
 	format = format.replace(/\s?a\s?/,"");	
-	Mojo.Dom.get('clock').innerHTML = Mojo.Format.formatDate(date, {time: format});				
+	var suffix = "";
+	if(new Date().getHours() < 12) {
+		suffix = "a";
+	} else {
+		suffix = "p";
+	}
+	Mojo.Dom.get('clock').innerHTML = Mojo.Format.formatDate(date, {time: format}) + suffix;
 	Mojo.Dom.get('today').innerHTML = Mojo.Format.formatDate(date,{date:'short'});
 },
 
