diff --git a/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js b/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
index 6fdbb39..d8ce7eb 100644
--- a/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
+++ b/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
@@ -2552,7 +2552,9 @@ updateClockEveryMinute: function() {
 		}
 	}
 	format = format.replace(/\s?a\s?/,"");	
-	$('clock').innerHTML = Mojo.Format.formatDate(date, {time: format});				
+  var formatDate = Mojo.Format.formatDate(date, {date:'short'});
+  formatDate = formatDate.substr(0, formatDate.lastIndexOf('/')+formatDate.lastIndexOf('.')+1);
+	$('clock').innerHTML = formatDate+'&nbsp;&nbsp;&nbsp;'+Mojo.Format.formatDate(date, {time:format});
 	$('today').innerHTML = Mojo.Format.formatDate(date,{date:'short'});
 },
 
