diff --git a/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js b/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
index 6d0a87a..6509bbb 100644
--- a/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
+++ b/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
@@ -1993,8 +1993,9 @@ handlePowerNotifications: function(payload) {
 			// Show the battery level if not charging
 			if (!this.isCharging) {
 				Mojo.Log.info("SystemUI- Is Charging "+ this.isCharging + " Battery level "+ i);
-				Mojo.Dom.get('power').className = 'battery-' + i;
-				
+				Mojo.Dom.get('power').className = 'battery';
+				Mojo.Dom.get('power').innerHTML = '<img src="images/battery-' + i + '.png" /><div style="-webkit-transform: rotate(-90deg); position: absolute; left: 162px; top: 1px; text-align: center; font-size: 10px">' + payload.percent_ui + '%</div>';
+
 				//Show Banner Message if the Battery level is below 20%
 				var batteryalert = $L("#{percent_ui}% battery remaining").interpolate(payload);	
 				if(this.batteryLevel <= 5 && !this.batteryLevel5Shown) {
@@ -2027,11 +2028,12 @@ handlePowerNotifications: function(payload) {
 			}
 			else {
 				Mojo.Log.info("SystemUI- Is Charging "+ this.isCharging );
-				if(payload.percent_ui == 100)
-					Mojo.Dom.get('power').className = 'battery-charged';
-				else
-					Mojo.Dom.get('power').className = 'battery-charging-' + i;
-					
+				$('power').className = 'battery';
+				if (payload.percent_ui == 100) {
+				    Mojo.Dom.get('power').innerHTML = '<img src="images/battery-charged.png" /><div style="-webkit-transform: rotate(-90deg); position: absolute; left: 162px; top: 1px; text-align: center; font-size: 10px">' + payload.percent_ui + '%</div>';
+				} else {
+				    Mojo.Dom.get('power').innerHTML = '<img src="images/battery-charging-' + i + '.png" /><div style="-webkit-transform: rotate(-90deg); position: absolute; left: 162px; top: 1px; text-align: center; font-size: 10px">' + payload.percent_ui + '%</div>';
+				}
 			}
 			
 			if(this.batteryLevel > 5 && this.batteryLevel <= 10) {				
@@ -2075,11 +2077,12 @@ handlePowerNotifications: function(payload) {
 						
 		if (this.isCharging) {
 			Mojo.Log.info("SystemUI- Charger Notification "+ this.isCharging + " Last battery level "+ this.lastBatteryLevel);
-			if(this.batteryLevel == 100)
-				Mojo.Dom.get('power').className = 'battery-charged';
-			else
-				Mojo.Dom.get('power').className = 'battery-charging-' + this.lastBatteryLevel;
-						
+			Mojo.Dom.get('power').className = 'battery';
+			if (this.batteryLevel == 100) {
+			    Mojo.Dom.get('power').innerHTML = '<img src="images/battery-charged.png" /><div style="-webkit-transform: rotate(-90deg); position: absolute; left: 162px; top: 1px; text-align: center; font-size: 10px">' + this.batteryLevel + '%</div>';
+			} else {
+			    Mojo.Dom.get('power').innerHTML = '<img src="images/battery-charging-' + this.lastBatteryLevel + '.png" /><div style="-webkit-transform: rotate(-90deg); position: absolute; left: 162px; top: 1px; text-align: center; font-size: 10px">' + this.batteryLevel + '%</div>';
+			}
 			var stageController = Mojo.Controller.getAppController().getStageController("LowBatteryAlert");
 			if (stageController) {
 				stageController.delegateToSceneAssistant("close");
@@ -2102,7 +2105,9 @@ handlePowerNotifications: function(payload) {
 		else {
 			this.chargingBannerShown = false;			
 			Mojo.Log.info("SystemUI- Charger Notification "+ this.isCharging + " Last battery level "+ this.lastBatteryLevel);
-			Mojo.Dom.get('power').className = 'battery-' + this.lastBatteryLevel;	
+			Mojo.Dom.get('power').className = 'battery';
+			Mojo.Dom.get('power').innerHTML = '<img src="images/battery-' + this.lastBatteryLevel + '.png" /><div style="-webkit-transform: rotate(-90deg); position: absolute; left: 162px; top: 1px; text-align: center; font-size: 10px">' + this.batteryLevel + '%</div>';
+
 			Mojo.Controller.getAppController().removeBanner('chargingAlert');	
 			if (this.batteryLevel < this.minBatThresholdForUpdate) {
 				Mojo.Log.info("SystemUI - OTA - Closing All Update Alerts due to battery level dropped below threshold level and charger disconnected.");
diff --git a/usr/lib/luna/system/luna-systemui/stylesheets/systemui.css b/usr/lib/luna/system/luna-systemui/stylesheets/systemui.css
index f3cacab..06164de 100644
--- a/usr/lib/luna/system/luna-systemui/stylesheets/systemui.css
+++ b/usr/lib/luna/system/luna-systemui/stylesheets/systemui.css
@@ -142,7 +142,10 @@ body.palm-default {
 
 #power {
 	margin-left: 2px;
-	width: 17px;		
+	width: 17px;
+}
+#power.battery {
+	width: 32px;
 }
 
 #power.error { background-image: url(../images/battery-error.png); }
