diff --git a/usr/palm/frameworks/mojo/builtins/palmInitFramework191_15.js b/usr/palm/frameworks/mojo/builtins/palmInitFramework191_15.js
index 4307893..b4a5635 100644
--- a/usr/palm/frameworks/mojo/builtins/palmInitFramework191_15.js
+++ b/usr/palm/frameworks/mojo/builtins/palmInitFramework191_15.js
@@ -674,6 +674,7 @@ _Dialog:'widget_dialog',
 Pager:null,
 Drawer:null,
 CharSelector:null,
+OnScreenKeyboard:null,
 FilterList:null,
 _Submenu:'widget_submenu',
 ExperimentalGridList:'widget_grid',
@@ -3717,6 +3718,12 @@ if(this.charSelector&&this.charSelector.element&&this.charSelector.element.mojo&
 return;
 }
 this.charSelector=this.createDynamicWidget('CharSelector',{selectionTarget:event.selectionTarget});
+}else if(event.type===Mojo.Event.renderOnScreenKeyboard){
+
+if(this.charSelector&&this.charSelector.element&&this.charSelector.element.mojo&&this.charSelector.element.mojo.isOpen()){
+return;
+}
+this.charSelector=this.createDynamicWidget('OnScreenKeyboard',{selectionTarget:event.selectionTarget});
 }
 },
 
@@ -3996,6 +4003,7 @@ kDefaultSceneName:'main',
 kSceneClassName:'palm-scene',
 kBrowserObject:'application/x-palm-browser',
 enableAltCharPicker:true,
+enableOnScreenKeyboard:true,
 
 
 
@@ -4891,6 +4899,15 @@ if(webView){
 webView.mojo.isEditing(this._sendCharpickerEventCallback.bind(this,Mojo.Event.renderAltCharacters,webView,null));
 }
 }
+}else if(this.enableOnScreenKeyboard&&event.keyCode===Mojo.Char.metaKey){
+if(this.doesTargetAcceptKeys(event.target)){
+this._sendCharpickerEvent(Mojo.Event.renderOnScreenKeyboard,event.target,null);
+}else{
+webView=this._getWebview(event.target);
+if(webView){
+webView.mojo.isEditing(this._sendCharpickerEventCallback.bind(this,Mojo.Event.renderOnScreenKeyboard,webView,null));
+}
+}
 }else if(event.keyCode===Mojo.Char.o&&event.ctrlKey&&event.shiftKey&&Mojo.Config.debuggingEnabled){
 
 
@@ -9431,6 +9448,9 @@ Mojo.Event.renderAltCharacters='mojo-altchars';
 Mojo.Event.renderChordedAltCharacters='mojo-altchars-chorded';
 
 
+Mojo.Event.renderOnScreenKeyboard='webos-internals-onscreen-keyboard';
+
+
 Mojo.Event.orientationChange='mojo-orientation';
 
 
@@ -10619,7 +10639,7 @@ return stringTable;
 
 Mojo.Locale.set=function(currentLocale,formatRegion){
 var deviceInfo=Mojo.Environment.DeviceInfo;
-var altCharFullTable,altCharTable;
+var altCharFullTable,altCharTable,kbCharTable,kbCharFullTable;
 
 if(Mojo.Locale.current!=currentLocale){
 Mojo.Locale.current=currentLocale;
@@ -10660,13 +10680,21 @@ Mojo.Locale.frameworkStrings=Mojo.Locale.readStringTable("strings.json",Mojo.Loc
 if(deviceInfo.keyboardType===Mojo.Environment.AZERTY){
 altCharTable="alternatechars_table_azerty.json";
 altCharFullTable="alternatechars_fulltable_azerty.json";
+kbCharTable="kbchars_table_azerty.json";
+kbCharFullTable="kbchars_fulltable_azerty.json";
 }else if(deviceInfo.keyboardType===Mojo.Environment.QWERTZ){
 altCharTable="alternatechars_table_qwertz.json";
 altCharFullTable="alternatechars_fulltable_qwertz.json";
+kbCharTable="kbchars_table_qwertz.json";
+kbCharFullTable="kbchars_fulltable_qwertz.json";
 }else{
 altCharTable="alternatechars_table.json";
 altCharFullTable="alternatechars_fulltable.json";
+kbCharTable="kbchars_table.json";
+kbCharFullTable="kbchars_fulltable.json";
 }
+Mojo.Locale.kbCharacters=Mojo.Locale.readStringTable(kbCharTable,Mojo.Locale.current,Mojo.Locale.frameworkResourcePath,altCharArrayMerge);
+Mojo.Locale.kbCharactersFull=Mojo.Locale.readStringTable(kbCharFullTable,Mojo.Locale.current,Mojo.Locale.frameworkResourcePath,altCharArrayMerge);
 Mojo.Locale.alternateCharacters=Mojo.Locale.readStringTable(altCharTable,Mojo.Locale.current,Mojo.Locale.frameworkResourcePath,altCharArrayMerge);
 Mojo.Locale.alternateCharactersFull=Mojo.Locale.readStringTable(altCharFullTable,Mojo.Locale.current,Mojo.Locale.frameworkResourcePath,altCharArrayMerge);
 Mojo.Locale.DateTimeStrings=Mojo.Locale.readStringTable("datetime_table.json",Mojo.Locale.current,Mojo.Locale.frameworkResourcePath);
@@ -34320,6 +34348,855 @@ return true;
 }
 return false;
 };
+
+/* oskb */
+Mojo.Widget.OnScreenKeyboard=Class.create({
+
+HI_PADDING_TOP:40,
+
+HI_PADDING_BOTTOM:20,
+
+HI_PADDING_LEFT:20,
+
+HI_PADDING_RIGHT:20,
+
+HI_COLUMNS:10,
+
+HI_MINIMUM_TOP:10,
+
+HI_MAX_BOTTOM:5,
+
+/* oskb */
+initialize:function(){
+this.ONSCREEN_KB_THEMES_PATH=Mojo.Config.IMAGES_HOME+'/virt_kb/';
+this.ONSCREEN_KB_OPEN=0;
+this.ONSCREEN_KB_FILTERING_STATE=1;
+this.ONSCREEN_KB_CLOSED=2;
+this.ONSCREEN_KB_EMPTY=3;
+
+this.SHIFT_OFF=0;
+this.SHIFT_ONCE=1;
+this.SHIFT_LOCK=2;
+
+this.state=this.ONSCREEN_KB_OPEN;
+this.shiftState=this.SHIFT_OFF;
+
+this.customStyle=undefined;
+this.theme="black_boxy";
+this.templateSuffix="";
+this.layout="portrait";
+this.scroll=false;
+this.textCase="lowercase";
+this.charList=[];
+this.localizedTable=Mojo.Locale.kbCharacters;
+this.localizedTableFull=Mojo.Locale.kbCharactersFull;
+},
+
+
+/* oskb */
+setup:function(){
+var model=this.controller.model;
+
+this.controller.exposeMethods(['close','isOpen']);
+if(this.controller.attributes.target){
+this.target=this.controller.get(this.controller.attributes.target);
+}else if(model.selectionTarget){
+this.target=this.controller.get(model.selectionTarget);
+}
+this.divPrefix=Mojo.View.makeUniqueId();
+this.currCode=this.controller.model.character;
+if(this.currCode!==undefined){
+this.chorded=true;
+}
+
+if(this.renderWidget(this.controller.model.character)){
+this.handleKeyEvent=this.handleKeyEvent.bind(this);
+this.handleKeyUpEvent=this.handleKeyUpEvent.bind(this);
+this.handleMouseEvent=this.handleMouseEvent.bind(this);
+this.handleOrientation=this.handleOrientation.bindAsEventListener(this);
+this.controller.listen(this.target,"keydown",this.handleKeyEvent,true);
+this.controller.listen(this.target,"keyup",this.handleKeyUpEvent,true);
+this.controller.listen(this.controller.document,Mojo.Event.tap,this.handleMouseEvent,true);
+this.controller.listen(this.controller.document,Mojo.Event.orientationChange,this.handleOrientation,true);
+
+if(this.chorded){
+
+this.state=this.ONSCREEN_KB_FILTERING_STATE;
+}else{
+this.enterOpenState();
+}
+this.controller.scene.pushContainer(this.controller.element,this.controller.scene.submenuContainerLayer,
+{cancelFunc:this._emptyAndClose.bind(this)});
+this.controller.scene.pushCommander(this);
+}
+},
+
+
+/* oskb */
+cleanup:function(){
+
+
+this.charPicker=undefined;
+this.selectedIndex=undefined;
+this.state=this.ONSCREEN_KB_CLOSED;
+this.cleanupEventListeners();
+},
+
+
+
+
+
+/* oskb */
+cleanupEventListeners:function(){
+this.controller.stopListening(this.target,"keydown",this.handleKeyEvent,true);
+this.controller.stopListening(this.target,"keyup",this.handleKeyUpEvent,true);
+this.controller.stopListening(this.controller.document,Mojo.Event.tap,this.handleMouseEvent,true);
+this.controller.stopListening(this.controller.document,Mojo.Event.orientationChange,this.handleOrientation,true);
+},
+
+
+/* oskb */
+loadTable:function(chr){
+var data,list;
+var i=0;
+var that=this;
+var table;
+var letter;
+var span;
+
+this.charList=[];
+table=this.localizedTableFull;
+
+table.each(function(c){
+/* TODO: get rid of list idea that doesn't need to exist for kb */
+  list = c.list;
+  letter = c.letter;
+  span = c.span;
+  if(list){
+    list.each(function(item){
+      data={
+        index:i,
+        character:item,
+        letter:letter,
+        span:span,
+        keyCode:c.keyCode
+      };
+
+      that.charList.push(data);
+      i++;
+    });
+  }
+  list=undefined;
+});
+
+if (this.theme !== "default") {
+  var themeTable = palmGetResource(this.ONSCREEN_KB_THEMES_PATH + this.theme + '/theme_config.json');
+  if (themeTable) {
+    var themeObject = Mojo.parseJSON(themeTable);
+    var idx;
+    var img;
+    var c;
+    var span;
+    var bg;
+    var class;
+    themeObject.each(function(t){
+      idx=t.index;
+      img=t.image;
+      c=t.character;
+      span=t.span;
+      class=t.class;
+      if ((idx>=0) && (idx<that.charList.length)) {
+        if (c) {
+          that.charList[idx].character=c;
+          if (img) {
+          that.charList[idx].character="<img name='" + idx + "' class='kb-img' src='"+that.ONSCREEN_KB_THEMES_PATH + that.theme + "/" + img + "' alt='" + c + "'>";
+          }
+          if (span) {
+            that.charList[idx].span = span;
+          }
+        }
+      }
+      if (!this.customStyle&&class) {
+        that.customStyle = t;
+      }
+    });
+  }
+}
+},
+
+/* oskb */
+_setPopupPositions:function(picker){
+var top='',left='';
+var cursorPos=Mojo.View.getCursorPosition(this.controller.window);
+var targetLeft;
+var pickerDims;
+var viewDims;
+var maxWidth,minWidth;
+
+/* TODO: go through this function */
+if(cursorPos){
+targetLeft=this.target.offsetLeft;
+viewDims=Mojo.View.getViewportDimensions(this.controller.document);
+
+
+pickerDims=Mojo.View.getDimensions(picker);
+
+
+if((pickerDims.height+this.HI_PADDING_BOTTOM+cursorPos.y)>viewDims.height){
+top=cursorPos.y-(pickerDims.height+this.HI_PADDING_TOP);
+if(top<this.HI_MINIMUM_TOP){
+top=this.HI_MINIMUM_TOP;
+}
+}else{
+top=cursorPos.y+cursorPos.height+this.HI_PADDING_BOTTOM;
+if((top+pickerDims.height)>(viewDims.height-this.HI_MAX_BOTTOM)){
+top=viewDims.height-this.HI_MAX_BOTTOM-pickerDims.height;
+}
+}
+
+left=cursorPos.x;
+maxWidth=viewDims.width-this.HI_PADDING_RIGHT;
+minWidth=targetLeft+this.HI_PADDING_LEFT;
+
+if((pickerDims.width+cursorPos.x)>maxWidth){
+
+left=maxWidth-pickerDims.width;
+}else if((cursorPos.x-pickerDims.width)<minWidth){
+
+left=minWidth;
+}
+
+left+='px';
+top+='px';
+}else if(this.target.type==='application/x-palm-browser'){
+
+}else{
+left='0px';
+top='0px';
+}
+
+left = '0px';
+this.charPicker.setStyle({'top':top,'left':left});
+},
+
+
+/* oskb */
+translateToRow:function(results){
+
+var finished=false;
+var result;
+var newOffset=0;
+var transformedResults=[];
+
+while(!finished){
+result={};
+
+
+result.characters=Mojo.View.render({collection:results.slice(newOffset,newOffset+this.HI_COLUMNS),attributes:{divPrefix:this.divPrefix},template:this.itemTemplate});
+newOffset+=this.HI_COLUMNS;
+transformedResults.push(result);
+if(newOffset>=results.length){
+finished=true;
+}
+}
+return transformedResults;
+},
+
+
+  /* oskb */
+renderWidget:function(chr){
+var data;
+var charContent;
+var charContentModel;
+var pickerContent;
+var parent;
+
+this.loadTable(chr);
+if(this.charList&&this.charList.length>0){
+
+charContentModel={
+divPrefix:this.divPrefix
+};
+
+this.templateSuffix = "";
+this.layout="portrait";
+
+if (this.controller.stageController) {
+  var orientation = this.controller.stageController.getWindowOrientation();
+  if (orientation === "left" || orientation === "right") {
+    this.templateSuffix = '-ls';
+    this.layout="landscape";
+  }
+}
+
+this.itemTemplate=Mojo.Widget.getSystemTemplatePath('/kbselector/char'+this.templateSuffix),
+this.itemsModel={items:this.translateToRow(this.charList)};
+this.charPicker=undefined;
+pickerContent=Mojo.View.render({object:charContentModel,template:Mojo.Widget.getSystemTemplatePath('/kbselector/charselector'+this.templateSuffix)});
+
+parent=Mojo.View.getScrollerForElement(this.target);
+if(!parent){
+parent=this.controller.scene.sceneElement;
+}
+if(this.controller.element.parentNode!==parent){
+this.controller.reparent(parent);
+}
+this.controller.element.innerHTML=pickerContent;
+/* If we are not set to scroll (i.e. symbol button not pressed), remove 
+ * the scroller widget before instantiating */
+if (!this.scroll) {
+  this.controller.get(this.divPrefix+"-char-selector").removeAttribute("x-mojo-element");
+}
+
+this.charPicker=this.controller.get(this.divPrefix+'-kb-char-selector-div');
+
+/*
+if (this.theme === "default") {
+  console.log("EG: ADD classname");
+  this.charPicker.addClassName("default");
+}
+*/
+
+if (this.customStyle) {
+  this.charPicker.setStyle(this.customStyle);
+}
+
+this.controller.scene.setupWidget('char-list',
+{itemTemplate:Mojo.Widget.getSystemTemplatePath('kbselector/char-selector-row'+this.templateSuffix),renderLimit:30},this.itemsModel);
+this.controller.instantiateChildWidgets(this.charPicker);
+/* Adjust the width based on the span number */
+for (var i=0; i<40; i++) {
+  if (this.charList[i].span) {
+    var newWidth = this.charList[i].span * 10 + "%";
+    this._getMatching(this.charPicker,i).setStyle({"width":newWidth});
+  }
+}
+this.controller.scene.showWidgetContainer(this.charPicker);
+this._setPopupPositions(this.charPicker);
+
+/*
+console.log("EG: HTML = " + this.charPicker.innerHTML);
+*/
+this.selectedIndex=0;
+/*
+this._updateSelected(null,this._selectedIdxElem());
+*/
+if(this._selectedIdxElem()){
+this.perLine=Math.floor(Element.getWidth(this.charPicker)/Element.getWidth(this._selectedIdxElem()));
+}else{
+this.perLine=0;
+}
+return true;
+}else{
+if(!this.chorded){
+this.exitSelector();
+return false;
+}else{
+return true;
+}
+}
+},
+
+
+/* oskb */
+enterOpenState:function(){
+
+this.state=this.ONSCREEN_KB_OPEN;
+},
+
+
+/* oskb */
+_maybeRemoveCharpicker:function(){
+if(this.charPicker){
+if(this.charPicker.parentNode){
+Element.remove(this.charPicker);
+}
+this.charPicker=undefined;
+}
+},
+
+
+/* oskb */
+enterFilteringState:function(keyCode){
+
+this.state=this.ONSCREEN_KB_FILTERING_STATE;
+
+if(this.currCode!==keyCode){
+this.currCode=keyCode;
+
+this._maybeRemoveCharpicker();
+this.renderWidget(this.currCode);
+}else{
+
+this.advance();
+}
+},
+
+
+/* oskb */
+handleModelChanged:function(model,what){
+Element.show(this.charPicker);
+if(Mojo.Char.isValid(this.controller.model.character)){
+this.enterFilteringState(this.controller.model.character);
+}
+},
+
+
+/* oskb */
+_emptyAndClose:function(){
+this.state=this.ONSCREEN_KB_EMPTY;
+this.close();
+},
+
+
+/* oskb */
+close:function(){
+
+if(this.state===this.ONSCREEN_KB_FILTERING_STATE||this.state===this.ONSCREEN_KB_OPEN){
+this.exitSelector(this.getEntered());
+return;
+}
+
+this._safeRemove();
+},
+
+
+/* oskb */
+isOpen:function(){
+return this.state!==this.ONSCREEN_KB_CLOSED;
+},
+
+/* oskb */
+isSpecialChar:function(keyCode){
+  return(Mojo.Char.isEnterKey(keyCode));
+},
+
+/* oskb */
+isSupported:function(keyCode){
+  return(keyCode && (keyCode !== Mojo.Char.altKey));
+},
+
+/* oskb */
+updateText:function(){
+  for (var i=0; i<28; i++) {
+    this.controller.get(this.divPrefix+"-"+i).setStyle({"text-transform":this.textCase});
+    switch(this.textCase) {
+      case 'uppercase':
+        this.charList[i].letter = this.charList[i].letter.toUpperCase();
+        break;
+      case 'lowercase':
+        this.charList[i].letter = this.charList[i].letter.toLowerCase();
+        break;
+    }
+  }
+},
+
+/* oskb */
+handleShift:function(keyCode){
+if (this.shiftState === this.SHIFT_OFF) {
+  this.shiftState = this.SHIFT_ONCE;
+  this.textCase = "uppercase";
+}
+else if (this.shiftState === this.SHIFT_ONCE) {
+  this.shiftState = this.SHIFT_LOCK;
+  this._updateSelected(null, this.controller.get(this.divPrefix+"-31"));
+  return;
+}
+else {
+  this.shiftState = this.SHIFT_OFF;
+  this.textCase = "lowercase";
+  this._updateSelected(this.controller.get(this.divPrefix+"-31"), null);
+}
+
+this.updateText();
+},
+
+/* oskb */
+exitSelector:function(chr){
+var letter;
+var characterVal,selection;
+var tagName=this.target.tagName;
+var selectionStart,selectionEnd;
+var isWebView=false;
+
+
+if(this.target.mojo&&this.target.mojo.insertStringAtCursor){
+isWebView=true;
+}
+
+if(chr){
+letter=chr.letter || chr.character || " ";
+selection=this.controller.window.getSelection();
+
+if (Mojo.Char.isDeleteKey(chr.keyCode)) {
+  this.controller.document.execCommand("Delete");
+}
+else if (this.isSpecialChar(chr.keyCode)) {
+  if (Mojo.Char.isEnterKey(chr.keyCode)) {
+    this.state=this.ONSCREEN_KB_CLOSED;
+    this.cleanupEventListeners();
+    this._safeRemove.bind(this).delay(0.2);
+  }
+  Mojo.Event.send(this.target,'keydown',{keyCode:chr.keyCode});
+  Mojo.Event.send(this.target,'keyup',{keyCode:chr.keyCode});
+}
+else if (chr.keyCode === Mojo.Char.shift) {
+  this.handleShift(chr.keyCode);
+}
+else if (this.isSymKey(chr.keyCode)) {
+  /* TODO: is this really the only way to dynamically start/stop scrolling? */
+  this.scroll = !this.scroll;
+  this.renderWidget();
+  if (this.scroll) {
+    this._updateSelected(null,this.controller.get(this.divPrefix+"-35"));
+  }
+  else {
+    this._updateSelected(this.controller.get(this.divPrefix+"-35"),null);
+  }
+  
+  return;
+}
+else if (!this.isSupported(chr.keyCode)) {
+  console.log("Char keycode " + chr.keyCode + " not supported (yet)");
+  return;
+}
+else if(selection&&selection.rangeCount>0&&selection.getRangeAt(0)){
+this.controller.document.execCommand("insertText",true,letter);
+}else if(isWebView&&letter!==null&&letter!==undefined){
+this.target.mojo.insertStringAtCursor(letter);
+}
+
+
+if(this.target.mojo&&this.target.mojo.setText){
+selectionStart=this.target.selectionStart;
+selectionEnd=this.target.selectionEnd;
+this.target.mojo.setText(this.target.value);
+this.target.selectionStart=selectionStart;
+this.target.selectionEnd=selectionEnd;
+}
+
+
+/*
+this.cleanupEventListeners();
+this._safeRemove.bind(this).delay(0.2);
+*/
+}else{
+this._safeRemove();
+}
+
+if(!isWebView){
+this.target.focus();
+}
+if (this.shiftState === this.SHIFT_ONCE && chr.keyCode !== Mojo.Char.shift){
+  this.shiftState = this.SHIFT_OFF;
+  this.textCase = "lowercase";
+  this.updateText();
+}
+},
+
+
+/* oskb */
+_safeRemove:function(){
+this.controller.scene.removeContainer(this.controller.element);
+if(this.controller.element&&this.controller.element.parentNode){
+Element.remove(this.controller.element);
+}
+},
+
+
+/* oskb */
+_insertChar:function(origValue,letter,start,end){
+var value='';
+if(origValue){
+value=origValue.substring(0,start);
+value+=letter;
+value+=origValue.substring(end,origValue.length);
+}else{
+value=letter;
+}
+return value;
+},
+
+
+/* oskb */
+advance:function(){
+var old=this._selectedIdxElem();
+var newElm;
+
+if(this.selectedIndex+1>this.charList.length-1){
+this.selectedIndex=0;
+}else{
+this.selectedIndex++;
+}
+
+/* oskb */
+newElm=this._selectedIdxElem();
+this._updateSelected(old,newElm);
+},
+
+
+/* oskb */
+retreat:function(){
+var old,newElm;
+
+old=this._selectedIdxElem();
+if(this.selectedIndex===0){
+this.selectedIndex=this.charList.length-1;
+}else{
+this.selectedIndex=this.selectedIndex-1;
+}
+newElm=this._selectedIdxElem();
+this._updateSelected(old,newElm);
+},
+
+
+/* oskb */
+_getMatching:function(element,query){
+if(!element){
+return;
+}
+return element.querySelector("[name='"+query+"']");
+},
+
+
+/* oskb */
+_updateSelected:function(oldSelection,newSelection){
+var node;
+if(oldSelection){
+node=this._getMatching(oldSelection,oldSelection.getAttribute("name"));
+if(node){
+node.removeClassName("selected-char");
+}
+}
+if(newSelection){
+node=this._getMatching(newSelection,newSelection.getAttribute("name"));
+if(node){
+node.addClassName("selected-char");
+}
+}
+},
+
+
+
+/* oskb */
+moveDown:function(){
+var old,newElm;
+if(this.selectedIndex+this.perLine<this.charList.length){
+old=this._selectedIdxElement();
+this.selectedIndex=this.selectedIndex+this.perLine;
+newElm=this._selectedIdxElement();
+this._updateSelected(old,newElm);
+}
+},
+
+
+/* oskb */
+moveUp:function(){
+var old,newElm;
+if(this.selectedIndex-this.perLine>=0){
+old=this._selectedIdxElem();
+this.selectedIndex=this.selectedIndex-this.perLine;
+newElm=this._selectedIdxElem();
+this._updatedSelected(old,newElm);
+}
+},
+
+
+/* oskb */
+updatePosition:function(key){
+switch(key){
+case Mojo.Char.leftArrow:
+this.retreat();
+break;
+case Mojo.Char.upArrow:
+this.moveUp();
+break;
+case Mojo.Char.rightArrow:
+this.advance();
+break;
+case Mojo.Char.downArrow:
+this.moveDown();
+break;
+default:
+break;
+}
+
+if(this.charPicker){
+this.controller.get(this.divPrefix+'-char-selector').mojo.revealElement(this._selectedIdxElem());
+}
+},
+
+
+
+/* oskb */
+handleKeyUpEvent:function(event){
+var keyCode=event.keyCode;
+var chr;
+
+this.exitSelector();
+return;
+
+if(this.isSymKey(keyCode)){
+if(this.state===this.ONSCREEN_KB_FILTERING_STATE){
+chr=this.getEntered();
+}
+this.exitSelector(chr);
+Event.stop(event);
+return;
+}
+},
+
+
+/* oskb */
+handleKeyEvent:function(event){
+var keyCode=event.keyCode;
+
+if(Mojo.Char.isEnterKey(keyCode)){
+this.exitSelector(this.getEntered());
+Event.stop(event);
+
+return;
+}
+if(Mojo.Char.isDeleteKey(keyCode)){
+this.exitSelector();
+Event.stop(event);
+return;
+}
+if(this.isDirectionalKey(keyCode)){
+this.updatePosition(keyCode);
+Event.stop(event);
+return;
+}
+
+if(!Mojo.Char.isValid(keyCode)){
+return;
+}
+
+switch(this.state){
+case this.ONSCREEN_KB_OPEN:
+case this.ONSCREEN_KB_FILTERING_STATE:
+case this.ONSCREEN_KB_EMPTY:
+this.enterFilteringState(keyCode);
+Event.stop(event);
+break;
+default:
+break;
+}
+},
+
+maybeChangeLayout:function(){
+  var orientation = this.controller.stageController.getWindowOrientation();
+  switch (this.layout) {
+    case "portrait":
+      if (orientation==="left" || orientation==="right") {
+        this.renderWidget();
+      }
+    break;
+    case "landscape":
+      if (orientation==="up" || orientation==="down") {
+        this.renderWidget();
+      }
+    break;
+  }
+},
+
+/* oskb */
+handleOrientation:function(event){
+/*TODO: I'm not sure why, but every change I get two events, back to back.
+ *      For now just ignore the 2nd orient event */
+console.log("ORIENT position" + event.orientation);
+
+if (this.ignoreSecond) {
+  this.ignoreSecond=false;
+  return;
+}
+else {
+  this.ignoreSecond=true;
+  this.maybeChangeLayout();
+}
+},
+
+/* oskb */
+handleMouseEvent:function(event){
+
+switch(this.state){
+case this.ONSCREEN_KB_OPEN:
+case this.ONSCREEN_KB_FILTERING_STATE:
+case this.ONSCREEN_KB_SINGLEFILTER_STATE:
+if(this.isInCharPicker(event.target)){
+console.log("kb CHAR MOUSE EVENT"+this.getSelected(event.target));
+this.exitSelector(this.getSelected(event.target));
+event.stop();
+}else{
+this.exitSelector();
+}
+break;
+default:
+break;
+}
+},
+
+
+/* oskb */
+getEntered:function(){
+return this.charList[this.selectedIndex];
+},
+
+
+/* oskb */
+getSelected:function(target){
+var chr=target.getAttribute('name');
+
+return this.charList[chr];
+},
+
+
+
+
+
+
+/* oskb */
+isDirectionalKey:function(key){
+if(key==Mojo.Char.leftArrow||key==Mojo.Char.upArrow||key==Mojo.Char.rightArrow||key==Mojo.Char.downArrow){
+return true;
+}
+return false;
+},
+
+
+/* oskb */
+isInCharPicker:function(target){
+if(!this.charPicker){
+return;
+}
+if(target.id==this.charPicker.id||Element.up(target,'div#'+this.charPicker.id)){
+return true;
+}
+return false;
+},
+
+
+/* oskb */
+isSymKey:function(keyCode){
+return keyCode===Mojo.Char.sym;
+},
+
+
+/* oskb */
+_selectedIdxElem:function(){
+return this.controller.get(this.divPrefix+"-"+this.selectedIndex);
+},
+
+
+/* oskb */
+handleCommand:function(commandEvent){
+if(commandEvent.type===Mojo.Event.back&&(this.state!==this.ONSCREEN_KB_CLOSED&&this.state!==this.ONSCREEN_KB_EMPTY)){
+this.exitSelector();
+Event.stop(commandEvent);
+}
+}
+});
+
 /* Compressed by the perl version of jsmin. */
 /* JavaScript::Minifier 0.02 */
 
diff --git a/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_boxy/@.png b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_boxy/@.png
new file mode 100644
index 0000000..3eed96d
Binary files /dev/null and b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_boxy/@.png differ
diff --git a/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_boxy/a.png b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_boxy/a.png
new file mode 100644
index 0000000..7a378c6
Binary files /dev/null and b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_boxy/a.png differ
diff --git a/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_boxy/b.png b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_boxy/b.png
new file mode 100644
index 0000000..68bbaad
Binary files /dev/null and b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_boxy/b.png differ
diff --git a/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_boxy/backspace.png b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_boxy/backspace.png
new file mode 100644
index 0000000..ffc68d4
Binary files /dev/null and b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_boxy/backspace.png differ
diff --git a/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_boxy/blank.png b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_boxy/blank.png
new file mode 100644
index 0000000..f273074
Binary files /dev/null and b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_boxy/blank.png differ
diff --git a/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_boxy/c.png b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_boxy/c.png
new file mode 100644
index 0000000..32fa7d1
Binary files /dev/null and b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_boxy/c.png differ
diff --git a/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_boxy/comma.png b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_boxy/comma.png
new file mode 100644
index 0000000..981c2b1
Binary files /dev/null and b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_boxy/comma.png differ
diff --git a/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_boxy/d.png b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_boxy/d.png
new file mode 100644
index 0000000..dd68b7a
Binary files /dev/null and b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_boxy/d.png differ
diff --git a/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_boxy/e.png b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_boxy/e.png
new file mode 100644
index 0000000..d939021
Binary files /dev/null and b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_boxy/e.png differ
diff --git a/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_boxy/enter.png b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_boxy/enter.png
new file mode 100644
index 0000000..2eaece3
Binary files /dev/null and b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_boxy/enter.png differ
diff --git a/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_boxy/f.png b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_boxy/f.png
new file mode 100644
index 0000000..9b7c223
Binary files /dev/null and b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_boxy/f.png differ
diff --git a/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_boxy/function.png b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_boxy/function.png
new file mode 100644
index 0000000..14ac6d1
Binary files /dev/null and b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_boxy/function.png differ
diff --git a/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_boxy/g.png b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_boxy/g.png
new file mode 100644
index 0000000..681ea61
Binary files /dev/null and b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_boxy/g.png differ
diff --git a/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_boxy/h.png b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_boxy/h.png
new file mode 100644
index 0000000..51f786b
Binary files /dev/null and b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_boxy/h.png differ
diff --git a/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_boxy/i.png b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_boxy/i.png
new file mode 100644
index 0000000..6f9913e
Binary files /dev/null and b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_boxy/i.png differ
diff --git a/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_boxy/j.png b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_boxy/j.png
new file mode 100644
index 0000000..b96a617
Binary files /dev/null and b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_boxy/j.png differ
diff --git a/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_boxy/k.png b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_boxy/k.png
new file mode 100644
index 0000000..4a17143
Binary files /dev/null and b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_boxy/k.png differ
diff --git a/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_boxy/l.png b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_boxy/l.png
new file mode 100644
index 0000000..a6c6f09
Binary files /dev/null and b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_boxy/l.png differ
diff --git a/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_boxy/m.png b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_boxy/m.png
new file mode 100644
index 0000000..8563da8
Binary files /dev/null and b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_boxy/m.png differ
diff --git a/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_boxy/n.png b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_boxy/n.png
new file mode 100644
index 0000000..456f89c
Binary files /dev/null and b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_boxy/n.png differ
diff --git a/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_boxy/o.png b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_boxy/o.png
new file mode 100644
index 0000000..ca36c55
Binary files /dev/null and b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_boxy/o.png differ
diff --git a/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_boxy/p.png b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_boxy/p.png
new file mode 100644
index 0000000..60e1b0e
Binary files /dev/null and b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_boxy/p.png differ
diff --git a/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_boxy/period.png b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_boxy/period.png
new file mode 100644
index 0000000..175dc28
Binary files /dev/null and b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_boxy/period.png differ
diff --git a/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_boxy/q.png b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_boxy/q.png
new file mode 100644
index 0000000..d9f7441
Binary files /dev/null and b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_boxy/q.png differ
diff --git a/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_boxy/r.png b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_boxy/r.png
new file mode 100644
index 0000000..70646b6
Binary files /dev/null and b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_boxy/r.png differ
diff --git a/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_boxy/s.png b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_boxy/s.png
new file mode 100644
index 0000000..cb2e49f
Binary files /dev/null and b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_boxy/s.png differ
diff --git a/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_boxy/shift.png b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_boxy/shift.png
new file mode 100644
index 0000000..6f17bac
Binary files /dev/null and b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_boxy/shift.png differ
diff --git a/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_boxy/space.png b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_boxy/space.png
new file mode 100644
index 0000000..91273e7
Binary files /dev/null and b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_boxy/space.png differ
diff --git a/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_boxy/sym.png b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_boxy/sym.png
new file mode 100644
index 0000000..5cf07de
Binary files /dev/null and b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_boxy/sym.png differ
diff --git a/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_boxy/t.png b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_boxy/t.png
new file mode 100644
index 0000000..b374c77
Binary files /dev/null and b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_boxy/t.png differ
diff --git a/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_boxy/theme_config.json b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_boxy/theme_config.json
new file mode 100644
index 0000000..a7dc887
--- /dev/null
+++ b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_boxy/theme_config.json
@@ -0,0 +1,43 @@
+[
+  {"index":0, "character":"q", "image":"q.png"},
+  {"index":1, "character":"w", "image":"w.png"},
+  {"index":2, "character":"e", "image":"e.png"},
+  {"index":3, "character":"r", "image":"r.png"},
+  {"index":4, "character":"t", "image":"t.png"},
+  {"index":5, "character":"y", "image":"y.png"},
+  {"index":6, "character":"u", "image":"u.png"},
+  {"index":7, "character":"i", "image":"i.png"},
+  {"index":8, "character":"o", "image":"o.png"},
+  {"index":9, "character":"p", "image":"p.png"},
+
+  {"index":10, "character":"a", "image":"a.png"},
+  {"index":11, "character":"s", "image":"s.png"},
+  {"index":12, "character":"d", "image":"d.png"},
+  {"index":13, "character":"f", "image":"f.png"},
+  {"index":14, "character":"g", "image":"g.png"},
+  {"index":15, "character":"h", "image":"h.png"},
+  {"index":16, "character":"j", "image":"j.png"},
+  {"index":17, "character":"k", "image":"k.png"},
+  {"index":18, "character":"l", "image":"l.png"},
+  {"index":19, "character":"\b", "image":"backspace.png"},
+
+  {"index":20, "character":"alt-orange", "image":"function.png"},
+  {"index":21, "character":"z", "image":"z.png"},
+  {"index":22, "character":"x", "image":"x.png"},
+  {"index":23, "character":"c", "image":"c.png"},
+  {"index":24, "character":"v", "image":"v.png"},
+  {"index":25, "character":"b", "image":"b.png"},
+  {"index":26, "character":"n", "image":"n.png"},
+  {"index":27, "character":"m", "image":"m.png"},
+  {"index":28, "character":",", "image":"comma.png"},
+  {"index":29, "character":"\n", "image":"enter.png"},
+
+  {"index":30, "character":"blank", "image":"blank.png"},
+  {"index":31, "character":"shift", "image":"shift.png"},
+  {"index":32, "character":"@", "image":"@.png"},
+  {"index":33, "character":" ", "image":"space.png"},
+  {"index":34, "character":".", "image":"period.png"},
+  {"index":35, "character":"sym-ctrl", "image":"sym.png"},
+  {"index":36, "character":"blank", "image":"blank.png"},
+
+]
diff --git a/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_boxy/u.png b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_boxy/u.png
new file mode 100644
index 0000000..d2de1cd
Binary files /dev/null and b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_boxy/u.png differ
diff --git a/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_boxy/v.png b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_boxy/v.png
new file mode 100644
index 0000000..82513bc
Binary files /dev/null and b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_boxy/v.png differ
diff --git a/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_boxy/w.png b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_boxy/w.png
new file mode 100644
index 0000000..8630f9a
Binary files /dev/null and b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_boxy/w.png differ
diff --git a/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_boxy/x.png b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_boxy/x.png
new file mode 100644
index 0000000..f8d287a
Binary files /dev/null and b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_boxy/x.png differ
diff --git a/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_boxy/y.png b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_boxy/y.png
new file mode 100644
index 0000000..37e6ecd
Binary files /dev/null and b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_boxy/y.png differ
diff --git a/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_boxy/z.png b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_boxy/z.png
new file mode 100644
index 0000000..4119966
Binary files /dev/null and b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_boxy/z.png differ
diff --git a/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_glossy/@.png b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_glossy/@.png
new file mode 100644
index 0000000..6504917
Binary files /dev/null and b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_glossy/@.png differ
diff --git a/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_glossy/a.png b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_glossy/a.png
new file mode 100644
index 0000000..7e15e46
Binary files /dev/null and b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_glossy/a.png differ
diff --git a/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_glossy/b.png b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_glossy/b.png
new file mode 100644
index 0000000..b995d25
Binary files /dev/null and b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_glossy/b.png differ
diff --git a/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_glossy/backspace.png b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_glossy/backspace.png
new file mode 100644
index 0000000..18212ad
Binary files /dev/null and b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_glossy/backspace.png differ
diff --git a/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_glossy/blank.png b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_glossy/blank.png
new file mode 100644
index 0000000..e3589cf
Binary files /dev/null and b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_glossy/blank.png differ
diff --git a/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_glossy/c.png b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_glossy/c.png
new file mode 100644
index 0000000..968c81d
Binary files /dev/null and b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_glossy/c.png differ
diff --git a/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_glossy/comma.png b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_glossy/comma.png
new file mode 100644
index 0000000..2c9f9d3
Binary files /dev/null and b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_glossy/comma.png differ
diff --git a/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_glossy/d.png b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_glossy/d.png
new file mode 100644
index 0000000..2cb8f78
Binary files /dev/null and b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_glossy/d.png differ
diff --git a/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_glossy/e.png b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_glossy/e.png
new file mode 100644
index 0000000..7c8456b
Binary files /dev/null and b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_glossy/e.png differ
diff --git a/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_glossy/enter.png b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_glossy/enter.png
new file mode 100644
index 0000000..5b4ee9e
Binary files /dev/null and b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_glossy/enter.png differ
diff --git a/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_glossy/f.png b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_glossy/f.png
new file mode 100644
index 0000000..3d0f471
Binary files /dev/null and b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_glossy/f.png differ
diff --git a/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_glossy/function.png b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_glossy/function.png
new file mode 100644
index 0000000..9d97f9b
Binary files /dev/null and b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_glossy/function.png differ
diff --git a/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_glossy/g.png b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_glossy/g.png
new file mode 100644
index 0000000..1da2a45
Binary files /dev/null and b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_glossy/g.png differ
diff --git a/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_glossy/h.png b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_glossy/h.png
new file mode 100644
index 0000000..358438b
Binary files /dev/null and b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_glossy/h.png differ
diff --git a/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_glossy/i.png b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_glossy/i.png
new file mode 100644
index 0000000..e042b65
Binary files /dev/null and b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_glossy/i.png differ
diff --git a/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_glossy/j.png b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_glossy/j.png
new file mode 100644
index 0000000..d2c1bff
Binary files /dev/null and b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_glossy/j.png differ
diff --git a/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_glossy/k.png b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_glossy/k.png
new file mode 100644
index 0000000..0cdecde
Binary files /dev/null and b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_glossy/k.png differ
diff --git a/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_glossy/l.png b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_glossy/l.png
new file mode 100644
index 0000000..f6692c1
Binary files /dev/null and b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_glossy/l.png differ
diff --git a/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_glossy/m.png b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_glossy/m.png
new file mode 100644
index 0000000..b6295cd
Binary files /dev/null and b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_glossy/m.png differ
diff --git a/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_glossy/n.png b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_glossy/n.png
new file mode 100644
index 0000000..113d404
Binary files /dev/null and b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_glossy/n.png differ
diff --git a/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_glossy/o.png b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_glossy/o.png
new file mode 100644
index 0000000..a8bb522
Binary files /dev/null and b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_glossy/o.png differ
diff --git a/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_glossy/p.png b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_glossy/p.png
new file mode 100644
index 0000000..fbdf8f5
Binary files /dev/null and b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_glossy/p.png differ
diff --git a/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_glossy/period.png b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_glossy/period.png
new file mode 100644
index 0000000..4af2e1e
Binary files /dev/null and b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_glossy/period.png differ
diff --git a/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_glossy/q.png b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_glossy/q.png
new file mode 100644
index 0000000..e82e1a9
Binary files /dev/null and b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_glossy/q.png differ
diff --git a/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_glossy/r.png b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_glossy/r.png
new file mode 100644
index 0000000..69933e7
Binary files /dev/null and b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_glossy/r.png differ
diff --git a/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_glossy/s.png b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_glossy/s.png
new file mode 100644
index 0000000..047cf16
Binary files /dev/null and b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_glossy/s.png differ
diff --git a/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_glossy/shift.png b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_glossy/shift.png
new file mode 100644
index 0000000..6707923
Binary files /dev/null and b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_glossy/shift.png differ
diff --git a/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_glossy/space.png b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_glossy/space.png
new file mode 100644
index 0000000..1f24711
Binary files /dev/null and b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_glossy/space.png differ
diff --git a/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_glossy/sym.png b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_glossy/sym.png
new file mode 100644
index 0000000..81597d8
Binary files /dev/null and b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_glossy/sym.png differ
diff --git a/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_glossy/t.png b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_glossy/t.png
new file mode 100644
index 0000000..a4324fe
Binary files /dev/null and b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_glossy/t.png differ
diff --git a/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_glossy/theme_config.json b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_glossy/theme_config.json
new file mode 100644
index 0000000..a7dc887
--- /dev/null
+++ b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_glossy/theme_config.json
@@ -0,0 +1,43 @@
+[
+  {"index":0, "character":"q", "image":"q.png"},
+  {"index":1, "character":"w", "image":"w.png"},
+  {"index":2, "character":"e", "image":"e.png"},
+  {"index":3, "character":"r", "image":"r.png"},
+  {"index":4, "character":"t", "image":"t.png"},
+  {"index":5, "character":"y", "image":"y.png"},
+  {"index":6, "character":"u", "image":"u.png"},
+  {"index":7, "character":"i", "image":"i.png"},
+  {"index":8, "character":"o", "image":"o.png"},
+  {"index":9, "character":"p", "image":"p.png"},
+
+  {"index":10, "character":"a", "image":"a.png"},
+  {"index":11, "character":"s", "image":"s.png"},
+  {"index":12, "character":"d", "image":"d.png"},
+  {"index":13, "character":"f", "image":"f.png"},
+  {"index":14, "character":"g", "image":"g.png"},
+  {"index":15, "character":"h", "image":"h.png"},
+  {"index":16, "character":"j", "image":"j.png"},
+  {"index":17, "character":"k", "image":"k.png"},
+  {"index":18, "character":"l", "image":"l.png"},
+  {"index":19, "character":"\b", "image":"backspace.png"},
+
+  {"index":20, "character":"alt-orange", "image":"function.png"},
+  {"index":21, "character":"z", "image":"z.png"},
+  {"index":22, "character":"x", "image":"x.png"},
+  {"index":23, "character":"c", "image":"c.png"},
+  {"index":24, "character":"v", "image":"v.png"},
+  {"index":25, "character":"b", "image":"b.png"},
+  {"index":26, "character":"n", "image":"n.png"},
+  {"index":27, "character":"m", "image":"m.png"},
+  {"index":28, "character":",", "image":"comma.png"},
+  {"index":29, "character":"\n", "image":"enter.png"},
+
+  {"index":30, "character":"blank", "image":"blank.png"},
+  {"index":31, "character":"shift", "image":"shift.png"},
+  {"index":32, "character":"@", "image":"@.png"},
+  {"index":33, "character":" ", "image":"space.png"},
+  {"index":34, "character":".", "image":"period.png"},
+  {"index":35, "character":"sym-ctrl", "image":"sym.png"},
+  {"index":36, "character":"blank", "image":"blank.png"},
+
+]
diff --git a/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_glossy/u.png b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_glossy/u.png
new file mode 100644
index 0000000..0077ad6
Binary files /dev/null and b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_glossy/u.png differ
diff --git a/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_glossy/v.png b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_glossy/v.png
new file mode 100644
index 0000000..92e09ab
Binary files /dev/null and b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_glossy/v.png differ
diff --git a/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_glossy/w.png b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_glossy/w.png
new file mode 100644
index 0000000..787be5b
Binary files /dev/null and b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_glossy/w.png differ
diff --git a/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_glossy/x.png b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_glossy/x.png
new file mode 100644
index 0000000..7b1f1ee
Binary files /dev/null and b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_glossy/x.png differ
diff --git a/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_glossy/y.png b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_glossy/y.png
new file mode 100644
index 0000000..795ebf3
Binary files /dev/null and b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_glossy/y.png differ
diff --git a/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_glossy/z.png b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_glossy/z.png
new file mode 100644
index 0000000..0f466d4
Binary files /dev/null and b/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/black_glossy/z.png differ
diff --git a/usr/palm/frameworks/mojo/submissions/191.15/resources/en_us/kbchars_fulltable.json b/usr/palm/frameworks/mojo/submissions/191.15/resources/en_us/kbchars_fulltable.json
new file mode 100644
index 0000000..e0e96d0
--- /dev/null
+++ b/usr/palm/frameworks/mojo/submissions/191.15/resources/en_us/kbchars_fulltable.json
@@ -0,0 +1,86 @@
+[
+  {"keyCode": 81, "letter":"q", "list": ["q"]},
+  {"keyCode": 87, "letter":"w", "list": ["w"]},
+  {"keyCode": 69, "letter":"e", "list": ["e"]},
+  {"keyCode": 82, "letter":"r", "list": ["r"]},
+  {"keyCode": 84, "letter":"t", "list": ["t"]},
+  {"keyCode": 89, "letter":"y", "list": ["y"]},
+  {"keyCode": 85, "letter":"u", "list": ["u"]},
+  {"keyCode": 73, "letter":"i", "list": ["i"]},
+  {"keyCode": 79, "letter":"o", "list": ["o"]},
+  {"keyCode": 80, "letter":"p", "list": ["p"]},
+
+  {"keyCode": 65, "letter":"a", "list": ["a"]},
+  {"keyCode": 83, "letter":"s", "list": ["s"]},
+  {"keyCode": 68, "letter":"d", "list": ["d"]},
+  {"keyCode": 70, "letter":"f", "list": ["f"]},
+  {"keyCode": 71, "letter":"g", "list": ["g"]},
+  {"keyCode": 72, "letter":"h", "list": ["h"]},
+  {"keyCode": 74, "letter":"j", "list": ["j"]},
+  {"keyCode": 75, "letter":"k", "list": ["k"]},
+  {"keyCode": 76, "letter":"l", "list": ["l"]},
+  {"keyCode": 8, "letter":"\b", "list": ["&laquo"]},
+
+  {"keyCode": 18, "letter":"alt-orange", "list": ["&curren"]},
+  {"keyCode": 90, "letter":"z", "list": ["z"]},
+  {"keyCode": 88, "letter":"x", "list": ["x"]},
+  {"keyCode": 67, "letter":"c", "list": ["c"]},
+  {"keyCode": 86, "letter":"v", "list": ["v"]},
+  {"keyCode": 66, "letter":"b", "list": ["b"]},
+  {"keyCode": 78, "letter":"n", "list": ["n"]},
+  {"keyCode": 77, "letter":"m", "list": ["m"]},
+  {"keyCode": 188, "letter":",", "list": [","]},
+  {"keyCode": 13, "letter":"\n", "list": ["&raquo"]},
+
+  {"keyCode": 00, "letter":"blank", "list": ["&deg"]},
+  {"keyCode": 16, "letter":"shift", "list": ["&yen"], "span": 1.5},
+  {"keyCode": 90, "letter":"@", "list": ["@"]},
+  {"keyCode": 32, "letter":" ", "list": ["&nbsp"], "span": 3},
+  {"keyCode": 190, "letter":".", "list": ["."]},
+  {"keyCode": 17, "letter":"sym-ctrl", "list": ["&iquest"], "span": 1.5},
+  {"keyCode": 00, "letter":"blank", "list": ["&deg"]},
+
+  {"keyCode": 190, "letter":"", "list": ["/","?","+","(",")","%","\"","=","&","-","$","!",":","'","&","#","_"]},
+  {"keyCode": 48, "letter":"0", "list": ["0"]},
+  {"keyCode": 49, "letter":"1", "list": ["1"]},
+  {"keyCode": 50, "letter":"2", "list": ["2"]},
+  {"keyCode": 51, "letter":"3", "list": ["3"]},
+  {"keyCode": 52, "letter":"4", "list": ["4"]},
+  {"keyCode": 53, "letter":"5", "list": ["5"]},
+  {"keyCode": 54, "letter":"6", "list": ["6"]},
+  {"keyCode": 55, "letter":"7", "list": ["7"]},
+  {"keyCode": 56, "letter":"8", "list": ["8"]},
+  {"keyCode": 57, "letter":"9", "list": ["9"]},
+
+  {"keyCode": 190, "letter":"", "list": ["~","\\","•","`","÷","^","[","]","{","}","<",">","«","»","Ø","µ","|"]},
+  {"keyCode": 65,  "letter":"", "list": ["á","à","ä","â","ã","å","æ"]},
+  {"keyCode": 65,  "letter":"", "list": ["Á","À","Ä","Â","Ã","Å","Æ"], "modifier": 16},
+  {"keyCode": 67,  "letter":"", "list": ["©","ç"]},
+  {"keyCode": 67,  "letter":"", "list": ["Ç"], "modifier": 16},
+  {"keyCode": 69,  "letter":"", "list": ["é","è","ë","ê","ē"]},
+  {"keyCode": 69,  "letter":"", "list": ["É","È","Ë","Ê"], "modifier": 16},
+  {"keyCode": 73,  "letter":"", "list": ["í","ì","ï","î"]},
+  {"keyCode": 73,  "letter":"", "list": ["Í","Ì","Ï","Î"], "modifier": 16},
+  {"keyCode": 78,  "letter":"", "list": ["ñ"]},
+  {"keyCode": 78,  "letter":"", "list": ["Ñ"], "modifier": 16},
+  {"keyCode": 79,  "letter":"", "list": ["ó","ò","ö","ô","œ","õ","ø"]},
+  {"keyCode": 79,  "letter":"", "list": ["Ó","Ò","Ö","Ô","Œ","Õ","Ø"], "modifier": 16},
+  {"keyCode": 83,  "letter":"", "list": ["ß"]},
+  {"keyCode": 85,  "letter":"", "list": ["ú","ù","ü","û"]},
+  {"keyCode": 85,  "letter":"", "list": ["Ú","Ù","Ü","Û"], "modifier": 16},
+  {"keyCode": 88,  "letter":"", "list": ["×","¤"]},
+  {"keyCode": 89,  "letter":"", "list": ["ÿ"]},
+  {"keyCode": 89,  "letter":"", "list": ["Ÿ"], "modifier": 16},
+  {"keyCode": 82,  "letter":"", "list": ["®"]},
+  {"keyCode": 80,  "letter":"", "list": ["¶"]},
+  {"keyCode": 84,  "letter":"", "list": ["™"]},
+  {"keyCode": 68,  "letter":"", "list": ["†","‡"], "modifier": 16},
+  {"keyCode": 96,  "letter":"", "list": ["°"]},
+  {"keyCode": 97,  "letter":"", "list": ["¹", "¼", "½"]},
+  {"keyCode": 98,  "letter":"", "list": ["²"]},
+  {"keyCode": 99,  "letter":"", "list": ["³", "¾"]},
+  {"keyCode": 67,  "letter":"", "list": ["¡"]},
+  {"keyCode": 67,  "letter":"", "list": ["¿"]},
+  {"keyCode": 67,  "letter":"", "list": [":-)", ":-(", ";-)"]},
+  {"keyCode": 67,  "letter":"", "list": ["$","¢","€","£","¥"]}
+]
diff --git a/usr/palm/frameworks/mojo/submissions/191.15/resources/en_us/kbchars_fulltable_qwertz.json b/usr/palm/frameworks/mojo/submissions/191.15/resources/en_us/kbchars_fulltable_qwertz.json
new file mode 100644
index 0000000..d4a8128
--- /dev/null
+++ b/usr/palm/frameworks/mojo/submissions/191.15/resources/en_us/kbchars_fulltable_qwertz.json
@@ -0,0 +1,58 @@
+[
+  {"keyCode": 65, "letter":"a", "list": ["a"]},
+  {"keyCode": 66, "letter":"b", "list": ["b"]},
+  {"keyCode": 67, "letter":"c", "list": ["c"]},
+  {"keyCode": 68, "letter":"d", "list": ["d"]},
+  {"keyCode": 69, "letter":"e", "list": ["e"]},
+  {"keyCode": 70, "letter":"f", "list": ["f"]},
+  {"keyCode": 70, "letter":"g", "list": ["g"]},
+  {"keyCode": 70, "letter":"h", "list": ["h"]},
+  {"keyCode": 70, "letter":"i", "list": ["i"]},
+  {"keyCode": 70, "letter":"j", "list": ["j"]},
+  {"keyCode": 70, "letter":"k", "list": ["k"]},
+  {"keyCode": 70, "letter":"l", "list": ["l"]},
+  {"keyCode": 70, "letter":"m", "list": ["m"]},
+  {"keyCode": 70, "letter":"n", "list": ["n"]},
+  {"keyCode": 70, "letter":"o", "list": ["o"]},
+  {"keyCode": 80, "letter":"p", "list": ["p"]},
+  {"keyCode": 80, "letter":"q", "list": ["q"]},
+  {"keyCode": 80, "letter":"r", "list": ["r"]},
+  {"keyCode": 80, "letter":"s", "list": ["s"]},
+  {"keyCode": 80, "letter":"t", "list": ["t"]},
+  {"keyCode": 80, "letter":"u", "list": ["u"]},
+  {"keyCode": 80, "letter":"v", "list": ["v"]},
+  {"keyCode": 80, "letter":"w", "list": ["w"]},
+  {"keyCode": 80, "letter":"x", "list": ["x"]},
+  {"keyCode": 80, "letter":"y", "list": ["y"]},
+  {"keyCode": 90, "letter":"z", "list": ["z"]},
+  {"keyCode": 90, "letter":"z", "list": ["z"]},
+  {"keyCode": 65, "letter":"A", "list": ["A"], "modifier": 16},
+  {"keyCode": 67, "letter":"c", "list": ["©","ç"]},
+  {"keyCode": 67, "letter":"C", "list": ["Ç"], "modifier": 16},
+  {"keyCode": 69, "letter":"e", "list": ["é","è","ë","ê","ē"]},
+  {"keyCode": 69, "letter":"E", "list": ["É","È","Ë","Ê"], "modifier": 16},
+  {"keyCode": 73, "letter":"i", "list": ["í","ì","ï","î"]},
+  {"keyCode": 73, "letter":"I", "list": ["Í","Ì","Ï","Î"], "modifier": 16},
+  {"keyCode": 78, "letter":"n", "list": ["ñ"]},
+  {"keyCode": 78, "letter":"N", "list": ["Ñ"], "modifier": 16},
+  {"keyCode": 79, "letter":"o", "list": ["ó","ò","ö","ô","œ","õ","ø"]},
+  {"keyCode": 79, "letter":"O", "list": ["Ó","Ò","Ö","Ô","Œ","Õ","Ø"], "modifier": 16},
+  {"keyCode": 83, "letter":"s", "list": ["ß"]},
+  {"keyCode": 85, "letter":"u", "list": ["ú","ù","ü","û"]},
+  {"keyCode": 85, "letter":"U", "list": ["Ú","Ù","Ü","Û"], "modifier": 16},
+  {"keyCode": 88, "letter":"x", "list": ["×","¤"]},
+  {"keyCode": 89, "letter":"y", "list": ["ÿ"]},
+  {"keyCode": 89, "letter":"Y", "list": ["Ÿ"], "modifier": 16},
+  {"keyCode": 82, "letter":"r", "list": ["®"]},
+  {"keyCode": 80, "letter":"p", "list": ["¶"]},
+  {"keyCode": 84, "letter":"t", "list": ["™"]},
+  {"keyCode": 68, "letter":"D", "list": ["†","‡"], "modifier": 16},
+  {"keyCode": 96, "letter":"0", "list": ["°"]},
+  {"keyCode": 97, "letter":"1", "list": ["¹", "¼", "½"]},
+  {"keyCode": 98, "letter":"2", "list": ["²"]},
+  {"keyCode": 99, "letter":"3", "list": ["³", "¾"]},
+  {"keyCode": 67, "letter":"!", "list": ["¡"]},
+  {"keyCode": 67, "letter":"?", "list": ["¿"]},
+  {"keyCode": 67, "letter":":", "list": [":-)", ":-(", ";-)"]},
+  {"keyCode": 67, "letter":"$", "list": ["$","¢","€","£","¥"]}
+]
diff --git a/usr/palm/frameworks/mojo/submissions/191.15/resources/en_us/kbchars_table.json b/usr/palm/frameworks/mojo/submissions/191.15/resources/en_us/kbchars_table.json
new file mode 100644
index 0000000..8544247
--- /dev/null
+++ b/usr/palm/frameworks/mojo/submissions/191.15/resources/en_us/kbchars_table.json
@@ -0,0 +1,30 @@
+[
+  {"keyCode": 188, "letter":",", "list": ["~","\\","`","•","÷","^","[","]","{","}","<",">","«","»","Ø","µ","|"]},
+  {"keyCode": 66,  "letter":"b", "list": ["b"]},
+  {"keyCode": 190, "letter":".", "list": ["…","~","\\","`","•","÷","^","[","]","{","}","<",">","«","»","Ø","µ","|"]},
+  {"keyCode": 48,  "letter":"@", "list": ["°","•","Ø"]},
+  {"keyCode": 65,  "letter":"a", "list": ["á","à","ä","â","ã","å","æ","Á","À","Ä","Â","Ã","Å","Æ","ª","~","\\","`","•","÷","^","[","]","{","}","<",">","«","»","Ø","µ","|"]},
+  {"keyCode": 66,  "letter":"b", "list": ["~","\\","`","•","÷","^","[","]","{","}","<",">","«","»","Ø","µ","|"]},
+  {"keyCode": 67,  "letter":"c", "list": ["©","ç","Ç","¢"]},
+  {"keyCode": 68,  "letter":"d", "list": ["†","‡"]},
+  {"keyCode": 69,  "letter":"e", "list": ["é","è","ë","ê","ē","É","È","Ë","Ê","€","¹","¼","½"]},
+  {"keyCode": 72,  "letter":"h", "list": ["¢","€","£","¥","ƒ"]},
+  {"keyCode": 73,  "letter":"i", "list": ["í","ì","ï","î","Í","Ì","Ï","Î","÷","‰"]},
+  {"keyCode": 74,  "letter":"j", "list": ["¡"]},
+  {"keyCode": 75,  "letter":"k", "list": [":-)",":-(",";-)"]},
+  {"keyCode": 76,  "letter":"l", "list": ["`","‘","’","‚","‛","\""]},
+  {"keyCode": 77,  "letter":"m", "list": ["µ",":-)",":-(",";-)"]},
+  {"keyCode": 78,  "letter":"n", "list": ["ñ","Ñ","¿"]},
+  {"keyCode": 79,  "letter":"o", "list": ["ó","ò","ö","ô","œ","õ","ø","Ó","Ò","Ö","Ô","Œ","Õ","Ø","º","“","”","„","<",">","«","»"]},
+  {"keyCode": 80,  "letter":"p", "list": ["¶","~","\\","`","•","÷","^","[","]","{","}","<",">","«","»","Ø","µ","|"]},
+  {"keyCode": 81,  "letter":"q", "list": ["\\","~","|"]},
+  {"keyCode": 82,  "letter":"r", "list": ["®","²"]},
+  {"keyCode": 83,  "letter":"s", "list": ["ß","š","Š","~","\\","`","•","÷","^","[","]","{","}","<",">","«","»","Ø","|"]},
+  {"keyCode": 84,  "letter":"t", "list": ["™","³","¾"]},
+  {"keyCode": 85,  "letter":"u", "list": ["ú","ù","ü","û","Ú","Ù","Ü","Û","[","]","{","}","<",">","«","»"]},
+  {"keyCode": 86,  "letter":"v", "list": ["^"]},
+  {"keyCode": 87,  "letter":"w", "list": ["÷","^","±"]},
+  {"keyCode": 88,  "letter":"x", "list": ["×","¤"]},
+  {"keyCode": 89,  "letter":"y", "list": ["ÿ","ý","Ÿ","Ý","¥","[","]","{","}","<",">","«","»"]},
+  {"keyCode": 90,  "letter":"z", "list": ["ž","Ž","•"]}
+]
diff --git a/usr/palm/frameworks/mojo/submissions/191.15/resources/en_us/kbchars_table_azerty.json b/usr/palm/frameworks/mojo/submissions/191.15/resources/en_us/kbchars_table_azerty.json
new file mode 100644
index 0000000..8180a5a
--- /dev/null
+++ b/usr/palm/frameworks/mojo/submissions/191.15/resources/en_us/kbchars_table_azerty.json
@@ -0,0 +1,29 @@
+[
+  {"keyCode": 188, "letter":",", "list": ["~","\\","`","•","÷","^","[","]","{","}","<",">","«","»","Ø","µ","|"]},
+  {"keyCode": 190, "letter":".", "list": ["…","~","\\","`","•","÷","^","[","]","{","}","<",">","«","»","Ø","µ","|"]},
+  {"keyCode": 48,  "letter":"@", "list": ["°","•","Ø"]},
+  {"keyCode": 65,  "letter":"a", "list": ["á","à","ä","â","ã","å","æ","Á","À","Ä","Â","Ã","Å","Æ","ª","~","\\","`","•","÷","^","[","]","{","}","<",">","«","»","Ø","µ","|"]},
+  {"keyCode": 66,  "letter":"b", "list": ["~","\\","`","•","÷","^","[","]","{","}","<",">","«","»","Ø","µ","|"]},
+  {"keyCode": 67,  "letter":"c", "list": ["©","ç","Ç","¢"]},
+  {"keyCode": 68,  "letter":"d", "list": ["†","‡"]},
+  {"keyCode": 69,  "letter":"e", "list": ["é","è","ë","ê","ē","É","È","Ë","Ê","€","¹","¼","½"]},
+  {"keyCode": 72,  "letter":"h", "list": ["¢","€","£","¥","ƒ"]},
+  {"keyCode": 73,  "letter":"i", "list": ["í","ì","ï","î","Í","Ì","Ï","Î","÷","‰"]},
+  {"keyCode": 74,  "letter":"j", "list": ["¡"]},
+  {"keyCode": 75,  "letter":"k", "list": [":-)",":-(",";-)"]},
+  {"keyCode": 76,  "letter":"l", "list": ["`","‘","’","‚","‛","\""]},
+  {"keyCode": 77,  "letter":"m", "list": ["µ",":-)",":-(",";-)"]},
+  {"keyCode": 78,  "letter":"n", "list": ["ñ","Ñ","¿"]},
+  {"keyCode": 79,  "letter":"o", "list": ["ó","ò","ö","ô","œ","õ","ø","Ó","Ò","Ö","Ô","Œ","Õ","Ø","º","“","”","„","<",">","«","»"]},
+  {"keyCode": 80,  "letter":"p", "list": ["¶","~","\\","`","•","÷","^","[","]","{","}","<",">","«","»","Ø","µ","|"]},
+  {"keyCode": 81,  "letter":"q", "list": ["\\","~","|"]},
+  {"keyCode": 82,  "letter":"r", "list": ["®","²"]},
+  {"keyCode": 83,  "letter":"s", "list": ["ß","š","Š","~","\\","`","•","÷","^","[","]","{","}","<",">","«","»","Ø","|"]},
+  {"keyCode": 84,  "letter":"t", "list": ["™","³","¾"]},
+  {"keyCode": 85,  "letter":"u", "list": ["ú","ù","ü","û","Ú","Ù","Ü","Û","[","]","{","}","<",">","«","»"]},
+  {"keyCode": 86,  "letter":"v", "list": ["^"]},
+  {"keyCode": 87,  "letter":"w", "list": ["÷","^","±"]},
+  {"keyCode": 88,  "letter":"x", "list": ["×","¤"]},
+  {"keyCode": 89,  "letter":"y", "list": ["ÿ","ý","Ÿ","Ý","¥","[","]","{","}","<",">","«","»"]},
+  {"keyCode": 90,  "letter":"z", "list": ["ž","Ž","•"]}
+]
\ No newline at end of file
diff --git a/usr/palm/frameworks/mojo/submissions/191.15/resources/en_us/kbchars_table_qwertz.json b/usr/palm/frameworks/mojo/submissions/191.15/resources/en_us/kbchars_table_qwertz.json
new file mode 100644
index 0000000..8180a5a
--- /dev/null
+++ b/usr/palm/frameworks/mojo/submissions/191.15/resources/en_us/kbchars_table_qwertz.json
@@ -0,0 +1,29 @@
+[
+  {"keyCode": 188, "letter":",", "list": ["~","\\","`","•","÷","^","[","]","{","}","<",">","«","»","Ø","µ","|"]},
+  {"keyCode": 190, "letter":".", "list": ["…","~","\\","`","•","÷","^","[","]","{","}","<",">","«","»","Ø","µ","|"]},
+  {"keyCode": 48,  "letter":"@", "list": ["°","•","Ø"]},
+  {"keyCode": 65,  "letter":"a", "list": ["á","à","ä","â","ã","å","æ","Á","À","Ä","Â","Ã","Å","Æ","ª","~","\\","`","•","÷","^","[","]","{","}","<",">","«","»","Ø","µ","|"]},
+  {"keyCode": 66,  "letter":"b", "list": ["~","\\","`","•","÷","^","[","]","{","}","<",">","«","»","Ø","µ","|"]},
+  {"keyCode": 67,  "letter":"c", "list": ["©","ç","Ç","¢"]},
+  {"keyCode": 68,  "letter":"d", "list": ["†","‡"]},
+  {"keyCode": 69,  "letter":"e", "list": ["é","è","ë","ê","ē","É","È","Ë","Ê","€","¹","¼","½"]},
+  {"keyCode": 72,  "letter":"h", "list": ["¢","€","£","¥","ƒ"]},
+  {"keyCode": 73,  "letter":"i", "list": ["í","ì","ï","î","Í","Ì","Ï","Î","÷","‰"]},
+  {"keyCode": 74,  "letter":"j", "list": ["¡"]},
+  {"keyCode": 75,  "letter":"k", "list": [":-)",":-(",";-)"]},
+  {"keyCode": 76,  "letter":"l", "list": ["`","‘","’","‚","‛","\""]},
+  {"keyCode": 77,  "letter":"m", "list": ["µ",":-)",":-(",";-)"]},
+  {"keyCode": 78,  "letter":"n", "list": ["ñ","Ñ","¿"]},
+  {"keyCode": 79,  "letter":"o", "list": ["ó","ò","ö","ô","œ","õ","ø","Ó","Ò","Ö","Ô","Œ","Õ","Ø","º","“","”","„","<",">","«","»"]},
+  {"keyCode": 80,  "letter":"p", "list": ["¶","~","\\","`","•","÷","^","[","]","{","}","<",">","«","»","Ø","µ","|"]},
+  {"keyCode": 81,  "letter":"q", "list": ["\\","~","|"]},
+  {"keyCode": 82,  "letter":"r", "list": ["®","²"]},
+  {"keyCode": 83,  "letter":"s", "list": ["ß","š","Š","~","\\","`","•","÷","^","[","]","{","}","<",">","«","»","Ø","|"]},
+  {"keyCode": 84,  "letter":"t", "list": ["™","³","¾"]},
+  {"keyCode": 85,  "letter":"u", "list": ["ú","ù","ü","û","Ú","Ù","Ü","Û","[","]","{","}","<",">","«","»"]},
+  {"keyCode": 86,  "letter":"v", "list": ["^"]},
+  {"keyCode": 87,  "letter":"w", "list": ["÷","^","±"]},
+  {"keyCode": 88,  "letter":"x", "list": ["×","¤"]},
+  {"keyCode": 89,  "letter":"y", "list": ["ÿ","ý","Ÿ","Ý","¥","[","]","{","}","<",">","«","»"]},
+  {"keyCode": 90,  "letter":"z", "list": ["ž","Ž","•"]}
+]
\ No newline at end of file
diff --git a/usr/palm/frameworks/mojo/submissions/191.15/resources/kbchars_fulltable.json b/usr/palm/frameworks/mojo/submissions/191.15/resources/kbchars_fulltable.json
new file mode 100644
index 0000000..6774f8f
--- /dev/null
+++ b/usr/palm/frameworks/mojo/submissions/191.15/resources/kbchars_fulltable.json
@@ -0,0 +1,87 @@
+[
+  {"keyCode": 80, "letter":"q", "list": ["q"]},
+  {"keyCode": 80, "letter":"w", "list": ["w"]},
+  {"keyCode": 69, "letter":"e", "list": ["e"]},
+  {"keyCode": 80, "letter":"r", "list": ["r"]},
+  {"keyCode": 80, "letter":"t", "list": ["t"]},
+  {"keyCode": 80, "letter":"y", "list": ["y"]},
+  {"keyCode": 80, "letter":"u", "list": ["u"]},
+  {"keyCode": 70, "letter":"i", "list": ["i"]},
+  {"keyCode": 70, "letter":"o", "list": ["o"]},
+  {"keyCode": 80, "letter":"p", "list": ["p"]},
+
+  {"keyCode": 65, "letter":"a", "list": ["a"]},
+  {"keyCode": 80, "letter":"s", "list": ["s"]},
+  {"keyCode": 68, "letter":"d", "list": ["d"]},
+  {"keyCode": 70, "letter":"f", "list": ["f"]},
+  {"keyCode": 70, "letter":"g", "list": ["g"]},
+  {"keyCode": 70, "letter":"h", "list": ["h"]},
+  {"keyCode": 70, "letter":"j", "list": ["j"]},
+  {"keyCode": 70, "letter":"k", "list": ["k"]},
+  {"keyCode": 70, "letter":"l", "list": ["l"]},
+  {"keyCode": 8, "letter":"\b", "list": ["&laquo"]},
+
+  {"keyCode": 90, "letter":"O", "list": ["O"]},
+  {"keyCode": 90, "letter":"z", "list": ["z"]},
+  {"keyCode": 80, "letter":"x", "list": ["x"]},
+  {"keyCode": 67, "letter":"c", "list": ["c"]},
+  {"keyCode": 80, "letter":"v", "list": ["v"]},
+  {"keyCode": 66, "letter":"b", "list": ["b"]},
+  {"keyCode": 70, "letter":"n", "list": ["n"]},
+  {"keyCode": 70, "letter":"m", "list": ["m"]},
+  {"keyCode": 90, "letter":",", "list": [","]},
+  {"keyCode": 90, "letter":"\n", "list": ["&raquo"]},
+
+  {"keyCode": 90, "letter":"blank", "list": ["&deg"]},
+  {"keyCode": 90, "letter":"shift", "list": ["&yen"]},
+  {"keyCode": 90, "letter":"@", "list": ["@"]},
+  {"keyCode": 90, "letter":" ", "list": ["&nbsp"]},
+  {"keyCode": 90, "letter":" ", "list": ["&nbsp"]},
+  {"keyCode": 90, "letter":" ", "list": ["&nbsp"]},
+  {"keyCode": 90, "letter":" ", "list": ["&nbsp"]},
+  {"keyCode": 90, "letter":".", "list": ["."]},
+  {"keyCode": 90, "letter":"sym", "list": ["&iquest"]},
+  {"keyCode": 90, "letter":"blank", "list": ["&deg"]},
+
+  {"keyCode": 90, "letter":"0", "list": ["0"]},
+  {"keyCode": 90, "letter":"1", "list": ["1"]},
+  {"keyCode": 90, "letter":"2", "list": ["2"]},
+  {"keyCode": 90, "letter":"3", "list": ["3"]},
+  {"keyCode": 90, "letter":"4", "list": ["4"]},
+  {"keyCode": 90, "letter":"5", "list": ["5"]},
+  {"keyCode": 90, "letter":"6", "list": ["6"]},
+  {"keyCode": 90, "letter":"7", "list": ["7"]},
+  {"keyCode": 90, "letter":"8", "list": ["8"]},
+  {"keyCode": 90, "letter":"9", "list": ["9"]},
+  {"keyCode": 90, "letter":".", "list": ["."]},
+
+  {"keyCode": 65, "letter":"A", "list": ["A"], "modifier": 16},
+  {"keyCode": 67, "letter":"c", "list": ["©","ç"]},
+  {"keyCode": 67, "letter":"C", "list": ["Ç"], "modifier": 16},
+  {"keyCode": 69, "letter":"e", "list": ["é","è","ë","ê","ē"]},
+  {"keyCode": 69, "letter":"E", "list": ["É","È","Ë","Ê"], "modifier": 16},
+  {"keyCode": 73, "letter":"i", "list": ["í","ì","ï","î"]},
+  {"keyCode": 73, "letter":"I", "list": ["Í","Ì","Ï","Î"], "modifier": 16},
+  {"keyCode": 78, "letter":"n", "list": ["ñ"]},
+  {"keyCode": 78, "letter":"N", "list": ["Ñ"], "modifier": 16},
+  {"keyCode": 79, "letter":"o", "list": ["ó","ò","ö","ô","œ","õ","ø"]},
+  {"keyCode": 79, "letter":"O", "list": ["Ó","Ò","Ö","Ô","Œ","Õ","Ø"], "modifier": 16},
+  {"keyCode": 83, "letter":"s", "list": ["ß"]},
+  {"keyCode": 85, "letter":"u", "list": ["ú","ù","ü","û"]},
+  {"keyCode": 85, "letter":"U", "list": ["Ú","Ù","ÜÛ"], "modifier": 16},
+  {"keyCode": 88, "letter":"x", "list": ["×","¤"]},
+  {"keyCode": 89, "letter":"y", "list": ["ÿ"]},
+  {"keyCode": 89, "letter":"Y", "list": ["Ÿ"], "modifier": 16},
+  {"keyCode": 82, "letter":"r", "list": ["®"]},
+  {"keyCode": 80, "letter":"p", "list": ["¶"]},
+  {"keyCode": 84, "letter":"t", "list": ["™"]},
+  {"keyCode": 68, "letter":"D", "list": ["†","‡"], "modifier": 16},
+  {"keyCode": 96, "letter":"0", "list": ["°"]},
+  {"keyCode": 97, "letter":"1", "list": ["¹", "¼", "½"]},
+  {"keyCode": 98, "letter":"2", "list": ["²"]},
+  {"keyCode": 99, "letter":"3", "list": ["³", "¾"]},
+  {"keyCode": 67, "letter":"!", "list": ["¡"]},
+  {"keyCode": 67, "letter":"?", "list": ["¿"]},
+  {"keyCode": 67, "letter":":", "list": [":-)", ":-(", ";-)"]},
+  {"keyCode": 67, "letter":"$", "list": ["$","¢","€","£","¥"]}
+]
diff --git a/usr/palm/frameworks/mojo/submissions/191.15/stylesheets/global.css b/usr/palm/frameworks/mojo/submissions/191.15/stylesheets/global.css
index 39fa4ec..8cb88a9 100644
--- a/usr/palm/frameworks/mojo/submissions/191.15/stylesheets/global.css
+++ b/usr/palm/frameworks/mojo/submissions/191.15/stylesheets/global.css
@@ -483,7 +483,7 @@ a:visited {
 }
 
 .char-selector-element {
-  color: #666;
+  color: red;
   width: 48px;
   height: 48px;
   line-height: 48px;
@@ -511,6 +511,73 @@ span.selected-char {
 }
 
   
+/* Styles for On Screen Keyboard */
+
+.kb-char-selector-container {
+  z-index: 199500; /* same as popup */
+  position:fixed;
+  width: 315px;
+  border-width: 24px;
+  -webkit-border-image: url(../images/palm-popup-background.png) 24 24 24 24 repeat repeat;
+  -webkit-box-sizing: border-box;
+}
+
+.kb-char-selector-container .kb-char-selector-wrapper {
+  margin: -15px -16px -15px -16px;
+}
+
+.kb-char-selector-content {
+  padding: 0px;
+  max-height: 192px; /* rows are 48px tall */
+  overflow: hidden;   
+  margin-left: -1px;
+}
+
+.kb-char-selector-column {
+  width: 304px;
+  height: 48px;
+}
+
+.kb-char-selector-element{
+  color: red;
+  width:10%;
+  height:48px;
+  line-height: 48px
+  display:inline-block;
+  float:left;
+  margin-right: -1px;
+  font-weight: normal;
+  text-align: center;
+  border-right: 1px solid #999;
+  border-bottom: 1px solid #999;
+}
+
+.kb-char-selector-element .kb-img{
+  width:100%;
+  height:48px;
+}
+
+.kb-char-selector-element.selected {
+  background: #444;
+  color: #fff !important;
+}
+
+span.selected-char {
+  font-weight: bold;
+  color: #1f75bf;
+  height: 48px;
+  background: url(../images/filter-search-highlight.png) bottom repeat-x;
+}
+
+/* Landscape rows */
+.kb-char-selector-container.landscape {
+  width: 480px;
+}
+
+.kb-char-selector-column.landscape {
+  width: 470px;
+}
+
 /** Slider and Progress Slider **/
 
 .palm-row-wrapper > .palm-slider {
@@ -665,4 +732,4 @@ span.selected-char {
   height: 111px;
   background: url(../images/disclaimer-beta.png) no-repeat;     
   -webkit-palm-mouse-target: ignore;
-}
\ No newline at end of file
+}
diff --git a/usr/palm/frameworks/mojo/submissions/191.15/templates/kbselector/char-ls.html b/usr/palm/frameworks/mojo/submissions/191.15/templates/kbselector/char-ls.html
new file mode 100644
index 0000000..f036112
--- /dev/null
+++ b/usr/palm/frameworks/mojo/submissions/191.15/templates/kbselector/char-ls.html
@@ -0,0 +1,4 @@
+<div id="#{divPrefix}-#{index}" name="#{index}" class="kb-char-selector-element landscape" x-mojo-tap-highlight="momentary">
+  <span name="#{index}">#{-character}</span>
+  <!-- <img class="kb-img" name="#{index}" src="/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/#{-character}.png" alt="#{-character}"> -->
+</div>
diff --git a/usr/palm/frameworks/mojo/submissions/191.15/templates/kbselector/char-selector-row-ls.html b/usr/palm/frameworks/mojo/submissions/191.15/templates/kbselector/char-selector-row-ls.html
new file mode 100644
index 0000000..a5a8217
--- /dev/null
+++ b/usr/palm/frameworks/mojo/submissions/191.15/templates/kbselector/char-selector-row-ls.html
@@ -0,0 +1 @@
+<div class="kb-char-selector-column landscape">#{-characters}</div>
diff --git a/usr/palm/frameworks/mojo/submissions/191.15/templates/kbselector/char-selector-row.html b/usr/palm/frameworks/mojo/submissions/191.15/templates/kbselector/char-selector-row.html
new file mode 100644
index 0000000..847f837
--- /dev/null
+++ b/usr/palm/frameworks/mojo/submissions/191.15/templates/kbselector/char-selector-row.html
@@ -0,0 +1 @@
+<div class="kb-char-selector-column">#{-characters}</div>
\ No newline at end of file
diff --git a/usr/palm/frameworks/mojo/submissions/191.15/templates/kbselector/char.html b/usr/palm/frameworks/mojo/submissions/191.15/templates/kbselector/char.html
new file mode 100644
index 0000000..752ec0c
--- /dev/null
+++ b/usr/palm/frameworks/mojo/submissions/191.15/templates/kbselector/char.html
@@ -0,0 +1,4 @@
+<div id="#{divPrefix}-#{index}" name="#{index}" class="kb-char-selector-element" x-mojo-tap-highlight="momentary">
+  <span name="#{index}">#{-character}</span>
+  <!--<img class="kb-img" name="#{index}" src="/usr/palm/frameworks/mojo/submissions/191.15/images/virt_kb/#{-character}.png" alt="#{-character}">-->
+</div>
diff --git a/usr/palm/frameworks/mojo/submissions/191.15/templates/kbselector/charselector-ls.html b/usr/palm/frameworks/mojo/submissions/191.15/templates/kbselector/charselector-ls.html
new file mode 100644
index 0000000..28aa368
--- /dev/null
+++ b/usr/palm/frameworks/mojo/submissions/191.15/templates/kbselector/charselector-ls.html
@@ -0,0 +1,7 @@
+<div  class="kb-char-selector-container landscape" id="#{divPrefix}-kb-char-selector-div">
+  <div class="kb-char-selector-wrapper">
+  	<div class="kb-char-selector-content landscape" x-mojo-element="Scroller" id="#{divPrefix}-char-selector">
+		<div class="kb-char-selector-content landscape" id="char-list" x-mojo-element="List">
+		</div>  
+  </div>
+</div> 
diff --git a/usr/palm/frameworks/mojo/submissions/191.15/templates/kbselector/charselector.html b/usr/palm/frameworks/mojo/submissions/191.15/templates/kbselector/charselector.html
new file mode 100644
index 0000000..b2b7ea5
--- /dev/null
+++ b/usr/palm/frameworks/mojo/submissions/191.15/templates/kbselector/charselector.html
@@ -0,0 +1,7 @@
+<div  class="kb-char-selector-container" id="#{divPrefix}-kb-char-selector-div">
+  <div class="kb-char-selector-wrapper">
+  	<div class="char-selector-content" x-mojo-element="Scroller" id="#{divPrefix}-char-selector">
+		<div class="kb-char-selector-content" id="char-list" x-mojo-element="List">
+		</div>  
+  </div>
+</div> 
