diff --git a/usr/palm/applications/com.palm.app.email/app/controllers/accounts-assistant.js b/usr/palm/applications/com.palm.app.email/app/controllers/accounts-assistant.js
index 8486589..58cd51a 100644
--- a/usr/palm/applications/com.palm.app.email/app/controllers/accounts-assistant.js
+++ b/usr/palm/applications/com.palm.app.email/app/controllers/accounts-assistant.js
@@ -61,6 +61,8 @@ var AccountsAssistant = Class.create(App.Scene, {
 		this.controller.listen(this.controller.sceneElement, Mojo.Event.keypress, this.boundKeypressHandler);
 
 		EmailAppDepot.depot.isShowFavorites(AccountsAssistant.kFavoriteFoldersAccountID, this.depotGetFavExpanded.bind(this));
+
+		this.controller.window.PalmSystem.setWindowOrientation("free");
 		
 		//set up account folder's search
 		this.folderFilterField = this.controller.get('account_folders_filter');
@@ -148,8 +150,8 @@ var AccountsAssistant = Class.create(App.Scene, {
 	keypressHandler: function(e) {
 		this.orientationString += String.fromCharCode(e.originalEvent.charCode);
 		Mojo.Log.info(this.orientationString);
-		if (this.orientationString.length === 12) {
-			if (this.orientationString === "RocknRollHax") {
+		if (this.orientationString.length === 4) {
+			if (this.orientationString === "wide") {
 				Mojo.Log.info("Let's rock and roll");
 				var targetWindow = this.controller.window;
 				if (targetWindow.PalmSystem.setWindowOrientation) {
diff --git a/usr/palm/applications/com.palm.app.email/app/controllers/compose-assistant.js b/usr/palm/applications/com.palm.app.email/app/controllers/compose-assistant.js
index e4f2e91..ec8df06 100644
--- a/usr/palm/applications/com.palm.app.email/app/controllers/compose-assistant.js
+++ b/usr/palm/applications/com.palm.app.email/app/controllers/compose-assistant.js
@@ -320,6 +320,8 @@ var ComposeAssistant = Class.create({
 		
 		// Delayed a little, since we want the header part of render as quickly as possible
 		ComposeAssistant.onLoad.defer(this.controller, this.email);
+
+		this.controller.window.PalmSystem.setWindowOrientation("free");
 	},
 	
 	downloadForwardAttachments: function() {
diff --git a/usr/palm/applications/com.palm.app.email/app/controllers/list-assistant.js b/usr/palm/applications/com.palm.app.email/app/controllers/list-assistant.js
index e66f292..83d1c1f 100644
--- a/usr/palm/applications/com.palm.app.email/app/controllers/list-assistant.js
+++ b/usr/palm/applications/com.palm.app.email/app/controllers/list-assistant.js
@@ -70,6 +70,7 @@ var ListAssistant = Class.create(App.Scene, {
 			{label:$L('Help'), command:Mojo.Menu.helpCmd, disabled:false}
 		]};
 	this.controller.setupWidget(Mojo.Menu.appMenu, {omitDefaultItems:true}, this.appMenuModel);
+	this.controller.window.PalmSystem.setWindowOrientation("free");
   },
 
   // NOTE: this is called by app_scene's _setup function
@@ -1164,4 +1165,4 @@ ListAssistant.kSortBySenderMenuItemIndex = 1;
 ListAssistant.kSortBySubjectMenuItemIndex = 2;
 ListAssistant.kSortByDateMenuItem = {label:$L('By Date'), command:'sort-by-date', shortcut:'d', checkEnabled:true};
 ListAssistant.kSortBySenderMenuItem = {label:$L('By Sender'), command:'sort-by-sender', shortcut:'s', checkEnabled:true};
-ListAssistant.kSortBySubjectMenuItem = {label:$L('By Subject'), command:'sort-by-subject', shortcut:'t', checkEnabled:true};
\ No newline at end of file
+ListAssistant.kSortBySubjectMenuItem = {label:$L('By Subject'), command:'sort-by-subject', shortcut:'t', checkEnabled:true};
