diff --git a/usr/palm/applications/com.palm.app.email/app/controllers/list-assistant.js b/usr/palm/applications/com.palm.app.email/app/controllers/list-assistant.js
index e66f292..e0101b4 100644
--- a/usr/palm/applications/com.palm.app.email/app/controllers/list-assistant.js
+++ b/usr/palm/applications/com.palm.app.email/app/controllers/list-assistant.js
@@ -89,7 +89,7 @@ var ListAssistant = Class.create(App.Scene, {
 					fixedHeightItems:true,
 					itemTemplate: 'list/email_entry',
 					swipeToDelete: true,
-					autoconfirmDelete: true,
+					autoconfirmDelete: false,
 					uniquenessProperty: 'id',
 					dividerTemplate: 'list/date_separator',
 					dividerFunction: function(email) { return email.category; },
