diff --git a/usr/palm/applications/com.palm.app.email/app/controllers/message-assistant.js b/usr/palm/applications/com.palm.app.email/app/controllers/message-assistant.js
index 761069e..1740474 100644
--- a/usr/palm/applications/com.palm.app.email/app/controllers/message-assistant.js
+++ b/usr/palm/applications/com.palm.app.email/app/controllers/message-assistant.js
@@ -2142,10 +2142,18 @@ var MessageAssistant = Class.create({
 		});
 	},
 
-	deleteEmail: function() {
-		Message.setDeleted(this.data.id, true);
-		this.controller.stageController.popScene();
-	}
+	
+       deleteEmail: function() {                                  
+                Message.setDeleted(this.data.id, true);       
+                if(this.nextMessages["newer"] !== undefined){       
+                        this.gotoNextEmail("newer");       
+                }else if(this.nextMessages["older"] !== undefined){
+                	this.gotoNextEmail("older");
+                }else{
+                        this.controller.stageController.popScene();
+                }                 
+                                              
+        }    
 });
 
 MessageAssistant.kAppMenuMarkRead = $L('Mark as Read');
