diff --git a/usr/palm/applications/com.palm.app.email/app/controllers/message-assistant.js b/usr/palm/applications/com.palm.app.email/app/controllers/message-assistant.js
index 878d9fb..2b4cf38 100644
--- a/usr/palm/applications/com.palm.app.email/app/controllers/message-assistant.js
+++ b/usr/palm/applications/com.palm.app.email/app/controllers/message-assistant.js
@@ -70,10 +70,10 @@ var MessageAssistant = Class.create({
 		this.cmdMenuModel = {
 				visible:true,
 				items: [
-					{label:$L('Reply'),     icon:'reply',     command:'reply'},
-					{label:$L('Reply all'), icon:'reply-all', command:'replyAll'},
-					{label:$L('Forward'),   icon:'forward-email', command:'forward'},
-					{label:$L('Delete'),    icon:'delete',    command:'delete'}
+					{label:$L('Reply'),     shortcut:'r',     command:'reply'},
+					{label:$L('Reply all'), shortcut:'e', command:'replyAll'},
+					{label:$L('Forward'),   shortcut:'f', command:'forward'},
+					{label:$L('Delete'),    shortcut:'d',    command:'delete'}
 				]};
 		this.controller.setupWidget(Mojo.Menu.commandMenu, undefined, this.cmdMenuModel);
 		
