diff --git a/usr/palm/applications/com.palm.app.email/data/AccountPreferences.js b/usr/palm/applications/com.palm.app.email/data/AccountPreferences.js
index 8827850..072ba39 100644
--- a/usr/palm/applications/com.palm.app.email/data/AccountPreferences.js
+++ b/usr/palm/applications/com.palm.app.email/data/AccountPreferences.js
@@ -136,8 +136,13 @@ AccountPreferences.SYNC_5_MINS	= 5;
 AccountPreferences.SYNC_10_MINS	= 10;
 AccountPreferences.SYNC_15_MINS	= 15;
 AccountPreferences.SYNC_30_MINS	= 30;
+AccountPreferences.SYNC_45_MINS	= 45;
 AccountPreferences.SYNC_1_HR	= 60;
+AccountPreferences.SYNC_2_HRS	= 120;
+AccountPreferences.SYNC_4_HRS	= 240;
 AccountPreferences.SYNC_6_HRS	= 360;
+AccountPreferences.SYNC_8_HRS	= 480;
+AccountPreferences.SYNC_10_HRS	= 600;
 AccountPreferences.SYNC_12_HRS	= 720;
 AccountPreferences.SYNC_24_HRS	= 1440;
 AccountPreferences.SYNC_PUSH	= -1;
@@ -148,6 +153,11 @@ AccountPreferences.LOOKBACK_3_DAYS 	= 3;
 AccountPreferences.LOOKBACK_7_DAYS 	= 7;
 AccountPreferences.LOOKBACK_2_WEEKS = 14;
 AccountPreferences.LOOKBACK_1_MONTH = 30;
+AccountPreferences.LOOKBACK_2_MONTHS = 60;
+AccountPreferences.LOOKBACK_3_MONTHS = 90;
+AccountPreferences.LOOKBACK_6_MONTHS = 180;
+AccountPreferences.LOOKBACK_9_MONTHS = 270;
+AccountPreferences.LOOKBACK_12_MONTHS = 365;
 AccountPreferences.LOOKBACK_FOREVER = 0;
 
 AccountPreferences.NO_SIGNATURE_BY_USER = '<div id="no_signature" style="overflow:hidden;"></div>';
diff --git a/usr/palm/applications/com.palm.app.email/mail/source/settings/AccountSettings.js b/usr/palm/applications/com.palm.app.email/mail/source/settings/AccountSettings.js
index df977dd..f73a040 100644
--- a/usr/palm/applications/com.palm.app.email/mail/source/settings/AccountSettings.js
+++ b/usr/palm/applications/com.palm.app.email/mail/source/settings/AccountSettings.js
@@ -126,6 +126,11 @@ enyo.kind({
 									 {caption: $L("7 days"), value: AccountPreferences.LOOKBACK_7_DAYS},
 									 {caption: $L("2 weeks"), value: AccountPreferences.LOOKBACK_2_WEEKS},
 									 {caption: $L("1 month"), value: AccountPreferences.LOOKBACK_1_MONTH},
+									 {caption: $L("2 months"), value: AccountPreferences.LOOKBACK_2_MONTHS},
+									 {caption: $L("3 months"), value: AccountPreferences.LOOKBACK_3_MONTHS},	
+ 									 {caption: $L("6 months"), value: AccountPreferences.LOOKBACK_6_MONTHS}, 
+									 {caption: $L("9 months"), value: AccountPreferences.LOOKBACK_9_MONTHS},		
+				 					 {caption: $L("12 months"), value: AccountPreferences.LOOKBACK_12_MONTHS},
 									 {caption: $L("All"), value: AccountPreferences.LOOKBACK_FOREVER}
 							    ],
 							onChange: "valueChanged"
@@ -142,7 +147,11 @@ enyo.kind({
 									 {caption: $L("15 minutes"), value: AccountPreferences.SYNC_15_MINS},
 									 {caption: $L("30 minutes"), value: AccountPreferences.SYNC_30_MINS},
 									 {caption: $L("1 Hour"), value: AccountPreferences.SYNC_1_HR},
+									 {caption: $L("2 Hours"), value: AccountPreferences.SYNC_2_HRS},
+									 {caption: $L("4 Hours"), value: AccountPreferences.SYNC_4_HRS},
 									 {caption: $L("6 Hours"), value: AccountPreferences.SYNC_6_HRS},
+									 {caption: $L("8 Hours"), value: AccountPreferences.SYNC_8_HRS},
+ 									 {caption: $L("10 Hours"), value: AccountPreferences.SYNC_10_HRS},
 									 {caption: $L("12 Hours"), value: AccountPreferences.SYNC_12_HRS},
 									 {caption: $L("24 Hours"), value: AccountPreferences.SYNC_24_HRS},
 									 {caption: $L("Manual"), value: AccountPreferences.SYNC_MANUAL}
