--- .orig/usr/palm/applications/com.palm.app.email/app/controllers/compose-assistant.js
+++ /usr/palm/applications/com.palm.app.email/app/controllers/compose-assistant.js
@@ -79,6 +79,7 @@
 				visible: true,
 				menuClass: 'palm-white',
 				items: [
+					{label:$L('Priority'), icon:'priority', command:'priority'},
 					{label:$L('Attach'), icon:'attach', command:'attach'},
 					{label:$L('Send'), icon:'send', command:'send'}
 				]};
@@ -80,7 +81,8 @@
 				menuClass: 'palm-white',
 				items: [
 					{label:$L('Attach'), icon:'attach', command:'attach'},
-					{label:$L('Send'), icon:'send', command:'send'}
+					{label:('Send'), icon:'send', command:'send'},
+					{label:$L('Delete'),icon:'delete', command:'cancel'}
 				]};
 
 		this.draftIsDirty = false;
@@ -236,7 +238,15 @@
 					break;
 
 				case 'cancel':
-					this.cancelCompose();
+					this.controller.showAlertDialog({
+					onChoose: function(value) {if (value == 'yes') {this.cancelCompose();}},
+					title: $L("Discard this Draft?"),
+					message: $L("This cannot be undone"),
+					choices:[
+						{label:$L('Yes'), value:"yes", type:'negative'},  
+						{label:$L("No"), value:"no", type: 'dismiss'}
+							]
+					});
 					break;
 						
 				case 'priority':
