diff --git a/usr/palm/applications/com.palm.app.email/mail/source/MailApp.js b/usr/palm/applications/com.palm.app.email/mail/source/MailApp.js
index afa3ae3..067b5ef 100644
--- a/usr/palm/applications/com.palm.app.email/mail/source/MailApp.js
+++ b/usr/palm/applications/com.palm.app.email/mail/source/MailApp.js
@@ -33,7 +33,7 @@ enyo.kind({
                     onHeaderTap: "headerTap"
 				}
 			]},
-			{flex: 1, name: "bodySliding", dragAnywhere: false, onResize: "resizeBody", components: [
+			{flex: 1, name: "bodySliding", dragAnywhere: false, onResize: "resizeBody", dismissible: true, components: [
 				{
 					name: "body",
 					kind: "Body",
@@ -468,6 +468,7 @@ enyo.kind({
 
 	/** note that this selectMessage takes a full message. mail.selectMessage takes a messageId */
 	selectMessage: function(inSender, message, userActivated) {
+		this.$.bodySliding.setShowing(true)
 		if (userActivated && (window.innerWidth < 900) && (this.$.slidingPane.view === this.$.folderSliding)) {
 			this.$.slidingPane.selectView(this.$.mailSliding);
 		}
