diff --git a/usr/palm/applications/com.palm.app.email/app/controllers/compose-assistant.js b/usr/palm/applications/com.palm.app.email/app/controllers/compose-assistant.js
index e4f2e91..c84057c 100644
--- a/usr/palm/applications/com.palm.app.email/app/controllers/compose-assistant.js
+++ b/usr/palm/applications/com.palm.app.email/app/controllers/compose-assistant.js
@@ -172,8 +172,11 @@ var ComposeAssistant = Class.create({
 				visible: true,
 				menuClass: 'palm-white',
 				items: [
+					{label:$L('Priority'), icon:'priority', command:'priority'},
 					{label:$L('Attach'), icon:'attach', command:'attach'},
-					{label:$L('Send'), icon:'send', command:'send'}
+					{label:$L('Save'), icon:'save', command:'save'},
+					{label:$L('Send'), icon:'send', command:'send'},
+					{label:$L('Delete'),icon:'delete', command:'cancel'}
 				]};
 
 		this.appMenuSaveIndex = 2;
@@ -383,6 +386,7 @@ var ComposeAssistant = Class.create({
 	},
 
 	orientationChanged: function(orientation) {
+		this.controller.stageController.setWindowOrientation(orientation);
 		if (orientation === "left" || orientation === "right") {
 			this.controller.sceneElement.addClassName('landscape');
 		} else {
@@ -609,16 +613,37 @@ var ComposeAssistant = Class.create({
 					break;
 
 				case 'save':
-					this.draftIsDirty = true; // set to true to make sure the save occurs
-					this.saveDraft();
+					this.controller.showAlertDialog({
+					onChoose: function(value) {if (value == 'yes') {this.save();}},
+					title: $L("Save This Message?"),
+					choices:[
+						{label:$L('Yes'), value:"yes", type:'negative'},  
+						{label:$L("No"), value:"no", type: 'dismiss'}
+							]
+					});
 					break;
 
 				case 'send':
-					this.send();
+					this.controller.showAlertDialog({
+					onChoose: function(value) {if (value == 'yes') {this.send();}},
+					title: $L("Send This Message?"),
+					choices:[
+						{label:$L('Yes'), value:"yes", type:'negative'},  
+						{label:$L("No"), value:"no", type: 'dismiss'}
+							]
+					});
 					break;
 
 				case 'cancel':
-					this.cancelCompose();
+					this.controller.showAlertDialog({
+					onChoose: function(value) {if (value == 'yes') {this.cancelCompose();}},
+					title: $L("Discard this Draft?"),
+					message: $L("This cannot be undone"),
+					choices:[
+						{label:$L('Yes'), value:"yes", type:'negative'},  
+						{label:$L("No"), value:"no", type: 'dismiss'}
+							]
+					});
 					break;
 						
 				case 'priority':
