diff --git a/usr/palm/applications/com.palm.app.email/app/controllers/message-assistant.js b/usr/palm/applications/com.palm.app.email/app/controllers/message-assistant.js
index 761069e..b961bfb 100644
--- a/usr/palm/applications/com.palm.app.email/app/controllers/message-assistant.js
+++ b/usr/palm/applications/com.palm.app.email/app/controllers/message-assistant.js
@@ -70,14 +70,14 @@ var MessageAssistant = Class.create({
 		this.cmdMenuModel = {
 				visible:true,
 				items: [
+					{label:$L('Delete'),    icon:'delete',    command:'delete'},
 					{label:$L('Reply'),     icon:'reply',     command:'reply'},
-					{label:$L('Reply all'), icon:'reply-all', command:'replyAll'},
-					{label:$L('Forward'),   icon:'forward-email', command:'forward'},
-					{label:$L('Delete'),    icon:'delete',    command:'delete'}
+					{label:$L('Move'),      icon:'file', command:'move'},
+					{label:$L('MarkUnread'),   icon:'back', command:'mark-unread'}
 				]};
 		this.controller.setupWidget(Mojo.Menu.commandMenu, undefined, this.cmdMenuModel);
 		
-		this.markUnreadMenuItem =     {label:MessageAssistant.kAppMenuMarkUnread, shortcut:'k', command:'mark-unread'};
+		this.markUnreadMenuItem =     {label:MessageAssistant.kAppMenuMarkUnread, shortcut:'u', command:'mark-unread'};
 		this.markSetFlagMenuItem =    {label:MessageAssistant.kAppMenuSetFlag, shortcut:'g', command:'flag'};
 		this.showRecipientsMenuItem = {label:MessageAssistant.kAppMenuShowRecipients, shortcut:'s', command:'show-recipients'};
 		
@@ -104,7 +104,8 @@ var MessageAssistant = Class.create({
 				emailEditItems,
 				this.markUnreadMenuItem,
 				this.markSetFlagMenuItem,
-				{label:$L('Move to Folder...'), shortcut:'m', command:'move'},
+				{label:$L('Forward...'), shortcut:'f', command:'forward'},
+				{label:$L('Reply to All...'), shortcut:'r', command:'replyAll'},
 				this.showRecipientsMenuItem,
 				Mojo.Menu.prefsItem,
 				Mojo.Menu.helpItem
@@ -1473,14 +1474,9 @@ var MessageAssistant = Class.create({
 						var currentLabel = this.markUnreadMenuItem.label;
 						var markRead = (currentLabel == MessageAssistant.kAppMenuMarkRead);
 						Message.setRead(this.data.id, markRead);
-						if (markRead) {
-							this.markUnreadMenuItem.label = MessageAssistant.kAppMenuMarkUnread;
-						} else {
-							this.markUnreadMenuItem.label = MessageAssistant.kAppMenuMarkRead;
-						}
-						
+						this.markUnreadMenuItem.label = MessageAssistant.kAppMenuMarkUnread;
 						this.updateMessageInParentScene();
-						
+						this.controller.stageController.popScene();
 						break;
 
 					case 'flag':
