diff --git a/usr/palm/applications/com.palm.app.email/app/controllers/message-assistant.js b/usr/palm/applications/com.palm.app.email/app/controllers/message-assistant.js
index 761069e..9c85c81 100644
--- a/usr/palm/applications/com.palm.app.email/app/controllers/message-assistant.js
+++ b/usr/palm/applications/com.palm.app.email/app/controllers/message-assistant.js
@@ -143,6 +143,7 @@ var MessageAssistant = Class.create({
 		Mojo.Event.listen(this.controller.stageController.document, Mojo.Event.activate, this.boundUpdateRecipientStatus);
 
 		Mojo.Event.listen(this.controller.getSceneScroller(), Mojo.Event.scrollStarting, this.addAsScrollListener.bind(this));
+    Mojo.Event.listen(this.controller.document, Mojo.Event.flick, this.flickHandler.bind(this));
 		
 		this.subjectElement = this.controller.get('email_subject');
 		this.readViewContainer = this.controller.get('email-readview-content-container');
@@ -1855,6 +1856,15 @@ var MessageAssistant = Class.create({
 		}
 	},
 
+  flickHandler: function(event) {
+                  if (event.velocity.x > 0) {
+                    this.boundGotoNextEmailNewer();
+                  }
+                  else {
+                    this.boundGotoNextEmailOlder();
+                  }
+  },
+
 	gotoNextEmail: function(direction) {
 		if (this.nextMessages !== undefined && this.nextMessages[direction] !== undefined) {
 			var details = this.nextMessages[direction];
