diff --git a/usr/palm/applications/com.palm.app.email/mail/source/Body.js b/usr/palm/applications/com.palm.app.email/mail/source/Body.js
index 8b6ed34..614c7ad 100644
--- a/usr/palm/applications/com.palm.app.email/mail/source/Body.js
+++ b/usr/palm/applications/com.palm.app.email/mail/source/Body.js
@@ -191,8 +191,8 @@ enyo.kind({
 			// {name: "moveToFolderCommandItem", icon: "../images/icons/toolbar-icon-moveto.png", onclick: "moveToFolderClick"},
 			// {kind: "Control", className:"toolbar-spacer"},
 			{name: "deleteCommandItem", icon: "../images/icons/toolbar-icon-delete.png",  onclick: "deleteClick"},
-			{kind: "Control", name: "navgroupSpacer", className:"toolbar-spacer", showing:false},
-			{kind:enyo.VFlexBox, name:"navgroup", style:"position:absolute;right:14px;margin:6px 0;", showing:false, width:"118px", components: [ 
+			{kind: "Control", name: "navgroupSpacer", className:"toolbar-spacer", showing:true},
+			{kind: "Control", className:"toolbar-spacer", width:"100px"},{kind:enyo.VFlexBox, name:"navgroup", style:"position:absolute;right:14px;margin:6px 0;", showing:true, width:"118px", components: [ 
 				{kind: "RadioGroup", name: "radioSelection", components: [
 					{icon: "../images/previous-email-icon.png", disabled:true, name:"prevButton", onclick: "goToPrevEmail"}, 
 					{icon: "../images/next-email-icon.png", disabled:true, name: "nextButton", onclick: "goToNextEmail"} 
diff --git a/usr/palm/applications/com.palm.app.email/mail/source/MailApp.js b/usr/palm/applications/com.palm.app.email/mail/source/MailApp.js
index 47ad47d..2e12439 100644
--- a/usr/palm/applications/com.palm.app.email/mail/source/MailApp.js
+++ b/usr/palm/applications/com.palm.app.email/mail/source/MailApp.js
@@ -269,7 +269,7 @@ enyo.kind({
 				var accountId = currentFolder.accountId;
 				enyo.application.dashboardManager.setFilter(accountId, currentFolder._id);
 			}
-			this.$.body.hideNextPrev();
+			this.$.body.exposeNextPrev();
 		}
 	},
 	rendered: function() {
