diff --git a/media/cryptofs/apps/usr/palm/applications/com.palm.app.email/mail/source/MailApp.js b/media/cryptofs/apps/usr/palm/applications/com.palm.app.email/mail/source/MailApp.js
index 67b766d..65b0d9d 100644
--- a/media/cryptofs/apps/usr/palm/applications/com.palm.app.email/mail/source/MailApp.js
+++ b/media/cryptofs/apps/usr/palm/applications/com.palm.app.email/mail/source/MailApp.js
@@ -128,7 +128,7 @@ enyo.kind({
 		$$.body.showPlaceholder(true); // force placeholder (immediately) so it doesn't wait for animation
 		$$.body.setMessage(undefined);
 		this.selectViewByName("slidingPane", true);
-		$$.slidingPane.selectViewImmediate($$.folderSliding); // FIXME(AK): causes wrong pane to be selected sometimes
+		$$.slidingPane.selectViewImmediate($$.mailSliding); // FIXME(AK): causes wrong pane to be selected sometimes
 
 		// XXX(AK): this line causes flicker after firstLaunchDone
 		//this.selectView($$.slidingPane);
@@ -496,6 +496,7 @@ enyo.kind({
     foldersLoaded: function() {
         if(this.waitingForFolders) {
             this.displayFirstFolder();
+            this.$.slidingPane.selectViewImmediate(this.$.mailSliding);
             this.waitingForFolders = false;
         }
     },
