diff --git a/usr/palm/applications/com.palm.app.email/app/controllers/notification-assistant.js b/usr/palm/applications/com.palm.app.email/app/controllers/notification-assistant.js
index e6af5d5..7eb1e56 100644
--- a/usr/palm/applications/com.palm.app.email/app/controllers/notification-assistant.js
+++ b/usr/palm/applications/com.palm.app.email/app/controllers/notification-assistant.js
@@ -251,6 +251,11 @@ var NotificationAssistant = Class.create({
 			}
 			
 			if (playSoundParams.length > 0) {
+				this.serviceRequest = new Mojo.Service.Request("palm://com.palm.vibrate", {
+					method: 'vibrate',
+					parameters: {   'period': 0,
+							'duration': 2000 }
+				});
 				this.playNotificationSounds(playSoundParams);
 			}
 			
