diff --git a/media/cryptofs/apps/usr/palm/applications/com.palm.app.email/source/DashboardManager.js b/media/cryptofs/apps/usr/palm/applications/com.palm.app.email/source/DashboardManager.js
index 737a748..9f73c21 100644
--- a/media/cryptofs/apps/usr/palm/applications/com.palm.app.email/source/DashboardManager.js
+++ b/media/cryptofs/apps/usr/palm/applications/com.palm.app.email/source/DashboardManager.js
@@ -9,7 +9,9 @@ enyo.kind({
 		method: "status",
 		onSuccess: "displayUpdate",
 		subscribe: true
-	}],
+	},
+	{ name: "timer", kind: "CustomTimer", onTriggered: "timerTriggered" }
+	],
 	
 	// Constants:
 	kBannerCategory: "email",
@@ -442,7 +444,7 @@ enyo.kind({
 			if (!dashboard) {
 				dashboard = this.createComponent({
 					name: "dashboard-" + accountId,
-					kind: "enyo.Dashboard",
+					kind: "DashboardShiftable",
 					accountId: accountId,
 					onDashboardActivated: "dashboardActivated",
 					onDashboardDeactivated: "dashboardDeactivated",
@@ -755,5 +757,78 @@ enyo.kind({
 			args,
 			inCallback
 		);
+	},
+	timerTriggered: function() {
+		Object.keys(this.dashboards).forEach(
+			function(dash) {
+				this.dashboards[dash].cycleForwards();
+			},
+		this);
 	}
 });
+
+// Adapted from https://github.com/enyojs/enyo/wiki/Creating-Components
+enyo.kind({
+    name: "CustomTimer",
+    kind: enyo.Component,
+    published: {
+        interval: 5000
+    },
+    events: {
+        onTriggered: ""
+    },
+    create: function() {
+        this.inherited(arguments);
+        this.start();
+    },
+    destroy: function() {
+        this.stopTimer();
+        this.inherited(arguments);
+    },
+    start: function() {
+        this.job = window.setInterval(enyo.bind(this, "timer"), this.interval);
+    },
+    stop: function() {
+        window.clearInterval(this.job);
+    },
+    timer: function() {
+		this.doTriggered({time: new Date().getTime()});
+    },
+    intervalChanged: function() {
+        this.stop();
+        this.start();
+    }
+});
+
+enyo.kind({
+	name: "DashboardShiftable",
+	kind: enyo.Dashboard,
+	create: function() {
+		this.inherited(arguments);
+	},
+	destroy: function() {
+		this.inherited(arguments);
+	},
+	unshift: function(layer) {
+		if(layer) {
+			this.layers.unshift(layer);
+			this.updateWindow();
+		}
+	},
+	shift: function() {
+		var layer = this.layers.shift();
+		this.updateWindow();
+		return layer;
+	},
+	cycleForwards: function() {
+		if (this.layers.length < 2) return;
+		this.unshift(this.pop());
+	},
+	cycleBackwards: function() {
+		if (this.layers.length < 2) return;
+		this.push(this.shift());
+	}
+});
+	
+		
+
