diff --git a/usr/palm/applications/com.palm.app.email/app/controllers/compose-assistant.js b/usr/palm/applications/com.palm.app.email/app/controllers/compose-assistant.js
index e4f2e91..029cfe4 100644
--- a/usr/palm/applications/com.palm.app.email/app/controllers/compose-assistant.js
+++ b/usr/palm/applications/com.palm.app.email/app/controllers/compose-assistant.js
@@ -614,7 +614,7 @@ var ComposeAssistant = Class.create({
 					break;
 
 				case 'send':
-					this.send();
+					this.asktosend();
 					break;
 
 				case 'cancel':
@@ -642,6 +642,18 @@ var ComposeAssistant = Class.create({
 		}
 	},
 
+	asktosend: function() {
+		this.controller.showAlertDialog({
+			onChoose: function(value) { if(value == "yes") { this.send(); }	},
+			title: "<b>" + $L("Confirm send") + "</b>",
+				message: $L("Are you sure you want to send this email?"),
+				choices: [
+					{label:$L("Yes"), value:"yes", type:"affirmative"},
+					{label:$L("No"), value:"no", type:"alert"}
+			]
+		});
+	},
+
 	setPriority: function(priority) {
 		this.priority = priority;
 		if (priority) {
diff --git a/usr/palm/applications/com.palm.app.email/resources/de_de/strings.json b/usr/palm/applications/com.palm.app.email/resources/de_de/strings.json
index 387596e..cf6a6a2 100644
--- a/usr/palm/applications/com.palm.app.email/resources/de_de/strings.json
+++ b/usr/palm/applications/com.palm.app.email/resources/de_de/strings.json
@@ -383,5 +383,7 @@
 	"page at #{host}": "Seite auf #{host}",
 	"picture at #{host}": "Bild auf #{host}",
 	"picture link": "Link zu Bild",
-	"the account does not yet exist": "Das Konto ist noch nicht vorhanden"
+	"the account does not yet exist": "Das Konto ist noch nicht vorhanden",
+	"Confirm send": "Senden bestätigen",
+	"Are you sure you want to send this email?": "Sind sie sicher, diese eMail versenden zu wollen?"
 }
\ No newline at end of file
