diff --git a/usr/palm/applications/com.palm.app.email/mail/source/MailApp.js b/usr/palm/applications/com.palm.app.email/mail/source/MailApp.js
index afa3ae3..b293b01 100644
--- a/usr/palm/applications/com.palm.app.email/mail/source/MailApp.js
+++ b/usr/palm/applications/com.palm.app.email/mail/source/MailApp.js
@@ -386,7 +386,11 @@ enyo.kind({
 	},
 	
 	showMainApp: function() {
-		this.$.slidingPane.selectViewImmediate(this.$.folderSliding);
+		if(this.$.mailSliding.getShowing()) {
+			this.$.slidingPane.selectViewImmediate(this.$.mailSliding);
+		} else {
+			this.$.slidingPane.selectViewImmediate(this.$.folderSliding);
+		}
 		this.selectView(this.$.slidingPane);
 	},
 	
@@ -462,6 +466,7 @@ enyo.kind({
     foldersLoaded: function() {
         if(this.waitingForFolders) {
             this.displayFirstFolder();
+            this.$.slidingPane.selectViewImmediate(this.$.mailSliding);
             this.waitingForFolders = false;
         }
     },
