Version: 1.3.1-1
Index: /usr/palm/applications/com.palm.app.email/app/controllers/accounts-assistant.js
===================================================================
--- .orig/usr/palm/applications/com.palm.app.email/app/controllers/accounts-assistant.js
+++ /usr/palm/applications/com.palm.app.email/app/controllers/accounts-assistant.js
@@ -55,6 +55,8 @@ var AccountsAssistant = Class.create(App
 		EmailAppDepot.depot.isShowFavorites(AccountsAssistant.kFavoriteFoldersAccountID, this.depotGetFavExpanded.bind(this));
 
 		this.controller.listen(this.controller.sceneElement, Mojo.Event.keypress, this.keypressHandler.bind(this));
+	
+		this.controller.window.PalmSystem.setWindowOrientation("free");
 		
 		//set up account folder's search
 		this.folderFilterField = this.controller.get('account_folders_filter');
@@ -120,8 +122,8 @@ var AccountsAssistant = Class.create(App
 	keypressHandler: function(e) {
 		this.orientationString += String.fromCharCode(e.originalEvent.charCode);
 		Mojo.Log.info(this.orientationString);
-		if (this.orientationString.length === 12) {
-			if (this.orientationString === "RocknRollHax") {
+		if (this.orientationString.length === 4) {
+			if (this.orientationString === "wide") {
 				Mojo.Log.info("Let's rock and roll");
 				var targetWindow = this.controller.window;
 				if (targetWindow.PalmSystem.setWindowOrientation) {
Index: /usr/palm/applications/com.palm.app.email/app/controllers/compose-assistant.js
===================================================================
--- .orig/usr/palm/applications/com.palm.app.email/app/controllers/compose-assistant.js
+++ /usr/palm/applications/com.palm.app.email/app/controllers/compose-assistant.js
@@ -186,6 +186,8 @@ var ComposeAssistant = Class.create({
 
 		// Delayed a little, since we want the header part of render as quickly as possible
 		ComposeAssistant.onLoad.defer(this.controller, this.email);
+	
+		this.controller.window.PalmSystem.setWindowOrientation("free");
 	},
 
 	cleanup: function() {
Index: /usr/palm/applications/com.palm.app.email/app/controllers/list-assistant.js
===================================================================
--- .orig/usr/palm/applications/com.palm.app.email/app/controllers/list-assistant.js
+++ /usr/palm/applications/com.palm.app.email/app/controllers/list-assistant.js
@@ -45,6 +45,7 @@
 		visible:true};
 		
 	this.controller.setupWidget(Mojo.Menu.appMenu, undefined, this.appMenuModel);
+	this.controller.window.PalmSystem.setWindowOrientation("free");	
   },
 
   // NOTE: this is called by app_scene's _setup function
\ No newline at end of file
