diff --git a/usr/palm/applications/com.palm.app.email/app/controllers/message-assistant.js b/usr/palm/applications/com.palm.app.email/app/controllers/message-assistant.js
index 761069e..8601688 100644
--- a/usr/palm/applications/com.palm.app.email/app/controllers/message-assistant.js
+++ b/usr/palm/applications/com.palm.app.email/app/controllers/message-assistant.js
@@ -663,6 +663,8 @@ var MessageAssistant = Class.create({
 			this.controller.get('invite-accept').observe(Mojo.Event.tap, this.boundHandleInviteResponseAccept);
 			this.controller.get('invite-tentative').observe(Mojo.Event.tap, this.boundHandleInviteResponseTentative);
 			this.controller.get('invite-decline').observe(Mojo.Event.tap, this.boundHandleInviteResponseDecline);
+			this.controller.get('invite-icon').observe(Mojo.Event.tap, this.handleInviteIconTapped.bind(this));
+			this.invitationStartDate = resp.whenStart;
 		}
 		else if (EmailFlags.isMeetingCancel(resp.flags)) {
 			content = Mojo.View.render({template: 'message/meeting_cancellation', object: {}});
@@ -673,7 +675,18 @@ var MessageAssistant = Class.create({
 			this.controller.get('email-readview-invitations').update("");
 		}
 	},
-
+	
+	handleInviteIconTapped: function(event){
+		var params = {date: this.invitationStartDate};
+		this.controller.serviceRequest("palm://com.palm.applicationManager", {
+			method: "launch",
+			parameters: {
+				id: "com.palm.app.calendar",
+				params: params
+			}
+		});		
+	},
+	
 	renderMessageRecipients: function(resp) {
 		Mojo.Log.info("Starting to render message recipients.");
 		var recips = resp.recipients;
diff --git a/usr/palm/applications/com.palm.app.email/app/views/message/meeting_invitation.html b/usr/palm/applications/com.palm.app.email/app/views/message/meeting_invitation.html
index 7d44f1d..39d38a3 100644
--- a/usr/palm/applications/com.palm.app.email/app/views/message/meeting_invitation.html
+++ b/usr/palm/applications/com.palm.app.email/app/views/message/meeting_invitation.html
@@ -1,5 +1,5 @@
 <div class="invite-info">
-  <div class="email-invite-icon"></div>
+  <div id="invite-icon" class="email-invite-icon"></div>
   <div class="when-info truncating-text">#{when}</div>
   <div class="where-info truncating-text">#{where}</div>
   <div class="conflict-info truncating-text">#{conflict}</div>
