diff --git a/usr/palm/applications/com.palm.app.email/mail/source/MailApp.js b/usr/palm/applications/com.palm.app.email/mail/source/MailApp.js
index 47ad47d..358dcf1 100644
--- a/usr/palm/applications/com.palm.app.email/mail/source/MailApp.js
+++ b/usr/palm/applications/com.palm.app.email/mail/source/MailApp.js
@@ -474,7 +474,11 @@ enyo.kind({
 	
 	showMainApp: function() {
 		this.$.slidingPane.setShowing(true);
-		this.$.slidingPane.selectViewImmediate(this.$.folderSliding);
+		if(this.$.mailSliding.getShowing()) {
+			this.$.slidingPane.selectViewImmediate(this.$.mailSliding);
+		} else {
+			this.$.slidingPane.selectViewImmediate(this.$.folderSliding);
+		}
 		this.selectView(this.$.slidingPane);
 		enyo.keyboard.setResizesWindow(false);
 	},
@@ -559,6 +563,7 @@ enyo.kind({
     foldersLoaded: function() {
         if(this.waitingForFolders) {
             this.displayFirstFolder();
+            this.$.slidingPane.selectViewImmediate(this.$.mailSliding);
             this.waitingForFolders = false;
         }
     },
