diff --git a/usr/palm/applications/com.palm.app.email/app/controllers/message-assistant.js b/usr/palm/applications/com.palm.app.email/app/controllers/message-assistant.js
index 95bb302..bf1beac 100644
--- a/usr/palm/applications/com.palm.app.email/app/controllers/message-assistant.js
+++ b/usr/palm/applications/com.palm.app.email/app/controllers/message-assistant.js
@@ -81,6 +81,7 @@ var MessageAssistant = function(targetEmail, focusStage, detailsObj){
 	this.handleWebViewSingleTap = this.handleWebViewSingleTap.bind(this);
 	this.handleDOMClicked = this.handleDOMClicked.bind(this);
 	this.handleMessageDownloadRetry = this.handleMessageDownloadRetry.bind(this);
+	this.boundHandleWebViewHoldTap = this.handleWebViewHoldTap.bind(this);
 	
 	this.displayContactAvatarAndPresence = this.displayContactAvatarAndPresence.bind(this);
 	
@@ -220,6 +221,7 @@ var MessageAssistant = function(targetEmail, focusStage, detailsObj){
 		this.webview.removeEventListener(Mojo.Event.webViewMimeHandoff, this.handleLinkClicked, false);
 		this.webview.removeEventListener(Mojo.Event.webViewImageSaved, this.handleInlineImageSaved, false);
 		this.webview.removeEventListener('singletap', this.handleWebViewSingleTap, true);
+		this.webview.removeEventListener(Mojo.Event.hold, this.boundHandleWebViewHoldTap, true);
 		
 		this.webview.removeEventListener(Mojo.Event.webViewLoadStarted, this.boundBodyRenderComplete, true);
 		
@@ -558,6 +560,7 @@ var MessageAssistant = function(targetEmail, focusStage, detailsObj){
 		this.webview.addEventListener(Mojo.Event.webViewMimeHandoff, this.handleLinkClicked, false);
 		this.webview.addEventListener(Mojo.Event.webViewImageSaved, this.handleInlineImageSaved, false);
 		this.webview.addEventListener('singletap', this.handleWebViewSingleTap, true);
+		this.webview.addEventListener(Mojo.Event.hold, this.boundHandleWebViewHoldTap, true);
 	
 		this.boundBodyRenderComplete = this.bodyRenderComplete.bind(this);
 		this.webview.addEventListener(Mojo.Event.webViewLoadStarted, this.boundBodyRenderComplete, true);
@@ -2014,8 +2017,15 @@ MessageAssistant.prototype._getAttachmentById = function(id) {
 	MessageAssistant.prototype.handleWebViewSingleTap = function(event) {	
 		try {
 			var tapPt = Mojo.Dom.viewportOffset(this.webview);
-			tapPt.left = event.centerX - tapPt.left;
-			tapPt.top  = event.centerY - tapPt.top;
+			if(event.isHoldTap) {
+				event.altKey = true;
+				tapPt.left = event.down.offsetX;
+				tapPt.top = event.down.offsetY;
+			}
+			else {
+				tapPt.left = event.centerX - tapPt.left;
+				tapPt.top  = event.centerY - tapPt.top;
+			}
 
 			//Mojo.Log.info("MessageAssistant.handleWebViewSingleTap(): event.altKey=%s, tapPt.left=%d, tapPt.top=%d", event.altKey, tapPt.left, tapPt.top);
 			if (event.altKey) {
@@ -2629,6 +2639,11 @@ MessageAssistant.prototype._getAttachmentById = function(id) {
 		});	
 	};
 	
+	MessageAssistant.prototype.handleWebViewHoldTap = function(event) {	
+		event.isHoldTap = true;
+		this.handleWebViewSingleTap(event);
+	},
+	
 
 
 MessageAssistant.kAppMenuMarkRead = $L('Mark as Read');
