diff --git a/usr/palm/applications/com.palm.app.email/app/controllers/accountmanager-assistant.js b/usr/palm/applications/com.palm.app.email/app/controllers/accountmanager-assistant.js
index de99e10..7df3e69 100644
--- a/usr/palm/applications/com.palm.app.email/app/controllers/accountmanager-assistant.js
+++ b/usr/palm/applications/com.palm.app.email/app/controllers/accountmanager-assistant.js
@@ -262,7 +262,7 @@ AccountmanagerAssistant.prototype.handleAccountsChanged = function() {
 							  Mojo.Event.stageActivate, this.renderDefaultAccount, true);
 
 		if (this.controller.stageController.setWindowOrientation) {
-			this.controller.stageController.setWindowOrientation("free");
+			this.controller.stageController.setWindowOrientation("up");
 		}
 	};
 
diff --git a/usr/palm/applications/com.palm.app.email/app/controllers/accountpreferences-assistant.js b/usr/palm/applications/com.palm.app.email/app/controllers/accountpreferences-assistant.js
index 15f6bac..b64b3ce 100644
--- a/usr/palm/applications/com.palm.app.email/app/controllers/accountpreferences-assistant.js
+++ b/usr/palm/applications/com.palm.app.email/app/controllers/accountpreferences-assistant.js
@@ -195,7 +195,7 @@ AccountpreferencesAssistant.prototype.cleanup = function() {
 		this.folderRequest.cancel();
 	}
 	if (this.controller.stageController.setWindowOrientation) {
-		this.controller.stageController.setWindowOrientation("free");
+		this.controller.stageController.setWindowOrientation("up");
 	}
 };
 
diff --git a/usr/palm/applications/com.palm.app.email/app/controllers/accounts-assistant.js b/usr/palm/applications/com.palm.app.email/app/controllers/accounts-assistant.js
index e6c1075..1ae1268 100644
--- a/usr/palm/applications/com.palm.app.email/app/controllers/accounts-assistant.js
+++ b/usr/palm/applications/com.palm.app.email/app/controllers/accounts-assistant.js
@@ -140,7 +140,7 @@ AccountsAssistant.prototype.setup = function() {
 	// allow landscape mode in this scene, and all stacked on it
 	// we explicitly disable in account prefs and related scenes
 	if (this.controller.stageController.setWindowOrientation) {
-		this.controller.stageController.setWindowOrientation("free");
+		this.controller.stageController.setWindowOrientation("up");
 	}
 
 	Mojo.Log.warn("NOTE: setup() for scene %s DONE", this.controller.sceneName);
diff --git a/usr/palm/applications/com.palm.app.email/app/controllers/create-assistant.js b/usr/palm/applications/com.palm.app.email/app/controllers/create-assistant.js
index 4a1cbd0..0b17eaf 100644
--- a/usr/palm/applications/com.palm.app.email/app/controllers/create-assistant.js
+++ b/usr/palm/applications/com.palm.app.email/app/controllers/create-assistant.js
@@ -259,7 +259,7 @@ CreateAssistant.prototype.cleanup = function() {
 		EmailAccount.saveAccountDetails(this.controller, this.accountDetails);
 	}
 	if (this.controller.stageController.setWindowOrientation) {
-		this.controller.stageController.setWindowOrientation("free");
+		this.controller.stageController.setWindowOrientation("up");
 	}
 };
 
diff --git a/usr/palm/applications/com.palm.app.email/app/controllers/setpin-assistant.js b/usr/palm/applications/com.palm.app.email/app/controllers/setpin-assistant.js
index 91078c5..451f9b8 100644
--- a/usr/palm/applications/com.palm.app.email/app/controllers/setpin-assistant.js
+++ b/usr/palm/applications/com.palm.app.email/app/controllers/setpin-assistant.js
@@ -103,7 +103,7 @@ var SetpinAssistant = function(policy, onSuccess){
 	
 	SetpinAssistant.prototype.cleanup = function(){
 		if (this.controller.stageController.setWindowOrientation) {
-			this.controller.stageController.setWindowOrientation("free");
+			this.controller.stageController.setWindowOrientation("up");
 		}
 
 	};
diff --git a/usr/palm/applications/com.palm.app.email/app/controllers/wizard-assistant.js b/usr/palm/applications/com.palm.app.email/app/controllers/wizard-assistant.js
index d42ea1d..963a2a6 100644
--- a/usr/palm/applications/com.palm.app.email/app/controllers/wizard-assistant.js
+++ b/usr/palm/applications/com.palm.app.email/app/controllers/wizard-assistant.js
@@ -489,7 +489,7 @@ WizardAssistant.prototype.popScene = function() {
 WizardAssistant.prototype.cleanup = function() {
 	this.wizard.cleanup();
 	if (this.controller.stageController.setWindowOrientation) {
-		this.controller.stageController.setWindowOrientation("free");
+		this.controller.stageController.setWindowOrientation("up");
 	}
 };
 
