diff --git a/usr/palm/applications/com.palm.app.email/app/controllers/compose-assistant.js b/usr/palm/applications/com.palm.app.email/app/controllers/compose-assistant.js
index 72daa3f..f4a1062 100644
--- a/usr/palm/applications/com.palm.app.email/app/controllers/compose-assistant.js
+++ b/usr/palm/applications/com.palm.app.email/app/controllers/compose-assistant.js
@@ -341,6 +341,10 @@ ComposeAssistant.prototype.setup = function() {
 	
 	// Delayed a little, since we want the header part of render as quickly as possible
 	this.onLoadDeferId = _.defer(this.onLoad.bind(this));
+
+	// enable free orientation
+	// was for 1.4 this.controller.window.PalmSystem.setWindowOrientation("free");
+	this.controller.stageController.setWindowOrientation("free");
 	
 	// update ui to disable save until email is dirty
 	this.setDirty(false);
diff --git a/usr/palm/applications/com.palm.app.email/app/controllers/list-assistant.js b/usr/palm/applications/com.palm.app.email/app/controllers/list-assistant.js
index d224d2a..0226364 100644
--- a/usr/palm/applications/com.palm.app.email/app/controllers/list-assistant.js
+++ b/usr/palm/applications/com.palm.app.email/app/controllers/list-assistant.js
@@ -77,6 +77,10 @@ ListAssistant.prototype.setup = function() {
 			]};
 	this.controller.setupWidget(Mojo.Menu.commandMenu, undefined, this.cmdMenuModel);
 	
+	// enable free orientation
+	// was for 1.4 this.controller.window.PalmSystem.setWindowOrientation("free");
+	this.controller.stageController.setWindowOrientation("free");
+	
 	this.appMenuModel = {
 		visible:true,
 		label:$L('Folder'),
