diff --git a/usr/palm/applications/com.palm.app.email/app/models/EmailAppDepot.js b/usr/palm/applications/com.palm.app.email/app/models/EmailAppDepot.js
index b2b1cd7..57a4157 100644
--- a/usr/palm/applications/com.palm.app.email/app/models/EmailAppDepot.js
+++ b/usr/palm/applications/com.palm.app.email/app/models/EmailAppDepot.js
@@ -47,6 +47,7 @@ var EmailAppDepot = Class.create({
 		if (accounts.list !== undefined && this.numAccounts !== accounts.list.length) {
 			// look for accounts that have been removed
 			var removedAccounts = [];
+      if (this.accounts.list) {
 			this.accounts.list.each(function(existingAcct){
 				var stillExists = false;
 				accounts.list.each(function(newAcct) {
@@ -59,6 +60,7 @@ var EmailAppDepot = Class.create({
 					removedAccounts.push(existingAccnt);
 				}
 			});
+      }
 			if (removedAccounts.length > 0) {
 				// delete the security policy that are associated with the deleted
 				// accounts. 
