diff --git a/usr/palm/applications/com.palm.app.email/app/models/Email.js b/usr/palm/applications/com.palm.app.email/app/models/Email.js
index 9392247..1c4e414 100644
--- a/usr/palm/applications/com.palm.app.email/app/models/Email.js
+++ b/usr/palm/applications/com.palm.app.email/app/models/Email.js
@@ -368,7 +368,7 @@ Email.getPriorityClass = function(priority) {
 }
 
 Email.kSignaturePlaceholder = "<span style='font-family:Prelude, Verdana, san-serif;'><br><br></span><span id='signature'></span>";
-Email.kReplyForwardSetupElems = "<br><span style='color:navy; font-family:Prelude, Verdana, san-serif; '>" + Email.kSignaturePlaceholder + "<hr align='left' style='width:75%'/>";
+Email.kReplyForwardSetupElems = "<br><span style='color:black; font-family:Prelude, Verdana, san-serif; '>" + Email.kSignaturePlaceholder + "<hr align='left' style='width:75%'/>";
 
 Email.kSubjectPrefixRe = $L("Re: ");
 Email.kSubjectPrefixFw = $L("Fw: ");
