diff --git a/usr/palm/applications/com.palm.app.email/app/models/Email.js b/usr/palm/applications/com.palm.app.email/app/models/Email.js
index 5063cd5..ef728ed 100644
--- a/usr/palm/applications/com.palm.app.email/app/models/Email.js
+++ b/usr/palm/applications/com.palm.app.email/app/models/Email.js
@@ -489,7 +489,7 @@
 }

 Email.kSignaturePlaceholder = "<span style='font-family:Prelude, Verdana, san-serif;'><br><br></span><span id='signature'></span>";
-Email.kReplyForwardSetupElems = "<span style='color:navy; font-family:Prelude, Verdana, san-serif; '>" + Email.kSignaturePlaceholder + "<hr align='left' style='width:75%'/>";
+Email.kReplyForwardSetupElems = "<span style='color:black; font-family:Prelude, Verdana, san-serif; '>" + Email.kSignaturePlaceholder + "<hr align='left' style='width:75%'/>";

 Email.kSubjectPrefixRe = $L("Re: ");
 Email.kSubjectPrefixFw = $L("Fw: ");
