diff --git a/usr/palm/applications/com.palm.app.email/app/models/Composition.js b/usr/palm/applications/com.palm.app.email/app/models/Composition.js
index bde9e8f..ef085ef 100644
--- a/usr/palm/applications/com.palm.app.email/app/models/Composition.js
+++ b/usr/palm/applications/com.palm.app.email/app/models/Composition.js
@@ -88,6 +88,16 @@ ComposeAssistant.prototype.Composition = function(options) {
 			this._setSender(EmailApp.accounts.getDefaultAccountId());
 		}
 	}
+	if (options && options.params && options.params.recipients ) {
+		if (options.params.recipients[0] === "you@self.com"  || (options.params.recipients[0].value && options.params.recipients[0].value === "you@self.com") ) {
+			if ( !this.email.to ) {
+						this.email.to=[];		// may be not needed. 'to' is currently not set/fixed in _setFromLaunchParams
+						this.email.to.push({addr: this.email.from.addr, name: "You", type: EmailRecipient.typeTo});
+					} else {
+						this.email.to[0]={addr: this.email.from.addr, name: "You", type: EmailRecipient.typeTo};
+			}
+		}
+	}
 	
 };
 
