diff --git a/usr/palm/applications/com.palm.app.email/app/models/EmailAppDepot.js b/usr/palm/applications/com.palm.app.email/app/models/EmailAppDepot.js
index 6179cfd..8c0b079 100644
--- a/usr/palm/applications/com.palm.app.email/app/models/EmailAppDepot.js
+++ b/usr/palm/applications/com.palm.app.email/app/models/EmailAppDepot.js
@@ -64,6 +64,7 @@ var EmailAppDepot = Class.create({
 		if (this.accounts.list && accounts.list && this.numAccounts !== accounts.list.length) {
 			// look for accounts that have been removed
 			var removedAccounts = [];
+      if (this.accounts.list) {
 			this.accounts.list.each(function(existingAcct){
 				var stillExists = false;
 				accounts.list.each(function(newAcct) {
@@ -76,6 +77,7 @@ var EmailAppDepot = Class.create({
 					removedAccounts.push(existingAccnt);
 				}
 			});
+      }
 			if (removedAccounts.length > 0) {
 				// delete the security policy that are associated with the deleted
 				// accounts. 
