diff --git a/usr/palm/applications/com.palm.app.email/compose/source/Composition.js b/usr/palm/applications/com.palm.app.email/compose/source/Composition.js
index faacdc8..1a6dfdd 100644
--- a/usr/palm/applications/com.palm.app.email/compose/source/Composition.js
+++ b/usr/palm/applications/com.palm.app.email/compose/source/Composition.js
@@ -38,7 +38,8 @@ EmailApp.Composition = function(options) {
 	// TODO: Refactor this. it's hella ugly.
 	var targetAcctId = options && options.accountId || enyo.application.accounts.getDefaultAccountId();
 	var signature = AccountPreferences.getSignature(targetAcctId);
-	var defaultBodyText = "<span style='font-family:Prelude, Verdana, san-serif;'><br><br></span><span id='signature'>" + signature + "</span>";
+//	var defaultBodyText = "<span style='font-family:Prelude, Verdana, san-serif;'><br><br></span><span id='signature'>" + signature + "</span>";
+	var defaultBodyText = "<span style=\"font-family:Prelude, Calibri, Arial, sans-serif; font-size: 15px;\"><br><br></span><span id='signature'>" + signature + "</span>";
 	
 	this.draftEmail.setBodyContent(defaultBodyText);
 	
@@ -72,16 +73,16 @@ EmailApp.Composition = function(options) {
 };
 
 // Need to break this out of the regular prototype assignment, since it's used in the def'n of REPLY_FWD_HTML
-EmailApp.Composition.prototype.SIGNATURE_PLACEHOLDER = "<span style='font-family:Prelude, Verdana, san-serif;'><br><br></span><span id='signature'></span>";
+EmailApp.Composition.prototype.SIGNATURE_PLACEHOLDER = "<span style=\"font-family:Prelude, Calibri, Arial, Sans-Serif; font-size: 15px;\"><br><br></span><span id='signature'></span>";
 
 EmailApp.Composition.prototype = {
 
 	SIGNATURE_PLACEHOLDER: EmailApp.Composition.prototype.SIGNATURE_PLACEHOLDER,
-	REPLY_FWD_HTML: "<br><br>" + EmailApp.Composition.prototype.SIGNATURE_PLACEHOLDER + "<span style='color:navy; font-family:Prelude, Verdana, san-serif; '><hr align='left' style='width:75%'/>",
-
-	SUBJECT_PREFIX_RE: $L("Re: "),
-	SUBJECT_PREFIX_FW: $L("Fw: "),
-
+	REPLY_FWD_HTML: EmailApp.Composition.prototype.SIGNATURE_PLACEHOLDER + "<span style=\"color:navy; font-family:Prelude, Calibri, Arial, sanS-seriF;\"><br><hr align='left' style='width:75%'/>",
+	
+	SUBJECT_PREFIX_RE: $L("RE: "), 
+	SUBJECT_PREFIX_FW: $L("FW: "), 
+	
 	isForward: function() {
 		return (this.action === "forward");
 	},
diff --git a/usr/palm/applications/com.palm.app.email/data/AccountPreferences.js b/usr/palm/applications/com.palm.app.email/data/AccountPreferences.js
index 1846377..4d72313 100644
--- a/usr/palm/applications/com.palm.app.email/data/AccountPreferences.js
+++ b/usr/palm/applications/com.palm.app.email/data/AccountPreferences.js
@@ -99,10 +99,11 @@ AccountPreferences._correctSignature = function(signature) {
 
 AccountPreferences.addStylingToSig = function(signature){
 	// If the signature is empty or already has styling, just return it. Otherwise add the styling
-	if (!signature || signature.indexOf('<div style="font-family: arial') === 0) {
+	if (!signature || signature.indexOf('<div style="font-family:') === 0) {
 		return signature;
 	} 
-	return '<div style="font-family: arial, sans-serif; font-size: 12px;color: #999999">' + signature + '</div>';
+	return '<div style="font-family:Prelude, Calibri, Arial, sans-serif; font-size:12px; color: #999999">' + signature + '</div>';
+
 };
 
 
diff --git a/usr/palm/applications/com.palm.app.email/data/EmailAccount.js b/usr/palm/applications/com.palm.app.email/data/EmailAccount.js
index 6eb5d19..0d176ce 100644
--- a/usr/palm/applications/com.palm.app.email/data/EmailAccount.js
+++ b/usr/palm/applications/com.palm.app.email/data/EmailAccount.js
@@ -531,10 +531,11 @@ EmailAccount.isNotSmartFolders = function(account) {
 
 EmailAccount.addStylingToSig = function(signature){
 	// If the signature is empty or already has styling, just return it. Otherwise add the styling
-	if (!signature || Foundations.StringUtils.startsWith(signature, '<div style="font-family: arial')) {
+//	if (!signature || Foundations.StringUtils.startsWith(signature, '<div style="font-family: arial')) {
+	if (!signature || Foundations.StringUtils.startsWith(signature, '<div style="font-family:')) {
 		return signature;
 	} else {
-		return EmailApp.Util.interpolate('<div style="font-family: arial, sans-serif; font-size: 12px;color: #999999;">#{signature}</div><br>', {
+		return EmailApp.Util.interpolate('<div style="font-family:Prelude, Calibri, Arial, sans-serif; font-size:12px; color: #999999">#{signature}</div><br>', {
 			signature: signature
 		});
 	}
