diff --git a/media/cryptofs/apps/usr/palm/applications/com.palm.app.email/mail/source/MailApp.js b/media/cryptofs/apps/usr/palm/applications/com.palm.app.email/mail/source/MailApp.js
index 67b766d..49e9022 100644
--- a/media/cryptofs/apps/usr/palm/applications/com.palm.app.email/mail/source/MailApp.js
+++ b/media/cryptofs/apps/usr/palm/applications/com.palm.app.email/mail/source/MailApp.js
@@ -17,7 +17,7 @@ enyo.kind({
 		{name: "checkFirstLaunch", kind: "Accounts.checkFirstLaunch", onCheckFirstLaunchResult: "_firstLaunchResponder", appId: "com.palm.app.email"},
 					// TODO: replace SlidingPane style below with this, once animations are added back in:  style: "background: url(../images/Mail-2-1a-256.png) center center no-repeat, url(../images/loading-bg.png) top left repeat-x;"
 		{name: "slidingPane", kind: "SlidingPane", style: "background: url(../images/loading-bg.png) top left repeat-x;", flex: 1, components: [
-			{width: "320px", name: "folderSliding", dragAnywhere: false, fixedWidth: true, components: [
+			{width: "200px", name: "folderSliding", dragAnywhere: false, fixedWidth: true, components: [
 				{name: "accounts", kind: "MailAccounts", flex: 1,
 					onSelectFolder: "folderChosen",
 					onComposeMessage: "composeMessage",
