diff --git a/usr/palm/applications/com.palm.app.email/app/controllers/compose-assistant.js b/usr/palm/applications/com.palm.app.email/app/controllers/compose-assistant.js
index e4f2e91..e604fd4 100644
--- a/usr/palm/applications/com.palm.app.email/app/controllers/compose-assistant.js
+++ b/usr/palm/applications/com.palm.app.email/app/controllers/compose-assistant.js
@@ -172,8 +172,11 @@ var ComposeAssistant = Class.create({
 				visible: true,
 				menuClass: 'palm-white',
 				items: [
+					{label:$L('Priority'), icon:'priority', command:'priority'},
 					{label:$L('Attach'), icon:'attach', command:'attach'},
-					{label:$L('Send'), icon:'send', command:'send'}
+					{label:$L('Save'), icon:'save', command:'save'},
+					{label:$L('Send'), icon:'send', command:'send'},
+					{label:$L('Delete'),icon:'delete', command:'cancel'}
 				]};
 
 		this.appMenuSaveIndex = 2;
@@ -383,6 +386,7 @@ var ComposeAssistant = Class.create({
 	},
 
 	orientationChanged: function(orientation) {
+		this.controller.stageController.setWindowOrientation(orientation);
 		if (orientation === "left" || orientation === "right") {
 			this.controller.sceneElement.addClassName('landscape');
 		} else {
@@ -614,11 +618,26 @@ var ComposeAssistant = Class.create({
 					break;
 
 				case 'send':
-					this.send();
+					this.controller.showAlertDialog({
+					onChoose: function(value) {if (value == 'yes') {this.send();}},
+					title: $L("Send This Message?"),
+					choices:[
+						{label:$L('Yes'), value:"yes", type:'negative'},  
+						{label:$L("No"), value:"no", type: 'dismiss'}
+							]
+					});
 					break;
 
 				case 'cancel':
-					this.cancelCompose();
+					this.controller.showAlertDialog({
+					onChoose: function(value) {if (value == 'yes') {this.cancelCompose();}},
+					title: $L("Discard this Draft?"),
+					message: $L("This cannot be undone"),
+					choices:[
+						{label:$L('Yes'), value:"yes", type:'negative'},  
+						{label:$L("No"), value:"no", type: 'dismiss'}
+							]
+					});
 					break;
 						
 				case 'priority':
@@ -1164,8 +1183,7 @@ ComposeAssistant.onLoad = function(controller, email) {
 	if (email.action === ComposeAssistant.kForwardAction) {
 		controller.get('forwarded_msg_body').innerHTML = originalText;
 	} else {
-		bodyText += originalText;
-		editor.setStyle({'min-height':'320px', 'font-size': '18px'});
+		controller.get('forwarded_msg_body').innerHTML = originalText;
 	}
 	editor.innerHTML = bodyText;
 	
@@ -1212,6 +1230,6 @@ ComposeAssistant.replaceURIs = function(originalText, email) {
 ComposeAssistant.kForwardAction = 'forward';
 ComposeAssistant.kAppMenuHighPriority = $L('Set as High Priority');
 ComposeAssistant.kAppMenuNormalPriority = $L('Set as Normal Priority');
-ComposeAssistant.kEmptySignatureRegex = /<span id="signature"><div style="font-family: arial, sans-serif; font-size: 12px;color: #999999;">[<br>]*<\/div>[<br>]*<\/span>/
+ComposeAssistant.kEmptySignatureRegex = /<span id="signature"><div style="font-family: Calibri, sans-serif; font-size: 15px;color: #999999;">[<br>]*<\/div>[<br>]*<\/span>/
 ComposeAssistant.kForwardDraftDelimeter = "<span id='FORWARD_DRAFT_TEXT' class='display:none'></span>"; 
 
