diff --git a/media/cryptofs/apps/usr/palm/applications/com.palm.app.email/mail/source/MailApp.js b/media/cryptofs/apps/usr/palm/applications/com.palm.app.email/mail/source/MailApp.js
index 67b766d..55c05b6 100644
--- a/media/cryptofs/apps/usr/palm/applications/com.palm.app.email/mail/source/MailApp.js
+++ b/media/cryptofs/apps/usr/palm/applications/com.palm.app.email/mail/source/MailApp.js
@@ -212,7 +212,7 @@ enyo.kind({
 				var accountId = currentFolder.accountId;
 				enyo.application.dashboardManager.setFilter(accountId, currentFolder._id);
 			}
-			this.$.body.hideNextPrev();
+			this.$.body.exposeNextPrev();
 		}
 	},
 	rendered: function() {
diff --git a/media/cryptofs/apps/usr/palm/applications/com.palm.app.email/mail/source/MessagePane.js b/media/cryptofs/apps/usr/palm/applications/com.palm.app.email/mail/source/MessagePane.js
index 9aac230..c663b05 100644
--- a/media/cryptofs/apps/usr/palm/applications/com.palm.app.email/mail/source/MessagePane.js
+++ b/media/cryptofs/apps/usr/palm/applications/com.palm.app.email/mail/source/MessagePane.js
@@ -79,8 +79,8 @@ enyo.kind({
 					{icon: "../images/icons/toolbar-icon-forward.png", onclick: "forwardClick"},
 					{kind: "Control", className:"toolbar-spacer"},
 					{name: "deleteCommandItem", icon: "../images/icons/toolbar-icon-delete.png",  onclick: "deleteClick"},
-					{kind: "Control", name: "navgroupSpacer", className:"toolbar-spacer", showing:false},
-					{kind:enyo.VFlexBox, name:"navgroup", style:"position:absolute;right:14px;margin:6px 0;", showing:false, width:"118px", components: [ 
+					{kind: "Control", name: "navgroupSpacer", className:"toolbar-spacer", showing:true},
+					{kind: "Control", className:"toolbar-spacer", width:"100px"},{kind:enyo.VFlexBox, name:"navgroup", style:"position:absolute;right:14px;margin:6px 0;", showing:true, width:"118px", components: [ 
 						{kind: "RadioGroup", name: "radioSelection", components: [
 							{icon: "../images/previous-email-icon.png", disabled:true, name:"prevButton", onclick: "goToPrevEmail"}, 
 							{icon: "../images/next-email-icon.png", disabled:true, name: "nextButton", onclick: "goToNextEmail"} 
