diff --git a/usr/palm/applications/com.palm.app.email/app/controllers/accountpreferences-assistant.js b/usr/palm/applications/com.palm.app.email/app/controllers/accountpreferences-assistant.js
index 15f6bac..f0a960a 100644
--- a/usr/palm/applications/com.palm.app.email/app/controllers/accountpreferences-assistant.js
+++ b/usr/palm/applications/com.palm.app.email/app/controllers/accountpreferences-assistant.js
@@ -522,6 +522,11 @@ AccountpreferencesAssistant.prototype.syncWindowOptions = {
 		{label: $L('7 days'), value: 7},
 		{label: $L('2 weeks'), value: 14},
 		{label: $L('1 month'), value: 30},
+		{label: $L('2 months'), value: 60},
+		{label: $L('3 months'), value: 90},
+		{label: $L('6 months'), value: 180},
+		{label: $L('9 months'), value: 270},
+		{label: $L('1 year'), value: 365},
 		{label: $L('All'), value: "0"}
 	]
 };
@@ -562,8 +567,13 @@ AccountpreferencesAssistant.prototype.fetchOptions = {
 		{label: $L('10 minutes'), value: AccountPreferences.SYNC_10_MINS + ""},
 		{label: $L('15 minutes'), value: AccountPreferences.SYNC_15_MINS + ""},
 		{label: $L('30 minutes'), value: AccountPreferences.SYNC_30_MINS + ""},
+		{label: $L('45 minutes'), value: AccountPreferences.SYNC_45_MINS + ""},
 		{label: $L('1 hour'), value: AccountPreferences.SYNC_1_HR + ""},
+		{label: $L('2 hours'), value: AccountPreferences.SYNC_2_HRS + ""},
+		{label: $L('4 hours'), value: AccountPreferences.SYNC_4_HRS + ""},
 		{label: $L('6 hours'), value: AccountPreferences.SYNC_6_HRS + ""},
+		{label: $L('8 hours'), value: AccountPreferences.SYNC_8_HRS + ""},
+		{label: $L('10 hours'), value: AccountPreferences.SYNC_10_HRS + ""},
 		{label: $L('12 hours'), value: AccountPreferences.SYNC_12_HRS + ""},
 		{label: $L('24 hours'), value: AccountPreferences.SYNC_24_HRS + ""},
 		{label: $L('Manual'), value: AccountPreferences.SYNC_MANUAL + ""}
diff --git a/usr/palm/applications/com.palm.app.email/app/controllers/compose-assistant.js b/usr/palm/applications/com.palm.app.email/app/controllers/compose-assistant.js
index 55f61d0..492ecad 100644
--- a/usr/palm/applications/com.palm.app.email/app/controllers/compose-assistant.js
+++ b/usr/palm/applications/com.palm.app.email/app/controllers/compose-assistant.js
@@ -254,8 +254,11 @@ ComposeAssistant.prototype.setup = function() {
 			visible: true,
 			menuClass: 'palm-white',
 			items: [
+				{label:$L('Priority'), icon:'priority', command:'priority'},
 				{label:$L('Attach'), icon:'attach', command:'attach'},
-				{label:$L('Send'), icon:'send', command:'send'}
+					{label:$L('Send'), icon:'send', command:'send'},
+					{label:$L('Save'), icon:'save', command:'save'},
+					{label:$L('Delete'),icon:'delete', command:'cancel'}
 			]};
 	this.controller.setupWidget(Mojo.Menu.commandMenu, undefined, this.cmdMenuModel);
 
@@ -669,16 +672,37 @@ ComposeAssistant.prototype.handleCommand = function(event) {
 				break;
 
 			case 'save':
-				this.draftIsDirty = true; // set to true to make sure the save occurs
-				this.saveDraft();
+					this.controller.showAlertDialog({
+					onChoose: function(value) {if (value == 'yes') {this.saveDraft();}},
+					title: $L("Save This Message?"),
+					choices:[
+						{label:$L('Yes'), value:"yes", type:'negative'},  
+						{label:$L("No"), value:"no", type: 'dismiss'}
+							]
+					});
 				break;
 
 			case 'send':
-				this.send();
-				break;
+					this.controller.showAlertDialog({
+					onChoose: function(value) {if (value == 'yes') {this.send();}},
+					title: $L("Send This Message?"),
+					choices:[
+						{label:$L('Yes'), value:"yes", type:'negative'},  
+						{label:$L("No"), value:"no", type: 'dismiss'}
+							]
+					});
+ 					break;
 
 			case 'cancel':
-				this.cancelCompose();
+					this.controller.showAlertDialog({
+					onChoose: function(value) {if (value == 'yes') {this.cancelCompose();}},
+					title: $L("Discard this Draft?"),
+					message: $L("This cannot be undone"),
+					choices:[
+						{label:$L('Yes'), value:"yes", type:'negative'},  
+						{label:$L("No"), value:"no", type: 'dismiss'}
+							]
+					});
 				break;
 					
 			case 'priority':
diff --git a/usr/palm/applications/com.palm.app.email/app/controllers/list-assistant.js b/usr/palm/applications/com.palm.app.email/app/controllers/list-assistant.js
index 2e239ef..1ca462e 100644
--- a/usr/palm/applications/com.palm.app.email/app/controllers/list-assistant.js
+++ b/usr/palm/applications/com.palm.app.email/app/controllers/list-assistant.js
@@ -72,7 +72,8 @@ ListAssistant.prototype.setup = function() {
 			visible:true,
 			items: [
 				{label:$L('Compose'), icon:'compose', command:'compose'},
-				{},
+				{label:$L('Delete All'), icon:'delete', command:'empty-folder'},
+				{label:$L('All Read'), icon:'priority', command:'mark-folder-read'},
 				{label:$L('Update'), icon:'sync', command:'sync'}
 			]};
 	this.controller.setupWidget(Mojo.Menu.commandMenu, undefined, this.cmdMenuModel);
diff --git a/usr/palm/applications/com.palm.app.email/app/controllers/message-assistant.js b/usr/palm/applications/com.palm.app.email/app/controllers/message-assistant.js
index 95bb302..6a7481e 100644
--- a/usr/palm/applications/com.palm.app.email/app/controllers/message-assistant.js
+++ b/usr/palm/applications/com.palm.app.email/app/controllers/message-assistant.js
@@ -128,6 +128,8 @@ var MessageAssistant = function(targetEmail, focusStage, detailsObj){
 		this.cmdMenuModel = {
 				visible:true,
 				items: [
+					{label:$L('Move'), 	icon:'file', 	command:'move'},
+					{label:$L('Set Flag'),  icon:'make-vip',      command:'flag'},
 					{label:$L('Reply'),     icon:'reply',     command:'reply'},
 					{label:$L('Reply all'), icon:'reply-all', command:'replyAll'},
 					{label:$L('Forward'),   icon:'forward-email', command:'forward'},
diff --git a/usr/palm/applications/com.palm.app.email/app/models/AccountPreferences.js b/usr/palm/applications/com.palm.app.email/app/models/AccountPreferences.js
index 3dd59f4..dc279c0 100644
--- a/usr/palm/applications/com.palm.app.email/app/models/AccountPreferences.js
+++ b/usr/palm/applications/com.palm.app.email/app/models/AccountPreferences.js
@@ -82,8 +82,13 @@ AccountPreferences.SYNC_5_MINS	= 5;
 AccountPreferences.SYNC_10_MINS	= 10;
 AccountPreferences.SYNC_15_MINS	= 15;
 AccountPreferences.SYNC_30_MINS	= 30;
+AccountPreferences.SYNC_45_MINS	= 45;
 AccountPreferences.SYNC_1_HR	= 60;
+AccountPreferences.SYNC_2_HRS	= 120;
+AccountPreferences.SYNC_4_HRS	= 240;
 AccountPreferences.SYNC_6_HRS	= 360;
+AccountPreferences.SYNC_8_HRS	= 480;
+AccountPreferences.SYNC_10_HRS	= 600;
 AccountPreferences.SYNC_12_HRS	= 720;
 AccountPreferences.SYNC_24_HRS	= 1440;
 AccountPreferences.SYNC_PUSH	= -1;
