diff --git a/usr/palm/applications/com.palm.app.email/app/controllers/accountpreferences-assistant.js b/usr/palm/applications/com.palm.app.email/app/controllers/accountpreferences-assistant.js
index 3c48b42..dd696a1 100644
--- a/usr/palm/applications/com.palm.app.email/app/controllers/accountpreferences-assistant.js
+++ b/usr/palm/applications/com.palm.app.email/app/controllers/accountpreferences-assistant.js
@@ -494,7 +494,16 @@ var AccountpreferencesAssistant = Class.create({
 			{label: $L('15 minutes'), value: 15},
 			{label: $L('30 minutes'), value: 30},
 			{label: $L('1 hour'), value: 60},
+			{label: '2 hours', value: 120},
+			{label: '3 hours', value: 180},
+			{label: '4 hours', value: 240},
+			{label: '5 hours', value: 300},
 			{label: $L('6 hours'), value: 360},
+			{label: '7 hours', value: 420},
+			{label: '8 hours', value: 480},
+			{label: '9 hours', value: 540},
+			{label: '10 hours', value: 600},
+			{label: '11 hours', value: 660},
 			{label: $L('12 hours'), value: 720},
 			{label: $L('24 hours'), value: 1440},
 			{label: $L('Manual'), value: 1000000}
