--- .orig/usr/palm/applications/com.palm.app.email/app/controllers/compose-assistant.js
+++ /usr/palm/applications/com.palm.app.email/app/controllers/compose-assistant.js
@@ -80,7 +80,9 @@
 				menuClass: 'palm-white',
 				items: [
 					{label:$L('Attach'), icon:'attach', command:'attach'},
-					{label:$L('Send'), icon:'send', command:'send'}
+					{label:$L('Priority'), icon:'priority', command:'priority'},
+					{label:$L('Delete'),icon:'delete', command:'cancel'},
+					{label:('Send'), icon:'send', command:'send'}
 				]};
 
 		this.draftIsDirty = false;
