diff --git a/usr/palm/applications/com.palm.app.email/mail/source/Body.js b/usr/palm/applications/com.palm.app.email/mail/source/Body.js
index 659e270..48d33ca 100644
--- a/usr/palm/applications/com.palm.app.email/mail/source/Body.js
+++ b/usr/palm/applications/com.palm.app.email/mail/source/Body.js
@@ -210,7 +210,7 @@ enyo.kind({
 			// {kind: "Control", className:"toolbar-spacer"},
 			{name: "deleteCommandItem", icon: "../images/icons/toolbar-icon-delete.png",  onclick: "deleteClick"},
 			{kind: "Control", className:"toolbar-spacer"},
-			{kind:enyo.VFlexBox, name:"navgroup", style:"position:absolute;right:14px;margin:6px 0;", showing:false, width:"118px", components: [ 
+			{kind: "Control", className:"toolbar-spacer", width:"75px"},{kind:enyo.VFlexBox, name:"navgroup", style:"position:absolute;right:6px;margin:6px 0;", showing:true, width:"118px", components: [ 
 				{kind: "RadioGroup", name: "radioSelection", components: [
 					{icon: "../images/previous-email-icon.png", disabled:true, name:"prevButton", onclick: "goToPrevEmail"}, 
 					{icon: "../images/next-email-icon.png", disabled:true, name: "nextButton", onclick: "goToNextEmail"} 
diff --git a/usr/palm/applications/com.palm.app.email/mail/source/MailApp.js b/usr/palm/applications/com.palm.app.email/mail/source/MailApp.js
index afa3ae3..e72cb5e 100644
--- a/usr/palm/applications/com.palm.app.email/mail/source/MailApp.js
+++ b/usr/palm/applications/com.palm.app.email/mail/source/MailApp.js
@@ -174,7 +174,7 @@ enyo.kind({
 				var accountId = currentFolder.accountId;
 				enyo.application.dashboardManager.setFilter(accountId, currentFolder._id);
 			}
-			this.$.body.hideNextPrev();
+			this.$.body.exposeNextPrev();
 		}
 	},
 	rendered: function() {
