diff --git a/usr/palm/applications/com.palm.app.email/app/controllers/message-assistant.js b/usr/palm/applications/com.palm.app.email/app/controllers/message-assistant.js
index 7e52e36..9e83ccf 100644
--- a/usr/palm/applications/com.palm.app.email/app/controllers/message-assistant.js
+++ b/usr/palm/applications/com.palm.app.email/app/controllers/message-assistant.js
@@ -135,12 +135,16 @@ var MessageAssistant = function(params, focusStage, detailsObj){
 				visible:true,
 				items: [
 					{label:$L('Reply'),     icon:'reply',     command:'reply'},
-					{label:$L('Reply all'), icon:'reply-all', command:'replyAll'},
-					{label:$L('Forward'),   icon:'forward-email', command:'forward'},
+					//{label:$L('Reply all'), icon:'reply-all', command:'replyAll'},
+					//{label:$L('Forward'),   icon:'forward-email', command:'forward'},
+					{label:$L('Move to Folder...'),   icon:'file', command:'move', disabled: this.readonly},
+					{label:MessageAssistant.kAppMenuMarkUnread, icon:'priority',  command:'mark-unread', disabled: this.readonly},
 					{label:$L('Delete'),    icon:'delete',    command:'delete', disabled: this.readonly}
 				]};
 		this.controller.setupWidget(Mojo.Menu.commandMenu, undefined, this.cmdMenuModel);
 		
+		this.replyAllMenuItem =     {label:$L('Reply all'), shortcut:'r', command:'replyAll'};
+		this.forwardMenuItem =     {label:$L('Forward'), shortcut:'f', command:'forward'};
 		this.markUnreadMenuItem =     {label:MessageAssistant.kAppMenuMarkUnread, shortcut:'k', command:'mark-unread', disabled: this.readonly};
 		this.markSetFlagMenuItem =    {label:MessageAssistant.kAppMenuSetFlag, shortcut:'g', command:'flag', disabled: this.readonly};
 		this.showRecipientsMenuItem = {label:MessageAssistant.kAppMenuShowRecipients, shortcut:'s', command:'show-recipients'};
@@ -157,9 +161,11 @@ var MessageAssistant = function(params, focusStage, detailsObj){
 			label: $L('Chat view menu'),
 			items: [
 				this.kEmailEditItems,
+				this.replyAllMenuItem,
+				this.forwardMenuItem,
 				this.markUnreadMenuItem,
 				this.markSetFlagMenuItem,
-				this.moveToFolderMenuItem,
+				//this.moveToFolderMenuItem,
 				this.showRecipientsMenuItem,
 				//this.openInNewCardMenuItem,
 				Mojo.Menu.prefsItem,
