diff --git a/usr/palm/applications/com.palm.app.email/app/controllers/compose-assistant.js b/usr/palm/applications/com.palm.app.email/app/controllers/compose-assistant.js
index e4f2e91..e53f7ea 100644
--- a/usr/palm/applications/com.palm.app.email/app/controllers/compose-assistant.js
+++ b/usr/palm/applications/com.palm.app.email/app/controllers/compose-assistant.js
@@ -172,8 +172,10 @@ var ComposeAssistant = Class.create({
 				visible: true,
 				menuClass: 'palm-white',
 				items: [
+					{label:$L('Priority'), icon:'priority', command:'priority'},
 					{label:$L('Attach'), icon:'attach', command:'attach'},
-					{label:$L('Send'), icon:'send', command:'send'}
+					{label:('Send'), icon:'send', command:'send'},
+					{label:$L('Delete'),icon:'delete', command:'cancel'}
 				]};
 
 		this.appMenuSaveIndex = 2;
@@ -618,7 +620,15 @@ var ComposeAssistant = Class.create({
 					break;
 
 				case 'cancel':
-					this.cancelCompose();
+					this.controller.showAlertDialog({
+					onChoose: function(value) {if (value == 'yes') {this.cancelCompose();}},
+					title: $L("Discard this Draft?"),
+					message: $L("This cannot be undone"),
+					choices:[
+						{label:$L('Yes'), value:"yes", type:'negative'},  
+						{label:$L("No"), value:"no", type: 'dismiss'}
+							]
+					});
 					break;
 						
 				case 'priority':
