diff --git a/usr/palm/applications/com.palm.app.email/app/controllers/notification-assistant.js b/usr/palm/applications/com.palm.app.email/app/controllers/notification-assistant.js
index 11a6861..4ea6190 100644
--- a/usr/palm/applications/com.palm.app.email/app/controllers/notification-assistant.js
+++ b/usr/palm/applications/com.palm.app.email/app/controllers/notification-assistant.js
@@ -169,7 +169,12 @@ var NotificationAssistant = Class.create({
 					// priority and trumps all other sound types.
 					if (folderObj.hasNew === true && 
 					    folderObj.playSound !== AccountpreferencesAssistant.kNotifyTypeNone &&
-					    playSound !== AccountpreferencesAssistant.kNotifyTypeAudio) {
+					    playSound !== AccountpreferencesAssistant.kNotifyTypeAudio)
+							this.serviceRequest = new Mojo.Service.Request					
+							("palm://com.palm.vibrate", {
+							method: 'vibrate', parameters: { 'period': 0,'duration': 1500 }
+							});
+{
 
 					    playSound = folderObj.playSound;
 					}
