diff --git a/usr/palm/applications/com.palm.app.email/data/AccountPreferences.js b/usr/palm/applications/com.palm.app.email/data/AccountPreferences.js
index 8827850..a451d9f 100644
--- a/usr/palm/applications/com.palm.app.email/data/AccountPreferences.js
+++ b/usr/palm/applications/com.palm.app.email/data/AccountPreferences.js
@@ -148,6 +148,11 @@ AccountPreferences.LOOKBACK_3_DAYS 	= 3;
 AccountPreferences.LOOKBACK_7_DAYS 	= 7;
 AccountPreferences.LOOKBACK_2_WEEKS = 14;
 AccountPreferences.LOOKBACK_1_MONTH = 30;
+AccountPreferences.LOOKBACK_2_MONTHS = 60;
+AccountPreferences.LOOKBACK_3_MONTHS = 90;
+AccountPreferences.LOOKBACK_6_MONTHS = 180;
+AccountPreferences.LOOKBACK_9_MONTHS = 270;
+AccountPreferences.LOOKBACK_12_MONTHS = 365;
 AccountPreferences.LOOKBACK_FOREVER = 0;
 
 AccountPreferences.NO_SIGNATURE_BY_USER = '<div id="no_signature" style="overflow:hidden;"></div>';
diff --git a/usr/palm/applications/com.palm.app.email/mail/source/settings/AccountSettings.js b/usr/palm/applications/com.palm.app.email/mail/source/settings/AccountSettings.js
index df977dd..ed19af0 100644
--- a/usr/palm/applications/com.palm.app.email/mail/source/settings/AccountSettings.js
+++ b/usr/palm/applications/com.palm.app.email/mail/source/settings/AccountSettings.js
@@ -126,6 +126,11 @@ enyo.kind({
 									 {caption: $L("7 days"), value: AccountPreferences.LOOKBACK_7_DAYS},
 									 {caption: $L("2 weeks"), value: AccountPreferences.LOOKBACK_2_WEEKS},
 									 {caption: $L("1 month"), value: AccountPreferences.LOOKBACK_1_MONTH},
+									 {caption: $L("2 months"), value: AccountPreferences.LOOKBACK_2_MONTHS},
+									 {caption: $L("3 months"), value: AccountPreferences.LOOKBACK_3_MONTHS},	
+ 									 {caption: $L("6 months"), value: AccountPreferences.LOOKBACK_6_MONTHS}, 
+									 {caption: $L("9 months"), value: AccountPreferences.LOOKBACK_9_MONTHS},		
+				 					 {caption: $L("12 months"), value: AccountPreferences.LOOKBACK_12_MONTHS},
 									 {caption: $L("All"), value: AccountPreferences.LOOKBACK_FOREVER}
 							    ],
 							onChange: "valueChanged"
