diff --git a/usr/palm/applications/com.palm.app.email/app/controllers/list-assistant.js b/usr/palm/applications/com.palm.app.email/app/controllers/list-assistant.js
index 74a2327..a3a8f6b 100644
--- a/usr/palm/applications/com.palm.app.email/app/controllers/list-assistant.js
+++ b/usr/palm/applications/com.palm.app.email/app/controllers/list-assistant.js
@@ -53,7 +53,7 @@ var ListAssistant = Class.create(App.Scene, {
 					itemTemplate: 'list/email_entry',
 					itemsCallback: this.dataSource.updateOffsetAndLimit.bind(this.dataSource),
 					swipeToDelete: true,
-					autoconfirmDelete: true,
+					autoconfirmDelete: false,
 					uniquenessProperty: 'id',
 					dividerTemplate: 'list/date_separator',
 					dividerFunction: function(email) { return email.dateCategory; }
