diff --git a/usr/palm/applications/com.palm.app.email/app/controllers/accountpreferences-assistant.js b/usr/palm/applications/com.palm.app.email/app/controllers/accountpreferences-assistant.js
index 15f6bac..f0a960a 100644
--- a/usr/palm/applications/com.palm.app.email/app/controllers/accountpreferences-assistant.js
+++ b/usr/palm/applications/com.palm.app.email/app/controllers/accountpreferences-assistant.js
@@ -522,6 +522,11 @@ AccountpreferencesAssistant.prototype.syncWindowOptions = {
 		{label: $L('7 days'), value: 7},
 		{label: $L('2 weeks'), value: 14},
 		{label: $L('1 month'), value: 30},
+		{label: $L('2 months'), value: 60},
+		{label: $L('3 months'), value: 90},
+		{label: $L('6 months'), value: 180},
+		{label: $L('9 months'), value: 270},
+		{label: $L('1 year'), value: 365},
 		{label: $L('All'), value: "0"}
 	]
 };
@@ -562,8 +567,13 @@ AccountpreferencesAssistant.prototype.fetchOptions = {
 		{label: $L('10 minutes'), value: AccountPreferences.SYNC_10_MINS + ""},
 		{label: $L('15 minutes'), value: AccountPreferences.SYNC_15_MINS + ""},
 		{label: $L('30 minutes'), value: AccountPreferences.SYNC_30_MINS + ""},
+		{label: $L('45 minutes'), value: AccountPreferences.SYNC_45_MINS + ""},
 		{label: $L('1 hour'), value: AccountPreferences.SYNC_1_HR + ""},
+		{label: $L('2 hours'), value: AccountPreferences.SYNC_2_HRS + ""},
+		{label: $L('4 hours'), value: AccountPreferences.SYNC_4_HRS + ""},
 		{label: $L('6 hours'), value: AccountPreferences.SYNC_6_HRS + ""},
+		{label: $L('8 hours'), value: AccountPreferences.SYNC_8_HRS + ""},
+		{label: $L('10 hours'), value: AccountPreferences.SYNC_10_HRS + ""},
 		{label: $L('12 hours'), value: AccountPreferences.SYNC_12_HRS + ""},
 		{label: $L('24 hours'), value: AccountPreferences.SYNC_24_HRS + ""},
 		{label: $L('Manual'), value: AccountPreferences.SYNC_MANUAL + ""}
