diff --git a/usr/palm/applications/com.palm.app.email/app/models/Email.js b/usr/palm/applications/com.palm.app.email/app/models/Email.js
index a42803b..d7ba7e7 100644
--- a/usr/palm/applications/com.palm.app.email/app/models/Email.js
+++ b/usr/palm/applications/com.palm.app.email/app/models/Email.js
@@ -438,7 +438,7 @@ Email.setReplied = function(messageId, value) {
 }
 
 Email.kSignaturePlaceholder = "<br><br><span id='signature'></span>";
-Email.kReplyForwardSetupElems = "<span style='color:navy'>" + Email.kSignaturePlaceholder + "<hr align='left' style='width:75%'/>";
+Email.kReplyForwardSetupElems = "<span style='color:black'>" + Email.kSignaturePlaceholder + "<hr align='left' style='width:75%'/>";
 
 Email.kSubjectPrefixRe = $L("Re: ");
 Email.kSubjectPrefixFw = $L("Fw: ");
