diff --git a/usr/palm/applications/com.palm.app.email/app/models/EmailAppDepot.js b/usr/palm/applications/com.palm.app.email/app/models/EmailAppDepot.js
index b2b1cd7..4e4e29b 100644
--- a/usr/palm/applications/com.palm.app.email/app/models/EmailAppDepot.js
+++ b/usr/palm/applications/com.palm.app.email/app/models/EmailAppDepot.js
@@ -47,7 +47,7 @@ var EmailAppDepot = Class.create({
 		if (accounts.list !== undefined && this.numAccounts !== accounts.list.length) {
 			// look for accounts that have been removed
 			var removedAccounts = [];
-			this.accounts.list.each(function(existingAcct){
+			accounts.list.each(function(existingAcct){
 				var stillExists = false;
 				accounts.list.each(function(newAcct) {
 					if (existingAcct.originalLogin === newAcct.originalLogin) {
