--- .orig/etc/palm/mojodb.conf
+++ /etc/palm/mojodb.conf
@@ -9,6 +9,7 @@
 	},	
 	"db" : {
 		"permissions" : [
+			{"type":"db.role","object":"admin","caller":"com.palm.systemui","operations":{"*":"allow"}},
 			{"type":"db.role","object":"admin","caller":"com.palm.configurator","operations":{"*":"allow"}},
 			{"type":"db.role","object":"admin","caller":"com.palm.service.backup","operations":{"*":"allow"}},
 			{"type":"db.role","object":"admin","caller":"com.palm.odd.service","operations":{"*":"allow"}},
--- .orig/usr/palm/frameworks/mojo2/submissions/98/stylesheets/global-menus.css
+++ /usr/palm/frameworks/mojo2/submissions/98/stylesheets/global-menus.css
@@ -864,3 +864,49 @@
   font-size: 16px;
   padding: 25px 17px 0px;
 }
+
+/* Make today menu items with smaller font and lower row height */
+
+#palm-today-menu #menu_items .palm-row,
+#palm-today-menu #edit_items .palm-row {
+	line-height: 42px;
+	min-height: 46px;
+	max-height: 46px;
+	margin:0px;
+	padding:0px;
+}
+
+#palm-today-menu #menu_items .palm-row .title,
+#palm-today-menu #edit_items .palm-row .title {
+	line-height: 42px;
+	min-height: 46px;
+	font-size: 18px;
+	margin: 0px 0px 0px 16px;
+	padding: 0px 0px 0px 0px;
+}
+
+#palm-today-menu #menu_items .palm-row .label,
+#palm-today-menu #edit_items .palm-row .label {
+	line-height: 42px;
+	min-height: 46px;
+	margin: 0px 16px 0px 0px;
+	padding: 0px 0px 0px 5px;
+}
+
+/* HACK: could not find a way to limit height of the selected row */
+
+#palm-today-menu #menu_items .palm-row.selected,
+#palm-today-menu #edit_items .palm-row.selected {
+	position: relative;
+	top:-2px;
+}
+#palm-today-menu #menu_items .palm-row.selected .title,
+#palm-today-menu #edit_items .palm-row.selected .title {
+	padding: 2px 0px 0px 0px;
+}
+
+#palm-today-menu #menu_items .palm-row.selected .label,
+#palm-today-menu #edit_items .palm-row.selected .label {
+	padding: 2px 0px 0px 5px;
+}
+
--- .orig/usr/lib/luna/system/luna-systemui/sources.json
+++ /usr/lib/luna/system/luna-systemui/sources.json
@@ -9,6 +9,9 @@
     "source": "app\/controllers\/devicemenu-assistant.js"
   },  
   {
+    "source": "app\/controllers\/todaymenu-assistant.js"
+  },  
+  {
     "source": "app\/controllers\/pbapalert-assistant.js"
   },
   {
@@ -120,6 +123,9 @@
     "source": "app\/models\/local-thumbs-model.js"
   }, 
   {
+    "source": "javascripts\/DateWeek.js"
+  },
+  {
     "source": "javascripts\/DeviceClass.js"
   },
   {
--- .orig/usr/lib/luna/system/luna-systemui/resources/es/strings.json
+++ /usr/lib/luna/system/luna-systemui/resources/es/strings.json
@@ -283,6 +283,11 @@
 	"Your phone will restart in #{counter} second. After restart, sign in to your Palm Profile to restore backed up data.": "El teléfono se reiniciará en #{counter} segundo. Después de reiniciar, inicia sesión en tu perfil de Palm para restaurar los datos de la copia de seguridad.",
 	"Your phone will restart in #{counter} seconds. After restart, sign in to your Palm Profile to restore backed up data.": "El teléfono se reiniciará en #{counter} segundos. Después de reiniciar, inicia sesión en tu perfil de Palm para restaurar los datos de la copia de seguridad.",
 	"Your service account does not allow internet connection sharing. Contact #{carrier} for help.": "Tu cuenta de servicio no permite compartir la conexión de Internet. Comunícate con #{carrier} para obtener ayuda.",
+	"Menu Content Settings": "",
+	"Show Today's Events": "",
+	"Show Tomorrow's Events": "",
+	"Show Events for 3 Days": "",
+	"Show Events for 7 Days": "",
 	"Zero": "Cero",
 	"drag up to answer": "arrastrar arriba para responder",
 	"drag up to unlock": "arrastrar arriba para desbloquear",
