diff --git a/usr/palm/applications/com.palm.app.tasks/app/controllers/app-assistant.js b/usr/palm/applications/com.palm.app.tasks/app/controllers/app-assistant.js
index be6a6be..f8077ba 100644
--- a/usr/palm/applications/com.palm.app.tasks/app/controllers/app-assistant.js
+++ b/usr/palm/applications/com.palm.app.tasks/app/controllers/app-assistant.js
@@ -156,7 +156,7 @@ var AppAssistant = Class.create(
 			
 		this.instantiateActivityMgrSingleton();
 		this.instantiateDbSingleton();
-		this.instantiateDashboardSingleton();		
+		//this.instantiateDashboardSingleton();		
 		
 		activityManager.createBootActivity(); //reschedule self first
 		activityManager.createDBActivity(this.callSuccess, this.callFailure);	
@@ -196,7 +196,7 @@ var AppAssistant = Class.create(
 		
 		this.instantiateDbSingleton();
 		this.instantiateActivityMgrSingleton();
-		this.instantiateDashboardSingleton();
+		//this.instantiateDashboardSingleton();
 		
 		activityManager.rescheduleDBActivity();
 
@@ -213,7 +213,7 @@ var AppAssistant = Class.create(
 			
 		this.instantiateDbSingleton();
 		this.instantiateActivityMgrSingleton();
-		this.instantiateDashboardSingleton();
+		//this.instantiateDashboardSingleton();
 		
 		activityManager.rescheduleAMActivity();
 		activityManager.rescheduleDBActivity();	
@@ -235,7 +235,7 @@ var AppAssistant = Class.create(
 			params, 
 			function(){
 				this.instantiateDbSingleton();
-				this.instantiateDashboardSingleton();
+				//this.instantiateDashboardSingleton();
 				db.updateTotals(true);
 				activityManager.rescheduleDBActivity();
 			}.bind(this)
@@ -247,7 +247,7 @@ var AppAssistant = Class.create(
 
 		this.instantiateActivityMgrSingleton();			
 		this.instantiateDbSingleton();
-		this.instantiateDashboardSingleton();
+		//this.instantiateDashboardSingleton();
 
 		db.markNewSyncedTaskPositions(this.callSuccess, this.callFailure); 
 
@@ -272,7 +272,7 @@ var AppAssistant = Class.create(
 DEBUG_TIME("LAUNCHREGULAR BEFORE DBSERVICE, DASHBOARD, ACTIVITYMGR INSTANTIATION"); //TASK: REMOVE DEBUG TIMING		
 		this.instantiateActivityMgrSingleton();			
 		this.instantiateDbSingleton();
-		this.instantiateDashboardSingleton();
+		//this.instantiateDashboardSingleton();
 DEBUG_TIME("LAUNCHREGULAR AFTER INST"); //TASK: REMOVE DEBUG TIMING
 		db.markNewSyncedTaskPositions(this.callSuccess, this.callFailure); 
 DEBUG_TIME("LAUNCHREGULAR AFTER MARKNEWSYNCEDPOS"); //TASK: REMOVE DEBUG TIMING		
