diff --git a/usr/palm/applications/com.palm.app.tasks/app/controllers/tasks-assistant.js b/usr/palm/applications/com.palm.app.tasks/app/controllers/tasks-assistant.js
index 3071858..d244951 100644
--- a/usr/palm/applications/com.palm.app.tasks/app/controllers/tasks-assistant.js
+++ b/usr/palm/applications/com.palm.app.tasks/app/controllers/tasks-assistant.js
@@ -920,7 +920,7 @@ var TasksAssistant = Class.create({
 			this.listElement.mojo.insertItemsAt(index, [task]);
 			this.listElement.mojo.revealItem(index);
 			this.addWidgets(this.listElement.mojo.getNodeByIndex(index).querySelector('[name="subject"]'), task);
-			this.listElement.mojo.focusItem(task);
+			this.state.edit.call(this, index);
 			
 			Mojo.Log.info("New task %j %j", task, this.task);
 		}
@@ -942,7 +942,7 @@ var TasksAssistant = Class.create({
 				{
 					// If we find the node we just added, focus on it.  We might not find the node however if the list is being refreshed
 					self.addWidgets(node.querySelector('[name="subject"]'), task);
-					self.listElement.mojo.focusItem(task);
+					self.state.edit.call(self, index);
 				}
 				
 				Mojo.Log.info("New task %j %j", task, self.task);
