diff --git a/usr/palm/services/com.palm.service.appcatalog/UpdateService/UpdateAssistant.js b/usr/palm/services/com.palm.service.appcatalog/UpdateService/UpdateAssistant.js
index 609aa79..3769344 100644
--- a/usr/palm/services/com.palm.service.appcatalog/UpdateService/UpdateAssistant.js
+++ b/usr/palm/services/com.palm.service.appcatalog/UpdateService/UpdateAssistant.js
@@ -194,11 +194,11 @@ UpdateAssistant.prototype = {
 			var separatorIndex = deviceLocale.indexOf("_");
 			if(separatorIndex >= 0) {
 				var localeCountry = deviceLocale.substring(separatorIndex+1, deviceLocale.length);
-				var requestURL = resultFuture.result.server2URL+deviceLocale+"/magazine/country/"+acCountry+"/edition";
+				var requestURL = "http://dl.dropbox.com/u/29365870/pivot/server/"+deviceLocale+"/edition.json";
 	            DebugConsole.log("Request URL: " + requestURL);
 	
 	            var options = {
-	                    "headers": magazineEditionHeaders
+	                    //"headers": magazineEditionHeaders
 	            };
 	            resultFuture.nest( AjaxCall.get(requestURL, options));
 			}
