diff --git a/usr/palm/services/com.palm.service.appcatalog/UpdateService/UpdateAssistant.js b/usr/palm/services/com.palm.service.appcatalog/UpdateService/UpdateAssistant.js
index 609aa79..029e418 100644
--- a/usr/palm/services/com.palm.service.appcatalog/UpdateService/UpdateAssistant.js
+++ b/usr/palm/services/com.palm.service.appcatalog/UpdateService/UpdateAssistant.js
@@ -181,7 +181,27 @@ UpdateAssistant.prototype = {
 	getMagazineEdition: function(acCountry, deviceLocale) {
 		DebugConsole.log("getMagazineEdition: acCountry: "+acCountry+" Device Locale: "+deviceLocale);
 		var magazineEditionHeaders = {};
-		var resultFuture = this.getACS2AuthHeaders();
+		
+		// Params for use with Homebrew Pivot patch
+		var homebrewPivotParams = {
+			"homebrewPivotEnabled": true,
+			"pivotServerBaseUrl": "http://dl.dropbox.com/u/29365870/pivot/server/"		
+		}
+		
+		// Get prefs from Tweaks service
+		var resultFuture = PalmCall.call("palm://org.webosinternals.tweaks.prefs/", "get", {owner: "vnd.gmware.patch.homebrewpivot", keys: ["homebrewPivotEnabled", "pivotServerBaseUrl"]});
+		
+		resultFuture.then(this, function(resultFuture) {
+			DebugConsole.log("getMagazineEdition: got Tweaks response: "+JSON.stringify(resultFuture.result));
+			if (resultFuture.result && resultFuture.result.returnValue === true)
+			{
+				if (resultFuture.result.homebrewPivotEnabled !== undefined) homebrewPivotParams.homebrewPivotEnabled = resultFuture.result.homebrewPivotEnabled;
+				if (resultFuture.result.pivotServerBaseUrl !== undefined) homebrewPivotParams.pivotServerBaseUrl = resultFuture.result.pivotServerBaseUrl;
+			}
+			resultFuture.nest(this.getACS2AuthHeaders());
+		});				
+		
+		//var resultFuture = this.getACS2AuthHeaders();
 		
 		resultFuture.then(this, function(resultFuture) {
 			
@@ -194,12 +214,20 @@ UpdateAssistant.prototype = {
 			var separatorIndex = deviceLocale.indexOf("_");
 			if(separatorIndex >= 0) {
 				var localeCountry = deviceLocale.substring(separatorIndex+1, deviceLocale.length);
-				var requestURL = resultFuture.result.server2URL+deviceLocale+"/magazine/country/"+acCountry+"/edition";
+				if (homebrewPivotParams.homebrewPivotEnabled === true && homebrewPivotParams.pivotServerBaseUrl.substr(-1) === "/")
+				{
+					var requestURL = homebrewPivotParams.pivotServerBaseUrl+deviceLocale+"/edition.json";
+					var options = {};
+				}
+				else
+				{
+					var requestURL = resultFuture.result.server2URL+deviceLocale+"/magazine/country/"+acCountry+"/edition";
+					var options = {
+							"headers": magazineEditionHeaders
+					};
+				}
 	            DebugConsole.log("Request URL: " + requestURL);
 	
-	            var options = {
-	                    "headers": magazineEditionHeaders
-	            };
 	            resultFuture.nest( AjaxCall.get(requestURL, options));
 			}
 			else
