diff --git a/media/cryptofs/apps/usr/palm/applications/com.palm.app.enyo-findapps/main/build.js.rej b/media/cryptofs/apps/usr/palm/applications/com.palm.app.enyo-findapps/main/build.js.rej
new file mode 100644
index 0000000..92f0e98
--- /dev/null
+++ b/media/cryptofs/apps/usr/palm/applications/com.palm.app.enyo-findapps/main/build.js.rej
@@ -0,0 +1,10 @@
+--- media/cryptofs/apps/usr/palm/applications/com.palm.app.enyo-findapps/main/build.js
++++ media/cryptofs/apps/usr/palm/applications/com.palm.app.enyo-findapps/main/build.js
+@@ -13930,6 +13930,7 @@
+ components: [ {
+ name: "backButton",
+ kind: "Button",
++style: "width:275px",
+ label: $L("Back"),
+ className: "back-button",
+ onclick: "goBack"
diff --git a/media/cryptofs/apps/usr/palm/applications/com.palm.app.enyo-findapps/main/build.js.~1~ b/media/cryptofs/apps/usr/palm/applications/com.palm.app.enyo-findapps/main/build.js.~1~
new file mode 100644
index 0000000..e69de29
