diff --git a/usr/palm/applications/com.palm.app.findapps/app/controllers/details-assistant.js b/usr/palm/applications/com.palm.app.findapps/app/controllers/details-assistant.js
index 6bd3877..f40d767 100644
--- a/usr/palm/applications/com.palm.app.findapps/app/controllers/details-assistant.js
+++ b/usr/palm/applications/com.palm.app.findapps/app/controllers/details-assistant.js
@@ -99,7 +99,7 @@ var DetailsAssistant = Class.create(
 				
 		this._detailsPage = this.controller.get('detailsPage');
 
-		this.drawerModel = {open:false};
+		this.drawerModel = {open:true};
 		this.controller.setupWidget('detailsDrawer', {modelProperty:'open', unstyled:true}, this.drawerModel);
 		this.drawer = this.controller.get('detailsDrawer');
 		
