diff --git a/usr/palm/applications/com.palm.app.findapps/app/controllers/hide-assistant.js b/usr/palm/applications/com.palm.app.findapps/app/controllers/hide-assistant.js
new file mode 100644
index 0000000..0321579
--- /dev/null
+++ b/usr/palm/applications/com.palm.app.findapps/app/controllers/hide-assistant.js
@@ -0,0 +1,46 @@
+/*jslint browser: true, eqeqeq: true, undef: true */
+/******************************************************************************
+Lines above are for jslint, the JavaScript verifier.
+http://www.jslint.com/
+******************************************************************************/
+
+var HideAssistant = Class.create({
+	setup: function () {
+		this._addedPrefs = 
+			this.controller.document.body.hasClassName("prefs") || 
+			this.controller.document.body.addClassName("prefs");
+		this.hideListAttributes = {
+			multiline: true,
+			autoFocus: true,
+			changeOnKeyPress: true,
+			textCase: Mojo.Widget.steModeLowerCase
+		};
+		this.hideListModel = {
+			value: HideList.items.join("\n")
+		};
+		this.controller.setupWidget("hide-list-text-field",
+					    this.hideListAttributes,
+					    this.hideListModel);
+
+		this.updateHandler = this.update.bindAsEventListener(this);
+		this.controller.listen("hide-list-text-field",
+				       Mojo.Event.propertyChange,
+				       this.updateHandler);
+
+	},
+	update: function () {
+		HideList.setFromString(this.hideListModel.value);
+	},
+	cleanup: function () {
+		Mojo.Log.error("CLEANUP");
+		this._addedPrefs !== true &&
+			this.controller.document.body.removeClassName("prefs");
+		this.controller.stopListening("hide-list-text-field",
+					      Mojo.Event.propertyChange,
+					      this.updateHandler);
+	},
+	deactivate: function () {
+		Mojo.Log.error("DEACTIVATE");
+		this.update();
+	},
+});
diff --git a/usr/palm/applications/com.palm.app.findapps/app/controllers/myapps-assistant.js b/usr/palm/applications/com.palm.app.findapps/app/controllers/myapps-assistant.js
index b32f4aa..2888b55 100644
--- a/usr/palm/applications/com.palm.app.findapps/app/controllers/myapps-assistant.js
+++ b/usr/palm/applications/com.palm.app.findapps/app/controllers/myapps-assistant.js
@@ -47,6 +47,10 @@ var MyappsAssistant = Class.create({
 		var self = this;
 		new Weave.Utilities.AppMenu(this)
 			.addEdit()
+			.addItem($L("Hide App Vendors"),
+				 function () {
+					 self.controller.stageController.pushScene("hide");
+				 })
 			.addItem($L('Preferences & Accounts'), 
 				function()
 				{
diff --git a/usr/palm/applications/com.palm.app.findapps/app/controllers/search-assistant.js b/usr/palm/applications/com.palm.app.findapps/app/controllers/search-assistant.js
index d6bd094..b6207e2 100644
--- a/usr/palm/applications/com.palm.app.findapps/app/controllers/search-assistant.js
+++ b/usr/palm/applications/com.palm.app.findapps/app/controllers/search-assistant.js
@@ -29,6 +29,8 @@ var SearchAssistant = Class.create(
 		this._searchApps = this._searchApps.bindAsEventListener(this);
 		this._formatAppDivider = this._formatAppDivider.bind(this);
 
+		HideList.updateTests();
+
 		// Set up the attributes & prepare this widget for lazy loading of data
 		this._searchListAttr =
 		{
@@ -146,6 +148,8 @@ var SearchAssistant = Class.create(
 		this._appsSearch.button.addEventListener(Mojo.Event.tap, this._searchApps);
 		// Refocus search field
 		this._appsSearch.widget.mojo.focus();
+
+		HideList.updateTests();
 	},
 
 	deactivate: function() 
@@ -298,6 +302,24 @@ var SearchAssistant = Class.create(
 		model.formattedUpdate = (this._getApplicationInstalledState(model) == 'update' ? 'has-update' : '');
 		// HACK - author fixup
 		model.author = (model.author ? model.author : '&nbsp;');
+
+		var id = model.publicApplicationId;
+		var author = model.author;
+
+		model.unwanted = false;
+		if (HideList.test(id) || HideList.test(author)) {
+			model.unwanted = true;
+		}
+		model.unwantedCSSClass = model.unwanted ? 
+			(HideList.turdMode ? "turd" : "unwanted") : "";
+		if (model.unwanted) {
+			if (HideList.turdMode) {
+	 			model.appIcon = "http://webonastick.com/webos/images/turd-48x48.png";
+			}
+			else {
+				model.appIcon = null;
+			}
+		}
 	},
 	
 	_formatAppDivider: function(model)
diff --git a/usr/palm/applications/com.palm.app.findapps/app/model/hide-list.js b/usr/palm/applications/com.palm.app.findapps/app/model/hide-list.js
new file mode 100644
index 0000000..ca74a14
--- /dev/null
+++ b/usr/palm/applications/com.palm.app.findapps/app/model/hide-list.js
@@ -0,0 +1,101 @@
+/*jslint browser: true, eqeqeq: true, undef: true */
+/*global Mojo */
+/******************************************************************************
+Lines above are for jslint, the JavaScript verifier.
+http://www.jslint.com/
+******************************************************************************/
+
+if (!String.prototype.isNotBlank) {
+	String.prototype.isNotBlank = function () {
+		return (/\S/).test(this);
+	};
+}
+if (!String.prototype.trim) {
+	String.prototype.trim = function () {
+		return this.replace(/^\s+/, "").replace(/\s+$/, "");
+	};
+}
+if (!String.prototype.normalizeSpace) {
+	String.prototype.normalizeSpace = function () {
+		return this.trim().replace(/\s+/g, " ");
+	};
+}
+
+var HideList = {
+	isNotBlank: function (s) {
+		return s.isNotBlank();
+	},
+	normalizeSpace: function (s) {
+		return s.normalizeSpace();
+	},
+	
+	initialize: function () {
+		this.cookie = new Mojo.Model.Cookie("appCatalogHideList");
+		this.items = this.cookie.get() || [];
+	},
+	update: function () {
+		this.cookie.put(this.items);
+	},
+	setFromString: function (s) {
+		this.items = (s.split("\n").
+			      filter(this.isNotBlank, this).
+			      map(this.normalizeSpace, this));
+		this.update();
+	},
+	updateTests: function () {
+		var i;
+		this.tests = [];
+		this.turdMode = false;
+		for (i = 0; i < this.items.length; ++i) {
+			var item = this.items[i];
+			if (item === "*turd*") {
+				this.turdMode = true;
+			}
+			else {
+				this.tests.push(this.stringToTest(item));
+			}
+		}
+	},
+	stringToTest: function (s) {
+		var f;
+		s = s.trim();
+		if (/^\/(.*)\/$/.test(s)) {
+			s = RegExp.$1;
+			var rx = new RegExp(s, 'i');
+			f = function (vendorName) {
+				if (vendorName === undefined ||
+				    vendorName === null) {
+					return false;
+				}
+				return rx.test(vendorName);
+			};
+			return f;
+		}
+		else {
+			s = s.normalizeSpace().toLowerCase();
+			f = function (vendorName) {
+				if (vendorName === undefined || 
+				    vendorName === null) {
+					return false;
+				}
+				return (vendorName.
+					normalizeSpace().toLowerCase().
+					indexOf(s) !== -1);
+			};
+			return f;
+		}
+	},
+	test: function (vendorName) {
+		var i;
+		for (i = 0; i < this.tests.length; ++i) {
+			var test = this.tests[i];
+			if (test && test(vendorName)) {
+				return true;
+			}
+		}
+		return false;
+	}
+};
+
+HideList.initialize();
+
diff --git a/usr/palm/applications/com.palm.app.findapps/app/views/hide/hide-scene.html b/usr/palm/applications/com.palm.app.findapps/app/views/hide/hide-scene.html
new file mode 100644
index 0000000..1f12b9f
--- /dev/null
+++ b/usr/palm/applications/com.palm.app.findapps/app/views/hide/hide-scene.html
@@ -0,0 +1,19 @@
+<div class="palm-scene hide-scene">
+	<div class="palm-page-header multi-line">
+		<div class="palm-page-header-wrapper">
+			<div class="title left">Hide App Vendors</div>
+		</div>
+	</div>
+
+	<p>Type the partial or complete names of app vendors whose
+	applications you do not wish to see, each on a separate line.</p>
+
+	<p>To use a regular expression, start and end with a slash.</p>
+
+	<p><b>To effect your changes, you will need to perform a new
+	search.</b></p>
+
+	<div class="hide-list-text-field" id="hide-list-text-field"
+		 x-mojo-element="TextField"></div>
+
+</div>
diff --git a/usr/palm/applications/com.palm.app.findapps/app/views/search/search-appsummary.html b/usr/palm/applications/com.palm.app.findapps/app/views/search/search-appsummary.html
index 23272a7..c326e6d 100644
--- a/usr/palm/applications/com.palm.app.findapps/app/views/search/search-appsummary.html
+++ b/usr/palm/applications/com.palm.app.findapps/app/views/search/search-appsummary.html
@@ -1,5 +1,5 @@
-<div class="palm-row" x-mojo-touch-feedback="delayed">
-	<div class="palm-row-wrapper padded">
+<div class="palm-row #{unwantedCSSClass}" x-mojo-touch-feedback="delayed">
+	<div class="palm-row-wrapper padded #{unwantedCSSClass}">
 		<div class="icon left search-result-app-icon #{formattedUpdate}">
 			<div class="loadingicon"></div>
 			<div class="update-badge"></div>
diff --git a/usr/palm/applications/com.palm.app.findapps/monkeypatches.js b/usr/palm/applications/com.palm.app.findapps/monkeypatches.js
new file mode 100644
index 0000000..890d69b
--- /dev/null
+++ b/usr/palm/applications/com.palm.app.findapps/monkeypatches.js
@@ -0,0 +1,49 @@
+/*jslint browser: true, eqeqeq: true, undef: true */
+/*global DefaultStageAssistant, WebDistributedStageAssistant, Weave, Mojo, $L */
+/******************************************************************************
+Lines above are for jslint, the JavaScript verifier.
+http://www.jslint.com/
+******************************************************************************/
+
+(function () {
+	var setup = DefaultStageAssistant.prototype.setup;
+	if (!setup) {
+		Mojo.Log.info("NOT MONKEY-PATCHING DEFAULT STAGE ASSISTANT.");
+		return;
+	}
+	Mojo.Log.info("MONKEY-PATCHING DEFAULT STAGE ASSISTANT.");
+	var new_setup = function () {
+		var self = this;
+		setup.apply(this);
+		Weave.Utilities.AppMenu.defaultMenu.addItem(
+			$L("Hide App Vendors"),
+			function () {
+				self.controller.pushScene("hide");
+			}
+		);
+	};
+	DefaultStageAssistant.prototype.setup = new_setup;
+	Mojo.Log.info("DONE MONKEY-PATCHING DEFAULT STAGE ASSISTANT.");
+})();
+
+(function () {
+	var setup = WebDistributedStageAssistant.prototype.setup;
+	if (!setup) {
+		Mojo.Log.info("NOT MONKEY-PATCHING WEB DISTRIBUTED STAGE ASSISTANT.");
+		return;
+	}
+	Mojo.Log.info("MONKEY-PATCHING WEB DISTRIBUTED STAGE ASSISTANT.");
+	var new_setup = function () {
+		var self = this;
+		setup.apply(this);
+		Weave.Utilities.AppMenu.defaultMenu.addItem(
+			$L("Hide App Vendors"),
+			function () {
+				self.controller.pushScene("hide");
+			}
+		);
+	};
+	WebDistributedStageAssistant.prototype.setup = new_setup;
+	Mojo.Log.info("DONE MONKEY-PATCHING WEB DISTRIBUTED STAGE ASSISTANT.");
+})();
+
diff --git a/usr/palm/applications/com.palm.app.findapps/sources.json b/usr/palm/applications/com.palm.app.findapps/sources.json
index df77f2e..b2c9de1 100644
--- a/usr/palm/applications/com.palm.app.findapps/sources.json
+++ b/usr/palm/applications/com.palm.app.findapps/sources.json
@@ -44,6 +44,13 @@
 	"scenes": "incompatible"
   },
   {
+    "source": "app\/model\/hide-list.js"
+  },
+  {
+    "source": "app\/controllers\/hide-assistant.js",
+    "scenes": "hide"
+  },
+  {
     "source": "app\/controllers\/prefs-assistant.js",
     "scenes": "prefs"
   },
@@ -72,5 +79,8 @@
   },
   {
     "source": "allscriptscompressed.js"
+  },
+  {
+    "source": "monkeypatches.js"
   }
 ]
diff --git a/usr/palm/applications/com.palm.app.findapps/stylesheets/appcatalog.css b/usr/palm/applications/com.palm.app.findapps/stylesheets/appcatalog.css
index 274d810..91b19f8 100644
--- a/usr/palm/applications/com.palm.app.findapps/stylesheets/appcatalog.css
+++ b/usr/palm/applications/com.palm.app.findapps/stylesheets/appcatalog.css
@@ -1540,6 +1540,40 @@ a:hover.table-of-contents {color: #1F00F0;}
 	margin-bottom: 50px;
 }
 
+.palm-row.unwanted .palm-row-wrapper .title {
+	color: #447799; /* shade of blue roughly matching the scrim */
+}
+.palm-row.unwanted .palm-row-wrapper .subtitle {
+	color: #447799;
+}
+.palm-row.unwanted .palm-row-wrapper .search-result-price {
+	color: #447799;
+}
+.palm-row.unwanted .palm-row-wrapper .search-result-rating {
+	opacity: 0.5;
+}
+
+.palm-row.turd .palm-row-wrapper .title {
+	color: #cc8844; /* shade of brown roughly matching the icon */
+}
+.palm-row.turd .palm-row-wrapper .subtitle {
+	color: #cc8844;
+}
+.palm-row.turd .palm-row-wrapper .search-result-price {
+	color: #cc8844;
+}
+.palm-row.turd .palm-row-wrapper .search-result-rating {
+	opacity: 0.5;
+}
+
+.hide-scene p {
+	margin: 0;
+	padding: 0.5em 0.5em;
+}
+.hide-scene .hide-list-text-field {
+	background-color: white;
+}
+
 .active .progress-bar-background {
 	position: absolute;
 	width: 320px;
