diff --git a/usr/palm/applications/com.palm.app.maps/appinfo.json.rej b/usr/palm/applications/com.palm.app.maps/appinfo.json.rej
new file mode 100644
index 0000000..8260fb9
--- /dev/null
+++ b/usr/palm/applications/com.palm.app.maps/appinfo.json.rej
@@ -0,0 +1,11 @@
+--- usr/palm/applications/com.palm.app.maps/appinfo.json
++++ usr/palm/applications/com.palm.app.maps/appinfo.json
+@@ -2,7 +2,7 @@
+ 	"title": "Google Maps",
+ 	"type": "web",
+ 	"main": "index.html",
+-	"id": "com.palm.app.maps",
++	"id": "com.palm.maps",
+ 	"icon": "icon.png",
+ 	"splashicon": "icon-256x256.png",
+ 	"version": "1.0.0",
diff --git a/usr/palm/applications/com.palm.app.maps/appinfo.json.~1~ b/usr/palm/applications/com.palm.app.maps/appinfo.json.~1~
new file mode 100644
index 0000000..e69de29
