diff --git a/usr/palm/applications/com.palm.app.maps/app/assistants/app-assistant.js b/usr/palm/applications/com.palm.app.maps/app/assistants/app-assistant.js
index f5c81a6..373da6f 100644
--- a/usr/palm/applications/com.palm.app.maps/app/assistants/app-assistant.js
+++ b/usr/palm/applications/com.palm.app.maps/app/assistants/app-assistant.js
@@ -1,6 +1,6 @@
 AppAssistant.url = "";
 AppAssistant.webViewWidget = undefined;
-AppAssistant.googleMapsURL = "http://maps.google.com/maps/m";
+AppAssistant.googleMapsURL = "http://maps.google.com/maps/m?hl=iw";
 //AppAssistant.googleMapsURL = "http://gjm.google.com/mwm?sky=tfb";
 AppAssistant.urlClientParam = "client=ms-mwm-palm";
 
