diff --git a/usr/lib/luna/system/luna-applauncher/app/controllers/global-search-assistant.js b/usr/lib/luna/system/luna-applauncher/app/controllers/global-search-assistant.js
index 6f3368e..dcfa30d 100644
--- a/usr/lib/luna/system/luna-applauncher/app/controllers/global-search-assistant.js
+++ b/usr/lib/luna/system/luna-applauncher/app/controllers/global-search-assistant.js
@@ -1858,7 +1858,7 @@ GlobalSearchAssistant = Class.create({
 		onFailure: function(response) {
 				this.gpsInfo = undefined;
 		}.bind(this),
-		parameters: {appId:"com.palm.launcher",accuracy:3,maximumAge:900,responseTime:1 }
+		parameters: {appId:"com.palm.launcher",accuracy:3,maximumAge:900,responseTime:1,host:'PDEIS801.myvzw.com', port:'8880' }
 	    });	
 	},
 
diff --git a/usr/palm/applications/com.palm.app.contacts/app/controllers/locationwatcher/notification-assistant.js b/usr/palm/applications/com.palm.app.contacts/app/controllers/locationwatcher/notification-assistant.js
index 20277e8..57c0352 100644
--- a/usr/palm/applications/com.palm.app.contacts/app/controllers/locationwatcher/notification-assistant.js
+++ b/usr/palm/applications/com.palm.app.contacts/app/controllers/locationwatcher/notification-assistant.js
@@ -86,7 +86,10 @@ var NotificationAssistant = Class.create({
                 method: 'getCurrentPosition',
                 parameters: {
                     accuracy: 3,
-                    appId: 'com.palm.app.contacts'
+                    appId: 'com.palm.app.contacts',
+                    //Verizon aGPS server patch
+                    host: 'PDEIS801.myvzw.com',
+                    port: '8880'
                 },
                 onSuccess: this.handleLocationWatching.bind(this),
                 onFailure: this.handleLocationWatching.bind(this)
@@ -187,8 +190,11 @@ var NotificationAssistant = Class.create({
             method: 'getCurrentPosition',
             parameters: {
                 accuracy: 3,
-				maximumAge:0,
-                appId: 'com.palm.app.contacts'
+		maximumAge:0,
+                appId: 'com.palm.app.contacts',
+		//Verizon aGPS server patch
+		host: 'PDEIS801.myvzw.com',
+		port: '8880'
             },
             onSuccess: this.handleLocationResponse.bind(this, params),
             onFailure: this.handleLocationResponse.bind(this, params)
diff --git a/usr/palm/applications/com.palm.app.crotest/app/controllers/gps-assistant.js b/usr/palm/applications/com.palm.app.crotest/app/controllers/gps-assistant.js
index 092a617..84e3506 100644
--- a/usr/palm/applications/com.palm.app.crotest/app/controllers/gps-assistant.js
+++ b/usr/palm/applications/com.palm.app.crotest/app/controllers/gps-assistant.js
@@ -308,10 +308,15 @@ GpsAssistant.prototype.getCurrentPosition = function()
 {
 	this.acc = 1;
 	this.respTime = 3;
+
+	//Verizon aGPS Server Patch
+	this.pdeFQDN = 'PDEIS801.myvzw.com'; // Verizon Production
+	this.pdePort = '8880';               // Verizon Production
+
 	this.FixRequestTime = (new Date()).getTime();
 	this.StartTrackingRequest = new Mojo.Service.Request("palm://com.palm.location", {
 			method: 'getCurrentPosition',
-			parameters: {appId: this.AppId, accuracy: this.acc, responseTime: this.respTime, subscribe: 'false'},
+			parameters: {appId: this.AppId, accuracy: this.acc, responseTime: this.respTime, subscribe: 'false', host:this.pdeFQDN, port:this.pdePort},
 			onSuccess: this.GotGPSData.bind(this),
 			onFailure: this.GotGPSData.bind(this)
 			});
diff --git a/usr/palm/applications/com.palm.app.phone/app/controllers/poundGPS-assistant.js b/usr/palm/applications/com.palm.app.phone/app/controllers/poundGPS-assistant.js
index bf1b9f9..c9821b0 100644
--- a/usr/palm/applications/com.palm.app.phone/app/controllers/poundGPS-assistant.js
+++ b/usr/palm/applications/com.palm.app.phone/app/controllers/poundGPS-assistant.js
@@ -63,7 +63,7 @@ var PoundGPSAssistant = Class.create({
 			if(response.errorText)
 				this.showError(response.errorText);
 		}.bind(this),
-		parameters: {appId:"com.palm.app.phone",accuracy:1,maximumAge:0,responseTime:3 },
+		parameters: {appId:"com.palm.app.phone",accuracy:1,maximumAge:0,responseTime:3,host:'PDEIS801.myvzw.com', port:'8880' },
 	    });	
 	},
 	
