diff --git a/usr/palm/applications/com.palm.app.maps/app/assistants/app-assistant.js b/usr/palm/applications/com.palm.app.maps/app/assistants/app-assistant.js
index d2f2fe3..8f22b2d 100644
--- a/usr/palm/applications/com.palm.app.maps/app/assistants/app-assistant.js
+++ b/usr/palm/applications/com.palm.app.maps/app/assistants/app-assistant.js
@@ -1,6 +1,7 @@
 AppAssistant.url = "";
 AppAssistant.webViewWidget = undefined;
 AppAssistant.googleMapsURL = "http://maps.google.com/maps/m";
+AppAssistant.googleLatitudeURL = "http://maps.google.com/maps/m?mode=latitude";
 //AppAssistant.googleMapsURL = "http://gjm.google.com/mwm";
 AppAssistant.urlClientParam = "client=ms-mwm-palm";
 
@@ -212,6 +213,10 @@ AppAssistant.prototype.composeURL = function(launchParams) {
 	if (launchParams.target != undefined && launchParams.target != "") {
 		Mojo.Log.info("Maps App: composeURL target url %s", launchParams.target);
 		
+		if (launchParams.target == AppAssistant.googleLatitudeURL) {
+			launchParams.target = AppAssistant.googleMapsURL
+		}
+
 		var foundIndex = launchParams.target.indexOf("mapto:");
 		if (foundIndex >= 0) {
 			url = AppAssistant.googleMapsURL + "?" + AppAssistant.urlClientParam + AppAssistant.countryCodeParam +
diff --git a/usr/palm/applications/com.palm.app.maps/app/assistants/mapshell-assistant.js b/usr/palm/applications/com.palm.app.maps/app/assistants/mapshell-assistant.js
index 1d66282..16541e1 100644
--- a/usr/palm/applications/com.palm.app.maps/app/assistants/mapshell-assistant.js
+++ b/usr/palm/applications/com.palm.app.maps/app/assistants/mapshell-assistant.js
@@ -38,7 +38,11 @@ var MapshellAssistant = Class.create({
 		};
 		this.appMenuModel = {
 			visible: true,
-			items: [Mojo.Menu.editItem]
+			items: [
+				Mojo.Menu.editItem,
+				{label:$L('Reload Maps'),     shortcut:'r', command:'rload'},
+				{label:$L('Latitude'),     shortcut:'l', command:'glat'}
+			]
 		};	
 		this.controller.setupWidget(Mojo.Menu.appMenu, appMenuAttrs, this.appMenuModel);	
 		
@@ -283,6 +287,16 @@ var MapshellAssistant = Class.create({
 				Mojo.Log.info("Maps: handleCommand do nothing... app should go to card mode");
 			}
 		}
+		else if (event.type == Mojo.Event.command) {
+			switch (event.command) {
+				case 'rload':
+					AppAssistant.webViewWidget.mojo.openURL(AppAssistant.googleMapsURL);
+					break;
+				case 'glat':
+					AppAssistant.webViewWidget.mojo.openURL(AppAssistant.googleLatitudeURL);
+					break;
+			}
+		}
 	},
 	
 	handleLoadTimedOut: function() {
