diff --git a/usr/palm/applications/com.palm.app.maps/app/assistants/app-assistant.js b/usr/palm/applications/com.palm.app.maps/app/assistants/app-assistant.js
index f5c81a6..4b8e353 100644
--- a/usr/palm/applications/com.palm.app.maps/app/assistants/app-assistant.js
+++ b/usr/palm/applications/com.palm.app.maps/app/assistants/app-assistant.js
@@ -3,6 +3,7 @@ AppAssistant.webViewWidget = undefined;
 AppAssistant.googleMapsURL = "http://maps.google.com/maps/m";
 //AppAssistant.googleMapsURL = "http://gjm.google.com/mwm?sky=tfb";
 AppAssistant.urlClientParam = "client=ms-mwm-palm";
+AppAssistant.gpsWasEnabled = false;
 
 function AppAssistant(appController) {
 	Mojo.Log.info("MAPS App: AppAssistant");
@@ -22,6 +23,21 @@ function AppAssistant(appController) {
 	window.document.addEventListener(Mojo.Event.activate, this.onAppActivate.bind(this));
 	window.document.addEventListener(Mojo.Event.deactivate, this.onAppDeactivate.bind(this));
 	
+    AppAssistant.gpsWasEnabled = false;
+    new Mojo.Service.Request('palm://com.palm.location', {
+        method: 'getUseGps',
+		parameters: {},
+		onSuccess: function(response){
+            AppAssistant.gpsWasEnabled = response.useGps;
+            if (AppAssistant.gpsWasEnabled==false){
+                new Mojo.Service.Request('palm://com.palm.location', {
+                    method: 'setUseGps',
+                    parameters: {useGps: true}
+                });
+            }
+        }      
+    });
+    
 	// subscribe for notifications from connection manager
 	this._startConnectionWatch();
 	
@@ -102,6 +118,16 @@ AppAssistant.prototype.handleLaunch = function(appLaunchParams) {
 
 AppAssistant.prototype.cleanup = function() {
 	Mojo.Log.info("Maps: AppAssistant cleanup");
+    
+    if (AppAssistant.gpsWasEnabled==false){
+        new Mojo.Service.Request('palm://com.palm.location', {
+			method: 'setUseGps',
+			parameters: {
+				useGps: false
+			}
+		});
+    }   
+    
 	this._cleanupConnectionWatch();	
 	this.storageNotificationSession.cancel();
 };
diff --git a/usr/palm/applications/com.palm.app.maps/app/assistants/mapshell-assistant.js b/usr/palm/applications/com.palm.app.maps/app/assistants/mapshell-assistant.js
index 75c81aa..7451f16 100644
--- a/usr/palm/applications/com.palm.app.maps/app/assistants/mapshell-assistant.js
+++ b/usr/palm/applications/com.palm.app.maps/app/assistants/mapshell-assistant.js
@@ -5,6 +5,8 @@ var MapshellAssistant = Class.create({
 		// when true, card the app else instruct webview to go back
 		this.goBack = false;
 		this.spinnerModel = {spinning: false};
+		// fs mode parameter
+		this.paramFS = true
 		
 		// event handlers
 		this._onWebViewCreatePageH = this._onWebViewCreatePage.bind(this);
@@ -16,6 +18,7 @@ var MapshellAssistant = Class.create({
 		this._onWebViewTitleUrlChangedH = this._onWebViewTitleUrlChanged.bind(this);
 		this._onAdapterConnectH = this._onAdapterConnect.bind(this);
 		this._onAdapterDisconnectH = this._onAdapterDisconnect.bind(this);
+		this._holdHandlerH = this._holdHandler.bind(this);
 		
 		/*
 		var curr = new Date();
@@ -42,6 +45,8 @@ var MapshellAssistant = Class.create({
 		};	
 		this.controller.setupWidget(Mojo.Menu.appMenu, appMenuAttrs, this.appMenuModel);	
 		
+		//set fs mode
+		this.controller.enableFullScreenMode(this.paramFS);
 		// setup spinner
 		this.controller.setupWidget('disconnected_spinner', 
 			{spinnerSize: Mojo.Widget.spinnerLarge}, this.spinnerModel);
@@ -66,6 +71,7 @@ var MapshellAssistant = Class.create({
 		this.controller.listen('webview', Mojo.Event.webViewTitleUrlChanged, this._onWebViewTitleUrlChangedH);
 		this.controller.listen('webview', Mojo.Event.webViewServerConnect, this._onAdapterConnectH);
 		this.controller.listen('webview', Mojo.Event.webViewServerDisconnect, this._onAdapterDisconnectH);
+		this.controller.listen('webview', Mojo.Event.hold, this._holdHandlerH);
 	},
 	
 	cleanup: function() {
@@ -79,6 +85,7 @@ var MapshellAssistant = Class.create({
 		this.controller.stopListening('webview', Mojo.Event.webViewTitleUrlChanged, this._onWebViewTitleUrlChangedH);
 		this.controller.stopListening('webview', Mojo.Event.webViewServerConnect, this._onAdapterConnectH);	
 		this.controller.stopListening('webview', Mojo.Event.webViewServerDisconnect, this._onAdapterDisconnectH);
+		this.controller.stopListening('webview', Mojo.Event.hold, this._holdHandlerH);
 	},
 	
 	activate: function() {
@@ -272,6 +279,16 @@ var MapshellAssistant = Class.create({
 		this.controller.modelChanged(this.spinnerModel);
 	},
 	
+	_holdHandler: function(event) {
+		if (this.paramFS) {
+			this.paramFS = false
+			this.controller.enableFullScreenMode(this.paramFS);
+		} 
+		else {
+			this.paramFS = true
+			this.controller.enableFullScreenMode(this.paramFS);
+		}
+	},
 
 	handleCommand: function(event) {
 		if (event.type == Mojo.Event.back) {
