diff --git a/usr/palm/applications/com.palm.app.maps/app/assistants/app-assistant.js b/usr/palm/applications/com.palm.app.maps/app/assistants/app-assistant.js
index f5c81a6..6e8ef3d 100644
--- a/usr/palm/applications/com.palm.app.maps/app/assistants/app-assistant.js
+++ b/usr/palm/applications/com.palm.app.maps/app/assistants/app-assistant.js
@@ -4,10 +4,48 @@ AppAssistant.googleMapsURL = "http://maps.google.com/maps/m";
 //AppAssistant.googleMapsURL = "http://gjm.google.com/mwm?sky=tfb";
 AppAssistant.urlClientParam = "client=ms-mwm-palm";
 
+
 function AppAssistant(appController) {
 	Mojo.Log.info("MAPS App: AppAssistant");
     this.appController = appController;
+
+	
+	this.cookie = new Mojo.Model.Cookie('googleMaps');	
+	
+	var gmaps = { url : "http://maps.google.com/maps/m"	};
+	var tmp = this.cookie.get();
+	if (tmp == undefined) 
+		this.cookie.put(gmaps);
+	else
+		AppAssistant.googleMapsURL = tmp.url;
+		
+	Mojo.Log.info("Google Maps URL: " + AppAssistant.googleMapsURL);
+	
+	// Avner, 
+	this.getTweaksPrefs = new Mojo.Service.Request("palm://org.webosinternals.tweaks.prefs/", {
+				method: 'get', 
+				parameters: {	
+								'owner': "com.palmpreacher.patches.googlemaps", 
+								keys: ["googleMapsLocale"]
+							}, 
+				onSuccess: function(response) {
+
+							if(response) {
+							Mojo.Log.error("onSuccess: Google Maps: " + response.googleMapsLocale );
+
+								if(response.googleMapsLocale != undefined) {
+									AppAssistant.googleMapsURL = "http://maps.google.com/maps/m?hl=" + response.googleMapsLocale;
+								
+									this.cookie = new Mojo.Model.Cookie('googleMaps');	
+									var gmaps = { url : "http://maps.google.com/maps/m?hl=" + response.googleMapsLocale };
+									this.cookie.put(gmaps);
+								}
+							}
+					}.bind(this)}  );
+
 	
+ 
+ 
 	var countryCode = Mojo.Locale.getCurrentLocale().slice(-2).toLowerCase();
 	// HACK for google: they need "uk" intead of "gb" that we set
 	if (countryCode == "gb") {
@@ -315,6 +353,17 @@ AppAssistant.prototype.composeURL = function(launchParams) {
 // return the URL with parameters that we always send
 AppAssistant.prototype.getBaseUrl = function() {
 	
+	this.cookie = new Mojo.Model.Cookie('googleMaps');	
+	var gmaps = { url : "http://maps.google.com/maps/m"	};
+	var tmp = this.cookie.get();
+	if (tmp == undefined) 
+		this.cookie.put(gmaps);
+	else
+		AppAssistant.googleMapsURL = tmp.url;
+
+	Mojo.Log.error("Maps App: composeURL final URL is: %s", AppAssistant.googleMapsURL);
+
+
 	var baseUrl = AppAssistant.googleMapsURL; 
 	if (baseUrl.indexOf("?") > 0)
 		baseUrl += "&";
