diff --git a/usr/palm/applications/com.palm.app.maps/app/assistants/app-assistant.js b/usr/palm/applications/com.palm.app.maps/app/assistants/app-assistant.js
index 194d62e..5ef7414 100644
--- a/usr/palm/applications/com.palm.app.maps/app/assistants/app-assistant.js
+++ b/usr/palm/applications/com.palm.app.maps/app/assistants/app-assistant.js
@@ -20,6 +20,21 @@ function AppAssistant(appController) {
 	
 	// subscribe for notifications from connection manager
 	this._startConnectionWatch();
+    
+    this.gpsWasEnabled = false;
+    this.gpsStatReq = new Mojo.Service.Request('palm://com.palm.location', {
+        method: 'getUseGps',
+		parameters: {},
+		onSuccess: function(response){
+            this.gpsWasEnabled = response.useGps;
+            if (this.gpsWasEnabled==false){
+                new Mojo.Service.Request('palm://com.palm.location', {
+                    method: 'setUseGps',
+                    parameters: {useGps: true}
+                });
+            }
+        }      
+    });
 	
 	// register for entering msm notifications
  	this.storageNotificationSession = new Mojo.Service.Request(
@@ -81,6 +96,16 @@ AppAssistant.prototype.handleLaunch = function(appLaunchParams) {
 
 AppAssistant.prototype.cleanup = function() {
 	Mojo.Log.info("Maps: AppAssistant cleanup");
+    
+    if (this.gpsWasEnabled==false){
+        new Mojo.Service.Request('palm://com.palm.location', {
+			method: 'setUseGps',
+			parameters: {
+				useGps: false
+			}
+		});
+    }   
+     
 	this._cleanupConnectionWatch();	
 	this.storageNotificationSession.cancel();
 };
