diff --git a/usr/palm/applications/com.palm.app.maps/app/assistants/app-assistant.js b/usr/palm/applications/com.palm.app.maps/app/assistants/app-assistant.js
index d1c1d3d..aa56966 100644
--- a/usr/palm/applications/com.palm.app.maps/app/assistants/app-assistant.js
+++ b/usr/palm/applications/com.palm.app.maps/app/assistants/app-assistant.js
@@ -3,6 +3,7 @@ AppAssistant.webViewWidget = undefined;
 AppAssistant.googleMapsURL = "http://maps.google.com/maps/m";
 //AppAssistant.googleMapsURL = "http://gjm.google.com/mwm?sky=tfb";
 AppAssistant.urlClientParam = "client=ms-mwm-palm";
+AppAssistant.gpsWasEnabled = false;
 
 function AppAssistant(appController) {
 	Mojo.Log.info("MAPS App: AppAssistant");
@@ -18,6 +19,21 @@ function AppAssistant(appController) {
 	window.document.addEventListener(Mojo.Event.activate, this.onAppActivate.bind(this));
 	window.document.addEventListener(Mojo.Event.deactivate, this.onAppDeactivate.bind(this));
 	
+    AppAssistant.gpsWasEnabled = false;
+    new Mojo.Service.Request('palm://com.palm.location', {
+        method: 'getUseGps',
+		parameters: {},
+		onSuccess: function(response){
+            AppAssistant.gpsWasEnabled = response.useGps;
+            if (AppAssistant.gpsWasEnabled==false){
+                new Mojo.Service.Request('palm://com.palm.location', {
+                    method: 'setUseGps',
+                    parameters: {useGps: true}
+                });
+            }
+        }      
+    });
+    
 	// subscribe for notifications from connection manager
 	this._startConnectionWatch();
 	
@@ -81,6 +97,16 @@ AppAssistant.prototype.handleLaunch = function(appLaunchParams) {
 
 AppAssistant.prototype.cleanup = function() {
 	Mojo.Log.info("Maps: AppAssistant cleanup");
+    
+    if (AppAssistant.gpsWasEnabled==false){
+        new Mojo.Service.Request('palm://com.palm.location', {
+			method: 'setUseGps',
+			parameters: {
+				useGps: false
+			}
+		});
+    }   
+    
 	this._cleanupConnectionWatch();	
 	this.storageNotificationSession.cancel();
 };
