diff --git a/usr/palm/applications/com.palm.app.calendar/app/controllers/datetime-assistant.js b/usr/palm/applications/com.palm.app.calendar/app/controllers/datetime-assistant.js
index 1bc0374..e2fc966 100644
--- a/usr/palm/applications/com.palm.app.calendar/app/controllers/datetime-assistant.js
+++ b/usr/palm/applications/com.palm.app.calendar/app/controllers/datetime-assistant.js
@@ -181,13 +181,13 @@ var DatetimeAssistant = Class.create({
         this.startTimeModel = {
             time: new Date(this.targetModifiedTime.start)
         };
-    this.controller.setupWidget('edit_startTime', {label: $L('Time'),modelProperty:'time'}, this.startTimeModel);
+    this.controller.setupWidget('edit_startTime', {label: $L('Time'),modelProperty:'time',minuteInterval:1}, this.startTimeModel);
         this.controller.listen('edit_startTime', Mojo.Event.propertyChange, this.startTimeCallback.bindAsEventListener(this));
         
         this.endTimeModel = {
             time: new Date(this.targetModifiedTime.end)
         };
-    this.controller.setupWidget('edit_endTime', {label: $L('Time'), modelProperty:'time'}, this.endTimeModel);
+    this.controller.setupWidget('edit_endTime', {label: $L('Time'), modelProperty:'time',minuteInterval:1}, this.endTimeModel);
         this.controller.listen('edit_endTime', Mojo.Event.propertyChange, this.endTimeCallback.bindAsEventListener(this));
         
         
