diff --git a/usr/palm/applications/com.palm.app.calendar/app/AppLaunch.js b/usr/palm/applications/com.palm.app.calendar/app/AppLaunch.js
index 3e022db..9a97040 100644
--- a/usr/palm/applications/com.palm.app.calendar/app/AppLaunch.js
+++ b/usr/palm/applications/com.palm.app.calendar/app/AppLaunch.js
@@ -218,7 +218,8 @@ enyo.kind ({
 		,	timeMachine		= this.timeMachine
 
 		// Show day view at launch.  Turn off auto-propogation of the previous view's date.
-		,	view			= [{view:"DayView", autoDate:false}]
+		////,	view			= [{view:"DayView", autoDate:false}]   
+                ,	view			= [{view:"MonthView", autoDate:false}]  
 		,	windowParams	= enyo.windowParams
 		;
 		view.keep = true;
diff --git a/usr/palm/applications/com.palm.app.calendar/app/AppView.js b/usr/palm/applications/com.palm.app.calendar/app/AppView.js
index a14ce89..d34bdc6 100644
--- a/usr/palm/applications/com.palm.app.calendar/app/AppView.js
+++ b/usr/palm/applications/com.palm.app.calendar/app/AppView.js
@@ -113,8 +113,14 @@ enyo.kind({
 		var enyoApp	= enyo.application
 		,	ui		= this.$
 		;
-		ui.daySwitch.clickHandler();
+               
+		//ui.daySwitch.clickHandler();
+                ui.monthSwitch.clickHandler();
 		ui.pane.selectView (ui.calendar);
+                //ui.monthSwitch.clickHandler();
+                //ui.pane.selectView (ui.calendar.month.MonthView);
+
+
 
 		this.createComponents (this.popups);
 
@@ -122,6 +128,8 @@ enyo.kind({
 		enyoApp.autoDate = true;				// When switching views (example: WeekView->DayView), use the previous view's date.
 
 		enyoApp.watch	({ calendars:this, is24Hr:this, showEvent:this, showView:this });
+                // ui.monthSwitch.clickHandler();
+               //  this.$.monthSwitch.clickHandler();
 	},
 
 	destroy: function destroy () {
