diff --git a/usr/palm/applications/com.palm.app.calendar/app/controllers/edit-assistant.js b/usr/palm/applications/com.palm.app.calendar/app/controllers/edit-assistant.js
index 9ba9c70..6760a29 100644
--- a/usr/palm/applications/com.palm.app.calendar/app/controllers/edit-assistant.js
+++ b/usr/palm/applications/com.palm.app.calendar/app/controllers/edit-assistant.js
@@ -8,6 +8,7 @@ var EditAssistant = Class.create({
 	                											 {label:$L('Get Directions'), command:'direction', disabled:false}] 
 	                													 				
 							},
+							{label:$L('Edit event as new'), command:'editAsNewEvent', disabled:false},
 							{label:$L('Delete event'), command:'deleteEvent', disabled:false},
 							{label:$L('Preferences & Accounts'), command:Mojo.Menu.prefsCmd, disabled: true},
 							{label:$L('Help'), command:Mojo.Menu.helpCmd, disabled:false}]
@@ -671,6 +672,12 @@ var EditAssistant = Class.create({
 			} else if (event.command == Mojo.Menu.prefsCmd) {
 				Event.stop(event);
 				this.controller.stageController.pushScene('prefs');
+			}else if(event.command =='editAsNewEvent'){
+				Event.stop(event);
+				this.targetEvent = Object.clone(this.targetEvent);
+				this.targetId = 0;
+				this.targetEvent.id = 0;
+				this.targetEvent.dirty = true;
 			}else if(event.command =='copy'){
 				//@todo
 			}else if(event.command == Mojo.Menu.helpCmd){
