diff --git a/usr/palm/applications/com.palm.app.calendar/app/controllers/day-assistant.js b/usr/palm/applications/com.palm.app.calendar/app/controllers/day-assistant.js
index 84263c0..4101587 100644
--- a/usr/palm/applications/com.palm.app.calendar/app/controllers/day-assistant.js
+++ b/usr/palm/applications/com.palm.app.calendar/app/controllers/day-assistant.js
@@ -458,7 +458,7 @@ var DayAssistant = Class.create({
 	},
 	
 	renderFreeTime: function(whichDay, free) {
-		if (free.duration > 150) {
+		if (free.duration > 1440) {
 			var ctx = this.divs[whichDay].ctxEvents;
 			
 			var top = this.compressPos(free.top, free.topCompress);
@@ -856,7 +856,7 @@ var DayAssistant = Class.create({
 			if (free.height > maxFreeDuration)
 				maxFreeDuration = free.height;
 			
-			if (free.duration > 150) {
+			if (free.duration > 1440) {
 				var template;
 				var templateModel = {};
 				templateModel.hour = parseInt(free.duration / 100.0);
@@ -946,7 +946,7 @@ var DayAssistant = Class.create({
 		do {
 			this._freeTimeIndex++;
 			free = (this._freeTimeIndex < this._freeTimes.length) ? this._freeTimes[this._freeTimeIndex] : null;
-		} while (free && (free.duration <= 150));
+		} while (free && (free.duration <= 1440));
 		return free;
 	},
 	
@@ -3494,7 +3494,7 @@ var DayAssistant = Class.create({
 		
 		for (i = 0; i < day.freeTimes.length; i++) {
 			var free = day.freeTimes[i];
-			if (free.duration <= 150)
+			if (free.duration <= 1440)
 				continue;
 				
 			// Note the values passed to compressPos below are intentionally reversed.
