diff --git a/usr/palm/services/com.palm.service.calendar.google/javascript/utils/utils.js b/usr/palm/services/com.palm.service.calendar.google/javascript/utils/utils.js
index 77da023..0b6fa6c 100644
--- a/usr/palm/services/com.palm.service.calendar.google/javascript/utils/utils.js
+++ b/usr/palm/services/com.palm.service.calendar.google/javascript/utils/utils.js
@@ -93,10 +93,14 @@ var Utils = {
 		throw new Error(errMsg);
 	},
 
+	encodeRegExp: function(string) {
+		return string.replace(/[\]\[\\\^$\*\+\?\.\(\)\|\{\}]/g, "\\$&");
+	},
+
 	cacheAuthToken: function (authToken) {
 		if (authToken && authToken !== '') {
 			// console.log("###### cached authToken: " + authToken);
-			Utils.authTokenRegex = new RegExp(authToken, 'g');
+			Utils.authTokenRegex = new RegExp(Utils.encodeRegExp(authToken), 'g');
 		}
 	},
 
@@ -105,7 +109,7 @@ var Utils = {
 		if (password && password !== '') {
 			// console.log("###### cached password: " + password);
 			//put the URI encoded password also into the regular expression 
-			reStr = password + "|" + encodeURIComponent(password);
+			reStr = Utils.encodeRegExp(password) + "|" + Utils.encodeRegExp(encodeURIComponent(password));
 			Utils.passwordRegex = new RegExp(reStr, 'g');
 		}
 	},
@@ -117,7 +121,7 @@ var Utils = {
 				username = username.slice(0, pos);
 			}
 			// console.log("###### cached username: " + username);
-			Utils.usernameRegex = new RegExp(username, 'g');
+			Utils.usernameRegex = new RegExp(Utils.encodeRegExp(username), 'g');
 		}
 	},
 
