diff --git a/usr/palm/applications/com.palm.app.calendar/app/controllers/app-assistant.js b/usr/palm/applications/com.palm.app.calendar/app/controllers/app-assistant.js
index f1a8cd5..ecd98c1 100644
--- a/usr/palm/applications/com.palm.app.calendar/app/controllers/app-assistant.js
+++ b/usr/palm/applications/com.palm.app.calendar/app/controllers/app-assistant.js
@@ -166,12 +166,12 @@ var AppAssistant = Class.create (
 			// If the top scene is not the Day View, try to pop to it.
 			// If the top scene is already the Day View, there's no point
 			// in popping the scene
-			if (stageController.topScene().sceneName != "day") {
-				stageController.popScenesTo("day");
+			if (stageController.topScene().sceneName != "month") {
+				stageController.popScenesTo("month");
 			}
 
 			// Check to see if the day scene is already up
-			if (stageController.topScene().sceneName == "day") {
+			if (stageController.topScene().sceneName == "month") {
 				sceneExists = true;
 			}
 
@@ -192,7 +192,7 @@ var AppAssistant = Class.create (
 
 		// Only push the Day View if it's not already visible
 		if (!sceneExists) {
-			stageController.pushScene({name: "day", transition: Mojo.Transition.crossFade, disableSceneScroller: true});
+			stageController.pushScene({name: "month", transition: Mojo.Transition.crossFade, disableSceneScroller: true});
 		}
 	},
 
@@ -205,7 +205,7 @@ var AppAssistant = Class.create (
 
 		if (launchParams && launchParams.reminders) {
 			if (!appExists) {
-				stageController.pushScene({name: "day", disableSceneScroller: true});
+				stageController.pushScene({name: "month", disableSceneScroller: true});
 			}
 			stageController.pushScene("reminder-list");
 			return;
@@ -213,7 +213,7 @@ var AppAssistant = Class.create (
 
 		if (launchParams && launchParams.details) {
 			if (!appExists) {
-				stageController.pushScene({name: "day", disableSceneScroller: true});
+				stageController.pushScene({name: "month", disableSceneScroller: true});
 			}
 			// TODO need to pass event occurence date instead of today's date
 			//TODO: What is being passed in for the id here? Does it have times or just a db-id?
@@ -230,7 +230,7 @@ var AppAssistant = Class.create (
 			while(i < sceneStackLength && !validUnderScene){
 				var scene = sceneStack[i];
 				switch (scene.sceneName){
-					case "day":
+					case "month":
 					case "month":
 					case "week":
 						validUnderScene = scene.sceneName;
@@ -242,7 +242,7 @@ var AppAssistant = Class.create (
 			
 			//if we couldn't find a valid underscene or the app doesn't exist, push dayview.
 			if(!validUnderScene || !appExists){
-				stageController.pushScene({name: "day", disableSceneScroller: true});
+				stageController.pushScene({name: "month", disableSceneScroller: true});
 			}
 			else {
 				stageController.popScenesTo(validUnderScene);
@@ -472,12 +472,12 @@ var AppAssistant = Class.create (
 				done: function() {
 // TODO: Only set first use to false if at least one account exists, i.e. Palm Profile:  
 					this.setFirstUseToFalse();
-					stageController.pushScene ({name: "day", transition: Mojo.Transition.crossFade, disableSceneScroller: true});
+					stageController.pushScene ({name: "month", transition: Mojo.Transition.crossFade, disableSceneScroller: true});
 				}.bind(this)
 			});
 		} else {
 			// Otherwise show the Day View
-			stageController.pushScene({name: "day", transition: Mojo.Transition.crossFade, disableSceneScroller: true});
+			stageController.pushScene({name: "month", transition: Mojo.Transition.crossFade, disableSceneScroller: true});
 		}
 	},
 
