diff --git a/usr/palm/applications/com.palm.app.calendar/app/AppLaunch.js b/usr/palm/applications/com.palm.app.calendar/app/AppLaunch.js
index 510366d..809b0c7 100644
--- a/usr/palm/applications/com.palm.app.calendar/app/AppLaunch.js
+++ b/usr/palm/applications/com.palm.app.calendar/app/AppLaunch.js
@@ -230,7 +230,7 @@ enyo.kind ({
 		,	timeMachine		= this.timeMachine
 
 		// Show day view at launch.  Turn off auto-propogation of the previous view's date.
-		,	view			= {data: {view:"DayView", autoDate:false}, keep: true}
+		,	view			= {data: {view:"WeekView", autoDate:false}, keep: true}
 		,	windowParams	= enyo.windowParams
 		;
 		calendarWindow && (view.wait = true);									// If the window already exists, make it switch views first.
diff --git a/usr/palm/applications/com.palm.app.calendar/app/AppView.js b/usr/palm/applications/com.palm.app.calendar/app/AppView.js
index e48ebc2..e3f69ca 100644
--- a/usr/palm/applications/com.palm.app.calendar/app/AppView.js
+++ b/usr/palm/applications/com.palm.app.calendar/app/AppView.js
@@ -112,7 +112,7 @@ enyo.kind({
 		var enyoApp	= enyo.application
 		,	ui		= this.$
 		;
-		ui.daySwitch.clickHandler();
+		ui.weekSwitch.clickHandler();
 		ui.pane.selectView (ui.calendar);
 
 		this.createComponents (this.popups);
@@ -162,7 +162,7 @@ enyo.kind({
 
 	windowHiddenHandler: function windowHiddenHandler () {		// TODO: Set start date & view based on user's app preferences.
 		DEBUG && this.log ("======= HIDDEN\t");
-		var	view	= {data: {view: calendar.day.DayView, autoDate: false}, wait: true};	// Since keep-alive suspends timers, make share synchronous.
+		var	view	= {data: {view: calendar.day.WeekView, autoDate: false}, wait: true};	// Since keep-alive suspends timers, make share synchronous.
 
 		enyo.application.share	({showView: view});							// Ensure that "today" view is already displayed when opened with keep-alive.
 		enyo.application.shareCurrentDate	({date:new Date(), wait:true});	// Ensure that the current date is already today when opened with keep-alive.
