diff --git a/usr/palm/services/com.palm.service.calendar.google/javascript/utils/xhr.js b/usr/palm/services/com.palm.service.calendar.google/javascript/utils/xhr.js
index 6240d9a..055e036 100644
--- a/usr/palm/services/com.palm.service.calendar.google/javascript/utils/xhr.js
+++ b/usr/palm/services/com.palm.service.calendar.google/javascript/utils/xhr.js
@@ -30,6 +30,9 @@ var XMLHttpRequest = Class.create({
 	
 	setRequestHeader: function (name, value) {
 		//Utils.log("\n\n\n--->>> new request header. name: " + name + " value: " + value + " \n\n");
+		if (name == "Content-Length") {
+			return;
+		}
 		this.options.headers[name] = value;
 	},
 	
