diff --git a/usr/palm/applications/com.palm.app.calendar/app/controllers/edit-assistant.js b/usr/palm/applications/com.palm.app.calendar/app/controllers/edit-assistant.js
index 9ba9c70..87cc938 100644
--- a/usr/palm/applications/com.palm.app.calendar/app/controllers/edit-assistant.js
+++ b/usr/palm/applications/com.palm.app.calendar/app/controllers/edit-assistant.js
@@ -8,6 +8,7 @@ var EditAssistant = Class.create({
 	                											 {label:$L('Get Directions'), command:'direction', disabled:false}] 
 	                													 				
 							},
+							{label:$L('Edit event as new'), command:'editAsNewEvent', disabled:false},
 							{label:$L('Delete event'), command:'deleteEvent', disabled:false},
 							{label:$L('Preferences & Accounts'), command:Mojo.Menu.prefsCmd, disabled: true},
 							{label:$L('Help'), command:Mojo.Menu.helpCmd, disabled:false}]
@@ -297,6 +298,12 @@ var EditAssistant = Class.create({
 		
 		// Delete event menu item.  Users can delete new and existing events
 		if(!this.isReadOnlyEvent)
+			this.appMenuModel.items[3].disabled = false;   
+		else
+			this.appMenuModel.items[3].disabled = true; 
+
+		// Edit as new event menu item. Users cannot edit new items as a new item
+		if(this.targetEvent.id > 0)
 			this.appMenuModel.items[2].disabled = false;   
 		else
 			this.appMenuModel.items[2].disabled = true; 
@@ -671,6 +678,14 @@ var EditAssistant = Class.create({
 			} else if (event.command == Mojo.Menu.prefsCmd) {
 				Event.stop(event);
 				this.controller.stageController.pushScene('prefs');
+			}else if(event.command =='editAsNewEvent'){
+				Event.stop(event);
+				this.targetEvent = Object.clone(this.targetEvent);
+				this.targetId = 0;
+				delete this.targetEvent.id;
+				this.targetEvent.parentId = 0;
+
+				this.renderEvent(this.targetEvent);
 			}else if(event.command =='copy'){
 				//@todo
 			}else if(event.command == Mojo.Menu.helpCmd){
