diff --git a/usr/palm/applications/com.palm.app.calendar/app/controllers/day-assistant.js b/usr/palm/applications/com.palm.app.calendar/app/controllers/day-assistant.js
index 8cc09e6..54e95b4 100644
--- a/usr/palm/applications/com.palm.app.calendar/app/controllers/day-assistant.js
+++ b/usr/palm/applications/com.palm.app.calendar/app/controllers/day-assistant.js
@@ -60,9 +60,9 @@ var DayAssistant = Class.create({
 		this.pendingCalendarSettingsUpdate = false;
 		//this.pendingOpenScratchEventInDetails = false;
 		this.snapEffectCount = 0;
-		this.hourHeight = 48;
-		this.halfHourHeight = 24;
-		this.fifteenMinuteHeight = 12;
+		this.hourHeight = 31;
+		this.halfHourHeight = 31;
+		this.fifteenMinuteHeight = 31;
 		this.totalWidth = 320;
 		this.totalHeight = this.hourHeight * 24;
 		// To make sure that the 11pm timeline can be scrolled above the command menu buttons, add an extra timeline
@@ -402,7 +402,7 @@ var DayAssistant = Class.create({
 	getXIndexForTimeText:function(ctx,i,text)
 	{		
 			
-			ctx.font = "bold 14px Prelude";
+			ctx.font = "12px Prelude";
 			
 
 			var metrics = ctx.measureText(text);
@@ -444,11 +444,17 @@ var DayAssistant = Class.create({
 				if (i < 24) {
 					var text = String(i);
 					if (this.currentTimeFormat == "HH12") {
+						if (i < 12) {
+							text = String(i + 'a');
+						}
+						if (i == 12) {
+							text = 'Noon';						
+						}
 						if (i === 0) {
-							text = '12';
+							text = '12a';
 						}
 						if (i > 12) {
-							text = String(i - 12);
+							text = String((i - 12) + 'p');
 						}
 					}
 					ctx.strokeText(text, this.getXIndexForTimeText(ctx, i, text), top + 15);
@@ -564,6 +570,9 @@ var DayAssistant = Class.create({
 		
 		var top = topPos || this.compressPos(event.top, event.topCompress);
 		var height = ((event.end_decimal - event.start_decimal) * this.hourHeight / 100);
+		if (height < this.hourHeight) {
+			height = this.hourHeight;
+		}		
 		var left = marginLeft;
 		var width = eventMaxWidth;
 		if (event.overlap_count > 1) {
@@ -792,6 +801,9 @@ var DayAssistant = Class.create({
 		
 		var top = this.compressPos(event.top, event.topCompress);
 		var height = ((event.end_decimal - event.start_decimal) * this.hourHeight / 100);
+		if (height < this.hourHeight) {
+			height = this.hourHeight;
+		}		
 		var left = marginLeft;
 		var width = eventMaxWidth;
 		if (event.overlap_count > 1) {
@@ -882,10 +894,10 @@ var DayAssistant = Class.create({
 		//Incorporates the padding-top of the first all day event 
 		//from the old css style .dv-allday-event:first-child
 		var marginTop = 10;
-		var eventHeight = 24;
+		var eventHeight = 17;
 		// Incorporates the padding-top 1px + padding-bottom 9px from 
 		// the old css style dv-allday-event
-		var totalHeight = 34;
+		var totalHeight = 27;
 		
 		return {
 			top : (totalHeight * index + marginTop),
@@ -920,7 +932,7 @@ var DayAssistant = Class.create({
 		if (selected) {
 			ctx.fillStyle = "rgb(255, 255, 255)";
 		}
-		ctx.font = "bold 14px Prelude";
+		ctx.font = "bold 15px Prelude";
 		this.fillText(event.subject, ctx, dim.left+10, dim.top+17, dim.width-10, dim.top+30, 15 /*line height*/);
 		
 		if (selected) {
@@ -1161,11 +1173,13 @@ var DayAssistant = Class.create({
 			}
 			//now that we have finished rendering the timelines and number 
 			//render am pm
+			/*
 			if (this.currentTimeFormat == "HH12") {
 				//now show the am/pm indicators
 				this.calculatePosToDrawAmPm(Math.abs(this.tds[this.THIS_DAY].scroller.mojo.getScrollPosition().top));
 				Mojo.Dom.show (this.controller.get('ampm_indicator'));
 			}
+			*/
 		}
 
 		if (!!GLOBAL.AppInspector && whichDay == this.THIS_DAY) {
@@ -1659,6 +1673,7 @@ var DayAssistant = Class.create({
 		//get the default hour's timeline position
 		var day			= data.days[0];
 		var hour		= this.days[whichDay].defaultTime.getHours();
+		hour = '8';
 		var scrollToPos = day.timelines[hour].top;
 
 		if (this.compressed) {
diff --git a/usr/palm/applications/com.palm.app.calendar/app/views/day/day-scene.html b/usr/palm/applications/com.palm.app.calendar/app/views/day/day-scene.html
index 03c6380..1731f8d 100644
--- a/usr/palm/applications/com.palm.app.calendar/app/views/day/day-scene.html
+++ b/usr/palm/applications/com.palm.app.calendar/app/views/day/day-scene.html
@@ -103,7 +103,7 @@
 				  				<canvas id="dv_events" width="320" height="1200">								
 				  				</canvas>
 								<div id="current_time" style="position:absolute;top=0px;left=4px;" >
-									<img  src="images/current-time-indicator.png" />
+									<img  src="images/current-time-indicator.png" style="height:16px;width:5px;" />
 								</div>
 								<div id="ampm_indicator" style="-webkit-transition: color .3s ease-in-out;color:rgba(0,0,0,0) ;font-weight:bold; position:absolute;top:0px;left:6px;font-size:12px;font-style:Prelude">#{-value}</div>								
 								<div id="dv_scratch_events"></div>
diff --git a/usr/palm/applications/com.palm.app.calendar/stylesheets/day.css b/usr/palm/applications/com.palm.app.calendar/stylesheets/day.css
index 33e0a75..f61b15c 100644
--- a/usr/palm/applications/com.palm.app.calendar/stylesheets/day.css
+++ b/usr/palm/applications/com.palm.app.calendar/stylesheets/day.css
@@ -234,12 +234,12 @@ Copyright 2009 Palm, Inc.  All rights reserved.
 	display: block !important;
 }
 .all-day-events-spacer {
-	height:9px;
+	height:2px;
 }
 .alldayeventsList {
 	position: relative;
 	top: 0px;
-	padding: 4px 0px 5px;
+	/*padding: 4px 0px 5px;*/
 	width: 100%;
 	overflow: hidden;
 }
@@ -293,9 +293,9 @@ Copyright 2009 Palm, Inc.  All rights reserved.
 .timeHeader {
 	width: 24px;
 	height: 32px;
-	font-size: 14px;
+	font-size: 9px;
 	text-align: center;
-	padding: 4px 0px 0px;
+	/*padding: 0px 0px 0px;*/
 	color: #777;
 	position: absolute;
 	z-index:3;
@@ -321,8 +321,8 @@ Copyright 2009 Palm, Inc.  All rights reserved.
 	margin-top:-6px;	
 }
 .dv-event.scratch-event .event-edit .hint-text {
-	margin-top:-5px;	
-	font-size: 14px;
+	margin-top:-7px;	
+	font-size: 15px;
 }
 .dv-event.scratch-event .event-edit input:focus {
 	margin-top:-14px;
@@ -383,7 +383,7 @@ Copyright 2009 Palm, Inc.  All rights reserved.
 		height: 48px;
 		position: absolute;
 		right: 0px;
-		top: -11px;
+		top: -22px;
 		background: url(../images/day-event-info-press.png) center center no-repeat;
 		background: url(../images/day-event-info.png) center center no-repeat;
 		z-index: 10000;
