diff --git a/usr/palm/applications/com.palm.app.calendar/app/controllers/day-assistant.js b/usr/palm/applications/com.palm.app.calendar/app/controllers/day-assistant.js
index 8cc09e6..228a3e3 100644
--- a/usr/palm/applications/com.palm.app.calendar/app/controllers/day-assistant.js
+++ b/usr/palm/applications/com.palm.app.calendar/app/controllers/day-assistant.js
@@ -243,6 +243,7 @@ var DayAssistant = Class.create({
 	buildDayTitle: function(doScrim) {
 		var today = Date.today().clearTime();
 		var dayDateNoTime = new Date(this.dayDate).clearTime();
+		var weekNumber = new Date(this.dayDate).getISO8601Week();
 		var formatObj = {};
 		
         // formatString = "EEE MMM d";
@@ -251,18 +252,23 @@ var DayAssistant = Class.create({
         formatObj.dateComponents = "md";
 		
 		if (dayDateNoTime.compareTo(today) === 0) {
-			var template = new Template($L("#{todayStr}, #{dateStr}"));
+			var template = new Template($L("#{todayStr}, #{dateStr}") + ", W#{weekStr}");
 			var todayStr= Mojo.Format.formatRelativeDate(today, {date: "short"}).capitalize();
 			
 			var templateModel = {};
 			templateModel.todayStr = todayStr;
 			//templateModel.dateStr = Mojo.Format.formatDate(this.dayDate, $L("EEE MMM d"));
 			templateModel.dateStr = Mojo.Format.formatDate(this.dayDate, formatObj);
+			templateModel.weekStr = weekNumber;
 			this.controller.get('dv_title').update(template.evaluate(templateModel)); // Localize this date format string
 		} else {
 			//this.controller.get('dv_title').update(Mojo.Format.formatDate(this.dayDate, $L("EEE MMM d, yyyy"))); // Localize this date format string
 			formatObj.dateComponents = "mdy";
-			this.controller.get('dv_title').update(Mojo.Format.formatDate(this.dayDate, formatObj)); // Localize this date format string
+			var template = new Template($L("#{dateStr}") + ", W#{weekStr}");
+			var templateModel = {};
+			templateModel.weekStr = weekNumber;
+			templateModel.dateStr = Mojo.Format.formatDate(this.dayDate, formatObj);
+			this.controller.get('dv_title').update(template.evaluate(templateModel)); // Localize this date format string
 		}
 		
 		if (doScrim) {
diff --git a/usr/palm/applications/com.palm.app.calendar/app/controllers/month-assistant.js b/usr/palm/applications/com.palm.app.calendar/app/controllers/month-assistant.js
index f20a9ad..8b48351 100644
--- a/usr/palm/applications/com.palm.app.calendar/app/controllers/month-assistant.js
+++ b/usr/palm/applications/com.palm.app.calendar/app/controllers/month-assistant.js
@@ -133,8 +133,14 @@ var MonthAssistant = Class.create({
         formatObj.date = "medium";
         formatObj.dateComponents = "my";
 
+		var wn = new Date(this.monthDate);
+		wn.setDate(1);
+		var firstWeek = wn.getISO8601Week();	
+		wn.setMonth(wn.getMonth()+1);
+		wn.setDate(wn.getDate()-1);
+		var lastWeek = wn.getISO8601Week();
 		//this.mvTitle.innerText = Mojo.Format.formatDate(this.monthDate, $L("MMM yyyy")); // Localize this date format string
-		this.mvTitle.innerText = Mojo.Format.formatDate(this.monthDate, formatObj); // Localize this date format string
+		this.mvTitle.innerText = Mojo.Format.formatDate(this.monthDate, formatObj) + ", " + $L("Week") + " " + firstWeek + "-" + lastWeek; // Localize this date format string
 	},
 
 	buildDayHeader: function() {
diff --git a/usr/palm/applications/com.palm.app.calendar/app/controllers/week-assistant.js b/usr/palm/applications/com.palm.app.calendar/app/controllers/week-assistant.js
index 4fa6e78..446997b 100644
--- a/usr/palm/applications/com.palm.app.calendar/app/controllers/week-assistant.js
+++ b/usr/palm/applications/com.palm.app.calendar/app/controllers/week-assistant.js
@@ -137,8 +137,9 @@ var WeekAssistant = Class.create({
 	buildWeekTitle: function() {
 		var today = Date.today().clearTime();
 		var dayOfWeek = this.getFirstDayOfWeek(this.weekDate);
+		var weekNumber = new Date(this.weekDate).getISO8601Week();
 		
-		var template = new Template($L("Week of #{dateStr}"));
+		var template = new Template($L("Week") + " #{weekStr}, " + $L("#{dateStr}"));
 		var templateModel = {};
 		var formatObj = {};
 		// formatString = "MMM d, yyyy";
@@ -146,6 +147,7 @@ var WeekAssistant = Class.create({
         formatObj.dateComponents = "mdy";
 		//templateModel.dateStr = Mojo.Format.formatDate(dayOfWeek, $L("MMM d, yyyy"));
 		templateModel.dateStr = Mojo.Format.formatDate(dayOfWeek, formatObj);
+		templateModel.weekStr = weekNumber;
 		this.title.innerHTML = template.evaluate(templateModel);
 		
 		if (this.changingWeeks) {
diff --git a/usr/palm/applications/com.palm.app.calendar/javascripts/datejs/week.js b/usr/palm/applications/com.palm.app.calendar/javascripts/datejs/week.js
new file mode 100644
index 0000000..5a8f688
--- /dev/null
+++ b/usr/palm/applications/com.palm.app.calendar/javascripts/datejs/week.js
@@ -0,0 +1,10 @@
+Date.prototype.getISO8601Week = function() {
+	var d = new Date(this);
+	d.setUTCDate(d.getUTCDate() - (d.getUTCDay() + 6) % 7 + 3);
+	var e = d.valueOf();
+	d.setUTCMonth(0, 4);
+	var w = Math.round((e - d.valueOf()) / 604800000) + 1;
+	if (w <= 9) { w = "0" + w; }
+	return w;
+}
+
diff --git a/usr/palm/applications/com.palm.app.calendar/sources.json b/usr/palm/applications/com.palm.app.calendar/sources.json
index 8488b5b..2f271ab 100644
--- a/usr/palm/applications/com.palm.app.calendar/sources.json
+++ b/usr/palm/applications/com.palm.app.calendar/sources.json
@@ -42,4 +42,5 @@
 ,	{	"source": "app\/controllers\/prefs-options-assistant.js"	}
 
 ,	{	"source": "javascripts\/automation\/appinspector.js"		}
+,	{	"source": "javascripts\/datejs\/week.js"		}
 ]
