diff --git a/usr/palm/applications/com.palm.app.calendar/app/controllers/prefs-assistant.js b/usr/palm/applications/com.palm.app.calendar/app/controllers/prefs-assistant.js
index cebe803..86afa6b 100644
--- a/usr/palm/applications/com.palm.app.calendar/app/controllers/prefs-assistant.js
+++ b/usr/palm/applications/com.palm.app.calendar/app/controllers/prefs-assistant.js
@@ -1,5 +1,9 @@
 /* Copyright 2009 Palm, Inc.  All rights reserved. */
 
+var ZERO_MINUTES = $L("0 minutes");
+var FIFTEEN_MINUTES = $L("15 minutes");
+var FOUR_HOURS = $L("4 hours");
+var EIGHT_HOURS = $L("8 hours");
 var THIRTY_MINUTES = $L("30 minutes");
 var ONE_HOUR = $L("1 hour");
 var TWO_HOURS = $L("2 hours");
@@ -43,13 +47,22 @@ var PrefsAssistant = Class.create({
 	renderNewEventDuration: function() {
 		var ned = {};
 		
-		if (this.prefs.defaultEventDuration == 30) {
-			ned.duration = THIRTY_MINUTES;
+		
+		if (this.prefs.defaultEventDuration == 0) {
+			ned.duration = ZERO_MINUTES;
+		} else if (this.prefs.defaultEventDuration == 15) {
+			ned.duration = FIFTEEN_MINUTES;
+		} else if (this.prefs.defaultEventDuration == 30) {
+ 			ned.duration = THIRTY_MINUTES;
 		} else if (this.prefs.defaultEventDuration == 60) {
 			ned.duration = ONE_HOUR;
+		} else if (this.prefs.defaultEventDuration == 120) {
+ 			ned.duration = TWO_HOURS;
+ 		} else if (this.prefs.defaultEventDuration == 240) {
+			ned.duration = FOUR_HOURS;
 		} else {
-			ned.duration = TWO_HOURS;
-		}
+			ned.duration = EIGHT_HOURS;
+ 		}
 		
 		this.controller.get('pv_new_event_duration').update(ned.duration);
 	},
@@ -692,14 +705,26 @@ var PrefsAssistant = Class.create({
 				onChoose: this.newEventDurationChoose.bind(this),
 				toggleCmd: this.prefs.defaultEventDuration,
 				items: [{
-					label: THIRTY_MINUTES,
-					command: 30
-				}, {
-					label: ONE_HOUR,
-					command: 60
-				}, {
-					label: TWO_HOURS,
-					command: 120
+					label: ZERO_MINUTES,
+					command: 0
+				},{	
+					label: FIFTEEN_MINUTES,
+					command: 15
+				},{	
+ 					label: THIRTY_MINUTES,
+ 					command: 30
+ 				}, {
+ 					label: ONE_HOUR,
+ 					command: 60
+ 				}, {
+ 					label: TWO_HOURS,
+ 					command: 120
+				},{	
+					label: FOUR_HOURS,
+					command: 240
+				},{	
+					label: EIGHT_HOURS,
+					command: 480
 				}]
 			});
 	},
