diff --git a/usr/palm/services/com.palm.service.calendar.google/javascript/utils/utils.js b/usr/palm/services/com.palm.service.calendar.google/javascript/utils/utils.js
index 08a8701..82f218a 100644
--- a/usr/palm/services/com.palm.service.calendar.google/javascript/utils/utils.js
+++ b/usr/palm/services/com.palm.service.calendar.google/javascript/utils/utils.js
@@ -43,17 +43,21 @@ var Utils = {
 		throw new Error(errMsg);
 	},
 
+	encodeRegExp: function(string) {
+		return new RegExp(string.replace(/[\]\[\\\^$\*\+\?\.\(\)\|\{\}]/g, "\\$&"), 'g');
+	},
+
 	cacheAuthToken: function (authToken) {
 		if (authToken && authToken !== '') {
 			// console.log("###### cached authToken: " + authToken);
-			Utils.authTokenRegex = new RegExp(authToken, 'g');
+			Utils.authTokenRegex = Utils.buildRegExp(authToken);
 		}
 	},
 
 	cachePassword: function (password) {
 		if (password && password !== '') {
 			// console.log("###### cached password: " + password);
-			Utils.passwordRegex = new RegExp(password, 'g');
+			Utils.passwordRegex = Utils.buildRegExp(password);
 		}
 	},
 
@@ -64,7 +68,7 @@ var Utils = {
 				username = username.slice(0, pos);
 			}
 			// console.log("###### cached username: " + username);
-			Utils.usernameRegex = new RegExp(username, 'g');
+			Utils.usernameRegex = Utils.buildRegExp(username);
 		}
 	},
 
