diff --git a/usr/palm/applications/com.palm.app.calendar/app/controllers/app-assistant.js b/usr/palm/applications/com.palm.app.calendar/app/controllers/app-assistant.js
index 5375929..b9011ed 100644
--- a/usr/palm/applications/com.palm.app.calendar/app/controllers/app-assistant.js
+++ b/usr/palm/applications/com.palm.app.calendar/app/controllers/app-assistant.js
@@ -154,12 +154,12 @@ var AppAssistant = Class.create({
 			// If the top scene is not the Day View, try to pop to it.
 			// If the top scene is already the Day View, there's no point
 			// in popping the scene
-			if (stageController.topScene().sceneName != "day") {
-				stageController.popScenesTo("day");
+			if (stageController.topScene().sceneName != "week") {
+				stageController.popScenesTo("week");
 			}
 			
 			// Check to see if the day scene is already up
-			if (stageController.topScene().sceneName == "day") {
+			if (stageController.topScene().sceneName == "week") {
 				sceneExists = true;
 			}
 			
@@ -180,7 +180,7 @@ var AppAssistant = Class.create({
 		
 		// Only push the Day View if it's not already visible
 		if (!sceneExists) {
-			stageController.pushScene({name: "day", transition: Mojo.Transition.crossFade, disableSceneScroller: true});
+			stageController.pushScene({name: "week", transition: Mojo.Transition.crossFade, disableSceneScroller: true});
 		}
 	},
 	
@@ -191,14 +191,14 @@ var AppAssistant = Class.create({
 		}
 		if (launchParams && launchParams.reminders) {
 			if (!appExists) {
-				stageController.pushScene({name: "day", disableSceneScroller: true});
+				stageController.pushScene({name: "week", disableSceneScroller: true});
 			}
 			stageController.pushScene("reminder-list");
 			return;
 		}
 		if (launchParams && launchParams.details) {
 			if (!appExists) {
-				stageController.pushScene({name: "day", disableSceneScroller: true});
+				stageController.pushScene({name: "week", disableSceneScroller: true});
 			}
 			// TODO need to pass event occurence date instead of today's date
 			stageController.pushScene('edit', launchParams.details, new Date(), false);
@@ -206,7 +206,7 @@ var AppAssistant = Class.create({
 		}
 		if (launchParams && launchParams.newEvent) {
 			if (!appExists) {
-				stageController.pushScene({name: "day", disableSceneScroller: true});
+				stageController.pushScene({name: "week", disableSceneScroller: true});
 			}
 			
 			this.createNewEventAndShowDetails(stageController, launchParams.newEvent);
@@ -489,7 +489,7 @@ var AppAssistant = Class.create({
 			this.firstUseManager = new FirstUseManager(stageController);
 		} else {
 			// Otherwise show the Day View
-			stageController.pushScene({name: "day", transition: Mojo.Transition.crossFade, disableSceneScroller: true});
+			stageController.pushScene({name: "week", transition: Mojo.Transition.crossFade, disableSceneScroller: true});
 		}
 	},
 	
