diff --git a/usr/palm/applications/com.palm.app.calendar/app/controllers/day-assistant.js b/usr/palm/applications/com.palm.app.calendar/app/controllers/day-assistant.js
index 84263c0..279bf4e 100644
--- a/usr/palm/applications/com.palm.app.calendar/app/controllers/day-assistant.js
+++ b/usr/palm/applications/com.palm.app.calendar/app/controllers/day-assistant.js
@@ -44,9 +44,10 @@ var DayAssistant = Class.create({
 		this.snapEffectCount = 0;
 
 		
-		this.hourHeight = 48;
-		this.halfHourHeight = 24;
-		this.fifteenMinuteHeight = 12;
+				// simply shrink hourHeight...  I'm setting half and quarter hours in case we want to make hourHeight variable
+		this.hourHeight = 16;
+		this.halfHourHeight = Math.round(this.hourHeight / 2);
+		this.fifteenMinuteHeight = Math.round(this.hourHeight / 4);
 		
 		this.totalWidth = 320;
 		this.totalHeight = this.hourHeight * 24;
@@ -345,7 +346,7 @@ var DayAssistant = Class.create({
 	getXIndexForTimeText:function(ctx,i,text)
 	{		
 			
-			ctx.font = "bold 14px Prelude";
+			ctx.font = "bold 10px Prelude";
 			
 
 			var metrics = ctx.measureText(text);
@@ -381,6 +382,7 @@ var DayAssistant = Class.create({
 				}
 				
 				if (i < 24) {
+				/*
 					var text = String(i);
 					if (this.currentTimeFormat == "HH12") {
 						if (i == 0) 
@@ -389,6 +391,25 @@ var DayAssistant = Class.create({
 							text = String(i - 12);
 					}
 					ctx.strokeText(text, this.getXIndexForTimeText(ctx, i, text), top + 15);
+				*/
+						// replace the time labels on the left with just three labels
+					var text = "";
+					if ( i == 12 ) {
+						text = "noon";
+						ctx.strokeText(text, this.getXIndexForTimeText(ctx, i, text), top + 7);
+					}
+					else if ( i == 6  || i == 18 ) {
+						text = "";
+						if (this.currentTimeFormat == "HH12") {
+							if ( i > 12 ) {
+								text = String(i - 12) + "pm";
+							}
+							else {
+								text = i + "am";
+							}
+						}
+						ctx.strokeText(text, this.getXIndexForTimeText(ctx, i, text), top + 7);
+					}
 				}
 			}
 		}
@@ -748,16 +769,53 @@ var DayAssistant = Class.create({
 		ctx.fillStyle = this.colors[event.calendarColor].text;
 		if (selected)
 			ctx.fillStyle = "rgb(255, 255, 255)";
-		ctx.font = "bold 14px Prelude";
+		ctx.font = "bold 10px Prelude";
 		bottom = textTop;
+		/*
 		var subjectTop = top+15;
 		if (bottom >= (top + height))
 			subjectTop = top+17;	// no room for location or note, therefore vertically center subject text
-		this.fillText(event.subject, ctx, left+4, subjectTop, width, bottom, 15 /*line height*/);	
+		*/
+				// prepend the subject with a human readable time stamp
+		var subjectTop = top+12;
+		var timeText = "";
+		var d = new Date(event.start);
+		var timeH = d.getHours();
+		var timeM = d.getMinutes();
+		if (this.currentTimeFormat == "HH12") {
+			if ( timeM > 0 ) {
+				if ( timeH == 0 ) {
+					timeText = "12:" + pad(timeM) + "a";
+				} else if ( timeH == 12 ) {
+					timeText = timeH + ":" + pad(timeM) + "p";
+				} else if ( timeH < 12 ) {
+					timeText = timeH + ":" + pad(timeM) + "a";
+				} else {
+					timeText = String(timeH - 12) + ":" + pad(timeM) + "p";
+				}
+			}
+			else {
+				if ( timeH == 0 ) {
+					timeText = "12a";
+				} else if ( timeH == 12 ) {
+					timeText = timeH + "p";
+				} else if ( timeH < 12 ) {
+					timeText = timeH + "a";
+				} else {
+					timeText = String(timeH - 12) + "p";
+				}
+			}
+		}
+		else {
+			timeText = timeH + ":" + pad(timeM);
+		}
+		//var str = Object.toJSON(event);
+		//console.error(str);
+		this.fillText(timeText + " " + event.subject, ctx, left+4, subjectTop, width, bottom, 15 /*line height*/);	
 		
 		// Location
 		if (event.location && (event.location.length > 0) && (textTop < (top + height))) {
-			ctx.font = "italic bold 14px Prelude";
+			ctx.font = "italic bold 10px Prelude";
 			bottom = textTop+15;
 			this.fillText(event.location,ctx,left+4,textTop,width,bottom,15 /*line height*/);		
 			textTop += 15;
@@ -841,7 +899,8 @@ var DayAssistant = Class.create({
 		var i;
 		
 		// Make sure that free times, busy times, and events are sorted by start times
-		day.freeTimes = day.freeTimes.sortBy(function(f) { return f.start_decimal; })
+		// removing free time compression by setting empty array for freeTimes
+		day.freeTimes = []; //day.freeTimes.sortBy(function(f) { return f.start_decimal; })
 		day.events = day.events.sortBy(function(e) { return e.start_decimal; })
 		day.busyTimes = day.busyTimes.sortBy(function(b) { return b.start_decimal; })
 
@@ -1108,11 +1167,14 @@ var DayAssistant = Class.create({
 				}
 				//now that we have finished rendering the timelines and number 
 				//render am pm
+						// remove the am/pm indicator that used to follow the top of the screen
+				/*
 				if (this.currentTimeFormat == "HH12") {
 				//now show the am/pm indicators
 					this.calculatePosToDrawAmPm(Math.abs(this.tds[this.THIS_DAY].scroller.mojo.getScrollPosition().top));
 					this.controller.get('ampm_indicator').show();
 				}
+				*/
 
 			}
 		}
@@ -1467,6 +1529,8 @@ var DayAssistant = Class.create({
 			top: yPixel+'px'
 		}).show();
 
+		this.controller.get('current_time_image').style.height = this.hourHeight+'px';
+		this.controller.get('current_time_image').style.width = img.width+'px';
 	},
 	
 	calculatePosWithCompression:function(day,hour)
@@ -1500,6 +1564,10 @@ var DayAssistant = Class.create({
 	
 	
 	scrollToDefault: function(whichDay) {
+			// disable the auto scrolling and just always follow the bottom of the page
+		this.tds[whichDay].scroller.mojo.revealBottom();
+		this.updateTimer(false);
+		/*
 		if (this.tds[whichDay].scroller && this.tds[whichDay].scroller.mojo) {
 			//Mojo.Log.info('BEFORE SCROLL TOP', this.tds[whichDay].scroller.mojo.getScrollPosition().top);
 			var ctx = this.divs[whichDay].ctxEvents;
@@ -1526,6 +1594,7 @@ var DayAssistant = Class.create({
 			this.updateTimer(false);
 			//Mojo.Log.info('AFTER SCROLL TOP', this.tds[whichDay].scroller.mojo.getScrollPosition().top);	
 		}
+		*/
 	},
 	
 	blurStageBound:function(){
@@ -3587,11 +3656,14 @@ var DayAssistant = Class.create({
 			Mojo.Animation.animateValue(Mojo.Animation.queueForElement(this.scrollDataDiv), 'bezier', function(value) {
 					this.animationValue = value;
 					this.renderDay(this.THIS_DAY, day, false);
+							// more scrolling to disable
+					/*
 					if (this.compressed) {
 						this.tds[this.THIS_DAY].scroller.mojo.scrollTo(undefined, Math.round(currentTop + scrollOffset * value));
 					} else {
 						this.tds[this.THIS_DAY].scroller.mojo.scrollTo(undefined, Math.round(currentTop - scrollOffset * value));
 					}
+					*/
 				}.bind(this), 
 				{
 				from: 0.0,
@@ -3683,3 +3755,12 @@ DayAssistant.MIN_EVENT_HEIGHT = 24;
 DayAssistant.MAX_CACHE_SIZE = 15;		// in days
 DayAssistant.cacheSize = 0;
 DayAssistant.dayCache = new Hash();
+
+		// this function pads single digit numbers with zeros.  sprintf wasn't working
+pad = function(number) {
+	var str = String(number);
+	if ( number < 10 ) {
+		str = "0" + String(number);
+	}
+	return str;
+}
diff --git a/usr/palm/applications/com.palm.app.calendar/app/controllers/week-assistant.js b/usr/palm/applications/com.palm.app.calendar/app/controllers/week-assistant.js
index 217a467..f145a26 100644
--- a/usr/palm/applications/com.palm.app.calendar/app/controllers/week-assistant.js
+++ b/usr/palm/applications/com.palm.app.calendar/app/controllers/week-assistant.js
@@ -36,7 +36,7 @@ var WeekAssistant = Class.create({
 		this.NEXT_WEEK = 3;
 		this.active = false;
 		this.requests = [];
-		this.HOUR_HEIGHT = 30;
+		this.HOUR_HEIGHT = 16;
 		this.TOTAL_HEIGHT = this.HOUR_HEIGHT * 24;
 		this.COL_WIDTH = [45, 46, 46, 46, 46, 46, 45];
 		this.COL_POS = [0, 45, 91, 137, 183, 229, 275];
@@ -367,13 +367,30 @@ var WeekAssistant = Class.create({
 		this.settingSnapIndex = false;
 	},
 
-	renderDayMarkers: function(whichWeek) {
+	renderStartEndShading: function(whichWeek) {
 		var ctx = this.weeks[whichWeek].ctxEvents;
 		
 		var startPos = this.startDayMarkerPos * this.HOUR_HEIGHT - 1;
 		var midPos = this.midDayMarkerPos * this.HOUR_HEIGHT - 1;
 		var endPos = this.endDayMarkerPos * this.HOUR_HEIGHT - 1;
 
+			// shaded
+		ctx.fillStyle = "rgb(206, 206, 204)";
+		ctx.fillRect(0, 0, 320, startPos);
+		ctx.fillRect(0, endPos, 320, (this.TOTAL_HEIGHT - endPos));
+			// lines
+		ctx.fillStyle = "rgb(159, 159, 158)";
+		ctx.fillRect(0, startPos, 320, 2);
+		ctx.fillRect(0, endPos, 320, 2);
+	},
+	
+	renderDayMarkers: function(whichWeek) {
+		var ctx = this.weeks[whichWeek].ctxEvents;
+		
+		var startPos = this.startDayMarkerPos * this.HOUR_HEIGHT - 1;
+		var midPos = this.midDayMarkerPos * this.HOUR_HEIGHT - 1;
+		var endPos = this.endDayMarkerPos * this.HOUR_HEIGHT - 1;       
+		
 		// Draw marker lines
 		ctx.fillStyle = "rgb(135, 135, 134)";
 		ctx.fillRect(0, startPos, 320, 2);
@@ -403,6 +420,33 @@ var WeekAssistant = Class.create({
 		ctx.strokeText(this.endDayMarkerTime, 17, endPos + 24);
 	},
 	
+	renderTimeLabels: function(whichWeek) {                                                                                                                       
+		var ctx = this.weeks[whichWeek].ctxEvents;
+	                
+				// print 6am/noon/6pm
+		ctx.font = "bold 10px Prelude";
+		if ( 6 > this.startDayMarkerPos && 6 <= this.endDayMarkerPos ) {
+			ctx.fillStyle = "rgb(85, 85, 85)";
+		} else {
+			ctx.fillStyle = "rgb(119, 119, 119)";
+		}
+		ctx.strokeText('6am', 4, (6 * this.HOUR_HEIGHT - 1) + 24);
+		
+		if ( 12 > this.startDayMarkerPos && 12 <= this.endDayMarkerPos ) {
+			ctx.fillStyle = "rgb(85, 85, 85)";
+		} else {
+			ctx.fillStyle = "rgb(119, 119, 119)";
+		}
+		ctx.strokeText('noon', 4, (12 * this.HOUR_HEIGHT - 1) + 24);
+		
+		if ( 18 > this.startDayMarkerPos && 18 <= this.endDayMarkerPos ) {
+			ctx.fillStyle = "rgb(85, 85, 85)";
+		} else {
+			ctx.fillStyle = "rgb(119, 119, 119)";
+		}
+		ctx.strokeText('6pm', 4, (18 * this.HOUR_HEIGHT - 1) + 24);
+	},
+	
 	renderEmptyWeek: function(whichWeek) {
 		var ctx = this.weeks[whichWeek].ctxEvents;
 		
@@ -410,6 +454,7 @@ var WeekAssistant = Class.create({
 		ctx.fillStyle = "rgb(228, 228, 226)";
 		ctx.fillRect(0, 0, 320, this.TOTAL_HEIGHT);
 
+		this.renderStartEndShading(whichWeek);
 		// Draw vertical day seperator lines
 		var x = 0;
 		for (var i = 0; i < 6; i++) {
@@ -420,7 +465,7 @@ var WeekAssistant = Class.create({
 		    ctx.fillRect(x+1, 0, 1, this.TOTAL_HEIGHT);
 		}
 		
-		this.renderDayMarkers(whichWeek);
+		//this.renderDayMarkers(whichWeek);
 	},	
 	
 	updateWeekHeight: function(whichWeek) {
@@ -585,7 +630,8 @@ var WeekAssistant = Class.create({
     		ctxAllDay.fillRect(x+1, 0, 1, this.weeks[whichWeek].alldayHeight);
 		}
 		
-		this.renderStartEndDayLabels(whichWeek);
+		//this.renderStartEndDayLabels(whichWeek);
+		this.renderTimeLabels(whichWeek);
 	},
 	
 	setup: function() {
@@ -858,7 +904,9 @@ var WeekAssistant = Class.create({
 	
  	scrollToDefault: function(animate) {
 		if (this.scroller) {
-			this.scroller.mojo.scrollTo(0, -(this.startDayMarkerPos * this.HOUR_HEIGHT - this.weeks[this.THIS_WEEK].alldayHeight) + 20, animate);
+			//this.scroller.mojo.scrollTo(0, -(this.startDayMarkerPos * this.HOUR_HEIGHT - this.weeks[this.THIS_WEEK].alldayHeight) + 20, animate);
+			//this.scroller.mojo.scrollTo(0, this.TOTAL_HEIGHT);
+			this.scroller.mojo.revealBottom();
 		}
 	},
 	
diff --git a/usr/palm/applications/com.palm.app.calendar/app/views/day/day-scene.html b/usr/palm/applications/com.palm.app.calendar/app/views/day/day-scene.html
index 03c6380..954eeee 100644
--- a/usr/palm/applications/com.palm.app.calendar/app/views/day/day-scene.html
+++ b/usr/palm/applications/com.palm.app.calendar/app/views/day/day-scene.html
@@ -103,7 +103,7 @@
 				  				<canvas id="dv_events" width="320" height="1200">								
 				  				</canvas>
 								<div id="current_time" style="position:absolute;top=0px;left=4px;" >
-									<img  src="images/current-time-indicator.png" />
+									<img id="current_time_image" src="images/current-time-indicator.png" />
 								</div>
 								<div id="ampm_indicator" style="-webkit-transition: color .3s ease-in-out;color:rgba(0,0,0,0) ;font-weight:bold; position:absolute;top:0px;left:6px;font-size:12px;font-style:Prelude">#{-value}</div>								
 								<div id="dv_scratch_events"></div>
diff --git a/usr/palm/applications/com.palm.app.calendar/app/views/week/week-scene.html b/usr/palm/applications/com.palm.app.calendar/app/views/week/week-scene.html
index 9dcd204..989911e 100644
--- a/usr/palm/applications/com.palm.app.calendar/app/views/week/week-scene.html
+++ b/usr/palm/applications/com.palm.app.calendar/app/views/week/week-scene.html
@@ -35,13 +35,13 @@
 			<div id="wv_scroll_events" x-mojo-element="Scroller">
 			  	<div id="wv_event_container">
 			  		<div id="wv_eventsDay0" class="wv-events-day">
-			  			<canvas id="events0" width="320" height="850"></canvas>
+			  			<canvas id="events0" width="320" height="450"></canvas>
 					</div>
 			  		<div id="wv_eventsDay1" class="wv-events-day">
-			  			<canvas id="events1" width="320" height="850"></canvas>
+			  			<canvas id="events1" width="320" height="450"></canvas>
 					</div>
 			  		<div id="wv_eventsDay2" class="wv-events-day">
-			  			<canvas id="events2" width="320" height="850"></canvas>
+			  			<canvas id="events2" width="320" height="450"></canvas>
 					</div>
 				</div>
 			</div>	
diff --git a/usr/palm/applications/com.palm.app.calendar/stylesheets/day.css b/usr/palm/applications/com.palm.app.calendar/stylesheets/day.css
index 02d0417..af5a05c 100644
--- a/usr/palm/applications/com.palm.app.calendar/stylesheets/day.css
+++ b/usr/palm/applications/com.palm.app.calendar/stylesheets/day.css
@@ -234,12 +234,12 @@ Copyright 2009 Palm, Inc.  All rights reserved.
 	display: block !important;
 }
 .all-day-events-spacer {
-	height:9px;
+	height:0px;
 }
 .alldayeventsList {
 	position: relative;
 	top: 0px;
-	padding: 4px 0px 5px;
+	/*padding: 4px 0px 5px;*/
 	width: 100%;
 	overflow: hidden;
 }
@@ -293,9 +293,9 @@ Copyright 2009 Palm, Inc.  All rights reserved.
 .timeHeader {
 	width: 24px;
 	height: 32px;
-	font-size: 14px;
+	font-size: 10px;
 	text-align: center;
-	padding: 4px 0px 0px;
+	/*padding: 0px 0px 0px;*/
 	color: #777;
 	position: absolute;
 	z-index:3;
@@ -321,8 +321,8 @@ Copyright 2009 Palm, Inc.  All rights reserved.
 	margin-top:-6px;	
 }
 .dv-event.scratch-event .event-edit .hint-text {
-	margin-top:-5px;	
-	font-size: 14px;
+	margin-top:-7px;	
+	font-size: 10px;
 }
 .dv-event.scratch-event .event-edit input:focus {
 	margin-top:-14px;
@@ -337,7 +337,7 @@ Copyright 2009 Palm, Inc.  All rights reserved.
 	vertical-align: middle;
 	border: none;
 	outline: none;
-	font-size: 14px;
+	font-size: 10px;
 	height: 16px;
 	font-weight: bold;
 	color: inherit;
@@ -388,7 +388,7 @@ Copyright 2009 Palm, Inc.  All rights reserved.
 		height: 48px;
 		position: absolute;
 		right: 0px;
-		top: -11px;
+		top: -28px;
 		background: url(../images/day-event-info-press.png) center center no-repeat;
 		background: url(../images/day-event-info.png) center center no-repeat;
 		z-index: 10000;
@@ -447,7 +447,7 @@ Copyright 2009 Palm, Inc.  All rights reserved.
 
 
 .dv-event .subject {
-	font-size: 14px;
+	font-size: 10px;
 	line-height: 14px;
 	height: 14px;
 	font-weight:bold;
@@ -456,7 +456,7 @@ Copyright 2009 Palm, Inc.  All rights reserved.
 }
 .dv-event .location {
 	margin-top:-1px;
-	font-size: 14px;
+	font-size: 10px;
 	line-height: 14px;
 	height: 14px;
 	font-weight:bold;
