diff --git a/usr/palm/applications/com.palm.app.calendar/app/controllers/edit-assistant.js b/usr/palm/applications/com.palm.app.calendar/app/controllers/edit-assistant.js
index 7b4c244..e1fdca5 100644
--- a/usr/palm/applications/com.palm.app.calendar/app/controllers/edit-assistant.js
+++ b/usr/palm/applications/com.palm.app.calendar/app/controllers/edit-assistant.js
@@ -581,20 +581,62 @@ var EditAssistant = Class.create({
                 list: [ 'none', // Do NOT Localize
 						'-PT0M', //DO NOT Localize
 						'-PT5M', // Do Not Localize
-						'-PT10M', //Do not localize
 						'-PT15M', // Do NOT Localize
 						'-PT30M', // Do NOT Localize
 						'-PT1H', // Do NOT Localize
-						'-P1D' // Do NOT Localize
+						'-PT90M', // Do NOT Localize
+						'-PT2H', // Do NOT Localize
+						'-PT3H', // Do NOT Localize
+						'-PT4H', // Do NOT Localize
+						'-PT6H', // Do NOT Localize
+						'-PT8H', // Do NOT Localize
+						'-PT10H', // Do NOT Localize
+						'-PT12H', // Do NOT Localize
+						'-PT23H', // Do NOT Localize
+						'-P1D', // Do NOT Localize
+						'-P36H', // Do NOT Localize
+	 					'-P2D', // Do NOT Localize
+						'-P60H', // Do NOT Localize	
+	 					'-P3D', // Do NOT Localize
+						'-P84H', // Do NOT Localize
+						'-P4D', // Do NOT Localize
+						'-P5D', // Do NOT Localize
+						'-P6D', // Do NOT Localize			
+	 					'-P1W', // Do NOT Localize
+						'-P2W', // Do NOT Localize
+						'-P3W', // Do NOT Localize
+						'-P4W', // Do NOT Localize
+						'-P8W', // Do NOT Localize
+						'-P12W', // Do NOT Localize
+						'-P16W', // Do NOT Localize
+						'-P20W', // Do NOT Localize
+						'-P24W', // Do NOT Localize
+						'-P52W' // Do NOT Localize
 				]
             },
             alarmsAllday: {
                 list: [ 'none', // Do NOT Localize
 						'-PT0M',// DO NOT Localize
+						'-PT12H', // DO NOT Localize
 						'-P1D', // Do NOT Localize
+						'-PT36H', // DO NOT Localize
 						'-P2D', // Do NOT Localize
+						'-PT60H', // DO NOT Localize
 						'-P3D', // Do NOT Localize
-						'-P1W'  // Do NOT Localize
+						'-PT84H', // DO NOT Localize
+						'-P4D', // Do NOT Localize
+						'-P5D', // Do NOT Localize
+						'-P6D', // Do NOT Localize
+						'-P1W',  // Do NOT Localize
+						'-P2W', // Do NOT Localize
+						'-P3W', // Do NOT Localize
+						'-P4W', // Do NOT Localize
+						'-P8W', // Do NOT Localize
+						'-P12W', // Do NOT Localize
+						'-P16W', // Do NOT Localize
+						'-P20W', // Do NOT Localize
+						'-P24W', // Do NOT Localize
+						'-P52W'  // Do NOT Localize
 				]
             }
         };
