diff --git a/usr/palm/services/com.palm.service.calendar.google/javascript/assistants/syncassistant.js b/usr/palm/services/com.palm.service.calendar.google/javascript/assistants/syncassistant.js
index c93d1bb..602c4dd 100644
--- a/usr/palm/services/com.palm.service.calendar.google/javascript/assistants/syncassistant.js
+++ b/usr/palm/services/com.palm.service.calendar.google/javascript/assistants/syncassistant.js
@@ -29,9 +29,9 @@ var SyncAssistant = Class.create(Sync.SyncCommand, {
 	initialize: function() {
 		// TODO: get "better" values for these
 		var now = new Date();
-		now.setMonth(now.getMonth() - 1);
+		now.setMonth(now.getMonth() - 12);
 		this._initialUpdatedMin = Calendar.IO.Utils.dateToIso8601(now);
-		now.setMonth(now.getMonth() + 3);
+		now.setMonth(now.getMonth() + 60);
 		this._initialUpdatedMax = Calendar.IO.Utils.dateToIso8601(now);
 		this.tzManager = new Calendar.TimezoneManager();
 	},
