diff --git a/usr/palm/applications/com.palm.app.calendar/app/controllers/edit-assistant.js b/usr/palm/applications/com.palm.app.calendar/app/controllers/edit-assistant.js
index 9ba9c70..3b82262 100644
--- a/usr/palm/applications/com.palm.app.calendar/app/controllers/edit-assistant.js
+++ b/usr/palm/applications/com.palm.app.calendar/app/controllers/edit-assistant.js
@@ -8,6 +8,7 @@ var EditAssistant = Class.create({
 	                											 {label:$L('Get Directions'), command:'direction', disabled:false}] 
 	                													 				
 							},
+							{label:$L('Edit event as new'), command:'editAsNewEvent', disabled:false},
 							{label:$L('Delete event'), command:'deleteEvent', disabled:false},
 							{label:$L('Preferences & Accounts'), command:Mojo.Menu.prefsCmd, disabled: true},
 							{label:$L('Help'), command:Mojo.Menu.helpCmd, disabled:false}]
@@ -297,6 +298,12 @@ var EditAssistant = Class.create({
 		
 		// Delete event menu item.  Users can delete new and existing events
 		if(!this.isReadOnlyEvent)
+			this.appMenuModel.items[3].disabled = false;   
+		else
+			this.appMenuModel.items[3].disabled = true; 
+
+		// Edit as new event menu item. Users cannot edit new items as a new item
+		if(this.targetEvent.id > 0)
 			this.appMenuModel.items[2].disabled = false;   
 		else
 			this.appMenuModel.items[2].disabled = true; 
@@ -440,21 +447,68 @@ var EditAssistant = Class.create({
 			alarmsTimed: {
 				list: [	 'none', // Do NOT Localize
 						 '-PT0M', //DO NOT Localize
-						 '-PT5M', // Do Not Localize
-						 '-PT10M', //Do not localize
 						 '-PT15M', // Do NOT Localize
 						 '-PT30M', // Do NOT Localize
+						 '-PT45M', // Do NOT Localize
 						 '-PT1H', // Do NOT Localize
-						 '-P1D' // Do NOT Localize
+						 '-PT90M', // Do NOT Localize
+						 '-PT2H', // Do NOT Localize
+						 '-PT150M', // Do NOT Localize
+						 '-PT3H', // Do NOT Localize
+						 '-PT210M', // Do NOT Localize
+						 '-PT4H', // Do NOT Localize
+						 '-PT5H', // Do NOT Localize
+						 '-PT6H', // Do NOT Localize
+						 '-PT7H', // Do NOT Localize
+						 '-PT8H', // Do NOT Localize
+						 '-PT10H', // Do NOT Localize
+						 '-PT12H', // Do NOT Localize	
+						 '-PT23H', // Do NOT Localize
+						 '-P1D', // Do NOT Localize
+						'-P36H', // Do NOT Localize
+	 					'-P2D', // Do NOT Localize
+						'-P60H', // Do NOT Localize	
+	 					'-P3D', // Do NOT Localize
+						'-P84H', // Do NOT Localize
+						'-P4D', // Do NOT Localize
+						'-P5D', // Do NOT Localize
+						'-P6D', // Do NOT Localize			
+	 					'-P1W', // Do NOT Localize
+						'-P2W', // Do NOT Localize
+						'-P3W', // Do NOT Localize
+						'-P4W', // Do NOT Localize
+						'-P8W', // Do NOT Localize
+						'-P12W', // Do NOT Localize
+						'-P16W', // Do NOT Localize
+						'-P20W', // Do NOT Localize
+						'-P24W', // Do NOT Localize
+						'-P52W' // Do NOT Localize
 						]
 						},
 			alarmsAllday: {
 				list: [ 'none', // Do NOT Localize
 						'-PT0M',// DO NOT Localize
-	 					'-P1D', // Do NOT Localize
+						'-PT12H', // Do NOT Localize	
+						'-PT23H', // Do NOT Localize						
+						'-P1D', // Do NOT Localize
+						'-P36H', // Do NOT Localize
 	 					'-P2D', // Do NOT Localize
+						'-P60H', // Do NOT Localize	
 	 					'-P3D', // Do NOT Localize
+						'-P84H', // Do NOT Localize
+						'-P4D', // Do NOT Localize
+						'-P5D', // Do NOT Localize
+						'-P6D', // Do NOT Localize			
 	 					'-P1W', // Do NOT Localize
+						'-P2W', // Do NOT Localize
+						'-P3W', // Do NOT Localize
+						'-P4W', // Do NOT Localize
+						'-P8W', // Do NOT Localize
+						'-P12W', // Do NOT Localize
+						'-P16W', // Do NOT Localize
+						'-P20W', // Do NOT Localize
+						'-P24W', // Do NOT Localize
+						'-P52W' // Do NOT Localize
 					 ]
 					}
 		};
@@ -512,6 +566,7 @@ var EditAssistant = Class.create({
 			modelProperty: 'location',
 			hintText: $L('Event location'),
 			acceptBack:true,
+			multiline:true,
 			runTextLinker:true
 	
 		};
@@ -671,6 +726,15 @@ var EditAssistant = Class.create({
 			} else if (event.command == Mojo.Menu.prefsCmd) {
 				Event.stop(event);
 				this.controller.stageController.pushScene('prefs');
+			}else if(event.command =='editAsNewEvent'){
+				Event.stop(event);
+				this.targetEvent = Object.clone(this.targetEvent);
+				this.targetId = 0;
+				delete this.targetEvent.id;
+				this.targetEvent.parentId = 0;
+
+				this.renderEvent(this.targetEvent);
+				this.edit_infoHandler();
 			}else if(event.command =='copy'){
 				//@todo
 			}else if(event.command == Mojo.Menu.helpCmd){
