Calendar: Allow Location field to wrap on multiple lines.
Index: /usr/palm/applications/com.palm.app.calendar/app/controllers/edit-assistant.js
===================================================================
--- .orig/usr/palm/applications/com.palm.app.calendar/app/controllers/edit-assistant.js
+++ /usr/palm/applications/com.palm.app.calendar/app/controllers/edit-assistant.js
@@ -516,7 +516,8 @@ var EditAssistant = Class.create({
 		var tfAttrs2 = {
 			modelProperty: 'location',
 			hintText: $L('Event location'),
-			acceptBack:true
+			acceptBack:true,
+			multiline:true
 	
 		};
 		 this.locationmodel = {
