diff --git a/usr/palm/applications/com.palm.app.calendar/app/controllers/edit-assistant.js b/usr/palm/applications/com.palm.app.calendar/app/controllers/edit-assistant.js
index cff0013..c084d8a 100644
--- a/usr/palm/applications/com.palm.app.calendar/app/controllers/edit-assistant.js
+++ b/usr/palm/applications/com.palm.app.calendar/app/controllers/edit-assistant.js
@@ -15,6 +15,7 @@ var EditAssistant = Class.create({
 																{label:$L('Get Directions'), command:'direction', disabled:false}] 
 							},
 							{label:$L('Delete event'), command:'deleteEvent', disabled:false},
+							{label:$L('Bridge Call Dialer'), command:'bridgeCallDialer', disabled:false},
 							{label:$L('Preferences & Accounts'), command:Mojo.Menu.prefsCmd, disabled: true},
 							{label:$L('Help'), command:Mojo.Menu.helpCmd, disabled:false}]
 				  },
@@ -828,6 +829,19 @@ var EditAssistant = Class.create({
 				this.handleSave(event);
 			}							
 		}
+		if (event.type == Mojo.Event.command && event.command == 'bridgeCallDialer') {
+					this.controller.serviceRequest("palm://com.palm.applicationManager", {
+						method: 'launch',
+						parameters: {
+							id: "com.ahub.bridgecalldialer",
+							params: {
+								subject: this.targetEvent.subject,
+								location: this.targetEvent.location,
+								note: this.targetEvent.note
+							}
+						}
+					});
+		}		
 	},
 	
 	noteChanged: function(e) {
