diff --git a/usr/palm/applications/com.palm.app.calendar/app/controllers/reminder-assistant.js b/usr/palm/applications/com.palm.app.calendar/app/controllers/reminder-assistant.js
index d190fd1..60c3429 100644
--- a/usr/palm/applications/com.palm.app.calendar/app/controllers/reminder-assistant.js
+++ b/usr/palm/applications/com.palm.app.calendar/app/controllers/reminder-assistant.js
@@ -4,7 +4,7 @@
 var ReminderAssistant = Class.create({
 	initialize: function(eventId,needThrobbing) {
 		Mojo.Log.info("$$$$$$$$ Reminder-assistant initialize: start: "+eventId);
-		this.DEFAULT_SNOOZE = 5;
+		this.DEFAULT_SNOOZE = 10;
 		this.reminders = getReminderManager();
 		getAppAssistant().openReminderAlert = this;
 		this.requests = [];
@@ -363,7 +363,7 @@ var ReminderAssistant = Class.create({
 		var template = new Template(templateStr);
 		var snoozeText=template.evaluate(templateModel);
 		Mojo.Log.info("$$$$$$$$ Reminder-assistant snooze: showBanner" + this.reminder.id);
-		Mojo.Controller.appController.showBanner({messageText: snoozeText}, null /*launchArguments*/, "calendar-snooze");
+	//	Mojo.Controller.appController.showBanner({messageText: snoozeText}, null /*launchArguments*/, "calendar-snooze");
 
 		Mojo.Log.info("$$$$$$$$ Reminder-assistant snooze: end" + this.reminder.id);
 	},
@@ -404,7 +404,7 @@ var ReminderAssistant = Class.create({
 	
 	scheduleAutoCloseTask: function() {
 		Mojo.Log.info("$$$$$$$$ Reminder-assistant scheduleAutoCloseTask: start " + this.reminder.id);
-		var minCloseTime = new Date(this.reminder.time).addMinutes(15);
+		var minCloseTime = new Date(this.reminder.time).addMinutes(2160);
 		var closeTime = new Date(this.reminder.time + this.reminder.duration);
 		if (closeTime < minCloseTime)
 			closeTime = minCloseTime;
