diff --git a/usr/palm/services/com.palm.service.calendar.google/javascript/config.js b/usr/palm/services/com.palm.service.calendar.google/javascript/config.js
index 8e9cb48..fb7af18 100644
--- a/usr/palm/services/com.palm.service.calendar.google/javascript/config.js
+++ b/usr/palm/services/com.palm.service.calendar.google/javascript/config.js
@@ -6,6 +6,6 @@ var Config = {
 	requiresInternet: true,
 	syncInterval: "20m",
 	maxSyncResults: 10,
-	syncWindowMonthsBefore: 1,
-	syncWindowMonthsAfter: 2
+	syncWindowMonthsBefore: 120,
+	syncWindowMonthsAfter: 120
 };
