diff --git a/usr/palm/applications/com.palm.app.calendar/app/controllers/day-assistant.js b/usr/palm/applications/com.palm.app.calendar/app/controllers/day-assistant.js
index 84263c0..0f85dbc 100644
--- a/usr/palm/applications/com.palm.app.calendar/app/controllers/day-assistant.js
+++ b/usr/palm/applications/com.palm.app.calendar/app/controllers/day-assistant.js
@@ -11,8 +11,8 @@ var DayAssistant = Class.create({
 		this.appMenuModel = { visible:true, 
 									label:$L('Calendar'), 
 									items: [ Mojo.Menu.editItem,
-											{label:$L('New'), items: [{label:$L('Event'), command:'newtimed', disabled:false},
-	                													{label:$L('All day event'), command:'newallday', disabled:false}] 
+											{label:$L('New'), items: [{label:$L('Event'),     shortcut:'n', command:'newtimed', disabled:false},
+	                													{label:$L('All day event'),     shortcut:'d', command:'newallday', disabled:false}] 
 	                													 				
 											},											
 											{label:$L('Sync Now'), command:'sync', id: 2},
@@ -1824,7 +1824,7 @@ var DayAssistant = Class.create({
 		this.menuModel = {
 			visible:true,
 			items: [
-					{},
+					{label: $L('Event'), icon: 'new-timed', command: 'newtimed'},
 						{label:$L('Views'), 
 						toggleCmd:'day', 
 						items:[
@@ -1832,7 +1832,7 @@ var DayAssistant = Class.create({
 							{label:$L('Week'), icon: 'menu-week', command:'week'}, 
 							{label:$L('Month'), icon: 'menu-month', command:'month'}
 					 ]},
-					{},
+					{label: $L('All day event'), icon: 'new-allday', command: 'newallday'},
 					]};
 			
 		this.controller.setupWidget(Mojo.Menu.commandMenu, undefined, this.menuModel);
diff --git a/usr/palm/applications/com.palm.app.calendar/app/controllers/month-assistant.js b/usr/palm/applications/com.palm.app.calendar/app/controllers/month-assistant.js
index 747fcb3..43b0652 100644
--- a/usr/palm/applications/com.palm.app.calendar/app/controllers/month-assistant.js
+++ b/usr/palm/applications/com.palm.app.calendar/app/controllers/month-assistant.js
@@ -11,6 +11,9 @@ var MonthAssistant = Class.create({
 					label:$L('Calendar'), 
 					items: [
 						Mojo.Menu.editItem,
+						{label:$L('New'), items: [{label:$L('Event'),     shortcut:'n', command:'newtimed', disabled:false},
+	                								{label:$L('All day event'),     shortcut:'d', command:'newallday', disabled:false}] 
+						},
 						{label:$L('Sync Now'), command:'sync', id: 0},
 						{label:$L('Show today'), command:'today', id: 1},
 						{label:$L('Jump to...'), command:'jumpto', id: 2},
@@ -637,7 +640,7 @@ var MonthAssistant = Class.create({
         this.menuModel = 
         {
             visible: true,
-            items: [{}, 
+            items: [{label: $L('Event'), icon: 'new-timed', command: 'newtimed'}, 
             {
                 label: $L('Views'),
                 toggleCmd: 'month',
@@ -657,7 +660,7 @@ var MonthAssistant = Class.create({
                     icon: 'menu-month',
                     command: 'month'
                 }]
-            }, {}, ]
+            }, {label: $L('All day event'), icon: 'new-allday', command: 'newallday'}, ]
         };
 		
 		this.controller.setupWidget(Mojo.Menu.commandMenu, undefined, this.menuModel);
@@ -665,6 +668,7 @@ var MonthAssistant = Class.create({
 		this.remindersUpdated(); 	// Update missed reminders menu item
 		
 		this.CalSelectorHandler = this.calSelector.bindAsEventListener(this);
+		this.getCalId = DayAssistant.getCalId();
   	},
   
 	showJumpTo: function() {
@@ -715,6 +719,12 @@ var MonthAssistant = Class.create({
 			} else if (event.command == Mojo.Menu.prefsCmd) {
 				Event.stop(event);
 				this.controller.stageController.pushScene('prefs');
+			} else if(event.command =="newtimed"){
+				Event.stop(event);
+				this.controller.stageController.pushScene('edit', 0, 0, false,this.getCalId/*calendarId*/,0/*subject*/);
+			} else if(event.command=="newallday"){
+				Event.stop(event);
+				this.controller.stageController.pushScene('edit', 0, 0, true,this.getCalId,0/*subject*/);
 			}else if(event.command == Mojo.Menu.helpCmd){
 				getAppManagerService().launchHelp(this.controller);
 			} else if (event.command == 'sync') {
diff --git a/usr/palm/applications/com.palm.app.calendar/app/controllers/week-assistant.js b/usr/palm/applications/com.palm.app.calendar/app/controllers/week-assistant.js
index 217a467..a4e3a16 100644
--- a/usr/palm/applications/com.palm.app.calendar/app/controllers/week-assistant.js
+++ b/usr/palm/applications/com.palm.app.calendar/app/controllers/week-assistant.js
@@ -8,6 +8,9 @@ var WeekAssistant = Class.create({
 		this.appMenuModel = { visible:true, 
 									label:$L('Calendar'), 
 									items: [ Mojo.Menu.editItem,
+											{label:$L('New'), items: [{label:$L('Event'),     shortcut:'n', command:'newtimed', disabled:false},
+	                													{label:$L('All day event'),     shortcut:'d', command:'newallday', disabled:false}] 
+											},
 											{label:$L('Sync Now'), command:'sync', id: 0},
 											{label:$L('Show today'), command:'today', id: 1},
 											{label:$L('Jump to...'), command:'jumpto', id: 2},
@@ -708,7 +711,7 @@ var WeekAssistant = Class.create({
 		this.menuModel = {
 			visible:true,
 			items: [ 
-						{},
+						{label: $L('Event'), icon: 'new-timed', command: 'newtimed'},
 						{	label:$L('Views'), 
 							toggleCmd:'week', 
 							items:[
@@ -716,7 +719,7 @@ var WeekAssistant = Class.create({
 								{label:$L('Week'), icon: 'menu-week', command:'week'}, 
 								{label:$L('Month'), icon: 'menu-month', command:'month'}
 								]},
-						{},
+						{label: $L('All day event'), icon: 'new-allday', command: 'newallday'},
 					]
 			};
 			
@@ -727,6 +730,7 @@ var WeekAssistant = Class.create({
     	this.CalSelectorHandler = this.calSelector.bindAsEventListener(this);
 		
 		this.scrolling = 0;
+		this.getCalId = DayAssistant.getCalId();
 	},
   
 	showJumpTo: function() {
@@ -776,6 +780,12 @@ var WeekAssistant = Class.create({
 			} else if (event.command == Mojo.Menu.prefsCmd) {
 				Event.stop(event);
 				this.controller.stageController.pushScene('prefs');
+			} else if(event.command =="newtimed"){
+				Event.stop(event);
+				this.controller.stageController.pushScene('edit', 0, 0, false,this.getCalId/*calendarId*/,0/*subject*/);
+			} else if(event.command=="newallday"){
+				Event.stop(event);
+				this.controller.stageController.pushScene('edit', 0, 0, true,this.getCalId,0/*subject*/);
 			} else if(event.command == Mojo.Menu.helpCmd){
 				getAppManagerService().launchHelp(this.controller);
 			} else if (event.command == 'sync') {
diff --git a/usr/palm/applications/com.palm.app.calendar/stylesheets/calendar.css b/usr/palm/applications/com.palm.app.calendar/stylesheets/calendar.css
index f30cb63..f47332d 100644
--- a/usr/palm/applications/com.palm.app.calendar/stylesheets/calendar.css
+++ b/usr/palm/applications/com.palm.app.calendar/stylesheets/calendar.css
@@ -59,6 +59,14 @@ body {
 		background-image: url(../images/menu-icon-day.png);
 }
 
+.palm-menu-icon.new-timed{
+		background-image: url(/usr/palm/applications/com.palm.app.calendar/images/menu-icon-newevent.png);
+}
+
+.palm-menu-icon.new-allday {
+		background-image: url(/usr/palm/applications/com.palm.app.calendar/images/menu-icon-today.png);
+}
+
 .palm-menu-icon.edit {
 		background-image: url(../images/menu-icon-compose.png);
 }
