diff --git a/usr/palm/applications/com.palm.app.agendaview/app/model/AgendaModel.js b/usr/palm/applications/com.palm.app.agendaview/app/model/AgendaModel.js
index 6db82b8..6f2f6b0 100644
--- a/usr/palm/applications/com.palm.app.agendaview/app/model/AgendaModel.js
+++ b/usr/palm/applications/com.palm.app.agendaview/app/model/AgendaModel.js
@@ -1,6 +1,6 @@
 var REFRESH_INTERVAL = 300000; // 5 min - how often events list is synced with the calendar db
 var ROCKS_URL = "/media/internal/wallpapers/12.jpg";
-var FLOWERS_URL = "images/bg-ORIENTATION.png";
+var FLOWERS_URL = "/media/internal/wallpapers/AgendaBG.png";
 var DEBUG_OLD_VERSION = false; // debug on a version which does not support the Calendar api
 
 if (!DEBUG_OLD_VERSION) {
@@ -215,14 +215,6 @@ AgendaModel.prototype.getBackgroundImage = function(callback, isLandscape)
 			callback("");
 		case AgendaModel.BgTypes.FLOWERS:
 			var url = FLOWERS_URL;
-			if (isLandscape) 
-			{
-				url = url.replace(/ORIENTATION/, "landscape");
-			} 
-			else
-			{
-				url = url.replace(/ORIENTATION/, "portrait")
-			}
 			callback(url)
 			break;
 	}
