diff --git a/usr/palm/applications/com.palm.app.calendar/app/controllers/day-assistant.js b/usr/palm/applications/com.palm.app.calendar/app/controllers/day-assistant.js
index 84263c0..f1cc276 100644
--- a/usr/palm/applications/com.palm.app.calendar/app/controllers/day-assistant.js
+++ b/usr/palm/applications/com.palm.app.calendar/app/controllers/day-assistant.js
@@ -458,7 +458,7 @@ var DayAssistant = Class.create({
 	},
 	
 	renderFreeTime: function(whichDay, free) {
-		if (free.duration > 150) {
+		if (free.duration > 1440) {
 			var ctx = this.divs[whichDay].ctxEvents;
 			
 			var top = this.compressPos(free.top, free.topCompress);
@@ -946,7 +946,7 @@ var DayAssistant = Class.create({
 		do {
 			this._freeTimeIndex++;
 			free = (this._freeTimeIndex < this._freeTimes.length) ? this._freeTimes[this._freeTimeIndex] : null;
-		} while (free && (free.duration <= 150));
+		} while (free && (free.duration <= 1440));
 		return free;
 	},
 	
diff --git a/usr/palm/applications/com.palm.app.calendar/app/controllers/day-assistant.js.rej b/usr/palm/applications/com.palm.app.calendar/app/controllers/day-assistant.js.rej
new file mode 100644
index 0000000..5db9f7d
--- /dev/null
+++ b/usr/palm/applications/com.palm.app.calendar/app/controllers/day-assistant.js.rej
@@ -0,0 +1,29 @@
+--- usr/palm/applications/com.palm.app.calendar/app/controllers/day-assistant.js
++++ usr/palm/applications/com.palm.app.calendar/app/controllers/day-assistant.js
+@@ -952,7 +952,7 @@
+ 				maxFreeDuration = free.height;
+ 			}
+ 			
+-			if (free.duration > 150) {
++			if (free.duration > 1440) {
+ 				var templateModel =
+ 				{	hour: parseInt((free.duration / 100.0), 10)
+ 				,	min	: (((free.duration % 100) / 100.0) * 60).toFixed (0) * 1
+@@ -1298,7 +1298,7 @@
+ 		for (i = 0; i < day.freeTimes.length; i++) {
+ 			free = day.freeTimes[i];
+ 
+-			if (free.duration <= 150) {
++			if (free.duration <= 1440) {
+ 				continue;
+ 			}
+ 
+@@ -3713,7 +3713,7 @@
+ 		
+ 		for (var i=0, j=day.freeTimes.length; i < j; i++) {
+ 			var free = day.freeTimes[i];
+-			if (free.duration <= 150) {
++			if (free.duration <= 1440) {
+ 				continue;
+ 			}
+ 
