diff --git a/usr/palm/applications/com.palm.app.calendar/app/controllers/edit-assistant.js b/usr/palm/applications/com.palm.app.calendar/app/controllers/edit-assistant.js
index 9ba9c70..3f54a2d 100644
--- a/usr/palm/applications/com.palm.app.calendar/app/controllers/edit-assistant.js
+++ b/usr/palm/applications/com.palm.app.calendar/app/controllers/edit-assistant.js
@@ -444,7 +444,10 @@ var EditAssistant = Class.create({
 						 '-PT10M', //Do not localize
 						 '-PT15M', // Do NOT Localize
 						 '-PT30M', // Do NOT Localize
+						 '-PT45M',      // Customized: Add 45 minute option
 						 '-PT1H', // Do NOT Localize
+						 '-PT90M',      // Customized: Add 90 minute option
+						 '-PT2H',       // Customized: Add 2 hour option
 						 '-P1D' // Do NOT Localize
 						]
 						},
diff --git a/usr/palm/applications/com.palm.app.calendar/app/controllers/prefs-assistant.js b/usr/palm/applications/com.palm.app.calendar/app/controllers/prefs-assistant.js
index cebe803..9591ca7 100644
--- a/usr/palm/applications/com.palm.app.calendar/app/controllers/prefs-assistant.js
+++ b/usr/palm/applications/com.palm.app.calendar/app/controllers/prefs-assistant.js
@@ -3,7 +3,7 @@
 var THIRTY_MINUTES = $L("30 minutes");
 var ONE_HOUR = $L("1 hour");
 var TWO_HOURS = $L("2 hours");
-var numTimedAlarm = 7;		
+var numTimedAlarm = 10;   // Customized: Changed 7 to 10. See corresponding change in BuildAlarmsMenu
 var PrefsAssistant = Class.create({
 				 
 	initialize: function() {
@@ -241,7 +241,9 @@ var PrefsAssistant = Class.create({
 					 this.alarms[4].command,
 					 this.alarms[5].command,
 					 this.alarms[6].command,
-				
+					 this.alarms[7].command,  // Customized: Added 3 new options, so need
+					 this.alarms[8].command,  // Customized: to account for 10 items in
+					 this.alarms[9].command,  // Customized: the list instead of 7.
 									
 					 this.allDayAlarms[1].command,
 					 this.allDayAlarms[2].command,
@@ -763,7 +765,10 @@ var PrefsAssistant = Class.create({
 		{label: '', command: '-PT10M'},		//Do Not localize
 		{label: '', command: '-PT15M'},		// Do NOT Localize
 		{label: '', command: '-PT30M'},		// Do NOT Localize
+		{label: '', command: '-PT45M'},          // Customized: Added 45 minute option
 		{label: '', command: '-PT1H'},		// Do NOT Localize
+		{label: '', command: '-PT90M'},          // Customized: Added 90 minute option
+		{label: '', command: '-PT2H'},           // Customized: Added 2 hour option
 		{label: '', command: '-P1D'}			// Do NOT Localize	
 	],
 	
@@ -774,4 +779,4 @@ var PrefsAssistant = Class.create({
 		{label: '', command: '-P3D'},			// Do NOT Localize
 		{label: '', command: '-P1W'}			// Do NOT Localize		
 	]
-});
\ No newline at end of file
+});
