diff --git a/usr/palm/applications/com.palm.app.calendar/app/controllers/edit-assistant.js b/usr/palm/applications/com.palm.app.calendar/app/controllers/edit-assistant.js
index 9ba9c70..9a7a9dc 100644
--- a/usr/palm/applications/com.palm.app.calendar/app/controllers/edit-assistant.js
+++ b/usr/palm/applications/com.palm.app.calendar/app/controllers/edit-assistant.js
@@ -512,6 +512,7 @@ var EditAssistant = Class.create({
 			modelProperty: 'location',
 			hintText: $L('Event location'),
 			acceptBack:true,
+			multiline:true,
 			runTextLinker:true
 	
 		};
