Index: /usr/palm/applications/com.palm.app.calculator/app/controllers/calculator-assistant.js
===================================================================
--- /usr/palm/applications/com.palm.app.calculator/app/controllers/~calculator-assistant.orig
+++ /usr/palm/applications/com.palm.app.calculator/app/controllers/calculator-assistant.js
@@ -160,12 +160,14 @@
   	onNumberPress: function(event, number) {
 		this.calculator.enterDigit(number);
 		this.updateResults();
+		this.vibrate();
 		event.stop();
 	},
 	
   	onDecimalPress: function(event) {
 		this.calculator.enterDecimal();
 		this.updateResults();
+		this.vibrate();
 		event.stop();
 	},
 	
@@ -178,6 +180,7 @@
 			this.calculator.divide();
 		}
 		this.updateResults();
+		this.vibrate();
 		event.stop();
 	},
 	
@@ -189,6 +192,7 @@
 			this.calculator.multiply();
 		}
 		this.updateResults();
+		this.vibrate();
 		event.stop();
 	},
 	
@@ -200,6 +204,7 @@
 			this.calculator.subtract();
 		}
 		this.updateResults();
+		this.vibrate();
 		event.stop();
 	},
 	
@@ -211,6 +216,7 @@
 			this.calculator.add();
 		}
 		this.updateResults();
+		this.vibrate();
 		event.stop();
 	},
 	
@@ -218,12 +224,14 @@
 
 		this.calculator.equals();
 		this.updateResults();
+		this.vibrate();
 		event.stop();
 	},
 	
   	onClearPress: function(event) {
 		this.calculator.clear();
 		this.updateResults();
+		this.vibrate();
 		event.stop();
 	},
 	
@@ -231,18 +239,21 @@
 		this.calculator.memPlus();
 		this.showMemoryIndicator();
 		this.updateResults();
+		this.vibrate();
 		event.stop();
 	},
 	
   	onMemRecallPress: function(event) {
 		this.calculator.memRecall();
 		this.updateResults();
+		this.vibrate();
 		event.stop();
 	},
 	
 	onBackspacePress: function(event) {
 		this.calculator.backspace();
 		this.updateResults();
+		this.vibrate();
 		event.stop();
 	},
 	
@@ -250,6 +261,7 @@
 		this.calculator.memClear();
 		this.hideMemoryIndicator();
 		this.updateResults();
+		this.vibrate();
 		event.stop();
 	},
 	
@@ -297,6 +309,16 @@
 		} else if (event.type == Mojo.Event.commandEnable && event.command == Mojo.Menu.helpCmd) {
 			event.stopPropagation();
 		}
+	},
+	
+	vibrate: function() {
+		this.controller.serviceRequest(
+			"palm://com.palm.vibrate", {
+				method: 'vibrate', parameters: {
+					 'period': 0,'duration': 50
+				}
+			}
+		);
 	},
 });
 
