diff --git a/usr/palm/applications/com.palm.app.calculator/app/controllers/calculator-assistant.js b/usr/palm/applications/com.palm.app.calculator/app/controllers/calculator-assistant.js
index c385ded..ec274c3 100644
--- a/usr/palm/applications/com.palm.app.calculator/app/controllers/calculator-assistant.js
+++ b/usr/palm/applications/com.palm.app.calculator/app/controllers/calculator-assistant.js
@@ -177,6 +177,7 @@ var CalculatorAssistant = Class.create({
 		this.clearOperatorHighlight();
 		this.calculator.enterDigit(number);
 		this.updateResults();
+		this.vibrate();
 		event.stop();
 	},
 	
@@ -184,6 +185,7 @@ var CalculatorAssistant = Class.create({
 		this.clearOperatorHighlight();
 		this.calculator.enterDecimal();
 		this.updateResults();
+		this.vibrate();
 		event.stop();
 	},
 	
@@ -198,6 +200,7 @@ var CalculatorAssistant = Class.create({
 			this.calculator.divide();
 		}
 		this.updateResults();
+		this.vibrate();
 		event.stop();
 	},
 	
@@ -211,6 +214,7 @@ var CalculatorAssistant = Class.create({
 			this.calculator.multiply();
 		}
 		this.updateResults();
+		this.vibrate();
 		event.stop();
 	},
 	
@@ -224,6 +228,7 @@ var CalculatorAssistant = Class.create({
 			this.calculator.subtract();
 		}
 		this.updateResults();
+		this.vibrate();
 		event.stop();
 	},
 	
@@ -237,6 +242,7 @@ var CalculatorAssistant = Class.create({
 			this.calculator.add();
 		}
 		this.updateResults();
+		this.vibrate();
 		event.stop();
 	},
 	
@@ -244,6 +250,7 @@ var CalculatorAssistant = Class.create({
 		this.clearOperatorHighlight();
 		this.calculator.equals();
 		this.updateResults();
+		this.vibrate();
 		event.stop();
 	},
 	
@@ -251,6 +258,7 @@ var CalculatorAssistant = Class.create({
 		this.clearOperatorHighlight();
 		this.calculator.clear();
 		this.updateResults();
+		this.vibrate();
 		event.stop();
 	},
 	
@@ -258,6 +266,7 @@ var CalculatorAssistant = Class.create({
 		this.calculator.memPlus();
 		this.showMemoryIndicator();
 		this.updateResults();
+		this.vibrate();
 		event.stop();
 	},
 	
@@ -265,12 +274,14 @@ var CalculatorAssistant = Class.create({
 		this.clearOperatorHighlight();
 		this.calculator.memRecall();
 		this.updateResults();
+		this.vibrate();
 		event.stop();
 	},
 	
 	onBackspacePress: function(event) {
 		this.calculator.backspace();
 		this.updateResults();
+		this.vibrate();
 		event.stop();
 	},
 	
@@ -278,6 +289,7 @@ var CalculatorAssistant = Class.create({
 		this.calculator.memClear();
 		this.hideMemoryIndicator();
 		this.updateResults();
+		this.vibrate();
 		event.stop();
 	},
 	
@@ -327,6 +339,15 @@ var CalculatorAssistant = Class.create({
 			event.stopPropagation();
 		}
 	},
+	vibrate: function() {
+		this.controller.serviceRequest(
+			"palm://com.palm.vibrate", {
+				method: 'vibrate', parameters: {
+					'period': 0,'duration': 50
+				}
+			}
+		);
+	},
 });
 
 
