--- /usr/lib/luna/system/luna-applauncher/app/controllers/launcher-assistant.js.orig	Sat Dec  5 11:48:12 2009
+++ /usr/lib/luna/system/luna-applauncher/app/controllers/launcher-assistant.js	Thu Dec  3 19:42:03 2009
@@ -183,6 +183,15 @@
 		delete this.launchRequest;
 		
 		this.removeLaunchFeedback();
+		
+		/* Reset launcher to first page */
+		$('launcher_root').mojo.setSnapIndex(0, false);
+		
+		/* Jump scroll position back to top for all pages */
+		for (var i=0; i<numPages; i++) {
+			var scroller = this.getPageScroller(i);
+			if (scroller && scroller.mojo) { scroller.mojo.revealTop(); }
+		}
 	},
 	
 	onResize: function(event) {
