diff --git a/usr/lib/luna/system/luna-applauncher/app/controllers/global-search-assistant.js b/usr/lib/luna/system/luna-applauncher/app/controllers/global-search-assistant.js
index 6f3368e..751ce0f 100644
--- a/usr/lib/luna/system/luna-applauncher/app/controllers/global-search-assistant.js
+++ b/usr/lib/luna/system/luna-applauncher/app/controllers/global-search-assistant.js
@@ -247,9 +247,6 @@ GlobalSearchAssistant = Class.create({
 			SystemManagerService.showLauncher(false);
 			this.launcherHiddenAtFirstKeydown = false;
 		}
-		else {
-			SystemManagerService.showQuickLaunch(true);
-		}
 		
 		this.galInit();
 	},
diff --git a/usr/lib/luna/system/luna-applauncher/app/controllers/launcher-assistant.js b/usr/lib/luna/system/luna-applauncher/app/controllers/launcher-assistant.js
index 9057138..13e79b7 100644
--- a/usr/lib/luna/system/luna-applauncher/app/controllers/launcher-assistant.js
+++ b/usr/lib/luna/system/luna-applauncher/app/controllers/launcher-assistant.js
@@ -8,7 +8,7 @@ var LauncherAssistant = Class.create({
 	 * hardcoding the quick launch height until we can find a better way. 
 	 * This value = ql graphic height - the offset from the top (small fade area)
 	 */
-	kQuickLaunchHeight: 67,
+	kQuickLaunchHeight: 0,
 	
 	kPageIndicatorSpacing: 6, 	/* spacing between each page indicator */
 	kPageMargin: 10, /* 10 pixel margin on each side of a page */
@@ -23,10 +23,15 @@ var LauncherAssistant = Class.create({
 		visible: true,
 		label: $L('Launcher'),
         items: [
-			/*{ label: $L('New page'), command: 'newpage' },
-			{ label: $L('Delete page'), command: 'deletepage' },*/
 			Mojo.Menu.editItem,
-			{ label: $L('List Apps...'), command: 'listapps' },
+			{ label: $L('Edit Pages'),
+				items: [
+					{ label: $L('New page'), command: 'newpage' },
+					{ label: $L('Delete page'), command: 'deletepage' }
+				]
+			},
+			{ label: $L('Edit Quick Launch'), command: 'quicklaunch' },
+			{ label: $L('List All Applications'), command: 'listapps' },
 			{ label: $L('Default Applications'), command: 'defaultapps'},
 			Mojo.Menu.helpItem
 		]
@@ -63,6 +68,7 @@ var LauncherAssistant = Class.create({
 	
 	setup: function() {
 		
+		Mojo.listen(this.controller.document, Mojo.Event.activate, this.onActivate.bindAsEventListener(this));
 		Mojo.listen(this.controller.document, Mojo.Event.deactivate, this.onDeactivate.bindAsEventListener(this));
 		Mojo.listen(this.controller.window, 'resize', this.onResize.bindAsEventListener(this));
 		
@@ -121,7 +127,11 @@ var LauncherAssistant = Class.create({
 				case 'defaultapps':
 					ApplicationService.launch(this.deviceInfo.id, this.deviceInfo.defaultAppParams);
 					break;
-				/*
+				case 'quicklaunch':
+					SystemManagerService.showQuickLaunch(true);
+					this.kQuickLaunchHeight = 67;
+					this.adjustLayout();
+					break;
 				case 'newpage':
 					if (this.pageDivs.length < 10) {
 						this.insertPage(this.activePageIndex, true);
@@ -135,7 +145,6 @@ var LauncherAssistant = Class.create({
 						}
 					}
 					break;
-				*/
 			}
 		}
 		else if (event.type === Mojo.Event.commandEnable) {
@@ -164,6 +173,14 @@ var LauncherAssistant = Class.create({
 		this.updatePageIndicators();
 	},
 	
+	/* hide quick launch */
+	onActivate: function(event) {
+		this.kQuickLaunchHeight = 0;
+		this.adjustLayout();
+
+		SystemManagerService.showQuickLaunch(false);
+	},
+	
 	/* clean and hide global search */
 	onDeactivate: function(event) {
 		
@@ -578,14 +595,12 @@ var LauncherAssistant = Class.create({
 			}
 		}
 		
-		SystemManagerService.showQuickLaunch(false);
 		this.globalSearchAssistant.enable(false);
 		
 		var appInfoAssistant = new AppInfoAssistant(appInfo,
 													this.controller,
 													function() {
 														this.appDialog = undefined;
-														SystemManagerService.showQuickLaunch(true);
 														this.globalSearchAssistant.enable(true);
 													}.bind(this));
 
