diff --git a/usr/lib/luna/system/luna-applauncher/app/controllers/launcher-assistant.js.rej b/usr/lib/luna/system/luna-applauncher/app/controllers/launcher-assistant.js.rej
new file mode 100644
index 0000000..69a9fd9
--- /dev/null
+++ b/usr/lib/luna/system/luna-applauncher/app/controllers/launcher-assistant.js.rej
@@ -0,0 +1,71 @@
+--- usr/lib/luna/system/luna-applauncher/app/controllers/launcher-assistant.js
++++ usr/lib/luna/system/luna-applauncher/app/controllers/launcher-assistant.js
+@@ -10,7 +10,7 @@
+ 	 */
+ 	kQuickLaunchHeight: 67,
+ 	
+-	kPageIndicatorSpacing: 6, 	/* spacing between each page indicator */
++	kPageIndicatorSpacing: 1, 	/* spacing between each page indicator */
+ 	kPageMargin: 10, /* 10 pixel margin on each side of a page */
+ 	kPageWidthNoMargin: NaN,
+ 	kPageWidth: NaN,
+@@ -23,8 +23,8 @@
+ 		visible: true,
+ 		label: $L('Launcher'),
+         items: [
+-			/*{ label: $L('New page'), command: 'newpage' },
+-			{ label: $L('Delete page'), command: 'deletepage' },*/
++			{ label: $L('New page'), command: 'newpage' },
++			{ label: $L('Delete page'), command: 'deletepage' },
+ 			Mojo.Menu.editItem,
+ 			{ label: $L('List Apps...'), command: 'listapps' },
+ 			{ label: $L('Default Applications'), command: 'defaultapps'},
+@@ -121,7 +121,7 @@
+ 				case 'defaultapps':
+ 					ApplicationService.launch(this.deviceInfo.id, this.deviceInfo.defaultAppParams);
+ 					break;
+-				/*
++				//start add del page code
+ 				case 'newpage':
+ 					if (this.pageDivs.length < 10) {
+ 						this.insertPage(this.activePageIndex, true);
+@@ -135,7 +135,7 @@
+ 						}
+ 					}
+ 					break;
+-				*/
++				//end add del page code
+ 			}
+ 		}
+ 		else if (event.type === Mojo.Event.commandEnable) {
+@@ -336,9 +336,9 @@
+ 			newAppDiv = $(newAppInfo.launchPointId);
+ 			Mojo.assert(newAppDiv !== undefined, "Application (%d) was inserted into the DOM but unable to be retrieved!", newAppInfo.launchPointId);
+ 			if (isNaN(this.kAppWidth) || isNaN(this.kAppHeight)) {
+-				this.kAppWidth = newAppDiv.getWidth();
+-				this.kAppHeight = newAppDiv.getHeight();
+-				this.kAppsPerRow = Math.round(this.kPageWidthNoMargin / this.kAppWidth);
++				this.kAppWidth = 80;
++				this.kAppHeight = 91;
++				this.kAppsPerRow = 4;
+ 			}
+ 			
+ 			Mojo.listen(newAppDiv, Mojo.Event.tap, this.onAppTapped.bindAsEventListener(this));
+@@ -536,7 +536,7 @@
+ 	/* determines the position of an app element at appIndex within a page */
+ 	calculateAppPosition: function(appIndex) {
+ 		return {
+-			left: (((appIndex % this.kAppsPerRow) * this.kAppWidth) + this.kPageMargin),
++			left: (((appIndex % this.kAppsPerRow) * this.kAppWidth) - 10),
+ 			top: (Math.floor(appIndex/this.kAppsPerRow) * this.kAppHeight)
+ 		};
+ 	},
+@@ -720,7 +720,7 @@
+ 	onAppTapped: function(event) {
+ 		var target = event.currentTarget;
+ 		var appInfo = target._info;
+-		if (event.up && event.up.altKey) {
++		if (event.up && event.up.altKey || event.up && event.up.metaKey) {
+ 			// open app-info dialog
+ 			this.showAppInfoMenu(appInfo);
+ 		} else if (!this.launchRequest) { // are we already trying to launch something?
diff --git a/usr/lib/luna/system/luna-applauncher/app/controllers/launcher-assistant.js.~1~ b/usr/lib/luna/system/luna-applauncher/app/controllers/launcher-assistant.js.~1~
new file mode 100644
index 0000000..e69de29
diff --git a/usr/lib/luna/system/luna-applauncher/stylesheets/launcher.css.rej b/usr/lib/luna/system/luna-applauncher/stylesheets/launcher.css.rej
new file mode 100644
index 0000000..559703b
--- /dev/null
+++ b/usr/lib/luna/system/luna-applauncher/stylesheets/launcher.css.rej
@@ -0,0 +1,108 @@
+--- usr/lib/luna/system/luna-applauncher/stylesheets/launcher.css
++++ usr/lib/luna/system/luna-applauncher/stylesheets/launcher.css
+@@ -25,7 +25,7 @@
+ .launcher_fade_bottom {
+ 	position: absolute;
+ 	bottom: 0px;
+-	background: url(../images/launcher-page-fade-bottom.png) center center no-repeat;
++/*	background: url(../images/launcher-page-fade-bottom.png) center center no-repeat; */
+ 	z-index:20;
+ 	left:0px;
+ 	width:320px;
+@@ -39,7 +39,7 @@
+ 	z-index: 29;
+ 	height: 24px;
+ 	bottom: 1px;
+-	background: url(../images/fade-arrow-down.png) center center no-repeat;
++/*	background: url(../images/fade-arrow-down.png) center center no-repeat; */
+ 	-webkit-palm-mouse-target: ignore;
+ }
+ 
+@@ -45,7 +45,7 @@
+ 
+ .launcher_fade_top {
+ 	position:absolute;
+-	background: url(../images/launcher-page-fade-top.png) center center no-repeat;
++/*	background: url(../images/launcher-page-fade-top.png) center center no-repeat; */
+ 	z-index:20;
+ 	left:0px;
+ 	top:0px;
+@@ -60,7 +60,7 @@
+ 	z-index: 29;
+ 	height: 24px;
+ 	top: 1px;
+-	background: url(../images/fade-arrow-up.png) center center no-repeat;
++/*	background: url(../images/fade-arrow-up.png) center center no-repeat; */
+ 	-webkit-palm-mouse-target: ignore;
+ }
+ 
+@@ -75,7 +75,7 @@
+ #page_indicators {
+ 	position:absolute;
+ 	bottom: 0px;
+-	height:20px;
++	height:7px;
+ 	width: 100%;
+ 	z-index:30;
+ 	-webkit-palm-mouse-target: ignore;
+@@ -86,7 +86,7 @@
+ 	background: url(../images/launcher-page-indicator.png) no-repeat;
+ 	/* left/top provided by JS */
+ 	width:7px;
+-	height:15px;
++	height:7px;
+ 	-webkit-palm-mouse-target: ignore;
+ }
+ 
+@@ -125,8 +125,9 @@
+ 
+ .launcher_page .name {	
+ 	position:absolute;
+-	top: 68px;
+-    width:100px;
++	top: 61px;
++	left:10px;
++    width:80px;
+ 	height: 34px;
+ 	max-height: 34px;
+     clear:both;
+@@ -130,7 +131,7 @@
+ 	height: 34px;
+ 	max-height: 34px;
+     clear:both;
+-	line-height:16px;
++	line-height:13px;
+     color:white;
+     text-align:center;
+ 	font-weight:bold;
+@@ -134,7 +135,7 @@
+     color:white;
+     text-align:center;
+ 	font-weight:bold;
+-    font-size: 14px;
++    font-size: 12px;
+ 	overflow: hidden;
+ 	text-overflow: ellipsis;
+ 	z-index:5;
+@@ -263,8 +264,8 @@
+ }
+ 
+ .draggable {
+-	width:64px;
+-	height:64px;
++	width:60px;
++	height:60px;
+ 	margin: 0 auto;
+ 	-webkit-user-drag: any;
+ 	-webkit-user-select: none;
+@@ -304,8 +305,8 @@
+ 
+ #app-icon {
+ 	float:left;
+-	width:64px;
+-	height:64px;
++	width:60px;
++	height:60px;
+ 	background: center center no-repeat;
+ }
+ 
diff --git a/usr/lib/luna/system/luna-applauncher/stylesheets/launcher.css.~1~ b/usr/lib/luna/system/luna-applauncher/stylesheets/launcher.css.~1~
new file mode 100644
index 0000000..a550b86
--- /dev/null
+++ b/usr/lib/luna/system/luna-applauncher/stylesheets/launcher.css.~1~
@@ -0,0 +1,67 @@
+/* Copyright 2010 Palm, Inc.  All rights reserved. */
+
+/* launcher-helper styles */
+
+.launcher-scrim {
+	background: url(../images/launcher-scrim-clear.png) left center repeat-x;
+	position: fixed;
+	top: 0;
+	left: 0;
+	bottom: 0;
+	right: 0;
+	z-index: 98000;
+}
+
+.editable-page-title input {
+	border-width: 22px;
+	-webkit-box-sizing: border-box;
+	-webkit-border-image: url(../images/launcher-edit-title-pill.png) 22 stretch;
+	background: none;
+	padding: 0;
+	margin: 0;
+	position: fixed;
+	top: 2px;
+	left: 45px;
+	width: 100%;
+	max-height: 48px;
+	
+	text-align: center;
+	color: #fff;
+	font-size: 18px;
+	font-weight: bold;
+}
+
+/* App Info */
+
+#app-icon {
+        float:left;
+        width:64px;
+        height:64px;
+        background: center center no-repeat;
+}
+
+#app-title,
+#app-version,
+#app-size {
+	position: absolute;
+	left: 74px;
+    font-size: 14px;
+	max-width: 200px;
+    overflow: hidden;
+    text-overflow: ellipsis;
+    white-space: nowrap;
+}
+
+#app-title {    
+        top: -1px; 
+        font-weight: bold;
+        font-size: 20px;
+}
+
+#app-version { 
+        bottom: 21px;
+}
+
+#app-size {  
+        bottom: 1px;
+}
