diff --git a/usr/lib/luna/system/luna-applauncher/app/controllers/launcher-assistant.js b/usr/lib/luna/system/luna-applauncher/app/controllers/launcher-assistant.js
index 9057138..036480e 100644
--- a/usr/lib/luna/system/luna-applauncher/app/controllers/launcher-assistant.js
+++ b/usr/lib/luna/system/luna-applauncher/app/controllers/launcher-assistant.js
@@ -720,7 +720,7 @@ var LauncherAssistant = Class.create({
 	onAppTapped: function(event) {
 		var target = event.currentTarget;
 		var appInfo = target._info;
-		if (event.up && event.up.altKey) {
+		if (event.up && event.up.altKey || event.up && event.up.metaKey) {
 			// open app-info dialog
 			this.showAppInfoMenu(appInfo);
 		} else if (!this.launchRequest) { // are we already trying to launch something?
