diff --git a/usr/palm/applications/com.palm.app.devmodeswitcher/appinfo.json.rej b/usr/palm/applications/com.palm.app.devmodeswitcher/appinfo.json.rej
new file mode 100644
index 0000000..7f47f92
--- /dev/null
+++ b/usr/palm/applications/com.palm.app.devmodeswitcher/appinfo.json.rej
@@ -0,0 +1,11 @@
+--- usr/palm/applications/com.palm.app.devmodeswitcher/appinfo.json
++++ usr/palm/applications/com.palm.app.devmodeswitcher/appinfo.json
+@@ -5,7 +5,7 @@
+ 	"id": "com.palm.app.devmodeswitcher",
+ 	"icon": "icon.png",
+ 	"noWindow": false,
+-	"visible": false,
++	"visible": true,
+ 	"version": "1.0.2",
+ 	"vendor":"palm"
+ }
diff --git a/usr/palm/applications/com.palm.app.devmodeswitcher/appinfo.json.~1~ b/usr/palm/applications/com.palm.app.devmodeswitcher/appinfo.json.~1~
new file mode 100644
index 0000000..7d9da85
--- /dev/null
+++ b/usr/palm/applications/com.palm.app.devmodeswitcher/appinfo.json.~1~
@@ -0,0 +1,10 @@
+{
+	"title": "DeveloperMode",
+	"type": "web",
+	"main": "index.html",
+	"id": "com.palm.app.devmodeswitcher",
+	"icon": "icon.png",
+	"noWindow": false,
+	"visible": false	
+}
+
