diff --git a/etc/palm/launcher3/layoutSettings.conf b/etc/palm/launcher3/layoutSettings.conf
index 2f50fbc..bb2dffb 100644
--- a/etc/palm/launcher3/layoutSettings.conf
+++ b/etc/palm/launcher3/layoutSettings.conf
@@ -1,3 +1,7 @@
+[QuickLaunch]
+quickLaunchBarUseAbsoluteSize=true
+quickLaunchBarHeightAbsolute=1
+quickLaunchItemAreaVerticalOffsetPx=47
 [DoneButton]
 doneButtonTextVerticalPosAdjust=-3
 doneButtonHorizontalPositionAdjustPx=42
\ No newline at end of file
