--- .orig/usr/lib/luna/system/luna-applauncher/app/controllers/launcher-assistant.js
+++ /usr/lib/luna/system/luna-applauncher/app/controllers/launcher-assistant.js
@@ -8,7 +8,7 @@
 	 * hardcoding the quick launch height until we can find a better way. 
 	 * This value = ql graphic height - the offset from the top (small fade area)
 	 */
-	kQuickLaunchHeight: 67,
+	kQuickLaunchHeight: 0,
 	
 	kPageIndicatorSpacing: 6, 	/* spacing between each page indicator */
 	kPageMargin: 10, /* 10 pixel margin on each side of a page */
\ No newline at end of file
