diff --git a/usr/lib/luna/system/luna-applauncher/app/controllers/launcher-assistant.js b/usr/lib/luna/system/luna-applauncher/app/controllers/launcher-assistant.js
index 64244b6..e68c9d2 100644
--- a/usr/lib/luna/system/luna-applauncher/app/controllers/launcher-assistant.js
+++ b/usr/lib/luna/system/luna-applauncher/app/controllers/launcher-assistant.js
@@ -157,6 +157,10 @@ var LauncherAssistant = Class.create({
 	
 	/* keep track of which page we are on */
 	onPageChange: function(event) {
+    var scroller = this.getPageScroller(this.activePageIndex);
+    if (scroller && scroller.mojo) {
+      scroller.mojo.revealTop(0);
+    }
 		this.activePageIndex = event.value;
 		this.updatePageIndicators();
 	},
