diff --git a/usr/lib/luna/system/luna-applauncher/app/controllers/launcher-assistant.js b/usr/lib/luna/system/luna-applauncher/app/controllers/launcher-assistant.js
index 9057138..88eed9d 100644
--- a/usr/lib/luna/system/luna-applauncher/app/controllers/launcher-assistant.js
+++ b/usr/lib/luna/system/luna-applauncher/app/controllers/launcher-assistant.js
@@ -178,6 +178,25 @@ var LauncherAssistant = Class.create({
 		SystemManagerService.showQuickLaunch(true);
 		
 		this.launchRequest = undefined;
+
+		/**
+		 * Calculate middle page. For an even number of pages,
+		 * this will be the lower side of the mid-point.
+		 */
+		var numPages=this.pagesModel.getNumPages();
+		var middlePage = (numPages % 2) ? (numPages - 1) / 2: numPages / 2 - 1;
+
+		/* Reset launcher to middle page - seems we need to do scrollTo AND setSnapIndex to be fully compatible with Wrap Pages */
+		$('launcher_root').mojo.scrollTo((middlePage) * -this.kPageWidth, undefined, false, true);
+		this.activePageIndex = Math.abs($('launcher_root').mojo.getScrollPosition().left) / this.kPageWidth;
+		$('launcher_root').mojo.setSnapIndex(middlePage, false);
+		this.updatePageIndicators();
+
+		/* Jump scroll position back to top for all pages */
+		for (var i=0; i<numPages; i++) {
+			var scroller = this.getPageScroller(i);
+			if (scroller && scroller.mojo) { scroller.mojo.revealTop(); }
+		}
 	},
 	
 	onResize: function(event) {
@@ -618,6 +637,18 @@ var LauncherAssistant = Class.create({
 		
 		this.globalSearchAssistant.enable(true);
 
+		/**
+		 * Calculate middle page. For an even number of pages,
+		 * this will be the lower side of the mid-point.
+		 */
+		var middlePage = (numPages % 2) ? (numPages - 1) / 2: numPages / 2 - 1;
+
+		/* Set launcher to middle page - seems we need to do scrollTo AND setSnapIndex to be fully compatible with Wrap Pages */
+		$('launcher_root').mojo.scrollTo((middlePage) * -this.kPageWidth, undefined, false, true);
+		this.activePageIndex = Math.abs($('launcher_root').mojo.getScrollPosition().left) / this.kPageWidth;
+		$('launcher_root').mojo.setSnapIndex(middlePage, false);
+		this.updatePageIndicators();
+
 		PalmSystem.launcherReady();
 	},
 	
