diff --git a/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js b/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
index 6fdbb39..0b09eda 100644
--- a/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
+++ b/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
@@ -61,6 +61,10 @@ var BarAssistant = Class.create({
 		this.storageInit();
 		
 		//Initialize and Subscribe for Power Off Notification.
+		this.controller.serviceRequest('palm://com.palm.systemmanager', {
+			method: 'systemUi',
+			parameters: {launcher:true}
+		});
 		this.powerOffInit();
 		
 		//Subscribe to Powerd Notification.
@@ -2391,6 +2395,22 @@ powerOffInit: function() {
 		},
 		onSuccess: this.powerOffHandleNotifications.bind(this)
 	});
+    this.displayChanged = this.controller.serviceRequest('palm://com.palm.display/control', {
+    		method:'status',
+ 		parameters:{
+ 			subscribe:true
+ 		},
+ 		onSuccess: function(response) {
+ 			if(response && response.event) {
+ 				if(response.event=="displayOn" && this.foregroundAppID==null) {
+ 					this.controller.serviceRequest('palm://com.palm.systemmanager', {
+						method: 'systemUi',
+						parameters: {launcher:true}
+					});
+ 				}
+ 			}
+ 		}.bind(this)
+	});
 },
 
 
