Add / Delete Pages in the Launcher
Index: /usr/lib/luna/system/luna-applauncher/app/controllers/launcher-assistant.js
===================================================================
--- .orig/usr/lib/luna/system/luna-applauncher/app/controllers/launcher-assistant.js
+++ /usr/lib/luna/system/luna-applauncher/app/controllers/launcher-assistant.js
@@ -23,8 +23,8 @@ var LauncherAssistant = Class.create({
 		visible: true,
 		label: $L('Launcher'),
         items: [
-			/*{ label: $L('New page'), command: 'newpage' },
-			{ label: $L('Delete page'), command: 'deletepage' },*/
+			{ label: $L('New page'), command: 'newpage' },
+			{ label: $L('Delete page'), command: 'deletepage' },
 			Mojo.Menu.editItem,
 			{ label: $L('List Apps...'), command: 'listapps' },
 			Mojo.Menu.helpItem
@@ -117,7 +117,6 @@ var LauncherAssistant = Class.create({
 				case 'listapps':
 					ApplicationService.launch(this.deviceInfo.id, this.deviceInfo.params);
 					break;
-				/*
 				case 'newpage':
 					if (this.pageDivs.length < 10) {
 						this.insertPage(this.activePageIndex, true);
@@ -131,7 +130,6 @@ var LauncherAssistant = Class.create({
 						}
 					}
 					break;
-				*/
 			}
 		}
 		else if (event.type === Mojo.Event.commandEnable) {
