diff --git a/usr/lib/luna/system/luna-applauncher/app/controllers/launcher-assistant.js b/usr/lib/luna/system/luna-applauncher/app/controllers/launcher-assistant.js
index 9057138..135562e 100644
--- a/usr/lib/luna/system/luna-applauncher/app/controllers/launcher-assistant.js
+++ b/usr/lib/luna/system/luna-applauncher/app/controllers/launcher-assistant.js
@@ -94,6 +94,10 @@ var LauncherAssistant = Class.create({
 		
 		// setup the initial dimensions for launcher/global-search
 		this.adjustLayout();
+		
+		// add a separate listener for resetting the scroll position on page changes, and also on deactivate
+		Mojo.listen($('launcher_root'), Mojo.Event.propertyChange, this.resetScrollPosition.bindAsEventListener(this));
+		Mojo.listen(this.controller.document, Mojo.Event.deactivate, this.resetScrollPosition.bindAsEventListener(this));
 	},
 	
 	ready: function() {		
@@ -184,6 +188,12 @@ var LauncherAssistant = Class.create({
 		this.adjustLayout();
 	},
 
+	resetScrollPosition: function(event) {
+		var pageIndex = (event.type === 'mojo-event-deactivate') ? this.activePageIndex : event.value;
+		var scroller = this.getPageScroller(pageIndex);
+		if (scroller && scroller.mojo) { scroller.mojo.revealTop(); }
+	},
+
 	/* remove page data and their corresponding HTML */
 	deleteAllPages: function() {
 		$('pages_container').update("");
