diff --git a/etc/palm/launcher3/launcher_icon_geom_settings.conf b/etc/palm/launcher3/launcher_icon_geom_settings.conf
index 8194eea..23335a6 100644
--- a/etc/palm/launcher3/launcher_icon_geom_settings.conf
+++ b/etc/palm/launcher3/launcher_icon_geom_settings.conf
@@ -1,8 +1,15 @@
 [Main]
-RemoveDeleteDecoratorHorizontalOffsetFromGeomOrigin=-50
-RemoveDeleteDecoratorVerticalOffsetFromGeomOrigin=-50
+MainIconBoxVerticalProportionToTotalGeom=1.6
+FrameBoxVerticalProportionToTotalGeom=.8
+UseAbsoluteMainIconGeom=true
+MainIconAbsoluteGeomSizeHeightPx=60
+UseAbsoluteFrameGeom=true
+FrameBoxAbsoluteGeomSizeHeightPx=1
+UseAbsoluteRemoceDeleteDecoratorOffsetFromGeomOrigin=true
+RemoveDeleteDecoratorHorizontalOffsetFromGeomOrigin=-35
+RemoveDeleteDecoratorVerticalOffsetFromGeomOrigin=-35
 InstallStatusDecoratorHorizontalOffsetFromGeomOrigin=50
 InstallStatusDecoratorVerticalOffsetFromGeomOrigin=-50
-MainIconVerticalOffsetFromGeomOriginPx=-11
+MainIconVerticalOffsetFromGeomOriginPx=-17
 [Label]
 LabelVerticalSpacingFromMainIcon=2
\ No newline at end of file
diff --git a/etc/palm/launcher3/launcher_icon_layoutsettings.conf b/etc/palm/launcher3/launcher_icon_layoutsettings.conf
index 20d6b5c..2d025f3 100644
--- a/etc/palm/launcher3/launcher_icon_layoutsettings.conf
+++ b/etc/palm/launcher3/launcher_icon_layoutsettings.conf
@@ -1,9 +1,13 @@
 [ReorderableLayout]
-InterRowSpaceInPixels=10
-IconHorizontalSpaceAdjustInPixels=12
-MaxIconsPerRow=7
-RowLeftMarginInPixels=27
+InterRowSpaceInPixels=0
+IconHorizontalSpaceAdjustInPixels=11
+MaxIconsPerRow=9
+RowLeftMarginInPixels=1
+RowTopMarginInPixels=1
+UseFixedLengthCellSize=true
+FixedIconCellHeight=104
+FixedIconCellWidth=100
 EmptyPageTextBoxWidthPx=350
-EmptyPageTextBoxCenterVerticalOffsetPx=190
+EmptyPageTextBoxCenterVerticalOffsetPx=350
 EmptyPageTextFontSizePx=18
 EmptyPageTextColor=#aaaaaa
\ No newline at end of file
diff --git a/etc/palm/launcher3/layoutSettings.conf b/etc/palm/launcher3/layoutSettings.conf
index 2f50fbc..2a6c0dd 100644
--- a/etc/palm/launcher3/layoutSettings.conf
+++ b/etc/palm/launcher3/layoutSettings.conf
@@ -1,3 +1,10 @@
+[QuickLaunch]
+quickLaunchBarLauncherAccessButtonVerticalOffsetPx=-61
+quickLaunchBarUseAbsoluteSize=true
+quickLaunchBarHeightAbsolute=65
+quickLaunchItemAreaVerticalOffsetPx=50
+quickLaunchItemAreaHorizontalOffsetPcx=0
+quickLaunchMaxItems=8
 [DoneButton]
 doneButtonTextVerticalPosAdjust=-3
 doneButtonHorizontalPositionAdjustPx=42
\ No newline at end of file
