diff --git a/usr/lib/luna/system/luna-applauncher/app/controllers/launcher-assistant.js b/usr/lib/luna/system/luna-applauncher/app/controllers/launcher-assistant.js
index 9057138..91a842e 100644
--- a/usr/lib/luna/system/luna-applauncher/app/controllers/launcher-assistant.js
+++ b/usr/lib/luna/system/luna-applauncher/app/controllers/launcher-assistant.js
@@ -23,8 +23,8 @@ var LauncherAssistant = Class.create({
 		visible: true,
 		label: $L('Launcher'),
         items: [
-			/*{ label: $L('New page'), command: 'newpage' },
-			{ label: $L('Delete page'), command: 'deletepage' },*/
+			{ label: $L('New page'), command: 'newpage' },
+			{ label: $L('Delete page'), command: 'deletepage' },
 			Mojo.Menu.editItem,
 			{ label: $L('List Apps...'), command: 'listapps' },
 			{ label: $L('Default Applications'), command: 'defaultapps'},
@@ -121,7 +121,6 @@ var LauncherAssistant = Class.create({
 				case 'defaultapps':
 					ApplicationService.launch(this.deviceInfo.id, this.deviceInfo.defaultAppParams);
 					break;
-				/*
 				case 'newpage':
 					if (this.pageDivs.length < 10) {
 						this.insertPage(this.activePageIndex, true);
@@ -135,7 +134,6 @@ var LauncherAssistant = Class.create({
 						}
 					}
 					break;
-				*/
 			}
 		}
 		else if (event.type === Mojo.Event.commandEnable) {
