diff --git a/usr/lib/luna/system/luna-applauncher/app/controllers/launcher-assistant.js b/usr/lib/luna/system/luna-applauncher/app/controllers/launcher-assistant.js
index 9057138..019fb80 100644
--- a/usr/lib/luna/system/luna-applauncher/app/controllers/launcher-assistant.js
+++ b/usr/lib/luna/system/luna-applauncher/app/controllers/launcher-assistant.js
@@ -178,6 +178,19 @@ var LauncherAssistant = Class.create({
 		SystemManagerService.showQuickLaunch(true);
 		
 		this.launchRequest = undefined;
+
+		/* Reset launcher to first page - seems we need to do scrollTo AND setSnapIndex to be fully compatible with Wrap Pages */
+		$('launcher_root').mojo.scrollTo(0, undefined, false, true);
+		this.activePageIndex = Math.abs($('launcher_root').mojo.getScrollPosition().left) / this.kPageWidth;
+		$('launcher_root').mojo.setSnapIndex(0, false);
+		this.updatePageIndicators();
+
+		/* Jump scroll position back to top for all pages */
+		var numPages=this.pagesModel.getNumPages();
+		for (var i=0; i<numPages; i++) {
+			var scroller = this.getPageScroller(i);
+			if (scroller && scroller.mojo) { scroller.mojo.revealTop(); }
+		}
 	},
 	
 	onResize: function(event) {
