diff --git a/etc/palm/launcher3/layoutSettings.conf b/etc/palm/launcher3/layoutSettings.conf
index 2f50fbc..c9b7944 100644
--- a/etc/palm/launcher3/layoutSettings.conf
+++ b/etc/palm/launcher3/layoutSettings.conf
@@ -1,3 +1,6 @@
+[QuickLaunch]
+quickLaunchBarLauncherAccessButtonVerticalOffsetPx=100
+quickLaunchMaxItems=8
 [DoneButton]
 doneButtonTextVerticalPosAdjust=-3
 doneButtonHorizontalPositionAdjustPx=42
\ No newline at end of file
