diff --git a/usr/lib/luna/system/luna-applauncher/app/controllers/global-search-assistant.js b/usr/lib/luna/system/luna-applauncher/app/controllers/global-search-assistant.js
index e4268cf..889875f 100644
--- a/usr/lib/luna/system/luna-applauncher/app/controllers/global-search-assistant.js
+++ b/usr/lib/luna/system/luna-applauncher/app/controllers/global-search-assistant.js
@@ -233,7 +233,7 @@ GlobalSearchAssistant = Class.create({
 			this.launcherHiddenAtFirstKeydown = false;
 		}
 		else {
-			SystemManagerService.showQuickLaunch(true);
+			SystemManagerService.showQuickLaunch(false);
 		}
 	},
 	
diff --git a/usr/lib/luna/system/luna-applauncher/app/controllers/launcher-assistant.js b/usr/lib/luna/system/luna-applauncher/app/controllers/launcher-assistant.js
index bebbad2..afefa2e 100644
--- a/usr/lib/luna/system/luna-applauncher/app/controllers/launcher-assistant.js
+++ b/usr/lib/luna/system/luna-applauncher/app/controllers/launcher-assistant.js
@@ -8,7 +8,7 @@ var LauncherAssistant = Class.create({
 	 * hardcoding the quick launch height until we can find a better way. 
 	 * This value = ql graphic height - the offset from the top (small fade area)
 	 */
-	kQuickLaunchHeight: 67,
+	kQuickLaunchHeight: 0,
 	
 	kPageIndicatorSpacing: 6, 	/* spacing between each page indicator */
 	kPageMargin: 10, /* 10 pixel margin on each side of a page */
@@ -590,7 +590,7 @@ var LauncherAssistant = Class.create({
 													this.controller,
 													function() {
 														delete this.appDialog;
-														SystemManagerService.showQuickLaunch(true);
+														SystemManagerService.showQuickLaunch(false);
 														this.globalSearchAssistant.enable(true);
 													}.bind(this));
 
diff --git a/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js b/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
index 6fdbb39..0709d07 100644
--- a/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
+++ b/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
@@ -260,6 +260,13 @@ handleAppName:function(payload){
 			}
 			$('appname').innerHTML = $L("Preferences");
 		}
+		else if (payload.id == "com.palm.launcher"){
+			this.controller.serviceRequest('palm://com.palm.systemmanager', {
+				method: 'systemUi',
+				parameters: {"quicklaunch": false}
+			});
+			$('appname').innerHTML = payload.appmenu || payload.title;
+		}
 		else 
 			$('appname').innerHTML = payload.appmenu || payload.title;
 		$('appname').show();
