--- /usr/lib/luna/system/luna-applauncher/app/controllers/launcher-assistant.js.orig	Sat Dec  5 11:48:12 2009
+++ /usr/lib/luna/system/luna-applauncher/app/controllers/launcher-assistant.js	Thu Dec  3 19:42:03 2009
@@ -98,5 +98,9 @@
 		// setup the initial dimensions for launcher/global-search
 		this.adjustLayout();
+		
+		// add a separate listener for resetting the scroll position on page changes, and also on deactivate
+		Mojo.listen($('launcher_root'), Mojo.Event.propertyChange, this.resetScrollPosition.bindAsEventListener(this));
+		Mojo.listen(this.controller.document, Mojo.Event.deactivate, this.resetScrollPosition.bindAsEventListener(this));
 	},
 	
 	ready: function() {	
@@ -188,6 +188,12 @@
 	onResize: function(event) {
 		this.adjustLayout();
 	},
 
+	resetScrollPosition: function(event) {
+		var pageIndex = (event.type === 'mojo-event-deactivate') ? this.activePageIndex : event.value;
+		var scroller = this.getPageScroller(pageIndex);
+		if (scroller && scroller.mojo) { scroller.mojo.revealTop(); }
+	},
+
 	/* remove page data and their corresponding HTML */
 	deleteAllPages: function() {

