--- .orig/usr/lib/luna/system/luna-applauncher/app/controllers/launcher-assistant.js
+++ /usr/lib/luna/system/luna-applauncher/app/controllers/launcher-assistant.js
@@ -630,7 +630,7 @@
 	onAppTapped: function(event) {
 		var target = event.currentTarget;
 		var appInfo = target._info;
-		if (event.up && event.up.altKey) {
+		if (event.up && event.up.altKey || event.up && event.up.metaKey) {
 			// open app-info dialog
 			this.showAppInfoMenu(appInfo);
 		} else if (!this.feedbackTimeout) { // are we already trying to launch something?
