diff --git a/etc/PmLog.conf b/etc/PmLog.conf
index 77409fe..98f79cb 100644
--- a/etc/PmLog.conf
+++ b/etc/PmLog.conf
@@ -146,10 +146,10 @@ Rule2=kern.*,kernlog
 Rule3=*.!err.UpdateDaemon,-stdlog
 
 [CONTEXT=LunaSysMgr]
-Rule1=*.*,stdlog
+Rule1=*.err,stdlog
 
 [CONTEXT=LunaSysMgrJS]
-Rule1=*.*,stdlog
+Rule1=*.err,stdlog
 
 #==========================================================
 # For the TIL.CCP context, only put errors in
@@ -157,7 +157,7 @@ Rule1=*.*,stdlog
 #==========================================================
 [CONTEXT=TIL.CCP]
 Rule1=*.err,stdlog
-Rule2=*,ccplog
+#Rule2=*,ccplog
 BufferSize=2K
 FlushLevel=err
 
@@ -167,7 +167,7 @@ FlushLevel=err
 #==========================================================
 [CONTEXT=TIL.LOG]
 Rule1=*.err,stdlog
-Rule2=*.*,tillog
+#Rule2=*.*,tillog
 
 #==========================================================
 # For the powermond context, only put errors in
@@ -175,7 +175,7 @@ Rule2=*.*,tillog
 #==========================================================
 [CONTEXT=powermond]
 Rule1=*.err,stdlog
-Rule2=*.warning,powermond
+#Rule2=*.warning,powermond
 BufferSize=2K
 FlushLevel=err
 
@@ -185,7 +185,7 @@ FlushLevel=err
 #==========================================================
 [CONTEXT=uploadd]
 Rule1=*.err,stdlog
-Rule2=*.warning,uploadd
+#Rule2=*.warning,uploadd
 BufferSize=2K
 FlushLevel=err
 
@@ -193,7 +193,7 @@ FlushLevel=err
 # Utility used in logging
 #==========================================================
 [CONTEXT=PmLogCtl]
-Rule1=*.*,stdlog
+Rule1=*.err,stdlog
 BufferSize=2K
 FlushLevel=err
 
@@ -203,25 +203,25 @@ FlushLevel=err
 #==========================================================
 [CONTEXT=PmBtEngine]
 Rule1=*.err,stdlog
-Rule2=*.*,btlog
+#Rule2=*.*,btlog
 
 [CONTEXT=PmBtStack]
 Rule1=*.err,stdlog
-Rule2=*.*,btlog
+#Rule2=*.*,btlog
 
 [CONTEXT=PmBtContacts]
 Rule1=*.err,btlog
-Rule2=*.*,btlog
+#Rule2=*.*,btlog
 
 [CONTEXT=BluetoothMonitor]
 Rule1=*.err,stdlog
-Rule2=*.*,btlog
+#Rule2=*.*,btlog
 
 #==========================================================
 # PmWanDaemon
 #==========================================================
 [CONTEXT=wan.mgr]
-Rule1=*,stdlog
+Rule1=*.err,stdlog
 BufferSize=8K
 FlushLevel=crit
 
@@ -229,7 +229,7 @@ FlushLevel=crit
 # PmWiFiService
 #==========================================================
 [CONTEXT=com.palm.wifi]
-Rule1=*,stdlog
+Rule1=*.err,stdlog
 BufferSize=2K
 FlushLevel=err
 
@@ -237,7 +237,7 @@ FlushLevel=err
 # Media       
 #==========================================================
 [CONTEXT=media]
-Rule1=*,stdlog
+Rule1=*.err,stdlog
 BufferSize=2K
 FlushLevel=err
 
@@ -245,7 +245,7 @@ FlushLevel=err
 # Wireless System Framework
 #==========================================================
 [CONTEXT=WSF.MessageMover]
-Rule1=*,stdlog
+Rule1=*.err,stdlog
 BufferSize=2K
 FlushLevel=err
 
@@ -255,81 +255,81 @@ FlushLevel=err
 #==========================================================
 [CONTEXT=wan.blade]
 Rule1=*.err,stdlog
-Rule2=*.*,wanlog
+#Rule2=*.*,wanlog
 
 [CONTEXT=wan.ipc]
 Rule1=*.err,stdlog
-Rule2=*.*,wanlog
+#Rule2=*.*,wanlog
 
 [CONTEXT=wan.pdpstate]
 Rule1=*.err,stdlog
-Rule2=*.*,wanlog
+#Rule2=*.*,wanlog
 
 [CONTEXT=wan.modem]
 Rule1=*.err,stdlog
-Rule2=*.*,wanlog
+#Rule2=*.*,wanlog
 
 [CONTEXT=wan.pdpstatemgr]
 Rule1=*.err,stdlog
-Rule2=*.*,wanlog
+#Rule2=*.*,wanlog
 
 [CONTEXT=wan.pdpslot]
 Rule1=*.err,stdlog
-Rule2=*.*,wanlog
+#Rule2=*.*,wanlog
 
 [CONTEXT=wan.profmgr]
 Rule1=*.err,stdlog
-Rule2=*.*,wanlog
+#Rule2=*.*,wanlog
 
 [CONTEXT=wan.radio]
 Rule1=*.err,stdlog
-Rule2=*.*,wanlog
+#Rule2=*.*,wanlog
 
 [CONTEXT=wan.connector]
 Rule1=*.err,stdlog
-Rule2=*.*,wanlog
+#Rule2=*.*,wanlog
 
 [CONTEXT=wan.auto]
 Rule1=*.err,stdlog
-Rule2=*.*,wanlog
+#Rule2=*.*,wanlog
 
 [CONTEXT=wan.settings]
 Rule1=*.err,stdlog
-Rule2=*.*,wanlog
+#Rule2=*.*,wanlog
 
 [CONTEXT=wan.radiopdp]
 Rule1=*.err,stdlog
-Rule2=*.*,wanlog
+#Rule2=*.*,wanlog
 
 [CONTEXT=wan.cdb]
 Rule1=*.err,stdlog
-Rule2=*.*,wanlog
+#Rule2=*.*,wanlog
 
 [CONTEXT=wan.power]
 Rule1=*.err,stdlog
-Rule2=*.*,wanlog
+#Rule2=*.*,wanlog
 
 [CONTEXT=wan.ip]
 Rule1=*.err,stdlog
-Rule2=*.*,wanlog
+#Rule2=*.*,wanlog
 
 [CONTEXT=wan.ppp]
 Rule1=*.err,stdlog
-Rule2=*.*,wanlog
+#Rule2=*.*,wanlog
 
 [CONTEXT=wan.qmi]
 Rule1=*.err,stdlog
-Rule2=*.*,wanlog
+#Rule2=*.*,wanlog
 
 [CONTEXT=wan.dss]
 Rule1=*.err,stdlog
-Rule2=*.*,wanlog
+#Rule2=*.*,wanlog
 
 #==========================================================
 # Passthru service
 #==========================================================
 [CONTEXT=passthrud]
-Rule1=*.*,passthrulog
+Rule1=*.err,passthrulog
 BufferSize=2M
 FlushLevel=info
 
diff --git a/etc/PmLogContexts.conf b/etc/PmLogContexts.conf
index 5824f33..4326dd1 100644
--- a/etc/PmLogContexts.conf
+++ b/etc/PmLogContexts.conf
@@ -52,32 +52,32 @@ LogToConsole=false
 <global>=err
 TIL.CCP=err
 TIL.LOG=err
-PmLogCtl=debug
+PmLogCtl=err
 com.palm.wifi=err
 media=err
-wan.mgr=notice
-WSF.MessageMover=warning
-wan.blade=warning
-wan.ipc=warning
-wan.pdpstate=warning
-wan.modem=warning
-wan.pdpstatemgr=warning
-wan.pdpslot=warning
-wan.profmgr=warning
-wan.radio=warning
-wan.connector=notice
-wan.auto=warning
-wan.settings=warning
-wan.radiopdp=warning
-wan.cdb=warning
-wan.power=warning
-wan.ip=warning
-wan.ppp=warning
-wan.qmi=warning
-wan.dss=warning
-PmBtEngine=debug
-PmBtStack=debug
-PmBtContacts=debug
-BluetoothMonitor=debug
-passthrud=info
-powerd=info
+wan.mgr=err
+WSF.MessageMover=err
+wan.blade=err
+wan.ipc=err
+wan.pdpstate=err
+wan.modem=err
+wan.pdpstatemgr=err
+wan.pdpslot=err
+wan.profmgr=err
+wan.radio=err
+wan.connector=err
+wan.auto=err
+wan.settings=err
+wan.radiopdp=err
+wan.cdb=err
+wan.power=err
+wan.ip=err
+wan.ppp=err
+wan.qmi=err
+wan.dss=err
+PmBtEngine=err
+PmBtStack=err
+PmBtContacts=err
+BluetoothMonitor=err
+passthrud=err
+powerd=err
diff --git a/etc/rdxd.conf b/etc/rdxd.conf
index 3cbd02f..78475b2 100644
--- a/etc/rdxd.conf
+++ b/etc/rdxd.conf
@@ -8,7 +8,7 @@
 #TODO: document this
 
 [rdxd]
-AutoUpload=true
+AutoUpload=false
 MaxPendingReports=100
 
 #TODO: document this
diff --git a/etc/uploadd.conf b/etc/uploadd.conf
index 363f82b..eb329ab 100644
--- a/etc/uploadd.conf
+++ b/etc/uploadd.conf
@@ -6,14 +6,14 @@
 
 [uploadd]
 MinTimeBetweenUploads=10
-MaxUploadsPerDay=20
+MaxUploadsPerDay=0
 MaxUploadSize=2M
 DefaultServer=rdx
 
 [SERVER=rdx]
-RepositoryURL=https://<HOST>/palmcsext/prefRequest?prefkey=APPLICATIONS,RDX_SRV
-UploadURL=https://<HOST>/palmcsext/RDFileReceiver
+RepositoryURL=http://127.0.0.1
+UploadURL=http://127.0.0.1
 
 [SERVER=context]
-RepositoryURL=https://<HOST>/palmcsext/prefRequest?prefkey=APPLICATIONS,RDX_SRV
-UploadURL=https://<HOST>//palmcsext/RDFileReceiver
+RepositoryURL=http://127.0.0.1
+UploadURL=http://127.0.0.1
