diff --git a/usr/palm/applications/com.palm.app.exhibitionpreferences/app/assistants/preferences-assistant.js b/usr/palm/applications/com.palm.app.exhibitionpreferences/app/assistants/preferences-assistant.js
index 932a4a7..24be8e2 100644
--- a/usr/palm/applications/com.palm.app.exhibitionpreferences/app/assistants/preferences-assistant.js
+++ b/usr/palm/applications/com.palm.app.exhibitionpreferences/app/assistants/preferences-assistant.js
@@ -40,6 +40,22 @@ PreferencesAssistant.prototype.setup = function () {
 	
 	this.getDockAppsList();
 	
+	this.controller.serviceRequest("palm://org.webosinternals.tweaks.prefs/", {
+		method: "get", 
+		parameters: {
+			owner: "org.webosinternals.patches.autostart-exhibition",
+			keys: ["autostartTweak"]
+		},
+		onSuccess: function(response) {
+			if((response) && (response.autostartTweak)) {
+				this.controller.window.setTimeout(this.startExhibitionMode.bind(this), 200);
+			}
+		}.bind(this),
+		onFailure: function(error) {
+			this.controller.window.setTimeout(this.startExhibitionMode.bind(this), 200);
+		}.bind(this)
+	});
+	
 };
 
 PreferencesAssistant.prototype.activate = function() {
