diff --git a/usr/palm/applications/com.palm.app.exhibitionpreferences/source/ExhibitionPreferencesApp.js b/usr/palm/applications/com.palm.app.exhibitionpreferences/source/ExhibitionPreferencesApp.js
index be55b6b..479a5f0 100644
--- a/usr/palm/applications/com.palm.app.exhibitionpreferences/source/ExhibitionPreferencesApp.js
+++ b/usr/palm/applications/com.palm.app.exhibitionpreferences/source/ExhibitionPreferencesApp.js
@@ -29,6 +29,7 @@ enyo.kind({
 			{name:"setDockEnabled"},
 			{name:"appLaunch", method:"open"}
 		]},
+		{kind:"PalmService", name:"getTweaks", service:"palm://org.webosinternals.tweaks.prefs/", method:"get", onSuccess:"gotTweaks", onFailure:"enterExhibitionMode"},
 		{kind:"PalmService", name:"startExhitbitionMode", service:"palm://com.palm.display/control/", method:"setState"}        
 	],
 	create:function() {
@@ -36,6 +37,14 @@ enyo.kind({
 		this.dockApps = [];
 		this.$.getDockList.call();
 	},
+	rendered:function(){
+		this.$.getTweaks.call({owner: "org.webosinternals.patches.autostart-exhibition", keys: ["autostartTweak"]});
+	},
+	gotTweaks:function(inSender, inResponse){
+		if((inResponse) && (inResponse.autostartTweak)) {
+			this.enterExhibitionMode();
+		};
+	},
 	handleDockAppList: function(inSender, inResponse) {
 		if(inResponse && !inResponse.returnValue)
 			return;
