diff --git a/usr/palm/frameworks/mojo/builtins/palmInitFramework347.js b/usr/palm/frameworks/mojo/builtins/palmInitFramework347.js
index aed4b02..3058606 100644
--- a/usr/palm/frameworks/mojo/builtins/palmInitFramework347.js
+++ b/usr/palm/frameworks/mojo/builtins/palmInitFramework347.js
@@ -18298,7 +18298,7 @@ this.sameTargetTime=0.5;
 this.dragTargetTime=0.4;
 this.flickScale=0.075;
 
-this.targetDecodeResolution=1536;
+this.targetDecodeResolution=12288;
 
 this.dragAnimationSteps=this.dragTargetTime*
 Mojo.Animation.targetFPS;
