diff --git a/etc/init.d/bootmisc.sh b/etc/init.d/bootmisc.sh
index 1f3be21..13a571a 100755
--- a/etc/init.d/bootmisc.sh
+++ b/etc/init.d/bootmisc.sh
@@ -88,3 +88,12 @@ then
 fi
 
 : exit 0
+
+blockdev --setra none /dev/mapper/store-swap
+blockdev --setra none /dev/mapper/store-var
+blockdev --setra none /dev/mapper/store-media
+blockdev --setra none /dev/mapper/store-update
+blockdev --setra none /dev/mapper/store-log
+blockdev --setra none /dev/mapper/store-root
+
+1024 > /sys/block/mmcblk0/queue/read_ahead_kb
