diff --git a/etc/sysctl.conf b/etc/sysctl.conf
new file mode 100644
index 0000000..1b0b4c1
--- /dev/null
+++ b/etc/sysctl.conf
@@ -0,0 +1,22 @@
+vm.swappiness = 10
+vm.vfs_cache_pressure = 20
+vm.dirty_expire_centisecs = 1200
+vm.dirty_writeback_centisecs = 500
+vm.dirty_background_ratio = 10
+vm.dirty_ratio = 30
+vm.min_free_kbytes = 1024
+vm.max_map_count = 65536
+vm.overcommit_memory = 1
+vm.overcommit_ratio = 10
+vm.zone_reclaim_mode = 2
+kernel.shmax = 268435456
+kernel.shmall = 16777216
+kernel.shmmni = 65576
+sched_compat_yield = 1
+vm.lowmem_reserve_ratio = 1024
+
+
+
+
+
+
