diff --git a/etc/event.d/compcache b/etc/event.d/compcache
index 9d370fd..f1cf26d 100644
--- a/etc/event.d/compcache
+++ b/etc/event.d/compcache
@@ -7,9 +7,8 @@ if [ -e /etc/palm/preferences/compcache_enabled ]; then
 	swapoff -a
 	sleep 1
 	modprobe xvmalloc
-	modprobe ramzswap memlimit_kb=25600 backing_swap=/dev/store/swap
 	sleep 3
-	swapon /dev/ramzswap0 -p 100
+	mkswap /dev/store/swap
+	swapon /dev/store/swap
 fi
 end script
-
