diff --git a/etc/PmLog.conf b/etc/PmLog.conf
index 77409fe..9c7bbb3 100644
--- a/etc/PmLog.conf
+++ b/etc/PmLog.conf
@@ -43,10 +43,10 @@ Rotations=5
 #==========================================================
 # kernel log definition
 #==========================================================
-[OUTPUT=kernlog]
-File=/var/log/kern.log
-MaxSize=50K
-Rotations=1
+#[OUTPUT=kernlog]
+#File=/var/log/kern.log
+#MaxSize=50K
+#Rotations=1
 
 #==========================================================
 # TIL log definition
@@ -138,11 +138,10 @@ Rotations=2
 # definition will use these rules.
 #
 # By default all messages are copied to stdlog.
-# Kernel messages are also copied to kernlog.
 #==========================================================
 [CONTEXT=<global>]
 Rule1=*.*,stdlog
-Rule2=kern.*,kernlog
+Rule2=kern.!info,-stdlog
 Rule3=*.!err.UpdateDaemon,-stdlog
 
 [CONTEXT=LunaSysMgr]
