--- a/usr/palm/services/com.palm.service.deviceprofile/handlers/DeviceProfileCommandAssistant.js
+++ b/usr/palm/services/com.palm.service.deviceprofile/handlers/DeviceProfileCommandAssistant.js
@@ -105,7 +105,17 @@
 				}
 		});
 		
-		future.result = {"deviceInfo" : info.toJSON()};
+		MojoDB.get(["org.webosinternals.impostah.deviceprofile"]).then(function (f) {
+			if ((f.result.returnValue == true) &&
+			    (f.result.results.length == 1)) {
+			    for (var field in f.result.results[0]) {
+				if (field in info) {
+				    info[field] = f.result.results[0][field];
+				}
+			    }
+			}
+			future.result = {"deviceInfo" : info.toJSON()};
+		}); 
 	},
 	
 	readTelephonyValues: function (info, response) {
