diff --git a/usr/palm/frameworks/mojo/builtins/palmInitFramework367.js b/usr/palm/frameworks/mojo/builtins/palmInitFramework367.js
index b062507..a0b0c36 100644
--- a/usr/palm/frameworks/mojo/builtins/palmInitFramework367.js
+++ b/usr/palm/frameworks/mojo/builtins/palmInitFramework367.js
@@ -18328,7 +18328,7 @@ this.sameTargetTime=0.5;
 this.dragTargetTime=0.4;
 this.flickScale=0.075;
 
-this.targetDecodeResolution=1536;
+this.targetDecodeResolution=12288;
 
 this.dragAnimationSteps=this.dragTargetTime*
 Mojo.Animation.targetFPS;
