diff --git a/etc/PmLog.conf b/etc/PmLog.conf
index e5b73cc..b3078e6 100644
--- a/etc/PmLog.conf
+++ b/etc/PmLog.conf
@@ -167,7 +167,7 @@ Rotations=5
 [CONTEXT=<global>]
 Rule1=*.*,stdlog
 Rule2=kern.!info,-stdlog
-Rule3=*.*.usbmon,usbmon
+#Rule3=*.*.usbmon,usbmon
 Rule4=*.*.usbmon,-stdlog
 #Rule5=*.err,errlog
 
@@ -177,7 +177,6 @@ Rule4=*.*.usbmon,-stdlog
 #==========================================================
 [CONTEXT=TIL.CCP]
 Rule1=*.err,stdlog
-Rule2=*.*,ccplog
 
 #==========================================================
 # For the TIL.LOG, TIL.IPC contexts, only put errors in
@@ -185,19 +184,15 @@ Rule2=*.*,ccplog
 #==========================================================
 [CONTEXT=TIL.LOG]
 Rule1=*.err,stdlog
-Rule2=*.*,tillog
 
 [CONTEXT=TIL.IPC]
 Rule1=*.err,stdlog
-Rule2=*.*,tillog
 
 [CONTEXT=TIL.SVC]
 Rule1=*.err,stdlog
-Rule2=*.*,tillog
 
 [CONTEXT=TIL.HCI]
 Rule1=*.err,stdlog
-Rule2=*.*,tillog
 
 #==========================================================
 # For the HLL.LOG context, only put errors in
@@ -205,14 +200,13 @@ Rule2=*.*,tillog
 #==========================================================
 [CONTEXT=TIL.HLL]
 Rule1=*.err,stdlog
-Rule2=*.*,hlllog
 
 #==========================================================
 # For the powermond context, put everything in
 # stdlog.  
 #==========================================================
 [CONTEXT=powermond]
-Rule1=*.*,stdlog
+Rule1=*.err,stdlog
 
 #==========================================================
 # For the uploadd context, only put errors in
@@ -220,7 +214,6 @@ Rule1=*.*,stdlog
 #==========================================================
 [CONTEXT=uploadd]
 Rule1=*.err,stdlog
-Rule2=*.*,uploadd
 
 #==========================================================
 # Example user context definition
@@ -230,28 +223,27 @@ Rule2=*.*,uploadd
 #==========================================================
 [CONTEXT=PmLogTest]
 Rule1=*.err,stdlog
-Rule2=*.*,PmLogTestLog
 
 [CONTEXT=PmLogTestRB1]
 BufferSize=2K
 FlushLevel=err
-Rule1=*.*,PmLogTestLog
+Rule1=*.err,PmLogTestLog
 
 [CONTEXT=PmLogTestRB2]
 BufferSize=2K
 FlushLevel=err
-Rule1=*.*,PmLogTestLog
+Rule1=*.err,PmLogTestLog
 
 [CONTEXT=PmLogTestRB3]
 BufferSize=2K
 FlushLevel=err
-Rule1=*.*,PmLogTestLog
+Rule1=*.err,PmLogTestLog
 
 #==========================================================
 # Utility used in logging
 #==========================================================
 [CONTEXT=PmLogCtl]
-Rule1=*,stdlog
+Rule1=*.err,stdlog
 BufferSize=2K
 FlushLevel=err
 
@@ -261,25 +253,21 @@ FlushLevel=err
 #==========================================================
 [CONTEXT=PmBtEngine]
 Rule1=*.err,stdlog
-Rule2=*.*,btlog
 
 [CONTEXT=PmBtStack]
 Rule1=*.err,stdlog
-Rule2=*.*,btlog
 
 [CONTEXT=PmBtContacts]
 Rule1=*.err,stdlog
-Rule2=*.*,btlog
 
 [CONTEXT=BluetoothMonitor]
 Rule1=*.err,stdlog
-Rule2=*.*,btlog
 
 #==========================================================
 # PmWanDaemon
 #==========================================================
 [CONTEXT=wan.mgr]
-Rule1=*,stdlog
+Rule1=*.err,stdlog
 BufferSize=10K
 FlushLevel=err
 
@@ -287,7 +275,7 @@ FlushLevel=err
 # PmWiFiService
 #==========================================================
 [CONTEXT=com.palm.wifi]
-Rule1=*,stdlog
+Rule1=*.err,stdlog
 BufferSize=10K
 FlushLevel=err
 
@@ -295,13 +283,13 @@ FlushLevel=err
 # Media       
 #==========================================================
 [CONTEXT=media]
-Rule1=*,stdlog
+Rule1=*.err,stdlog
 
 #==========================================================
 # Wireless System Framework
 #==========================================================
 [CONTEXT=WSF.MessageMover]
-Rule1=*,stdlog
+Rule1=*.err,stdlog
 
 #==========================================================
 # For the VPN contexts, only put errors in
@@ -309,19 +297,15 @@ Rule1=*,stdlog
 #==========================================================
 [CONTEXT=vpn.mgr]
 Rule1=*.err,stdlog
-Rule2=*.*,vpnlog
 
 [CONTEXT=vpn.vpncagent]
 Rule1=*.err,stdlog
-Rule2=*.*,vpnlog
 
 [CONTEXT=vpn.vpnacagent]
 Rule1=*.err,stdlog
-Rule2=*.*,vpnlog
 
 [CONTEXT=vpnagent.acoal]
 Rule1=*.err,stdlog
-Rule2=*.*,vpnlog
 
 #==========================================================
 # For the WAN contexts, only put errors in
@@ -329,81 +313,63 @@ Rule2=*.*,vpnlog
 #==========================================================
 [CONTEXT=wan.blade]
 Rule1=*.err,stdlog
-Rule2=*.*,wanlog
 
 [CONTEXT=wan.ipc]
 Rule1=*.err,stdlog
-Rule2=*.*,wanlog
 
 [CONTEXT=wan.pdpstate]
 Rule1=*.err,stdlog
-Rule2=*.*,wanlog
 
 [CONTEXT=wan.modem]
 Rule1=*.err,stdlog
-Rule2=*.*,wanlog
 
 [CONTEXT=wan.pdpstatemgr]
 Rule1=*.err,stdlog
-Rule2=*.*,wanlog
 
 [CONTEXT=wan.pdpslot]
 Rule1=*.err,stdlog
-Rule2=*.*,wanlog
 
 [CONTEXT=wan.profmgr]
 Rule1=*.err,stdlog
-Rule2=*.*,wanlog
 
 [CONTEXT=wan.radio]
 Rule1=*.err,stdlog
-Rule2=*.*,wanlog
 
 [CONTEXT=wan.connector]
 Rule1=*.err,stdlog
-Rule2=*.*,wanlog
 
 [CONTEXT=wan.auto]
 Rule1=*.err,stdlog
-Rule2=*.*,wanlog
 
 [CONTEXT=wan.settings]
 Rule1=*.err,stdlog
-Rule2=*.*,wanlog
 
 [CONTEXT=wan.radiopdp]
 Rule1=*.err,stdlog
-Rule2=*.*,wanlog
 
 [CONTEXT=wan.cdb]
 Rule1=*.err,stdlog
-Rule2=*.*,wanlog
 
 [CONTEXT=wan.power]
 Rule1=*.err,stdlog
-Rule2=*.*,wanlog
 
 [CONTEXT=wan.ip]
 Rule1=*.err,stdlog
-Rule2=*.*,wanlog
 
 [CONTEXT=wan.ppp]
 Rule1=*.err,stdlog
-Rule2=*.*,wanlog
 
 [CONTEXT=wan.qmi]
 Rule1=*.err,stdlog
-Rule2=*.*,wanlog
 
 [CONTEXT=wan.dss]
 Rule1=*.err,stdlog
-Rule2=*.*,wanlog
 
 #==========================================================
 # Passthru service
 #==========================================================
 [CONTEXT=passthrud]
-Rule1=*.*,passthrulog
+Rule1=*.err,passthrulog
 BufferSize=500K
 FlushLevel=info
 
@@ -411,6 +377,6 @@ FlushLevel=info
 # usbsniff service
 #==========================================================
 [CONTEXT=usbsniff]
-Rule1=*.*,usbsnifflog
+Rule1=*.err,usbsnifflog
 BufferSize=500K
 FlushLevel=info
diff --git a/etc/PmLogContexts.conf b/etc/PmLogContexts.conf
index c5ddaf1..7979335 100644
--- a/etc/PmLogContexts.conf
+++ b/etc/PmLogContexts.conf
@@ -49,46 +49,46 @@ LogToConsole=false
 # see real time changes without rebooting.
 ###########################################################
 [Contexts]
-<global>=debug
+<global>=err
 TIL.CCP=err
-TIL.LOG=notice
-TIL.IPC=notice
-TIL.SVC=notice
-TIL.HCI=notice
-TIL.HLL=notice
-PmLogCtl=info
-com.palm.wifi=info
-media=debug
-WSF.MessageMover=info
-wan.blade=debug
-wan.ipc=debug
-wan.pdpstate=debug
-wan.modem=debug
-wan.pdpstatemgr=debug
-wan.pdpslot=debug
-wan.profmgr=debug
-wan.radio=debug
-wan.connector=debug
-wan.auto=debug
-wan.settings=debug
-wan.radiopdp=debug
-wan.cdb=debug
-wan.power=debug
-wan.ip=debug
-wan.ppp=debug
-wan.qmi=debug
-wan.dss=debug
-PmBtEngine=debug
-PmBtStack=debug
-PmBtContacts=debug
-BluetoothMonitor=debug
-passthrud=info
-usbsniff=debug
-powerd=info
-com.palm.dhcp=info
-com.palm.mobilehotspot=info
-com.palm.zeroconf=info
-com.palm.wifi.authenticator=info
-com.palm.wifi.supplicant=info
-com.palm.wifi.carrierhotspot=info
-libpalmsocket=info
+TIL.LOG=err
+TIL.IPC=err
+TIL.SVC=err
+TIL.HCI=err
+TIL.HLL=err
+PmLogCtl=err
+com.palm.wifi=err
+media=err
+WSF.MessageMover=err
+wan.blade=err
+wan.ipc=err
+wan.pdpstate=err
+wan.modem=err
+wan.pdpstatemgr=err
+wan.pdpslot=err
+wan.profmgr=err
+wan.radio=err
+wan.connector=err
+wan.auto=err
+wan.settings=err
+wan.radiopdp=err
+wan.cdb=err
+wan.power=err
+wan.ip=err
+wan.ppp=err
+wan.qmi=err
+wan.dss=err
+PmBtEngine=err
+PmBtStack=err
+PmBtContacts=err
+BluetoothMonitor=err
+passthrud=err
+usbsniff=err
+powerd=err
+com.palm.dhcp=err
+com.palm.mobilehotspot=err
+com.palm.zeroconf=err
+com.palm.wifi.authenticator=err
+com.palm.wifi.supplicant=err
+com.palm.wifi.carrierhotspot=err
+libpalmsocket=err
