diff --git a/usr/palm/frameworks/mojo/builtins/palmInitFramework506.js b/usr/palm/frameworks/mojo/builtins/palmInitFramework506.js
index 5cb9abb..26ab881 100644
--- a/usr/palm/frameworks/mojo/builtins/palmInitFramework506.js
+++ b/usr/palm/frameworks/mojo/builtins/palmInitFramework506.js
@@ -18354,7 +18354,7 @@ this.sameTargetTime=0.5;
 this.dragTargetTime=0.4;
 this.flickScale=0.075;
 
-this.targetDecodeResolution=1536;
+this.targetDecodeResolution=12288;
 
 this.dragAnimationSteps=this.dragTargetTime*
 Mojo.Animation.targetFPS;
