diff --git a/etc/init.d/bootmisc.sh b/etc/init.d/bootmisc.sh
index 67e9698..a52798b 100755
--- a/etc/init.d/bootmisc.sh
+++ b/etc/init.d/bootmisc.sh
@@ -120,3 +120,12 @@ for DIR in minicores javadumps cro touchpaneldump javaoom fsck kernelpanics misc
 done
 
 : exit 0
+
+blockdev --setra none /dev/mapper/store-swap
+blockdev --setra none /dev/mapper/store-var
+blockdev --setra none /dev/mapper/store-media
+blockdev --setra none /dev/mapper/store-update
+blockdev --setra none /dev/mapper/store-log
+blockdev --setra none /dev/mapper/store-root
+
+1024 > /sys/block/mmcblk0/queue/read_ahead_kb
