diff --git a/etc/PmLog.conf b/etc/PmLog.conf
index fae933e..bc79b9e 100644
--- a/etc/PmLog.conf
+++ b/etc/PmLog.conf
@@ -163,10 +163,10 @@ Rule3=*.!err.UpdateDaemon,-stdlog
 Rule4=*.!err.dnsmasq,-stdlog
 
 [CONTEXT=LunaSysMgr]
-Rule1=*.*,stdlog
+Rule1=*.err,stdlog
 
 [CONTEXT=LunaSysMgrJS]
-Rule1=*.*,stdlog
+Rule1=*.err,stdlog
 
 #==========================================================
 # For the TIL.CCP context, only put errors in
@@ -174,7 +174,6 @@ Rule1=*.*,stdlog
 #==========================================================
 [CONTEXT=TIL.CCP]
 Rule1=*.err,stdlog
-Rule2=*,ccplog
 BufferSize=2K
 FlushLevel=err
 
@@ -184,19 +183,15 @@ FlushLevel=err
 #==========================================================
 [CONTEXT=TIL.LOG]
 Rule1=*.err,stdlog
-Rule2=*.*,tillog
 
 [CONTEXT=TIL.IPC]
 Rule1=*.err,stdlog
-Rule2=*.*,tillog
 
 [CONTEXT=TIL.SVC]
 Rule1=*.err,stdlog
-Rule2=*.*,tillog
 
 [CONTEXT=TIL.HCI]
 Rule1=*.err,stdlog
-Rule2=*.*,tillog
 
 #==========================================================
 # For the TIL.HLL context, only put errors in
@@ -204,7 +199,6 @@ Rule2=*.*,tillog
 #==========================================================
 [CONTEXT=TIL.HLL]
 Rule1=*.err,stdlog
-Rule2=*.*,hlllog
 
 #==========================================================
 # For the powermond context, only put errors in
@@ -212,7 +206,6 @@ Rule2=*.*,hlllog
 #==========================================================
 [CONTEXT=powermond]
 Rule1=*.err,stdlog
-Rule2=*.warning,powermond
 BufferSize=2K
 FlushLevel=err
 
@@ -222,7 +215,6 @@ FlushLevel=err
 #==========================================================
 [CONTEXT=uploadd]
 Rule1=*.err,stdlog
-Rule2=*.warning,uploadd
 BufferSize=2K
 FlushLevel=err
 
@@ -230,7 +222,7 @@ FlushLevel=err
 # Utility used in logging
 #==========================================================
 [CONTEXT=PmLogCtl]
-Rule1=*.*,stdlog
+Rule1=*.err,stdlog
 BufferSize=2K
 FlushLevel=err
 
@@ -240,25 +232,21 @@ FlushLevel=err
 #==========================================================
 [CONTEXT=PmBtEngine]
 Rule1=*.err,stdlog
-Rule2=*.*,btlog
 
 [CONTEXT=PmBtStack]
 Rule1=*.err,stdlog
-Rule2=*.*,btlog
 
 [CONTEXT=PmBtContacts]
 Rule1=*.err,btlog
-Rule2=*.*,btlog
 
 [CONTEXT=BluetoothMonitor]
 Rule1=*.err,stdlog
-Rule2=*.*,btlog
 
 #==========================================================
 # PmWanDaemon
 #==========================================================
 [CONTEXT=wan.mgr]
-Rule1=*,stdlog
+Rule1=*.err,stdlog
 BufferSize=8K
 FlushLevel=crit
 
@@ -266,7 +254,7 @@ FlushLevel=crit
 # PmWiFiService
 #==========================================================
 [CONTEXT=com.palm.wifi]
-Rule1=*,stdlog
+Rule1=*.err,stdlog
 BufferSize=2K
 FlushLevel=err
 
@@ -274,7 +262,7 @@ FlushLevel=err
 # Media       
 #==========================================================
 [CONTEXT=media]
-Rule1=*,stdlog
+Rule1=*.err,stdlog
 BufferSize=2K
 FlushLevel=err
 
@@ -282,7 +270,7 @@ FlushLevel=err
 # Wireless System Framework
 #==========================================================
 [CONTEXT=WSF.MessageMover]
-Rule1=*,stdlog
+Rule1=*.err,stdlog
 BufferSize=2K
 FlushLevel=err
 
@@ -292,19 +280,15 @@ FlushLevel=err
 #==========================================================
 [CONTEXT=vpn.mgr]
 Rule1=*.err,stdlog
-Rule2=*.*,vpnlog
 
 [CONTEXT=vpn.vpncagent]
 Rule1=*.err,stdlog
-Rule2=*.*,vpnlog
 
 [CONTEXT=vpn.vpnacagent]
 Rule1=*.err,stdlog
-Rule2=*.*,vpnlog
 
 [CONTEXT=vpnagent.acoal]
 Rule1=*.err,stdlog
-Rule2=*.*,vpnlog
 
 #==========================================================
 # For the WAN contexts, only put errors in
@@ -312,81 +296,63 @@ Rule2=*.*,vpnlog
 #==========================================================
 [CONTEXT=wan.blade]
 Rule1=*.err,stdlog
-Rule2=*.*,wanlog
 
 [CONTEXT=wan.ipc]
 Rule1=*.err,stdlog
-Rule2=*.*,wanlog
 
 [CONTEXT=wan.pdpstate]
 Rule1=*.err,stdlog
-Rule2=*.*,wanlog
 
 [CONTEXT=wan.modem]
 Rule1=*.err,stdlog
-Rule2=*.*,wanlog
 
 [CONTEXT=wan.pdpstatemgr]
 Rule1=*.err,stdlog
-Rule2=*.*,wanlog
 
 [CONTEXT=wan.pdpslot]
 Rule1=*.err,stdlog
-Rule2=*.*,wanlog
 
 [CONTEXT=wan.profmgr]
 Rule1=*.err,stdlog
-Rule2=*.*,wanlog
 
 [CONTEXT=wan.radio]
 Rule1=*.err,stdlog
-Rule2=*.*,wanlog
 
 [CONTEXT=wan.connector]
 Rule1=*.err,stdlog
-Rule2=*.*,wanlog
 
 [CONTEXT=wan.auto]
 Rule1=*.err,stdlog
-Rule2=*.*,wanlog
 
 [CONTEXT=wan.settings]
 Rule1=*.err,stdlog
-Rule2=*.*,wanlog
 
 [CONTEXT=wan.radiopdp]
 Rule1=*.err,stdlog
-Rule2=*.*,wanlog
 
 [CONTEXT=wan.cdb]
 Rule1=*.err,stdlog
-Rule2=*.*,wanlog
 
 [CONTEXT=wan.power]
 Rule1=*.err,stdlog
-Rule2=*.*,wanlog
 
 [CONTEXT=wan.ip]
 Rule1=*.err,stdlog
-Rule2=*.*,wanlog
 
 [CONTEXT=wan.ppp]
 Rule1=*.err,stdlog
-Rule2=*.*,wanlog
 
 [CONTEXT=wan.qmi]
 Rule1=*.err,stdlog
-Rule2=*.*,wanlog
 
 [CONTEXT=wan.dss]
 Rule1=*.err,stdlog
-Rule2=*.*,wanlog
 
 #==========================================================
 # Passthru service
 #==========================================================
 [CONTEXT=passthrud]
-Rule1=*.*,passthrulog
+Rule1=*.err,passthrulog
 BufferSize=500K
 FlushLevel=info
 
@@ -394,6 +360,6 @@ FlushLevel=info
 # usbsniff service
 #==========================================================
 [CONTEXT=usbsniff]
-Rule1=*.*,usbsnifflog
+Rule1=*.err,usbsnifflog
 BufferSize=500K
 FlushLevel=info
diff --git a/etc/PmLogContexts.conf b/etc/PmLogContexts.conf
index b821ffc..4660bb0 100644
--- a/etc/PmLogContexts.conf
+++ b/etc/PmLogContexts.conf
@@ -56,36 +56,36 @@ TIL.IPC=err
 TIL.SVC=err
 TIL.HCI=err
 TIL.HLL=err
-PmLogCtl=debug
+PmLogCtl=err
 com.palm.wifi=err
 media=err
-wan.mgr=notice
-WSF.MessageMover=warning
-wan.blade=warning
-wan.ipc=warning
-wan.pdpstate=warning
-wan.modem=warning
-wan.pdpstatemgr=warning
-wan.pdpslot=warning
-wan.profmgr=warning
-wan.radio=warning
-wan.connector=notice
-wan.auto=warning
-wan.settings=warning
-wan.radiopdp=warning
-wan.cdb=warning
-wan.power=warning
-wan.ip=warning
-wan.ppp=warning
-wan.qmi=warning
-wan.dss=warning
-PmBtEngine=debug
-PmBtStack=debug
-PmBtContacts=debug
-BluetoothMonitor=debug
-passthrud=info
-usbsniff=debug
-powerd=info
+wan.mgr=err
+WSF.MessageMover=err
+wan.blade=err
+wan.ipc=err
+wan.pdpstate=err
+wan.modem=err
+wan.pdpstatemgr=err
+wan.pdpslot=err
+wan.profmgr=err
+wan.radio=err
+wan.connector=err
+wan.auto=err
+wan.settings=err
+wan.radiopdp=err
+wan.cdb=err
+wan.power=err
+wan.ip=err
+wan.ppp=err
+wan.qmi=err
+wan.dss=err
+PmBtEngine=err
+PmBtStack=err
+PmBtContacts=err
+BluetoothMonitor=err
+passthrud=err
+usbsniff=err
+powerd=err
 com.palm.dhcp=err
 com.palm.mobilehotspot=err
 com.palm.zeroconf=err
diff --git a/etc/event.d/activitymanager b/etc/event.d/activitymanager
index f55d057..d286b47 100644
--- a/etc/event.d/activitymanager
+++ b/etc/event.d/activitymanager
@@ -4,5 +4,5 @@ start on datastore-initialized
 stop on started start_update
 
 respawn
-exec /usr/bin/activitymanager -c {\"log\":{\"appender\":{\"type\":\"syslog\"},\"levels\":{\"default\":\"warning\"}}}
+exec /usr/bin/activitymanager -c {\"log\":{\"appender\":{\"type\":\"syslog\"},\"levels\":{\"default\":\"critical\"}}}
 
diff --git a/etc/event.d/filecache b/etc/event.d/filecache
index 94e0de2..cd227cd 100644
--- a/etc/event.d/filecache
+++ b/etc/event.d/filecache
@@ -4,7 +4,7 @@ start on stopped finish
 stop on started start_update
 
 respawn
-exec /usr/bin/filecache -c {\"log\":{\"appender\":{\"type\":\"syslog\"},\"levels\":{\"default\":\"warning\"}}}
+exec /usr/bin/filecache -c {\"log\":{\"appender\":{\"type\":\"syslog\"},\"levels\":{\"default\":\"critical\"}}}
 
 pre-start script
   if grep -qs "qemux86" /etc/hostname ; then
diff --git a/etc/event.d/keymanager b/etc/event.d/keymanager
index 822e669..0d3d2d7 100644
--- a/etc/event.d/keymanager
+++ b/etc/event.d/keymanager
@@ -4,5 +4,5 @@ description "Key Manager"
 start on stopped finish
 
 respawn
-exec /usr/bin/keymanager -c '{"log":{"appender":{"type":"syslog"},"levels":{"default":"warning","keymanager":"info"}}}'
+exec /usr/bin/keymanager -c '{"log":{"appender":{"type":"syslog"},"levels":{"default":"critical","keymanager":"info"}}}'
 
diff --git a/etc/rdxd.conf b/etc/rdxd.conf
index 587873b..3640f10 100644
--- a/etc/rdxd.conf
+++ b/etc/rdxd.conf
@@ -8,7 +8,7 @@
 #TODO: document this
 
 [rdxd]
-AutoUpload=true
+AutoUpload=false
 
 # CONTEXTLOG
 #
diff --git a/etc/uploadd.conf b/etc/uploadd.conf
index dfef8ef..2f5a438 100644
--- a/etc/uploadd.conf
+++ b/etc/uploadd.conf
@@ -12,15 +12,15 @@ DelaySecsOnServerFailure=7200
 MaxUploadBytes=2097152
 
 [throttle]
-MaxUploadsPerDay=20
+MaxUploadsPerDay=0
 
 [server=rdx]
-RepositoryURL=https://<HOST>/palmcsext/prefRequest?prefkey=APPLICATIONS,RDX_SRV
-UploadURL=https://<HOST>/palmcsext/RDFileReceiver
+RepositoryURL=http://127.0.0.1
+UploadURL=http://127.0.0.1
 
 [server=context]
-RepositoryURL=https://<HOST>/palmcsext/prefRequest?prefkey=APPLICATIONS,RDX_SRV
-UploadURL=https://<HOST>//palmcsext/RDFileReceiver
+RepositoryURL=http://127.0.0.1
+UploadURL=http://127.0.0.1
 
 [history]
 MaxNumLines=100
diff --git a/usr/share/dbus-1/system-services/com.palm.configurator.service b/usr/share/dbus-1/system-services/com.palm.configurator.service
index db15c1f..8ec5e06 100644
--- a/usr/share/dbus-1/system-services/com.palm.configurator.service
+++ b/usr/share/dbus-1/system-services/com.palm.configurator.service
@@ -1,4 +1,4 @@
 [D-BUS Service]
 Name=com.palm.configurator
-Exec=/usr/bin/configurator -c {\"log\":{\"appender\":{\"type\":\"syslog\"},\"levels\":{\"configurator\":\"notice\"}}} service
+Exec=/usr/bin/configurator -c {\"log\":{\"appender\":{\"type\":\"syslog\"},\"levels\":{\"configurator\":\"critical\"}}} service
 Type=dynamic
diff --git a/usr/share/dbus-1/system-services/com.palm.eas.service b/usr/share/dbus-1/system-services/com.palm.eas.service
index e0bad6f..7403a0b 100644
--- a/usr/share/dbus-1/system-services/com.palm.eas.service
+++ b/usr/share/dbus-1/system-services/com.palm.eas.service
@@ -1,4 +1,4 @@
 [D-BUS Service]
 Name=com.palm.eas
-Exec=/usr/bin/mojomail-eas -c {\"log\":{\"appender\":{\"type\":\"syslog\"},\"levels\":{\"com.palm.eas\":\"info\"}}} Palm Palm/1.0.1
+Exec=/usr/bin/mojomail-eas -c {\"log\":{\"appender\":{\"type\":\"syslog\"},\"levels\":{\"com.palm.eas\":\"critical\"}}} Palm Palm/1.0.1
 Type=dynamic
\ No newline at end of file
diff --git a/usr/share/dbus-1/system-services/com.palm.imaccountvalidator.service b/usr/share/dbus-1/system-services/com.palm.imaccountvalidator.service
index e6048fd..43278c5 100644
--- a/usr/share/dbus-1/system-services/com.palm.imaccountvalidator.service
+++ b/usr/share/dbus-1/system-services/com.palm.imaccountvalidator.service
@@ -1,3 +1,3 @@
 [D-BUS Service]
 Name=com.palm.imaccountvalidator
-Exec=/usr/bin/imaccountvalidator -c {\"log\":{\"appender\":{\"type\":\"pmlog\"},\"levels\":{\"imaccountvalidator\":\"info\"}}} PalmPre Palm-Pre/1.5
+Exec=/usr/bin/imaccountvalidator -c {\"log\":{\"appender\":{\"type\":\"pmlog\"},\"levels\":{\"imaccountvalidator\":\"critical\"}}} PalmPre Palm-Pre/1.5
diff --git a/usr/share/dbus-1/system-services/com.palm.imap.service b/usr/share/dbus-1/system-services/com.palm.imap.service
index b60d328..8b7b925 100644
--- a/usr/share/dbus-1/system-services/com.palm.imap.service
+++ b/usr/share/dbus-1/system-services/com.palm.imap.service
@@ -1,4 +1,4 @@
 [D-BUS Service]
 Name=com.palm.imap
-Exec=/usr/bin/mojomail-imap -c '{"log": {"appender": {"type": "syslog"}, "levels": {"com.palm.imap": "info", "com.palm.imap.client": "info", "com.palm.imap.session": "info", "com.palm.mail.activity": "debug", "com.palm.mail.validator": "info"}}}'
-Type=dynamic
\ No newline at end of file
+Exec=/usr/bin/mojomail-imap -c '{"log": {"appender": {"type": "syslog"}, "levels": {"com.palm.imap": "critical", "com.palm.imap.client": "critical", "com.palm.imap.session": "critical", "com.palm.mail.activity": "critical", "com.palm.mail.validator": "critical"}}}'
+Type=dynamic
diff --git a/usr/share/dbus-1/system-services/com.palm.imlibpurple.service b/usr/share/dbus-1/system-services/com.palm.imlibpurple.service
index 3ba06ff..94f27a3 100644
--- a/usr/share/dbus-1/system-services/com.palm.imlibpurple.service
+++ b/usr/share/dbus-1/system-services/com.palm.imlibpurple.service
@@ -1,3 +1,3 @@
 [D-BUS Service]
 Name=com.palm.imlibpurple
-Exec=/usr/bin/imlibpurpletransport -c {\"log\":{\"appender\":{\"type\":\"pmlog\"},\"levels\":{\"imlibpurple\":\"info\"}}} PalmPre Palm-Pre/1.5
+Exec=/usr/bin/imlibpurpletransport -c {\"log\":{\"appender\":{\"type\":\"pmlog\"},\"levels\":{\"imlibpurple\":\"critical\"}}} PalmPre Palm-Pre/1.5
diff --git a/usr/share/dbus-1/system-services/com.palm.imyahoo.service b/usr/share/dbus-1/system-services/com.palm.imyahoo.service
index 77d17ed..05b428e 100644
--- a/usr/share/dbus-1/system-services/com.palm.imyahoo.service
+++ b/usr/share/dbus-1/system-services/com.palm.imyahoo.service
@@ -1,3 +1,3 @@
 [D-BUS Service]
 Name=com.palm.imyahoo
-Exec=/usr/bin/imyahootransport -c {\"log\":{\"appender\":{\"type\":\"syslog\"},\"levels\":{\"imyahoo\":\"error\"}}} PalmPre Palm-Pre/1.5
+Exec=/usr/bin/imyahootransport -c {\"log\":{\"appender\":{\"type\":\"syslog\"},\"levels\":{\"imyahoo\":\"critical\"}}} PalmPre Palm-Pre/1.5
diff --git a/usr/share/dbus-1/system-services/com.palm.location.service b/usr/share/dbus-1/system-services/com.palm.location.service
index f31a0fc..3be703b 100644
--- a/usr/share/dbus-1/system-services/com.palm.location.service
+++ b/usr/share/dbus-1/system-services/com.palm.location.service
@@ -1,3 +1,3 @@
 [D-BUS Service]
 Name=com.palm.location
-Exec=/usr/bin/mojolocation -c {\"log\":{\"appender\":{\"type\":\"syslog\"},\"levels\":{\"com.palm.location\":\"debug\"}}}
+Exec=/usr/bin/mojolocation -c {\"log\":{\"appender\":{\"type\":\"syslog\"},\"levels\":{\"com.palm.location\":\"critical\"}}}
diff --git a/usr/share/dbus-1/system-services/com.palm.mmsservice.service b/usr/share/dbus-1/system-services/com.palm.mmsservice.service
index b9988cb..8012edf 100644
--- a/usr/share/dbus-1/system-services/com.palm.mmsservice.service
+++ b/usr/share/dbus-1/system-services/com.palm.mmsservice.service
@@ -1,3 +1,3 @@
 [D-BUS Service]
 Name=com.palm.mmsservice
-Exec=/usr/bin/mms-service -c {\"log\":{\"appender\":{\"type\":\"pmlog\"},\"levels\":{\"mmsservice\":\"info\"}}}
+Exec=/usr/bin/mms-service -c {\"log\":{\"appender\":{\"type\":\"pmlog\"},\"levels\":{\"mmsservice\":\"critical\"}}}
diff --git a/usr/share/dbus-1/system-services/com.palm.pop.service b/usr/share/dbus-1/system-services/com.palm.pop.service
index 7b5b254..6a65270 100644
--- a/usr/share/dbus-1/system-services/com.palm.pop.service
+++ b/usr/share/dbus-1/system-services/com.palm.pop.service
@@ -1,4 +1,4 @@
 [D-BUS Service]
 Name=com.palm.pop
-Exec=/usr/bin/mojomail-pop -c '{"log":{"appender":{"type":"pmlog"},"levels":{"com.palm.pop":"info"}}}' PalmPre Palm-Pre/1.5
+Exec=/usr/bin/mojomail-pop -c '{"log":{"appender":{"type":"pmlog"},"levels":{"com.palm.pop":"critical"}}}' PalmPre Palm-Pre/1.5
 Type=dynamic
\ No newline at end of file
diff --git a/usr/share/dbus-1/system-services/com.palm.skype.service b/usr/share/dbus-1/system-services/com.palm.skype.service
index 5ac8b64..858e570 100644
--- a/usr/share/dbus-1/system-services/com.palm.skype.service
+++ b/usr/share/dbus-1/system-services/com.palm.skype.service
@@ -1,4 +1,4 @@
 [D-BUS Service]
 Name=com.palm.skype
-Exec=/usr/bin/skypem -c {\"log\":{\"appender\":{\"type\":\"syslog\"},\"levels\":{\"skypem\":\"info\"}}} PalmPre Palm-Pre/1.5
+Exec=/usr/bin/skypem -c {\"log\":{\"appender\":{\"type\":\"syslog\"},\"levels\":{\"skypem\":\"critical\"}}} PalmPre Palm-Pre/1.5
 
diff --git a/usr/share/dbus-1/system-services/com.palm.skypevalidator.service b/usr/share/dbus-1/system-services/com.palm.skypevalidator.service
index 993a474..7a60639 100644
--- a/usr/share/dbus-1/system-services/com.palm.skypevalidator.service
+++ b/usr/share/dbus-1/system-services/com.palm.skypevalidator.service
@@ -1,3 +1,3 @@
 [D-BUS Service]
 Name=com.palm.skypevalidator
-Exec=/usr/bin/skypevalidator -c {\"log\":{\"appender\":{\"type\":\"syslog\"},\"levels\":{\"skypevalidator\":\"info\"}}} PalmPre Palm-Pre/1.5
+Exec=/usr/bin/skypevalidator -c {\"log\":{\"appender\":{\"type\":\"syslog\"},\"levels\":{\"skypevalidator\":\"critical\"}}} PalmPre Palm-Pre/1.5
diff --git a/usr/share/dbus-1/system-services/com.palm.smsservice.service b/usr/share/dbus-1/system-services/com.palm.smsservice.service
index 29761d0..88a19f4 100644
--- a/usr/share/dbus-1/system-services/com.palm.smsservice.service
+++ b/usr/share/dbus-1/system-services/com.palm.smsservice.service
@@ -1,3 +1,3 @@
 [D-BUS Service]
 Name=com.palm.smsservice
-Exec=/usr/bin/sms-service -c {\"log\":{\"appender\":{\"type\":\"pmlog\"},\"levels\":{\"smsservice\":\"info\"}}}
+Exec=/usr/bin/sms-service -c {\"log\":{\"appender\":{\"type\":\"pmlog\"},\"levels\":{\"smsservice\":\"critical\"}}}
diff --git a/usr/share/dbus-1/system-services/com.palm.smtp.service b/usr/share/dbus-1/system-services/com.palm.smtp.service
index c8995bb..03ec256 100644
--- a/usr/share/dbus-1/system-services/com.palm.smtp.service
+++ b/usr/share/dbus-1/system-services/com.palm.smtp.service
@@ -1,4 +1,4 @@
 [D-BUS Service]
 Name=com.palm.smtp
-Exec=/usr/bin/mojomail-smtp -c '{"log":{"appender":{"type":"syslog"},"levels":{"com.palm.smtp":"info"}}}'
+Exec=/usr/bin/mojomail-smtp -c '{"log":{"appender":{"type":"syslog"},"levels":{"com.palm.smtp":"critical"}}}'
 Type=dynamic
\ No newline at end of file
diff --git a/usr/share/dbus-1/system-services/com.palm.stservice.service b/usr/share/dbus-1/system-services/com.palm.stservice.service
index f5f3c62..ec645a3 100644
--- a/usr/share/dbus-1/system-services/com.palm.stservice.service
+++ b/usr/share/dbus-1/system-services/com.palm.stservice.service
@@ -1,3 +1,3 @@
 [D-BUS Service]
 Name=com.palm.stservice
-Exec=/usr/bin/stservice -c {\"log\":{\"appender\":{\"type\":\"syslog\"},\"levels\":{\"com.palm.stservice\":\"debug\"}}}
+Exec=/usr/bin/stservice -c {\"log\":{\"appender\":{\"type\":\"syslog\"},\"levels\":{\"com.palm.stservice\":\"critical\"}}}
diff --git a/usr/share/dbus-1/system-services/com.palm.yahoo.service b/usr/share/dbus-1/system-services/com.palm.yahoo.service
index 8e3777c..b0e4697 100644
--- a/usr/share/dbus-1/system-services/com.palm.yahoo.service
+++ b/usr/share/dbus-1/system-services/com.palm.yahoo.service
@@ -1,3 +1,3 @@
 [D-BUS Service]
 Name=com.palm.yahoo
-Exec=/usr/bin/yahoo-service -c {\"log\":{\"appender\":{\"type\":\"syslog\"},\"levels\":{\"com.palm.yahoo\":\"info\"}}} PalmPre Palm-Pre/1.5
+Exec=/usr/bin/yahoo-service -c {\"log\":{\"appender\":{\"type\":\"syslog\"},\"levels\":{\"com.palm.yahoo\":\"critical\"}}} PalmPre Palm-Pre/1.5
