diff --git a/etc/PmLog.conf b/etc/PmLog.conf
index e9f7fc9..7bcabca 100644
--- a/etc/PmLog.conf
+++ b/etc/PmLog.conf
@@ -155,10 +155,10 @@ Rule3=*.!err.UpdateDaemon,-stdlog
 Rule4=*.!err.dnsmasq,-stdlog
 
 [CONTEXT=LunaSysMgr]
-Rule1=*.*,stdlog
+Rule1=*.err,stdlog
 
 [CONTEXT=LunaSysMgrJS]
-Rule1=*.*,stdlog
+Rule1=*.err,stdlog
 
 #==========================================================
 # For the TIL.CCP context, only put errors in
@@ -166,7 +166,6 @@ Rule1=*.*,stdlog
 #==========================================================
 [CONTEXT=TIL.CCP]
 Rule1=*.err,stdlog
-Rule2=*,ccplog
 BufferSize=2K
 FlushLevel=err
 
@@ -176,19 +175,15 @@ FlushLevel=err
 #==========================================================
 [CONTEXT=TIL.LOG]
 Rule1=*.err,stdlog
-Rule2=*.*,tillog
 
 [CONTEXT=TIL.IPC]
 Rule1=*.err,stdlog
-Rule2=*.*,tillog
 
 [CONTEXT=TIL.SVC]
 Rule1=*.err,stdlog
-Rule2=*.*,tillog
 
 [CONTEXT=TIL.HCI]
 Rule1=*.err,stdlog
-Rule2=*.*,tillog
 
 #==========================================================
 # For the TIL.HLL context, only put errors in
@@ -196,7 +191,6 @@ Rule2=*.*,tillog
 #==========================================================
 [CONTEXT=TIL.HLL]
 Rule1=*.err,stdlog
-Rule2=*.*,hlllog
 
 #==========================================================
 # For the powermond context, only put errors in
@@ -204,7 +198,6 @@ Rule2=*.*,hlllog
 #==========================================================
 [CONTEXT=powermond]
 Rule1=*.err,stdlog
-Rule2=*.warning,powermond
 BufferSize=2K
 FlushLevel=err
 
@@ -214,7 +207,6 @@ FlushLevel=err
 #==========================================================
 [CONTEXT=uploadd]
 Rule1=*.err,stdlog
-Rule2=*.warning,uploadd
 BufferSize=2K
 FlushLevel=err
 
@@ -222,7 +214,7 @@ FlushLevel=err
 # Utility used in logging
 #==========================================================
 [CONTEXT=PmLogCtl]
-Rule1=*.*,stdlog
+Rule1=*.err,stdlog
 BufferSize=2K
 FlushLevel=err
 
@@ -232,25 +224,21 @@ FlushLevel=err
 #==========================================================
 [CONTEXT=PmBtEngine]
 Rule1=*.err,stdlog
-Rule2=*.*,btlog
 
 [CONTEXT=PmBtStack]
 Rule1=*.err,stdlog
-Rule2=*.*,btlog
 
 [CONTEXT=PmBtContacts]
 Rule1=*.err,btlog
-Rule2=*.*,btlog
 
 [CONTEXT=BluetoothMonitor]
 Rule1=*.err,stdlog
-Rule2=*.*,btlog
 
 #==========================================================
 # PmWanDaemon
 #==========================================================
 [CONTEXT=wan.mgr]
-Rule1=*,stdlog
+Rule1=*.err,stdlog
 BufferSize=8K
 FlushLevel=crit
 
@@ -258,7 +246,7 @@ FlushLevel=crit
 # PmWiFiService
 #==========================================================
 [CONTEXT=com.palm.wifi]
-Rule1=*,stdlog
+Rule1=*.err,stdlog
 BufferSize=2K
 FlushLevel=err
 
@@ -266,7 +254,7 @@ FlushLevel=err
 # Media       
 #==========================================================
 [CONTEXT=media]
-Rule1=*,stdlog
+Rule1=*.err,stdlog
 BufferSize=2K
 FlushLevel=err
 
@@ -274,7 +262,7 @@ FlushLevel=err
 # Wireless System Framework
 #==========================================================
 [CONTEXT=WSF.MessageMover]
-Rule1=*,stdlog
+Rule1=*.err,stdlog
 BufferSize=2K
 FlushLevel=err
 
@@ -284,81 +272,63 @@ FlushLevel=err
 #==========================================================
 [CONTEXT=wan.blade]
 Rule1=*.err,stdlog
-Rule2=*.*,wanlog
 
 [CONTEXT=wan.ipc]
 Rule1=*.err,stdlog
-Rule2=*.*,wanlog
 
 [CONTEXT=wan.pdpstate]
 Rule1=*.err,stdlog
-Rule2=*.*,wanlog
 
 [CONTEXT=wan.modem]
 Rule1=*.err,stdlog
-Rule2=*.*,wanlog
 
 [CONTEXT=wan.pdpstatemgr]
 Rule1=*.err,stdlog
-Rule2=*.*,wanlog
 
 [CONTEXT=wan.pdpslot]
 Rule1=*.err,stdlog
-Rule2=*.*,wanlog
 
 [CONTEXT=wan.profmgr]
 Rule1=*.err,stdlog
-Rule2=*.*,wanlog
 
 [CONTEXT=wan.radio]
 Rule1=*.err,stdlog
-Rule2=*.*,wanlog
 
 [CONTEXT=wan.connector]
 Rule1=*.err,stdlog
-Rule2=*.*,wanlog
 
 [CONTEXT=wan.auto]
 Rule1=*.err,stdlog
-Rule2=*.*,wanlog
 
 [CONTEXT=wan.settings]
 Rule1=*.err,stdlog
-Rule2=*.*,wanlog
 
 [CONTEXT=wan.radiopdp]
 Rule1=*.err,stdlog
-Rule2=*.*,wanlog
 
 [CONTEXT=wan.cdb]
 Rule1=*.err,stdlog
-Rule2=*.*,wanlog
 
 [CONTEXT=wan.power]
 Rule1=*.err,stdlog
-Rule2=*.*,wanlog
 
 [CONTEXT=wan.ip]
 Rule1=*.err,stdlog
-Rule2=*.*,wanlog
 
 [CONTEXT=wan.ppp]
 Rule1=*.err,stdlog
-Rule2=*.*,wanlog
 
 [CONTEXT=wan.qmi]
 Rule1=*.err,stdlog
-Rule2=*.*,wanlog
 
 [CONTEXT=wan.dss]
 Rule1=*.err,stdlog
-Rule2=*.*,wanlog
 
 #==========================================================
 # Passthru service
 #==========================================================
 [CONTEXT=passthrud]
-Rule1=*.*,passthrulog
+Rule1=*.err,passthrulog
 BufferSize=500K
 FlushLevel=info
 
@@ -366,6 +336,6 @@ FlushLevel=info
 # usbsniff service
 #==========================================================
 [CONTEXT=usbsniff]
-Rule1=*.*,usbsnifflog
+Rule1=*.err,usbsnifflog
 BufferSize=500K
 FlushLevel=info
diff --git a/etc/PmLogContexts.conf b/etc/PmLogContexts.conf
index b821ffc..4660bb0 100644
--- a/etc/PmLogContexts.conf
+++ b/etc/PmLogContexts.conf
@@ -56,36 +56,36 @@ TIL.IPC=err
 TIL.SVC=err
 TIL.HCI=err
 TIL.HLL=err
-PmLogCtl=debug
+PmLogCtl=err
 com.palm.wifi=err
 media=err
-wan.mgr=notice
-WSF.MessageMover=warning
-wan.blade=warning
-wan.ipc=warning
-wan.pdpstate=warning
-wan.modem=warning
-wan.pdpstatemgr=warning
-wan.pdpslot=warning
-wan.profmgr=warning
-wan.radio=warning
-wan.connector=notice
-wan.auto=warning
-wan.settings=warning
-wan.radiopdp=warning
-wan.cdb=warning
-wan.power=warning
-wan.ip=warning
-wan.ppp=warning
-wan.qmi=warning
-wan.dss=warning
-PmBtEngine=debug
-PmBtStack=debug
-PmBtContacts=debug
-BluetoothMonitor=debug
-passthrud=info
-usbsniff=debug
-powerd=info
+wan.mgr=err
+WSF.MessageMover=err
+wan.blade=err
+wan.ipc=err
+wan.pdpstate=err
+wan.modem=err
+wan.pdpstatemgr=err
+wan.pdpslot=err
+wan.profmgr=err
+wan.radio=err
+wan.connector=err
+wan.auto=err
+wan.settings=err
+wan.radiopdp=err
+wan.cdb=err
+wan.power=err
+wan.ip=err
+wan.ppp=err
+wan.qmi=err
+wan.dss=err
+PmBtEngine=err
+PmBtStack=err
+PmBtContacts=err
+BluetoothMonitor=err
+passthrud=err
+usbsniff=err
+powerd=err
 com.palm.dhcp=err
 com.palm.mobilehotspot=err
 com.palm.zeroconf=err
diff --git a/etc/rdxd.conf b/etc/rdxd.conf
index 587873b..3640f10 100644
--- a/etc/rdxd.conf
+++ b/etc/rdxd.conf
@@ -8,7 +8,7 @@
 #TODO: document this
 
 [rdxd]
-AutoUpload=true
+AutoUpload=false
 
 # CONTEXTLOG
 #
diff --git a/etc/uploadd.conf b/etc/uploadd.conf
index dfef8ef..2f5a438 100644
--- a/etc/uploadd.conf
+++ b/etc/uploadd.conf
@@ -12,15 +12,15 @@ DelaySecsOnServerFailure=7200
 MaxUploadBytes=2097152
 
 [throttle]
-MaxUploadsPerDay=20
+MaxUploadsPerDay=0
 
 [server=rdx]
-RepositoryURL=https://<HOST>/palmcsext/prefRequest?prefkey=APPLICATIONS,RDX_SRV
-UploadURL=https://<HOST>/palmcsext/RDFileReceiver
+RepositoryURL=http://127.0.0.1
+UploadURL=http://127.0.0.1
 
 [server=context]
-RepositoryURL=https://<HOST>/palmcsext/prefRequest?prefkey=APPLICATIONS,RDX_SRV
-UploadURL=https://<HOST>//palmcsext/RDFileReceiver
+RepositoryURL=http://127.0.0.1
+UploadURL=http://127.0.0.1
 
 [history]
 MaxNumLines=100
