diff --git a/usr/palm/applications/com.palm.app.exhibitionpreferences/app/assistants/preferences-assistant.js b/usr/palm/applications/com.palm.app.exhibitionpreferences/app/assistants/preferences-assistant.js
index 932a4a7..4ac0718 100644
--- a/usr/palm/applications/com.palm.app.exhibitionpreferences/app/assistants/preferences-assistant.js
+++ b/usr/palm/applications/com.palm.app.exhibitionpreferences/app/assistants/preferences-assistant.js
@@ -40,6 +40,8 @@ PreferencesAssistant.prototype.setup = function () {
 	
 	this.getDockAppsList();
 	
+	this.controller.window.setTimeout(this.startExhibitionMode.bind(this), 200);
+	
 };
 
 PreferencesAssistant.prototype.activate = function() {
