diff --git a/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js b/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js
index 0c816ba..e69aa92 100644
--- a/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js
+++ b/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js
@@ -160,7 +160,12 @@ var ChatviewAssistant = Class.create({
 				segmentCountContainer: this.controller.get('segmentCounterContainer'),
 				segmentCountElement: this.controller.get('segmentCounter'),
 				setTextFieldValueFn: this.setTextFieldValue.bind(this)
-			}
+			},
+      charCounter: {
+        charCountContainer: this.controller.get('charCounterContainer'),
+        charCountElement: this.controller.get('charCounter'),
+        setTextFieldValueFn: this.setTextFieldValue.bind(this)
+       }
 		};
 
 		this.characterCounter = new CharacterCounter(this.controller,this.Messaging.messagingPrefs.platformType,this.messageTextElement,charCounterParams);
@@ -2115,4 +2120,4 @@ var ChatviewAssistant = Class.create({
 		this.controller.sceneScroller.mojo.revealBottom();
 		this.controller.sceneScroller.mojo.revealBottom();
 	}
-});
\ No newline at end of file
+});
diff --git a/usr/palm/applications/com.palm.app.messaging/app/controllers/compose-assistant.js b/usr/palm/applications/com.palm.app.messaging/app/controllers/compose-assistant.js
index e023a89..2d4c30d 100644
--- a/usr/palm/applications/com.palm.app.messaging/app/controllers/compose-assistant.js
+++ b/usr/palm/applications/com.palm.app.messaging/app/controllers/compose-assistant.js
@@ -140,7 +140,12 @@ var ComposeAssistant = Class.create({
 			segmentCountContainer: this.controller.get('segmentCounterContainer'),
 			segmentCountElement: this.controller.get('segmentCounter'),
 			setTextFieldValueFn: this.setTextFieldValue.bind(this)
-		}
+		},
+    charCounter: {
+      charCountContainer: this.controller.get('charCounterContainer'),
+      charCountElement: this.controller.get('charCounter'),
+      setTextFieldValueFn: this.setTextFieldValue.bind(this)
+     }
 	};
 
 	this.characterCounter = new CharacterCounter(this.controller,this.Messaging.messagingPrefs.platformType,this.messageTextElement,charCounterParams);
diff --git a/usr/palm/applications/com.palm.app.messaging/app/utilities/CharacterCounter.js b/usr/palm/applications/com.palm.app.messaging/app/utilities/CharacterCounter.js
index 9e0cbc6..ea22829 100644
--- a/usr/palm/applications/com.palm.app.messaging/app/utilities/CharacterCounter.js
+++ b/usr/palm/applications/com.palm.app.messaging/app/utilities/CharacterCounter.js
@@ -39,6 +39,10 @@ var CharacterCounter = Class.create({
 				containerElement: null,
 				valueElement: null
 			};
+      var charCounterUI = {
+        containerElement: null,
+        valueElement: null
+      };
 			
 						
 			// TODO: eventually we might want to implement some cleverness that counts
@@ -51,6 +55,11 @@ var CharacterCounter = Class.create({
 				var message = textAreaElement.value;
 				return {count:message.length, message:message};
 			};
+      var charCounterUI = {
+        containerElement: null,
+        valueElement: null
+      };
+
 
 			var setOverLimit = function(isOver) {
 				if (isOver) {
@@ -185,6 +194,20 @@ var CharacterCounter = Class.create({
 				}
 			};
 			
+      var setCurrentCharCount = function(newCharCount) { 
+        if(charCounterUI.valueElement) 
+          charCounterUI.valueElement.update(newCharCount); 
+        if(charCounterUI.containerElement) { 
+          if (newCharCount == 0) { 
+            if (charCounterUI.containerElement.visible()) 
+              charCounterUI.containerElement.hide(); 
+          } else { 
+            if (!charCounterUI.containerElement.visible()) 
+              charCounterUI.containerElement.show(); 
+          } 
+        } 
+      };
+
 			return {
 				
 				init: function(controller,platform,textElement,params) {
@@ -265,7 +288,13 @@ var CharacterCounter = Class.create({
 								setTextFieldValueFn = params.segmentCounter.setTextFieldValueFn;
 							}						
 						}
-
+            if (params.charCounter) {
+              charCounterUI.containerElement = params.charCounter.charCountContainer;
+              charCounterUI.valueElement = params.charCounter.charCountElement;
+              if (params.charCounter.setTextFieldValueFn) {
+                setTextFieldValueFn = params.charCounter.setTextFieldValueFn;
+              }
+            }
 					}					
 				},
 				
@@ -304,6 +333,7 @@ var CharacterCounter = Class.create({
 					var rawCharacterData = getRawCharacterData();
 					var messageData = parseMessage(rawCharacterData.message,rawCharacterData.count,_maxLength,true);
 					adjustedCharacterCount = messageData.adjustedCharacterCount;
+          setCurrentCharCount(rawCharacterData.count);
 					if (_maxLength > 0)
 						setOverLimit(messageData.isOverLimit);
 					setCurrentSegmentCount(messageData.segmentCount);				
@@ -329,4 +359,4 @@ var CharacterCounter = Class.create({
 		characterCounter.init(controller,platformType,textAreaElement,params);
 		Object.extend(this,characterCounter);
 	}
-});
\ No newline at end of file
+});
diff --git a/usr/palm/applications/com.palm.app.messaging/app/views/chatview/chatview-scene.html b/usr/palm/applications/com.palm.app.messaging/app/views/chatview/chatview-scene.html
index d0d8745..7ecb65a 100644
--- a/usr/palm/applications/com.palm.app.messaging/app/views/chatview/chatview-scene.html
+++ b/usr/palm/applications/com.palm.app.messaging/app/views/chatview/chatview-scene.html
@@ -22,6 +22,10 @@
 			<div id="messageContainer" class='palm-row'>
 				<div class="palm-row-wrapper textfield-group focused">
 					<div class="title">
+            <div id="charCounterContainer">
+              <div id="charCounter">
+              </div>
+            </div>
 						<div id="attachmentContainer" style="display: none;">
 							<div id="cancelAttachment">
 							</div>
diff --git a/usr/palm/applications/com.palm.app.messaging/app/views/compose/compose-scene.html b/usr/palm/applications/com.palm.app.messaging/app/views/compose/compose-scene.html
index 30daa88..238d597 100644
--- a/usr/palm/applications/com.palm.app.messaging/app/views/compose/compose-scene.html
+++ b/usr/palm/applications/com.palm.app.messaging/app/views/compose/compose-scene.html
@@ -6,6 +6,10 @@
 		<div id="messageContainer" class='palm-row'>
 			<div class="palm-row-wrapper textfield-group focused">
 				<div class="title">
+          <div id="charCounterContainer">
+            <div id="charCounter">
+            </div>
+          </div>
 					<div id="attachmentContainer" style="display: none;">
 						<div id="cancelAttachment">
 						</div>
diff --git a/usr/palm/applications/com.palm.app.messaging/stylesheets/messaging.css b/usr/palm/applications/com.palm.app.messaging/stylesheets/messaging.css
index fe6127a..ae0c424 100644
--- a/usr/palm/applications/com.palm.app.messaging/stylesheets/messaging.css
+++ b/usr/palm/applications/com.palm.app.messaging/stylesheets/messaging.css
@@ -837,6 +837,23 @@ img.avatar-overlay {
 #messageContainer.palm-row .icon.right {
 	height: 61px;
 }
+#messageContainer #charCounterContainer { 
+  line-height: 20px; 
+  display:block; 
+  height: 20px; 
+  border-width: 0px 10px 0px 9px; 
+  -webkit-border-image: url(../images/message-segment-badge.png) 0 10 0 9 stretch stretch; 
+  position: absolute; 
+  z-index: 3; 
+  top: 2px; 
+  left: 2px;
+}
+#messageContainer #charCounterContainer #charCounter {
+  font-size: 12px;
+  font-weight: bold;
+  color: #679BC2;
+  margin: 0px -4px 3px -3px;
+}
 #messageContainer #attachmentContainer {
 	position: relative;
 	margin-top:10px;
@@ -1152,4 +1169,4 @@ img.avatar-overlay {
 }
 #im-links .palm-row .title {
 	padding: 9px 14px;
-}
\ No newline at end of file
+}
