diff -ur /usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js /usr/palm/app-modified/com.palm.app.messaging/app/controllers/chatview-assistant.js
--- /usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js	2009-07-18 00:22:14.000000000 -0500
+++ /usr/palm/app-modified/com.palm.app.messaging/app/controllers/chatview-assistant.js	2009-07-23 23:13:19.000000000 -0500
@@ -160,6 +160,11 @@
 				segmentCountContainer: this.controller.get('segmentCounterContainer'),
 				segmentCountElement: this.controller.get('segmentCounter'),
 				setTextFieldValueFn: this.setTextFieldValue.bind(this)
+			},
+			charCounter: {
+				charCountContainer: this.controller.get('charCounterContainer'),
+				charCountElement: this.controller.get('charCounter'),
+				setTextFieldValueFn: this.setTextFieldValue.bind(this)
 			}
 		};
 
diff -ur /usr/palm/applications/com.palm.app.messaging/app/controllers/compose-assistant.js /usr/palm/app-modified/com.palm.app.messaging/app/controllers/compose-assistant.js
--- /usr/palm/applications/com.palm.app.messaging/app/controllers/compose-assistant.js	2009-07-18 00:22:14.000000000 -0500
+++ /usr/palm/app-modified/com.palm.app.messaging/app/controllers/compose-assistant.js	2009-07-23 23:14:11.000000000 -0500
@@ -140,6 +140,11 @@
 			segmentCountContainer: this.controller.get('segmentCounterContainer'),
 			segmentCountElement: this.controller.get('segmentCounter'),
 			setTextFieldValueFn: this.setTextFieldValue.bind(this)
+		},
+		charCounter: {
+			charCountContainer: this.controller.get('charCounterContainer'),
+			charCountElement: this.controller.get('charCounter'),
+			setTextFieldValueFn: this.setTextFieldValue.bind(this)
 		}
 	};
 
diff -ur /usr/palm/applications/com.palm.app.messaging/app/views/chatview/chatview-scene.html /usr/palm/app-modified/com.palm.app.messaging/app/views/chatview/chatview-scene.html
--- /usr/palm/applications/com.palm.app.messaging/app/views/chatview/chatview-scene.html	2009-07-18 00:22:14.000000000 -0500
+++ /usr/palm/app-modified/com.palm.app.messaging/app/views/chatview/chatview-scene.html	2009-07-23 23:16:55.000000000 -0500
@@ -22,6 +22,10 @@
 			<div id="messageContainer" class='palm-row'>
 				<div class="palm-row-wrapper textfield-group focused">
 					<div class="title">
+						<div id="charCounterContainer">
+							<div id="charCounter">
+							</div>
+						</div>
 						<div id="attachmentContainer" style="display: none;">
 							<div id="cancelAttachment">
 							</div>
diff -ur /usr/palm/applications/com.palm.app.messaging/app/views/compose/compose-scene.html /usr/palm/app-modified/com.palm.app.messaging/app/views/compose/compose-scene.html
--- /usr/palm/applications/com.palm.app.messaging/app/views/compose/compose-scene.html	2009-07-18 00:22:14.000000000 -0500
+++ /usr/palm/app-modified/com.palm.app.messaging/app/views/compose/compose-scene.html	2009-07-23 23:17:14.000000000 -0500
@@ -6,6 +6,10 @@
 		<div id="messageContainer" class='palm-row'>
 			<div class="palm-row-wrapper textfield-group focused">
 				<div class="title">
+					<div id="charCounterContainer">
+						<div id="charCounter">
+						</div>
+					</div>
 					<div id="attachmentContainer" style="display: none;">
 						<div id="cancelAttachment">
 						</div>
diff -ur /usr/palm/applications/com.palm.app.messaging/stylesheets/messaging.css /usr/palm/app-modified/com.palm.app.messaging/stylesheets/messaging.css
--- /usr/palm/applications/com.palm.app.messaging/stylesheets/messaging.css	2009-07-18 00:22:22.000000000 -0500
+++ /usr/palm/app-modified/com.palm.app.messaging/stylesheets/messaging.css	2009-07-23 23:19:25.000000000 -0500
@@ -837,6 +837,23 @@
 #messageContainer.palm-row .icon.right {
 	height: 61px;
 }
+#messageContainer #charCounterContainer {
+	line-height: 20px;
+	display:block;
+	height: 20px;
+	border-width: 0px 10px 0px 9px;
+	-webkit-border-image: url(../images/message-segment-badge.png) 0 10 0 9 stretch stretch;
+	position: absolute;
+	z-index: 3;
+	top: 2px;
+	left: 2px;
+}
+#messageContainer #charCounterContainer #charCounter {
+	font-size: 12px;
+	font-weight: bold;
+	color: #679BC2;
+	margin: 0px -4px 3px -3px;
+}
 #messageContainer #attachmentContainer {
 	position: relative;
 	margin-top:10px;

diff --git a/usr/palm/applications/com.palm.app.messaging/app/utilities/CharacterCounter.js b/usr/palm/applications/com.palm.app.messaging/app/utilities/CharacterCounter.js
index 9e0cbc6..ea22829 100644
--- a/usr/palm/applications/com.palm.app.messaging/app/utilities/CharacterCounter.js
+++ b/usr/palm/applications/com.palm.app.messaging/app/utilities/CharacterCounter.js
@@ -39,6 +39,10 @@ var CharacterCounter = Class.create({
 				containerElement: null,
 				valueElement: null
 			};
+      var charCounterUI = {
+        containerElement: null,
+        valueElement: null
+      };
 			
 						
 			// TODO: eventually we might want to implement some cleverness that counts
@@ -51,6 +55,11 @@ var CharacterCounter = Class.create({
 				var message = textAreaElement.value;
 				return {count:message.length, message:message};
 			};
+      var charCounterUI = {
+        containerElement: null,
+        valueElement: null
+      };
+
 
 			var setOverLimit = function(isOver) {
 				if (isOver) {
@@ -185,6 +194,20 @@ var CharacterCounter = Class.create({
 				}
 			};
 			
+      var setCurrentCharCount = function(newCharCount) { 
+        if(charCounterUI.valueElement) 
+          charCounterUI.valueElement.update(newCharCount); 
+        if(charCounterUI.containerElement) { 
+          if (newCharCount == 0) { 
+            if (charCounterUI.containerElement.visible()) 
+              charCounterUI.containerElement.hide(); 
+          } else { 
+            if (!charCounterUI.containerElement.visible()) 
+              charCounterUI.containerElement.show(); 
+          } 
+        } 
+      };
+
 			return {
 				
 				init: function(controller,platform,textElement,params) {
@@ -265,7 +288,13 @@ var CharacterCounter = Class.create({
 								setTextFieldValueFn = params.segmentCounter.setTextFieldValueFn;
 							}						
 						}
-
+            if (params.charCounter) {
+              charCounterUI.containerElement = params.charCounter.charCountContainer;
+              charCounterUI.valueElement = params.charCounter.charCountElement;
+              if (params.charCounter.setTextFieldValueFn) {
+                setTextFieldValueFn = params.charCounter.setTextFieldValueFn;
+              }
+            }
 					}					
 				},
 				
@@ -304,6 +333,7 @@ var CharacterCounter = Class.create({
 					var rawCharacterData = getRawCharacterData();
 					var messageData = parseMessage(rawCharacterData.message,rawCharacterData.count,_maxLength,true);
 					adjustedCharacterCount = messageData.adjustedCharacterCount;
+          setCurrentCharCount(rawCharacterData.count);
 					if (_maxLength > 0)
 						setOverLimit(messageData.isOverLimit);
 					setCurrentSegmentCount(messageData.segmentCount);				
@@ -329,4 +359,4 @@ var CharacterCounter = Class.create({
 		characterCounter.init(controller,platformType,textAreaElement,params);
 		Object.extend(this,characterCounter);
 	}
-});
\ No newline at end of file
+});

