Index: /usr/palm/applications/com.palm.app.messaging/app/controllers/compose-assistant.js
===================================================================
--- .orig/usr/palm/applications/com.palm.app.messaging/app/controllers/compose-assistant.js
+++ /usr/palm/applications/com.palm.app.messaging/app/controllers/compose-assistant.js
@@ -76,7 +76,7 @@ var ComposeAssistant = Class.create({
 
 	this.searchList = {
 		"property": "to$A",
-		"includeEmails": false,
+		"includeEmails": true,
 		"includePhones": true,
 		"includeIMs": true,
 		"includeShortCode": true,
