diff --git a/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js b/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js
index 484cd45..acc7acc 100644
--- a/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js
+++ b/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js
@@ -60,6 +60,13 @@ var ChatviewAssistant = Class.create({
 			template: new Template($L("#{chatAddress} is offline")),
 			isTransientFn: ChatFlags.isTransientOffline
 		};
+		
+		this.appMenuComposeNewItem = {
+			label: $L('Compose new...'),
+			command: 'composeNew',
+			disabled: false
+		};	
+		
 		this.messageTemplatePath = 'chatview/message/';
 		
 		this.appMenuAddPictureItem = {
@@ -84,6 +91,7 @@ var ChatviewAssistant = Class.create({
 				Mojo.Menu.cutItem,
 				Mojo.Menu.copyItem,
 				{label: $L('Copy All'), command: 'copyall'},
+				Mojo.Menu.selectAllItem,
 				Mojo.Menu.pasteItem
 			]
 		};
@@ -92,6 +100,7 @@ var ChatviewAssistant = Class.create({
 			visible: true,
 			label: $L('Chat view menu'),
 			items: [
+			    this.appMenuComposeNewItem,
 				chatEditItems,
 				this.appMenuAddPictureItem,
 				Mojo.Menu.helpItem
@@ -235,7 +244,11 @@ var ChatviewAssistant = Class.create({
 			this.handleStageFocus();
 		}
 	},
-
+	
+	loadComposeView: function(params) {
+		this.controller.stageController.pushScene('compose', params);
+	},
+   
 	cleanup: function() {
 		this.chatViewHeaderClickTarget.stopObserving(Mojo.Event.tap, this.handleHeaderTap);
 		this.chatListWidget.stopObserving(Mojo.Event.listTap, this.handleMessageTap);
@@ -325,7 +338,10 @@ var ChatviewAssistant = Class.create({
 	
 	handleCommand: function(event) {
 		if (event.type === Mojo.Event.command) {
-			if (event.command === 'addPicture') {
+			if (event.command == 'composeNew') {
+				this.loadComposeView();
+				event.stop();
+			} else if (event.command == 'addPicture') {
 				this.attachFilePicker();
 				event.stop();
 			} else if (event.command === 'blockUsername') {
@@ -1800,6 +1816,8 @@ var ChatviewAssistant = Class.create({
 			};
 				
 			this.controller.stageController.pushScene("compose", composeParams);
+		} else if (value == 'compose-new-cmd') {
+				this.loadComposeView();				
 		} else if (value === 'forward-as-email-cmd') {
 			this.forwardAsEmail({text: text});
 		} else if (value === 'copy-cmd') {
@@ -1827,6 +1845,8 @@ var ChatviewAssistant = Class.create({
 			};
 
 			this.controller.stageController.pushScene("mmsforward", composeParams);
+		} else if (value == 'compose-new-cmd') {
+				this.loadComposeView();				
 		} else if (value === 'forward-as-email-cmd') {
 			var composeParams = {
 				summary: paramsIn.summary,
@@ -1950,6 +1970,7 @@ var ChatviewAssistant = Class.create({
 					onChoose: this.handleContextMenuSelectSmsIm.bind(this, event.item.textForCopying, messageId, event.index),
 					placeNear: eventTarget.target,
 					items: [
+					    {label: $L('Compose New'), command: 'compose-new-cmd'},
 						{label: $L('Forward'), command: 'forward-cmd'},
 						{label: $L('Forward Via Email'), command: 'forward-as-email-cmd'},
 						{label: $L('Copy Text'), command: 'copy-cmd'},
@@ -1996,6 +2017,7 @@ var ChatviewAssistant = Class.create({
 						onChoose: this.handleContextMenuSelectMms.bind(this, params),
 						placeNear: mmsTarget,
 						items: [
+						    {label: $L('Compose New'), command: 'compose-new-cmd'},
 							{label: $L('View'), command: 'view-image-cmd'},
 							{label: $L('Forward'), command: 'forward-cmd'},
 							{label: $L('Forward Via Email'), command: 'forward-as-email-cmd'},
@@ -2017,6 +2039,7 @@ var ChatviewAssistant = Class.create({
 						onChoose: this.handleContextMenuSelectMms.bind(this, params),
 						placeNear: mmsTarget,
 						items: [
+						    {label: $L('Compose New'), command: 'compose-new-cmd'},
 							{label: $L('Play'), command: 'play-video-cmd'},
 							{label: $L('Forward'), command: 'forward-cmd'},
 							{label: $L('Forward Via Email'), command: 'forward-as-email-cmd'},
@@ -2036,6 +2059,7 @@ var ChatviewAssistant = Class.create({
 						onChoose: this.handleContextMenuSelectMms.bind(this, params),
 						placeNear: mmsTarget,
 						items: [
+						    {label: $L('Compose New'), command: 'compose-new-cmd'},
 							{label: $L('Add to Contacts'), command: 'add-contact-cmd'},
 							{label: $L('Forward'), command: 'forward-cmd'},
 							{label: $L('Forward Via Email'), command: 'forward-as-email-cmd'},
@@ -2106,6 +2130,7 @@ var ChatviewAssistant = Class.create({
 					onChoose: this.handleContextMenuSelectMms.bind(this, params),
 					placeNear: eventTarget.target,
 					items: [
+					    {label: $L('Compose New'), command: 'compose-new-cmd'},
 						{label: $L('Forward'), command: 'forward-cmd'},
 						{label: $L('Forward Via Email'), command: 'forward-as-email-cmd'},
 						{label: $L('Copy Text'), command: 'copy-text-cmd'},
