diff --git a/usr/palm/applications/com.palm.app.messaging/app/controllers/compose-assistant.js b/usr/palm/applications/com.palm.app.messaging/app/controllers/compose-assistant.js
index 8f9b17e..1fa84e8 100644
--- a/usr/palm/applications/com.palm.app.messaging/app/controllers/compose-assistant.js
+++ b/usr/palm/applications/com.palm.app.messaging/app/controllers/compose-assistant.js
@@ -85,7 +85,7 @@ var ComposeAssistant = Class.create({
 		this.handleRecipientsChanged = this.handleRecipientsChanged.bind(this);
 
 		this.monkeyPatchAddressingWidget();
-		this.saveDraft = true;
+		this.saveDraft = false;
 	},
 
   setup: function() {
@@ -450,7 +450,7 @@ var ComposeAssistant = Class.create({
   },
   
   cleanup: function() {
-	this.saveToDraft();
+	//this.saveToDraft();
   	
     var that = this;
 	
