diff --git a/usr/palm/applications/com.palm.app.messaging/app/controllers/listview-assistant.js b/usr/palm/applications/com.palm.app.messaging/app/controllers/listview-assistant.js
index c301f25..66ef44a 100644
--- a/usr/palm/applications/com.palm.app.messaging/app/controllers/listview-assistant.js
+++ b/usr/palm/applications/com.palm.app.messaging/app/controllers/listview-assistant.js
@@ -829,8 +829,14 @@ var ListviewAssistant = Class.create(App.Scene, {
 	// Runs on all history list item models when received from the service, before giving them to the List widget.
 	preformatHistoryItem: function(item) {
 		if (ChatFlags.isOutgoing(item.flags)) {
+			item.css_class = "my-history-chat";
 			item.messageActionImage = "images/outgoing.png";
+
+		} else if (!(ChatFlags.isRead(item.flags))) {
+			item.css_class = "new-chat";
+			item.messageActionImage = "images/incoming.png";
 		} else {
+			item.css_class = "their-chat";
 			item.messageActionImage = "images/incoming.png";
 		}
 		
diff --git a/usr/palm/applications/com.palm.app.messaging/app/views/listview/historyList-row.html b/usr/palm/applications/com.palm.app.messaging/app/views/listview/historyList-row.html
index 355c0f4..18b4480 100644
--- a/usr/palm/applications/com.palm.app.messaging/app/views/listview/historyList-row.html
+++ b/usr/palm/applications/com.palm.app.messaging/app/views/listview/historyList-row.html
@@ -1,4 +1,4 @@
-<div id="#{id}" class="palm-row chat-entry historyRow removable #{classNames}">
+<div id="#{id}" class="palm-row chat-entry historyRow removable #{css_class} #{classNames}">
 	<div class="palm-row-wrapper">
 		<div class="chat-balloon" x-mojo-tap-highlight="momentary">
 			<div class="chat-balloon-wrapper">
diff --git a/usr/palm/applications/com.palm.app.messaging/stylesheets/messaging.css b/usr/palm/applications/com.palm.app.messaging/stylesheets/messaging.css
index 019deb5..417d19c 100644
--- a/usr/palm/applications/com.palm.app.messaging/stylesheets/messaging.css
+++ b/usr/palm/applications/com.palm.app.messaging/stylesheets/messaging.css
@@ -629,6 +629,10 @@ img.avatar-overlay {
 		-webkit-border-image: url(../images/chatview/chat-balloon-left-error.png) 10 10 10 10 stretch stretch;
 }
 
+.chat-entry.new-chat .chat-balloon {
+		-webkit-border-image: url(/media/internal/scrims/chat-balloon-left-new.png) 10 10 10 10 stretch stretch;
+}
+
 .chat-entry.my-chat .chat-balloon {
 		-webkit-border-image: url(../images/chatview/chat-balloon-right.png) 10 10 10 10 stretch stretch;
 }
@@ -641,6 +645,10 @@ img.avatar-overlay {
 		-webkit-border-image: url(../images/chatview/chat-balloon-right-error.png) 10 10 10 10 stretch stretch;
 }
 
+.chat-entry.my-history-chat .chat-balloon {
+		-webkit-border-image: url(/media/internal/scrims/chat-balloon-history-left.png) 10 10 10 10 stretch stretch;
+}
+
 .palm-row.chat-entry.historyRow .chat-balloon.selected {
 		border-width: 10px;
 		-webkit-border-image: url(../images/chatview/chat-balloon-left-selected.png) 10 10 10 10 stretch stretch;
