Formats the messages to landscape when phone is rotated.
Index: /usr/palm/applications/com.palm.app.email/app/controllers/accounts-assistant.js
===================================================================
--- .orig/usr/palm/applications/com.palm.app.email/app/controllers/accounts-assistant.js
+++ /usr/palm/applications/com.palm.app.email/app/controllers/accounts-assistant.js
@@ -53,6 +53,7 @@ var AccountsAssistant = Class.create(App
 		EmailAppDepot.depot.isShowFavorites(AccountsAssistant.kFavoriteFoldersAccountID, this.depotGetFavExpanded.bind(this));
 
 		this.controller.listen(this.controller.sceneElement, Mojo.Event.keypress, this.keypressHandler.bind(this));
+                this.controller.window.PalmSystem.setWindowOrientation("free");
 	},
 
 	cleanup: function() {
Index: /usr/palm/applications/com.palm.app.email/app/controllers/compose-assistant.js
===================================================================
--- .orig/usr/palm/applications/com.palm.app.email/app/controllers/compose-assistant.js
+++ /usr/palm/applications/com.palm.app.email/app/controllers/compose-assistant.js
@@ -186,6 +186,7 @@ var ComposeAssistant = Class.create({
 
 		// Delayed a little, since we want the header part of render as quickly as possible
 		ComposeAssistant.onLoad.defer(this.controller, this.email);
+                this.controller.window.PalmSystem.setWindowOrientation("free");
 	},
 
 	cleanup: function() {
@@ -641,7 +642,8 @@ ComposeAssistant.onLoad = function(contr
 	var bodyText = email.text;
 	var originalText = email.originalText;
 	// Reduce bad scripts, styles, and margin settings by removing the html up to (and including) the body tag.
-	var splitIndex = originalText.indexOf("<body");
+	/* -START- REPLACE BY D2GLOBALINC MODIFICATION TO FIX REPLY-FORWARD FORMATTING
+        var splitIndex = originalText.indexOf("<body");
 	if (splitIndex > 0) {
 		splitIndex = originalText.indexOf('>', splitIndex);
 		if (splitIndex > 0) {
@@ -651,7 +653,23 @@ ComposeAssistant.onLoad = function(contr
 		}
 	}
 
-	// Truncate to 100K if the message is too long. 
+	*/
+        // -END- REPLACE BY D2GLOBALINC MODIFICATION TO FIX REPLY-FORWARD FORMATTING
+
+        // FIX BROKEN FORMATTING IN REPLAY-FORWARD
+        var splitIndex = originalText.indexOf("<body");
+        if (splitIndex > 0) {
+                splitIndex = originalText.indexOf('>', splitIndex);
+                if (splitIndex > 0) {
+                        splitIndex++;
+                        Mojo.Log.info("Slicing header starting at", splitIndex);
+                        originalText = originalText.slice(splitIndex);
+                }
+        } else {
+                originalText = originalText.gsub("\n","<br/>");
+        }
+
+        // Truncate to 100K if the message is too long. 
 	// cutting it off conservatively).
 	var maxBodyLength = 100000;
 	if (originalText.length > maxBodyLength) {
Index: /usr/palm/applications/com.palm.app.email/app/controllers/list-assistant.js
===================================================================
--- .orig/usr/palm/applications/com.palm.app.email/app/controllers/list-assistant.js
+++ /usr/palm/applications/com.palm.app.email/app/controllers/list-assistant.js
@@ -37,6 +37,7 @@ var ListAssistant = Class.create(App.Sce
 				{label:$L('Update'), icon:'sync', command:'sync'}
 			]};
 	this.controller.setupWidget(Mojo.Menu.commandMenu, undefined, this.cmdMenuModel);
+        this.controller.window.PalmSystem.setWindowOrientation("free");
   },
 
   // NOTE: this is called by app_scene's _setup function
