diff --git a/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js b/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js
index 484cd45..dd9de1b 100644
--- a/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js
+++ b/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js
@@ -1807,6 +1807,20 @@ var ChatviewAssistant = Class.create({
 			this.addTextToClipboard(text);
 		} else if (value === 'delete-cmd') {
 			this.handleMessageDeleteFromContextMenu(messageId, index);
+		} else if (value == 'browser-cmd') {
+			this.controller.serviceRequest("palm://com.palm.applicationManager", {
+				method: "open",
+				parameters: {
+					id: "com.palm.app.browser",
+					params: {
+					target: text}}})
+		} else if (value == 'google-cmd') {
+			this.controller.serviceRequest("palm://com.palm.applicationManager", {
+				method: "open",
+				parameters: {
+					id: "com.palm.app.browser",
+					params: {
+					target: "http://www.google.com?q=" + text}}})
 		}
 	},
 	
@@ -1953,6 +1967,8 @@ var ChatviewAssistant = Class.create({
 						{label: $L('Forward'), command: 'forward-cmd'},
 						{label: $L('Forward Via Email'), command: 'forward-as-email-cmd'},
 						{label: $L('Copy Text'), command: 'copy-cmd'},
+						{label: $L('Search in Google'), command: 'google-cmd'},
+						{label: $L('Open Link in Browser'), command: 'browser-cmd'},
 						{label: $L('Delete'), command: 'delete-cmd'}
 					]
 				});
