diff --git a/usr/palm/applications/com.palm.app.messaging/app/controllers/listview-assistant.js b/usr/palm/applications/com.palm.app.messaging/app/controllers/listview-assistant.js
index c301f25..00afc47 100644
--- a/usr/palm/applications/com.palm.app.messaging/app/controllers/listview-assistant.js
+++ b/usr/palm/applications/com.palm.app.messaging/app/controllers/listview-assistant.js
@@ -39,7 +39,8 @@ var ListviewAssistant = Class.create(App.Scene, {
     		loadHistoryView: {command:'cmdLoadHistoryView',target: this.loadHistoryView.bind(this)},
     		loadBuddyView: {command: 'cmdLoadBuddyView',target: this.loadBuddyView.bind(this)},
 			toggleOfflineBuddies: {command: 'cmdShowOfflineBuddies',target: this.toggleOfflineBuddies.bind(this)},
-			loadDebugView: {command: 'cmdLoadDebugView',target: this.loadDebugView.bind(this)}
+			loadDebugView: {command: 'cmdLoadDebugView',target: this.loadDebugView.bind(this)},
+			deleteAll: {command: 'cmdDeleteAll',target: this.handleHistoryListDeleteAllConfirm.bind(this)}//--> DELETE ALL COMMAND
 		};
 		
 		var selectedView = this.COMMAND_MENU.loadHistoryView.command;
@@ -299,7 +300,47 @@ var ListviewAssistant = Class.create(App.Scene, {
 		MessagingMojoService.deleteChatThread(this.controller, chatThreadId);
 		this.clearSpecificListBadgeForChatThreadId(chatThreadId, "buddyBageContainer", this.buddyListWidget);
 	},
+
 	
+	//--> DELETE ALL CONFIRMATION FUNCTION
+	handleHistoryListDeleteAllConfirm: function(event){
+		if (this.historyListWidget.mojo.getLength() == 0){
+			this.controller.showAlertDialog({
+				title: $L("No Conversations"),
+				message: $L("There are no conversations to delete."),
+				choices:[{label:$L("Ok"), value:"ok", type:'dismiss'}]
+			});
+		}else{
+			this.controller.showAlertDialog({
+				title: $L("Delete Conversation Logs"),
+				message: $L("Are you sure you want to delete all Conversation Logs? This cannot be undone."),
+				preventCancel: true,
+				choices:[
+					 {label:$L('Yes'), value:"yes", type:'negative'},  
+					 {label:$L("No"), value:"no", type:'affirmative'}    
+				],
+				onChoose: function(value) {
+					if (value=="yes"){
+						for (var i = 0; i < this.historyListWidget.mojo.getLength(); i++ ){
+							var chatThreadId = this.historyListWidget.mojo.getNodeByIndex(i).id;
+							Mojo.Controller.appController.notificationAssistant.clearNotificationsForChat(chatThreadId);
+							MessagingMojoService.deleteChatThread(this.controller, chatThreadId);
+							this.clearSpecificListBadgeForChatThreadId(chatThreadId, "buddyBageContainer", this.buddyListWidget);
+						}
+						
+						this.controller.modelChanged(this.historyListModel);
+				
+						this.controller.showAlertDialog({
+							title: $L("All Conversations Deleted"),
+							message: $L("All Conversations have been deleted."),
+							choices:[{label:$L("Ok"), value:"ok", type:'dismiss'}]
+						});
+					}
+				}
+			});
+		}
+	},
+
 	handleBuddyListTap: function(e) {
 		this.launchChatView(e.item.chatId, { selectIMTransport: true });
 	},
@@ -467,6 +508,16 @@ var ListviewAssistant = Class.create(App.Scene, {
 			items.push(appMenuDebugItem);
 		}
 		
+		
+		//--> DELETE ALL PATCH MENU ITEM
+		var appMenuDeleteAll = {
+			label: 'Delete All Conversations',
+			command: 'cmdDeleteAll',
+			enabled: true
+		};
+		items.push(appMenuDeleteAll);
+		
+		
 		return items;
 	},
 	
