diff --git a/usr/palm/applications/com.palm.app.messaging/app/utilities/CharacterCounter.js b/usr/palm/applications/com.palm.app.messaging/app/utilities/CharacterCounter.js
index 2aca4c6..3d27e63 100644
--- a/usr/palm/applications/com.palm.app.messaging/app/utilities/CharacterCounter.js
+++ b/usr/palm/applications/com.palm.app.messaging/app/utilities/CharacterCounter.js
@@ -46,7 +46,7 @@ var CharacterCounter = Class.create({
 			var textSavedBeforePaste;		// used to revert a paste event if the max length is exceeded
 			var selectionStartSavedBeforePaste = 0;
 			var selectionEndSavedBeforePaste = 0;
-			var multiMessageHeaderTemplate = new Template($L("(#{currentSegmentIndex} of #{lastSegmentIndex}) "));
+           var multiMessageHeaderTemplate = new Template($L(""));
 
 			var segmentCounterUI = {
 				containerElement: null,
