diff --git a/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js b/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js
index 484cd45..051205a 100644
--- a/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js
+++ b/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js
@@ -2373,6 +2373,32 @@ var ChatviewAssistant = Class.create({
 	},
 
 	handleTextAreaKeyUp: function(event) {
+    if (event && event.ctrlKey && event.keyCode==Mojo.Char.f) { 
+      currentText=this.messageTextElement.value; 
+      oldPos = this.messageTextElement.selectionStart; 
+      newPos = currentText.lastIndexOf(' ',oldPos-1); 
+      if (newPos>-1) { 
+        this.messageTextElement.setSelectionRange(newPos,newPos); 
+      } 
+      else { 
+        this.messageTextElement.setSelectionRange(0,0); 
+      } 
+      Event.stop(event); 
+    }
+
+    if (event && event.ctrlKey && event.keyCode==Mojo.Char.g ) { 
+      currentText=this.messageTextElement.value; 
+      oldPos = this.messageTextElement.selectionStart; 
+      newPos = currentText.indexOf(' ',oldPos); 
+      if (newPos>-1) { 
+        this.messageTextElement.setSelectionRange(newPos+1,newPos+1); 
+      } 
+      else { 
+        this.messageTextElement.setSelectionRange(currentText.length,currentText.length); 
+      } 
+      Event.stop(event); 
+    }
+
 		if (event && Mojo.Char.isEnterKey(event.keyCode)) {
 			this.considerForSend();
 			Event.stop(event);
