diff --git a/etc/palm/db/permissions/com.palm.message b/etc/palm/db/permissions/com.palm.message
index f072ae2..357326e 100644
--- a/etc/palm/db/permissions/com.palm.message
+++ b/etc/palm/db/permissions/com.palm.message
@@ -8,6 +8,14 @@
             "read": "allow"
         }
     },
+	{
+		"type": "db.kind",
+		"object": "com.palm.message:1",
+		"caller": "info.mobo.*",
+		"operations": {
+			"read": "allow"
+		}
+	},
     {
         "type": "db.kind",
         "object": "com.palm.message:1",
