diff --git a/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js b/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js
index 84e13e3..b5e6e89 100644
--- a/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js
+++ b/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js
@@ -335,8 +335,8 @@ var ChatviewAssistant = Class.create({
 				this.copyConversationToClipboard();
 			}
 		} else if (event.type === Mojo.Event.back) {
-			event.stop();
-			this.popAndRevealListView();
+			//event.stop();
+			//this.popAndRevealListView();
 			return;
 		}
 	},
diff --git a/usr/palm/applications/com.palm.app.messaging/app/controllers/listview-assistant.js b/usr/palm/applications/com.palm.app.messaging/app/controllers/listview-assistant.js
index c301f25..1a4e938 100644
--- a/usr/palm/applications/com.palm.app.messaging/app/controllers/listview-assistant.js
+++ b/usr/palm/applications/com.palm.app.messaging/app/controllers/listview-assistant.js
@@ -1312,34 +1312,27 @@ var ListviewAssistant = Class.create(App.Scene, {
 		});
 	},
 	
-	launchChatView: function(chatThreadId, params) {
-		if (params === undefined) {
-			params = {};
+launchChatView: function(chatThreadId) {
+	//Need to jump to that stage if it exists
+	var stageController = Mojo.Controller.appController.getStageController("messaging"+chatThreadId);
+		if (stageController) {
+			stageController.activate();
+			return;
 		}
-		params.clearListBadgeFn = this.clearListBadgeForChatThreadId.bind(this, chatThreadId);
-		this.controller.stageController.pushScene('chatview', chatThreadId, params);
-	},
-	
-	clearListBadgeForChatThreadId: function(chatThreadId) {
-		this.clearSpecificListBadgeForChatThreadId(chatThreadId, "buddyBageContainer", this.buddyListWidget);
-		this.clearSpecificListBadgeForChatThreadId(chatThreadId, "historyBageContainer", this.historyListWidget);
-	},
 
-	clearSpecificListBadgeForChatThreadId: function(chatThreadId, prefix, list) {
-		var listItem = this.controller.get(prefix + chatThreadId);
-		var itemModel;
-		
-		if (listItem) {
-			listItem.addClassName('hide-unread-count');
-			
-			itemModel = list.mojo.getItemByNode(listItem);
-			if(itemModel) {
-				itemModel.unreadCount = '0';
-			}
-		}
-	
-	},
-		
+	var params = {name: 'messaging' + chatThreadId,
+			lightweight: Mojo.Controller.appInfo.lwStages
+	};
+
+	var callback = function(controller) {
+		controller.pushScene('chatview',chatThreadId,{
+					focusWindow: true
+				});		
+	};
+
+	Mojo.Controller.getAppController().createStageWithCallback(params, callback); //doesnt create ifit exists sooo
+   },
+
 	loadDebugView: function() {
 		this.controller.stageController.pushScene('debug');
 	},
