diff --git a/usr/palm/applications/com.palm.app.messaging/app/controllers/notification-assistant.js b/usr/palm/applications/com.palm.app.messaging/app/controllers/notification-assistant.js
index 1692085..998dfdc 100644
--- a/usr/palm/applications/com.palm.app.messaging/app/controllers/notification-assistant.js
+++ b/usr/palm/applications/com.palm.app.messaging/app/controllers/notification-assistant.js
@@ -51,6 +51,19 @@ NotificationAssistant.prototype.sendNotification = function(notificationType, re
 	}
 };
 
+NotificationAssistant.prototype.doBanner = function(bannerParams,bannerLaunchParams,bannerType,payload) {
+	if (payload.messagetone.fullPath)
+		bannerParams.soundFile = payload.messagetone.fullPath;
+	this.controller.showBanner(bannerParams,bannerLaunchParams,bannerType);
+};
+
+NotificationAssistant.prototype.playMessagetone = function(payload){
+	if (payload.messagetone.fullPath)
+		this.controller.playSoundNotification('alerts',payload.messagetone.fullPath);
+	else
+		this.controller.playSoundNotification('alerts','');
+};
+
 /*
  * <ReminderCode>
  */
@@ -140,7 +153,7 @@ NotificationAssistant.prototype.considerForNotification = function(notificationD
 	
 	// check if we should only play a sound (when you are already in a chat & a new message comes in)
 	if(notificationData.get('playSoundOnly') && this.Messaging.messagingPrefs.enableNotificationSound) {
-		this.controller.playSoundNotification('alerts','');
+		MessagingMojoService.getMessagetone(this.playMessagetone.bind(this));
 		return; // don't display any visual notification
 	}	
 	
@@ -176,10 +189,10 @@ NotificationAssistant.prototype.sendClass0PopupAlert = function(notificationData
 
 	notificationData.set('alertTitle',alertTitle);
 	this.class0AlertData.list.push(notificationData);
-	this.renderClass0PopupAlert(true);
+	MessagingMojoService.getMessagetone(this.renderClass0PopupAlert.bind(this,true));
 };
 
-NotificationAssistant.prototype.renderClass0PopupAlert = function(playSound) {
+NotificationAssistant.prototype.renderClass0PopupAlert = function(playSound,messageTone) {
 	var class0Stage = this.controller.getStageController(Class0AlertStageName);
 
 	if(class0Stage) {
@@ -188,6 +201,8 @@ NotificationAssistant.prototype.renderClass0PopupAlert = function(playSound) {
 		var soundClass = 'none';
 		if(playSound) {
 			soundClass = 'alerts';
+			if(messageTone)
+				soundFile = messageTone;
 		}		
 		
 		var pushClass0AlertScene = function(stageController) {
@@ -198,7 +213,8 @@ NotificationAssistant.prototype.renderClass0PopupAlert = function(playSound) {
 			name: Class0AlertStageName,
 			lightweight: true,
 			height: 300,
-			soundclass: soundClass
+			soundclass: soundClass,
+			soundfile: soundFile
 		}, pushClass0AlertScene, 'popupalert');		
 	}
 	
@@ -253,8 +269,6 @@ NotificationAssistant.prototype.sendNewMessageNotificationAndUpdateDashboard = f
 	var bannerParams = {
 		messageText: notificationText
 	};
-	if (this.Messaging.messagingPrefs.enableNotificationSound)
-		bannerParams.soundClass = "alerts";
 	var bannerLaunchParams =  {
 		chatThreadId: chatThreadId,
 		clearBanner: true		
@@ -262,10 +276,14 @@ NotificationAssistant.prototype.sendNewMessageNotificationAndUpdateDashboard = f
 	
 	if (this.Messaging.DisplayState.isDisplayOn()) {
 		Mojo.Log.info("notificationAssistant - executing full banner notification");	
+		if (this.Messaging.messagingPrefs.enableNotificationSound) {
+			bannerParams.soundClass = "alerts";
+			MessagingMojoService.getMessagetone(this.doBanner.bind(this,bannerParams,bannerLaunchParams,'chat'));
+		} else
 		this.controller.showBanner(bannerParams, bannerLaunchParams, 'chat');		
 	} else if (this.Messaging.messagingPrefs.enableNotificationSound) {
 		Mojo.Log.info("notificationAssistant - playing sound notification only");
-		this.controller.playSoundNotification('alerts','');
+		MessagingMojoService.getMessagetone(this.playMessagetone.bind(this));
 	}
 	
 	// Store the data so it can be used in the dashboard
@@ -324,14 +342,14 @@ NotificationAssistant.prototype.sendNewMessageNotificationAndUpdateDashboard = f
 		// delay creating the dashboard window for the case where the banner is clicked on
 		// to take you to the chat view.  This will likely result in the dashboard data
 		// being cleared.  If the dashboard data is empty, we do not need to create the dashboard.		
-		if (this.Messaging.DisplayState.isDisplayOn()) {
+		//if (this.Messaging.DisplayState.isDisplayOn()) {
 			if (!this.isNewMessageDashboardPending) {
 				this.isNewMessageDashboardPending = true; 
 				createDashboard.delay(5);
 			}
-		} else {
-			createDashboard(); // if the screen is off, create the dashboard right away
-		}
+	//	} else {
+	//		createDashboard(); // if the screen is off, create the dashboard right away
+	//	}
 	}
 };
 
diff --git a/usr/palm/applications/com.palm.app.messaging/app/models/messaging-luna-service.js b/usr/palm/applications/com.palm.app.messaging/app/models/messaging-luna-service.js
index 808aaa7..862d04c 100644
--- a/usr/palm/applications/com.palm.app.messaging/app/models/messaging-luna-service.js
+++ b/usr/palm/applications/com.palm.app.messaging/app/models/messaging-luna-service.js
@@ -9,6 +9,16 @@ var MessagingMojoService = {
   telephonyIdentifier: 'palm://com.palm.telephony',
   systemManagerIdentifier: 'palm://com.palm.systemmanager',
 
+  getMessagetone:  function(callback) {
+  	var request = new Mojo.Service.Request('palm://com.palm.systemservice', {
+  		method: 'getPreferences',
+  		parameters: {"keys":["messagetone"]},
+  		onSuccess: callback,
+  		onFailure: callback
+  	});
+  	return request;
+  },
+  
   launchMessaging: function(sceneController,params) {  
 	return sceneController.serviceRequest("palm://com.palm.applicationManager", {
 		method:'launch',
