diff --git a/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js b/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js
index 484cd45..d8c85ef 100644
--- a/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js
+++ b/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js
@@ -81,6 +81,7 @@ var ChatviewAssistant = Class.create({
 		var chatEditItems = {
 			label: $L('Edit'),
 			items: [
+				Mojo.Menu.selectAllItem,
 				Mojo.Menu.cutItem,
 				Mojo.Menu.copyItem,
 				{label: $L('Copy All'), command: 'copyall'},
