diff --git a/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js b/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js
index 484cd45..89d1a48 100644
--- a/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js
+++ b/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js
@@ -1140,6 +1140,7 @@ var ChatviewAssistant = Class.create({
 		} else if (update.descriptions && update.descriptions.msgDeleted) {
 			// do nothing.  The list will remove the item so we can just ignore this event
 			// We already reset the list when we sent the delete request to the service
+			this.updateMsgCount();
 		} else {
 			// update the count and reload the window
 			var currentListSize = listWidget.mojo.getLength();
@@ -1290,7 +1291,7 @@ var ChatviewAssistant = Class.create({
 				}
 			}).defer();
 		}
-		
+		this.updateMsgCount();
 		this.debouncedListLoaded();
 		//Mojo.Timing.pause("ChatviewAssistant#itemRendered");
 	},
@@ -1767,6 +1768,7 @@ var ChatviewAssistant = Class.create({
 			
 					this.listReset = true;
 					MessagingMojoService.deleteMessage(this.controller, messageId);
+					this.updateMsgCount();
 				}
 			},
 			title: $L("Delete Message"),
@@ -1866,6 +1868,11 @@ var ChatviewAssistant = Class.create({
 		}
 	},
 	
+	updateMsgCount: function() {
+		var newCount = this.chatListWidget.mojo.getLength();
+		this.chatViewRecipientName.update("(" + newCount + ")" + " " + this.chatDisplayName.escapeHTML());
+	},
+	
 	/**
 	 * handler for message tap within chat list
 	 *
