diff --git a/media/cryptofs/apps/usr/palm/applications/com.palm.app.messaging/app/conversations/ConversationList.js b/media/cryptofs/apps/usr/palm/applications/com.palm.app.messaging/app/conversations/ConversationList.js
index afec0ca..a6c097f 100644
--- a/media/cryptofs/apps/usr/palm/applications/com.palm.app.messaging/app/conversations/ConversationList.js
+++ b/media/cryptofs/apps/usr/palm/applications/com.palm.app.messaging/app/conversations/ConversationList.js
@@ -864,7 +864,7 @@ enyo.kind({
 		this.$.richText.setValue("");
 		
 		// play a sound for sending message
-		this.playSoundNotification({ isSent: true });
+		//this.playSoundNotification({ isSent: true });
 		
 //todo: keep it to remind something similiar
 //		MessagingUtils.checkAirplaneMode(params);
