diff --git a/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js b/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js
index 0c816ba..37f0501 100644
--- a/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js
+++ b/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js
@@ -1212,6 +1212,9 @@ var ChatviewAssistant = Class.create({
 	preFormatChatList: function(stuff){
 		for(var i=0; i<stuff.list.length; i++) {
 			var msg = stuff.list[i];
+			var ts = new Date();
+			ts.setTime(msg.timeStamp);
+			msg.timeStampText = Mojo.Format.formatDate(ts,'short');
 			if(msg.summary && !ChatFlags.isTransient(msg.flags)) {
 				msg.summary = msg.summary.escapeHTML();
 				msg.summary = PalmSystem.runTextIndexer(msg.summary);
@@ -1235,7 +1238,7 @@ var ChatviewAssistant = Class.create({
 		 	var deviceTimeStampMinutes = d.getMinutes();
 			d.setMinutes((deviceTimeStampMinutes - (deviceTimeStampMinutes % 5)));
 			d.setSeconds(0);
-			msg.dividerText = msg.transportClass + BucketDateFormatter.getDateBucket(d,true,true);
+			msg.dividerText = msg.transportClass; // + BucketDateFormatter.getDateBucket(d,true,true);
 	
 			// if the INCOMING message was sent longer than 10 minutes before it arrived on device,
 			// add sent date to indicate that this delay happened
diff --git a/usr/palm/applications/com.palm.app.messaging/app/views/chatview/message/message-text.html b/usr/palm/applications/com.palm.app.messaging/app/views/chatview/message/message-text.html
index 97ff521..95b2ba6 100644
--- a/usr/palm/applications/com.palm.app.messaging/app/views/chatview/message/message-text.html
+++ b/usr/palm/applications/com.palm.app.messaging/app/views/chatview/message/message-text.html
@@ -1 +1 @@
-<div class="messageText">#{-messageText} #{-errorContent}</div>
\ No newline at end of file
+<div class="messageText"><span class="timeStamp">#{-timeStampText}</span><pre>#{-messageText}</pre> #{-errorContent}</div>
diff --git a/usr/palm/applications/com.palm.app.messaging/stylesheets/messaging.css b/usr/palm/applications/com.palm.app.messaging/stylesheets/messaging.css
index fe6127a..5eceb3d 100644
--- a/usr/palm/applications/com.palm.app.messaging/stylesheets/messaging.css
+++ b/usr/palm/applications/com.palm.app.messaging/stylesheets/messaging.css
@@ -659,6 +659,18 @@ img.avatar-overlay {
 	font-size:14px;
 }
 
+.messageText pre {
+	font-family: inherit;
+	font-size: inherit;
+	color: inherit;
+}
+.timeStamp {
+	font-size: 12px;
+font-st	yle: italic;
+	color: #1111ff;
+}
+
+
 .my-chat .chat-balloon-wrapper, 
 .their-chat .chat-balloon-wrapper {
   padding: 1px 0;
