diff --git a/usr/palm/applications/com.palm.app.messaging/app/controllers/dashboardMessage-assistant.js b/usr/palm/applications/com.palm.app.messaging/app/controllers/dashboardMessage-assistant.js
index fe431e5..1074f14 100644
--- a/usr/palm/applications/com.palm.app.messaging/app/controllers/dashboardMessage-assistant.js
+++ b/usr/palm/applications/com.palm.app.messaging/app/controllers/dashboardMessage-assistant.js
@@ -73,6 +73,25 @@ DashboardMessageAssistant.prototype.update = function(isNewContent) {
 	if(isNewContent) {
 		this.controller.stageController.indicateNewContent(true);
 	}
+
+    this.neato_cookie = new Mojo.Model.Cookie('neatoMessages');
+    var neato = this.neato_cookie.get();
+    if(!neato)
+    {
+    	this.neato_cookie.put({neatoMessages: 'heckyeah'});
+		neato = {neatoMessages: "heckyeah"};
+    }
+	
+	if (neato.neatoMessages == "heckyeah")     
+	{
+	    this.controller.serviceRequest('palm://com.palm.applicationManager', {
+	        method: 'open',
+	        parameters: {
+	            id: 'com.zhephree.neato',
+	            params: {send: '{"a":"text","c":"'+ conversationText +'"}'} 
+	        }
+	    });		
+	}
 };
 
 DashboardMessageAssistant.prototype.removePresence = function(response) {
diff --git a/usr/palm/applications/com.palm.app.messaging/app/controllers/listview-assistant.js b/usr/palm/applications/com.palm.app.messaging/app/controllers/listview-assistant.js
index c301f25..26a95b2 100644
--- a/usr/palm/applications/com.palm.app.messaging/app/controllers/listview-assistant.js
+++ b/usr/palm/applications/com.palm.app.messaging/app/controllers/listview-assistant.js
@@ -56,8 +56,17 @@ var ListviewAssistant = Class.create(App.Scene, {
 			]}
 		];
 		
+		this.neato_cookie = new Mojo.Model.Cookie('neatoMessages');
+	    var neato = this.neato_cookie.get();
+	    if(!neato)
+	    {
+	    	this.neato_cookie.put({neatoMessages: 'heckyeah'});
+			neato = {neatoMessages: "heckyeah"};
+	    }	
+		
 		this.commandMenuModel = [
 			{label:$L('Compose new message'), icon:'newchat', command:this.COMMAND_MENU.loadComposeView.command}
+			,{},{toggleCmd: neato.neatoMessages, items:[{label: $L('Yes'), command: 'heckyeah'}, {label: $L('No'), command: 'heckno'} ] }
 		];
 
 		this.availabilities = [
@@ -405,6 +414,8 @@ var ListviewAssistant = Class.create(App.Scene, {
 	handleCommand: function(event) {
 		// handle menu button command events
 		if (event.type == Mojo.Event.command) {
+			if(event.command=="heckyeah") this.neato_cookie.put({neatoMessages: 'heckyeah'});
+			if(event.command=="heckno") this.neato_cookie.put({neatoMessages: 'heckno'});
 			// determine if we have a target function for this command
 			var menu = $H(this.COMMAND_MENU);
 			menu.keys().each(function(key) {
