diff --git a/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js b/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js
index 46c3d51..ca7860f 100644
--- a/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js
+++ b/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js
@@ -2076,6 +2076,7 @@ var ChatviewAssistant = Class.create({
 				if (isOutgoingMsg) {
 					msg.messageText = msg.messageText.escapeHTML();
 				}
+				msg.messageText=msg.messageText.replace(/\n/g,"<br>");
 				//msg.messageText = PalmSystem.runTextIndexer(msg.messageText);
 				delayed.messageText = msg.messageText;
 			}
