diff --git a/usr/palm/applications/com.palm.app.messaging/app/utilities/CharacterCounter.js b/usr/palm/applications/com.palm.app.messaging/app/utilities/CharacterCounter.js
index 2aca4c6..d0b973f 100644
--- a/usr/palm/applications/com.palm.app.messaging/app/utilities/CharacterCounter.js
+++ b/usr/palm/applications/com.palm.app.messaging/app/utilities/CharacterCounter.js
@@ -52,6 +52,10 @@ var CharacterCounter = Class.create({
 				containerElement: null,
 				valueElement: null
 			};
+			var charCounterUI = {
+				containerElement: null,
+				valueElement: null
+			};
 			
 						
 			// TODO: eventually we might want to implement some cleverness that counts
@@ -407,6 +411,20 @@ var CharacterCounter = Class.create({
 				}
 			};
 			
+			var setCurrentCharCount = function(newCharCount) {
+				if(charCounterUI.valueElement)
+					charCounterUI.valueElement.update(currentSegmentCount + ':' + newCharCount);
+				if(charCounterUI.containerElement) {
+					if (newCharCount == 0) {
+						if (charCounterUI.containerElement.visible())
+							charCounterUI.containerElement.hide();
+					} else {
+						if (!charCounterUI.containerElement.visible())
+							charCounterUI.containerElement.show();
+					}
+				}
+			};
+			
 			return {
 				
 				init: function(controller,platform,textElement,params) {
@@ -514,7 +532,13 @@ var CharacterCounter = Class.create({
 								setTextFieldValueFn = params.segmentCounter.setTextFieldValueFn;
 							}						
 						}
-
+						if(params.charCounter) {
+							charCounterUI.containerElement = params.charCounter.charCountContainer;
+							charCounterUI.valueElement = params.charCounter.charCountElement;
+							if(params.charCounter.setTextFieldValueFn) {
+								setTextFieldValueFn = params.charCounter.setTextFieldValueFn;
+							}						
+						}
 					}					
 				},
 				
@@ -584,7 +608,8 @@ var CharacterCounter = Class.create({
 					adjustedCharacterCount = messageData.adjustedCharacterCount;
 					if (adjustedMaxLen > 0)
 						setOverLimit(messageData.isOverLimit);
-					setCurrentSegmentCount(messageData.segmentCount);				
+					setCurrentSegmentCount(messageData.segmentCount);	
+					setCurrentCharCount(rawCharacterData.count);							
 				},
 				
 				deferredResetCount: function() {
