diff --git a/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js b/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js
index 71c2226..1ac2b58 100644
--- a/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js
+++ b/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js
@@ -170,12 +170,13 @@ var ChatviewAssistant = Class.create({
 			visible: true
 		};
 		
-		this.appMenuAddPictureItem = {
+		/*this.appMenuAddPictureItem = {
 			label: $L('Attach file...'),
 			command: ChatviewAssistant.kMenuCmdAddPicture,
 			disabled: false,
 			visible: true
-		};
+		};*/
+		this.appMenuAddPictureItem = {label: $L('Attach file...'), shortcut: 'p', command:'addPicture', enabled:true};
 
 		this.appMenuBlockUsernameItem = {
 			label: $L('Block this person'),
@@ -688,6 +689,8 @@ var ChatviewAssistant = Class.create({
 					Mojo.Log.info("[CV]: unhandled command", event.command);
 					break;
 			}
+		} else if (event.type == Mojo.Event.forward) { 
+			this.considerForSend();			
 		} else if (event.type === Mojo.Event.back) {
 			event.stop();
 			this.popAndRevealListView();
diff --git a/usr/palm/applications/com.palm.app.messaging/app/controllers/compose-assistant.js b/usr/palm/applications/com.palm.app.messaging/app/controllers/compose-assistant.js
index f9eda28..55425d3 100644
--- a/usr/palm/applications/com.palm.app.messaging/app/controllers/compose-assistant.js
+++ b/usr/palm/applications/com.palm.app.messaging/app/controllers/compose-assistant.js
@@ -15,7 +15,7 @@ var ComposeAssistant = Class.create({
 			this.toggleDebugMenu = listviewSceneAssistant.toggleDebugMenu.bind(listviewSceneAssistant,true);
 		}
 
-		this.appMenuAddPictureItem = {label: $L('Add file...'), command:'addPicture', enabled:true};
+		this.appMenuAddPictureItem = {label: $L('Attach file...'), shortcut: 'p', command:'addPicture', enabled:true};
 
 		this.appMenuAttrs = {omitDefaultItems: true};	
 		this.appMenuModel = { visible:true, label: $L('Chat view menu'), items: [Mojo.Menu.editItem,this.appMenuAddPictureItem,Mojo.Menu.helpItem]};   
@@ -826,7 +826,9 @@ var ComposeAssistant = Class.create({
 			if(event.command === 'addPicture') {
 				this.attachFilePicker();
 				event.stop();
-			} 
+			}
+		} else if (event.type == Mojo.Event.forward) {	
+	        this.considerForSend();				
 		} else if(event.type === Mojo.Event.back) {
 			if (this.launchedAsForward === undefined) {
 				event.stop();
diff --git a/usr/palm/applications/com.palm.app.messaging/app/views/chatview/chatview-scene.html b/usr/palm/applications/com.palm.app.messaging/app/views/chatview/chatview-scene.html
index e1f7235..9ce791a 100644
--- a/usr/palm/applications/com.palm.app.messaging/app/views/chatview/chatview-scene.html
+++ b/usr/palm/applications/com.palm.app.messaging/app/views/chatview/chatview-scene.html
@@ -30,10 +30,10 @@
 							</div>
 							<img id="attachmentImage" src=""/>
 						</div>
-						<div id="ActionButton" class="icon right empty">
+						<div id="ActionButton" class="icon right empty" style="display:none;">
 							<div id="AttachButton" class="stuck attachButton" style="display:none;" x-mojo-tap-highlight="momentary">
 							</div>
-							<div id="SendButton" class="stuck sendButton" x-mojo-tap-highlight="immediate">
+							<div id="SendButton" class="stuck sendButton" style="display:none;" x-mojo-tap-highlight="immediate">
 								<div id="segmentCounterContainer">
 									<div id="segmentCounter">
 									</div>
diff --git a/usr/palm/applications/com.palm.app.messaging/app/views/compose/compose-scene.html b/usr/palm/applications/com.palm.app.messaging/app/views/compose/compose-scene.html
index 79fca4a..c3252e3 100644
--- a/usr/palm/applications/com.palm.app.messaging/app/views/compose/compose-scene.html
+++ b/usr/palm/applications/com.palm.app.messaging/app/views/compose/compose-scene.html
@@ -15,10 +15,10 @@
 						</div>
 						<img id="attachmentImage" src=""/>
 					</div>
-					<div id="ActionButton" class="icon right empty">
+					<div id="ActionButton" class="icon right empty" style="display:none;">
 						<div id="AttachButton" class="stuck attachButton" x-mojo-tap-highlight="momentary">
 						</div>
-						<div id="SendButton" class="stuck sendButton" x-mojo-tap-highlight="momentary">
+						<div id="SendButton" class="stuck sendButton" style="display:none;" x-mojo-tap-highlight="momentary">
 							<div id="segmentCounterContainer">
 								<div id="segmentCounter">
 								</div>
