diff --git a/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js b/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js
index 46c3d51..753b0a1 100644
--- a/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js
+++ b/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js
@@ -316,7 +316,7 @@ var ChatviewAssistant = Class.create({
 			hintText: $L('Enter message here...'),
 			multiline: true,
 			focus: false,
-			enterSubmits: true,
+			enterSubmits: false,
 			inputName: "messageTextElement",
 			className: " ",
 			changeOnKeyPress: true
@@ -2078,6 +2078,7 @@ var ChatviewAssistant = Class.create({
 				}
 				//msg.messageText = PalmSystem.runTextIndexer(msg.messageText);
 				delayed.messageText = msg.messageText;
+				msg.messageText = msg.messageText.replace(/\n/g,"<br>");
 			}
 			
 			var icons = this.Messaging.IMAccounts.getIcons(msg.serviceName);
@@ -2954,7 +2955,7 @@ var ChatviewAssistant = Class.create({
 		// Check for return keys, and put a fake highlight on the send button.
 		// It will be cleared on key up.
 		if (event) {
-			if (Mojo.Char.isEnterKey(event.keyCode)) {
+			if (Mojo.Char.isEnterKey(event.keyCode) && !this.ctrlKeyPressed) {
 				this.sendButton.addClassName('selected');
 			} else if (event.keyCode===129) {
 				this.ctrlKeyPressed = true;
@@ -2966,21 +2967,19 @@ var ChatviewAssistant = Class.create({
 				};
 				var emoticonPicker = new EmoticonPickerDialogAssistant(this, onselect);
 				emoticonPicker.show();
-			}
+			} 
 		}
 	},
 
 	handleTextAreaKeyUp: function(event) {
 		if (event) {
-			if (Mojo.Char.isEnterKey(event.keyCode)) {
+			if (event.keyCode===129) {
+				this.ctrlKeyPressed = false;
+			} else if (Mojo.Char.isEnterKey(event.keyCode) && !this.ctrlKeyPressed) {
 				this.considerForSend();
 				Event.stop(event);
-				
-				// Remove highlight that was added on keydown.
-				this.sendButton.removeClassName('selected');
-			} else if (event.keyCode===129) {
-				this.ctrlKeyPressed = false;
-			}
+ 				this.sendButton.removeClassName('selected');
+			} 
 		}
 	},
 
diff --git a/usr/palm/applications/com.palm.app.messaging/app/controllers/compose-assistant.js b/usr/palm/applications/com.palm.app.messaging/app/controllers/compose-assistant.js
index f37e579..f80efce 100644
--- a/usr/palm/applications/com.palm.app.messaging/app/controllers/compose-assistant.js
+++ b/usr/palm/applications/com.palm.app.messaging/app/controllers/compose-assistant.js
@@ -122,6 +122,7 @@ var ComposeAssistant = Class.create({
 		this.considerForSend = this.considerForSend.bind(this);
 		this.attachFilePicker = this.attachFilePicker.bind(this);
 		this.handleTextAreaChange = this.handleTextAreaChange.bind(this);
+		this.handleTextAreaKeyDown = this.handleTextAreaKeyDown.bind(this);
 		this.handleTextAreaKeyUp = this.handleTextAreaKeyUp.bind(this);
 		this.handleCancelAttachment = this.resetTextBox.bind(this,false);		
 		this.handleRecipientsChanged = this.handleRecipientsChanged.bind(this);
@@ -170,7 +171,7 @@ var ComposeAssistant = Class.create({
 		    hintText: $L('Enter message here...'),
 		    multiline: true,
 		    focus: (this.prePopulatedRecipients.length > 0),
-		    enterSubmits: true,
+		    enterSubmits: false,
 			requiresEnterKey: true,
 		    modelProperty: "value",
 			changeOnKeyPress: true
@@ -209,6 +210,7 @@ var ComposeAssistant = Class.create({
 		this.addressingWidget = this.controller.get('searchList');
 		this.addressingWidgetSpacer = this.controller.get('addressingWidgetSpacer');
 
+		this.messageTextElement.observe('keydown', this.handleTextAreaKeyDown);
 		this.messageTextElement.observe('keyup', this.handleTextAreaKeyUp);
 		this.messageTextWidget.observe(Mojo.Event.propertyChange, this.handleTextAreaChange);
 	
@@ -687,12 +689,24 @@ var ComposeAssistant = Class.create({
 			this.handleTextAreaChange();			
 		} 
 	},
-
+	
+	handleTextAreaKeyDown: function(event) {
+		// Check for return keys, and put a fake highlight on the send button.
+		// It will be cleared on key up.
+		if (event && event.keyCode===129) {
+				this.ctrlKeyPressed = true;
+		}
+	},
+	
 	handleTextAreaKeyUp: function(event) {
-		if (event && Mojo.Char.isEnterKey(event.keyCode)) {
-			this.considerForSend();
-			Event.stop(event);
-		}		
+		if (event) {
+			if (event.keyCode===129) {
+				this.ctrlKeyPressed = false;
+			} else if (Mojo.Char.isEnterKey(event.keyCode) && !this.ctrlKeyPressed) {
+				this.considerForSend();
+				Event.stop(event);
+			}
+		}	
 	},
 
 	handleTextAreaChange: function() {
@@ -846,6 +860,7 @@ var ComposeAssistant = Class.create({
 	
 		this.controller.stopListening('searchList', Mojo.Event.addressingMaxRecipients, this.handleMaxRecipients);
 	
+		this.messageTextElement.stopObserving('keydown', this.handleTextAreaKeyDown);
 		this.messageTextElement.stopObserving('keyup', this.handleTextAreaKeyUp);
 		this.messageTextWidget.stopObserving(Mojo.Event.propertyChange, this.handleTextAreaChange);
 	
