--- .orig/usr/palm/applications/com.palm.app.messaging/app/controllers/notification-assistant.js
+++ /usr/palm/applications/com.palm.app.messaging/app/controllers/notification-assistant.js
@@ -157,6 +157,10 @@
 	
 	if (Object.keys(notificationData).length > 0 && this.Messaging.messagingPrefs.enableNotification) {
 		var notificationType = notificationData.notificationType;
+		this.serviceRequest = new Mojo.Service.Request					
+		("palm://com.palm.vibrate", {
+		method: 'vibrate', parameters: { 'period': 0,'duration': 1500 }
+		});
 		if (notificationType == this.Messaging.notificationTypes.newMessage) {
 			// if the new message is an MMS, retrieve the part data
 			if (notificationData.messageType == "MMS") {
