diff --git a/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js b/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js
index 484cd45..d35d6ee 100644
--- a/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js
+++ b/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js
@@ -134,7 +134,7 @@ var ChatviewAssistant = Class.create({
 			hintText: $L('Enter message here...'),
 			multiline: true,
 			focus: false,
-			enterSubmits: true,
+			enterSubmits: false,
 			textFieldName: "messageTextElement",
 			className: " ",
 			changeOnKeyPress: true
@@ -1463,6 +1463,7 @@ var ChatviewAssistant = Class.create({
 				if (ChatFlags.isOutgoing(msg.flags)) {
 					msg.messageText = msg.messageText.escapeHTML();
 				}
+				msg.messageText=msg.messageText.replace(/\n/g,"<br>");
 				//msg.messageText = PalmSystem.runTextIndexer(msg.messageText);
 				delayed.messageText = msg.messageText;
 			}
@@ -2373,13 +2374,13 @@ var ChatviewAssistant = Class.create({
 	},
 
 	handleTextAreaKeyUp: function(event) {
-		if (event && Mojo.Char.isEnterKey(event.keyCode)) {
-			this.considerForSend();
-			Event.stop(event);
+		//if (event && Mojo.Char.isEnterKey(event.keyCode)) {
+		//	this.considerForSend();
+		//	Event.stop(event);
 			
 			// Remove highlight that was added on keydown.
-			this.sendButton.removeClassName('selected');
-		}
+		//	this.sendButton.removeClassName('selected');
+		//}
 	},
 
 	handleTextAreaChange: function() {
diff --git a/usr/palm/applications/com.palm.app.messaging/app/controllers/compose-assistant.js b/usr/palm/applications/com.palm.app.messaging/app/controllers/compose-assistant.js
index 3183de4..61ee5c2 100644
--- a/usr/palm/applications/com.palm.app.messaging/app/controllers/compose-assistant.js
+++ b/usr/palm/applications/com.palm.app.messaging/app/controllers/compose-assistant.js
@@ -120,7 +120,7 @@ var ComposeAssistant = Class.create({
 	    hintText: $L('Enter message here...'),
 	    multiline: true,
 	    focus: false,
-	    enterSubmits: true,
+	    enterSubmits: false,
 		requiresEnterKey: true,
 	    modelProperty: "value",
 		changeOnKeyPress: true
@@ -327,10 +327,10 @@ var ComposeAssistant = Class.create({
 	},
 
 	handleTextAreaKeyUp: function(event) {
-		if (event && Mojo.Char.isEnterKey(event.keyCode)) {
-			this.considerForSend();
-			Event.stop(event);
-		}		
+		//if (event && Mojo.Char.isEnterKey(event.keyCode)) {
+		//	this.considerForSend();
+		//	Event.stop(event);
+		//}		
 	},
 
 	handleTextAreaChange: function() {
