Each messaging conversation opens in a new window.
Index: /usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js
===================================================================
--- .orig/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js
+++ /usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js
@@ -318,8 +318,8 @@ var ChatviewAssistant = Class.create({
 				this.copyConversationToClipboard();
 			}
 		} else if(event.type == Mojo.Event.back) {
-			event.stop();
-			this.popAndRevealListView();
+			// event.stop();
+			// this.popAndRevealListView();
 		 	return;
 		 }
 	},
Index: /usr/palm/applications/com.palm.app.messaging/app/controllers/listview-assistant.js
===================================================================
--- .orig/usr/palm/applications/com.palm.app.messaging/app/controllers/listview-assistant.js
+++ /usr/palm/applications/com.palm.app.messaging/app/controllers/listview-assistant.js
@@ -1069,20 +1069,27 @@ var ListviewAssistant = Class.create(App
     }, {popScenesTo: CONSTANTS.SCENE_ID_ACCOUNT_SUMMARY});
   },
   
-  launchChatView: function(chatThreadId) {
-    this.controller.stageController.pushScene('chatview',chatThreadId,{clearListBadgeFn:this.clearListBadgeForChatThreadId.bind(this,chatThreadId)});		
-  },
-  
-  clearListBadgeForChatThreadId: function(chatThreadId) {
-	var badgeContainers = ["buddyBageContainer","historyBageContainer"];
-	var listItem;
-	for (var i = 0; i < badgeContainers.length; i++) {
-		listItem = this.controller.get(badgeContainers[i] + chatThreadId);
-		if (listItem && !listItem.hasClassName('hide-unread-count')) {
-			listItem.addClassName('hide-unread-count');
-		}
-	}	
+launchChatView: function(chatThreadId) {
+	//Need to jump to that stage if it exists
+	var stageController = Mojo.Controller.appController.getStageController("messaging"+chatThreadId);
+		if (stageController) {
+			stageController.activate();
+			return;
+		}
+
+	var params = {name: 'messaging' + chatThreadId,
+			lightweight: Mojo.Controller.appInfo.lwStages
+	};
+
+	var callback = function(controller) {
+		controller.pushScene('chatview',chatThreadId,{
+					focusWindow: true
+				});		
+	};
+
+	Mojo.Controller.getAppController().createStageWithCallback(params, callback); //doesnt create ifit exists sooo
   },
+
 	  
   loadDebugView: function() {
     this.controller.stageController.pushScene('debug');
