diff --git a/usr/palm/applications/com.palm.app.messaging/app/controllers/notification-assistant.js b/usr/palm/applications/com.palm.app.messaging/app/controllers/notification-assistant.js
index 5a5aec0..7c02c7a 100644
--- a/usr/palm/applications/com.palm.app.messaging/app/controllers/notification-assistant.js
+++ b/usr/palm/applications/com.palm.app.messaging/app/controllers/notification-assistant.js
@@ -245,6 +245,10 @@ NotificationAssistant.prototype.considerForNotification = function(notificationD
 	//TODO: does sendFailure and connectionFailure need displayName?
 	if (Object.keys(notificationData).length > 0 && this.Messaging.messagingPrefs.getNotificationsEnabled()) {
 		var notificationType = notificationData.notificationType;
+		this.serviceRequest = new Mojo.Service.Request					
+		("palm://com.palm.vibrate", {
+		method: 'vibrate', parameters: { 'period': 0,'duration': 1000 }
+		});
 		if (notificationType === this.Messaging.notificationTypes.newMessage) {
 			this.preSendNewMessageNotificationAndUpdateDashboard(notificationData);
 		} else if (notificationType === this.Messaging.notificationTypes.connectionFailure) {
