diff --git a/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js b/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js
index 57d3375..48629b2 100644
--- a/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js
+++ b/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js
@@ -190,6 +190,11 @@ var ChatviewAssistant = Class.create({
 				segmentCountContainer: this.controller.get('segmentCounterContainer'),
 				segmentCountElement: this.controller.get('segmentCounter'),
 				setTextFieldValueFn: this.setTextFieldValue.bind(this)
+			},
+			charCounter: {
+				charCountContainer: this.controller.get('charCounterContainer'),
+				charCountElement: this.controller.get('charCounter'),
+				setTextFieldValueFn: this.setTextFieldValue.bind(this)
 			}
 		};
 
diff --git a/usr/palm/applications/com.palm.app.messaging/app/controllers/compose-assistant.js b/usr/palm/applications/com.palm.app.messaging/app/controllers/compose-assistant.js
index 9715c78..bde693c 100644
--- a/usr/palm/applications/com.palm.app.messaging/app/controllers/compose-assistant.js
+++ b/usr/palm/applications/com.palm.app.messaging/app/controllers/compose-assistant.js
@@ -147,6 +147,11 @@ var ComposeAssistant = Class.create({
 			segmentCountContainer: this.controller.get('segmentCounterContainer'),
 			segmentCountElement: this.controller.get('segmentCounter'),
 			setTextFieldValueFn: this.setTextFieldValue.bind(this)
+		},
+		charCounter: {
+			charCountContainer: this.controller.get('charCounterContainer'),
+			charCountElement: this.controller.get('charCounter'),
+			setTextFieldValueFn: this.setTextFieldValue.bind(this)
 		}
 	};
 
diff --git a/usr/palm/applications/com.palm.app.messaging/app/utilities/CharacterCounter.js b/usr/palm/applications/com.palm.app.messaging/app/utilities/CharacterCounter.js
index a5efef9..24c54ab 100644
--- a/usr/palm/applications/com.palm.app.messaging/app/utilities/CharacterCounter.js
+++ b/usr/palm/applications/com.palm.app.messaging/app/utilities/CharacterCounter.js
@@ -54,6 +54,10 @@ var CharacterCounter = Class.create({
 				containerElement: null,
 				valueElement: null
 			};
+			var charCounterUI = {
+				containerElement: null,
+				valueElement: null
+			};
 			
 						
 			// TODO: eventually we might want to implement some cleverness that counts
@@ -297,6 +301,20 @@ var CharacterCounter = Class.create({
 				}
 			};
 			
+			var setCurrentCharCount = function(newCharCount) {
+				if(charCounterUI.valueElement)
+					charCounterUI.valueElement.update(newCharCount);
+				if(charCounterUI.containerElement) {
+					if (newCharCount == 0) {
+						if (charCounterUI.containerElement.visible())
+							charCounterUI.containerElement.hide();
+					} else {
+						if (!charCounterUI.containerElement.visible())
+							charCounterUI.containerElement.show();
+					}
+				}
+			};
+			
 			return {
 				
 				init: function(controller,platform,textElement,params) {
@@ -404,7 +422,13 @@ var CharacterCounter = Class.create({
 								setTextFieldValueFn = params.segmentCounter.setTextFieldValueFn;
 							}						
 						}
-
+						if(params.charCounter) {
+							charCounterUI.containerElement = params.charCounter.charCountContainer;
+							charCounterUI.valueElement = params.charCounter.charCountElement;
+							if(params.charCounter.setTextFieldValueFn) {
+								setTextFieldValueFn = params.charCounter.setTextFieldValueFn;
+							}						
+						}
 					}					
 				},
 				
@@ -472,6 +496,7 @@ var CharacterCounter = Class.create({
 					
 					var messageData = parseMessage(rawCharacterData.message,rawCharacterData.count,true);
 					adjustedCharacterCount = messageData.adjustedCharacterCount;
+					setCurrentCharCount(rawCharacterData.count);
 					if (adjustedMaxLen > 0)
 						setOverLimit(messageData.isOverLimit);
 					setCurrentSegmentCount(messageData.segmentCount);				
diff --git a/usr/palm/applications/com.palm.app.messaging/app/views/chatview/chatview-scene.html b/usr/palm/applications/com.palm.app.messaging/app/views/chatview/chatview-scene.html
index d0d8745..fbc1f82 100644
--- a/usr/palm/applications/com.palm.app.messaging/app/views/chatview/chatview-scene.html
+++ b/usr/palm/applications/com.palm.app.messaging/app/views/chatview/chatview-scene.html
@@ -22,6 +22,10 @@
 			<div id="messageContainer" class='palm-row'>
 				<div class="palm-row-wrapper textfield-group focused">
 					<div class="title">
+						<div id="charCounterContainer" style="display: none;">
+							<div id="charCounter">
+							</div>
+						</div>
 						<div id="attachmentContainer" style="display: none;">
 							<div id="cancelAttachment">
 							</div>
diff --git a/usr/palm/applications/com.palm.app.messaging/app/views/compose/compose-scene.html b/usr/palm/applications/com.palm.app.messaging/app/views/compose/compose-scene.html
index 30daa88..aa9c440 100644
--- a/usr/palm/applications/com.palm.app.messaging/app/views/compose/compose-scene.html
+++ b/usr/palm/applications/com.palm.app.messaging/app/views/compose/compose-scene.html
@@ -6,6 +6,10 @@
 		<div id="messageContainer" class='palm-row'>
 			<div class="palm-row-wrapper textfield-group focused">
 				<div class="title">
+					<div id="charCounterContainer" style="display: none;">
+						<div id="charCounter">
+						</div>
+					</div>
 					<div id="attachmentContainer" style="display: none;">
 						<div id="cancelAttachment">
 						</div>
diff --git a/usr/palm/applications/com.palm.app.messaging/stylesheets/messaging.css b/usr/palm/applications/com.palm.app.messaging/stylesheets/messaging.css
index 70065e0..dcfb418 100644
--- a/usr/palm/applications/com.palm.app.messaging/stylesheets/messaging.css
+++ b/usr/palm/applications/com.palm.app.messaging/stylesheets/messaging.css
@@ -831,6 +831,23 @@ img.avatar-overlay {
 #messageContainer.palm-row .icon.right {
 	height: 61px;
 }
+#messageContainer #charCounterContainer {
+	line-height: 20px;
+	display:block;
+	height: 20px;
+	border-width: 0px 10px 0px 9px;
+	-webkit-border-image: url(../images/message-segment-badge.png) 0 10 0 9 stretch stretch;
+	position: absolute;
+	z-index: 3;
+	top: 2px;
+	left: 2px;
+}
+#messageContainer #charCounterContainer #charCounter {
+	font-size: 12px;
+	font-weight: bold;
+	color: #679BC2;
+	margin: 0px -4px 3px -3px;
+}
 #messageContainer #attachmentContainer {
 	position: relative;
 	margin-top:10px;
