diff --git a/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js b/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js
index 29204c4..6ec5710 100644
--- a/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js
+++ b/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js
@@ -2446,6 +2446,8 @@ var ChatviewAssistant = Class.create({
 			this.controller.stageController.pushScene("compose", composeParams);
 		} else if (value === 'forward-as-email-cmd') {
 			this.forwardAsEmail({text: text});
+		} else if (value === 'resend-cmd') {
+			this.resend(text);
 		} else if (value === 'copy-cmd') {
 			Mojo.Log.info("text to copy : " + text);
 			this.addTextToClipboard(text);
@@ -2582,6 +2584,7 @@ var ChatviewAssistant = Class.create({
 					popupItems = [{label: $L(deliveryReceiptMsg), command: '', disabled: true}];
 				}
 				popupItems = popupItems.concat([
+					{label: $L('Resend'), command: 'resend-cmd', disabled: this.chatIsLocked},
 					{label: $L('Forward'), command: 'forward-cmd'},
 					{label: $L('Forward Via Email'), command: 'forward-as-email-cmd'},
 					{label: $L('Copy Text'), command: 'copy-cmd'},
@@ -3516,6 +3519,11 @@ var ChatviewAssistant = Class.create({
 		}
 	},
 	
+	resend: function(text) {
+		this.setTextFieldValue(text);
+		this.considerForSend();
+	},
+	
 	stopTextFieldAutoFocus: function() {
 		this.disabledMessageTextElement = this.messageTextElement;
 		this.messageTextElement = undefined;
