diff --git a/usr/palm/applications/com.palm.app.messaging/app/compose/ComposeView.js b/usr/palm/applications/com.palm.app.messaging/app/compose/ComposeView.js
index 53667ba..e835a2b 100644
--- a/usr/palm/applications/com.palm.app.messaging/app/compose/ComposeView.js
+++ b/usr/palm/applications/com.palm.app.messaging/app/compose/ComposeView.js
@@ -202,7 +202,7 @@ enyo.kind({
 		}
 	},
 	considerForSend: function(){
-		var message = this.$.richText.getValue();
+		var message = this.$.richText.getHtml();
 		if (message.length > 0) {//todo: check max length?
 			var contacts = this.$.addressingPopup.getContacts();
 			enyo.log("ComposeView::considerForSend::contacts:",contacts);
@@ -425,7 +425,7 @@ enyo.kind({
 		// Pressing "enter" should send a message and clear the input
 		if (inEvent.keyCode === 13) {
 			inEvent.preventDefault();
-			messageText = this.$.richText.getValue();
+			messageText = this.$.richText.getHtml();
 
 			// Only send non-empty messages
 			if (messageText) {
@@ -440,7 +440,7 @@ enyo.kind({
 		}
 	},
 	saveDraft: function(){
-		var message = this.$.richText.getValue();
+		var message = this.$.richText.getHtml();
 		if (message.length > 0) {
 			var contacts = this.$.addressingPopup.getContacts();
 			
diff --git a/usr/palm/applications/com.palm.app.messaging/app/conversations/ConversationList.js b/usr/palm/applications/com.palm.app.messaging/app/conversations/ConversationList.js
index cb1f8c3..723bb2f 100644
--- a/usr/palm/applications/com.palm.app.messaging/app/conversations/ConversationList.js
+++ b/usr/palm/applications/com.palm.app.messaging/app/conversations/ConversationList.js
@@ -281,7 +281,7 @@ enyo.kind({
 		if (chatThreadId) {
 			if (!skipSaveDraft) {
 				//save draft
-				this.saveMessageToDraft(this.$.richText.getValue(), chatThreadId);			
+				this.saveMessageToDraft(this.$.richText.getHtml(), chatThreadId);			
 			}
 			
 			this.$.richText.setValue("");
@@ -601,7 +601,7 @@ enyo.kind({
 		if (inEvent.keyCode === 13) {
 			enyo.log(" ENYO PERF: TRANSITION START time: "+ Date.now());
 			inEvent.preventDefault();
-			messageText = this.$.richText.getValue();
+			messageText = this.$.richText.getHtml();
 
 			// Only send non-empty messages
 			if (messageText) {
@@ -814,7 +814,7 @@ enyo.kind({
 				deliveryReport: deliveryReport
 			},
 			to: [recipient],
-			messageText: this.$.richText.getValue(),
+			messageText: this.$.richText.getHtml(),
 			serviceName: selectedTransport.serviceName
 		};
 		
