diff --git a/media/cryptofs/apps/usr/palm/applications/com.palm.app.messaging/app/imstatus/AccountLoginState.js b/media/cryptofs/apps/usr/palm/applications/com.palm.app.messaging/app/imstatus/AccountLoginState.js
index 3b2960d..cc16c5e 100644
--- a/media/cryptofs/apps/usr/palm/applications/com.palm.app.messaging/app/imstatus/AccountLoginState.js
+++ b/media/cryptofs/apps/usr/palm/applications/com.palm.app.messaging/app/imstatus/AccountLoginState.js
@@ -41,7 +41,7 @@ enyo.kind({
 	loginStateChanged: function() {
 		var availability = enyo.messaging.imLoginState.getAvailability(this.loginState);
 		this.$.statuses.setAvailability([availability]);
-		this.$.statuses.setInvisibilitySupported(this.loginState.supportsInvisibleStatus);
+		// GFN this.$.statuses.setInvisibilitySupported(this.loginState.supportsInvisibleStatus);
 		if (availability <= enyo.messaging.im.availability.BUSY) {
 			this.$.customMessage.setContent(this.loginState.customMessage);
 		} else {
diff --git a/media/cryptofs/apps/usr/palm/applications/com.palm.app.messaging/app/imstatus/AccountLoginStatesPopup.js b/media/cryptofs/apps/usr/palm/applications/com.palm.app.messaging/app/imstatus/AccountLoginStatesPopup.js
index 75612a0..6187844 100644
--- a/media/cryptofs/apps/usr/palm/applications/com.palm.app.messaging/app/imstatus/AccountLoginStatesPopup.js
+++ b/media/cryptofs/apps/usr/palm/applications/com.palm.app.messaging/app/imstatus/AccountLoginStatesPopup.js
@@ -23,7 +23,7 @@ enyo.kind({
 	loginStatesChanged: function() {
 		this.$.allAccountsDivider.setShowing(this.loginStates.length > 1);
 		this.$.indAccountsDivider.setShowing(this.loginStates.length > 1);
-		this.checkInvisibilitySupport();
+		// GFN this.checkInvisibilitySupport();
 		this.updateAllAccountsAvailability();
 		this.updateLoginStateComponents();
 		this.$.db.setLoginStates(this.loginStates);
diff --git a/media/cryptofs/apps/usr/palm/applications/com.palm.app.messaging/app/imstatus/AccountStatuses.js b/media/cryptofs/apps/usr/palm/applications/com.palm.app.messaging/app/imstatus/AccountStatuses.js
index a362f6a..533e971 100644
--- a/media/cryptofs/apps/usr/palm/applications/com.palm.app.messaging/app/imstatus/AccountStatuses.js
+++ b/media/cryptofs/apps/usr/palm/applications/com.palm.app.messaging/app/imstatus/AccountStatuses.js
@@ -4,7 +4,7 @@ enyo.kind({
 	layoutKind: "VFlexLayout",
 	published: {
 		availability: [],
-		invisibilitySupported: false
+		invisibilitySupported: true
 	},
 	events: {
 		onAvailabilityChanged: "",
@@ -14,7 +14,7 @@ enyo.kind({
 		{name: "availabilities", className: "availabilities", components: [
 			{name: "available", kind: "Availability", caption: $L("Available"), value: enyo.messaging.im.availability.AVAILABLE, onclick:"updateAvailability", className: "enyo-first available-status"},
 			{name: "busy", kind: "Availability", caption: $L("Busy"), value: enyo.messaging.im.availability.BUSY, onclick:"updateAvailability"},
-			{name: "invisible", kind: "Availability", showing: false, caption: $L("Invisible"), value: enyo.messaging.im.availability.INVISIBLE, onclick:"updateAvailability"},
+			{name: "invisible", kind: "Availability", showing: true, caption: $L("Invisible - not GTalk"), value: enyo.messaging.im.availability.INVISIBLE, onclick:"updateAvailability"},
 			{name: "offline", kind: "Availability", caption: $L("Offline"), value: enyo.messaging.im.availability.OFFLINE, onclick:"updateAvailability"},
 			{name: "customStatus", kind: "MenuItem", caption: $L("Set custom status"), onclick:"setCustomMessage", className: "custom-status"}
 		]}
diff --git a/media/cryptofs/apps/usr/palm/applications/com.palm.app.messaging/app/imstatus/ImStatus.js b/media/cryptofs/apps/usr/palm/applications/com.palm.app.messaging/app/imstatus/ImStatus.js
index 13a8345..d7ca003 100644
--- a/media/cryptofs/apps/usr/palm/applications/com.palm.app.messaging/app/imstatus/ImStatus.js
+++ b/media/cryptofs/apps/usr/palm/applications/com.palm.app.messaging/app/imstatus/ImStatus.js
@@ -66,7 +66,7 @@ enyo.kind({
 		for (var i = 0; i < loginStates.length; i++) {
 			loginStates[i].accountTypeName = this.getAccountTypeName(loginStates[i]);
 			// The check for type_yahoo can be removed once the supportsInvisibleStatus flag is added to the Yahoo account template
-			loginStates[i].supportsInvisibleStatus = (this.supportsInvisibleStatus(loginStates[i]) || loginStates[i].serviceName === "type_yahoo");
+			// loginStates[i].supportsInvisibleStatus = (this.supportsInvisibleStatus(loginStates[i]) || loginStates[i].serviceName === "type_yahoo");
 		}		
 		loginStates.sort(function(a, b) {
 			return a.accountTypeName > b.accountTypeName || (a.accountTypeName === b.accountTypeName && a.username > b.username);
