Index: /usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js
===================================================================
--- .orig/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js
+++ /usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js
@@ -1541,59 +1541,104 @@ var ChatviewAssistant = Class.create({
 		MessagingMojoService.deleteMessage(this.controller, event.item.messageId);
 	},
 
-	handleMessageTap: function(event){
-		var eventTarget = this.controller.get(event.originalEvent.target);
-			
-		var mmsImageTarget = MessagingUtils.getClassUpChain(eventTarget,'MMSImageObject');
-		if(mmsImageTarget) {
-			this.controller.stageController.pushScene('imageview', mmsImageTarget.getAttribute('originalSrc'));
-			return;		
-		}
+        handleMessageTap: function(event){
+                var eventTarget = this.controller.get(event.originalEvent.target);
 
-		var mmsVideoTarget = MessagingUtils.getClassUpChain(eventTarget,'mms-video');
-		if(mmsVideoTarget) {
-			var videoPath = mmsVideoTarget.getAttribute('filePath');
-			var videoName = mmsVideoTarget.getAttribute('fileInfo');
-			
-			var args = {
-				appId: "com.palm.app.videoplayer",
-				name: "nowplaying"
-			};
-			var params = {
-				target: videoPath,
-				title: videoName
-			};
-			this.controller.stageController.pushScene(args, params); 			
-			return;		
-		}
-		
-		var mmsVcardTarget = MessagingUtils.getClassUpChain(eventTarget,'mms-vcard');
-		if(mmsVcardTarget) {
-			var filePath = mmsVcardTarget.getAttribute('filePath');
-			MessagingMojoService.launchFile(this.controller, 'text/x-vcard', 'file://'+filePath);
-			return;		
-		}
-		
-		var mmsVcalTarget = MessagingUtils.getClassUpChain(eventTarget,'mms-vcal');
-		if(mmsVcalTarget) {
-			var filePath = mmsVcalTarget.getAttribute('filePath');
-			MessagingMojoService.launchFile(this.controller, 'text/x-vcalendar', 'file://'+filePath);
-			return;		
-		}		
-				
-		MessagingUtils.simpleListClick(this.controller.get(event.originalEvent.target), "chatRow", function(targetRow){
-			var messageData = {
-				errorCode: targetRow.getAttribute("errorCode"),
-				status: targetRow.getAttribute("status"),
-				messageId: targetRow.getAttribute("messageId"),
-				flags: targetRow.getAttribute("flags"),
-				messageType: targetRow.getAttribute("messageType")
-			};
-			
-			MessagingMojoService.getMessageErrorInfo(this.controller, messageData.messageId, messageData.flags, this.handleMessageErrorPopup.bind(this,messageData));
+                var mmsImageTarget = MessagingUtils.getClassUpChain(eventTarget,'MMSImageObject');
+                if(mmsImageTarget) {
+                        var imagePath = mmsImageTarget.getAttribute('originalSrc');
+                        this.controller.showAlertDialog({
+                                onChoose: function(value) {if(value == "forward"){MessagingMojoService.forwardMessage(this.controller, event.item.messageText, imagePath);} else {this.controller.stageController.pushScene('imageview', imagePath);}},
+                                title: $L("Forward Message"),
+                                message: $L("Do you want to forward this message?"),
+                                choices:[
+                                        {label:$L("Yes"), value:"forward", type:"affirmative"},
+                                        {label:$L("No"), value:"", type:"negative"}
+                                ]
+                        });
+                        return;
+                }
 
-		}.bind(this), false);
-	},
+                var mmsVideoTarget = MessagingUtils.getClassUpChain(eventTarget,'mms-video');
+                if(mmsVideoTarget) {
+                        var videoPath = mmsVideoTarget.getAttribute('filePath');
+                        var videoName = mmsVideoTarget.getAttribute('fileInfo');
+
+                        var args = {
+                                appId: "com.palm.app.videoplayer",
+                                name: "nowplaying"
+                        };
+                        var params = {
+                                target: videoPath,
+                                title: videoName
+                        };
+                        this.controller.showAlertDialog({
+                                onChoose: function(value) {if(value == "forward"){MessagingMojoService.forwardMessage(this.controller, event.item.messageText, videoPath);} else {this.controller.stageController.pushScene(args, params);}},
+                                title: $L("Forward Message"),
+                                message: $L("Do you want to forward this message?"),
+                                choices:[
+                                        {label:$L("Yes"), value:"forward", type:"affirmative"},
+                                        {label:$L("No"), value:"", type:"negative"}
+                                ]
+                        });
+                        return;
+                }
+
+                var mmsVcardTarget = MessagingUtils.getClassUpChain(eventTarget,'mms-vcard');
+                if(mmsVcardTarget) {
+                        var filePath = mmsVcardTarget.getAttribute('filePath');
+                        this.controller.showAlertDialog({
+                                onChoose: function(value) {if(value == "forward"){MessagingMojoService.forwardMessage(this.controller, event.item.messageText, filePath);} else {this.controller.stageController.pushScene('mmsTextAttachment', filePath);}},
+                                title: $L("Forward Message"),
+                                message: $L("Do you want to forward this message?"),
+                                choices:[
+                                        {label:$L("Yes"), value:"forward", type:"affirmative"},
+                                        {label:$L("No"), value:"", type:"negative"}
+                                ]
+                        });
+                        return;
+                }
+
+                var mmsVcalTarget = MessagingUtils.getClassUpChain(eventTarget,'mms-vcal');
+                if(mmsVcalTarget) {
+                        var filePath = mmsVcalTarget.getAttribute('filePath');
+                        this.controller.showAlertDialog({
+                                onChoose: function(value) {if(value == "forward"){MessagingMojoService.forwardMessage(this.controller, event.item.messageText, filePath);} else {this.controller.stageController.pushScene('mmsTextAttachment', filePath);}},
+                                title: $L("Forward Message"),
+                                message: $L("Do you want to forward this message?"),
+                                choices:[
+                                        {label:$L("Yes"), value:"forward", type:"affirmative"},
+                                        {label:$L("No"), value:"", type:"negative"}
+                                ]
+                        });
+                        return;
+                }
+
+                if (!mmsImageTarget && !mmsVideoTarget && !mmsVcardTarget && !mmsVcalTarget) {
+                        this.controller.showAlertDialog({
+                                onChoose: function(value) {if(value == "forward"){MessagingMojoService.forwardMessage(this.controller, event.item.messageText, '');}},
+                                title: $L("Forward Message"),
+                                message: $L("Do you want to forward this message?"),
+                                choices:[
+                                        {label:$L("Yes"), value:"forward", type:"affirmative"},
+                                        {label:$L("No"), value:"", type:"negative"}
+                                ]
+                        });
+                }
+
+                MessagingUtils.simpleListClick(this.controller.get(event.originalEvent.target), "chatRow", function(targetRow){
+                        var messageData = {
+                                errorCode: targetRow.getAttribute("errorCode"),
+                                status: targetRow.getAttribute("status"),
+                                messageId: targetRow.getAttribute("messageId"),
+                                flags: targetRow.getAttribute("flags"),
+                                messageType: targetRow.getAttribute("messageType")
+                        };
+
+                        MessagingMojoService.getMessageErrorInfo(this.controller, messageData.messageId, messageData.flags, this.handleMessageErrorPopup.bind(this,messageData));
+
+                }.bind(this), false);
+        },
 	
 	
 	handleMessageErrorPopup: function(messageData,messageErrorData) {
Index: /usr/palm/applications/com.palm.app.messaging/app/models/messaging-luna-service.js
===================================================================
--- .orig/usr/palm/applications/com.palm.app.messaging/app/models/messaging-luna-service.js
+++ /usr/palm/applications/com.palm.app.messaging/app/models/messaging-luna-service.js
@@ -8,7 +8,25 @@ var MessagingMojoService = {
   displayIdentifier: 'palm://com.palm.display/control',
   telephonyIdentifier: 'palm://com.palm.telephony',
   systemManagerIdentifier: 'palm://com.palm.systemmanager',
+  forwardIdentifier: 'palm://com.palm.applicationManager',
 
+  forwardMessage: function(sceneController,messageText,attachment) {
+	var opts = {
+                method: 'launch',
+                parameters: {
+                        id: 'com.palm.app.messaging',
+                        params: {
+                        }
+                }
+        };
+        if (messageText)
+                opts.parameters.params.messageText = 'FWD: '+messageText;
+        if (attachment)
+                opts.parameters.params.attachment = attachment;
+        return sceneController.serviceRequest(MessagingMojoService.forwardIdentifier,opts);
+  },
+  
+  
   launchMessaging: function(sceneController,params) {  
 	return sceneController.serviceRequest("palm://com.palm.applicationManager", {
 		method:'launch',
