diff --git a/usr/palm/applications/com.palm.app.messaging/app/conversations/ConversationList.js b/usr/palm/applications/com.palm.app.messaging/app/conversations/ConversationList.js
index e712c30..dc61b0e 100644
--- a/usr/palm/applications/com.palm.app.messaging/app/conversations/ConversationList.js
+++ b/usr/palm/applications/com.palm.app.messaging/app/conversations/ConversationList.js
@@ -1049,7 +1049,7 @@ enyo.kind({
 		this.$.richText.setValue("");
 		
 		// play a sound for sending message
-		this.playSoundNotification({ isSent: true });
+		//this.playSoundNotification({ isSent: true });
 		
 //if app is throw away, and closeConversation() is not called, then following line should be called to clear unread count everytime user send a message
 //todo:if we have "pane" cleaup event, then following line should be called.
