diff --git a/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js b/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js
index 84e13e3..0dd1a4b 100644
--- a/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js
+++ b/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js
@@ -408,7 +408,8 @@ var ChatviewAssistant = Class.create({
 			response.record.displayName = MessagingUtils.formatPhoneNumber(response.record.displayName);
 		}
 		if (!this.chatDisplayName || this.chatDisplayName != response.record.displayName) {
-			this.chatViewRecipientName.update(response.record.displayName.escapeHTML());
+			this.lastTotalCount = response.record.totalCount;
+			this.chatViewRecipientName.update("(" + this.lastTotalCount + ")" + " " + response.record.displayName.escapeHTML());
 			this.chatDisplayName = response.record.displayName;
 		}
 		
