diff --git a/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js b/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js
index 484cd45..c3cad1c 100644
--- a/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js
+++ b/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js
@@ -2586,7 +2586,7 @@ var ChatviewAssistant = Class.create({
 		Mojo.Log.info("++++++ maxMessageAttachmentSize " + this.Messaging.carrierSettings.mmscMessageSizeLimit);
 		var params = {
 			actionType: 'attach',
-			kinds: ['image','video'], // TODO: integrate other media types
+			kinds: ['image','audio','video'], // TODO: integrate other media types
 			defaultKind: 'image',
 			maxVideoSize: this.Messaging.carrierSettings.mmscMessageSizeLimit,
 			tempVideoFilePath: '/media/internal/.attachments/mms/mmsVideo2',
diff --git a/usr/palm/applications/com.palm.app.messaging/app/controllers/compose-assistant.js b/usr/palm/applications/com.palm.app.messaging/app/controllers/compose-assistant.js
index 3183de4..3b1d8d7 100644
--- a/usr/palm/applications/com.palm.app.messaging/app/controllers/compose-assistant.js
+++ b/usr/palm/applications/com.palm.app.messaging/app/controllers/compose-assistant.js
@@ -350,7 +350,7 @@ var ComposeAssistant = Class.create({
     Mojo.Log.info("++++++ maxMessageAttachmentSize " + this.Messaging.carrierSettings.mmscMessageSizeLimit);
     var params = {
 		actionType: 'attach',
-        kinds: ['image','video'], // TODO: integrate other media types
+        kinds: ['image','audio','video'], // TODO: integrate other media types
         defaultKind: 'image',
 		maxVideoSize: this.Messaging.carrierSettings.mmscMessageSizeLimit,
 		tempVideoFilePath: '/media/internal/.attachments/mms/mmsVideo1',
