diff --git a/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js b/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js
index 98ab9b2..3362d61 100644
--- a/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js
+++ b/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js
@@ -2391,7 +2391,7 @@ var ChatviewAssistant = Class.create({
 	attachFilePicker: function(event) {
 		var params = {
 			actionType: 'attach',
-			kinds: ['image'], // TODO: integrate other media types
+			kinds: ['image','audio','video'], // TODO: integrate other media types
 			defaultKind: 'image',
 			onSelect: this.handleFilePickerSelect.bind(this)
 		};
diff --git a/usr/palm/applications/com.palm.app.messaging/app/controllers/compose-assistant.js b/usr/palm/applications/com.palm.app.messaging/app/controllers/compose-assistant.js
index 9715c78..19346cc 100644
--- a/usr/palm/applications/com.palm.app.messaging/app/controllers/compose-assistant.js
+++ b/usr/palm/applications/com.palm.app.messaging/app/controllers/compose-assistant.js
@@ -330,7 +330,7 @@ var ComposeAssistant = Class.create({
   attachFilePicker: function(event) {   
     var params = {
 		actionType: 'attach',
-        kinds: ['image'], // TODO: integrate other media types
+        kinds: ['image','audio','video'], // TODO: integrate other media types
         defaultKind: 'image',
         onSelect: this.handleFilePickerSelect.bind(this)
     };
