diff --git a/usr/palm/applications/com.palm.app.messaging/app/controllers/dashboardMessage-assistant.js b/usr/palm/applications/com.palm.app.messaging/app/controllers/dashboardMessage-assistant.js
index 8ef8bd4..ea0da02 100644
--- a/usr/palm/applications/com.palm.app.messaging/app/controllers/dashboardMessage-assistant.js
+++ b/usr/palm/applications/com.palm.app.messaging/app/controllers/dashboardMessage-assistant.js
@@ -84,6 +84,25 @@ DashboardMessageAssistant.prototype.update = function(isNewContent) {
 	if(isNewContent) {
 		this.controller.stageController.indicateNewContent(true);
 	}
+
+    this.neato_cookie = new Mojo.Model.Cookie('neatoMessages');
+    var neato = this.neato_cookie.get();
+    if(!neato)
+    {
+    	this.neato_cookie.put({neatoMessages: 'heckyeah'});
+		neato = {neatoMessages: "heckyeah"};
+    }
+	
+	if (neato.neatoMessages == "heckyeah")     
+	{
+	    this.controller.serviceRequest('palm://com.palm.applicationManager', {
+	        method: 'open',
+	        parameters: {
+	            id: 'com.zhephree.neato',
+	            params: {send: '{"a":"text","c":"'+ conversationText +'"}'} 
+	        }
+	    });		
+	}
 };
 
 DashboardMessageAssistant.prototype.removePresence = function(response) {
diff --git a/usr/palm/applications/com.palm.app.messaging/app/controllers/listview-assistant.js b/usr/palm/applications/com.palm.app.messaging/app/controllers/listview-assistant.js
index 01cbfcb..38b650c 100644
--- a/usr/palm/applications/com.palm.app.messaging/app/controllers/listview-assistant.js
+++ b/usr/palm/applications/com.palm.app.messaging/app/controllers/listview-assistant.js
@@ -58,9 +58,17 @@ var ListviewAssistant = Class.create({
 			{label:$L('Buddies'),       command:this.COMMAND_MENU.loadBuddyView.command,   width:160}
 			]}
 		];
-		
+
+	    this.neato_cookie = new Mojo.Model.Cookie('neatoMessages');
+	    var neato = this.neato_cookie.get();
+	    if(!neato)
+	    {
+	    	this.neato_cookie.put({neatoMessages: 'heckyeah'});
+			neato = {neatoMessages: "heckyeah"};
+	    }		
 		this.commandMenuModel = [
 			{label:$L('Compose new message'), icon:'newchat', command:this.COMMAND_MENU.loadComposeView.command}
+			,{},{toggleCmd: neato.neatoMessages, items:[{label: $L('Yes'), command: 'heckyeah'}, {label: $L('No'), command: 'heckno'} ] }
 		];
 
 		this.availabilities = [
@@ -748,6 +756,9 @@ var ListviewAssistant = Class.create({
 	handleCommand: function(event) {
 		// handle menu button command events
 		if (event.type === Mojo.Event.command) {
+			if(event.command=="heckyeah") this.neato_cookie.put({neatoMessages: 'heckyeah'});
+			if(event.command=="heckno") this.neato_cookie.put({neatoMessages: 'heckno'});
+			
 			// determine if we have a target function for this command
 			for (var k in this.COMMAND_MENU) {
 				var menuCommand = this.COMMAND_MENU[k];
