diff --git a/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js b/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js
index 549661e..ac46c8a 100644
--- a/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js
+++ b/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js
@@ -2466,6 +2466,8 @@ var ChatviewAssistant = Class.create({
 			this.controller.stageController.pushScene("compose", composeParams);
 		} else if (value === 'forward-as-email-cmd') {
 			this.forwardAsEmail({text: text});
+		} else if (value === 'resend-cmd') {
+			this.resend(text);
 		} else if (value === 'copy-cmd') {
 			Mojo.Log.info("text to copy : " + text);
 			this.addTextToClipboard(text);
@@ -2604,6 +2606,7 @@ var ChatviewAssistant = Class.create({
 				popupItems = popupItems.concat([
 					{label: $L('Forward'), command: 'forward-cmd'},
 					{label: $L('Forward Via Email'), command: 'forward-as-email-cmd'},
+					{label: $L('Resend'), command: 'resend-cmd', disabled: this.chatIsLocked},
 					{label: $L('Copy Text'), command: 'copy-cmd'},
 					{label: $L('Delete'), command: 'delete-cmd'}
 				]);
@@ -3554,6 +3557,10 @@ var ChatviewAssistant = Class.create({
 		}
 	},
 	
+	resend: function(text) {
+		this.setTextFieldValue(text);
+	},
+
 	stopTextFieldAutoFocus: function() {
 		this.disabledMessageTextElement = this.messageTextElement;
 		this.messageTextElement = undefined;
