diff --git a/etc/dnsmasq.palm.conf b/etc/dnsmasq.palm.conf
index 47b7908..4bebd01 100644
--- a/etc/dnsmasq.palm.conf
+++ b/etc/dnsmasq.palm.conf
@@ -286,5 +286,11 @@ interface=bridge0
 # dnsmasq.
 #log-queries
 
+#use the public google dns servers to resolve the domains gmail.com and googlemail.com.
+server=/gmail.com/8.8.8.8
+server=/gmail.com/8.8.4.4
+server=/googlemail.com/8.8.8.8
+server=/googlemail.com/8.8.4.4
+
 # Include a another lot of configuration options.
 conf-file=/tmp/pmnetconfig/dnsmasq.server.conf
