diff --git a/usr/palm/applications/com.palm.app.notes/app/controllers/note-assistant.js b/usr/palm/applications/com.palm.app.notes/app/controllers/note-assistant.js
index b685358..2f8d445 100644
--- a/usr/palm/applications/com.palm.app.notes/app/controllers/note-assistant.js
+++ b/usr/palm/applications/com.palm.app.notes/app/controllers/note-assistant.js
@@ -185,6 +185,22 @@ var NoteAssistant = Class.create(
 			}
 		}));
 	},
+        
+        $menu_text_memo_cmd: function(event)
+	{
+		(new Mojo.Service.Request('palm://com.palm.applicationManager',
+		{
+			method: 'open',
+			parameters:
+			{
+				id: 'com.palm.app.messaging',
+				params:
+				{
+					messageText: this._getTEXTNote()
+				}
+			}
+		}));
+	},
 	
 	$menu_delete_memo_cmd: function(event)
 	{
@@ -388,5 +404,11 @@ var NoteAssistant = Class.create(
 		// Convert multiple spaces into &nbsp;...
 		val = val.replace(/ {2}/g, " &nbsp;");
 		return val;
+	},
+        
+        _getTEXTNote: function()
+	{
+		var val = this.$widget_noteText.getValue();
+		return val;
 	}
 });
diff --git a/usr/palm/applications/com.palm.app.notes/app/views/note/note-scene.html b/usr/palm/applications/com.palm.app.notes/app/views/note/note-scene.html
index 38257c8..a438dec 100644
--- a/usr/palm/applications/com.palm.app.notes/app/views/note/note-scene.html
+++ b/usr/palm/applications/com.palm.app.notes/app/views/note/note-scene.html
@@ -7,6 +7,7 @@
 			<div x-mojo-action="palm-paste-cmd">Paste</div>
 		</div>
 		<div x-mojo-loc="" x-mojo-action="email-memo-cmd">Email memo...</div>
+                <div x-mojo-loc="" x-mojo-action="text-memo-cmd">Text memo...</div>
 		<div x-mojo-loc="" x-mojo-action="delete-memo-cmd">Delete</div>
 		<div x-mojo-loc="" x-mojo-action="palm-help-cmd">Help</div>
 	</div>
