diff --git a/media/cryptofs/apps/usr/palm/applications/com.palm.app.messaging/app/conversations/ConversationList.js b/media/cryptofs/apps/usr/palm/applications/com.palm.app.messaging/app/conversations/ConversationList.js
index afec0ca..a042615 100644
--- a/media/cryptofs/apps/usr/palm/applications/com.palm.app.messaging/app/conversations/ConversationList.js
+++ b/media/cryptofs/apps/usr/palm/applications/com.palm.app.messaging/app/conversations/ConversationList.js
@@ -678,9 +678,9 @@ enyo.kind({
 		
 		if (inMessage.flags && !inMessage.flags.read && enyo.application.messageDashboardManager.getAppDeactivated() === false) {
 			// play sound notification only when this is a new unread message in either inbox or transient folder
-			if (inMessage.folder !== enyo.messaging.message.FOLDERS.OUTBOX) {
+			/*if (inMessage.folder !== enyo.messaging.message.FOLDERS.OUTBOX) {
 				this.playSoundNotification({ isSent: false });
-			}
+			}*/
 		}				
 	},
 	updateListButtons: function(thread) {
