diff --git a/usr/palm/applications/com.palm.app.messaging/app/controllers/listview-assistant.js b/usr/palm/applications/com.palm.app.messaging/app/controllers/listview-assistant.js
index 01cbfcb..30fdd6a 100644
--- a/usr/palm/applications/com.palm.app.messaging/app/controllers/listview-assistant.js
+++ b/usr/palm/applications/com.palm.app.messaging/app/controllers/listview-assistant.js
@@ -1266,8 +1266,13 @@ Mojo.Log.error("------!!!!!!----should not comment out historyListQuery:");
 	// Runs on all history list item models when received from the service, before giving them to the List widget.
 	preformatHistoryItem: function(item) {
 		if (item.flags.outgoing) {
+			item.css_class = "my-history-chat";
 			item.messageActionImage = "images/outgoing.png";
+		} else if (item.unreadCount) {
+			item.css_class = "new-chat";
+			item.messageActionImage = "images/incoming.png";
 		} else {
+			item.css_class = "their-chat";
 			item.messageActionImage = "images/incoming.png";
 		}
 		
diff --git a/usr/palm/applications/com.palm.app.messaging/app/views/listview/historyList-row.html b/usr/palm/applications/com.palm.app.messaging/app/views/listview/historyList-row.html
index 02c3aad..93b1537 100644
--- a/usr/palm/applications/com.palm.app.messaging/app/views/listview/historyList-row.html
+++ b/usr/palm/applications/com.palm.app.messaging/app/views/listview/historyList-row.html
@@ -1,4 +1,4 @@
-<div id="#{id}" class="palm-row chat-entry historyRow removable #{classNames}">
+<div id="#{id}" class="palm-row chat-entry historyRow removable #{css_class} #{classNames}">
 	<div class="palm-row-wrapper">
 		<div class="chat-balloon" x-mojo-tap-highlight="momentary">
 			<div class="chat-balloon-wrapper">
diff --git a/usr/palm/applications/com.palm.app.messaging/stylesheets/messaging.css b/usr/palm/applications/com.palm.app.messaging/stylesheets/messaging.css
index 59607ca..dc9a884 100644
--- a/usr/palm/applications/com.palm.app.messaging/stylesheets/messaging.css
+++ b/usr/palm/applications/com.palm.app.messaging/stylesheets/messaging.css
@@ -618,6 +618,10 @@
 		color: #747474;
 }
 
+.chat-entry.new-chat .chat-balloon {
+		-webkit-border-image: url(/media/internal/.scrims/chat-balloon-left-new.png) 10 10 10 10 stretch stretch;
+}
+
 /* Chat Balloons - sending */
 
 .chat-entry.my-chat .chat-balloon {
@@ -637,6 +641,10 @@
 
 /* Chat Balloons - History List tapped */
 
+.chat-entry.my-history-chat .chat-balloon {
+		-webkit-border-image: url(../images/chatview/chat-balloon-right.png) 10 10 10 10 stretch stretch;
+}
+
 .palm-row.chat-entry.historyRow .chat-balloon.selected {
 		border-width: 10px;
 		-webkit-border-image: url(../images/chatview/chat-balloon-left-selected.png) 10 10 10 10 stretch stretch;
