diff --git a/usr/palm/applications/com.palm.app.messaging/app/controllers/notification-assistant.js b/usr/palm/applications/com.palm.app.messaging/app/controllers/notification-assistant.js
index 5a5aec0..ff2699f 100644
--- a/usr/palm/applications/com.palm.app.messaging/app/controllers/notification-assistant.js
+++ b/usr/palm/applications/com.palm.app.messaging/app/controllers/notification-assistant.js
@@ -404,7 +404,8 @@ NotificationAssistant.prototype.sendNewMessageNotificationAndUpdateDashboard = f
 	}else{
 		notificationText = $L("#{name}: #{message}").interpolate({
 			name: name,
-			message: message
+			//message: message
+			message: message.substring(0,20)
 		});
 	}
 	
