diff --git a/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js b/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js
index 71c2226..ef42484 100644
--- a/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js
+++ b/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js
@@ -3265,7 +3265,7 @@ var ChatviewAssistant = Class.create({
 	attachFilePicker: function(event) {
 		var params = {
 			actionType: 'attach',
-			kinds: ['image','video'], // TODO: integrate other media types
+			kinds: ['image','audio','video'], // TODO: integrate other media types
 			defaultKind: 'image',
 			convertBMP: true,
 			maxVideoSize: (this.Messaging.carrierSettings.mmscMessageSizeLimit-MessagingDB.MMS.headerSize)*1024,
diff --git a/usr/palm/applications/com.palm.app.messaging/app/controllers/compose-assistant.js b/usr/palm/applications/com.palm.app.messaging/app/controllers/compose-assistant.js
index f9eda28..77c3f9e 100644
--- a/usr/palm/applications/com.palm.app.messaging/app/controllers/compose-assistant.js
+++ b/usr/palm/applications/com.palm.app.messaging/app/controllers/compose-assistant.js
@@ -691,7 +691,7 @@ var ComposeAssistant = Class.create({
 	attachFilePicker: function(event) {   
 		var params = {
 			actionType: 'attach',
-			kinds: ['image','video'], // TODO: integrate other media types
+			kinds: ['image','audio','video'], // TODO: integrate other media types
 			defaultKind: 'image',
 			convertBMP: true,
 			maxVideoSize: (this.Messaging.carrierSettings.mmscMessageSizeLimit-MessagingDB.MMS.headerSize)*1024,
