diff --git a/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js b/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js
index aa6a2b7..332411a 100644
--- a/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js
+++ b/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js
@@ -326,8 +326,8 @@ var ChatviewAssistant = Class.create({
 				this.copyConversationToClipboard();
 			}
 		} else if (event.type == Mojo.Event.back) {
-			event.stop();
-			this.popAndRevealListView();
+			// event.stop();
+			// this.popAndRevealListView();
 			return;
 		}
 	},
diff --git a/usr/palm/applications/com.palm.app.messaging/app/controllers/listview-assistant.js b/usr/palm/applications/com.palm.app.messaging/app/controllers/listview-assistant.js
index 5058e58..fa6281d 100644
--- a/usr/palm/applications/com.palm.app.messaging/app/controllers/listview-assistant.js
+++ b/usr/palm/applications/com.palm.app.messaging/app/controllers/listview-assistant.js
@@ -1071,23 +1071,23 @@ var ListviewAssistant = Class.create(App.Scene, {
 		});
 	},
 	
-	launchChatView: function(chatThreadId, params) {
-		if (params === undefined) {
-			params = {};
+	launchChatView: function(chatThreadId) {
+		//Need to jump to that stage if it exists
+		var stageController = Mojo.Controller.appController.getStageController("messaging"+chatThreadId);
+		if (stageController) {
+			stageController.activate();
+			return;
 		}
-		params.clearListBadgeFn = this.clearListBadgeForChatThreadId.bind(this, chatThreadId);
-		this.controller.stageController.pushScene('chatview', chatThreadId, params);
-	},
 	
-	clearListBadgeForChatThreadId: function(chatThreadId) {
-		var badgeContainers = ["buddyBageContainer", "historyBageContainer"];
-		var listItem;
-		for (var i = 0; i < badgeContainers.length; i++) {
-			listItem = this.controller.get(badgeContainers[i] + chatThreadId);
-			if (listItem && !listItem.hasClassName('hide-unread-count')) {
-				listItem.addClassName('hide-unread-count');
-			}
-		}
+		var params = {name: 'messaging' + chatThreadId,	lightweight: Mojo.Controller.appInfo.lwStages};
+
+		var callback = function(controller) {
+			controller.pushScene('chatview',chatThreadId,{
+					focusWindow: true
+			});		
+		};
+
+		Mojo.Controller.getAppController().createStageWithCallback(params, callback); //doesnt create ifit exists sooo
 	},
 	
 	loadDebugView: function() {
