diff --git a/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js b/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js
index 599caf0..693e9be 100644
--- a/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js
+++ b/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js
@@ -1411,6 +1411,9 @@ var ChatviewAssistant = Class.create({
 		for (var i = 0; i < stuff.list.length; i++) {
 			var delayed = {};
 			var msg = stuff.list[i];
+			var ts = new Date();
+			ts.setTime(msg.timeStamp);
+			msg.timeStampText = Mojo.Format.formatDate(ts,'short');
 			if (msg.summary && !ChatFlags.isTransient(msg.flags)) {
 				msg.summary = msg.summary.escapeHTML();
 				//msg.summary = PalmSystem.runTextIndexer(msg.summary);
@@ -1457,7 +1460,7 @@ var ChatviewAssistant = Class.create({
 			if (prevAddress) {
 				this.dividerAddress = prevAddress;
 			}
-			msg.dividerText = msg.transportClass + BucketDateFormatter.getDateBucket(d, true, true) + this.dividerAddress;
+			msg.dividerText = msg.transportClass; // + BucketDateFormatter.getDateBucket(d, true, true) + this.dividerAddress;
 	
 			// if the INCOMING message was sent longer than 10 minutes before it arrived on device,
 			// add sent date to indicate that this delay happened
diff --git a/usr/palm/applications/com.palm.app.messaging/app/views/chatview/message/message-text.html b/usr/palm/applications/com.palm.app.messaging/app/views/chatview/message/message-text.html
index 97ff521..95b2ba6 100644
--- a/usr/palm/applications/com.palm.app.messaging/app/views/chatview/message/message-text.html
+++ b/usr/palm/applications/com.palm.app.messaging/app/views/chatview/message/message-text.html
@@ -1 +1 @@
-<div class="messageText">#{-messageText} #{-errorContent}</div>
\ No newline at end of file
+<div class="messageText"><span class="timeStamp">#{-timeStampText}</span><pre>#{-messageText}</pre> #{-errorContent}</div>
diff --git a/usr/palm/applications/com.palm.app.messaging/stylesheets/messaging.css b/usr/palm/applications/com.palm.app.messaging/stylesheets/messaging.css
index 6930649..2fe676a 100644
--- a/usr/palm/applications/com.palm.app.messaging/stylesheets/messaging.css
+++ b/usr/palm/applications/com.palm.app.messaging/stylesheets/messaging.css
@@ -653,6 +653,18 @@ img.avatar-overlay {
 	font-size:14px;
 }
 
+.messageText pre {
+	font-family: inherit;
+	font-size: inherit;
+	color: inherit;
+}
+.timeStamp {
+	font-size: 12px;
+	font-style: italic;
+	color: #1111ff;
+}
+
+
 .my-chat .chat-balloon-wrapper, 
 .their-chat .chat-balloon-wrapper {
   padding: 1px 0;
