diff --git a/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js b/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js
index 484cd45..cca84fe 100644
--- a/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js
+++ b/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js
@@ -126,6 +126,9 @@ var ChatviewAssistant = Class.create({
 	},
 	
 	setup: function() {
+		var cookie = new Mojo.Model.Cookie("myAvatarPath");
+		this.myAvatar = cookie.get() || "images/list-view-default-avatar.png";
+		
 		Mojo.Log.info("++++++ChatviewAssistant::setup");
 		this.controller.setupWidget(Mojo.Menu.appMenu, this.appMenuAttrs, this.appMenuModel);
 		
@@ -1540,7 +1543,7 @@ var ChatviewAssistant = Class.create({
 			if (ChatFlags.isOutgoing(msg.flags)) {
 				tmpClass.push("my-chat");
 				// Disabling MY avatar because we are not doing this for 1.0  2008-11-05
-				//msg.avatar = this.myAvatar || 'images/my-avatar.png';
+				msg.avatar = this.myAvatar;
 			} else {
 				tmpClass.push("their-chat");
 				msg.avatar = this.avatarHash[msg.fromAddress] || this.contactAvatar || "images/list-view-default-avatar.png";
diff --git a/usr/palm/applications/com.palm.app.messaging/app/controllers/listview-assistant.js b/usr/palm/applications/com.palm.app.messaging/app/controllers/listview-assistant.js
index c301f25..14f4d4a 100644
--- a/usr/palm/applications/com.palm.app.messaging/app/controllers/listview-assistant.js
+++ b/usr/palm/applications/com.palm.app.messaging/app/controllers/listview-assistant.js
@@ -179,6 +179,24 @@ var ListviewAssistant = Class.create(App.Scene, {
 	},
 	
 	setup: function() {
+		var cookie = new Mojo.Model.Cookie("myAvatarPath");
+		this.myAvatar = cookie.get();
+		if(this.myAvatar) {
+			this.controller.serviceRequest("palm://ca.canucksoftware.filemgr", {
+				method: "exists",
+				parameters: {file: this.myAvatar},
+				onSuccess: function(response) {
+					if(!response.exists) {
+						cookie.remove();
+					}
+				}.bind(this)
+			});
+		}
+		this.controller.serviceRequest("palm://ca.canucksoftware.filemgr", {
+			method: "createDir",
+			parameters: {path: "/var/luna/files/"}
+		});
+
 		this.appMenuAttrs = {
 			omitDefaultItems: true
 		};
@@ -402,9 +420,54 @@ var ListviewAssistant = Class.create(App.Scene, {
 		}
 	},
 	
+	deleteAvatar: function() {
+		if(this.myAvatar) {
+			this.controller.serviceRequest("palm://ca.canucksoftware.filemgr", {
+				method: "delete",
+				parameters: {
+					file: this.myAvatar,
+				}
+			});
+		}
+	},
+	
+	copyAvatar: function(filePath) {
+		this.timestamp = new Date().getTime();
+		this.avatarExt = filePath.substring(filePath.lastIndexOf("."));
+		this.controller.serviceRequest("palm://ca.canucksoftware.filemgr", {
+			method: "copy",
+			parameters: {
+				from: filePath,
+				to: "/var/luna/files/_my-avatar-" + this.timestamp + this.avatarExt,
+			},
+			onSuccess: function() {
+				this.myAvatar = "/var/luna/files/_my-avatar-" + this.timestamp
+						+ this.avatarExt;
+				var cookie = new Mojo.Model.Cookie("myAvatarPath");
+				cookie.put(this.myAvatar);
+			}.bind(this)
+		});
+	},
+	
 	handleCommand: function(event) {
 		// handle menu button command events
 		if (event.type == Mojo.Event.command) {
+			if (event.command === 'set-avatar') {
+				var self = this;
+				var params = {
+					extensions: ["bmp", "jpg", "jpeg", "png", "gif"],
+					kinds: ["image", "file"],
+					defaultKind: "image",
+					actionName: "Set Avatar",
+					onSelect: function(file){
+						self.deleteAvatar();
+						self.copyAvatar(file.fullPath);
+					}
+				};
+				Mojo.FilePicker.pickFile(params, this.controller.stageController);
+				event.stop();
+				return;
+			}
 			// determine if we have a target function for this command
 			var menu = $H(this.COMMAND_MENU);
 			menu.keys().each(function(key) {
@@ -439,6 +502,7 @@ var ListviewAssistant = Class.create(App.Scene, {
 			enabled: true
 		};
 		var items = [Mojo.Menu.editItem];
+		items.push({label: "Set Avatar", command: "set-avatar"});
 
 		if (this.isOfflineCollapsed != undefined && this.currentListView == this.Messaging.Views.BUDDY) {
 			var appMenuOfflineItem = {
diff --git a/usr/palm/applications/com.palm.app.messaging/stylesheets/messaging.css b/usr/palm/applications/com.palm.app.messaging/stylesheets/messaging.css
index 019deb5..23146f6 100644
--- a/usr/palm/applications/com.palm.app.messaging/stylesheets/messaging.css
+++ b/usr/palm/applications/com.palm.app.messaging/stylesheets/messaging.css
@@ -98,16 +98,11 @@ img.avatar-overlay {
 }
 
 .chatRow img.avatar-location {
-		margin-top: -18px;
+		margin-top: -39px;
 }
 
 .chatRow img.avatar-overlay {
-		margin-top: -21px;
-}
-
-.chatRow .avatar-container {
-		vertical-align: middle;
-		display: inline-block;
+		margin-top: -42.5px;
 }
 
 .login-animation-available,
@@ -727,10 +722,7 @@ img.avatar-overlay {
 		padding: 1px 0;
 		font-size: 18px;
 		word-wrap: break-word !important;
-}
-
-.their-chat .chat-balloon-wrapper {
-		padding-right: 8px;
+		padding-right: 40px;
 }
 
 .their-chat .avatar {
@@ -740,13 +732,6 @@ img.avatar-overlay {
 
 /* Hide user avatar */
 
-.my-chat .avatar-location,
-.my-chat .avatar-overlay,
-.my-chat .avatar-container {
-		display: none;
-		width: 40px;
-		height: 40px;
-}
 
 .message.status-success {
 		color: #000000;
