diff --git a/usr/palm/frameworks/mojo/builtins/palmInitFramework347.js b/usr/palm/frameworks/mojo/builtins/palmInitFramework347.js
index aed4b02..94019f7 100644
--- a/usr/palm/frameworks/mojo/builtins/palmInitFramework347.js
+++ b/usr/palm/frameworks/mojo/builtins/palmInitFramework347.js
@@ -11689,6 +11689,7 @@ altCharFullTable="alternatechars_fulltable.json";
 Mojo.Locale.alternateCharacters=Mojo.Locale.readStringTable(altCharTable,Mojo.Locale.current,Mojo.Locale.frameworkResourcePath,altCharArrayMerge);
 Mojo.Locale.alternateCharactersFull=Mojo.Locale.readStringTable(altCharFullTable,Mojo.Locale.current,Mojo.Locale.frameworkResourcePath,altCharArrayMerge);
 Mojo.Locale.DateTimeStrings=Mojo.Locale.readStringTable("datetime_table.json",Mojo.Locale.current,Mojo.Locale.frameworkResourcePath);
+Mojo.Locale.alternateCharactersHack=Mojo.Locale.readStringTable("alternatechars_table_hack.json","xxxxx","/usr/palm/frameworks/mojo/submissions/347/resources",altCharArrayMerge);
 
 
 
@@ -37181,6 +37182,7 @@ this.state=this.CHARSELECTOR_OPEN;
 this.charList=[];
 this.localizedTable=Mojo.Locale.alternateCharacters;
 this.localizedTableFull=Mojo.Locale.alternateCharactersFull;
+this.localizedTableHack=Mojo.Locale.alternateCharactersHack;
 },
 
 
@@ -37248,7 +37250,11 @@ var table;
 
 this.charList=[];
 if(chr){
+if(chr==75 || chr==90){
+table=this.localizedTableHack;
+}else{
 table=this.localizedTable;
+}
 }else if(chr===undefined){
 table=this.localizedTableFull;
 }else{
diff --git a/usr/palm/frameworks/mojo/submissions/347/resources/xxxxx/alternatechars_table_hack.json b/usr/palm/frameworks/mojo/submissions/347/resources/xxxxx/alternatechars_table_hack.json
new file mode 100644
index 0000000..0c71c3a
--- /dev/null
+++ b/usr/palm/frameworks/mojo/submissions/347/resources/xxxxx/alternatechars_table_hack.json
@@ -0,0 +1,4 @@
+[
+    {"keyCode": 75, "letter":"k", "list": [":doh",":mad",":wtf",":cool",":cry",":eek",":sad",":omg",":grin",":kiss",":lol",":evil",":meh",":sick",":smile",":wink",":yuck",":eww",":heart",":angel",":redface"]},
+    {"keyCode": 90, "letter":"z", "list": [":-)",":-(",";-)","o_O","8)",":'(",":[",":S",":!",":O","^_^","O:-)",":-*",":-D","=-@",":/",":P",":|",">:-)",">:-(","<3","ž","Ž","•"]}
+]
