diff --git a/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js b/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js
index 549661e..f874080 100644
--- a/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js
+++ b/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js
@@ -247,6 +247,7 @@ var ChatviewAssistant = Class.create({
 				Mojo.Menu.cutItem,
 				Mojo.Menu.copyItem,
 				{label: $L('Copy All'), command: 'copyall'},
+				Mojo.Menu.selectAllItem,
 				Mojo.Menu.pasteItem
 			]
 		};
