diff --git a/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js b/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js
index 71c2226..3656303 100644
--- a/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js
+++ b/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js
@@ -265,7 +265,7 @@ var ChatviewAssistant = Class.create({
 			hintText: $L('Enter message here...'),
 			multiline: true,
 			focus: false,
-			enterSubmits: true,
+			enterSubmits: false,
 			inputName: "messageTextElement",
 			className: " ",
 			changeOnKeyPress: true
@@ -2003,6 +2003,7 @@ var ChatviewAssistant = Class.create({
 				if (isOutgoingMsg) {
 					msg.messageText = msg.messageText.escapeHTML();
 				}
+				msg.messageText=msg.messageText.replace(/\n/g,"<br>");
 				//msg.messageText = PalmSystem.runTextIndexer(msg.messageText);
 				delayed.messageText = msg.messageText;
 			}
@@ -2870,13 +2871,13 @@ var ChatviewAssistant = Class.create({
 	},
 
 	handleTextAreaKeyUp: function(event) {
-		if (event && Mojo.Char.isEnterKey(event.keyCode)) {
-			this.considerForSend();
-			Event.stop(event);
+		//if (event && Mojo.Char.isEnterKey(event.keyCode)) {
+		//	this.considerForSend();
+		//	Event.stop(event);
 			
 			// Remove highlight that was added on keydown.
-			this.sendButton.removeClassName('selected');
-		}
+		//	this.sendButton.removeClassName('selected');
+		//}
 	},
 
 	handleTextAreaChange: function() {
diff --git a/usr/palm/applications/com.palm.app.messaging/app/controllers/compose-assistant.js b/usr/palm/applications/com.palm.app.messaging/app/controllers/compose-assistant.js
index f9eda28..cfaa315 100644
--- a/usr/palm/applications/com.palm.app.messaging/app/controllers/compose-assistant.js
+++ b/usr/palm/applications/com.palm.app.messaging/app/controllers/compose-assistant.js
@@ -164,7 +164,7 @@ var ComposeAssistant = Class.create({
 		    hintText: $L('Enter message here...'),
 		    multiline: true,
 		    focus: (this.prePopulatedRecipients.length > 0),
-		    enterSubmits: true,
+		    enterSubmits: false,
 			requiresEnterKey: true,
 		    modelProperty: "value",
 			changeOnKeyPress: true
@@ -662,10 +662,10 @@ var ComposeAssistant = Class.create({
 	},
 
 	handleTextAreaKeyUp: function(event) {
-		if (event && Mojo.Char.isEnterKey(event.keyCode)) {
-			this.considerForSend();
-			Event.stop(event);
-		}		
+		//if (event && Mojo.Char.isEnterKey(event.keyCode)) {
+		//	this.considerForSend();
+		//	Event.stop(event);
+		//}		
 	},
 
 	handleTextAreaChange: function() {
