diff --git a/usr/palm/applications/com.palm.app.messaging/app/controllers/dashboardMessage-assistant.js b/usr/palm/applications/com.palm.app.messaging/app/controllers/dashboardMessage-assistant.js
index 8ef8bd4..2e0c857 100644
--- a/usr/palm/applications/com.palm.app.messaging/app/controllers/dashboardMessage-assistant.js
+++ b/usr/palm/applications/com.palm.app.messaging/app/controllers/dashboardMessage-assistant.js
@@ -65,7 +65,7 @@ DashboardMessageAssistant.prototype.update = function(isNewContent) {
 		}
 	}
 
-	var conversationText = data.last().message;
+	var conversationText = '[New Message]';
 	dModel.topLine = data[0].name;
 	// TODO: this was added as a temporary patch to display number of messages from a given person.
 	if (data.length > 1) {
diff --git a/usr/palm/applications/com.palm.app.messaging/app/controllers/notification-assistant.js b/usr/palm/applications/com.palm.app.messaging/app/controllers/notification-assistant.js
index ffab634..201bd5a 100644
--- a/usr/palm/applications/com.palm.app.messaging/app/controllers/notification-assistant.js
+++ b/usr/palm/applications/com.palm.app.messaging/app/controllers/notification-assistant.js
@@ -404,7 +404,7 @@ NotificationAssistant.prototype.sendNewMessageNotificationAndUpdateDashboard = f
 	}else{
 		notificationText = $L("#{name}: #{message}").interpolate({
 			name: name,
-			message: message
+			message: '[New Message]'
 		});
 	}
 	
