diff --git a/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js b/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js
index 29204c4..c2da6e6 100644
--- a/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js
+++ b/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js
@@ -3353,7 +3353,7 @@ var ChatviewAssistant = Class.create({
 	attachFilePicker: function(event) {
 		var params = {
 			actionType: 'attach',
-			kinds: ['image','video'], // TODO: integrate other media types
+			kinds: ['image','audio','video'], // TODO: integrate other media types
 			defaultKind: 'image',
 			convertBMP: true,
 			maxVideoSize: (this.Messaging.carrierSettings.mmscMessageSizeLimit-MessagingDB.MMS.headerSize)*1024,
diff --git a/usr/palm/applications/com.palm.app.messaging/app/controllers/compose-assistant.js b/usr/palm/applications/com.palm.app.messaging/app/controllers/compose-assistant.js
index 57dfb0f..09b8156 100644
--- a/usr/palm/applications/com.palm.app.messaging/app/controllers/compose-assistant.js
+++ b/usr/palm/applications/com.palm.app.messaging/app/controllers/compose-assistant.js
@@ -697,7 +697,7 @@ var ComposeAssistant = Class.create({
 	attachFilePicker: function(event) {   
 		var params = {
 			actionType: 'attach',
-			kinds: ['image','video'], // TODO: integrate other media types
+			kinds: ['image','audio','video'], // TODO: integrate other media types
 			defaultKind: 'image',
 			convertBMP: true,
 			maxVideoSize: (this.Messaging.carrierSettings.mmscMessageSizeLimit-MessagingDB.MMS.headerSize)*1024,
