diff -ur /usr/palm/app-original/com.palm.app.messaging/app/controllers/chatview-assistant.js /usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js
--- /usr/palm/app-original/com.palm.app.messaging/app/controllers/chatview-assistant.js	2009-09-13 23:09:56.000000000 -0500
+++ /usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js	2009-09-28 15:29:40.000000000 -0500
@@ -187,6 +187,11 @@
 				segmentCountContainer: this.controller.get('segmentCounterContainer'),
 				segmentCountElement: this.controller.get('segmentCounter'),
 				setTextFieldValueFn: this.setTextFieldValue.bind(this)
+			},
+			charCounter: {
+				charCountContainer: this.controller.get('charCounterContainer'),
+				charCountElement: this.controller.get('charCounter'),
+				setTextFieldValueFn: this.setTextFieldValue.bind(this)
 			}
 		};
 
diff -ur /usr/palm/app-original/com.palm.app.messaging/app/controllers/compose-assistant.js /usr/palm/applications/com.palm.app.messaging/app/controllers/compose-assistant.js
--- /usr/palm/app-original/com.palm.app.messaging/app/controllers/compose-assistant.js	2009-09-13 23:09:56.000000000 -0500
+++ /usr/palm/applications/com.palm.app.messaging/app/controllers/compose-assistant.js	2009-09-28 15:30:36.000000000 -0500
@@ -140,6 +140,11 @@
 			segmentCountContainer: this.controller.get('segmentCounterContainer'),
 			segmentCountElement: this.controller.get('segmentCounter'),
 			setTextFieldValueFn: this.setTextFieldValue.bind(this)
+		},
+		charCounter: {
+			charCountContainer: this.controller.get('charCounterContainer'),
+			charCountElement: this.controller.get('charCounter'),
+			setTextFieldValueFn: this.setTextFieldValue.bind(this)
 		}
 	};
 
diff -ur /usr/palm/app-original/com.palm.app.messaging/app/utilities/CharacterCounter.js /usr/palm/applications/com.palm.app.messaging/app/utilities/CharacterCounter.js
--- /usr/palm/app-original/com.palm.app.messaging/app/utilities/CharacterCounter.js	2009-09-13 23:09:56.000000000 -0500
+++ /usr/palm/applications/com.palm.app.messaging/app/utilities/CharacterCounter.js	2009-09-28 15:34:44.000000000 -0500
@@ -58,6 +58,10 @@
 				containerElement: null,
 				valueElement: null
 			};
+			var charCounterUI = {
+				containerElement: null,
+				valueElement: null
+			};
 			
 						
 			// TODO: eventually we might want to implement some cleverness that counts
@@ -291,6 +295,20 @@
 				}
 			};
 			
+			var setCurrentCharCount = function(newCharCount) {
+				if(charCounterUI.valueElement)
+					charCounterUI.valueElement.update(newCharCount);
+				if(charCounterUI.containerElement) {
+					if (newCharCount == 0) {
+						if (charCounterUI.containerElement.visible())
+							charCounterUI.containerElement.hide();
+					} else {
+						if (!charCounterUI.containerElement.visible())
+							charCounterUI.containerElement.show();
+					}
+				}
+			};
+			
 			return {
 				
 				init: function(controller,platform,textElement,params) {
@@ -451,7 +469,13 @@
 								setTextFieldValueFn = params.segmentCounter.setTextFieldValueFn;
 							}						
 						}
-
+						if(params.charCounter) {
+							charCounterUI.containerElement = params.charCounter.charCountContainer;
+							charCounterUI.valueElement = params.charCounter.charCountElement;
+							if(params.charCounter.setTextFieldValueFn) {
+								setTextFieldValueFn = params.charCounter.setTextFieldValueFn;
+							}						
+						}
 					}					
 				},
 				
@@ -517,6 +541,7 @@
 					
 					var messageData = parseMessage(rawCharacterData.message,rawCharacterData.count,true);
 					adjustedCharacterCount = messageData.adjustedCharacterCount;
+					setCurrentCharCount(rawCharacterData.count);
 					if (adjustedMaxLen > 0)
 						setOverLimit(messageData.isOverLimit);
 					setCurrentSegmentCount(messageData.segmentCount);				
diff -ur /usr/palm/app-original/com.palm.app.messaging/app/views/chatview/chatview-scene.html /usr/palm/applications/com.palm.app.messaging/app/views/chatview/chatview-scene.html
--- /usr/palm/app-original/com.palm.app.messaging/app/views/chatview/chatview-scene.html	2009-09-13 23:09:57.000000000 -0500
+++ /usr/palm/applications/com.palm.app.messaging/app/views/chatview/chatview-scene.html	2009-09-28 15:35:16.000000000 -0500
@@ -22,6 +22,10 @@
 			<div id="messageContainer" class='palm-row'>
 				<div class="palm-row-wrapper textfield-group focused">
 					<div class="title">
+						<div id="charCounterContainer" style="display: none;">
+							<div id="charCounter">
+							</div>
+						</div>
 						<div id="attachmentContainer" style="display: none;">
 							<div id="cancelAttachment">
 							</div>
diff -ur /usr/palm/app-original/com.palm.app.messaging/app/views/compose/compose-scene.html /usr/palm/applications/com.palm.app.messaging/app/views/compose/compose-scene.html
--- /usr/palm/app-original/com.palm.app.messaging/app/views/compose/compose-scene.html	2009-09-13 23:09:57.000000000 -0500
+++ /usr/palm/applications/com.palm.app.messaging/app/views/compose/compose-scene.html	2009-09-28 15:36:15.000000000 -0500
@@ -6,6 +6,10 @@
 		<div id="messageContainer" class='palm-row'>
 			<div class="palm-row-wrapper textfield-group focused">
 				<div class="title">
+					<div id="charCounterContainer" style="display: none;">
+						<div id="charCounter">
+						</div>
+					</div>
 					<div id="attachmentContainer" style="display: none;">
 						<div id="cancelAttachment">
 						</div>
diff -ur /usr/palm/app-original/com.palm.app.messaging/stylesheets/messaging.css /usr/palm/applications/com.palm.app.messaging/stylesheets/messaging.css
--- /usr/palm/app-original/com.palm.app.messaging/stylesheets/messaging.css	2009-09-13 23:10:04.000000000 -0500
+++ /usr/palm/applications/com.palm.app.messaging/stylesheets/messaging.css	2009-09-28 15:37:25.000000000 -0500
@@ -837,6 +837,23 @@
 #messageContainer.palm-row .icon.right {
 	height: 61px;
 }
+#messageContainer #charCounterContainer {
+	line-height: 20px;
+	display:block;
+	height: 20px;
+	border-width: 0px 10px 0px 9px;
+	-webkit-border-image: url(../images/message-segment-badge.png) 0 10 0 9 stretch stretch;
+	position: absolute;
+	z-index: 3;
+	top: 2px;
+	left: 2px;
+}
+#messageContainer #charCounterContainer #charCounter {
+	font-size: 12px;
+	font-weight: bold;
+	color: #679BC2;
+	margin: 0px -4px 3px -3px;
+}
 #messageContainer #attachmentContainer {
 	position: relative;
 	margin-top:10px;
