diff --git a/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js b/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js
index 484cd45..d95f82c 100644
--- a/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js
+++ b/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js
@@ -62,11 +62,12 @@ var ChatviewAssistant = Class.create({
 		};
 		this.messageTemplatePath = 'chatview/message/';
 		
-		this.appMenuAddPictureItem = {
+		/* this.appMenuAddPictureItem = {
 			label: $L('Attach file...'),
 			command: 'addPicture',
 			disabled: false
-		};
+		}; */
+		this.appMenuAddPictureItem = {label: $L('Attach file...'), shortcut: 'p', command:'addPicture', enabled:true};
 
 		this.appMenuBlockUsernameItem = {
 			label: $L('Block this person'),
@@ -334,6 +335,8 @@ var ChatviewAssistant = Class.create({
 			} else if (event.command === 'copyall') {
 				this.copyConversationToClipboard();
 			}
+		} else if (event.type == Mojo.Event.forward) { 
+			this.considerForSend();
 		} else if (event.type === Mojo.Event.back) {
 			event.stop();
 			this.popAndRevealListView();
diff --git a/usr/palm/applications/com.palm.app.messaging/app/controllers/compose-assistant.js b/usr/palm/applications/com.palm.app.messaging/app/controllers/compose-assistant.js
index 3183de4..7bdbacf 100644
--- a/usr/palm/applications/com.palm.app.messaging/app/controllers/compose-assistant.js
+++ b/usr/palm/applications/com.palm.app.messaging/app/controllers/compose-assistant.js
@@ -15,7 +15,7 @@ var ComposeAssistant = Class.create({
 			this.toggleDebugMenu = listviewSceneAssistant.toggleDebugMenu.bind(listviewSceneAssistant,true);
 		}
 
-		this.appMenuAddPictureItem = {label: $L('Attach file...'), command:'addPicture', enabled:true};
+		this.appMenuAddPictureItem = {label: $L('Attach file...'), shortcut: 'p', command:'addPicture', enabled:true};
 
 		this.appMenuAttrs = {omitDefaultItems: true};	
 		this.appMenuModel = { visible:true, label: $L('Chat view menu'), items: [Mojo.Menu.editItem,this.appMenuAddPictureItem,Mojo.Menu.helpItem]};   
@@ -484,6 +484,8 @@ var ComposeAssistant = Class.create({
 				this.attachFilePicker();
 				event.stop();
 			} 
+		} else if (event.type == Mojo.Event.forward) {	
+	        this.considerForSend();				
 		} else if(event.type == Mojo.Event.back) {
 			if (this.launchedAsForward === undefined) {
 				event.stop();
diff --git a/usr/palm/applications/com.palm.app.messaging/app/views/chatview/chatview-scene.html b/usr/palm/applications/com.palm.app.messaging/app/views/chatview/chatview-scene.html
index d0d8745..7576a6e 100644
--- a/usr/palm/applications/com.palm.app.messaging/app/views/chatview/chatview-scene.html
+++ b/usr/palm/applications/com.palm.app.messaging/app/views/chatview/chatview-scene.html
@@ -27,10 +27,10 @@
 							</div>
 							<img id="attachmentImage" src=""/>
 						</div>
-						<div id="ActionButton" class="icon right empty">
+						<div id="ActionButton" class="icon right empty" style="display:none;">
 							<div id="AttachButton" class="stuck attachButton" style="display:none;" x-mojo-tap-highlight="momentary">
 							</div>
-							<div id="SendButton" class="stuck sendButton" x-mojo-tap-highlight="immediate">
+							<div id="SendButton" class="stuck sendButton" style="display:none;" x-mojo-tap-highlight="immediate">
 								<div id="segmentCounterContainer">
 									<div id="segmentCounter">
 									</div>
diff --git a/usr/palm/applications/com.palm.app.messaging/app/views/compose/compose-scene.html b/usr/palm/applications/com.palm.app.messaging/app/views/compose/compose-scene.html
index 30daa88..d1e2948 100644
--- a/usr/palm/applications/com.palm.app.messaging/app/views/compose/compose-scene.html
+++ b/usr/palm/applications/com.palm.app.messaging/app/views/compose/compose-scene.html
@@ -11,10 +11,10 @@
 						</div>
 						<img id="attachmentImage" src=""/>
 					</div>
-					<div id="ActionButton" class="icon right empty">
+					<div id="ActionButton" class="icon right empty" style="display:none;">
 						<div id="AttachButton" class="stuck attachButton" x-mojo-tap-highlight="momentary">
 						</div>
-						<div id="SendButton" class="stuck sendButton" x-mojo-tap-highlight="momentary">
+						<div id="SendButton" class="stuck sendButton" style="display:none;" x-mojo-tap-highlight="momentary">
 							<div id="segmentCounterContainer">
 								<div id="segmentCounter">
 								</div>
