diff --git a/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js b/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js
index 29204c4..5b36143 100644
--- a/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js
+++ b/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js
@@ -315,7 +315,7 @@ var ChatviewAssistant = Class.create({
 			hintText: $L('Enter message here...'),
 			multiline: true,
 			focus: false,
-			enterSubmits: true,
+			enterSubmits: false,
 			inputName: "messageTextElement",
 			className: " ",
 			changeOnKeyPress: true
@@ -2066,6 +2066,7 @@ var ChatviewAssistant = Class.create({
 				}
 				//msg.messageText = PalmSystem.runTextIndexer(msg.messageText);
 				delayed.messageText = msg.messageText;
+				msg.messageText = msg.messageText.replace(/\n/g,"<br>");
 			}
 			
 			var icons = this.Messaging.IMAccounts.getIcons(msg.serviceName);
@@ -2929,7 +2930,7 @@ var ChatviewAssistant = Class.create({
 		// It will be cleared on key up.
 		if (event) {
 			if (Mojo.Char.isEnterKey(event.keyCode)) {
-				this.sendButton.addClassName('selected');
+				//this.sendButton.addClassName('selected');
 			} else if (event.keyCode===129) {
 				this.ctrlKeyPressed = true;
 			} else if (event.keyCode===Mojo.Char.spaceBar && this.ctrlKeyPressed) {
@@ -2947,11 +2948,11 @@ var ChatviewAssistant = Class.create({
 	handleTextAreaKeyUp: function(event) {
 		if (event) {
 			if (Mojo.Char.isEnterKey(event.keyCode)) {
-				this.considerForSend();
-				Event.stop(event);
+				//this.considerForSend();
+				//Event.stop(event);
 				
 				// Remove highlight that was added on keydown.
-				this.sendButton.removeClassName('selected');
+				//this.sendButton.removeClassName('selected');
 			} else if (event.keyCode===129) {
 				this.ctrlKeyPressed = false;
 			}
diff --git a/usr/palm/applications/com.palm.app.messaging/app/controllers/compose-assistant.js b/usr/palm/applications/com.palm.app.messaging/app/controllers/compose-assistant.js
index 57dfb0f..484bebb 100644
--- a/usr/palm/applications/com.palm.app.messaging/app/controllers/compose-assistant.js
+++ b/usr/palm/applications/com.palm.app.messaging/app/controllers/compose-assistant.js
@@ -164,7 +164,7 @@ var ComposeAssistant = Class.create({
 		    hintText: $L('Enter message here...'),
 		    multiline: true,
 		    focus: (this.prePopulatedRecipients.length > 0),
-		    enterSubmits: true,
+		    enterSubmits: false,
 			requiresEnterKey: true,
 		    modelProperty: "value",
 			changeOnKeyPress: true
@@ -669,8 +669,8 @@ var ComposeAssistant = Class.create({
 
 	handleTextAreaKeyUp: function(event) {
 		if (event && Mojo.Char.isEnterKey(event.keyCode)) {
-			this.considerForSend();
-			Event.stop(event);
+			//this.considerForSend();
+			//Event.stop(event);
 		}		
 	},
 
