diff --git a/usr/palm/applications/com.palm.app.messaging/app/controllers/notification-assistant.js b/usr/palm/applications/com.palm.app.messaging/app/controllers/notification-assistant.js
index 8238c68..b36a0c9 100644
--- a/usr/palm/applications/com.palm.app.messaging/app/controllers/notification-assistant.js
+++ b/usr/palm/applications/com.palm.app.messaging/app/controllers/notification-assistant.js
@@ -272,7 +272,8 @@ NotificationAssistant.prototype.sendNewMessageNotificationAndUpdateDashboard = f
 	}else{
 		notificationText = $L("#{name}: #{message}").interpolate({
 			name: name,
-			message: message
+			//message: message
+			message: message.substring(0,10)
 		});
 	}
 	
