diff --git a/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js b/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js
index aa6a2b7..a3681e9 100644
--- a/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js
+++ b/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js
@@ -1367,16 +1367,16 @@ var ChatviewAssistant = Class.create({
 	
 			// if the INCOMING message was sent longer than 10 minutes before it arrived on device,
 			// add sent date to indicate that this delay happened
-			if (!ChatFlags.isOutgoing(msg.flags) && msg.deviceTimeStamp && msg.timeStamp && (msg.deviceTimeStamp - msg.timeStamp) > (1000 * 60 * 10)) {
+//			if (!ChatFlags.isOutgoing(msg.flags) && msg.deviceTimeStamp && msg.timeStamp && (msg.deviceTimeStamp - msg.timeStamp) > (1000 * 60 * 10)) {
 				d.setTime(msg.timeStamp);
 				var sentTimeStamp = d;
 				
 				// Only show the message sent time if it is less than two days old
 				// from when it arrived on device
-				if (today - msg.timeStamp <= 172800000) { //172800000 = (1000*60*60*24*2)
+//				if (today - msg.timeStamp <= 172800000) { //172800000 = (1000*60*60*24*2)
 					msg.timeStampDeltaMsg = new Template($L("Message Sent: #{dateTime}")).evaluate({dateTime:Mojo.Format.formatDate(sentTimeStamp,'short')});
-				}
-			}
+//				}
+//			}
 
 			if (msg.callbackNumber) {
 				msg.callbackNumber = PalmSystem.runTextIndexer(msg.callbackNumber);
