diff --git a/usr/palm/applications/com.palm.app.messaging/app/controllers/compose-assistant.js b/usr/palm/applications/com.palm.app.messaging/app/controllers/compose-assistant.js
index 8f9b17e..3d0e43d 100644
--- a/usr/palm/applications/com.palm.app.messaging/app/controllers/compose-assistant.js
+++ b/usr/palm/applications/com.palm.app.messaging/app/controllers/compose-assistant.js
@@ -84,7 +84,7 @@ var ComposeAssistant = Class.create({
 		this.handleCancelAttachment = this.resetTextBox.bind(this,false);		
 		this.handleRecipientsChanged = this.handleRecipientsChanged.bind(this);
 
-		this.monkeyPatchAddressingWidget();
+		/*this.monkeyPatchAddressingWidget();*/
 		this.saveDraft = true;
 	},
 
@@ -593,11 +593,11 @@ var ComposeAssistant = Class.create({
 				}.bind(this)
 			});
 		}
-	},
+	}/*,
 
 	/*******
 	 * Monkey patch the addressing widget if the carrier requires limiting recipients
-	 *******/
+	 *******
 	monkeyPatchAddressingWidget: function() {
 		if (this.Messaging.carrierSettings && this.Messaging.carrierSettings.smsMaxRecipients > 0) {
 			Mojo.Log.warn("carrierSettings.smsMaxRecipients=%i, monkey patching addRecipients", this.Messaging.carrierSettings.smsMaxRecipients);
@@ -648,5 +648,5 @@ var ComposeAssistant = Class.create({
 				this.totalRecips++;
 			}
 		}
-	}
+	}*/
 });
