diff --git a/usr/palm/applications/com.palm.app.messaging/app/controllers/dashboardMessage-assistant.js b/usr/palm/applications/com.palm.app.messaging/app/controllers/dashboardMessage-assistant.js
index fe431e5..d33cd4d 100644
--- a/usr/palm/applications/com.palm.app.messaging/app/controllers/dashboardMessage-assistant.js
+++ b/usr/palm/applications/com.palm.app.messaging/app/controllers/dashboardMessage-assistant.js
@@ -55,7 +55,7 @@ DashboardMessageAssistant.prototype.update = function(isNewContent) {
 		}
 	}
 
-	var conversationText = data.last().message;
+	var conversationText = '[New Message]';
 	dModel.topLine = data[0].name;
 	// TODO: this was added as a temporary patch to display number of messages from a given person.
 	if (data.length > 1) {
diff --git a/usr/palm/applications/com.palm.app.messaging/app/controllers/notification-assistant.js b/usr/palm/applications/com.palm.app.messaging/app/controllers/notification-assistant.js
index 8238c68..4bbfeef 100644
--- a/usr/palm/applications/com.palm.app.messaging/app/controllers/notification-assistant.js
+++ b/usr/palm/applications/com.palm.app.messaging/app/controllers/notification-assistant.js
@@ -272,7 +272,7 @@ NotificationAssistant.prototype.sendNewMessageNotificationAndUpdateDashboard = f
 	}else{
 		notificationText = $L("#{name}: #{message}").interpolate({
 			name: name,
-			message: message
+			message: '[New Message]'
 		});
 	}
 	
