diff --git a/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js b/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js
index 29204c4..3b6df6d 100644
--- a/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js
+++ b/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js
@@ -568,7 +568,12 @@ var ChatviewAssistant = Class.create({
 				segmentCountContainer: this.controller.get('segmentCounterContainer'),
 				segmentCountElement: this.controller.get('segmentCounter'),
 				setTextFieldValueFn: this.setTextFieldValue.bind(this)
-			}
+			},
+			charCounter: {
+				charCountContainer: this.controller.get('charCounterContainer'),
+				charCountElement: this.controller.get('charCounter'),
+				setTextFieldValueFn: this.setTextFieldValue.bind(this)
+ 			}
 		};
 
 		this.characterCounter = new CharacterCounter(this.controller, this.Messaging.platformType, this.Messaging.carrier, this.messageTextElement, charCounterParams);
diff --git a/usr/palm/applications/com.palm.app.messaging/app/controllers/compose-assistant.js b/usr/palm/applications/com.palm.app.messaging/app/controllers/compose-assistant.js
index 57dfb0f..35b9b71 100644
--- a/usr/palm/applications/com.palm.app.messaging/app/controllers/compose-assistant.js
+++ b/usr/palm/applications/com.palm.app.messaging/app/controllers/compose-assistant.js
@@ -217,7 +217,12 @@ var ComposeAssistant = Class.create({
 				segmentCountContainer: this.controller.get('segmentCounterContainer'),
 				segmentCountElement: this.controller.get('segmentCounter'),
 				setTextFieldValueFn: this.setTextFieldValue.bind(this)
-			}
+			},
+			charCounter: {
+				charCountContainer: this.controller.get('charCounterContainer'),
+				charCountElement: this.controller.get('charCounter'),
+				setTextFieldValueFn: this.setTextFieldValue.bind(this)
+ 			}
 		};
 
 		this.characterCounter = new CharacterCounter(this.controller,this.Messaging.platformType,this.Messaging.carrier,this.messageTextElement,charCounterParams);
diff --git a/usr/palm/applications/com.palm.app.messaging/app/utilities/CharacterCounter.js b/usr/palm/applications/com.palm.app.messaging/app/utilities/CharacterCounter.js
index 96decfe..f4c9ff9 100644
--- a/usr/palm/applications/com.palm.app.messaging/app/utilities/CharacterCounter.js
+++ b/usr/palm/applications/com.palm.app.messaging/app/utilities/CharacterCounter.js
@@ -54,7 +54,10 @@ var CharacterCounter = Class.create({
 				containerElement: null,
 				valueElement: null
 			};
-			
+			var charCounterUI = {
+				containerElement: null,
+				valueElement: null
+			};
 						
 			// TODO: eventually we might want to implement some cleverness that counts
 			// bytes so we can accurately limit text that includes UTF-8 characters.
@@ -414,6 +417,20 @@ var CharacterCounter = Class.create({
 				}
 			};
 			
+			var setCurrentCharCount = function(newCharCount) {
+				if(charCounterUI.valueElement)
+					charCounterUI.valueElement.update(newCharCount);
+				if(charCounterUI.containerElement) {
+					if (newCharCount == 0) {
+						if (charCounterUI.containerElement.visible())
+							charCounterUI.containerElement.hide();
+					} else {
+						if (!charCounterUI.containerElement.visible())
+							charCounterUI.containerElement.show();
+					}
+				}
+			};
+		
 			return {
 				
 				init: function(controller,platform,carrier,textElement,params) {
@@ -522,6 +539,13 @@ var CharacterCounter = Class.create({
 								setTextFieldValueFn = params.segmentCounter.setTextFieldValueFn;
 							}						
 						}
+						if(params.charCounter) {
+							charCounterUI.containerElement = params.charCounter.charCountContainer;
+							charCounterUI.valueElement = params.charCounter.charCountElement;
+							if(params.charCounter.setTextFieldValueFn) {
+								setTextFieldValueFn = params.charCounter.setTextFieldValueFn;
+							}						
+						}
 
 					}					
 				},
@@ -582,7 +606,7 @@ var CharacterCounter = Class.create({
 					this.resetTimer = undefined;
 
 					var rawCharacterData = getRawCharacterData();
-					
+					setCurrentCharCount(rawCharacterData.count);
 //					if (rawCharacterData.count > 69){
 //						recheckAllTextForNonGsm = true;
 //						Mojo.Log.info("text changed, looking for char size");
diff --git a/usr/palm/applications/com.palm.app.messaging/app/views/chatview/chatview-scene.html b/usr/palm/applications/com.palm.app.messaging/app/views/chatview/chatview-scene.html
index e1f7235..e293b25 100644
--- a/usr/palm/applications/com.palm.app.messaging/app/views/chatview/chatview-scene.html
+++ b/usr/palm/applications/com.palm.app.messaging/app/views/chatview/chatview-scene.html
@@ -25,6 +25,10 @@
 			<div id="messageContainer" class="palm-row view-#{availabilityClass}">
 				<div class="palm-row-wrapper textfield-group focused">
 					<div class="title">
+						<div id="charCounterContainer" style="display: none;">
+							<div id="charCounter">
+							</div>
+						</div>
 						<div id="attachmentContainer" style="display: none;">
 							<div id="cancelAttachment">
 							</div>
diff --git a/usr/palm/applications/com.palm.app.messaging/app/views/compose/compose-scene.html b/usr/palm/applications/com.palm.app.messaging/app/views/compose/compose-scene.html
index 79fca4a..8be374b 100644
--- a/usr/palm/applications/com.palm.app.messaging/app/views/compose/compose-scene.html
+++ b/usr/palm/applications/com.palm.app.messaging/app/views/compose/compose-scene.html
@@ -10,6 +10,10 @@
 		<div id="messageContainer" class='palm-row'>
 			<div class="palm-row-wrapper textfield-group focused">
 				<div class="title">
+					<div id="charCounterContainer" style="display: none;">
+						<div id="charCounter">
+						</div>
+					</div>
 					<div id="attachmentContainer" style="display: none;">
 						<div id="cancelAttachment">
 						</div>
diff --git a/usr/palm/applications/com.palm.app.messaging/resources/de/views/chatview/chatview-scene.html b/usr/palm/applications/com.palm.app.messaging/resources/de/views/chatview/chatview-scene.html
index 02f03c4..a747e36 100644
--- a/usr/palm/applications/com.palm.app.messaging/resources/de/views/chatview/chatview-scene.html
+++ b/usr/palm/applications/com.palm.app.messaging/resources/de/views/chatview/chatview-scene.html
@@ -25,6 +25,10 @@
 			<div id="messageContainer" class="palm-row view-#{availabilityClass}">
 				<div class="palm-row-wrapper textfield-group focused">
 					<div class="title">
+						<div id="charCounterContainer" style="display: none;">
+							<div id="charCounter">
+							</div>
+						</div>
 						<div id="attachmentContainer" style="display: none;">
 							<div id="cancelAttachment">
 							</div>
diff --git a/usr/palm/applications/com.palm.app.messaging/resources/es/views/chatview/chatview-scene.html b/usr/palm/applications/com.palm.app.messaging/resources/es/views/chatview/chatview-scene.html
index 3bff455..94a41f4 100644
--- a/usr/palm/applications/com.palm.app.messaging/resources/es/views/chatview/chatview-scene.html
+++ b/usr/palm/applications/com.palm.app.messaging/resources/es/views/chatview/chatview-scene.html
@@ -25,6 +25,10 @@
 			<div id="messageContainer" class="palm-row view-#{availabilityClass}">
 				<div class="palm-row-wrapper textfield-group focused">
 					<div class="title">
+						<div id="charCounterContainer" style="display: none;">
+							<div id="charCounter">
+							</div>
+						</div>
 						<div id="attachmentContainer" style="display: none;">
 							<div id="cancelAttachment">
 							</div>
diff --git a/usr/palm/applications/com.palm.app.messaging/resources/fr/views/chatview/chatview-scene.html b/usr/palm/applications/com.palm.app.messaging/resources/fr/views/chatview/chatview-scene.html
index b2b21d9..e5bfa48 100644
--- a/usr/palm/applications/com.palm.app.messaging/resources/fr/views/chatview/chatview-scene.html
+++ b/usr/palm/applications/com.palm.app.messaging/resources/fr/views/chatview/chatview-scene.html
@@ -25,6 +25,10 @@
 			<div id="messageContainer" class="palm-row view-#{availabilityClass}">
 				<div class="palm-row-wrapper textfield-group focused">
 					<div class="title">
+						<div id="charCounterContainer" style="display: none;">
+							<div id="charCounter">
+							</div>
+						</div>
 						<div id="attachmentContainer" style="display: none;">
 							<div id="cancelAttachment">
 							</div>
diff --git a/usr/palm/applications/com.palm.app.messaging/resources/it/views/chatview/chatview-scene.html b/usr/palm/applications/com.palm.app.messaging/resources/it/views/chatview/chatview-scene.html
index ca4d044..f958106 100644
--- a/usr/palm/applications/com.palm.app.messaging/resources/it/views/chatview/chatview-scene.html
+++ b/usr/palm/applications/com.palm.app.messaging/resources/it/views/chatview/chatview-scene.html
@@ -25,6 +25,10 @@
 			<div id="messageContainer" class="palm-row view-#{availabilityClass}">
 				<div class="palm-row-wrapper textfield-group focused">
 					<div class="title">
+						<div id="charCounterContainer" style="display: none;">
+							<div id="charCounter">
+							</div>
+						</div>
 						<div id="attachmentContainer" style="display: none;">
 							<div id="cancelAttachment">
 							</div>
diff --git a/usr/palm/applications/com.palm.app.messaging/stylesheets/messaging.css b/usr/palm/applications/com.palm.app.messaging/stylesheets/messaging.css
index 59607ca..d9bfda2 100644
--- a/usr/palm/applications/com.palm.app.messaging/stylesheets/messaging.css
+++ b/usr/palm/applications/com.palm.app.messaging/stylesheets/messaging.css
@@ -993,6 +993,25 @@
 		height: 61px;
 }
 
+#messageContainer #charCounterContainer {
+	line-height: 20px;
+	display:block;
+	height: 20px;
+	border-width: 0px 10px 0px 9px;
+	-webkit-border-image: url(../images/message-segment-badge.png) 0 10 0 9 stretch stretch;
+	position: absolute;
+	z-index: 3;
+	top: 2px;
+	left: 2px;
+}
+
+#messageContainer #charCounterContainer #charCounter {
+	font-size: 12px;
+	font-weight: bold;
+	color: #679BC2;
+	margin: 0px -4px 3px -3px;
+}
+
 #messageContainer #attachmentContainer {
 		position: relative;
 		margin-top: 10px;
