diff --git a/usr/palm/applications/com.palm.app.messaging/app/conversations/ConversationList.js b/usr/palm/applications/com.palm.app.messaging/app/conversations/ConversationList.js
index cb1f8c3..5ca834f 100644
--- a/usr/palm/applications/com.palm.app.messaging/app/conversations/ConversationList.js
+++ b/usr/palm/applications/com.palm.app.messaging/app/conversations/ConversationList.js
@@ -857,7 +857,7 @@ enyo.kind({
 		this.$.richText.setValue("");
 		
 		// play a sound for sending message
-		this.playSoundNotification({ isSent: true });
+		//this.playSoundNotification({ isSent: true });
 		
 //todo: keep it to remind something similiar
 //		MessagingUtils.checkAirplaneMode(params);
