Index: /usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js
===================================================================
--- .orig/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js
+++ /usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js
@@ -1837,7 +1837,7 @@ var ChatviewAssistant = Class.create({
 		// This will sometimes be an object because we pass this method directly when we observe
 		// Mojo.Event.tap on the send button
 		if(forceSendIfOffline && forceSendIfOffline != true)
-			forceSendIfOffline = false;		
+			forceSendIfOffline = true;		
 		
 		// prevent blank messages from going through
 		var message = this.messageTextElement.value.strip();
