diff --git a/usr/palm/applications/com.palm.app.messaging/app/controllers/compose-assistant.js b/usr/palm/applications/com.palm.app.messaging/app/controllers/compose-assistant.js
index 3183de4..f61eda4 100644
--- a/usr/palm/applications/com.palm.app.messaging/app/controllers/compose-assistant.js
+++ b/usr/palm/applications/com.palm.app.messaging/app/controllers/compose-assistant.js
@@ -97,7 +97,7 @@ var ComposeAssistant = Class.create({
 
 	this.searchList = {
 		"property": "to$A",
-		"includeEmails": false,
+		"includeEmails": true,
 		"includePhones": true,
 		"includeIMs": true,
 		"includeShortCode": true,
