diff --git a/usr/palm/applications/com.palm.app.messaging/app/controllers/listview-assistant.js b/usr/palm/applications/com.palm.app.messaging/app/controllers/listview-assistant.js
index c301f25..6abb40e 100644
--- a/usr/palm/applications/com.palm.app.messaging/app/controllers/listview-assistant.js
+++ b/usr/palm/applications/com.palm.app.messaging/app/controllers/listview-assistant.js
@@ -140,6 +140,8 @@ var ListviewAssistant = Class.create(App.Scene, {
 		this.filter = this.filter.bind(this);
 		this.handleHistoryListTap = this.handleHistoryListTap.bindAsEventListener(this);
 		this.handleHistoryListDelete = this.handleHistoryListDelete.bindAsEventListener(this);
+        this.handleHistoryListHold = this.handleHistoryListHold.bindAsEventListener(this);
+        this.handleBuddyListHold = this.handleBuddyListHold.bindAsEventListener(this);		
 		this.handleBuddyListTap = this.handleBuddyListTap.bindAsEventListener(this);
 		this.handleAvailabilityPickerTap = this.handleAvailabilityPickerTap.bindAsEventListener(this);
 		this.customMessageUpdate = this.customMessageUpdate.bind(this);
@@ -270,6 +272,8 @@ var ListviewAssistant = Class.create(App.Scene, {
 
 		// observe clicks on the buddy list & history list
 		this.historyListWidget.observe(Mojo.Event.listTap, this.handleHistoryListTap);
+	    this.historyListWidget.observe(Mojo.Event.hold, this.handleHistoryListHold);
+    	this.BuddyListWidget.observe(Mojo.Event.hold, this.handleBuddyListHold);		
 		// observe swipe to delete on the history list
 		this.historyListWidget.observe(Mojo.Event.listDelete, this.handleHistoryListDelete);
 		this.buddyListWidget.observe(Mojo.Event.listTap, this.handleBuddyListTap);
@@ -300,6 +304,12 @@ var ListviewAssistant = Class.create(App.Scene, {
 		this.clearSpecificListBadgeForChatThreadId(chatThreadId, "buddyBageContainer", this.buddyListWidget);
 	},
 	
+	handleHistoryListHold: function(e) {
+	    MessagingUtils.simpleListClick(this.controller.get(e.target),"historyRow",this.launchChatInNewCard.bind(this),true);
+	},
+	handleBuddyListHold: function(e) {
+	    MessagingUtils.simpleListClick(this.controller.get(e.target),"buddyList",this.launchChatInNewCard.bind(this),true);
+	},	
 	handleBuddyListTap: function(e) {
 		this.launchChatView(e.item.chatId, { selectIMTransport: true });
 	},
@@ -1311,6 +1321,26 @@ var ListviewAssistant = Class.create(App.Scene, {
 			popScenesTo: CONSTANTS.SCENE_ID_ACCOUNT_SUMMARY
 		});
 	},
+
+  launchChatInNewCard: function(chatThreadId) {
+	var stageController = Mojo.Controller.appController.getStageController("messaging"+chatThreadId);
+		if (stageController) {
+			stageController.activate();
+			return;
+		}
+
+	var params = {name: 'messaging' + chatThreadId,
+			lightweight: Mojo.Controller.appInfo.lwStages
+	};
+
+	var callback = function(controller) {
+		controller.pushScene('chatview',chatThreadId,{
+					focusWindow: true
+				});		
+	};
+
+	Mojo.Controller.getAppController().createStageWithCallback(params, callback);
+  },
 	
 	launchChatView: function(chatThreadId, params) {
 		if (params === undefined) {
@@ -1352,6 +1382,8 @@ var ListviewAssistant = Class.create(App.Scene, {
 		this.filterField.stopObserving(Mojo.Event.filter, this.filter);
 		this.historyListWidget.stopObserving(Mojo.Event.listTap, this.handleHistoryListTap);
 		this.historyListWidget.stopObserving(Mojo.Event.listDelete, this.handleHistoryListDelete);
+    	this.historyListWidget.stopObserving(Mojo.Event.hold, this.handleHistoryListHold);	
+	 	this.buddyListWidget.stopObserving(Mojo.Event.hold, this.handleBuddyListHold);
 		this.buddyListWidget.stopObserving(Mojo.Event.listTap, this.handleBuddyListTap);
 		this.customMessageField.stopObserving('blur', this.customMessageUpdate);
 		this.customMessageField.stopObserving('keyup', this.handleCustomMessageFieldKeyUp);
