diff --git a/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js b/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js
index aa6a2b7..792b7ce 100644
--- a/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js
+++ b/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js
@@ -128,6 +128,9 @@ var ChatviewAssistant = Class.create({
 		Mojo.Log.info("++++++ChatviewAssistant::setup");
 		this.controller.setupWidget(Mojo.Menu.appMenu, this.appMenuAttrs, this.appMenuModel);
 		
+		// enable free orientation
+		this.controller.window.PalmSystem.setWindowOrientation("free");
+		
 		var messageTextWidgetAttributes = {
 			modelProperty: 'value',
 			hintText: $L('Enter message here...'),
@@ -232,6 +235,14 @@ var ChatviewAssistant = Class.create({
 		}
 	},
 
+	orientationChanged: function(orientation) {
+		if (orientation === "left" || orientation === "right") {
+			this.controller.sceneElement.addClassName('landscape');
+		} else {
+			this.controller.sceneElement.removeClassName('landscape');
+		}
+	},
+
 	cleanup: function() {
 		this.chatViewHeaderClickTarget.stopObserving(Mojo.Event.tap, this.handleHeaderTap);
 		this.chatListWidget.stopObserving(Mojo.Event.listTap, this.handleMessageTap);
diff --git a/usr/palm/applications/com.palm.app.messaging/app/controllers/compose-assistant.js b/usr/palm/applications/com.palm.app.messaging/app/controllers/compose-assistant.js
index 9715c78..83a4521 100644
--- a/usr/palm/applications/com.palm.app.messaging/app/controllers/compose-assistant.js
+++ b/usr/palm/applications/com.palm.app.messaging/app/controllers/compose-assistant.js
@@ -81,6 +81,9 @@ var ComposeAssistant = Class.create({
   setup: function() {
     this.controller.setupWidget(Mojo.Menu.appMenu, this.appMenuAttrs, this.appMenuModel);
 
+	// enable free orientation
+	this.controller.window.PalmSystem.setWindowOrientation("free");
+
 	this.searchList = {
 		"property": "to$A",
 		"includeEmails": false,
@@ -400,6 +403,14 @@ var ComposeAssistant = Class.create({
 	this.setCharacterCounterMaxLength();
   },
   
+	orientationChanged: function(orientation) {
+		if (orientation === "left" || orientation === "right") {
+			this.controller.sceneElement.addClassName('landscape');
+		} else {
+			this.controller.sceneElement.removeClassName('landscape');
+		}
+	},
+
   cleanup: function() {
     var that = this;
 	
diff --git a/usr/palm/applications/com.palm.app.messaging/app/controllers/listview-assistant.js b/usr/palm/applications/com.palm.app.messaging/app/controllers/listview-assistant.js
index 5058e58..b7a22cf 100644
--- a/usr/palm/applications/com.palm.app.messaging/app/controllers/listview-assistant.js
+++ b/usr/palm/applications/com.palm.app.messaging/app/controllers/listview-assistant.js
@@ -214,7 +214,9 @@ var ListviewAssistant = Class.create(App.Scene, {
 			items: this.commandMenuModel
 		};
 	
-		this.controller.setupWidget(Mojo.Menu.commandMenu, {}, this.cmdMenuModel);
+		this.controller.setupWidget(Mojo.Menu.commandMenu, undefined, this.cmdMenuModel);
+		// enable free orientation
+		this.controller.window.PalmSystem.setWindowOrientation("free");
 	
 		this.filterField = this.controller.get('filterField');
 		this.buddyListHeader = this.controller.get('buddyListHeader');
@@ -1094,6 +1096,14 @@ var ListviewAssistant = Class.create(App.Scene, {
 		this.controller.stageController.pushScene('debug');
 	},
 	
+	orientationChanged: function(orientation) {
+		if (orientation === "left" || orientation === "right") {
+			this.controller.sceneElement.addClassName('landscape');
+		} else {
+			this.controller.sceneElement.removeClassName('landscape');
+		}
+	},
+	
 	cleanup: function() {
 		// save the listview state
 		MessagingMojoService.setIsHistoryViewSelected(this.controller, (this.currentListView == this.Messaging.Views.HISTORY));
diff --git a/usr/palm/applications/com.palm.app.messaging/stylesheets/messaging.css b/usr/palm/applications/com.palm.app.messaging/stylesheets/messaging.css
index 73af153..ea73f5e 100644
--- a/usr/palm/applications/com.palm.app.messaging/stylesheets/messaging.css
+++ b/usr/palm/applications/com.palm.app.messaging/stylesheets/messaging.css
@@ -605,10 +605,10 @@ img.avatar-overlay {
 }
 
 .their-chat .message-container {
-	width: 255px;
+	width: 85%;
 }
 .my-chat .message-container {
-	width: 280px;
+	width: 95%;
 }
 .chat-entry .chat-balloon {
 	min-height: 26px;
@@ -839,7 +839,7 @@ img.avatar-overlay {
 	position: fixed;
 	bottom:0px;
 	left:0px;
-	width: 320px;
+	width: 100%;
 }
 
 /* Force addressing widget to sit on top of
