diff --git a/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js b/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js
index 599caf0..112e7a2 100644
--- a/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js
+++ b/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js
@@ -63,7 +63,7 @@ var ChatviewAssistant = Class.create({
 		this.messageTemplatePath = 'chatview/message/';
 		
 		this.appMenuAddPictureItem = {
-			label: $L('Add picture...'),
+			label: $L('Attach...'),
 			command: 'addPicture',
 			disabled: false
 		};
@@ -2561,7 +2561,7 @@ var ChatviewAssistant = Class.create({
 		Mojo.Log.info("++++++ maxMessageAttachmentSize " + this.Messaging.carrierSettings.mmscMessageSizeLimit);
 		var params = {
 			actionType: 'attach',
-			kinds: ['image','video'], // TODO: integrate other media types
+			kinds: ['image','audio','video'], // TODO: integrate other media types
 			defaultKind: 'image',
 			maxVideoSize: this.Messaging.carrierSettings.mmscMessageSizeLimit,
 			tempVideoFilePath: '/media/internal/.attachments/mms/mmsVideo2',
diff --git a/usr/palm/applications/com.palm.app.messaging/app/controllers/compose-assistant.js b/usr/palm/applications/com.palm.app.messaging/app/controllers/compose-assistant.js
index ec90d3e..67bcfce 100644
--- a/usr/palm/applications/com.palm.app.messaging/app/controllers/compose-assistant.js
+++ b/usr/palm/applications/com.palm.app.messaging/app/controllers/compose-assistant.js
@@ -15,7 +15,7 @@ var ComposeAssistant = Class.create({
 			this.toggleDebugMenu = listviewSceneAssistant.toggleDebugMenu.bind(listviewSceneAssistant,true);
 		}
 
-		this.appMenuAddPictureItem = {label: $L('Add picture...'), command:'addPicture', enabled:true};
+		this.appMenuAddPictureItem = {label: $L('Attach...'), command:'addPicture', enabled:true};
 
 		this.appMenuAttrs = {omitDefaultItems: true};	
 		this.appMenuModel = { visible:true, label: $L('Chat view menu'), items: [Mojo.Menu.editItem,this.appMenuAddPictureItem,Mojo.Menu.helpItem]};   
@@ -344,7 +344,7 @@ var ComposeAssistant = Class.create({
     Mojo.Log.info("++++++ maxMessageAttachmentSize " + this.Messaging.carrierSettings.mmscMessageSizeLimit);
     var params = {
 		actionType: 'attach',
-        kinds: ['image','video'], // TODO: integrate other media types
+        kinds: ['image','audio','video'], // TODO: integrate other media types
         defaultKind: 'image',
 		maxVideoSize: this.Messaging.carrierSettings.mmscMessageSizeLimit,
 		tempVideoFilePath: '/media/internal/.attachments/mms/mmsVideo1',
