diff --git a/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js b/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js
index 84e13e3..c9a40ff 100644
--- a/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js
+++ b/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js
@@ -1786,7 +1786,7 @@ var ChatviewAssistant = Class.create({
 		
 		if (value === 'forward-cmd') {
 			var composeParams = {
-				messageText: text,
+				messageText: "Fwd: " + text,
 				focusWindow: true,
 				launchedAsForward: true
 			};
