diff --git a/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js b/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js
index 484cd45..db6cc9c 100644
--- a/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js
+++ b/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js
@@ -129,6 +129,9 @@ var ChatviewAssistant = Class.create({
 		Mojo.Log.info("++++++ChatviewAssistant::setup");
 		this.controller.setupWidget(Mojo.Menu.appMenu, this.appMenuAttrs, this.appMenuModel);
 		
+		// enable free orientation
+		this.controller.window.PalmSystem.setWindowOrientation("free");
+		
 		var messageTextWidgetAttributes = {
 			modelProperty: 'value',
 			hintText: $L('Enter message here...'),
@@ -229,6 +232,14 @@ var ChatviewAssistant = Class.create({
 		this.handleClearNotifications();
 	},
 	
+	orientationChanged: function(orientation) {
+		if (orientation === "left" || orientation === "right") {
+			this.controller.sceneElement.addClassName('landscape');
+		} else {
+			this.controller.sceneElement.removeClassName('landscape');
+		}
+	},
+
 	handleClearNotifications: function() {
 		if (this.isWindowFocused && this.isSceneActive && this.isScreenOn && !this.isScreenLocked) {
 			Mojo.Log.info("[CV] ****** handleClearNotifications ---- calling handleStageFocus()");
diff --git a/usr/palm/applications/com.palm.app.messaging/app/controllers/compose-assistant.js b/usr/palm/applications/com.palm.app.messaging/app/controllers/compose-assistant.js
index 3183de4..a7e81be 100644
--- a/usr/palm/applications/com.palm.app.messaging/app/controllers/compose-assistant.js
+++ b/usr/palm/applications/com.palm.app.messaging/app/controllers/compose-assistant.js
@@ -95,6 +95,9 @@ var ComposeAssistant = Class.create({
 
     this.controller.setupWidget(Mojo.Menu.appMenu, this.appMenuAttrs, this.appMenuModel);
 
+	// enable free orientation
+	this.controller.window.PalmSystem.setWindowOrientation("free");
+
 	this.searchList = {
 		"property": "to$A",
 		"includeEmails": false,
@@ -449,6 +452,14 @@ var ComposeAssistant = Class.create({
 	this.setCharacterCounterMaxLength();
   },
   
+	orientationChanged: function(orientation) {
+		if (orientation === "left" || orientation === "right") {
+			this.controller.sceneElement.addClassName('landscape');
+		} else {
+			this.controller.sceneElement.removeClassName('landscape');
+		}
+	},
+
   cleanup: function() {
 	this.saveToDraft();
   	
diff --git a/usr/palm/applications/com.palm.app.messaging/app/controllers/listview-assistant.js b/usr/palm/applications/com.palm.app.messaging/app/controllers/listview-assistant.js
index c301f25..d8a28dc 100644
--- a/usr/palm/applications/com.palm.app.messaging/app/controllers/listview-assistant.js
+++ b/usr/palm/applications/com.palm.app.messaging/app/controllers/listview-assistant.js
@@ -203,7 +203,9 @@ var ListviewAssistant = Class.create(App.Scene, {
 			items: this.commandMenuModel
 		};
 	
-		this.controller.setupWidget(Mojo.Menu.commandMenu, {}, this.cmdMenuModel);
+		this.controller.setupWidget(Mojo.Menu.commandMenu, undefined, this.cmdMenuModel);
+		// enable free orientation
+		this.controller.window.PalmSystem.setWindowOrientation("free");
 	
 		this.filterField = this.controller.get('filterField');
 		this.buddyListHeader = this.controller.get('buddyListHeader');
@@ -1344,6 +1346,14 @@ var ListviewAssistant = Class.create(App.Scene, {
 		this.controller.stageController.pushScene('debug');
 	},
 	
+	orientationChanged: function(orientation) {
+		if (orientation === "left" || orientation === "right") {
+			this.controller.sceneElement.addClassName('landscape');
+		} else {
+			this.controller.sceneElement.removeClassName('landscape');
+		}
+	},
+	
 	cleanup: function() {
 		// save the listview state
 		MessagingMojoService.setIsHistoryViewSelected(this.controller, (this.currentListView == this.Messaging.Views.HISTORY));
diff --git a/usr/palm/applications/com.palm.app.messaging/stylesheets/messaging.css b/usr/palm/applications/com.palm.app.messaging/stylesheets/messaging.css
index 019deb5..acd0763 100644
--- a/usr/palm/applications/com.palm.app.messaging/stylesheets/messaging.css
+++ b/usr/palm/applications/com.palm.app.messaging/stylesheets/messaging.css
@@ -701,11 +701,11 @@ img.avatar-overlay {
 }
 
 .their-chat .message-container {
-		width: 255px;
+	width: 85%;
 }
 
 .my-chat .message-container {
-		width: 280px;
+	width: 95%;
 }
 
 .chat-entry .chat-balloon {
@@ -994,7 +994,7 @@ img.avatar-overlay {
 		position: fixed;
 		bottom: 0px;
 		left: 0px;
-		width: 320px;
+		width: 100%;
 }
 
 /* Force addressing widget to sit on top of
