--- .orig/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js
+++ /usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js
@@ -60,7 +60,7 @@
 		this.messageTemplatePath = 'chatview/message/';
 		
 		this.appMenuAddPictureItem = {
-			label: $L('Add picture...'),
+			label: $L('Attach...'),
 			command: 'addPicture',
 			disabled: false
 		};
@@ -2422,7 +2422,7 @@
 	attachFilePicker: function(event){
 		var params = {
 			actionType: 'attach',
-			kinds: ['image'], // TODO: integrate other media types
+			kinds: ['image','audio','video'], // TODO: integrate other media types
 			defaultKind: 'image',
 			onSelect: this.handleFilePickerSelect.bind(this)
 		};
\ No newline at end of file
--- .orig/usr/palm/applications/com.palm.app.messaging/app/controllers/compose-assistant.js
+++ /usr/palm/applications/com.palm.app.messaging/app/controllers/compose-assistant.js
@@ -15,7 +15,7 @@
 			this.toggleDebugMenu = listviewSceneAssistant.toggleDebugMenu.bind(listviewSceneAssistant,true);
 		}
 
-		this.appMenuAddPictureItem = {label: $L('Add picture...'), command:'addPicture', enabled:true};
+		this.appMenuAddPictureItem = {label: $L('Attach...'), command:'addPicture', enabled:true};
 
 		this.appMenuAttrs = {omitDefaultItems: true};	
 		this.appMenuModel = { visible:true, label: $L('Chat view menu'), items: [Mojo.Menu.editItem,this.appMenuAddPictureItem,Mojo.Menu.helpItem]};   
@@ -330,7 +330,7 @@
   attachFilePicker: function(event) {   
     var params = {
 		actionType: 'attach',
-        kinds: ['image'], // TODO: integrate other media types
+        kinds: ['image','audio','video'], // TODO: integrate other media types
         defaultKind: 'image',
         onSelect: this.handleFilePickerSelect.bind(this)
     };
