Index: /usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js
===================================================================
--- .orig/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js
+++ /usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js
@@ -2013,7 +2013,7 @@ var ChatviewAssistant = Class.create({
 	attachFilePicker: function(event){
 		var params = {
 			actionType: 'attach',
-			kinds: ['image'], // TODO: integrate other media types
+			kinds: ['image','audio','video'], // TODO: integrate other media types
 			defaultKind: 'image',
 			onSelect: this.handleFilePickerSelect.bind(this)
 		};
Index: /usr/palm/applications/com.palm.app.messaging/app/controllers/compose-assistant.js
===================================================================
--- .orig/usr/palm/applications/com.palm.app.messaging/app/controllers/compose-assistant.js
+++ /usr/palm/applications/com.palm.app.messaging/app/controllers/compose-assistant.js
@@ -324,7 +324,7 @@ var ComposeAssistant = Class.create({
   attachFilePicker: function(event) {   
     var params = {
 		actionType: 'attach',
-        kinds: ['image'], // TODO: integrate other media types
+        kinds: ['image','audio','video'], // TODO: integrate other media types
         defaultKind: 'image',
         onSelect: this.handleFilePickerSelect.bind(this)
     };
