diff --git a/usr/palm/applications/com.palm.app.messaging/app/controllers/listview-assistant.js b/usr/palm/applications/com.palm.app.messaging/app/controllers/listview-assistant.js
index c301f25..68fe9c6 100644
--- a/usr/palm/applications/com.palm.app.messaging/app/controllers/listview-assistant.js
+++ b/usr/palm/applications/com.palm.app.messaging/app/controllers/listview-assistant.js
@@ -64,7 +64,7 @@ var ListviewAssistant = Class.create(App.Scene, {
 			{label: $L('Available'), command:this.Messaging.Availability.getAvailabilityAsConstText(this.Messaging.Availability.AVAILABLE), secondaryIcon:'status-available'},
 			{label: $L('Busy'),      command:this.Messaging.Availability.getAvailabilityAsConstText(this.Messaging.Availability.BUSY),      secondaryIcon:'status-away'},
 			// libpurple doesn't support invisible status for gtalk so commenting this out for now.
-			//{label: $L('Invisible'), command:this.Messaging.Availability.getAvailabilityAsConstText(this.Messaging.Availability.INVISIBLE),  secondaryIcon:'status-invisible'},
+			{label: $L('Invisible'), command:this.Messaging.Availability.getAvailabilityAsConstText(this.Messaging.Availability.INVISIBLE),  secondaryIcon:'status-invisible'},
 			{label: $L('Sign off'),  command:this.Messaging.Availability.getAvailabilityAsConstText(this.Messaging.Availability.OFFLINE),   secondaryIcon:'status-offline'}
 		];
 		
@@ -1382,7 +1382,8 @@ var ListviewAssistant = Class.create(App.Scene, {
 		
 		if (selectedValue != this.Messaging.Availability.OFFLINE &&
 		this.currentAvailability != this.Messaging.Availability.AVAILABLE &&
-		this.currentAvailability != this.Messaging.Availability.BUSY) {
+	    this.currentAvailability != this.Messaging.Availability.BUSY &&
+    	this.currentAvailability != this.Messaging.Availability.INVISIBLE) {
 			this.setAvailabilityIndicator(this.Messaging.Availability.PENDING);
 		} else {
 			this.setAvailabilityIndicator(selectedValue);
