diff --git a/usr/palm/applications/com.palm.app.messaging/app/compose/ComposeView.js b/usr/palm/applications/com.palm.app.messaging/app/compose/ComposeView.js
index 69986f2..8b345bd 100644
--- a/usr/palm/applications/com.palm.app.messaging/app/compose/ComposeView.js
+++ b/usr/palm/applications/com.palm.app.messaging/app/compose/ComposeView.js
@@ -200,7 +200,7 @@ enyo.kind({
 		}
 	},
 	considerForSend: function(){
-		var message = this.$.richText.getValue();
+		var message = this.$.richText.getHtml();
 		if (message.length > 0) {//todo: check max length?
 			var contacts = this.$.addressingPopup.getContacts();
 			enyo.log("ComposeView::considerForSend::contacts:",contacts);
@@ -422,7 +422,7 @@ enyo.kind({
 		// Pressing "enter" should send a message and clear the input
 		if (inEvent.keyCode === 13) {
 			inEvent.preventDefault();
-			messageText = this.$.richText.getValue();
+			messageText = this.$.richText.getHtml();
 
 			// Only send non-empty messages
 			if (messageText) {
@@ -437,7 +437,7 @@ enyo.kind({
 		}
 	},
 	saveDraft: function(){
-		var message = this.$.richText.getValue();
+		var message = this.$.richText.getHtml();
 		if (message.length > 0) {
 			var contacts = this.$.addressingPopup.getContacts();
 			
diff --git a/usr/palm/applications/com.palm.app.messaging/app/conversations/ConversationList.js b/usr/palm/applications/com.palm.app.messaging/app/conversations/ConversationList.js
index e712c30..2b93c08 100644
--- a/usr/palm/applications/com.palm.app.messaging/app/conversations/ConversationList.js
+++ b/usr/palm/applications/com.palm.app.messaging/app/conversations/ConversationList.js
@@ -306,7 +306,7 @@ enyo.kind({
 			this.forceSendIfOffline = false;
 			this.person = undefined;
 			//save draft
-			this.saveMessageToDraft(this.$.richText.getValue(), inOldChatThread._id);
+			this.saveMessageToDraft(this.$.richText.getHtml(), inOldChatThread._id);
 			this.$.richText.setValue("");
 			this.deleteTransientMessages(inOldChatThread._id);
 			this.deleteEmptyChatThread(inOldChatThread._id);
@@ -618,7 +618,7 @@ enyo.kind({
 			this.updateDashboard({_id: undefined}, enyo.application.messageDashboardManager);
 			this.updateDashboard({_id: undefined}, enyo.application.inviteDashboardManager);
 			//save draft
-			this.saveMessageToDraft(this.$.richText.getValue(), this.chatThread._id);
+			this.saveMessageToDraft(this.$.richText.getHtml(), this.chatThread._id);
 			this.$.richText.setValue("");
 		}
 		this.chatThread = undefined;
@@ -710,7 +710,7 @@ enyo.kind({
 		// Pressing "enter" should send a message and clear the input
 		if (inEvent.keyCode === 13) {
 			inEvent.preventDefault();
-			messageText = this.$.richText.getValue();
+			messageText = this.$.richText.getHtml();
 
 			// Only send non-empty messages
 			if (messageText) {
@@ -912,7 +912,7 @@ enyo.kind({
 		}
 	},
 	sendButtonClick: function() {
-		if(this.$.richText.getValue().length>0){
+		if(this.$.richText.getHtml().length>0){
 			this.considerForSend();
 		}
 	},
@@ -1005,7 +1005,7 @@ enyo.kind({
 				deliveryReport: deliveryReport
 			},
 			to: [recipient],
-			messageText: this.$.richText.getValue(),
+			messageText: this.$.richText.getHtml(),
 			serviceName: selectedTransport.serviceName
 		};
 		
