Patch to add time stamps to all text messages.
Index: /usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js
===================================================================
--- .orig/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js
+++ /usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js
@@ -1212,6 +1212,9 @@ var ChatviewAssistant = Class.create({
 	preFormatChatList: function(stuff){
 		for(var i=0; i<stuff.list.length; i++) {
 			var msg = stuff.list[i];
+			var ts = new Date();
+			ts.setTime(msg.timeStamp);
+			msg.timeStampText = Mojo.Format.formatDate(ts,'short');
 			if(msg.summary && !ChatFlags.isTransient(msg.flags)) {
 				msg.summary = msg.summary.escapeHTML();
 				msg.summary = PalmSystem.runTextIndexer(msg.summary);
@@ -1235,7 +1238,7 @@ var ChatviewAssistant = Class.create({
 		 	var deviceTimeStampMinutes = d.getMinutes();
 			d.setMinutes((deviceTimeStampMinutes - (deviceTimeStampMinutes % 5)));
 			d.setSeconds(0);
-			msg.dividerText = msg.transportClass + BucketDateFormatter.getDateBucket(d,true,true);
+			msg.dividerText = msg.transportClass; // + BucketDateFormatter.getDateBucket(d,true,true);
 	
 			// if the INCOMING message was sent longer than 10 minutes before it arrived on device,
 			// add sent date to indicate that this delay happened
@@ -2115,4 +2118,4 @@ var ChatviewAssistant = Class.create({
 		this.controller.sceneScroller.mojo.revealBottom();
 		this.controller.sceneScroller.mojo.revealBottom();
 	}
-});
\ No newline at end of file
+});
Index: /usr/palm/applications/com.palm.app.messaging/app/views/chatview/message/message-text.html
===================================================================
--- .orig/usr/palm/applications/com.palm.app.messaging/app/views/chatview/message/message-text.html
+++ /usr/palm/applications/com.palm.app.messaging/app/views/chatview/message/message-text.html
@@ -1 +1 @@
-<div class="messageText">#{-messageText} #{-errorContent}</div>
\ No newline at end of file
+<div class="messageText"><span class="timeStamp">#{-timeStampText}</span><pre>#{-messageText}</pre> #{-errorContent}</div>
Index: /usr/palm/applications/com.palm.app.messaging/stylesheets/messaging.css
===================================================================
--- .orig/usr/palm/applications/com.palm.app.messaging/stylesheets/messaging.css
+++ /usr/palm/applications/com.palm.app.messaging/stylesheets/messaging.css
@@ -659,6 +659,18 @@ img.avatar-overlay {
 	font-size:14px;
 }
 
+.messageText pre {
+	font-family: inherit;
+	font-size: inherit;
+	color: inherit;
+}
+.timeStamp {
+	font-size: 12px;
+font-st	yle: italic;
+	color: #1111ff;
+}
+
+
 .my-chat .chat-balloon-wrapper, 
 .their-chat .chat-balloon-wrapper {
   padding: 1px 0;
@@ -1152,4 +1164,4 @@ img.avatar-overlay {
 }
 #im-links .palm-row .title {
 	padding: 9px 14px;
-}
\ No newline at end of file
+}
