diff --git a/usr/lib/luna/system/luna-applauncher/app/controllers/global-search-assistant.js b/usr/lib/luna/system/luna-applauncher/app/controllers/global-search-assistant.js
index b9db2c2..0d4656c 100644
--- a/usr/lib/luna/system/luna-applauncher/app/controllers/global-search-assistant.js
+++ b/usr/lib/luna/system/luna-applauncher/app/controllers/global-search-assistant.js
@@ -617,6 +617,13 @@ GlobalSearchAssistant = Class.create({
 		}
 		if (data.record.phoneNumbers && data.record.phoneNumbers.length > 0) {
 			$A(data.record.phoneNumbers).each(function(a){
+				if (a.label == 5){
+					//Dont show fax numbers
+				}
+				else if (a.label == 6){
+					//Dont show fax numbers
+				}
+				else{
 				a.contactId = contact.id;
 				a.value = Mojo.Format.formatPhoneNumber(a.value);
 				a.icon = "im-image";
@@ -625,6 +632,7 @@ GlobalSearchAssistant = Class.create({
 				else 
 					a.label = that.phoneLabel[a.label];
 				phoneList.push(a);
+				}
 			});
 		}
 		if (data.record.imNames && data.record.imNames.length > 0) {
