diff --git a/usr/lib/luna/system/luna-applauncher/app/models/utils.js b/usr/lib/luna/system/luna-applauncher/app/models/utils.js
index c5f5d21..39f8115 100644
--- a/usr/lib/luna/system/luna-applauncher/app/models/utils.js
+++ b/usr/lib/luna/system/luna-applauncher/app/models/utils.js
@@ -159,7 +159,7 @@ var Utils = {
 	
 	launchBrowser: function(url) {
 		var params = {scene: 'page', target: url};
-		ApplicationService.launch('com.palm.app.browser', params);		
+		ApplicationService.launch('com.openmobl.app.universe', params);		
 	},
 	
 	launchApp:function(appinfo){
