diff --git a/usr/lib/luna/system/luna-applauncher/app/controllers/global-search-assistant.js b/usr/lib/luna/system/luna-applauncher/app/controllers/global-search-assistant.js
index b9db2c2..d04306c 100644
--- a/usr/lib/luna/system/luna-applauncher/app/controllers/global-search-assistant.js
+++ b/usr/lib/luna/system/luna-applauncher/app/controllers/global-search-assistant.js
@@ -45,7 +45,7 @@ GlobalSearchAssistant = Class.create({
 		this.offset=0;
 		this.limit=40;
 		this.URLs = {
-			'google':$L("www.google.com/m/search?client=ms-palm-webOS&channel=iss&q="),
+			'google':$L("www.google.com/m/ncr/search?client=ms-palm-webOS&channel=iss&q="),
 			'wikipedia':$L("http://en.m.wikipedia.org/wiki/Special:Search?search="),
 			'twitter': $L("http://search.twitter.com/search?q="),
 			'yahoo': $L("http://search.yahoo.com/search?p="),
diff --git a/usr/palm/applications/com.palm.app.browser/app/controllers/preferences-assistant.js b/usr/palm/applications/com.palm.app.browser/app/controllers/preferences-assistant.js
index fba2ffb..5550112 100644
--- a/usr/palm/applications/com.palm.app.browser/app/controllers/preferences-assistant.js
+++ b/usr/palm/applications/com.palm.app.browser/app/controllers/preferences-assistant.js
@@ -30,7 +30,7 @@ PreferencesAssistant.DefaultSearchProvider = {
 
 	id: 'google',
 	title: $L('Google'),
-	weburl: $L('http://www.google.com/m/search?client=ms-palm-webOS&channel=bm&q=#{query}')
+	weburl: $L('http://www.google.com/m/ncr/search?client=ms-palm-webOS&channel=bm&q=#{query}')
 };
 
 PreferencesAssistant.prototype.setup = function() {
