diff --git a/usr/lib/luna/system/luna-applauncher/app/controllers/global-search-assistant.js b/usr/lib/luna/system/luna-applauncher/app/controllers/global-search-assistant.js
index 6f3368e..ac38cba 100644
--- a/usr/lib/luna/system/luna-applauncher/app/controllers/global-search-assistant.js
+++ b/usr/lib/luna/system/luna-applauncher/app/controllers/global-search-assistant.js
@@ -45,7 +45,7 @@ GlobalSearchAssistant = Class.create({
 		this.offset=0;
 		this.limit=40;
 		this.URLs = {
-			'google':$L("www.google.com/m/search?client=ms-palm-webOS&channel=iss&q="),
+			'google':$L("www.google.com/m/ncr/search?client=ms-palm-webOS&channel=iss&q="),
 			'wikipedia':$L("http://en.m.wikipedia.org/wiki/Special:Search?search="),
 			'twitter': $L("http://search.twitter.com/search?q="),
 			'yahoo': $L("http://search.yahoo.com/search?p="),
