diff --git a/usr/lib/luna/system/luna-applauncher/app/HomeView.js b/usr/lib/luna/system/luna-applauncher/app/HomeView.js
index 16e426a..1499e92 100644
--- a/usr/lib/luna/system/luna-applauncher/app/HomeView.js
+++ b/usr/lib/luna/system/luna-applauncher/app/HomeView.js
@@ -27,13 +27,13 @@ enyo.kind({
 					className:"recent-search-container",
 					caption: $L("Recent Searches")
 			},
-			/*{
+			{ // Enable "Clear All" button
 				kind:"Button",
 				name:"clearAllButton",
-				label:$L("CLEAR ALL"),
-				style: "width: 150px; margin-left:auto;margin-right:auto",
+				label:$L("Clear Recent Searches"),
+				style: "width: 300px; margin-left:auto;margin-right:auto",
 				onclick:"clearRecentSearches"
-			} */
+			}
 			]
 		}
 	],
@@ -76,12 +76,12 @@ enyo.kind({
 		this.populateRecentSearchList();
 		if(this.recentSearchStrings.length > 0) {
 			this.$.recentSearchListGroup.setShowing(true);
-			//this.$.clearAllButton.setShowing(true);
+			this.$.clearAllButton.setShowing(true);
 			this.$.emptyMain.setShowing(false);
 		}
 		else {
 			this.$.recentSearchListGroup.setShowing(false);
-			//this.$.clearAllButton.setShowing(false);
+			this.$.clearAllButton.setShowing(false);
 			this.$.emptyMain.setShowing(true);
 		}	
 	},
