diff --git a/usr/lib/luna/system/luna-applauncher/app/controllers/global-search-assistant.js b/usr/lib/luna/system/luna-applauncher/app/controllers/global-search-assistant.js
index b9db2c2..a950278 100644
--- a/usr/lib/luna/system/luna-applauncher/app/controllers/global-search-assistant.js
+++ b/usr/lib/luna/system/luna-applauncher/app/controllers/global-search-assistant.js
@@ -619,6 +619,8 @@ GlobalSearchAssistant = Class.create({
 			$A(data.record.phoneNumbers).each(function(a){
 				a.contactId = contact.id;
 				a.value = Mojo.Format.formatPhoneNumber(a.value);
+				//a.icon = "im-image";
+				if (a.label == 3)
 				a.icon = "im-image";
 				if (a.label == 2 && a.customLabel) 
 					a.label = a.customLabel;
