diff --git a/usr/lib/luna/system/luna-applauncher/app/controllers/global-search-assistant.js b/usr/lib/luna/system/luna-applauncher/app/controllers/global-search-assistant.js
index b9db2c2..5c12059 100644
--- a/usr/lib/luna/system/luna-applauncher/app/controllers/global-search-assistant.js
+++ b/usr/lib/luna/system/luna-applauncher/app/controllers/global-search-assistant.js
@@ -48,8 +48,8 @@ GlobalSearchAssistant = Class.create({
 			'google':$L("www.google.com/m/search?client=ms-palm-webOS&channel=iss&q="),
 			'wikipedia':$L("http://en.m.wikipedia.org/wiki/Special:Search?search="),
 			'twitter': $L("http://search.twitter.com/search?q="),
-			'yahoo': $L("http://search.yahoo.com/search?p="),
-			'bing': $L("http://www.bing.com/search?q=")
+			'yahoo': $L("http://m.yahoo.com/w/search?p="),
+			'bing': $L("http://m.bing.com/TouchCommonPage.aspx?go=&d=&dl=&a=results&MID=1&asq=&asems=&q=")
 		};
 		
 		this.searchEngineImages = {
@@ -144,6 +144,8 @@ GlobalSearchAssistant = Class.create({
 		this.controller.setupWidget('web_drawer', {unstyled:true, property:'showWeb'}, this.webDrawer);
 		
 		this.searchEngineList = [{id:"google",title:"Google",imgFile:"images/search-google.png", showImage:"block", showName:"none"},
+								 {id:"yahoo",title:"Yahoo",imgFile:"images/search-yahoo.png", showImage:"block", showName:"none"},
+								 {id:"bing",title:"Bing",imgFile:"images/search-bing.png", showImage:"block", showName:"none"},
 								 {id:"map",title:"Google Maps",imgFile:"images/search-maps.png", showImage:"block", showName:"none"},
 								 {id:"wikipedia",title:"Wikipedia",imgFile:"images/search-wikipedia.png", showImage:"block", showName:"none"},
 								 {id:"twitter",title:"Twitter",imgFile:"images/search-twitter.png", showImage:"block", showName:"none"}
