--- .orig/usr/palm/applications/com.palm.app.browser/app/controllers/preferences-assistant.js
+++ /usr/palm/applications/com.palm.app.browser/app/controllers/preferences-assistant.js
@@ -30,7 +30,7 @@
 
 	id: 'google',
 	title: $L('Google'),
-	weburl: $L('http://www.google.com/m/search?client=ms-palm-webOS&channel=bm&q=#{query}')
+	weburl: $L('http://www.google.com/m/ncr/search?client=ms-palm-webOS&channel=bm&q=#{query}')
 };
 
 PreferencesAssistant.prototype.setup = function() {
--- .orig/usr/lib/luna/system/luna-applauncher/app/controllers/global-search-assistant.js
+++ /usr/lib/luna/system/luna-applauncher/app/controllers/global-search-assistant.js
@@ -48,1 +48,1 @@
-			'google':$L("www.google.com/m/search?client=ms-palm-webOS&channel=iss&q="),
+			'google':$L("www.google.com/m/ncr/search?client=ms-palm-webOS&channel=iss&q="),
