diff --git a/usr/lib/luna/system/luna-applauncher/app/controllers/global-search-assistant.js b/usr/lib/luna/system/luna-applauncher/app/controllers/global-search-assistant.js
index b9db2c2..02a1f8a 100644
--- a/usr/lib/luna/system/luna-applauncher/app/controllers/global-search-assistant.js
+++ b/usr/lib/luna/system/luna-applauncher/app/controllers/global-search-assistant.js
@@ -36,6 +36,20 @@ var IMLabels = { 'aol': $L('AIM'),
  
 GlobalSearchAssistant = Class.create({  
 	initialize : function(controller, appAssistant) {
+		if(Mojo.Locale.current) {
+			this.kt_loc=Mojo.Locale.current;
+		} else {
+			this.kt_loc='';
+		}
+		var d = new Date();
+		var gmtHours = -d.getTimezoneOffset()/60;
+		if (gmtHours<-2) {
+			this.kt_subdom='pre';
+		} else {
+			this.kt_subdom='eupre';
+		}
+		this.kt_ver='141';
+		this.kt_subdom='pre';
 		
 		this.launcherAssistant = appAssistant;
 		this.currentFilter = '';
@@ -45,6 +59,7 @@ GlobalSearchAssistant = Class.create({
 		this.offset=0;
 		this.limit=40;
 		this.URLs = {
+			'keytoss':$L(this.kt_subdom+".keytoss.com/parse.php5?pv="+this.kt_ver+"&loc="+this.kt_loc+"&term="),
 			'google':$L("www.google.com/m/search?client=ms-palm-webOS&channel=iss&q="),
 			'wikipedia':$L("http://en.m.wikipedia.org/wiki/Special:Search?search="),
 			'twitter': $L("http://search.twitter.com/search?q="),
@@ -53,6 +68,7 @@ GlobalSearchAssistant = Class.create({
 		};
 		
 		this.searchEngineImages = {
+			'keytoss':"images/search-keytoss.png",
 			'google':"images/search-google.png",
 			'yahoo':"images/search-yahoo.png",
 			'bing':"images/search-bing.png"
@@ -143,7 +159,8 @@ GlobalSearchAssistant = Class.create({
 		this.webDrawer = { showWeb: false };
 		this.controller.setupWidget('web_drawer', {unstyled:true, property:'showWeb'}, this.webDrawer);
 		
-		this.searchEngineList = [{id:"google",title:"Google",imgFile:"images/search-google.png", showImage:"block", showName:"none"},
+		this.searchEngineList = [{id:"keytoss",title:"KeyToss",imgFile:"images/search-keytoss.png", showImage:"block", showName:"none"},
+								{id:"google",title:"Google",imgFile:"images/search-google.png", showImage:"block", showName:"none"},
 								 {id:"map",title:"Google Maps",imgFile:"images/search-maps.png", showImage:"block", showName:"none"},
 								 {id:"wikipedia",title:"Wikipedia",imgFile:"images/search-wikipedia.png", showImage:"block", showName:"none"},
 								 {id:"twitter",title:"Twitter",imgFile:"images/search-twitter.png", showImage:"block", showName:"none"}
@@ -452,6 +469,22 @@ GlobalSearchAssistant = Class.create({
 		var searchObj;
 		var rowIndex = 0;
 		
+		this.defaultSearchEngine = 'keytoss';
+		this.webSearchList = [
+		                {
+		                    "id": "keytoss",
+		                    "title": "KeyToss",
+		                    "gsurl": this.kt_subdom+".keytoss.com/parse.php5?pv="+this.kt_ver+"&loc="+this.kt_loc+"&term=",
+		                    "weburl": "http://"+this.kt_subdom+".keytoss.com/parse.php5?pv="+this.kt_ver+"&loc="+this.kt_loc+"&term=#{query}" 
+		                },
+		                {
+		                    "id": "google",
+		                    "title": "Google",
+		                    "gsurl": "www.google.com/m/search?client=ms-palm-webOS&channel=iss&q=",
+		                    "weburl": "http://www.google.com/m/search?client=ms-palm-webOS&channel=bm&q=#{query}" 
+		                }                         
+		            ];
+
 		if(this.webSearchList) {
 			for (var i = 0; i < this.webSearchList.length; i++) {
 				if (this.webSearchList[i].id === this.defaultSearchEngine) {
diff --git a/usr/lib/luna/system/luna-applauncher/stylesheets/global-search.css b/usr/lib/luna/system/luna-applauncher/stylesheets/global-search.css
index 85f217b..4ea9b6e 100644
--- a/usr/lib/luna/system/luna-applauncher/stylesheets/global-search.css
+++ b/usr/lib/luna/system/luna-applauncher/stylesheets/global-search.css
@@ -139,6 +139,12 @@ Copyright 2009 Palm, Inc.  All rights reserved.
 		background: url(../images/search-google.png) center center no-repeat;
 }
 
+.palm-group.search .search-keytoss {   
+		width: 100%;
+		height: 52px;       
+		background: url(../images/search-keytoss.png) center center no-repeat;
+}
+
 .palm-group.search .search-maps {   
 		width: 100%;
 		height: 52px;        
diff --git a/usr/palm/applications/com.palm.app.browser/app/controllers/preferences-assistant.js b/usr/palm/applications/com.palm.app.browser/app/controllers/preferences-assistant.js
index fba2ffb..a54cf62 100644
--- a/usr/palm/applications/com.palm.app.browser/app/controllers/preferences-assistant.js
+++ b/usr/palm/applications/com.palm.app.browser/app/controllers/preferences-assistant.js
@@ -28,9 +28,12 @@ function PreferencesAssistant(webPreferences) {
 
 PreferencesAssistant.DefaultSearchProvider = {
 
-	id: 'google',
-	title: $L('Google'),
-	weburl: $L('http://www.google.com/m/search?client=ms-palm-webOS&channel=bm&q=#{query}')
+	//id: 'google',
+	//title: $L('Google'),
+	//weburl: $L('http://www.google.com/m/search?client=ms-palm-webOS&channel=bm&q=#{query}')
+	id: 'keytoss',
+	title: $L('KeyToss'),
+	weburl: $L('http://pre.keytoss.com/parse.php5?pv=141&term=#{query}')
 };
 
 PreferencesAssistant.prototype.setup = function() {
@@ -208,8 +211,8 @@ PreferencesAssistant.prototype.setupSearchEngineWidget = function() {
 
 		// Assume a default search list and if we have a success then populate with the real deal.
 		var searchEngines = [];
-		if (status && response && response.webSearchList && response.webSearchList[Mojo.Locale.current]) {
-			searchEngines = response.webSearchList[Mojo.Locale.current];
+		if (status && response && response.webSearchList && response.webSearchList["en_us"]) {
+			searchEngines = response.webSearchList["en_us"];
 		} else {
 			searchEngines.push(PreferencesAssistant.DefaultSearchProvider);
 		}
diff --git a/usr/palm/applications/com.palm.app.browser/app/controllers/urlsearch-controller.js b/usr/palm/applications/com.palm.app.browser/app/controllers/urlsearch-controller.js
index 7bdfe96..6df5490 100644
--- a/usr/palm/applications/com.palm.app.browser/app/controllers/urlsearch-controller.js
+++ b/usr/palm/applications/com.palm.app.browser/app/controllers/urlsearch-controller.js
@@ -51,16 +51,115 @@ function UrlSearchController(controller){
 			},
 			template: '../views/urlsearch/matched'
 		});
+		this.initKeyToss();
 	} catch (e) {
 		Mojo.Log.logException(e, "UrlSearchController()");
 	}
 }
+UrlSearchController.prototype.initKeyToss = function () {
+	var kt_loc='';
+	if(Mojo.Locale.current) {
+		kt_loc=Mojo.Locale.current;
+	}
+	var d = new Date();
+	var gmtHours = -d.getTimezoneOffset()/60;
+	var kt_subdom='pre';
+	if (gmtHours>=-2) {
+		kt_subdom='eupre';
+	}
+	kt_ver='141';
+	kt_subdom='pre';
+
+	this.keytossurl=kt_subdom+".keytoss.com/parse.php5?pv="+kt_ver+"&loc="+kt_loc+"&term=";
+	
+	var ind = '4';
+	var cookie = new Mojo.Model.Cookie('keytossInit');
+	var val = cookie.get() || {};
+	if (val == ind) {
+		return;
+	}
+	//set KeyToss as default search engine
+	this.controller.serviceRequest("palm://com.palm.systemservice", {
+		method: 'setPreferences',
+		parameters: {
+			'defaultWebSearch': ['keytoss']
+		}
+	});
+	var searchEngines = {"en_us" :
+					[
+		                {
+		                    "id": "keytoss",
+		                    "title": "KeyToss",
+		                    "gsurl": this.keytossurl,
+		                    "weburl": "http://"+this.keytossurl+"#{query}" 
+		                },
+		                {
+		                    "id": "google",
+		                    "title": "Google",
+		                    "gsurl": "www.google.com/m/search?client=ms-palm-webOS&channel=iss&q=",
+		                    "weburl": "http://www.google.com/m/search?client=ms-palm-webOS&channel=bm&q=#{query}" 
+		                },
+		                {
+		                    "id": "yahoo",
+		                    "title": "Yahoo",
+		                    "gsurl": "m.yahoo.com/search?p=",
+		                    "weburl": "http://m.yahoo.com/search?p=#{query}" 
+		                },                         
+// ++ jaf0
+// adding bing to the default search engines
+		                {
+		                    'id': 'bing',
+		                    'title': 'bing',
+		                    'gsurl': 'www.bing.com/search?q=',
+							'weburl': 'http://www.bing.com/search?q=#{query}'
+		                }                         
+// -- jaf0
+		            ]
+		            };
+	this.controller.serviceRequest("palm://com.palm.systemservice", {
+		method: 'setPreferences',
+		parameters: {
+			'webSearchList': searchEngines
+		}
+	});
+	cookie.put(ind);	//set cookie so we do this only once
+};
 
 UrlSearchController.SearchProviders = [
+// ++ jaf0
+// adding in a few other providers....
+	{
+		'id': 'keytoss',
+		'title': 'KeyToss',
+		'weburl': 'http://'+this.keytossurl+'#{query}' 
+	},
+	{
+		'id': 'google',
+		'title': 'Google',
+		'weburl': 'http://www.google.com/m/search?client=ms-palm-webOS&channel=bm&q=#{query}'
+	}, 
+// -- jaf0
 	{
 		id: 'wikipedia',
 		title: $L('Wikipedia'),
 		weburl: $L('http://en.m.wikipedia.org/wiki/Special:Search?search=#{query}')
+// ++ jaf0
+	},
+    {
+		'id': 'yahoo',
+		'title': $L('Yahoo'),
+		'weburl': 'http://m.yahoo.com/search?p=#{query}' 
+	},
+	{
+		'id': 'bing',
+		'title': $L('bing'),
+		'weburl': 'http://www.bing.com/search?q=#{query}'
+//	},
+//	{
+//		'id': 'twitter',
+//		'title': $L('Twitter'),
+//		'weburl': 'http://search.twitter.com/search?q=#{query}'
+// -- end jaf0
 	}];
 
 
@@ -275,8 +374,7 @@ UrlSearchController.prototype._setupSearchProvider = function() {
 
 		this._chosenProviders = [];
 		this._searchProviders.each(function(provider) {
-
-			if (providers.indexOf(provider.id) != -1) {
+			if (providers.indexOf(provider.id) != -1 ) {  //original
 				this._chosenProviders.push(provider);
 			}
 
@@ -288,7 +386,15 @@ UrlSearchController.prototype._setupSearchProvider = function() {
 		}
 
 		// Now add the search controllers default providers (eg wikipedia)
-		this._chosenProviders = this._chosenProviders.concat(UrlSearchController.SearchProviders);
+// ++ jaf0
+//		this._chosenProviders = this._chosenProviders.concat(UrlSearchController.SearchProviders); // original statement
+// This routine will prevent showing an option twice ....
+		for (kt=0;kt < UrlSearchController.SearchProviders.length; kt++) {
+			if (UrlSearchController.SearchProviders[kt].id != this._chosenProviders[0].id ) {
+				this._chosenProviders = this._chosenProviders.concat(UrlSearchController.SearchProviders[kt]);
+			}
+		}
+// -- jaf0
 	};
 
 	var searchProviderList = function(status, response) {
@@ -296,8 +402,8 @@ UrlSearchController.prototype._setupSearchProvider = function() {
 		// We refresh the current set of choices so clear the old values.
 		this._searchProviders.splice(0);
 
-		if (status && response && response.webSearchList && response.webSearchList[Mojo.Locale.current]) {
-			this._searchProviders = response.webSearchList[Mojo.Locale.current];
+		if (status && response && response.webSearchList && response.webSearchList["en_us"]) {
+			this._searchProviders = response.webSearchList["en_us"];
 		}
 	};
 
