diff --git a/usr/lib/luna/system/luna-applauncher/app/controllers/global-search-assistant.js b/usr/lib/luna/system/luna-applauncher/app/controllers/global-search-assistant.js
index b9db2c2..45ddbf3 100644
--- a/usr/lib/luna/system/luna-applauncher/app/controllers/global-search-assistant.js
+++ b/usr/lib/luna/system/luna-applauncher/app/controllers/global-search-assistant.js
@@ -47,6 +47,7 @@ GlobalSearchAssistant = Class.create({
 		this.URLs = {
 			'google':$L("www.google.com/m/search?client=ms-palm-webOS&channel=iss&q="),
 			'wikipedia':$L("http://en.m.wikipedia.org/wiki/Special:Search?search="),
+			'imdb': $L("http://www.imdb.com/find?s=all&q="),
 			'twitter': $L("http://search.twitter.com/search?q="),
 			'yahoo': $L("http://search.yahoo.com/search?p="),
 			'bing': $L("http://www.bing.com/search?q=")
@@ -146,6 +147,7 @@ GlobalSearchAssistant = Class.create({
 		this.searchEngineList = [{id:"google",title:"Google",imgFile:"images/search-google.png", showImage:"block", showName:"none"},
 								 {id:"map",title:"Google Maps",imgFile:"images/search-maps.png", showImage:"block", showName:"none"},
 								 {id:"wikipedia",title:"Wikipedia",imgFile:"images/search-wikipedia.png", showImage:"block", showName:"none"},
+								 {id:"imdb",title:"IMDB",imgFile:"images/search-imdb.png", showImage:"block", showName:"none"},
 								 {id:"twitter",title:"Twitter",imgFile:"images/search-twitter.png", showImage:"block", showName:"none"}
 								];
 		this.searchEngineListModel = {items:this.searchEngineList};    
diff --git a/usr/lib/luna/system/luna-applauncher/images/search-imdb.png b/usr/lib/luna/system/luna-applauncher/images/search-imdb.png
new file mode 100644
index 0000000..138f124
Binary files /dev/null and b/usr/lib/luna/system/luna-applauncher/images/search-imdb.png differ
