diff --git a/usr/lib/luna/system/luna-systemui/data/PowerdService.js b/usr/lib/luna/system/luna-systemui/data/PowerdService.js
index 5200ee8..c405761 100644
--- a/usr/lib/luna/system/luna-systemui/data/PowerdService.js
+++ b/usr/lib/luna/system/luna-systemui/data/PowerdService.js
@@ -119,7 +119,7 @@ enyo.kind({
 					
 				if(!this.chargingBannerShown && (inResponse.DockConnected || inResponse.USBName == "wall")) {
 					var soundClassName = enyo.application.getTelephonyService().getOnActiveCall() ? "none" : "notifications";
-					enyo.windows.addBannerMessage($L("Charging Battery"), "{}",'/usr/lib/luna/system/luna-systemui/images/notification-small-charging.png', soundClassName, "/usr/palm/sounds/charging.mp3");
+					enyo.windows.addBannerMessage($L("Charging Battery"), "{}",'/usr/lib/luna/system/luna-systemui/images/notification-small-charging.png', "vibrate" );
 					this.chargingBannerShown = true;
 				}  				
 			}
