diff --git a/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js b/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
index 6d0a87a..4c15c91 100644
--- a/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
+++ b/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
@@ -2016,7 +2016,7 @@ handlePowerNotifications: function(payload) {
 					this.controller.showBanner({
 						messageText: batteryalert,
 						icon: "/usr/lib/luna/system/luna-systemui/images/battery-"+i+'.png',
-						soundClass: "notifications"
+						soundClass: "none"
 					},{}, 'batteryAlert');	
 				}
 				
