diff --git a/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js b/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
index c78c5de..77bd434 100644
--- a/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
+++ b/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
@@ -1622,7 +1622,7 @@ handlePowerNotifications: function(payload) {
 				this.controller.showBanner({
 						messageText: batteryalert,
 						icon: "/usr/lib/luna/system/luna-systemui/images/notification-small-charging.png",
-						soundClass: soundClassName
+						soundClass: ""
 					},{}, 'chargingAlert'); 				
 			}
 			this.chargingBannerShown = true;			
