diff --git a/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js b/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
index d4ae043..e38a4b5 100644
--- a/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
+++ b/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
@@ -60,6 +60,7 @@ var BarAssistant = Class.create({
 		
 		// Initialize and Subscribe for audio Status Notification
 		//this.initOnServerStart("com.palm.audio", this.audioInit.bind(this));
+		this.initOnServerStart("com.palm.keys", this.muteControl.bind(this));
 				
 		// Initialize and Subscribe for Update Notification
 		this.initOnServerStart("com.palm.update", this.updateInit.bind(this));
@@ -3104,6 +3105,50 @@ audioInit: function() {
 		onSuccess: this.handleVoiceDialingAudioNotifications.bind(this)});
 },
 
+muteControl: function() {
+	this.controller.serviceRequest('palm://com.palm.keys/switches', {
+		method:'status',
+		parameters: {"subscribe":true},
+		onSuccess: function(payload) {
+			this.controller.serviceRequest('palm://com.palm.audio/media', {
+				method:'status',
+				parameters: {},
+				onSuccess: function(response) {
+					this.setMutedSystem(!response['ringer switch']);
+					this.setMutedRingtone(!response['ringer switch']);
+					this.setMutedMedia(!response['ringer switch']
+							&& (response.scenario=="media_back_speaker"));
+				}.bind(this)
+			});
+		}.bind(this)
+	});
+},
+
+setMutedSystem: function(mute) {
+	this.controller.serviceRequest('palm://com.palm.audio/system', {
+		method:'setMuted',
+		parameters:{muted:mute}
+	});
+},
+
+setMutedRingtone: function(mute) {
+	this.controller.serviceRequest('palm://com.palm.audio/ringtone', {
+		method:'setMuted',
+		parameters:{muted:mute}
+	});
+},
+
+setMutedMedia: function(mute) {
+	var value = "up";
+	if(mute) {
+		value = "down";
+	}
+	this.controller.serviceRequest('palm://com.palm.audio/media', {
+		method:'ramp',
+		parameters:{command:value}
+	});
+},
+
 handlePhoneAudioNotifications: function(payload) {
 	
 	if(!payload)
