diff --git a/etc/audio/a/phone_front_speaker.txt.rej b/etc/audio/a/phone_front_speaker.txt.rej
new file mode 100644
index 0000000..4942385
--- /dev/null
+++ b/etc/audio/a/phone_front_speaker.txt.rej
@@ -0,0 +1,10 @@
+--- etc/audio/a/phone_front_speaker.txt
++++ etc/audio/a/phone_front_speaker.txt
+@@ -87,6 +87,6 @@
+     setreg 0x15 0x1e
+ end
+ begin phone_front_speaker_volume_9
+-    setreg 0x14 0x25
++    setreg 0x14 0x31
+     setreg 0x15 0x1e
+ end
diff --git a/etc/audio/a/phone_front_speaker.txt.~1~ b/etc/audio/a/phone_front_speaker.txt.~1~
new file mode 100644
index 0000000..e69de29
