Workaround for Touchlink Charging Event Problem
Index: /usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
===================================================================
--- .orig/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
+++ /usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
@@ -1477,7 +1477,8 @@ handlePowerNotifications: function(paylo
 				stageController.delegateToSceneAssistant("close");
 			}
 			
-			if(!this.chargingBannerShown && payload.name && (payload.name == "wall" || payload.name == "puck")) {
+			if(!this.chargingBannerShown && payload.name && ((payload.name == "wall") || 
+			                                                 (payload.name == "puck" && this.batteryLevel < 100))) {
 				var batteryalert = $L("Charging Battery");
 				var soundClassName = this.onActiveCall ? "none" : "alerts";			
 				this.controller.showBanner({
