diff --git a/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js b/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
index 0e239cc..837ddce 100644
--- a/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
+++ b/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
@@ -2162,7 +2162,7 @@ handlePowerNotifications: function(payload) {
 				this.controller.showBanner({
 						messageText: batteryalert,
 						icon: "/usr/lib/luna/system/luna-systemui/images/notification-small-charging.png",
-						soundClass: soundClassName
+						soundClass: "vibrate"
 					},{}, 'chargingAlert'); 
 			}
 			// This flag is set to true only when there is "name" property is set in the payload. 
