diff --git a/etc/audio/a/phone_headset_mic.txt.rej b/etc/audio/a/phone_headset_mic.txt.rej
new file mode 100644
index 0000000..22cc649
--- /dev/null
+++ b/etc/audio/a/phone_headset_mic.txt.rej
@@ -0,0 +1,9 @@
+--- etc/audio/a/phone_headset_mic.txt
++++ etc/audio/a/phone_headset_mic.txt
+@@ -85,5 +85,5 @@
+     setreg 0x14 0x22
+ end
+ begin phone_headset_mic_volume_9
+-    setreg 0x14 0x24
++    setreg 0x14 0x30
+ end
diff --git a/etc/audio/a/phone_headset_mic.txt.~1~ b/etc/audio/a/phone_headset_mic.txt.~1~
new file mode 100644
index 0000000..e69de29
