diff --git a/usr/palm/applications/com.palm.app.phone/app/controllers/pin-assistant.js b/usr/palm/applications/com.palm.app.phone/app/controllers/pin-assistant.js
index a4e40b7..c6037a1 100644
--- a/usr/palm/applications/com.palm.app.phone/app/controllers/pin-assistant.js
+++ b/usr/palm/applications/com.palm.app.phone/app/controllers/pin-assistant.js
@@ -30,6 +30,8 @@ var PinAssistant = Class.create({
 	},
 	
   	setup: function() {
+		this.unlock();
+		return;
 		try {
 			Mojo.Log.info("Phone App - Pin Assistant -- Setup Called");
 			this.controller.document.body.className = "firstuse"; 
