diff --git a/usr/palm/applications/com.palm.app.screenlock/source/ScreenLockPrefApp.js b/usr/palm/applications/com.palm.app.screenlock/source/ScreenLockPrefApp.js
index a8be86f..f8fb5f8 100644
--- a/usr/palm/applications/com.palm.app.screenlock/source/ScreenLockPrefApp.js
+++ b/usr/palm/applications/com.palm.app.screenlock/source/ScreenLockPrefApp.js
@@ -29,10 +29,17 @@ enyo.kind({
 					        components: [
 					                     {flex:1, content:$L("Turn off After")},
 					                     {kind: "ListSelector", value: 30, name:"displayTimer", onChange:"handleDisplayTimerChange", items: [
+					                                            					{caption: $L('15 seconds'), value:15},
+					                                            					{caption: $L('30 seconds'), value:30},
 					                                            					{caption: $L('1 minute'), value:60},
 					                                            					{caption: $L('2 minutes'), value:120},
 					                                            					{caption: $L('5 minutes'), value:300},
-					                                            					{caption: $L('10 minutes'), value:600}
+					                                            					{caption: $L('10 minutes'), value:600},
+					                                            					{caption: $L('30 minutes'), value:1800},
+					                                            					{caption: $L('1 hour'), value:3600},
+					                                            					{caption: $L('3 hours'), value:10800},
+					                                            					{caption: $L('8 hours'), value:28800},
+					                                            					{caption: $L('24 hours'), value:86400}
 					                                            				]}
 					]}
 	        	]},
