diff --git a/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js b/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
index 786f7ae..e94aaa1 100644
--- a/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
+++ b/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
@@ -1896,7 +1896,12 @@ handlePowerNotifications: function(payload) {
 			this.chargingBannerShown = false;			
 			Mojo.Log.info("SystemUI- Charger Notification "+ this.isCharging + " Last battery level "+ this.lastBatteryLevel);
 			$('power').className = 'battery-' + this.lastBatteryLevel;	
-			Mojo.Controller.getAppController().removeBanner('chargingAlert');		
+			Mojo.Controller.getAppController().removeBanner('chargingAlert');
+			this.controller.serviceRequest('palm://com.palm.display', {
+                                 method: 'control/setState',
+                                parameters: {state: "off"}
+                        });
+		
 		}	
 	}
 },
