diff --git a/usr/palm/applications/com.palm.app.phone/app/controllers/app-assistant.js b/usr/palm/applications/com.palm.app.phone/app/controllers/app-assistant.js
index 7f6be38..ae9c67b 100644
--- a/usr/palm/applications/com.palm.app.phone/app/controllers/app-assistant.js
+++ b/usr/palm/applications/com.palm.app.phone/app/controllers/app-assistant.js
@@ -1062,7 +1062,7 @@ AppAssistant.prototype.popEmergencyActiveCall = function() {
 // show dialpad for emergency call and screenlock
 AppAssistant.prototype.showLimitedDialpad = function(sc, dialParams) {
 	dialParams.limited = true;
-		
+	dialParams.number = this.emergencyNumbers;		
 	sc.pushScene({
 		"name": "dialpad",
 		automaticFocusAdvance: false,
