diff --git a/usr/palm/applications/com.palm.app.phone/phoneAppPinCode/source/PinUnlock.js b/usr/palm/applications/com.palm.app.phone/phoneAppPinCode/source/PinUnlock.js
index e651e4f..4aa3fd6 100644
--- a/usr/palm/applications/com.palm.app.phone/phoneAppPinCode/source/PinUnlock.js
+++ b/usr/palm/applications/com.palm.app.phone/phoneAppPinCode/source/PinUnlock.js
@@ -40,6 +40,7 @@ enyo.kind({
 			cancelButtonCaption: null,
 			modal: true
 		},
+		{name: "hapticFeedback", kind: enyo.PalmService, service: "palm://com.palm.vibrate/", method: "vibrate", onSuccess: "", onFailure:""}
 	],
 	
 	create: function(){
@@ -104,6 +105,7 @@ enyo.kind({
 				this.pin = this.pin + value;
 			}
 			this.updatePinDisplay();
+			 
 		}
 	},	
 	handleClear: function() {
@@ -118,7 +120,11 @@ enyo.kind({
 	}, 		
 	updatePinDisplay: function() {
 		var maskString = this.pin.replace(/./g,'.');
-		this.$.pinDisplay.setContent(maskString);	
+		this.$.pinDisplay.setContent(maskString);
+		this.$.hapticFeedback.call({
+						"period": 0,
+						"duration": 75
+								});//vibrate		
 	},
 	updateRetriesDisplay: function(){
 		var t = new enyo.g11n.Template($L("1#One try remaining|##{tries} tries remaining")); 
