diff --git a/usr/palm/applications/com.palm.app.screenlock/app/controllers/securityconfig-assistant.js b/usr/palm/applications/com.palm.app.screenlock/app/controllers/securityconfig-assistant.js
index f88bdca..069252d 100644
--- a/usr/palm/applications/com.palm.app.screenlock/app/controllers/securityconfig-assistant.js
+++ b/usr/palm/applications/com.palm.app.screenlock/app/controllers/securityconfig-assistant.js
@@ -148,7 +148,7 @@ var SecurityconfigAssistant = Class.create({
 	timerModel: {currentTimer: 30},
 	
 	//Available Timer Values.
-	availableTimers: [{label: $L('30 seconds'), value:30},{label: $L('1 minute'), value:60},{label: $L('2 minutes'), value:120},{label: $L('3 minutes'), value:180}],
+	availableTimers: [{label: $L('15 seconds'), value: 15}, {label: $L('30 seconds'), value:30},{label: $L('1 minute'), value:60},{label: $L('2 minutes'), value:120},{label: $L('3 minutes'), value:180},{label: $L('5 minutes'), value:300},{label: $L('Never'), value:10800}],
 	
 	//Model for Security Selection List
 	securityModel: {currentSecurity: 'none'},
