diff --git a/usr/palm/applications/com.palm.app.screenlock/app/controllers/securityconfig-assistant.js b/usr/palm/applications/com.palm.app.screenlock/app/controllers/securityconfig-assistant.js
index f88bdca..045c06d 100644
--- a/usr/palm/applications/com.palm.app.screenlock/app/controllers/securityconfig-assistant.js
+++ b/usr/palm/applications/com.palm.app.screenlock/app/controllers/securityconfig-assistant.js
@@ -38,7 +38,7 @@ var SecurityconfigAssistant = Class.create({
 	sliderattributes: {
 			modelProperty: 'value',
 			maxValue: 100,
-			minValue: 10,
+			minValue: 5,
 			round: false,
 			updateInterval: 0.1
 	},	
