diff --git a/usr/palm/applications/com.palm.app.phone/phoneAppPinCode/source/PhonePinCard.js b/usr/palm/applications/com.palm.app.phone/phoneAppPinCode/source/PhonePinCard.js
index 5a31b51..499edfb 100644
--- a/usr/palm/applications/com.palm.app.phone/phoneAppPinCode/source/PhonePinCard.js
+++ b/usr/palm/applications/com.palm.app.phone/phoneAppPinCode/source/PhonePinCard.js
@@ -63,7 +63,7 @@ enyo.kind({
 		// DEFAULT: just handle known launch mode
 		} else {
 			//always unlock - for testing
-			//this.$.updatePinAppState.call({state: 'unlock'}); return;
+			this.$.updatePinAppState.call({state: 'unlock'}); return;
 			
 			this.handleLaunchLockMode();
 		}
