--- .orig/usr/palm/applications/com.palm.app.phone/app/controllers/app-assistant.js
+++ /usr/palm/applications/com.palm.app.phone/app/controllers/app-assistant.js
@@ -996,6 +996,7 @@
 // show dialpad for emergency call and screenlock
 AppAssistant.prototype.showLimitedDialpad = function(sc, dialParams) {
 	dialParams.limited = true;
+	dialParams.number = this.emergencyNumbers;
 	sc.pushScene({
 		"name": "dialpad",
 		automaticFocusAdvance: false,
