diff --git a/usr/palm/applications/com.palm.app.screenlock/source/ScreenLockPrefApp.js b/usr/palm/applications/com.palm.app.screenlock/source/ScreenLockPrefApp.js
index 12efc9e..d501736 100644
--- a/usr/palm/applications/com.palm.app.screenlock/source/ScreenLockPrefApp.js
+++ b/usr/palm/applications/com.palm.app.screenlock/source/ScreenLockPrefApp.js
@@ -23,7 +23,7 @@ enyo.kind({
 					]},
 					{kind: "Item", layoutKind:"VFlexLayout", tapHighlight: false, components: [
 					                                                   						{content: $L("Brightness")},
-					                                                   						{kind: "Slider", name:"brightnessSlider", position: 0, minimum:10, maximum:100, onChange:"brightnessChange"}
+					                                                   						{kind: "Slider", name:"brightnessSlider", position: 0, minimum:1, maximum:100, onChange:"brightnessChange"}
 					                                                   					]},
 					{kind: "Item", layoutKind:"HFlexLayout", tapHighlight: false, 
 					        components: [
