diff --git a/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js b/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
index 2888b7c..3a4abb0 100644
--- a/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
+++ b/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
@@ -1527,6 +1527,8 @@ handlePowerNotifications: function(payload) {
 				this.isCharging = true;			
 		}.bind(this));
 						
+		this.setForAutoOff();
+						
 		if (this.isCharging) {
 			Mojo.Log.info("SystemUI- Charger Notification "+ this.isCharging + " Last battery level "+ this.lastBatteryLevel);
 			if(this.batteryLevel == 100)
@@ -1559,6 +1561,66 @@ handlePowerNotifications: function(payload) {
 	}
 },
 
+setForAutoOff: function() {
+	if(this.powerSource['inductive']==true) {
+		this.controller.serviceRequest('palm://com.palm.display/control', {
+			method: 'getProperty',
+			parameters: {
+				properties: ["onWhenConnected"]
+			},
+			onSuccess: function(payload) {
+				if(payload.onWhenConnected) {
+					this.getAutoOffTimeout(
+						function(response) {
+							if(response.timeout!=604800) {
+								this.setAutoOffCookie(response.timeout);
+								this.setAutoOffTimeout(604800);
+							}
+						}.bind(this)
+					);
+				}
+			}.bind(this)
+		});
+	} else {
+		var timeout = this.getAutoOffCookie();
+		if(timeout.seconds != 604800) {
+			this.setAutoOffTimeout(timeout.seconds);
+			this.setAutoOffCookie(604800);
+		}
+	}
+},
+
+getAutoOffCookie: function() {
+	var cookie = new Mojo.Model.Cookie("auto-off");
+	var val = cookie.get();
+	if(!val) {
+		val = {seconds:604800};
+		cookie.put(val);
+	}
+	return val;
+},
+
+setAutoOffCookie: function(value) {
+	var cookie = new Mojo.Model.Cookie("auto-off");
+	cookie.put({seconds:value});
+},
+
+getAutoOffTimeout: function(callback) {
+	this.controller.serviceRequest('palm://com.palm.display/control', {
+		method: 'getProperty',
+		parameters:{properties:['timeout']},
+		onSuccess: callback
+	});
+},
+
+setAutoOffTimeout: function(value) {
+	this.controller.serviceRequest('palm://com.palm.display/control', {
+		method: 'setProperty',
+		parameters:{timeout:value}
+	});
+
+},
+
 handleCriticalPowerNotifications: function(payload) {
 	if(payload.cause) {
 		Mojo.Log.error("SystemUI - Battery Level is critical. Shutting down the phone ");
diff --git a/usr/palm/applications/com.palm.app.screenlock/app/controllers/securityconfig-assistant.js b/usr/palm/applications/com.palm.app.screenlock/app/controllers/securityconfig-assistant.js
index 86eb675..1e1c0a3 100644
--- a/usr/palm/applications/com.palm.app.screenlock/app/controllers/securityconfig-assistant.js
+++ b/usr/palm/applications/com.palm.app.screenlock/app/controllers/securityconfig-assistant.js
@@ -19,6 +19,10 @@ var SecurityconfigAssistant = Class.create({
         enableProp: 'enabled',        
     },
     
+    autoOffChargingtoggleModel: {
+        value: false
+    },
+    
     switchApptoggleModel: {
         value: false,
     },
@@ -53,6 +57,9 @@ var SecurityconfigAssistant = Class.create({
 		//this.controller.setupWidget('blinkAlerts', this.onOffToggleOpt, this.ledThrobberToggleModel);
 		//Mojo.Event.listen($('blinkAlerts'),'mojo-property-change', this.toggleLEDThrobber.bindAsEventListener(this));
 		
+		this.controller.setupWidget('autoOffCharging', this.onOffToggleOpt, this.autoOffChargingtoggleModel);
+		Mojo.Event.listen($('autoOffCharging'),'mojo-property-change', this.toggleautoOffCharging.bindAsEventListener(this));
+		
 		this.controller.setupWidget('switchApps', this.onOffToggleOpt, this.switchApptoggleModel);
 		Mojo.Event.listen($('switchApps'),'mojo-property-change', this.toggleSwitchApps.bindAsEventListener(this));
 		
@@ -236,6 +243,7 @@ var SecurityconfigAssistant = Class.create({
 		
 		this.getSystemlockModeReq = SystemService.getSystemlockMode(this.handleSystemlockMode.bind(this));
 		this.getSystemlockTimeoutReq = SystemService.getSystemlockTimeout(this.handleSystemlockTimeout.bind(this));
+		this.getAutoOffCharging();
 	},
 	
 	updateUI: function(timeout,maximumBrightness) {
@@ -551,6 +559,33 @@ var SecurityconfigAssistant = Class.create({
 		}
 	},
 	
+	toggleautoOffCharging: function(event) {
+		if(!event)
+			return;
+		this.setAutoOffCharging();
+	},
+	
+	setAutoOffCharging: function(value) {
+		this.controller.serviceRequest('palm://com.palm.display/control', {
+			method: 'setProperty',
+			parameters: {
+				onWhenConnected: !this.autoOffChargingtoggleModel.value
+			}
+		});	
+	},
+	
+	getAutoOffCharging: function() {
+		this.controller.serviceRequest('palm://com.palm.display/control', {
+			method: 'getProperty',
+			parameters: {
+				properties: ["onWhenConnected"]
+			},
+			onSuccess: function(payload) {
+				this.autoOffChargingtoggleModel.value = !payload.onWhenConnected;
+				this.controller.modelChanged(this.autoOffChargingtoggleModel);
+			}.bind(this)
+		});		
+	},
 	
 	
 	launchFilePicker: function(event) {
diff --git a/usr/palm/applications/com.palm.app.screenlock/app/views/securityconfig/securityconfig-scene.html b/usr/palm/applications/com.palm.app.screenlock/app/views/securityconfig/securityconfig-scene.html
index a97b61b..ab92777 100644
--- a/usr/palm/applications/com.palm.app.screenlock/app/views/securityconfig/securityconfig-scene.html
+++ b/usr/palm/applications/com.palm.app.screenlock/app/views/securityconfig/securityconfig-scene.html
@@ -19,11 +19,17 @@
 				</div>
 			</div>
 
-			<div id='screentimerrow' class="palm-row last">
+			<div id='screentimerrow' class="palm-row">
 				<div id='timeroptions' class="palm-row-wrapper">
 					<div id="screenOffTimer" x-mojo-element="ListSelector" class="right"></div>
 				</div>
 			</div>
+			<div class="palm-row last">
+				<div class="palm-row-wrapper">
+					<div x-mojo-element="ToggleButton" id="autoOffCharging"></div>
+					<div class="title capitalize" style="font-size:0.85em" x-mojo-loc=''>Auto-off while charging</div>
+				</div>
+			</div>
 		</div>
 	 </div>
 	 <div class="palm-group">
