diff --git a/usr/palm/applications/com.palm.app.email/app/controllers/email-dashboard-assistant.js b/usr/palm/applications/com.palm.app.email/app/controllers/email-dashboard-assistant.js
index 479f88e..93e6a20 100644
--- a/usr/palm/applications/com.palm.app.email/app/controllers/email-dashboard-assistant.js
+++ b/usr/palm/applications/com.palm.app.email/app/controllers/email-dashboard-assistant.js
@@ -74,9 +74,6 @@ var EmailDashboardAssistant = Class.create({
 		var content;
 		var latestDetails;
 
-		// turn on throbber
-		Mojo.Log.info('Turning on throbber for new email notification');
-		this.controller.stageController.indicateNewContent(true);
 		
 		// update notifications if they are changed.
 		this.notifications = notificationInfo;
