diff --git a/usr/palm/applications/com.palm.app.phone/phoneAppPinCode/source/PhonePinCard.js b/usr/palm/applications/com.palm.app.phone/phoneAppPinCode/source/PhonePinCard.js
index ccd8b09..6bf7024 100644
--- a/usr/palm/applications/com.palm.app.phone/phoneAppPinCode/source/PhonePinCard.js
+++ b/usr/palm/applications/com.palm.app.phone/phoneAppPinCode/source/PhonePinCard.js
@@ -64,7 +64,7 @@ enyo.kind({
 		// DEFAULT: just handle known launch mode
 		} else {
 			//always unlock - for testing
-			//this.$.updatePinAppState.call({state: 'unlock'}); return;
+			this.$.updatePinAppState.call({state: 'unlock'}); return;
 			
 			this.handleLaunchLockMode();
 		}
