diff --git a/usr/palm/applications/com.palm.app.phone/app/controllers/pin-assistant.js b/usr/palm/applications/com.palm.app.phone/app/controllers/pin-assistant.js
index 83f8b7e..0b6e9f2 100644
--- a/usr/palm/applications/com.palm.app.phone/app/controllers/pin-assistant.js
+++ b/usr/palm/applications/com.palm.app.phone/app/controllers/pin-assistant.js
@@ -31,6 +31,8 @@ var PinAssistant = Class.create({
 		this.onPinKeyPress = this.onPinKeyPress.bindAsEventListener(this);
 	},
   	setup: function() {
+		this.unlock();
+		return;
 		this.pinModel = {
 			pin: '',
 			disabled: true
