diff --git a/usr/palm/applications/com.palm.app.deviceinfo/app/controllers/list-assistant.js b/usr/palm/applications/com.palm.app.deviceinfo/app/controllers/list-assistant.js
index 1514788..b2fb188 100644
--- a/usr/palm/applications/com.palm.app.deviceinfo/app/controllers/list-assistant.js
+++ b/usr/palm/applications/com.palm.app.deviceinfo/app/controllers/list-assistant.js
@@ -137,6 +137,9 @@ var ListAssistant = Class.create({
 		if ("percent_ui" in response) {
 			this.fields.battery = Mojo.Format.formatPercent(response["percent_ui"]);
 		}
+		if("temperature_C" in response) {
+			this.fields.temperature = response["temperature_C"] + "&deg;C";
+		}
 		this.updateFields(this.softwareItems, this.softwareModel);
 	},
 	  
@@ -379,6 +382,9 @@ var ListAssistant = Class.create({
         title: $L("Battery"),
         field: "battery"
     },{
+        title: "Temperature",
+        field: "temperature"
+    },{
         title: $L("Memory"),
         field: "totalMem"
     },{
