diff --git a/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js b/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
index 0e239cc..9e222a4 100644
--- a/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
+++ b/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
@@ -4022,6 +4022,10 @@ showTimezoneErrorAlert: function(showTimezoneAlert, launchParameter) {
 },
 
 appMgrInit: function() {
+	this.clockDockModeRequest = new Mojo.Service.Request('palm://com.palm.applicationManager', {
+		method: 'addDockModeLaunchPoint',
+		parameters: {appId:"com.palm.app.clock"}
+	});
 	//Get all launch points
 	this.getAppListServiceRequest = new Mojo.Service.Request('luna://com.palm.applicationManager', {
 	        method: 'listApps',
