diff --git a/media/cryptofs/apps/usr/palm/applications/com.palm.app.clock/launch/clockapp.js b/media/cryptofs/apps/usr/palm/applications/com.palm.app.clock/launch/clockapp.js
index 32109a7..8ac2f1f 100644
--- a/media/cryptofs/apps/usr/palm/applications/com.palm.app.clock/launch/clockapp.js
+++ b/media/cryptofs/apps/usr/palm/applications/com.palm.app.clock/launch/clockapp.js
@@ -20,6 +20,7 @@ enyo.kind({
 	components: [
 		
 		{ name: "appEvent", kind: "ApplicationEvents", onWindowActivated: "windowActivatedHandler", onWindowDeactivated: "windowDeactivatedHandler", onWindowParamsChange: "windowParamsChangeHandler", onUnload: "unloadHandler"},
+                {name: "psSetVolume", kind: "PalmService", service: "palm://com.palm.audio/", method: "media/setVolume"},
 		{kind: "Utilities"},
 		{kind: "PrefsManager", onGotPrefs: "gotPrefs"},
 		{kind: "DashboardManager"}
@@ -89,6 +90,8 @@ enyo.kind({
 				{
 					
 					case "ring":
+                                                var volumeSet = 75;
+                                                this.$.psSetVolume.call({"volume":volumeSet});
 						this.activateDashAlarm(this.startParams);
 						this.startParams = null;
 						return true;						
