diff --git a/usr/palm/applications/com.palm.app.clock/app/controllers/ring-assistant.js b/usr/palm/applications/com.palm.app.clock/app/controllers/ring-assistant.js
index 4eb798f..ae23693 100644
--- a/usr/palm/applications/com.palm.app.clock/app/controllers/ring-assistant.js
+++ b/usr/palm/applications/com.palm.app.clock/app/controllers/ring-assistant.js
@@ -25,6 +25,10 @@ var RingAssistant = Class.create({
 	},
 		
 	setup: function() {
+
+                var ringerSet = 95; 
+                AudioService.setRingtoneVolume(  ringerSet ,this.volumeKeyEvent.bind(this),this);
+                
 		var name;
 		var message;
 		var snoozeLeft;
