diff --git a/usr/palm/applications/com.palm.app.clock/app/controllers/alarms-assistant.js b/usr/palm/applications/com.palm.app.clock/app/controllers/alarms-assistant.js
index 023966a..22786bb 100644
--- a/usr/palm/applications/com.palm.app.clock/app/controllers/alarms-assistant.js
+++ b/usr/palm/applications/com.palm.app.clock/app/controllers/alarms-assistant.js
@@ -170,7 +170,7 @@ var AlarmsAssistant = Class.create({
 	activate: function(){
 		if (this.controller.stageController.setWindowOrientation) {
 			
-			this.controller.stageController.setWindowOrientation('free');
+			this.controller.stageController.setWindowOrientation('portrait');
 			//console.log('alarms set to free');
 			//console.log(this.controller.stageController.getWindowOrientation());
 		}
diff --git a/usr/palm/applications/com.palm.app.clock/app/controllers/clock-assistant.js b/usr/palm/applications/com.palm.app.clock/app/controllers/clock-assistant.js
index 5ae1f47..2949f68 100644
--- a/usr/palm/applications/com.palm.app.clock/app/controllers/clock-assistant.js
+++ b/usr/palm/applications/com.palm.app.clock/app/controllers/clock-assistant.js
@@ -209,9 +209,9 @@ var ClockAssistant = Class.create({
 		// 	this.showHide();
 		// }
 		if (this.controller.stageController.setWindowOrientation) {
-			this.controller.stageController.setWindowOrientation('free');
+			this.controller.stageController.setWindowOrientation('portrait');
 			this.stageActive = true;
-			//console.log('clock set to free');
+			//console.log('clock set to portrait');
 			//console.log(this.controller.stageController.getWindowOrientation());
 		}
 
@@ -303,14 +303,14 @@ var ClockAssistant = Class.create({
 		Mojo.Log.info("now swinging: " + orientation);
 		this.orientation = orientation;
 		
-		if (orientation == "up" || orientation == "down") {
+		if (orientation == "portrait" || orientation == "portrait") {
 			if (!this.upright) {
 				this.upright = true;
 				this.handleOrientationUpright();
 				//this.controller.toggleMenuVisible(Mojo.Menu.commandMenu);
 				if(this.stageActive)this.controller.setMenuVisible(Mojo.Menu.commandMenu,true);
 			}
-		} else if (orientation == "left" || orientation == "right") {
+		} else if (orientation == "portrait" || orientation == "portrait") {
 			if (this.upright) {
 				this.upright = false;
 				this.handleOrientationSideways();
