Index: /usr/palm/applications/com.palm.app.screenlock/app/controllers/securityconfig-assistant.js
===================================================================
--- .orig/usr/palm/applications/com.palm.app.screenlock/app/controllers/securityconfig-assistant.js
+++ /usr/palm/applications/com.palm.app.screenlock/app/controllers/securityconfig-assistant.js
@@ -35,7 +35,7 @@
 	sliderattributes: {
 			modelProperty: 'value',
 			maxValue: 100,
-			minValue: 10,
+			minValue: 0,
 			round: false,
 			updateInterval: 0.1
 	},	
