diff --git a/usr/palm/applications/com.palm.app.contacts/app/models/phonenumber-model.js b/usr/palm/applications/com.palm.app.contacts/app/models/phonenumber-model.js
index 68dd4c9..da8ff77 100644
--- a/usr/palm/applications/com.palm.app.contacts/app/models/phonenumber-model.js
+++ b/usr/palm/applications/com.palm.app.contacts/app/models/phonenumber-model.js
@@ -25,7 +25,10 @@ PhoneNumber.strip = function(numberStr){
     if (typeof numberStr === 'number') {
         numberStr = "" + numberStr;
     }
-    return numberStr.replace(/[^A-Za-z0-9+*#]+/g, "");
+//========================begin======patch 1 of 1, display phone number as entered
+	return numberStr; 
+	// original     return numberStr.replace(/[^A-Za-z0-9+*#]+/g, "");
+//=======================end
 };
 
 // US phone number formatter
