diff --git a/usr/palm/applications/com.palm.app.contacts/app/models/phonenumber-model.js b/usr/palm/applications/com.palm.app.contacts/app/models/phonenumber-model.js
index 68dd4c9..e9a3bcd 100644
--- a/usr/palm/applications/com.palm.app.contacts/app/models/phonenumber-model.js
+++ b/usr/palm/applications/com.palm.app.contacts/app/models/phonenumber-model.js
@@ -52,6 +52,13 @@ PhoneNumber.MOBILE = 3;
 PhoneNumber.PAGER = 4;
 PhoneNumber.PERSONAL_FAX = 5;
 PhoneNumber.FAX = 6;
+PhoneNumber.WORK_1 = 7;
+PhoneNumber.HOME_2 = 8;
+PhoneNumber.MOBILE_2 = 9;
+PhoneNumber.OFFICE = 10;
+PhoneNumber.OFFICE_2 = 11;
+PhoneNumber.FAX_2 = 12;
+PhoneNumber.DIRECT = 13;
 
 PhoneNumber.popupLabels = [{
     value: 3,
@@ -73,6 +80,22 @@ PhoneNumber.popupLabels = [{
     value: 2,
     label: $L("Other"),
     command: 2
+}, {
+    value: 10,
+    label: $L("Office"),
+    command: 10
+}, {
+    value: 9,
+    label: $L("Mobile 2"),
+    command: 9
+}, {
+    value: 13,
+    label: $L("Direct"),
+    command: 13
+}, {
+    value: 12,
+    label: $L("Fax 2"),
+    command: 12
 }];
 
 PhoneNumber.labels = {
@@ -81,7 +104,14 @@ PhoneNumber.labels = {
     1: $L('Work'),
     7: $L('Main'),
     6: $L('Fax'),
-    5: $L('Fax'),
+    5: $L('Per Fax'),
     4: $L('Pager'),
-    2: $L('Other')
+    2: $L('Other'),
+    8: $L('Pager'),
+    9: $L('Mobile 2'),
+   10: $L('Office'),
+   11: $L('Office 2'),
+   12: $L('Fax 2'),	
+   12: $L('Direct')
+
 };
