diff --git a/usr/palm/frameworks/contacts/submission/111/concatenated.js b/usr/palm/frameworks/contacts/submission/111/concatenated.js
index 07dff89..0a71f66 100755
--- a/usr/palm/frameworks/contacts/submission/111/concatenated.js
+++ b/usr/palm/frameworks/contacts/submission/111/concatenated.js
@@ -5259,7 +5259,7 @@ PhoneNumber.Labels = Utils.createLabelFunctions([{
 	value: PhoneNumber.TYPE.COMPANY,
 	displayValue: RB.$L('Company'),
 	shortDisplayValue: RB.$L('C'),
-	isPopupLabel: false
+	isPopupLabel: true
 }, {
 	value: PhoneNumber.TYPE.WORK2,
 	displayValue: RB.$L('Work 2'),
diff --git a/usr/palm/frameworks/contacts/submission/111/javascript/properties/PhoneNumber.js b/usr/palm/frameworks/contacts/submission/111/javascript/properties/PhoneNumber.js
index bb60b8a..98a3d80 100644
--- a/usr/palm/frameworks/contacts/submission/111/javascript/properties/PhoneNumber.js
+++ b/usr/palm/frameworks/contacts/submission/111/javascript/properties/PhoneNumber.js
@@ -284,7 +284,7 @@ PhoneNumber.Labels = Utils.createLabelFunctions([{
 	value: PhoneNumber.TYPE.PERSONAL,
 	displayValue: RB.$L('Personal'),
 	shortDisplayValue: RB.$L('Pe'),
-	isPopupLabel: false
+	isPopupLabel: true
 }, {
 	value: PhoneNumber.TYPE.SIM,
 	displayValue: RB.$L('SIM'),
@@ -294,12 +294,12 @@ PhoneNumber.Labels = Utils.createLabelFunctions([{
 	value: PhoneNumber.TYPE.MAIN,
 	displayValue: RB.$L('Main'),
 	shortDisplayValue: RB.$L('Ma'),
-	isPopupLabel: false
+	isPopupLabel: true
 }, {
 	value: PhoneNumber.TYPE.PERSONAL_FAX,
 	displayValue: RB.$L('Fax'),
 	shortDisplayValue: RB.$L('P'),
-	isPopupLabel: false
+	isPopupLabel: true
 }, {
 	value: PhoneNumber.TYPE.ASSISTANT,
 	displayValue: RB.$L('Assistant'),
@@ -319,7 +319,7 @@ PhoneNumber.Labels = Utils.createLabelFunctions([{
 	value: PhoneNumber.TYPE.COMPANY,
 	displayValue: RB.$L('Company'),
 	shortDisplayValue: RB.$L('C'),
-	isPopupLabel: false
+	isPopupLabel: true
 }, {
 	value: PhoneNumber.TYPE.WORK2,
 	displayValue: RB.$L('Work 2'),
diff --git a/usr/palm/frameworks/mojo/builtins/palmcontactsVersion1_0.js b/usr/palm/frameworks/mojo/builtins/palmcontactsVersion1_0.js
index bcaeb79..cab1ec5 100644
--- a/usr/palm/frameworks/mojo/builtins/palmcontactsVersion1_0.js
+++ b/usr/palm/frameworks/mojo/builtins/palmcontactsVersion1_0.js
@@ -217,8 +217,8 @@ PhoneNumber.strip=function(numberStr){if(!numberStr){console.warn("PhoneNumber.s
 PhoneNumber.unformatForVCard=function(numberStr){var toReturn="";numberStr.replace(/[0123456789\+\*wpt#]/gi,function(substr){toReturn+=substr});return toReturn};PhoneNumber.getDisplayType=function(type){return PhoneNumber.Labels.getLabel(type)||PhoneNumber.Labels.getLabel(PhoneNumber.TYPE.MOBILE)};PhoneNumber.PART_DELIMITER="-";
 PhoneNumber.TYPE=Utils.defineConstants({MOBILE:"type_mobile",HOME:"type_home",HOME2:"type_home2",WORK:"type_work",WORK2:"type_work2",MAIN:"type_main",PERSONAL_FAX:"type_personal_fax",WORK_FAX:"type_work_fax",PAGER:"type_pager",PERSONAL:"type_personal",SIM:"type_sim",ASSISTANT:"type_assistant",CAR:"type_car",RADIO:"type_radio",COMPANY:"type_company",OTHER:"type_other"});
 PhoneNumber.Labels=Utils.createLabelFunctions([{value:PhoneNumber.TYPE.MOBILE,displayValue:RB.$L("Mobile"),shortDisplayValue:RB.$L("M"),isPopupLabel:true},{value:PhoneNumber.TYPE.HOME,displayValue:RB.$L("Home"),shortDisplayValue:RB.$L("H"),isPopupLabel:true},{value:PhoneNumber.TYPE.WORK,displayValue:RB.$L("Work"),shortDisplayValue:RB.$L("W"),isPopupLabel:true},{value:PhoneNumber.TYPE.WORK_FAX,displayValue:RB.$L("Fax"),shortDisplayValue:RB.$L("F"),isPopupLabel:true},{value:PhoneNumber.TYPE.OTHER,displayValue:RB.$L("Other"),
-shortDisplayValue:RB.$L("O"),isPopupLabel:true},{value:PhoneNumber.TYPE.PAGER,displayValue:RB.$L("Pager"),shortDisplayValue:RB.$L("P"),isPopupLabel:false},{value:PhoneNumber.TYPE.PERSONAL,displayValue:RB.$L("Personal"),shortDisplayValue:RB.$L("Pe"),isPopupLabel:false},{value:PhoneNumber.TYPE.SIM,displayValue:RB.$L("SIM"),shortDisplayValue:RB.$L("S"),isPopupLabel:false},{value:PhoneNumber.TYPE.MAIN,displayValue:RB.$L("Main"),shortDisplayValue:RB.$L("Ma"),isPopupLabel:false},{value:PhoneNumber.TYPE.PERSONAL_FAX,
-displayValue:RB.$L("Fax"),shortDisplayValue:RB.$L("P"),isPopupLabel:false},{value:PhoneNumber.TYPE.ASSISTANT,displayValue:RB.$L("Assistant"),shortDisplayValue:RB.$L("A"),isPopupLabel:false},{value:PhoneNumber.TYPE.CAR,displayValue:RB.$L("Car"),shortDisplayValue:RB.$L("Ca"),isPopupLabel:false},{value:PhoneNumber.TYPE.RADIO,displayValue:RB.$L("Radio"),shortDisplayValue:RB.$L("R"),isPopupLabel:false},{value:PhoneNumber.TYPE.COMPANY,displayValue:RB.$L("Company"),shortDisplayValue:RB.$L("C"),isPopupLabel:false},
+shortDisplayValue:RB.$L("O"),isPopupLabel:true},{value:PhoneNumber.TYPE.PAGER,displayValue:RB.$L("Pager"),shortDisplayValue:RB.$L("P"),isPopupLabel:false},{value:PhoneNumber.TYPE.PERSONAL,displayValue:RB.$L("Personal"),shortDisplayValue:RB.$L("Pe"),isPopupLabel:false},{value:PhoneNumber.TYPE.SIM,displayValue:RB.$L("SIM"),shortDisplayValue:RB.$L("S"),isPopupLabel:false},{value:PhoneNumber.TYPE.MAIN,displayValue:RB.$L("Main"),shortDisplayValue:RB.$L("Ma"),isPopupLabel:true},{value:PhoneNumber.TYPE.PERSONAL_FAX,
+displayValue:RB.$L("Fax"),shortDisplayValue:RB.$L("P"),isPopupLabel:true},{value:PhoneNumber.TYPE.ASSISTANT,displayValue:RB.$L("Assistant"),shortDisplayValue:RB.$L("A"),isPopupLabel:false},{value:PhoneNumber.TYPE.CAR,displayValue:RB.$L("Car"),shortDisplayValue:RB.$L("Ca"),isPopupLabel:false},{value:PhoneNumber.TYPE.RADIO,displayValue:RB.$L("Radio"),shortDisplayValue:RB.$L("R"),isPopupLabel:false},{value:PhoneNumber.TYPE.COMPANY,displayValue:RB.$L("Company"),shortDisplayValue:RB.$L("C"),isPopupLabel:true},
 {value:PhoneNumber.TYPE.WORK2,displayValue:RB.$L("Work 2"),shortDisplayValue:RB.$L("W2"),isPopupLabel:false},{value:PhoneNumber.TYPE.HOME2,displayValue:RB.$L("Home 2"),shortDisplayValue:RB.$L("H2"),isPopupLabel:false}]);var FavoritablePhoneNumber=PropertyBase.create({superClass:PhoneNumber,data:FavoritablePersonField.data});FavoritablePhoneNumber.prototype.addFavoriteData=FavoritablePersonField.functions.addFavoriteData;FavoritablePhoneNumber.prototype.hasFavoriteDataForAnyApp=FavoritablePersonField.functions.hasFavoriteDataForAnyApp;FavoritablePhoneNumber.prototype.getFavoriteDataForAppWithId=FavoritablePersonField.functions.getFavoriteDataForAppWithId;
 FavoritablePhoneNumber.prototype.removeFavoriteDefaultForAppWithId=FavoritablePersonField.functions.removeFavoriteDefaultForAppWithId;FavoritablePhoneNumber.prototype.removeAllFavoriteData=FavoritablePersonField.functions.removeAllFavoriteData;var PersonPhotos=exports.PersonPhotos=PropertyBase.create({data:[{dbFieldName:"bigPhotoPath",defaultValue:"",setterName:"setBigPhotoPath",getterName:"getBigPhotoPath"},{dbFieldName:"squarePhotoPath",defaultValue:"",setterName:"setSquarePhotoPath",getterName:"getSquarePhotoPath"},{dbFieldName:"listPhotoPath",defaultValue:"",setterName:"setListPhotoPath",getterName:"getListPhotoPath"},{dbFieldName:"listPhotoSource",defaultValue:"",setterName:"setListPhotoSource",getterName:"getListPhotoSource"},{dbFieldName:"bigPhotoId",
 defaultValue:"",setterName:"setBigPhotoId",getterName:"getBigPhotoId"},{dbFieldName:"squarePhotoId",defaultValue:"",setterName:"setSquarePhotoId",getterName:"getSquarePhotoId"},{dbFieldName:"contactId",defaultValue:"",setterName:"setContactId",getterName:"getContactId"},{dbFieldName:"accountId",defaultValue:"",setterName:"setAccountId",getterName:"getAccountId"}]});
