diff --git a/usr/palm/applications/com.palm.app.contacts/app/controllers/speeddial-assistant.js b/usr/palm/applications/com.palm.app.contacts/app/controllers/speeddial-assistant.js
index 9700535..d55396b 100644
--- a/usr/palm/applications/com.palm.app.contacts/app/controllers/speeddial-assistant.js
+++ b/usr/palm/applications/com.palm.app.contacts/app/controllers/speeddial-assistant.js
@@ -54,6 +54,23 @@ var SpeeddialAssistant = Class.create({
             letter: 'V',
 			symbol: '9',
             hasNumber: 'has-number'
+        }, {                                                                                                              
+            number: 0,                                                                                                    
+            letter: '@',                                                                                                  
+            symbol: '0',                                                                                      
+            hasNumber: 'has-number'                                                                                       
+        }, {                                                                                                              
+            letter: 'Z',                                                                                                  
+                        symbol: '*'                       
+        }, {                                                                                                              
+            letter: 'B',                                                                                                  
+                        symbol: '#'                     
+        }, {                                                                                                              
+            letter: '.',                                                                                                  
+                        symbol: '.'                                                                                       
+        }, {                                                                                                              
+            letter: ',',                                                                                                  
+                        symbol: '_'                       
         }, {
             letter: 'A',
 			symbol: '&'
@@ -173,7 +190,7 @@ var SpeeddialAssistant = Class.create({
 		
 		var ch = String.fromCharCode(evt.keyCode).capitalize();
 		
-		if ("abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ".indexOf(ch) < 0) {
+		if ("abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ,.@".indexOf(ch) < 0) {
 			for (var i = 0; i < this.buttons.size(); i++) {
 				var thisButton = this.buttons[i];
 				if (thisButton.symbol == ch) {
@@ -190,7 +207,7 @@ var SpeeddialAssistant = Class.create({
 	
 	isValidSpeedDialChar: function(keyCode) {
 		var s = String.fromCharCode(keyCode);
-		var valid="acdfghijklmnopqrstuvwxyACDFGHIJKLMNOPQRSTUVWXY23456789/+()%\"=&-$!:'*#?;";
+		var valid="acdfghijklmnopqrstuvwxyACDFGHIJKLMNOPQRSTUVWXY234567890.,_/+()%\"=&-$!:'*#?;@"; 
 		
 		if( valid.indexOf(s) >= 0 ) {
 			return s;
