diff --git a/usr/palm/services/com.palm.service.contacts.google/javascript/google_contact_handler.js b/usr/palm/services/com.palm.service.contacts.google/javascript/google_contact_handler.js
index 25bdb09..775746c 100644
--- a/usr/palm/services/com.palm.service.contacts.google/javascript/google_contact_handler.js
+++ b/usr/palm/services/com.palm.service.contacts.google/javascript/google_contact_handler.js
@@ -337,7 +337,26 @@ GoogleContactHandler.copyFieldsToLocal = function (remoteContact, localContact)
 	return localContact;
 };
 
+String.prototype.rtrim = function () {
+	return this.replace(/[ \t\r\n]+$/g, "");
+}
+GoogleContactHandler._ltrim = function (object) {
+	for (var name in object) {
+		var value = object[name];
+		switch (typeof value) {
+		case 'string':
+			object[name] = value.rtrim();
+			break;
+
+		case 'object':
+			this._ltrim(object[name]);
+			break;
+		}
+	}
+}
+
 GoogleContactHandler.copyFieldsToRemote = function (localContact, remoteContact) {
+	this._ltrim(localContact);
 	GoogleContactHandler._copyNameToRemote(localContact, remoteContact);
 	
 	GoogleContactHandler._copyNicknameToRemote(localContact, remoteContact);
diff --git a/usr/palm/services/com.palm.service.contacts.google/javascript/utils/xhr.js b/usr/palm/services/com.palm.service.contacts.google/javascript/utils/xhr.js
index bbd7562..9106809 100644
--- a/usr/palm/services/com.palm.service.contacts.google/javascript/utils/xhr.js
+++ b/usr/palm/services/com.palm.service.contacts.google/javascript/utils/xhr.js
@@ -41,6 +41,9 @@ var XMLHttpRequest = Class.create({
 	
 	setRequestHeader: function (name, value) {
 		this._consoleLog("\n\n\n--->>> new request header. name: ", name, " value: ", value, " \n\n");
+		if (name == "Content-Length") {
+			return;
+		}
 		this.options.headers[name] = value;
 	},
 	
