diff --git a/usr/lib/luna/system/luna-applauncher/app/controllers/global-search-assistant.js b/usr/lib/luna/system/luna-applauncher/app/controllers/global-search-assistant.js
index b9db2c2..e599d0d 100644
--- a/usr/lib/luna/system/luna-applauncher/app/controllers/global-search-assistant.js
+++ b/usr/lib/luna/system/luna-applauncher/app/controllers/global-search-assistant.js
@@ -71,15 +71,20 @@ GlobalSearchAssistant = Class.create({
 		this.imAccount={};
 		//Phone label and mail label should match with contact app
 		this.phoneLabel = {
-			'0': $L('Home'),
-			'1': $L('Work'),
-			'2': $L('Other'),
-			'3': $L('Mobile'),
-			'4': $L('Pager'),
-			'5': $L('Fax'),
-			'6': $L('Fax'),
-			'7': $L('Main'),
-			'8': $L('SIM')
+		    3: $L('Mobile'),
+		    0: $L('Home'),
+  		    1: $L('Work'),
+    		    7: $L('Main'),
+		    6: $L('Fax'),
+		    5: $L('Per Fax'),
+		    4: $L('Pager'),
+		    2: $L('Other'),
+		    8: $L('Home 2'),
+		    9: $L('Google Voice'),
+		   10: $L('Office'),
+		   11: $L('Office 2'),
+		   12: $L('Fax 2'),	
+		   13: $L('Direct')
 		};
 		this.mailLabel = {
 			'0': $L("Home"), 
diff --git a/usr/palm/applications/com.palm.app.contacts/app/models/phonenumber-model.js b/usr/palm/applications/com.palm.app.contacts/app/models/phonenumber-model.js
index 68dd4c9..8eccbcc 100644
--- a/usr/palm/applications/com.palm.app.contacts/app/models/phonenumber-model.js
+++ b/usr/palm/applications/com.palm.app.contacts/app/models/phonenumber-model.js
@@ -52,6 +52,13 @@ PhoneNumber.MOBILE = 3;
 PhoneNumber.PAGER = 4;
 PhoneNumber.PERSONAL_FAX = 5;
 PhoneNumber.FAX = 6;
+PhoneNumber.WORK_1 = 7;
+PhoneNumber.HOME_2 = 8;
+PhoneNumber.GOOGLE_VOICE = 9;
+PhoneNumber.OFFICE = 10;
+PhoneNumber.OFFICE_2 = 11;
+PhoneNumber.FAX_2 = 12;
+PhoneNumber.DIRECT = 13;
 
 PhoneNumber.popupLabels = [{
     value: 3,
@@ -73,6 +80,26 @@ PhoneNumber.popupLabels = [{
     value: 2,
     label: $L("Other"),
     command: 2
+}, {
+    value: 10,
+    label: $L("Office"),
+    command: 10
+}, {
+    value: 9,
+    label: $L("Google Voice"),
+    command: 9
+}, {
+    value: 13,
+    label: $L("Direct"),
+    command: 13
+}, {
+    value: 12,
+    label: $L("Fax 2"),
+    command: 12
+}, {
+    value: 4,
+    label: $L("Pager"),
+    command: 4
 }];
 
 PhoneNumber.labels = {
@@ -81,7 +108,14 @@ PhoneNumber.labels = {
     1: $L('Work'),
     7: $L('Main'),
     6: $L('Fax'),
-    5: $L('Fax'),
+    5: $L('Per Fax'),
     4: $L('Pager'),
-    2: $L('Other')
+    2: $L('Other'),
+    8: $L('Home 2'),
+    9: $L('Google Voice'),
+   10: $L('Office'),
+   11: $L('Office 2'),
+   12: $L('Fax 2'),	
+   13: $L('Direct')
+
 };
