diff --git a/usr/palm/applications/com.palm.app.contacts/app/controllers/list-assistant.js b/usr/palm/applications/com.palm.app.contacts/app/controllers/list-assistant.js
index fbb9529..43f4f60 100644
--- a/usr/palm/applications/com.palm.app.contacts/app/controllers/list-assistant.js
+++ b/usr/palm/applications/com.palm.app.contacts/app/controllers/list-assistant.js
@@ -287,6 +287,8 @@ ListAssistant = Class.create(App.Scene,{
 				items: [Mojo.Menu.editItem,{
 					label:$L("Send All to Car Kit"),
 					command:"sendcontacts"
+				}, {label:$L("Sync Now"),
+					command:"syncnow"
 				}, prefsItem, {
 					label: $LL('Help'),
 					command: Mojo.Menu.helpCmd
@@ -466,6 +468,8 @@ ListAssistant = Class.create(App.Scene,{
 			this.controller.stageController.pushScene('prefs');
 		}else if (event.type == Mojo.Event.command && event.command == "sendcontacts") {
 			this.sendContacts();
+		}else if (event.type == Mojo.Event.command && event.command == "syncnow") {
+			this.syncAccounts();
 		}
 		if (event.type == Mojo.Event.commandEnable && event.command == Mojo.Menu.prefsCmd) {
 			// funkay!  this is apparently how you declare you want to activate the prefs menu item
@@ -691,6 +695,14 @@ ListAssistant = Class.create(App.Scene,{
 					
 				}.bind(this)
 			});
+	},
+	
+	syncAccounts: function() {
+		AppAssistant.pimSyncService.syncAll(this.controller, this.easAccounts, this.showSyncing.bind(this));
+	},
+	
+	showSyncing: function() {
+		this.controller.showBanner($L("Syncing Accounts..."), {});
 	}
 	
 	
