diff --git a/usr/palm/services/com.palm.service.contacts.google/javascript/google_contact_handler.js b/usr/palm/services/com.palm.service.contacts.google/javascript/google_contact_handler.js
index 25bdb09..65d903c 100644
--- a/usr/palm/services/com.palm.service.contacts.google/javascript/google_contact_handler.js
+++ b/usr/palm/services/com.palm.service.contacts.google/javascript/google_contact_handler.js
@@ -332,7 +332,7 @@ GoogleContactHandler.copyFieldsToLocal = function (remoteContact, localContact)
 	
 	GoogleContactHandler._copyRelationsToLocal(localContact, remoteContact);
 	
-	GoogleContactHandler._copyPhotoToLocal(localContact, remoteContact);
+//	GoogleContactHandler._copyPhotoToLocal(localContact, remoteContact);
 		
 	return localContact;
 };
