diff --git a/usr/palm/applications/com.palm.app.contacts/app/controllers/speeddial-assistant.js b/usr/palm/applications/com.palm.app.contacts/app/controllers/speeddial-assistant.js
index bed2865..e4b35b6 100644
--- a/usr/palm/applications/com.palm.app.contacts/app/controllers/speeddial-assistant.js
+++ b/usr/palm/applications/com.palm.app.contacts/app/controllers/speeddial-assistant.js
@@ -139,9 +139,9 @@ SpeeddialAssistant.prototype.generateButtonsArray = function () {
 		curSymbol = keys.opt;
 		
 		//don't include the keys that map to * and #, since the phone app doesn't let the user type them
-		if (curSymbol === "*" || curSymbol === "#") {
-			continue;
-		}
+//		if (curSymbol === "*" || curSymbol === "#") {
+//			continue;
+//		}
 		
 		this.validLetters += keys.shift;
 		
@@ -166,6 +166,34 @@ SpeeddialAssistant.prototype.generateButtonsArray = function () {
 		buttonsArray.push(newButton);
 		this.buttonsHash[newButton.letter] = newButton;
 	}
+				
+		this.validLetters += "@";
+		newButton = {
+			letter: "@",
+			symbol: "0"
+		};
+			newButton.number = 0;
+			newButton.hasNumber = "has-number";
+		buttonsArray.push(newButton);
+		this.buttonsHash[newButton.letter] = newButton;
+		
+	
+		this.validLetters += ".";
+		newButton = {
+			letter: ".",
+			symbol: "."
+		};
+		buttonsArray.push(newButton);
+		this.buttonsHash[newButton.letter] = newButton;
+
+		
+		this.validLetters += ",";
+		newButton = {
+			letter: ",",
+			symbol: "_"
+		};
+		buttonsArray.push(newButton);
+		this.buttonsHash[newButton.letter] = newButton;
 	
 	//now sort the array based on 1) number and 2) letter
 	buttonsArray.sort(this.buttonCompare);
