Index: /usr/palm/applications/com.palm.app.phone/app/controllers/calllog-assistant.js
===================================================================
--- .orig/usr/palm/applications/com.palm.app.phone/app/controllers/calllog-assistant.js
+++ /usr/palm/applications/com.palm.app.phone/app/controllers/calllog-assistant.js
@@ -10,7 +10,7 @@
 		this.stageController = Mojo.Controller.getAppController().getStageController("PhoneApp");
 		this.telListener = this.appAssistant.telephonyEventListener;
 		// if it's true, then we hide call length
-		this.hideduration = this.appAssistant.hideCallLogCallDuration;
+		this.hideduration = false;
 		this.zeroOffset = new Date(0, 0, 0, 0, 0, 0, 0).getTime();
 		
 		if (contactId !== undefined && contactId !== "")
