diff --git a/usr/palm/applications/com.palm.app.phone/app/controllers/incomingcall-assistant.js b/usr/palm/applications/com.palm.app.phone/app/controllers/incomingcall-assistant.js
index b4b2a70..06f803e 100644
--- a/usr/palm/applications/com.palm.app.phone/app/controllers/incomingcall-assistant.js
+++ b/usr/palm/applications/com.palm.app.phone/app/controllers/incomingcall-assistant.js
@@ -373,7 +373,7 @@ IncomingcallAssistant = Class.create({
 		// and we were previously on the puck
 		if (response && response.type == "inductive") {
 			if (this.appAssistant.puckMode && this.exposed && this.puckConnected === true && response.connected === false) {
-				this.answerIfStillOffPuck.bind(this).delay(0.750);
+				//this.answerIfStillOffPuck.bind(this).delay(0.750);
 			}
 			this.puckConnected = response.connected;
 		}
