diff --git a/usr/palm/applications/com.palm.app.phone/shared/dialpad/Dialpad.js b/usr/palm/applications/com.palm.app.phone/shared/dialpad/Dialpad.js
index f632702..87d6fa7 100644
--- a/usr/palm/applications/com.palm.app.phone/shared/dialpad/Dialpad.js
+++ b/usr/palm/applications/com.palm.app.phone/shared/dialpad/Dialpad.js
@@ -36,7 +36,7 @@ enyo.kind({
 		]},
 		{name: "client", flex: 1, layoutKind: "VFlexLayout"},
 		
-		{name: "playDialTone", kind: enyo.PalmService, service: "palm://com.palm.audio/dtmf/", method: "playDTMF", onSuccess: "", onFailure:""},
+		{name: "hapticFeedback", kind: enyo.PalmService, service: "palm://com.palm.vibrate/", method: "vibrate", onSuccess: "", onFailure:""},
 		{name: "stopDialTone", kind: enyo.PalmService, service: "palm://com.palm.audio/dtmf/", method: "stopDTMF", onSuccess: "", onFailure:""}
 	],
 	create: function() {
@@ -206,11 +206,10 @@ enyo.kind({
 			// assumes the id has the key at position 5
 			//enyo.log( "sendShortDTMF : " + tone);
 			if (this.useFeedback && this.feedbackOnly) {
-				 this.$.playDialTone.call({
-				 		"id": String(tone),
-				 		"oneshot": true,
-				 		"feedbackOnly": this.feedbackOnly
-		                        });//sendDTMF
+				 this.$.hapticFeedback.call({
+						"period": 0,
+						"duration": 75
+					});//vibrate
                         } else {
 	                        var held = false;
 				enyo.application.CallSynergizer.callDtmf(tone, held);
@@ -224,11 +223,10 @@ enyo.kind({
 			this.continuousDTMFInProgress = true;
 			//enyo.log( "sendContinuousDTMF : " + tone);
 			if (this.useFeedback && this.feedbackOnly) {
-				 this.$.playDialTone.call({
-				 		"id": String(tone),
-				 		"oneshot": false,
-				 		"feedbackOnly": this.feedbackOnly
-		                        });
+				 this.$.hapticFeedback.call({
+						"period": 0,
+						"duration": 75
+					});//vibrate
 			} else {
 				var held = true;
 				enyo.application.CallSynergizer.callDtmf(tone, held);
