Index: /usr/palm/applications/com.palm.app.phone/app/controllers/app-assistant.js
===================================================================
--- /usr/palm/applications/com.palm.app.phone/app/controllers/~app-assistant.js	Fri Oct 16 21:20:07 2009
+++ /usr/palm/applications/com.palm.app.phone/app/controllers/app-assistant.js	Sat Oct 17 18:11:03 2009
@@ -197,7 +197,7 @@
			var stageController = appControl.getStageController(this.stageName);

				if (!stageController) {
-					this.showDialpad();
+					this.showCallLog();
				}

			this.focusStage();
@@ -467,7 +467,7 @@
		this.sceneStack.push(sceneName);
		sc.pushScene({
			"name": sceneName,
-			transition: Mojo.Transition.crossFade
+			transition: Mojo.Transition.none
		}, contactId, number, type);
	}.bind(this);
    
