diff --git a/usr/palm/applications/com.palm.app.phone/app/controllers/activecall-assistant.js b/usr/palm/applications/com.palm.app.phone/app/controllers/activecall-assistant.js
index bab675d..1c83657 100644
--- a/usr/palm/applications/com.palm.app.phone/app/controllers/activecall-assistant.js
+++ b/usr/palm/applications/com.palm.app.phone/app/controllers/activecall-assistant.js
@@ -763,6 +763,7 @@ break;
 		// save formatting
 		this.lastFormattedCalls = formattedCall;
 
+		this.toggleDTMF();
 	},
     
 	// configures class names and styles for a non-conference call. 
