Adds button to answer call via speaker phone
Index: /usr/palm/applications/com.palm.app.phone/app/controllers/incomingcall-assistant.js
===================================================================
--- .orig/usr/palm/applications/com.palm.app.phone/app/controllers/incomingcall-assistant.js
+++ /usr/palm/applications/com.palm.app.phone/app/controllers/incomingcall-assistant.js
@@ -336,6 +336,7 @@ var IncomingcallAssistant = Class.create
 				&& this.sliderOpen === false 
 				&& newSliderOpenState === true) {
 					this.answerCall();
+					TelephonyCommands.setAudioScenario("phone_back_speaker");
 			}
 			this.sliderOpen = newSliderOpenState
 		} 
