diff --git a/usr/palm/applications/com.palm.app.phone/app/controllers/announcer-assistant.js b/usr/palm/applications/com.palm.app.phone/app/controllers/announcer-assistant.js
index 15ff747..d171e1d 100644
--- a/usr/palm/applications/com.palm.app.phone/app/controllers/announcer-assistant.js
+++ b/usr/palm/applications/com.palm.app.phone/app/controllers/announcer-assistant.js
@@ -351,19 +351,24 @@ var AnnouncerAssistant = Class.create({
 			call = this.missed.pop();
 			count = this.missed.length;
 			
-			// don't try to call Blocked numbers back
-			if ( call.contact.canBeCalled() ) {
-				this.appAssistant.launchDial(call.contact);
-			} else {
-				UIStateMachine.event("missedcall");
-			}
+//			// don't try to call Blocked numbers back
+//			if ( call.contact.canBeCalled() ) {
+//				this.appAssistant.launchDial(call.contact);
+//			} else {
+//				UIStateMachine.event("missedcall");
+//			}
+//			
+//			if (count === 0) {
+//				this.announceMissedDashboardClear();
+//			} else {
+//				lastCall = this.missed[count - 1];
+//				this.announceUpdate("misseddash", lastCall.displayName, lastCall.message, count, lastCall.timestamp);
+//			}
+
+			UIStateMachine.event("missedcall");
+			lastCall = this.missed[0];
+			this.announceMissedDashboardClear();
 			
-			if (count === 0) {
-				this.announceMissedDashboardClear();
-			} else {
-				lastCall = this.missed[count - 1];
-				this.announceUpdate("misseddash", lastCall.displayName, lastCall.message, count, lastCall.timestamp);
-			}
 		}.bind(this);
 		
 		this.announce("misseddash", "notification-small-missed.png", false, "missed", this.missed.length, call.displayName, call.message, iconCallback, messageCallback, timeStamp);
