diff --git a/usr/palm/applications/com.palm.app.phone/app/controllers/app-assistant.js b/usr/palm/applications/com.palm.app.phone/app/controllers/app-assistant.js
index aa92cd0..c96be23 100644
--- a/usr/palm/applications/com.palm.app.phone/app/controllers/app-assistant.js
+++ b/usr/palm/applications/com.palm.app.phone/app/controllers/app-assistant.js
@@ -188,20 +188,14 @@ AppAssistant.prototype.handleLaunch = function(params){
 			if(PalmSystem.isMinimal)
 				return 
 			else
-				this.showDialpad();
+				this.showCallLog(); // start up to call log instead of dialpad
 		} 
     } else {
 		
 		if (!(this.telephonyEventListener.isPendingOrActive())) {
 			if (this.initialLaunch === false) {
 				QDLogger.log( "handleLaunch", "no args, no calls: focusing stage");
-				var appControl = Mojo.Controller.getAppController();
-    			var stageController = appControl.getStageController(this.stageName);
-					
-				if (!stageController) {
-					this.showDialpad();
-				}
-				
+				this.showCallLog(); // just show call log here
 				this.focusStage();
 			} else {
 				QDLogger.log( "handleLaunch", "initial launch, so doing nothing");
@@ -499,7 +493,7 @@ AppAssistant.prototype.showCallLog = function(type, keepHidden){
 		
 		sc.pushScene({
 			name: sceneName,
-			transition: Mojo.Transition.crossFade,
+			transition: Mojo.Transition.none, // we don't need the transition
 			templateModel: {
 				voicemailCount: voicemailCount,
 				voicemailCountDisplayClass: voicemailCountDisplayClass
@@ -522,6 +516,8 @@ AppAssistant.prototype.showCallLog = function(type, keepHidden){
 			if ( type == "missed" ) {
 				stageController.delegateToSceneAssistant("onMissedCallsTap", true);
 			}
+			stageController.delegateToSceneAssistant("closeDrawer"); // need this so drawers don't get stuck
+			pushLogScene(stageController); // make sure we go to all calls list
 		}
 	}
 };
@@ -773,14 +769,7 @@ AppAssistant.prototype.popActiveCall = function(params) {
 	if (stageController !== undefined) {
 		this.sceneStack = [];
 		stageController.popScenesTo();
-		if (this.backToCalllog) {
-			if(this.calllogType == "missed")
-				this.showCallLog("missed", true);
-			else
-				this.showCallLog("", true);
-		}
-		else 
-			this.showDialpad(null, true);
+		this.showCallLog("", true); // return to call log when call ended
 	} else {
 		QDLogger.log( "popActiveCall", "undefined stage");
 	}
