diff --git a/usr/palm/applications/com.palm.app.phone/app/models/states/CallLog.js b/usr/palm/applications/com.palm.app.phone/app/models/states/CallLog.js
index 9d42483..548e7f1 100644
--- a/usr/palm/applications/com.palm.app.phone/app/models/states/CallLog.js
+++ b/usr/palm/applications/com.palm.app.phone/app/models/states/CallLog.js
@@ -15,7 +15,7 @@ UIStateMachine.STATES.CallLogAbstractState = Class.create(UIStateMachine.STATES.
 		
 	},
 	event_back: function(commandEvent) {
-		this.machine.enter('dialpad_card');
+		commandEvent.stopPropagation();
 	},
 	event_focus: function() {
 		this.focusStage(this.machine.STAGE_MAIN);
diff --git a/usr/palm/applications/com.palm.app.phone/app/models/states/DialpadCard.js b/usr/palm/applications/com.palm.app.phone/app/models/states/DialpadCard.js
index 439856c..82a18f1 100644
--- a/usr/palm/applications/com.palm.app.phone/app/models/states/DialpadCard.js
+++ b/usr/palm/applications/com.palm.app.phone/app/models/states/DialpadCard.js
@@ -18,10 +18,12 @@ UIStateMachine.STATES.DialpadCardState = Class.create(UIStateMachine.STATES.Abst
 			this.appController.closeStage(this.machine.STAGE_MAIN);
 			Event.stop(commandEvent);
 		}
-		commandEvent.stopPropagation(); // stop mojo propogation, but allow app to be carded
+		//commandEvent.stopPropagation(); // stop mojo propogation, but allow app to be carded
+		this.machine.enter("calllog", undefined, true);
 	},
 	event_closed: function() {
 		this.appController.getStageController(this.machine.STAGE_MAIN).delegateToSceneAssistant("resetOnHide");
+		this.machine.enter("calllog", undefined, true);
 	},
 	event_emergency: function(isEnabled) {
 		if ( isEnabled ) {
diff --git a/usr/palm/applications/com.palm.app.phone/app/models/states/Favorites.js b/usr/palm/applications/com.palm.app.phone/app/models/states/Favorites.js
index 54955e7..688a761 100644
--- a/usr/palm/applications/com.palm.app.phone/app/models/states/Favorites.js
+++ b/usr/palm/applications/com.palm.app.phone/app/models/states/Favorites.js
@@ -9,8 +9,12 @@ UIStateMachine.STATES.FavoritesState = Class.create(UIStateMachine.STATES.Abstra
 		
 	},
 	event_back: function(commandEvent) {
-		this.machine.enter('dialpad_card');
+		this.machine.enter('calllog');
 	},
+	
+	 event_closed: function() {
+ 		this.machine.enter("calllog", undefined, true);
+ 	},
 	event_emergency: function(isEnabled) {
 		if ( isEnabled ) {
 			this.machine.enter("emergency_card");
diff --git a/usr/palm/applications/com.palm.app.phone/app/models/states/Poundapp.js b/usr/palm/applications/com.palm.app.phone/app/models/states/Poundapp.js
index 6ef8789..df266b9 100644
--- a/usr/palm/applications/com.palm.app.phone/app/models/states/Poundapp.js
+++ b/usr/palm/applications/com.palm.app.phone/app/models/states/Poundapp.js
@@ -23,7 +23,7 @@ UIStateMachine.STATES.PoundAppState = Class.create(UIStateMachine.STATES.Abstrac
 		}
 	},
 	event_closed: function() {
-		this.machine.enter('dialpad_card');
+		this.machine.enter('calllog', undefined, true);
 	},
 	event_emergency: function(isEnabled) {
 		this.machine.enter('emergency_card');
diff --git a/usr/palm/applications/com.palm.app.phone/app/models/states/Start.js b/usr/palm/applications/com.palm.app.phone/app/models/states/Start.js
index 979ed83..982cb96 100644
--- a/usr/palm/applications/com.palm.app.phone/app/models/states/Start.js
+++ b/usr/palm/applications/com.palm.app.phone/app/models/states/Start.js
@@ -8,7 +8,7 @@ UIStateMachine.STATES.StartState = Class.create(UIStateMachine.STATES.AbstractSt
 	},
 	// always launch to dialpad first
 	event_launch: function() {
-		this.machine.enter('dialpad_card');
+		this.machine.enter('calllog');
 	},
 	event_emergency: function(isEnabled) {
 		if ( isEnabled ) {
diff --git a/usr/palm/applications/com.palm.app.phone/app/models/states/Voicecommand.js b/usr/palm/applications/com.palm.app.phone/app/models/states/Voicecommand.js
index d92fbca..3caaa76 100644
--- a/usr/palm/applications/com.palm.app.phone/app/models/states/Voicecommand.js
+++ b/usr/palm/applications/com.palm.app.phone/app/models/states/Voicecommand.js
@@ -9,14 +9,14 @@ UIStateMachine.STATES.VoiceCommandState = Class.create(UIStateMachine.STATES.Abs
 	},
 	event_back: function(commandEvent) {
 		// todo go back to previous state, not always the dialpad
-		this.machine.enter("dialpad_card");
+		this.machine.enter("calllog");
 		if ( commandEvent ) {
         	Event.stop(commandEvent);
 		}
 	},
 	event_closed: function() {
 		// always go back to dialpad for now
-		this.machine.enter("dialpad_card", undefined, true);
+		this.machine.enter("calllog", undefined, true);
 	},
 	event_emergency: function(isEnabled) {
 		if ( isEnabled ) {
diff --git a/usr/palm/applications/com.palm.app.phone/app/models/states/Voicemail.js b/usr/palm/applications/com.palm.app.phone/app/models/states/Voicemail.js
index 2a24ebe..dd12db1 100644
--- a/usr/palm/applications/com.palm.app.phone/app/models/states/Voicemail.js
+++ b/usr/palm/applications/com.palm.app.phone/app/models/states/Voicemail.js
@@ -17,13 +17,14 @@ UIStateMachine.STATES.VoicemailState = Class.create(UIStateMachine.STATES.Abstra
 		}
 	},
 	event_back: function(commandEvent) {
-		this.machine.enter('dialpad_card');
+		this.machine.enter('calllog');
 	},
 	event_closed: function() {
 		var stageController = Mojo.Controller.getAppController().getStageController(this.machine.STAGE_MAIN);
 		if (stageController) {
 			stageController.delegateToSceneAssistant("reset");
 		}
+		this.machine.enter("calllog", undefined, true);
 	},
 	event_focus: function() {
 		this.focusStage(this.machine.STAGE_MAIN);
