diff --git a/usr/palm/applications/com.palm.app.phone/phoneApp/built.js b/usr/palm/applications/com.palm.app.phone/phoneApp/built.js
index 15b56f0..b3fb87c 100644
--- a/usr/palm/applications/com.palm.app.phone/phoneApp/built.js
+++ b/usr/palm/applications/com.palm.app.phone/phoneApp/built.js
@@ -111,7 +111,7 @@ components: [ {
 icon: "../images/menu-icon-voicemail.png",
 value: "voicemail",
 className: "voicemail-icon",
-style: "position:relative;",
+style: "position:relative; display:none !important",
 components: [ {
 name: "voicemailCountWrapper",
 className: "voicemail-count-wrapper",
diff --git a/usr/palm/applications/com.palm.app.phone/phoneApp/source/PhoneTabsMenu.js b/usr/palm/applications/com.palm.app.phone/phoneApp/source/PhoneTabsMenu.js
index 442f337..a5d780d 100644
--- a/usr/palm/applications/com.palm.app.phone/phoneApp/source/PhoneTabsMenu.js
+++ b/usr/palm/applications/com.palm.app.phone/phoneApp/source/PhoneTabsMenu.js
@@ -6,7 +6,7 @@ enyo.kind({
 		{icon: "../images/menu-icon-contacts.png",label: $L("CONTACTS"), value: "contactlookup"},  
 		{icon: "../images/menu-icon-favorites.png", label: $L("FAVORITES"),value: "favorites"},
 		{icon: "../images/menu-icon-call-log.png", label: $L("CALL-LOG"), value: "calllog"},   
- 	  {icon: "../images/menu-icon-voicemail.png", label: $L("VOICEMAIL"), value: "voicemail", className: "voicemail-icon", components: [
+ 	  {icon: "../images/menu-icon-voicemail.png", label: $L("VOICEMAIL"), value: "voicemail", className: "voicemail-icon", style:"display:none !important;", components: [
 			{name: "voicemailCountWrapper",  className: "voicemail-count-wrapper", showing: false, components: [
 				{name: "voicemailCount", className: "voicemail-count"},
 				{kind:"Image", src: "../images/voicemail-play-icon.png", name: "unknownpositiveCountImg"},
