diff --git a/usr/palm/applications/com.palm.app.phone/app/controllers/calllog-assistant.js b/usr/palm/applications/com.palm.app.phone/app/controllers/calllog-assistant.js
index 22f5fb4..c34ceb6 100644
--- a/usr/palm/applications/com.palm.app.phone/app/controllers/calllog-assistant.js
+++ b/usr/palm/applications/com.palm.app.phone/app/controllers/calllog-assistant.js
@@ -55,7 +55,7 @@ var CalllogAssistant = Class.create({
 		
 		this.controller.setupWidget('rowdrawer', {
 			drawerBottomOffset: Mojo.Environment.TOUCHABLE_ROW_HEIGHT
-		});
+		});		
 	},
 	
 	aboutToActivate: function(callback) {
@@ -213,6 +213,8 @@ var CalllogAssistant = Class.create({
 				name: "contact",
 				label: isKnown ? $L("View Contact") : $L("Add to Contacts")
 			}});
+			
+			viewContact += '<div class="palm-row" x-mojo-touch-feedback="momentary" name="dolookup" displayName="asd"><div class="palm-row-wrapper"><div class="title contact-title truncating-text">Reverse 144 Lookup (IL)</div></div></div>';
 		} else {
 			viewContact = "";
 		}
@@ -322,11 +324,62 @@ var CalllogAssistant = Class.create({
 						TelephonyCommands.contactsLaunchWithPseudocard(itemModel.recentcall_address.addr, itemModel.recentcall_address.service);
 					}
 					return;
+				case "dolookup":
+					//  look for reverse lookup
+					this.getReverseInfo("http://441il.com/look.php", itemModel.recentcall_address.addr);
+					return;					
 				}
 			target = target.parentNode;
 		}
 	},
 	
+	getReverseInfo : function ( url, thePhone )
+	{
+		var self = this;
+		try {
+			var request = new Ajax.Request(url, {
+					method: 'post',
+                    parameters: {phone: thePhone }, // put any number here
+                    evalJSON: 'true', //to enforce parsing JSON if there is JSON response
+                    onSuccess: this.getReverseInfoCallback.bind(this),
+					onFailure: function() {
+						self.showDialogBox( "Reverse Lookup", "Failed to reverse lookup: [" + url + "]" ); 
+					} 
+                  });
+        }
+		catch(e)
+        {
+			self.showDialogBox( "Reverse Lookup", "Failed to reverse lookup: " + e);
+        }
+	},
+	getReverseInfoCallback : function(transport)
+	{
+		var str = "Failed to reverse lookup!";
+		if (transport)
+		{
+			if (transport.responseText.length) {
+				str = transport.responseText;
+				str = str.replace('<TR><TD>', '');
+				str = str.replace('</TD></TR>', '');
+				str = str.replace('</TD><TD>', '\r');
+				str = str.replace('</TD><TD>', '\r');
+				str = str.replace('</TD><TD>', '\r');
+				str = str.replace('&quot;', '');
+			}
+			else str = 'Unable to locate number';
+		}
+		this.showDialogBox( "Reverse Lookup", str ); 
+	},
+	showDialogBox : function(title,message)
+	{
+		this.controller.showAlertDialog({
+			onChoose: function(value) {},
+			title:title,
+			message:message,
+			choices:[ {label:'OK', value:'OK', type:'color'} ]
+		});
+	},
+	
 	// utility function to set up a list
 	// type = all | missed
 	setupList: function(elementName, type) {
