Index: /usr/palm/applications/com.palm.app.phone/app/controllers/activecall-assistant.js
===================================================================
--- /usr/palm/applications/com.palm.app.phone/app/controllers/~activecall-assistant.orig
+++ /usr/palm/applications/com.palm.app.phone/app/controllers/activecall-assistant.js
@@ -1186,17 +1186,7 @@
 		
 		// always pop active call
 		// FIXME: actually don't if dialpad is on top
-		var closeStage = false;
-		if ( this.lastLines && 
-			this.lastLines.length > 0 &&
-			this.lastLines[0] &&
-			(this.lastLines[0].origin === TelephonyCallState.DIALING) &&
-			this.appAssistant.firstuse !== true) {
-			QDLogger.log( "activeCallAssistant::updateCalls", "last call outgoing, so back to the pad: ");
-		} else {
-			QDLogger.log( "activeCallAssistant::updateCalls", "last call incoming, so hiding stage");
-			closeStage = true;
-		}
+		var closeStage = true;
 		
 		this.lastLines = [];
 		this.appAssistant.popActiveCall();
