diff --git a/usr/palm/applications/com.palm.app.phone/shared/dialpad/Dialpad.js b/usr/palm/applications/com.palm.app.phone/shared/dialpad/Dialpad.js
index 3caaecd..c59130b 100644
--- a/usr/palm/applications/com.palm.app.phone/shared/dialpad/Dialpad.js
+++ b/usr/palm/applications/com.palm.app.phone/shared/dialpad/Dialpad.js
@@ -335,7 +335,7 @@ enyo.kind({
 	name: "PhoneDialpad",
 	kind: "Dialpad",
  	useFeedback: true,
-	feedbackOnly: true, 
+	feedbackOnly: false, 
 	className: "dialpad-container", 
 	create: function(){
 		this.inherited(arguments);
