diff --git a/usr/palm/applications/com.palm.app.phone/source/uistates/ActiveCallCard.js b/usr/palm/applications/com.palm.app.phone/source/uistates/ActiveCallCard.js
index 3e85f39..9de0f20 100644
--- a/usr/palm/applications/com.palm.app.phone/source/uistates/ActiveCallCard.js
+++ b/usr/palm/applications/com.palm.app.phone/source/uistates/ActiveCallCard.js
@@ -39,8 +39,10 @@ enyo.kind({
 		// states that are ok to go back to after hangup, else go to dialpad
 		if (['voicemail','calllog','calllog_missed','preferences','favorites'].indexOf(prevState) >= 0) {
 			enyo.application.UI.enter(prevState, undefined, true);
+			enyo.application.closeMainCard();
 		} else {
 			enyo.application.UI.enter('dialpad_card', undefined, true);
+			enyo.application.closeMainCard();
 		}
 		this.isHangingUp = false;
 	},
