--- /usr/lib/luna/system/luna-applauncher/app/controllers/global-search-assistant.js.bak	Wed Oct 21 13:41:29 2009
+++ /usr/lib/luna/system/luna-applauncher/app/controllers/global-search-assistant.js	Wed Oct 21 13:43:03 2009
@@ -889,7 +889,7 @@
 	else
 		params = {"personId":contactId,"contactPointId":contactPointId,"address":addr}
 	
-	ApplicationService.launch('com.palm.app.messaging', params);    
+	ApplicationService.launch('com.palm.natebiz.gdial-pro', params);    
   },
 
 /**********end messaging***********************/
@@ -913,7 +913,7 @@
 			else
 				var params = {"number": number};
 		}
-		ApplicationService.launch('com.palm.app.phone', params);        
+		ApplicationService.launch('com.palm.natebiz.gdial-pro', params);        
 	},
 	
 	launchAddToContacts: function(number) {
