diff --git a/usr/palm/applications/com.palm.app.phone/shared/dialpad/Dialpad.js b/usr/palm/applications/com.palm.app.phone/shared/dialpad/Dialpad.js
index 3caaecd..31cc04d 100644
--- a/usr/palm/applications/com.palm.app.phone/shared/dialpad/Dialpad.js
+++ b/usr/palm/applications/com.palm.app.phone/shared/dialpad/Dialpad.js
@@ -96,6 +96,7 @@ enyo.kind({
 			{name: "dialpadButton6", number: "6", subtext: "MNO", className: "dialpadbutton-right"},
 			{name: "dialpadButton9", number: "9", subtext: "WXYZ", className: "dialpadbutton-right"}
 		]},
+		{name: "hapticFeedback", kind: enyo.PalmService, service: "palm://com.palm.vibrate/", method: "vibrate", onSuccess: "", onFailure:""},
 		{name: "playDialTone", kind: enyo.PalmService, service: "palm://com.palm.audio/dtmf/", method: "playDTMF", onSuccess: "", onFailure:""},
 		{name: "stopDialTone", kind: enyo.PalmService, service: "palm://com.palm.audio/dtmf/", method: "stopDTMF", onSuccess: "", onFailure:""}
 	],
@@ -271,6 +272,10 @@ enyo.kind({
 				 		"oneshot": true,
 				 		"feedbackOnly": this.feedbackOnly
 		                        });//sendDTMF
+				 this.$.hapticFeedback.call({
+						"period": 0,
+						"duration": 75
+								});//vibrate
                         } else {
 	                        var held = false;
 				enyo.application.CallSynergizer.callDtmf(tone, held);
@@ -289,6 +294,10 @@ enyo.kind({
 				 		"oneshot": false,
 				 		"feedbackOnly": this.feedbackOnly
 		                        });
+				 this.$.hapticFeedback.call({
+						"period": 0,
+						"duration": 75
+								});//vibrate
 			} else {
 				var held = true;
 				enyo.application.CallSynergizer.callDtmf(tone, held);
