diff --git a/usr/palm/applications/com.palm.app.phone/app/controllers/activecall-assistant.js b/usr/palm/applications/com.palm.app.phone/app/controllers/activecall-assistant.js
index c0e08cf..26a69ae 100644
--- a/usr/palm/applications/com.palm.app.phone/app/controllers/activecall-assistant.js
+++ b/usr/palm/applications/com.palm.app.phone/app/controllers/activecall-assistant.js
@@ -130,6 +130,9 @@ var ActivecallAssistant = Class.create({
 		}
 		this.updateCalls(initialCallState);
 		
+//BEGIN PowerButton Patch block 1 of 3
+		this.powerButtonSubscriptionx =TelephonyCommands.powerButtonSubscribe(true, 'activecall', this.onPowerButtonEventDuringCall.bind(this));
+//END PowerButton Patch block 1 of 3
 		//start the timer
 		this.timerInterval = this.controller.window.setInterval(this.updateDisplayTimer.bind(this), 1000);
     	
@@ -179,6 +182,28 @@ var ActivecallAssistant = Class.create({
 				event.stopPropagation();
 			}
 	},
+//BEGIN PowerButton Patch block 2 of 3
+	onPowerButtonEventDuringCall: function(results) {
+		if (results&&results.powerKey&&results.powerKey=='released') {
+			Mojo.Log.error("this.audioActiveProfile: ", this.audioActiveProfile);
+			if (this.audioActiveProfile !== "phone_back_speaker") {
+				//First, "tap" each Disconnect Button.
+				for (i = 0; i < 3; i++) {
+					if (this.controller.get("disc_button_" + i)) {
+						//QDLogger.log( "**** disconnecting line:" + i + "****");
+						Mojo.Event.send(this.controller.get("disc_button_" + i), Mojo.Event.tap);
+					}
+				}
+				//In case the Disconnect Buttons don't work, or there are calls not disconnected, just call the function to disconnect all calls.
+				//The reason we don't just do this one function is that the screen redraw is not pretty.
+				//QDLogger.log( "********disconnecting all calls********");
+				this.eventListener.disconnectAllCalls();
+				//QDLogger.log( "********finished routine********");
+			}
+		}
+	},
+	
+//END PowerButton Patch block 2 of 3
 		
 	setupDtmfPad: function(index) {
 		this.showDTMF = false;
@@ -1345,6 +1370,9 @@ break;
 		this.appAssistant.telephonyEventListener.removeCallStateListener(this);
 		this.appAssistant.telephonyEventListener.removeAudioStateListener(this);
 		
+//BEGIN PowerButton Patch block 3 of 3
+		this.powerButtonSubscriptionx.cancel();	//remove slider listener
+//END PowerButton Patch block 3 of 3
 		this.appAssistant.activeCallFocused = false;
 
 		if (!this.unloaded) {
