diff --git a/usr/palm/applications/com.palm.app.phone/app/controllers/announcer-assistant.js b/usr/palm/applications/com.palm.app.phone/app/controllers/announcer-assistant.js
index cabc06e..11e2665 100644
--- a/usr/palm/applications/com.palm.app.phone/app/controllers/announcer-assistant.js
+++ b/usr/palm/applications/com.palm.app.phone/app/controllers/announcer-assistant.js
@@ -374,14 +374,8 @@ var AnnouncerAssistant = Class.create({
 		// tapping the message calls that number back and decrements the missed count
 		var messageCallback = function() {
 			var call = this.missed.pop();
-			this.appAssistant.launchDial(call.number);
-			var count = this.missed.length
-			if (count === 0) {
+			this.appAssistant.launchMissed();
 				this.announceMissedDashboardClear();
-			} else {
-				var lastCall = this.missed[count-1];
-				this.announceUpdate("misseddash",lastCall.displayName, lastCall.message, count,lastCall.timestamp);
-			}
 		}.bind(this);
 		
 		this.announce("misseddash", "notification-small-missed.png", false, "missed", 
