diff --git a/usr/palm/applications/com.palm.app.phone/shared/dialpad/Dialpad.js b/usr/palm/applications/com.palm.app.phone/shared/dialpad/Dialpad.js
index 3caaecd..c938378 100644
--- a/usr/palm/applications/com.palm.app.phone/shared/dialpad/Dialpad.js
+++ b/usr/palm/applications/com.palm.app.phone/shared/dialpad/Dialpad.js
@@ -96,7 +96,7 @@ enyo.kind({
 			{name: "dialpadButton6", number: "6", subtext: "MNO", className: "dialpadbutton-right"},
 			{name: "dialpadButton9", number: "9", subtext: "WXYZ", className: "dialpadbutton-right"}
 		]},
-		{name: "playDialTone", kind: enyo.PalmService, service: "palm://com.palm.audio/dtmf/", method: "playDTMF", onSuccess: "", onFailure:""},
+		{name: "hapticFeedback", kind: enyo.PalmService, service: "palm://com.palm.vibrate/", method: "vibrate", onSuccess: "", onFailure:""},
 		{name: "stopDialTone", kind: enyo.PalmService, service: "palm://com.palm.audio/dtmf/", method: "stopDTMF", onSuccess: "", onFailure:""}
 	],
 	create: function() {
@@ -266,11 +266,10 @@ enyo.kind({
 			// assumes the id has the key at position 5
 			//enyo.log( "sendShortDTMF : " + tone);
 			if (this.useFeedback && this.feedbackOnly) {
-				 this.$.playDialTone.call({
-				 		"id": String(tone),
-				 		"oneshot": true,
-				 		"feedbackOnly": this.feedbackOnly
-		                        });//sendDTMF
+				 this.$.hapticFeedback.call({
+						"period": 0,
+						"duration": 75
+					});//vibrate
                         } else {
 	                        var held = false;
 				enyo.application.CallSynergizer.callDtmf(tone, held);
@@ -284,11 +283,10 @@ enyo.kind({
 			this.continuousDTMFInProgress = true;
 			//enyo.log( "sendContinuousDTMF : " + tone);
 			if (this.useFeedback && this.feedbackOnly) {
-				 this.$.playDialTone.call({
-				 		"id": String(tone),
-				 		"oneshot": false,
-				 		"feedbackOnly": this.feedbackOnly
-		                        });
+				 this.$.hapticFeedback.call({
+						"period": 0,
+						"duration": 75
+					});//vibrate
 			} else {
 				var held = true;
 				enyo.application.CallSynergizer.callDtmf(tone, held);
