diff --git a/usr/palm/applications/com.palm.app.usbpassthrough/app/controllers/usbpassthrough-assistant.js b/usr/palm/applications/com.palm.app.usbpassthrough/app/controllers/usbpassthrough-assistant.js
index 2cd9f70..2089b43 100644
--- a/usr/palm/applications/com.palm.app.usbpassthrough/app/controllers/usbpassthrough-assistant.js
+++ b/usr/palm/applications/com.palm.app.usbpassthrough/app/controllers/usbpassthrough-assistant.js
@@ -185,8 +185,8 @@ var UsbpassthroughAssistant = Class.create({
 			
 			if (this.isActivationFile != true) {
 				$('masterGroup').style.display = 'none';
-				$('menuGroup').style.display = 'none'; // show menu
-				$('entryGroup').style.display = 'block';
+				$('menuGroup').style.display = 'block'; // show menu
+				$('entryGroup').style.display = 'none';
 			}
 		} catch (e) {
 			Mojo.Log.error ("main.activate(): exception: " + e)
diff --git a/usr/palm/applications/com.palm.app.usbpassthrough/appinfo.json b/usr/palm/applications/com.palm.app.usbpassthrough/appinfo.json
index c888755..5f60749 100644
--- a/usr/palm/applications/com.palm.app.usbpassthrough/appinfo.json
+++ b/usr/palm/applications/com.palm.app.usbpassthrough/appinfo.json
@@ -6,7 +6,7 @@
 	"icon": "icon.png",
 	"version": "1.0.31",
 	"noWindow": false,
-	"visible": false,
+	"visible": true,
 	"vendor": "Palm, Inc."
 }
 
