diff --git a/usr/palm/applications/com.palm.app.phone/app/controllers/calllog-assistant.js b/usr/palm/applications/com.palm.app.phone/app/controllers/calllog-assistant.js
index 2cb234a..0876ab2 100644
--- a/usr/palm/applications/com.palm.app.phone/app/controllers/calllog-assistant.js
+++ b/usr/palm/applications/com.palm.app.phone/app/controllers/calllog-assistant.js
@@ -480,15 +480,17 @@ var CalllogAssistant = Class.create({
 			};
 			view_contact = Mojo.View.render({template: 'calllog/listitem_contact', object: template_model});
 		}
-		
+
 		var element = Mojo.View.convertToNode('<div name="rowdrawer" class="calllogdetails" x-mojo-element="Drawer"> '
-											  + call_details + '<div name="contact_numbers_placeholder"></div>' + view_contact
+											  + call_details + '<div name="contact_numbers_placeholder"></div>'
+											  + '<div class="palm-row" x-mojo-touch-feedback="momentary" name="dolookup" value="' + itemModel.contactId + '" number="' + itemModel.number + '"><div class="palm-row-wrapper"><div class="title truncating-text">Rückwärtssuche (DasOertliche.de)</div></div></div>'
+											  + view_contact
 									  		  + '</div>', this.controller.window.document);
 		
 		itemNode.querySelector(".palm-row-wrapper").appendChild(element);
 		
 		this.loadContactNumbers(element, itemModel, is_valid);
-		
+
 		// no 'add to contact' or altnum row
 		if ( ! is_valid && itemModel.contactId == -1 ) {
 			element.querySelector('div[name="call-log-details-row"]').removeClassName('first').addClassName('single');
@@ -620,6 +622,34 @@ var CalllogAssistant = Class.create({
 						this.dialNumber(details.number, itemModel.contactId, details.label);
 					}
 					return;
+
+				case "dolookup":
+					//Mojo.Log.error("WhitePages Lookup");
+					//Mojo.Log.error("itemModel.recentcall_address.addr = " + itemModel.recentcall_address.addr);
+
+					// Launch the Browser with Das Oertliche reverse lookup
+					DasOertlicheURL = "http://mobil.dasoertliche.de/oe/go_rueckwaertssuche?number=" + target.getAttribute('number'); 
+					DasOertlicheURL =  DasOertlicheURL.replace("+","00");
+
+			
+					this.controller.serviceRequest("palm://com.palm.applicationManager", {
+					   method: "open",
+					   parameters:  {
+					       id: 'com.palm.app.browser',
+					       params: {
+						   target: DasOertlicheURL
+					       }
+					   }
+					 }); 
+					
+					//this.controller.serviceRequest('palm://com.palm.applicationManager', {
+					//	method: 'launch',
+					//	parameters: {
+					//		id: "com.whitepages.mobile",
+					//		params:{action: "search", scene: "reverse", search_number: itemModel.recentcall_address.addr, autosearch:true}
+					//	},
+					return;
+				
 					
 				case "contact":
 					if ( target.getAttribute('value') == -1 ) {
