Index: /usr/palm/applications/com.palm.app.phone/app/controllers/activecall-assistant.js
===================================================================
--- /usr/palm/applications/com.palm.app.phone/app/controllers/activecall-assistant.js
+++ /usr/palm/applications/com.palm.app.phone/app/controllers/activecall-assistant.js
@@ -33,6 +33,9 @@
 		this.lastLines = [];
 		this.pauseWaitDigits = [];
 		this.showWaitButtonOnActive = [];
+//BEGIN PowerButton Patch block 1 of 3
+		this.powerButtonSubscriptionx =TelephonyCommands.powerButtonSubscribe(true, 'activecall', this.onPowerButtonEventDuringCall.bind(this));
+//END PowerButton Patch block 1 of 3
 		this.waitMenuItems = [];
 		
 		// select multicall rules
@@ -180,6 +183,25 @@
 			}
 	},
 		
+//BEGIN PowerButton Patch block 2 of 3
+	onPowerButtonEventDuringCall: function(results) {
+		//QDLogger.log( "********ActivecallAssistant#onPowerButtonEvent********");
+		if (results&&results.powerKey&&results.powerKey=='released') {
+			//First, "tap" each Disconnect Button.
+			for (i = 0; i < 3; i++) {
+				if (this.controller.get("disc_button_" + i)) {
+					//QDLogger.log( "**** disconnecting line:" + i + "****");
+					Mojo.Event.send(this.controller.get("disc_button_" + i), Mojo.Event.tap);
+				}
+			}
+			//In case the Disconnect Buttons don't work, or there are calls not disconnected, just call the function to disconnect all calls.
+			//The reason we don't just do this one function is that the screen redraw is not pretty.
+			//QDLogger.log( "********disconnecting all calls********");
+			this.eventListener.disconnectAllCalls();
+			//QDLogger.log( "********finished routine********");
+		}
+	},
+//END PowerButton Patch block 2 of 3
 	setupDtmfPad: function(index) {
 		this.showDTMF = false;
 		
@@ -1345,6 +1367,9 @@
 		this.appAssistant.telephonyEventListener.removeCallStateListener(this);
 		this.appAssistant.telephonyEventListener.removeAudioStateListener(this);
 		
+//BEGIN PowerButton Patch block 3 of 3
+		this.powerButtonSubscriptionx.cancel();	//remove slider listener
+//END PowerButton Patch block 3 of 3
 		this.appAssistant.activeCallFocused = false;
 
 		if (!this.unloaded) {
