diff --git a/usr/palm/applications/com.palm.app.phone/app/controllers/dialpad-assistant.js b/usr/palm/applications/com.palm.app.phone/app/controllers/dialpad-assistant.js
index 19e28e2..c7907cd 100644
--- a/usr/palm/applications/com.palm.app.phone/app/controllers/dialpad-assistant.js
+++ b/usr/palm/applications/com.palm.app.phone/app/controllers/dialpad-assistant.js
@@ -642,6 +642,13 @@ var DialpadAssistant = Class.create({
 	numberClick: function(event, key){
 		
 		if (key) {
+			// tjb
+			this.controller.serviceRequest("palm://com.palm.vibrate", {
+			method: 'vibrate', parameters: { 
+			period: '0',
+			duration: '50' }
+			});
+			// END tjb
 			TelephonyCommands.sendDTMF(key, true);
 			this.formatAndUpdateDialString(key,event);
 			if (this.dialString.length == 1 && this.speedDialString.length == 0) {
@@ -943,6 +950,14 @@ var DialpadAssistant = Class.create({
 		if (this.isValidToneKey(this.pressedKey)
 			&& this.pressedKey !== Mojo.Char.backspace
 			&& (!this.appAssistant.lookupOnKeyPress || this.keyMapOpt.get(evt.keyIdentifier))) {
+			// tjb
+			this.controller.serviceRequest("palm://com.palm.vibrate", {
+			method: 'vibrate', parameters: { 
+			period: '0',
+			duration: '50' }
+			});
+			// END tjb
+
 			TelephonyCommands.sendDTMF(this.pressedKey, true);
 		}
 		
