Index: /usr/palm/applications/com.palm.app.phone/app/controllers/app-assistant.js
===================================================================
--- /usr/palm/applications/com.palm.app.phone/app/controllers/~app-assistant.js	Fri Oct 16 21:20:07 2009
+++ /usr/palm/applications/com.palm.app.phone/app/controllers/app-assistant.js	Sat Oct 17 18:11:03 2009
@@ -194,7 +194,7 @@
 		if (!(this.telephonyEventListener.isPendingOrActive())) {
 			if (this.initialLaunch === false) {
 				QDLogger.log( "handleLaunch", "no args, no calls: showing dialpad");
-				this.showDialpad();
+				this.showCallLog();
 				this.focusStage();
 			} else {
 				QDLogger.log( "handleLaunch", "initial launch, so doing nothing");
@@ -461,7 +461,7 @@
 		this.sceneStack.push(sceneName);
 		sc.pushScene({
 			"name": sceneName,
-			transition: Mojo.Transition.crossFade
+			transition: Mojo.Transition.none
 		}, contactId, number, type);
 	}.bind(this);
    
