diff --git a/usr/palm/applications/com.palm.app.phone/app/controllers/activecall-assistant.js b/usr/palm/applications/com.palm.app.phone/app/controllers/activecall-assistant.js
index 2443f5b..495465f 100644
--- a/usr/palm/applications/com.palm.app.phone/app/controllers/activecall-assistant.js
+++ b/usr/palm/applications/com.palm.app.phone/app/controllers/activecall-assistant.js
@@ -1440,7 +1440,7 @@ break;
 		
 		this.lastLines = [];
 		if (closeStage) {
-			this.appAssistant.closePhoneStage();
+			//this.appAssistant.closePhoneStage();
 		}
 		
 		this.appAssistant.popActiveCall();
