diff --git a/usr/palm/applications/com.palm.app.phoneprefs/app/controllers/preflist-assistant.js b/usr/palm/applications/com.palm.app.phoneprefs/app/controllers/preflist-assistant.js
index 124990a..707b163 100644
--- a/usr/palm/applications/com.palm.app.phoneprefs/app/controllers/preflist-assistant.js
+++ b/usr/palm/applications/com.palm.app.phoneprefs/app/controllers/preflist-assistant.js
@@ -93,7 +93,7 @@ var PreflistAssistant = Class.create({
 		this.controller.setupWidget('callforward-number', this.dsAttributes, this.callFWDNumbermodel);
 		
 		//Hide the Row until we get the response from network
-		$('callfwdfirstrow').hide();
+		$('callfwdfirstrow').show();
 		$('callfwdlastrow').hide();
 		$('callWaitingRowStatus').hide();
 		$('showOutgoingCallerIDRowStatus').hide();
