diff --git a/usr/palm/applications/com.palm.app.phone/app/controllers/app-assistant.js b/usr/palm/applications/com.palm.app.phone/app/controllers/app-assistant.js
index 7c6d6ca..6d489d8 100644
--- a/usr/palm/applications/com.palm.app.phone/app/controllers/app-assistant.js
+++ b/usr/palm/applications/com.palm.app.phone/app/controllers/app-assistant.js
@@ -1039,7 +1039,7 @@ AppAssistant.prototype.showLimitedDialpad = function(sc, dialParams) {
 
 AppAssistant.prototype.loadSettings = function() {
 	this.stickyDialpad = this.getStickyDialpad();
-	this.hideCallLogCallDuration = this.getHideCallLogDuration(); // true hides call length in call log; false shows call duration
+	this.hideCallLogCallDuration = false; // true hides call length in call log; false shows call duration
 	this.numberMaskEnabled = PhoneNumberFormatter.prototype.masked = this.getNumberMaskEnabled();
 
 	if (this.hideCallLogCallDuration === undefined) {
