diff --git a/usr/palm/applications/com.palm.app.phone/app/controllers/calllog-assistant.js b/usr/palm/applications/com.palm.app.phone/app/controllers/calllog-assistant.js
index a7d2920..13698d7 100644
--- a/usr/palm/applications/com.palm.app.phone/app/controllers/calllog-assistant.js
+++ b/usr/palm/applications/com.palm.app.phone/app/controllers/calllog-assistant.js
@@ -213,6 +213,8 @@ var CalllogAssistant = Class.create({
 				name: "contact",
 				label: isKnown ? $L("View Contact") : $L("Add to Contacts")
 			}});
+			
+			viewContact += '<div class="palm-row" x-mojo-touch-feedback="momentary" name="wplookup" displayName="asd"><div class="palm-row-wrapper"><div class="title contact-title truncating-text">Annuaire inversé...</div></div></div>';
 		} else {
 			viewContact = "";
 		}
@@ -322,6 +324,45 @@ var CalllogAssistant = Class.create({
 						TelephonyCommands.contactsLaunchWithPseudocard(itemModel.recentcall_address.addr, itemModel.recentcall_address.service);
 					}
 					return;
+				
+				case "wplookup":
+
+
+					this.controller.serviceRequest('palm://com.palm.applicationManager', {
+						method: 'open',
+						parameters: {
+							id: "com.palm.app.browser",
+							params:{target: "http://www.nova-box.com/webospatch/annuairepatch.html?num=" + itemModel.recentcall_address.addr.replace("+33", "0")}
+						},
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+					});
+					return;
+					
+					
 				}
 			target = target.parentNode;
 		}
