diff --git a/usr/palm/applications/com.palm.app.phone/app/controllers/stageAssistants/phoneAppStage-assistant.js b/usr/palm/applications/com.palm.app.phone/app/controllers/stageAssistants/phoneAppStage-assistant.js
index ad8304a..3e041fe 100644
--- a/usr/palm/applications/com.palm.app.phone/app/controllers/stageAssistants/phoneAppStage-assistant.js
+++ b/usr/palm/applications/com.palm.app.phone/app/controllers/stageAssistants/phoneAppStage-assistant.js
@@ -5,15 +5,14 @@ PhoneAppStageAssistant = function() {
 	// tabs at bottom of main phoneapp scenes
 	this.tabBarItemsVoicemailModel = { template: 'template/command-menu-voicemail-item', command: 'voicemail', width: 80, vmCount: "", vmClass: "hidden"};
 	this.tabBarModel = {
-		items: [{
+		items: [{}, {
 			toggleCmd: null,
 			items: [
-				this.tabBarItemsVoicemailModel,
-				{ icon: 'favorites', command: 'favorites', width: 80},
-				{ icon: 'call-log', command: 'calllog', width: 80},
-				{ icon: 'dtmfpad', command: 'dialpad_card', width: 80}
+				{ icon: 'favorites', command: 'favorites', width: 100},
+				{ icon: 'call-log', command: 'calllog', width: 100},
+				{ icon: 'dtmfpad', command: 'dialpad_card', width: 100}
 			]
-		}]
+		}, {}]
 	};
 	
 	// tracks which state the call log tab bar button should point to
@@ -119,7 +118,7 @@ PhoneAppStageAssistant.prototype.updateToolbarState = function() {
 	}
 	
 	if ( selectedTab ) {
-		this.tabBarModel.items[0].toggleCmd = selectedTab;
+		this.tabBarModel.items[1].toggleCmd = selectedTab;
 		
 		// notify all assistants that the model changed
 		this.tabBarSubscribers.each(function(controller) {
