diff --git a/usr/palm/applications/com.palm.app.phone/app/controllers/announcer-assistant.js b/usr/palm/applications/com.palm.app.phone/app/controllers/announcer-assistant.js
index d7df1b5..3025954 100644
--- a/usr/palm/applications/com.palm.app.phone/app/controllers/announcer-assistant.js
+++ b/usr/palm/applications/com.palm.app.phone/app/controllers/announcer-assistant.js
@@ -412,7 +412,7 @@ var AnnouncerAssistant = Class.create({
 		this.appControl.showBanner(
 			{
 				"messageText": message,
-				"soundClass": 'notifications'
+				"soundClass": ''
 			},
 			launchArgs
 		);
@@ -711,7 +711,7 @@ var AnnouncerAssistant = Class.create({
 						lightweight: true,
 						name: stageName,
 						height: DroppedcallAssistant.getPopupHeight(contact),
-						soundclass: "alerts"
+						soundclass: "none"
 					},
 					function(stagecontroller) {
 						stagecontroller.pushScene("droppedcall", message, contact);
@@ -759,7 +759,7 @@ var AnnouncerAssistant = Class.create({
 		this.appControl.showBanner(
 			{
 				messageText: message,
-				soundClass: 'notifications'
+				soundClass: ''
 			},
 			{
 				prefill: contact.addr // todo need to pass a full contact to prefill
