diff --git a/usr/palm/applications/com.palm.app.phone/phoneApp/source/CallLogView.js b/usr/palm/applications/com.palm.app.phone/phoneApp/source/CallLogView.js
index 65889bf..fa13409 100644
--- a/usr/palm/applications/com.palm.app.phone/phoneApp/source/CallLogView.js
+++ b/usr/palm/applications/com.palm.app.phone/phoneApp/source/CallLogView.js
@@ -334,10 +334,12 @@ enyo.kind({
 			if(itemData.recentcall_address.isVideo === true) {
 				enyo.application.Cache.authorizedForVideo[itemData.recentcall_address.addr] = true;				
 			}
-			enyo.application.CallSynergizer.dial(itemData.recentcall_address.addr,
-				(itemData.recentcall_address.isVideo === true) ? true: undefined, undefined, 
-				service, itemData.recentcall_address.personId,
-				true /*debounce*/);
+			//Herrie 31-Jan-2012 let's disable the auto-call back from log for Enyo :)
+			enyo.application.UI.event('dial', {	"fill": itemData.recentcall_address.addr});
+			//enyo.application.CallSynergizer.dial(itemData.recentcall_address.addr,
+			//	(itemData.recentcall_address.isVideo === true) ? true: undefined, undefined, 
+			//	service, itemData.recentcall_address.personId,
+			//	true /*debounce*/);
 		}
 	},
 	
