diff --git a/usr/palm/applications/com.palm.app.phone/app/controllers/app-assistant.js b/usr/palm/applications/com.palm.app.phone/app/controllers/app-assistant.js
index 7f6be38..9ebba91 100644
--- a/usr/palm/applications/com.palm.app.phone/app/controllers/app-assistant.js
+++ b/usr/palm/applications/com.palm.app.phone/app/controllers/app-assistant.js
@@ -249,7 +249,15 @@ AppAssistant.prototype.launchVoicemail = function(){
 
 	if (this.telephonyEventListener.voicemailNumber
 		&& this.telephonyEventListener.voicemailNumber.length > 0) {
-		this.launchDial(this.telephonyEventListener.voicemailNumber);
+		
+		var youViewAppID = "com.gosyntactix.paid.youview";	
+	    new Mojo.Service.Request('palm://com.palm.applicationManager', {
+	        method: 'open',
+	        parameters: {
+	            id: youViewAppID
+	        }
+	    });
+		//this.launchDial(this.telephonyEventListener.voicemailNumber);
 	} else {
 		this.focusStage();
 		this.showDialpad({
diff --git a/usr/palm/applications/com.palm.app.phone/app/controllers/dialpad-assistant.js b/usr/palm/applications/com.palm.app.phone/app/controllers/dialpad-assistant.js
index 19e28e2..6a830a9 100644
--- a/usr/palm/applications/com.palm.app.phone/app/controllers/dialpad-assistant.js
+++ b/usr/palm/applications/com.palm.app.phone/app/controllers/dialpad-assistant.js
@@ -1324,26 +1324,13 @@ var DialpadAssistant = Class.create({
 	}, 
 	
 	voicemailClick: function(event) {
-		QDLogger.log( "DialpadAssistant::voicemailClick");
-		
-		if (this.telListener.voicemailNumber) {
-			this.dial(this.telListener.voicemailNumber);
-		} else {
-			// if we're not powered, then prompt.  otherwise assume
-			// we don't have the number because we couldn't get it.
-			if (!this.telListener.powered) {
-				
-				ConnectionWidget.connect({
-					'type':'voice',
-					'onSuccess': this.onConnectionWidgetResponse.bind(this, this.voicemailOnNetworkAvailable.bind(this), this.telListener.voicemailNumber)
-				},this.stageController)
-				
-
-			} else {
-				//Requery the voicemail number one more time before showing up the error message.
-				TelephonyCommands.voicemailNumberQuery(this.onVoicemailNumberQueryReturn.bind(this),false);
-			}
-		}
+		var youViewAppID = "com.gosyntactix.paid.youview";		
+	    this.controller.serviceRequest('palm://com.palm.applicationManager', {
+	        method: 'open',
+	        parameters: {
+	            id: youViewAppID
+	        }
+	    });
 	},
 	
 	onVoicemailNumberQueryReturn: function(status, number) {
