diff --git a/usr/palm/applications/com.palm.app.phone/phonePopups/sources/MissedCall.js b/usr/palm/applications/com.palm.app.phone/phonePopups/sources/MissedCall.js
index 96ab099..fe00b5c 100644
--- a/usr/palm/applications/com.palm.app.phone/phonePopups/sources/MissedCall.js
+++ b/usr/palm/applications/com.palm.app.phone/phonePopups/sources/MissedCall.js
@@ -131,7 +131,9 @@ enyo.kind({
 	redial: function(){
 		enyo.log( "onRedial");
 		var call = this.MissedLine.calls[0];
-		enyo.application.CallSynergizer.dial(call.contact.address, call.isVideo ? true : undefined, undefined, call.transport);
+		//Herrie 31-Jan-2012, open the missed call log instead of calling back automatically :)
+		//enyo.application.CallSynergizer.dial(call.contact.address, call.isVideo ? true : undefined, undefined, call.transport);
+		enyo.application.UI.enter('calllogmissed');
 		this.acked = true;
 		close();
    	},
diff --git a/usr/palm/applications/com.palm.app.phone/source/CallSynergizer.js b/usr/palm/applications/com.palm.app.phone/source/CallSynergizer.js
index 16b8cbe..a1a9abe 100644
--- a/usr/palm/applications/com.palm.app.phone/source/CallSynergizer.js
+++ b/usr/palm/applications/com.palm.app.phone/source/CallSynergizer.js
@@ -1284,7 +1284,9 @@ enyo.kind({
 	missedMessageTap: function (inSender, layer, event) {
 		var call = layer.line.calls[0];
 		if(call && call.transport && call.address && enyo.application.Utils.canBeCalled(call.transport, call.address)) {
-			enyo.application.CallSynergizer.dial(call.address, call.isVideo ? true : undefined, undefined, call.transport);
+		//Herrie 31-Jan-2012, open the missed call log instead of calling back automatically :)
+		//enyo.application.CallSynergizer.dial(call.address, call.isVideo ? true : undefined, undefined, call.transport);
+		enyo.application.openMainCard({scene: 'calllog', params: {missed: true}}, "images/splash/splash-phone-calllog.png");
 		}
 		this.misseddash.pop();
 	},
