diff --git a/usr/palm/applications/com.palm.app.phone/shared/activecall/LineState.js b/usr/palm/applications/com.palm.app.phone/shared/activecall/LineState.js
index 9648e36..62a0cd4 100644
--- a/usr/palm/applications/com.palm.app.phone/shared/activecall/LineState.js
+++ b/usr/palm/applications/com.palm.app.phone/shared/activecall/LineState.js
@@ -6,7 +6,10 @@ enyo.kind({
 	pack: "center",
 	components: [
 		{name: "prefixLbl", style: "font-weight: bold; margin-right: 5px;", content: $L("MUTE"), showing: false},
-		{name: "timerLbl"}
+		{name: "timerLbl"},
+		//Twitter @Androidcn Email:flatlandcn@gmail.com
+		{name:"Vibrate", kind:"PalmService", service: "palm://com.palm.vibrate", method: "vibrate",params: {'period': 0,'duration': 100}},
+
 	],
 	published: {
 		line: ''
@@ -21,8 +24,9 @@ enyo.kind({
 		if (this.line.state == enyo.application.CallSynergizer.STATES.ACTIVE ) {
 			this.$.timerLbl.setContent(enyo.application.Utils.getElaspedTime(this.startTime));
 			this.callTimer = window.setInterval(enyo.hitch(this, "timer"), 1000);
-			
-		} else if ( this.line.state == enyo.application.CallSynergizer.STATES.DIALING
+			//Twitter @Androidcn Email:flatlandcn@gmail.com
+     		this.$.Vibrate.call();
+     	} else if ( this.line.state == enyo.application.CallSynergizer.STATES.DIALING
 			 	|| this.line.state == enyo.application.CallSynergizer.STATES.DIALPENDING ) {
 			this.$.timerLbl.setContent($L("Connecting"));
 			
@@ -34,7 +38,8 @@ enyo.kind({
 			
 		} else if ( this.line.state == enyo.application.CallSynergizer.STATES.DISCONNECTED ) {
 			this.$.timerLbl.setContent($L("Ended"));
-			
+			//Twitter @Androidcn Email:flatlandcn@gmail.com
+			this.$.Vibrate.call();
 		} else {
 			this.error("unknown state: " + this.line.state)
 		}
@@ -44,14 +49,20 @@ enyo.kind({
 		this.inherited(arguments);
 	},
 	timer: function() {
-		if(enyo.application.Cache.muteStatus) {
+			if(enyo.application.Cache.muteStatus) {
 			if (!this.$.prefixLbl.getShowing())
 				this.$.prefixLbl.setShowing(true);
 		} else {
 			if (this.$.prefixLbl.getShowing())
 				this.$.prefixLbl.setShowing(false);
 		}
-
-		this.$.timerLbl.setContent(enyo.application.Utils.getElaspedTime(this.startTime));
+	    this.$.timerLbl.setContent(enyo.application.Utils.getElaspedTime(this.startTime));
+	    //Twitter @Androidcn Email:flatlandcn@gmail.com
+	    var currentTime = new Date().getTime();
+	    var elapsed = currentTime - this.startTime;
+	    var Elapsedtimer = new Date(elapsed);
+	    callTimerSeconds = Elapsedtimer.getSeconds();
+     		if(callTimerSeconds == 50)
+     			this.$.Vibrate.call();
 	},
 });
