diff --git a/usr/palm/applications/com.palm.app.phoneprefs/app/controllers/preflist-assistant.js b/usr/palm/applications/com.palm.app.phoneprefs/app/controllers/preflist-assistant.js
index 498ffa0..8bd1899 100644
--- a/usr/palm/applications/com.palm.app.phoneprefs/app/controllers/preflist-assistant.js
+++ b/usr/palm/applications/com.palm.app.phoneprefs/app/controllers/preflist-assistant.js
@@ -103,6 +103,10 @@ var PreflistAssistant = Class.create({
 			value: true
 		};
 		       
+	this.phoneAppHideCallDurationModel = {
+			value: false
+		};
+		       
         this.callFwdModel = {
 			value: false
 		};
@@ -145,6 +149,11 @@ var PreflistAssistant = Class.create({
         }, this.showContactMatchModel);
         this.controller.get('showContactMatch').addEventListener(Mojo.Event.propertyChange, this.doPropertyChanged.bindAsEventListener(this, 'showContactMatch'));
 
+		this.controller.setupWidget('phoneAppHideCallDuration', {
+	    modelProperty: 'value'
+	}, this.phoneAppHideCallDurationModel);
+	this.controller.get('phoneAppHideCallDuration').addEventListener(Mojo.Event.propertyChange, this.doPropertyChanged.bindAsEventListener(this, 'phoneAppHideCallDuration'));
+
 		this.controller.setupWidget('callfwd', {
             modelProperty: 'value'
         }, this.callFwdModel);
@@ -393,6 +402,7 @@ var PreflistAssistant = Class.create({
 			['phoneInternationalDialingActive', 'phoneInternationalDialingRegionId'],this.updateInternationalDialingSettings.bind(this));
 		this.networkStatusQueryReq = TelephonyService.networkStatusQuery(this.updateNetworkname.bind(this));
         this.getSystemPreferencesServiceShowcontactmatchReq = SystemService.getSystemPreferencesService('showcontactmatch',this.updateShowcontactmatch.bind(this));
+	this.getSystemPreferencesServicephoneAppHideCallDurationReq = SystemService.getSystemPreferencesService('phoneAppHideCallDuration',this.updatephoneAppHideCallDuration.bind(this));
 		this.updateManualDataSettingsReq = SystemService.getManualDataSettings(this.updateManualDataSettings.bind(this));
 		this.getVoiceNumberEditableReq = SystemService.getSystemPreferencesService('allowEditVoicemail',this.updateVoicenumberEditableQueryDone.bind(this));
 		this.getBandSelectionAllowed = SystemService.getSystemPreferencesService('hideRatSelection',this.updateRatSelection.bind(this));
@@ -1002,7 +1012,14 @@ voiceNumberSetDone: function(payload){
         
     },
 	
+        //phoneAppHideCallDuration
+    updatephoneAppHideCallDuration: function(payload){
+                if(payload.phoneAppHideCallDuration == undefined)
+                        return;
+        this.phoneAppHideCallDurationModel.value = payload.phoneAppHideCallDuration;
+        this.controller.modelChanged(this.phoneAppHideCallDurationModel);
 	
+    },
 	
 	//DTMF
     updateDTMF: function(payload){
@@ -2174,6 +2191,9 @@ break;
 			case 'showContactMatch':
    	            this.setSystemPreferencesServiceReq = SystemService.setSystemPreferencesService('showcontactmatch',value,null);
 				break;
+			case 'phoneAppHideCallDuration':
+		    this.setSystemPreferencesServiceReq = SystemService.setSystemPreferencesService('phoneAppHideCallDuration',value,null);
+				break;
             case 'showOutgoingCallerID':
                 this.clirSetReq = TelephonyService.clirSet(!value, this.handleError.bind(this));
             	break;            
diff --git a/usr/palm/applications/com.palm.app.phoneprefs/app/views/preflist/preflist-scene.html b/usr/palm/applications/com.palm.app.phoneprefs/app/views/preflist/preflist-scene.html
index 9912c08..8448dd1 100644
--- a/usr/palm/applications/com.palm.app.phoneprefs/app/views/preflist/preflist-scene.html
+++ b/usr/palm/applications/com.palm.app.phoneprefs/app/views/preflist/preflist-scene.html
@@ -42,6 +42,22 @@
         </div>
     </div>
 
+  <div class="palm-group">
+        <div x-mojo-loc="" class="palm-group-title">
+            Call Log
+        </div>
+        <div class="palm-list">
+            <div class="palm-row single">
+                                <div class="palm-row-wrapper">
+                        <div id="phoneAppHideCallDuration" x-mojo-element="ToggleButton"></div>
+                        <div class="title capitalize" x-mojo-loc="">
+                            Hide Call Duration
+                        </div>
+                    </div>
+            </div>
+        </div>
+    </div>
+
     <div class="palm-group" id="calls">
         <div x-mojo-loc="" class="palm-group-title">
             Calls
