diff --git a/usr/palm/applications/com.palm.app.phone/app/controllers/announcer-assistant.js b/usr/palm/applications/com.palm.app.phone/app/controllers/announcer-assistant.js
index ced1495..18c2257 100644
--- a/usr/palm/applications/com.palm.app.phone/app/controllers/announcer-assistant.js
+++ b/usr/palm/applications/com.palm.app.phone/app/controllers/announcer-assistant.js
@@ -372,17 +372,11 @@ var AnnouncerAssistant = Class.create({
 		// tapping the message calls that number back and decrements the missed count
 		var messageCallback = function() {
 			var call = this.missed.pop();
-			this.appAssistant.launchDial(call.number);
-			var count = this.missed.length
-			if (count === 0) {
-				this.announceMissedDashboardClear();
-			} else {
-				var lastCall = this.missed[count-1];
-				this.announceUpdate("misseddash",lastCall.displayName, lastCall.message, count,lastCall.timestamp);
-			}
-		}.bind(this);
-		
-		this.announce("misseddash", "notification-small-missed.png", false, "missed", 
++			this.appAssistant.launchMissed();
++			this.announceMissedDashboardClear();
+ 		}.bind(this);
+ 		
+ 		this.announce("misseddash", "notification-small-missed.png", false, "missed", 
 			count, call.displayName, call.message, iconCallback, messageCallback, timeStamp);
 	},
 	
