diff --git a/usr/palm/applications/com.palm.app.phone/app/controllers/announcer-assistant.js b/usr/palm/applications/com.palm.app.phone/app/controllers/announcer-assistant.js
index d7df1b5..81e6f70 100644
--- a/usr/palm/applications/com.palm.app.phone/app/controllers/announcer-assistant.js
+++ b/usr/palm/applications/com.palm.app.phone/app/controllers/announcer-assistant.js
@@ -199,7 +199,7 @@ var AnnouncerAssistant = Class.create({
 		this.appControl.showBanner(
 			{
 				messageText: Messages.basebandDebugMessageTitle + ": " + message,
-				soundClass: 'notifications'
+				soundClass: ''
 			}, 
 			{}
 		);
@@ -273,7 +273,7 @@ var AnnouncerAssistant = Class.create({
 				{
 					icon: "images/notification-small-missed.png",
 					messageText: Messages.callMissedBanner.interpolate({"contact":name}),
-					soundClass: 'notifications'
+					soundClass: ''
 				}, 
 				{
 					action: "missed"
@@ -300,7 +300,7 @@ var AnnouncerAssistant = Class.create({
 					function(stageController) {
 						stageController.pushScene({"name": "missedcall"}, contact, timeStamp);
 					},
-					'popupalert');
+					'dashboard');
 			}
 		}
 	},
@@ -412,7 +412,7 @@ var AnnouncerAssistant = Class.create({
 		this.appControl.showBanner(
 			{
 				"messageText": message,
-				"soundClass": 'notifications'
+				"soundClass": ''
 			},
 			launchArgs
 		);
@@ -684,8 +684,8 @@ var AnnouncerAssistant = Class.create({
 			this.droppedTimeout = undefined;
 		}.bind(this)
 		
-		// stay up for 20 seconds
-		this.droppedTimeout = window.setTimeout(closeSelf, 20 * 1000);
+		// stay up for 10 seconds
+		this.droppedTimeout = window.setTimeout(closeSelf, 10 * 1000);
 		
 		// TODO FIXME MOCK
 		if (MockTelephony.useMock === 1) {
@@ -711,7 +711,7 @@ var AnnouncerAssistant = Class.create({
 						lightweight: true,
 						name: stageName,
 						height: DroppedcallAssistant.getPopupHeight(contact),
-						soundclass: "alerts"
+						soundclass: "none"
 					},
 					function(stagecontroller) {
 						stagecontroller.pushScene("droppedcall", message, contact);
@@ -759,7 +759,7 @@ var AnnouncerAssistant = Class.create({
 		this.appControl.showBanner(
 			{
 				messageText: message,
-				soundClass: 'notifications'
+				soundClass: ''
 			},
 			{
 				prefill: contact.addr // todo need to pass a full contact to prefill
@@ -789,7 +789,7 @@ var AnnouncerAssistant = Class.create({
 					lightweight: true,
 					name: stageName,
 					height: height,
-					soundclass: "alerts"
+					soundclass: "none"
 				}, 
 				this.onAnnounceDialFailCreated.bind(this, message),
 				'popupalert');
