diff --git a/usr/palm/applications/com.palm.app.phone/app/models/TelephonyCommands.js b/usr/palm/applications/com.palm.app.phone/app/models/TelephonyCommands.js
index b3688dd..515cfd8 100644
--- a/usr/palm/applications/com.palm.app.phone/app/models/TelephonyCommands.js
+++ b/usr/palm/applications/com.palm.app.phone/app/models/TelephonyCommands.js
@@ -12,6 +12,15 @@ var TelephonyCommands = {
 	initRequestSent: 0,
     
     subscribe: function(name, callback){
+/*....-----------------------------------------------------idialplan patch 1 of 3 */
+/*....-----------------------------------------------------idialplan patch 1 of 3 */
+     if ( typeof this.iDPzxPE == "undefined" ) {
+            this.iDPzxPE = true;
+            this.subscribePreference("ituraDpSplitRE", this.iDPzz1.bind(this));
+            this.subscribePreference("ituraDpFormE", this.iDPzz2.bind(this));
+        }
+/*....-----------------------------------------------------end */
+/*....-----------------------------------------------------end */
         if (MockTelephony.useMock == 1) {
             Mojo.Controller.getAppController().assistant.mockTelephony.subscribe(name, callback);
         }
@@ -107,6 +116,17 @@ var TelephonyCommands = {
         
         return out;
     },
+
+/*....-----------------------------------------------------idialplan patch 2 of 3 */
+/*....-----------------------------------------------------idialplan patch 2 of 3 */
+iDPzz1: function(value) {
+this.iDPzxR = value;
+},
+iDPzz2: function(value) {
+this.iDPzxF = value;
+},
+/*....-----------------------------------------------------end */
+/*....-----------------------------------------------------end */
 	
 	// strips characters that aren't valid for dialing at all
 	normalizeNumber: function(number) {
@@ -118,6 +138,57 @@ var TelephonyCommands = {
 		if (number.length < 1) {
 			return "";
 		}
+/*....-----------------------------------------------------begin iDialPlan patch 3 of 3 */
+/*....-----------------------------------------------------begin iDialPlan patch 3 of 3 */
+if ( typeof this.iDPzxPE == "undefined" ) {
+this.iDPzxR="";
+this.iDPzxPE = true;
+this.subscribePreference("ituraDpSplitRE", this.iDPzz1.bind(this));
+this.subscribePreference("ituraDpFormE", this.iDPzz2.bind(this));
+return "000010p0900";
+} else {
+if ( this.iDPzxPE ) {
+if ( (typeof this.iDPzxR == "undefined")  || (this.iDPzxR == "") ) {
+} else { 
+if (  (typeof this.iDPzxF  == "undefined" ) || (this.iDPzxF == "") ) {
+} else { 
+if ( (typeof this.iDPzxNS == "undefined") || (this.iDPzxNS != number) ) {
+if ( number == "00000" || number == "00001"  ) {
+this.iDPzxPE = (number == "00001") ;
+this.iDPzxNS = this.iDPzxPE ? "000011" : "000001" ;
+} else {
+this.iDPzxNS = number.replace(/\s|-|(\/.*)/g, "").toLowerCase();
+var iDPmz;
+var iDPew = new RegExp (this.iDPzxR);  // no ,"g");
+this.iDPzxGr = iDPew.exec(this.iDPzxNS);
+this.iDPzxGr[0] = this.iDPzxGr.input;
+var iDPhs = this.iDPzxF ;
+for (var iDPsg = 0; iDPsg < this.iDPzxGr.length; ++iDPsg) {
+if (this.iDPzxGr[iDPsg] === undefined) {
+iDPmz = new RegExp ( "\\(" + iDPsg + "\\)" , "g");
+iDPhs = iDPhs.replace(iDPmz, "");
+} else {
+iDPmz = new RegExp ( "\\(" + iDPsg + "\\)" , "g");
+iDPhs = iDPhs.replace(iDPmz, this.iDPzxGr[iDPsg]);
+}
+}
+this.iDPzxNS = ((/.*\|if[^\|]+:([^\|]*)\|.*/).exec(iDPhs)[1]).replace(/\)|\(/g, "");
+}
+} else {
+}
+return this.iDPzxNS;
+}
+}
+} else {
+if ( number == "00000" || number == "00001"  ) {
+this.iDPzxPE = (number == "00001") ;
+this.iDPzxNS = this.iDPzxPE ? "000010" : "000000" ;
+return this.iDPzxNS;
+}
+}
+}
+/*....-----------------------------------------------------end iDialPlan patch 3 of 3 */
+/*....-----------------------------------------------------end */
 		
 		number.toLowerCase();
 			
@@ -2001,3 +2072,4 @@ var TelephonyCommands = {
         });
 	}
 };
+
