diff --git a/usr/palm/applications/com.palm.app.phoneprefs/app/controllers/preflist-assistant.js b/usr/palm/applications/com.palm.app.phoneprefs/app/controllers/preflist-assistant.js
index 124990a..81c27cb 100644
--- a/usr/palm/applications/com.palm.app.phoneprefs/app/controllers/preflist-assistant.js
+++ b/usr/palm/applications/com.palm.app.phoneprefs/app/controllers/preflist-assistant.js
@@ -242,7 +242,8 @@ var PreflistAssistant = Class.create({
 		
 		this.voiceRoamingChoices =  [
 			{label: this.carrierOnlyLabel,value: "homeonly"},
-			{label : $L("Automatic"),value: "any"}
+			{label : $L("Automatic"),value: "any"},
+			{label : $L("Roam Only"),value: "roamonly"}
 		];
 	
 		this.voiceRoamingModel = {
@@ -1546,18 +1547,23 @@ updateHAC: function(payload){
 			// if update happens while we are open and band selection
 			// was previously allowed (assuming based on choices == 3)
 			// put things back
-			if (this.voiceRoamingChoices.length == 3) {
-				this.voiceRoamingChoices[1] = {
-					label: $L("Automatic"),
-					value: "any"	
-				};
-				this.voiceRoamingChoices.pop();
-			}
+			//if (this.voiceRoamingChoices.length == 3) {
+				//this.voiceRoamingChoices[1] = {
+					//label: $L("Automatic"),
+					//value: "any"	
+				//};
+				//this.voiceRoamingChoices.pop();
+			//}
 			
 			if (payload.extended.mode == 'any') {
 				this.voiceRoamingModel.currentVoiceRoaming = "any";
 				$('dataroamingrow').show();
-			} else {
+		}
+			else if(payload.extended.mode == 'roamonly') {
+				this.voiceRoamingModel.currentVoiceRoaming = "roamonly";
+				$('dataroamingrow').show();
+		} 
+			else {
 				this.voiceRoamingModel.currentVoiceRoaming = "homeonly";
 				$('dataroamingrow').hide();
 			}
