diff --git a/usr/palm/applications/com.palm.app.phone/app/controllers/activecall-assistant.js b/usr/palm/applications/com.palm.app.phone/app/controllers/activecall-assistant.js
index bab675d..8a27e39 100644
--- a/usr/palm/applications/com.palm.app.phone/app/controllers/activecall-assistant.js
+++ b/usr/palm/applications/com.palm.app.phone/app/controllers/activecall-assistant.js
@@ -1430,17 +1430,7 @@ break;
 		
 		// always pop active call
 		// FIXME: actually don't if dialpad is on top
-		var closeStage = false;
-		if ( this.lastLines && 
-			this.lastLines.length > 0 &&
-			this.lastLines[0] &&
-			(this.lastLines[0].origin === TelephonyCallState.DIALING) &&
-			this.appAssistant.firstuse !== true) {
-			QDLogger.log( "activeCallAssistant::updateCalls", "last call outgoing, so back to the pad: ");
-		} else {
-			QDLogger.log( "activeCallAssistant::updateCalls", "last call incoming, so hiding stage");
-			closeStage = true;
-		}
+		var closeStage = true;
 		
 		this.lastLines = [];
 		if (closeStage) {
