diff --git a/usr/palm/applications/com.palm.app.phone/phoneApp/source/PhoneTabsMenu.js b/usr/palm/applications/com.palm.app.phone/phoneApp/source/PhoneTabsMenu.js
index 8a15f51..fea84ac 100644
--- a/usr/palm/applications/com.palm.app.phone/phoneApp/source/PhoneTabsMenu.js
+++ b/usr/palm/applications/com.palm.app.phone/phoneApp/source/PhoneTabsMenu.js
@@ -3,13 +3,13 @@ enyo.kind({
 	kind: "RadioToolButtonGroup", 
 	flex: 1,                                                                           
 	components: [
-		{icon: "../images/menu-icon-voicemail.png", value: "voicemail", className: "voicemail-icon", style: "position:relative;", components: [
+		{icon: "../images/menu-icon-favorites.png", value: "favorites"},
+		{icon: "../images/menu-icon-voicemail.png", value: "voicemail", className: "voicemail-icon", style: "position:relative; display:none !important;", components: [
 			{name: "voicemailCountWrapper", className: "voicemail-count-wrapper", showing: false, components: [
 				{name: "voicemailCount", className: "voicemail-count"},
 				{kind:"Image", src: "../images/voicemail-play-icon.png", name: "unknownpositiveCountImg"},
 			]}
 		]},
-		{icon: "../images/menu-icon-favorites.png", value: "favorites"},
 		{icon: "../images/menu-icon-call-log.png", value: "calllog"},
 		{icon: "../images/menu-icon-dtmfpad.png", value: "dialpad_card"}
 	],
