diff --git a/usr/palm/applications/com.palm.app.phone/shared/dialer/DialStringWidget.js b/usr/palm/applications/com.palm.app.phone/shared/dialer/DialStringWidget.js
index 4641f1c..513392f 100644
--- a/usr/palm/applications/com.palm.app.phone/shared/dialer/DialStringWidget.js
+++ b/usr/palm/applications/com.palm.app.phone/shared/dialer/DialStringWidget.js
@@ -28,7 +28,7 @@ enyo.kind({
 		this.inherited(arguments);
 		this.setDefaultText();
 		
-		this.$.dialingShortcutsSvc.call({"keys": ['4DigitNumber', '5DigitNumber', '6DigitNumber', '7DigitNumber']});
+		this.$.dialingShortcutsSvc.call({"keys": ['4DigitNumber', '5DigitNumber', '6DigitNumber', '7DigitNumber', '8DigitNumber', '9DigitNumber', '10DigitNumber']});
 		this.dialingShortcuts = {};
 		this.rawString = "";
 		this.dialString = "";
@@ -190,10 +190,10 @@ enyo.kind({
 
 		for (key in payload) {
 			if (typeof(payload[key]) == 'string' && enyo.string.trim(payload[key]).length > 0) {
-				this.dialingShortcuts[parseInt(key.charAt(0))] = payload[key];
+				this.dialingShortcuts[parseInt(key.match(/\d+/))] = payload[key];
 				//enyo.log("Adding dialingShortcut: " + this.dialingShortcuts[parseInt(key.charAt(0))]);
-			} else if (this.dialingShortcuts[parseInt(key.charAt(0))]) {
-				delete this.dialingShortcuts[parseInt(key.charAt(0))];
+			} else if (this.dialingShortcuts[parseInt(key.match(/\d+/))]) {
+				delete this.dialingShortcuts[parseInt(key.match(/\d+/))];
 				//enyo.log("Remove shortcut at: " + key);
 			}
 		}
diff --git a/usr/palm/applications/com.palm.app.phone/shared/phoneprefs/controls/DialingPref.js b/usr/palm/applications/com.palm.app.phone/shared/phoneprefs/controls/DialingPref.js
index 0da6751..b11503c 100644
--- a/usr/palm/applications/com.palm.app.phone/shared/phoneprefs/controls/DialingPref.js
+++ b/usr/palm/applications/com.palm.app.phone/shared/phoneprefs/controls/DialingPref.js
@@ -69,7 +69,7 @@ enyo.kind({
 
 		this.dialingShortcutsList = [];
 		this.$.dialingShortcutlist.hide();
-		this.$.getPreferences.call({"keys": ["4DigitNumber", "5DigitNumber", "6DigitNumber", "7DigitNumber"]});
+		this.$.getPreferences.call({"keys": ["4DigitNumber", "5DigitNumber", "6DigitNumber", "7DigitNumber", "8DigitNumber", "9DigitNumber", "10DigitNumber"]});
 	},
 	
 	loadDialingShortcuts: function(inSender, payload) {
@@ -83,7 +83,7 @@ enyo.kind({
 		this.dialingShortcutsList.splice(0, this.dialingShortcutsList.length); // clear the array
 		for (key in payload) {
 			if (key != "returnValue" && typeof(payload[key]) == 'string') {
-				var str = key.charAt(0);
+				var str = key.match(/\d+/);
 				//according to localization team we need to use formatchoice even though we only have plurals
 				//so the translation for different language can be appropriate
 				var dispText = enyo.application.Utils.formatChoice($L("1#1 digit|##{n} digits"), str, {n: str}); 
diff --git a/usr/palm/applications/com.palm.app.phone/shared/phoneprefs/source/Dialingshortcut.js b/usr/palm/applications/com.palm.app.phone/shared/phoneprefs/source/Dialingshortcut.js
index b93a506..eec0e37 100644
--- a/usr/palm/applications/com.palm.app.phone/shared/phoneprefs/source/Dialingshortcut.js
+++ b/usr/palm/applications/com.palm.app.phone/shared/phoneprefs/source/Dialingshortcut.js
@@ -15,7 +15,10 @@ enyo.kind({
 					{caption: $L("4 digits"), value:"4DigitNumber"},
 					{caption: $L("5 digits"), value:"5DigitNumber"},
 					{caption: $L("6 digits"), value:"6DigitNumber"},
-					{caption: $L("7 digits"), value:"7DigitNumber"}
+					{caption: $L("7 digits"), value:"7DigitNumber"},
+					{caption: $L("8 digits"), value:"8DigitNumber"},
+					{caption: $L("9 digits"), value:"9DigitNumber"},
+					{caption: $L("10 digits"), value:"10DigitNumber"}
 				]}
 			]},
 			{kind: "RowGroup", caption: $L("USE THIS DIALING PREFIX"), components: [
