diff --git a/usr/palm/applications/com.palm.app.phone/app/controllers/activecall-assistant.js b/usr/palm/applications/com.palm.app.phone/app/controllers/activecall-assistant.js
index bab675d..8b29399 100644
--- a/usr/palm/applications/com.palm.app.phone/app/controllers/activecall-assistant.js
+++ b/usr/palm/applications/com.palm.app.phone/app/controllers/activecall-assistant.js
@@ -1444,7 +1444,7 @@ break;
 		
 		this.lastLines = [];
 		if (closeStage) {
-			this.appAssistant.closePhoneStage();
+			//this.appAssistant.closePhoneStage();
 		}
 		
 		this.appAssistant.popActiveCall();
