diff --git a/usr/palm/applications/com.palm.app.phone/app/controllers/calllog-assistant.js b/usr/palm/applications/com.palm.app.phone/app/controllers/calllog-assistant.js
index 10d1d80..1efdb7d 100644
--- a/usr/palm/applications/com.palm.app.phone/app/controllers/calllog-assistant.js
+++ b/usr/palm/applications/com.palm.app.phone/app/controllers/calllog-assistant.js
@@ -10,7 +10,7 @@ var CalllogAssistant = Class.create({
 		this.stageController = Mojo.Controller.getAppController().getStageController("PhoneApp");
 		this.telListener = this.appAssistant.telephonyEventListener;
 		// if it's true, then we hide call length
-		this.hideduration = this.appAssistant.hideCallLogCallDuration;
+		this.hideduration = false;
 		this.zeroOffset = new Date(0, 0, 0, 0, 0, 0, 0).getTime();
 		
 		if (contactId !== undefined && contactId !== "")
