diff --git a/usr/palm/applications/com.palm.app.phone/app/controllers/poundGPS-assistant.js b/usr/palm/applications/com.palm.app.phone/app/controllers/poundGPS-assistant.js
index bf1b9f9..c8ef914 100644
--- a/usr/palm/applications/com.palm.app.phone/app/controllers/poundGPS-assistant.js
+++ b/usr/palm/applications/com.palm.app.phone/app/controllers/poundGPS-assistant.js
@@ -41,7 +41,15 @@ var PoundGPSAssistant = Class.create({
 					this.controller.get('vaccuracy').innerHTML = response.vertAccuracy;
 					this.controller.get('gpsinfo').show();
 				} else {
-					this.showError($L("Position Unavailable."));
+					this.showError($L("Accuracy appears to be poor."));
+					this.controller.get('lat').innerHTML = response.latitude;
+					this.controller.get('long').innerHTML = response.longitude;
+					this.controller.get('velocity').innerHTML = response.velocity;
+					this.controller.get('altitude').innerHTML = response.altitude;
+					this.controller.get('heading').innerHTML = response.heading;
+					this.controller.get('haccuracy').innerHTML = response.horizAccuracy;
+					this.controller.get('vaccuracy').innerHTML = response.vertAccuracy;
+					this.controller.get('gpsinfo').show();
 				}				
 			}
 			else {
@@ -49,6 +57,18 @@ var PoundGPSAssistant = Class.create({
 					this.showError($L("Timeout."));
 				else if(response.errorCode == 2)
 					this.showError($L("Position Unavailable."));
+				else if(response.errorCode == 3)
+					this.showError($L("GPS Returned Unknown Error."));
+				else if(response.errorCode == 4)
+					this.showError($L("Permanent Error."));
+				else if(response.errorCode == 5)
+					this.showError($L("Location Service Off."));
+				else if(response.errorCode == 6)
+					this.showError($L("Permission Denied."));
+				else if(response.errorCode == 7)
+					this.showError($L("Application already has a pending message."));
+				else if(response.errorCode == 8)
+					this.showError($L("Application Blacklisted."));
 				else 
 					this.showError($L("Unknown Error."));
 			}
