diff --git a/usr/palm/applications/com.palm.app.phone/app/models/TelephonyEventListener.js b/usr/palm/applications/com.palm.app.phone/app/models/TelephonyEventListener.js
index c7190b5..267f562 100644
--- a/usr/palm/applications/com.palm.app.phone/app/models/TelephonyEventListener.js
+++ b/usr/palm/applications/com.palm.app.phone/app/models/TelephonyEventListener.js
@@ -711,7 +711,9 @@ var TelephonyEventListener = Class.create({
 	
 	// callback after a contact lookup -- announces incoming call or passes on CNAP update
 	onIncomingLookupComplete: function(callId, cnapName, number, update, contact) {
-		
+		if (!(contact.name)) {
+			contact.ringtoneLoc = "/media/internal/ringtones/ringtoneforunknown.mp3";
+		}		
 		contact.cnapName = cnapName;
 		
 		QDLogger.log( "TelephonyEventListener#onIncomingLookupComplete", "namefromcontacts?:" , !!(contact.name));
