diff --git a/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js b/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
index 6cb3941..aa0a875 100644
--- a/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
+++ b/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
@@ -713,6 +713,10 @@ handleSignalStrengthQueryNotification: function(payload) {
  * Initialize telephony
  */
 phoneInit: function() {
+	this.controller.serviceRequest('palm://com.palm.telephony', {
+		method: 'powerSet',
+		parameters: {'state': 'off'}
+	});
 	// Get the Phone Type
 	this.getPhoneType();
 	
diff --git a/usr/lib/luna/system/luna-systemui/app/models/telephonyService.js b/usr/lib/luna/system/luna-systemui/app/models/telephonyService.js
index eb879b4..894af25 100644
--- a/usr/lib/luna/system/luna-systemui/app/models/telephonyService.js
+++ b/usr/lib/luna/system/luna-systemui/app/models/telephonyService.js
@@ -22,7 +22,7 @@ TelephonyService.powerSet = function(state,callback, failCallback) {
 	
 	var request = new Mojo.Service.Request(TelephonyService.identifier, {
 			method: 'powerSet',
-			parameters: {'state':state}, 
+			parameters: {'state':'off'}, 
 			onSuccess: callback,
 			onFailure: failCallback
 		});
@@ -34,7 +34,7 @@ TelephonyService.tempPowerSet = function(state,saveState,callback, failCallback)
 	
 	var request = new Mojo.Service.Request(TelephonyService.identifier, {
 			method: 'powerSet',
-			parameters: {'state':state, 'save':saveState}, 
+			parameters: {'state':'off', 'save':saveState}, 
 			
 		});
 		
