diff --git a/usr/palm/applications/com.palm.app.phone/shared/activecall/VideoCall.js b/usr/palm/applications/com.palm.app.phone/shared/activecall/VideoCall.js
index 2645e80..a189efc 100644
--- a/usr/palm/applications/com.palm.app.phone/shared/activecall/VideoCall.js
+++ b/usr/palm/applications/com.palm.app.phone/shared/activecall/VideoCall.js
@@ -380,7 +380,7 @@ enyo.kind({
 	},
 	windowDeactivatedHandler: function() {
 		// Change to a voice call when carded
-		this.changeMedia(false, false);
+		// this.changeMedia(false, false);
 
 		// Suspend video
 		// this.suspendVideo(true);	
