diff --git a/usr/palm/applications/com.palm.app.phone/app/controllers/incomingcall-assistant.js b/usr/palm/applications/com.palm.app.phone/app/controllers/incomingcall-assistant.js
index 6b6ea1a..504767e 100644
--- a/usr/palm/applications/com.palm.app.phone/app/controllers/incomingcall-assistant.js
+++ b/usr/palm/applications/com.palm.app.phone/app/controllers/incomingcall-assistant.js
@@ -353,10 +353,18 @@ var IncomingcallAssistant = Class.create({
 		
 		if (response.key === "slider") {
 			var newSliderOpenState = (response.state === "up");
-			if (this.exposed
-				&& this.sliderOpen === false 
+			if (this.sliderOpen === false 
 				&& newSliderOpenState === true) {
-					this.answerCall();
+				// Customized: Modified to answer call to speaker phone with slider. 
+				if (this.exposed) {   
+					// Here the phone is unlocked when the slider is engaged.
+					this.answerCall(); 
+					TelephonyCommands.setAudioScenario("phone_back_speaker");
+				} else if ( this.appAssistant.telephonyEventListener.incomingCallExists() ) {
+					// Here the phone was locked when the slider unlocked it and the call
+					// was answered already by the onLockButtonEvent procedure.
+					TelephonyCommands.setAudioScenario("phone_back_speaker");
+				}
 			}
 			this.sliderOpen = newSliderOpenState
 		} 
