diff --git a/usr/palm/applications/com.palm.app.phone/app/models/TelephonyCommands.js b/usr/palm/applications/com.palm.app.phone/app/models/TelephonyCommands.js
index b3688dd..9f562be 100644
--- a/usr/palm/applications/com.palm.app.phone/app/models/TelephonyCommands.js
+++ b/usr/palm/applications/com.palm.app.phone/app/models/TelephonyCommands.js
@@ -108,6 +108,16 @@ var TelephonyCommands = {
         return out;
     },
 	
+/*....-----------------------------------------------------idialplan patch 1 of 2 */
+iDPzz1: function(value) {
+this.iDPzxR = value;
+},
+iDPzz2: function(value) {
+this.iDPzxF = value;
+},
+/*....-----------------------------------------------------end */
+
+
 	// strips characters that aren't valid for dialing at all
 	normalizeNumber: function(number) {
 		if (!number)  
@@ -119,6 +129,59 @@ var TelephonyCommands = {
 			return "";
 		}
 		
+/*....-----------------------------------------------------begin iDialPlan patch 2 of 2 */
+/*....-----------------------------------------------------begin iDialPlan patch 2 of 2 */
+if ( typeof this.iDPzxS == "undefined" ) {
+this.iDPzxR="";
+this.iDPzxS = true;
+this.subscribePreference("ituraDpSplitRE", this.iDPzz1.bind(this));
+this.subscribePreference("ituraDpFormE", this.iDPzz2.bind(this));
+return "999910p0601";
+} else {
+if ( this.iDPzxS ) {
+if ( (typeof this.iDPzxR == "undefined")  || (this.iDPzxR == "") ) {
+} else { 
+if (  (typeof this.iDPzxF  == "undefined" ) || (this.iDPzxF == "") ) {
+} else { 
+if ( (typeof this.iDPav == "undefined") || (this.iDPav != number) ) {
+if ( number == "99990" || number == "99991"  ) {
+this.iDPzxS = (number == "99991") ;
+this.iDPav = this.iDPzxS ? "999911" : "999901" ;
+} else {
+this.iDPav = number.replace(/\s|-|(\/.*)/g, "").toLowerCase();
+var iDPcv;
+var iDPag = new RegExp (this.iDPzxR);  // no ,"g");
+this.iDPzxGr = iDPag.exec(this.iDPav);
+this.iDPcx = this.iDPzxGr.input;
+var iDPca = this.iDPzxF ;
+for (var iDPcz = 0; iDPcz < this.iDPzxGr.length; ++iDPcz) {
+if (this.iDPzxGr[iDPcz] === undefined) {
+iDPcv = new RegExp ( "\\(" + iDPcz + "\\)" , "g");
+iDPca = iDPca.replace(iDPcv, "");
+} else {
+iDPcv = new RegExp ( "\\(" + iDPcz + "\\)" , "g");
+iDPca = iDPca.replace(iDPcv, this.iDPzxGr[iDPcz]);
+}
+}
+this.iDPav = ((/.*\|if[^\|]+:([^\|]*)\|.*/).exec(iDPca)[1]).replace(/\)|\(/g, "");
+}
+} else {
+}
+return this.iDPav;
+}
+}
+} else {
+if ( number == "99990" || number == "99991"  ) {
+this.iDPzxS = (number == "99991") ;
+this.iDPav = this.iDPzxS ? "999910" : "999900" ;
+return this.iDPav;
+}
+}
+}
+/*....-----------------------------------------------------end iDialPlan patch 2 of 2 */
+
+/*....-----------------------------------------------------end iDialPlan patch 2 of 2 */
+	
 		number.toLowerCase();
 			
 		var validDigits = "+01234567890*#pwt";
