diff --git a/usr/palm/applications/com.palm.app.phone/app/controllers/announcer-assistant.js b/usr/palm/applications/com.palm.app.phone/app/controllers/announcer-assistant.js
index ced1495..7dc2a15 100644
--- a/usr/palm/applications/com.palm.app.phone/app/controllers/announcer-assistant.js
+++ b/usr/palm/applications/com.palm.app.phone/app/controllers/announcer-assistant.js
@@ -317,7 +317,7 @@ var AnnouncerAssistant = Class.create({
 					"height": height, 
 				}, 
 				this.onAnnounceMissedCreated.bind(this, name, number, picture, contact, Messages.missedMessage, timeStamp),
-				'popupalert');
+				'dashboard');
 		}
 
 	},
@@ -434,7 +434,7 @@ var AnnouncerAssistant = Class.create({
 		this.appControl.showBanner(
 			{
 				"messageText": message,
-				"soundClass": 'notifications'
+				"soundClass": ''
 			}, 
 			
 			{
@@ -786,7 +786,7 @@ var AnnouncerAssistant = Class.create({
 					lightweight: true,
 					name: stageName, 
 					"height": height, 
-					soundclass: "alerts"
+					soundclass: "none"
 				}, 
 				this.onAnnounceDisconnectAbnormalCreated.bind(this, name, callState.number, label, picture, message, callerDetails),
 				'popupalert');
@@ -858,7 +858,7 @@ var AnnouncerAssistant = Class.create({
 		this.appControl.showBanner(
 			{
 				messageText: message,
-				soundClass: 'notifications'
+				soundClass: ''
 			}, 
 			{
 				prefill: callbackNumber
@@ -892,7 +892,7 @@ var AnnouncerAssistant = Class.create({
 					lightweight: true,
 					name: stageName, 
 					"height": height, 
-					soundclass: "alerts"
+					soundclass: "none"
 				}, 
 				this.onAnnounceDialFailCreated.bind(this, message),
 				'popupalert');
