diff --git a/usr/palm/applications/com.palm.app.phone/phonePopups/sources/IncomingCallImpl.js b/usr/palm/applications/com.palm.app.phone/phonePopups/sources/IncomingCallImpl.js
index af46e6d..ad317f7 100644
--- a/usr/palm/applications/com.palm.app.phone/phonePopups/sources/IncomingCallImpl.js
+++ b/usr/palm/applications/com.palm.app.phone/phonePopups/sources/IncomingCallImpl.js
@@ -333,7 +333,7 @@ enyo.kind({
 	    if (response && response.type == "inductive") {
 	        if (this.exposed && this.puckConnected === true && response.connected === false) {	        
 			enyo.job("answerCallIfStillOffPuck", enyo.bind(this, function() {
-			            this.answerIfStillOffPuck();
+			            //this.answerIfStillOffPuck();
 			}),0.750);
 	        }
 	        this.puckConnected = response.connected;
