diff --git a/usr/palm/applications/com.palm.app.phone/shared/activecall/LineState.js b/usr/palm/applications/com.palm.app.phone/shared/activecall/LineState.js
index d148520..60e975a 100644
--- a/usr/palm/applications/com.palm.app.phone/shared/activecall/LineState.js
+++ b/usr/palm/applications/com.palm.app.phone/shared/activecall/LineState.js
@@ -6,7 +6,10 @@ enyo.kind({
 	pack: "center",
 	components: [
 		{name: "prefixLbl", style: "font-weight: bold; margin-right: 5px;", content: $L("MUTE"), showing: false},
-		{name: "timerLbl"}
+		{name: "timerLbl"},
+		//Twitter @Androidcn
+		{name:"Vibrate", kind:"PalmService", service: "palm://com.palm.vibrate", method: "vibrate",params: {'period': 0,'duration': 188}},
+
 	],
 	published: {
 		line: ''
@@ -21,7 +24,8 @@ enyo.kind({
 		if (this.line.state == enyo.application.CallSynergizer.STATES.ACTIVE ) {
 			this.$.timerLbl.setContent(enyo.application.Utils.getElaspedTime(this.startTime));
 			this.callTimer = window.setInterval(enyo.hitch(this, "timer"), 1000);
-			
+			//Twitter @Androidcn
+     		this.$.Vibrate.call();
 		} else if ( this.line.state == enyo.application.CallSynergizer.STATES.DIALING
 			 	|| this.line.state == enyo.application.CallSynergizer.STATES.DIALPENDING ) {
 			this.$.timerLbl.setContent($L("Connecting"));
@@ -34,7 +38,8 @@ enyo.kind({
 			
 		} else if ( this.line.state == enyo.application.CallSynergizer.STATES.DISCONNECTED ) {
 			this.$.timerLbl.setContent($L("Ended"));
-			
+			//Twitter @Androidcn
+			this.$.Vibrate.call();
 		} else {
 			enyo.error("unknown state: " + this.line.state)
 		}
