
--- .orig/usr/palm/applications/com.palm.app.phone/app/controllers/dialpad-assistant.js
+++ /usr/palm/applications/com.palm.app.phone/app/controllers/dialpad-assistant.js
@@ -634,7 +634,7 @@
 	numberClick: function(event, key){
 		
 		if (key) {
-			TelephonyCommands.sendDTMF(key, true);
+//			TelephonyCommands.sendDTMF(key, true);
 			this.formatAndUpdateDialString(key,event);
 			if (this.dialString.length == 1 && this.speedDialString.length == 0) {
 				QDLogger.log( "DialpadAssistant#numberClick", "about to lookup speed dial");
@@ -929,7 +929,7 @@
 		if (this.isValidToneKey(this.pressedKey)
 			&& this.pressedKey !== Mojo.Char.backspace
 			&& (!this.appAssistant.lookupOnKeyPress || this.keyMapOpt.get(evt.keyIdentifier))) {
-			TelephonyCommands.sendDTMF(this.pressedKey, true);
+//			TelephonyCommands.sendDTMF(this.pressedKey, true);
 		}
 		
 		Event.stop(evt);
