diff --git a/usr/palm/applications/com.palm.app.phone/app/controllers/calllog-assistant.js b/usr/palm/applications/com.palm.app.phone/app/controllers/calllog-assistant.js
index 22f5fb4..b224f76 100644
--- a/usr/palm/applications/com.palm.app.phone/app/controllers/calllog-assistant.js
+++ b/usr/palm/applications/com.palm.app.phone/app/controllers/calllog-assistant.js
@@ -213,6 +213,8 @@ var CalllogAssistant = Class.create({
 				name: "contact",
 				label: isKnown ? $L("View Contact") : $L("Add to Contacts")
 			}});
+			
+			viewContact += '<div class="palm-row" x-mojo-touch-feedback="momentary" name="dolookup" displayName="asd"><div class="palm-row-wrapper"><div class="title contact-title truncating-text">Anrufer identifizieren (search.ch)</div></div></div>';
 		} else {
 			viewContact = "";
 		}
@@ -322,6 +324,33 @@ var CalllogAssistant = Class.create({
 						TelephonyCommands.contactsLaunchWithPseudocard(itemModel.recentcall_address.addr, itemModel.recentcall_address.service);
 					}
 					return;
+				
+				case "dolookup":
+					//Mojo.Log.error("TelSearch Lookup");
+					//Mojo.Log.error("itemModel.recentcall_address.addr = " + itemModel.recentcall_address.addr);
+
+					// Launch the Browser with tel.search.ch reverse lookup
+					TelSearchURL = "http://tel.search.ch?was=" + itemModel.recentcall_address.addr; 
+			
+					this.controller.serviceRequest("palm://com.palm.applicationManager", {
+					   method: "open",
+					   parameters:  {
+					       id: 'com.palm.app.browser',
+					       params: {
+						   target: TelSearchURL
+					       }
+					   }
+					 }); 
+					
+					//this.controller.serviceRequest('palm://com.palm.applicationManager', {
+					//	method: 'launch',
+					//	parameters: {
+					//		id: "com.whitepages.mobile",
+					//		params:{action: "search", scene: "reverse", search_number: itemModel.recentcall_address.addr, autosearch:true}
+					//	},
+					return;
+					
+					
 				}
 			target = target.parentNode;
 		}
