diff --git a/usr/palm/applications/com.palm.app.phone/app/controllers/activecall-assistant.js b/usr/palm/applications/com.palm.app.phone/app/controllers/activecall-assistant.js
index bab675d..3cd2a4a 100644
--- a/usr/palm/applications/com.palm.app.phone/app/controllers/activecall-assistant.js
+++ b/usr/palm/applications/com.palm.app.phone/app/controllers/activecall-assistant.js
@@ -139,6 +139,10 @@ var ActivecallAssistant = Class.create({
 		var appMenuModel = {
 			visible: true,
             items: [{
+                label: $L('Call History'),
+                command: 'callhistory'
+		 },
+		 {
                 label: $L('Sounds & Ringtones'),
                 command: 'ringtones'
             }]
@@ -153,6 +157,9 @@ var ActivecallAssistant = Class.create({
 	handleCommand: function(event) {
 		if (event.type == Mojo.Event.command) {
 			switch (event.command) {
+				case 'callhistory':
+					this.appAssistant.showCallLog();
+					break;
 				case 'ringtones':
 					MenuController.showRingtones(this.controller.stageController);
 					break;
diff --git a/usr/palm/applications/com.palm.app.phone/app/controllers/calllog-assistant.js b/usr/palm/applications/com.palm.app.phone/app/controllers/calllog-assistant.js
index 2cb234a..5defabc 100644
--- a/usr/palm/applications/com.palm.app.phone/app/controllers/calllog-assistant.js
+++ b/usr/palm/applications/com.palm.app.phone/app/controllers/calllog-assistant.js
@@ -199,7 +199,12 @@ var CalllogAssistant = Class.create({
 			// manually deal with back; always return to dialpad even if there's
 			// no dialpad scene yet created
 			QDLogger.log( "CalllogAssistant#handleCommand", "back!");
-			this.appAssistant.showDialpad();
+			if (this.telListener.callExists() == true) {
+				this.appAssistant.showActiveCall();
+			}
+			else {
+				this.appAssistant.showDialpad();
+			}
 			Event.stop(event);
 			
 		} else if (event.type == Mojo.Event.command) {
@@ -249,7 +254,12 @@ var CalllogAssistant = Class.create({
 	},
     
     switchToDialpad: function() {
-       this.appAssistant.showDialpad();
+	if (this.telListener.callExists() == true) {
+		this.appAssistant.showActiveCall();
+	}
+	else {
+       	this.appAssistant.showDialpad();
+	}
     },
 	
     onAllCallsTap: function(force) {
