diff --git a/usr/palm/applications/com.palm.app.phone/app/controllers/calllog-assistant.js b/usr/palm/applications/com.palm.app.phone/app/controllers/calllog-assistant.js
index a7d2920..033af2a 100644
--- a/usr/palm/applications/com.palm.app.phone/app/controllers/calllog-assistant.js
+++ b/usr/palm/applications/com.palm.app.phone/app/controllers/calllog-assistant.js
@@ -213,6 +213,8 @@ var CalllogAssistant = Class.create({
 				name: "contact",
 				label: isKnown ? $L("View Contact") : $L("Add to Contacts")
 			}});
+			
+			viewContact += '<div class="palm-row" x-mojo-touch-feedback="momentary" name="wplookup" displayName="asd"><div class="palm-row-wrapper"><div class="title contact-title truncating-text">Look up on WhitePages</div></div></div>';
 		} else {
 			viewContact = "";
 		}
@@ -322,6 +324,45 @@ var CalllogAssistant = Class.create({
 						TelephonyCommands.contactsLaunchWithPseudocard(itemModel.recentcall_address.addr, itemModel.recentcall_address.service);
 					}
 					return;
+				
+				case "wplookup":
+					//Mojo.Log.error("WhitePages Lookup");
+					//Mojo.Log.error("itemModel.recentcall_address.addr = " + itemModel.recentcall_address.addr);
+					this.controller.serviceRequest('palm://com.palm.applicationManager', {
+						method: 'launch',
+						parameters: {
+							id: "com.whitepages.mobile",
+							params:{action: "search", scene: "reverse", search_number: itemModel.recentcall_address.addr, autosearch:true}
+						},
+						onSuccess:function(){
+							Mojo.Log.error("launched just fine.");
+						}.bind(this),
+						onFailure:function(){
+							this.controller.showAlertDialog({
+								onChoose: function(value){
+									if (value=="open"){
+										this.controller.serviceRequest('palm://com.palm.applicationManager', {
+											method:'open',
+											parameters:{
+												target: "http://developer.palm.com/appredirect/?packageid=com.whitepages.mobile"
+											}
+										});
+					
+									}
+								},
+								preventCancel: false,
+								title: $L("WhitePages App Not Installed"),
+								message: $L("Sorry! WhitePages is not installed on your device. You can download this app from the App Catalog (Free)."),
+								choices:[
+									{label:$L('Nevermind'), value:"ok", type:'dismissal'},
+									{label:$L('Open in App Catalog'), value:"open", type:'affirmative'}
+								]
+							});
+						}.bind(this)
+					});
+					return;
+					
+					
 				}
 			target = target.parentNode;
 		}
