diff --git a/usr/palm/applications/com.palm.app.phone/app/models/TelephonyEventListener.js b/usr/palm/applications/com.palm.app.phone/app/models/TelephonyEventListener.js
index 3d17f47..20bbce5 100644
--- a/usr/palm/applications/com.palm.app.phone/app/models/TelephonyEventListener.js
+++ b/usr/palm/applications/com.palm.app.phone/app/models/TelephonyEventListener.js
@@ -724,7 +724,12 @@ var TelephonyEventListener = Class.create({
 		// wait to show incoming popup until the contact has been decorated
 		future = this.incomingPending.contact.decorated();
 		future.then(this, function() {
-			this.announcer.announceIncoming(callState.affectedCallId, future.result, this.isConnected());
+			if (contact.name === 'Do Not Answer'){
+				this.flagCallRejected(callState.affectedCallId);
+				TelephonyCommands.disconnect(callState.affectedCallId);
+			} else {
+				this.announcer.announceIncoming(callState.affectedCallId, future.result, this.isConnected());
+			}
 			return true; // continue
 		});
 		
