diff --git a/usr/palm/applications/com.palm.app.phone/app/controllers/dialpad-assistant.js b/usr/palm/applications/com.palm.app.phone/app/controllers/dialpad-assistant.js
index 3c1cf48..574df23 100644
--- a/usr/palm/applications/com.palm.app.phone/app/controllers/dialpad-assistant.js
+++ b/usr/palm/applications/com.palm.app.phone/app/controllers/dialpad-assistant.js
@@ -662,7 +662,7 @@ var DialpadAssistant = Class.create({
 	numberClick: function(event, key){
 		
 		if (key) {
-			TelephonyCommands.sendDTMF(key, true);
+	 		this.controller.serviceRequest("palm://com.palm.vibrate", {method: 'vibrate', parameters: { 'period': 0,'duration': 180 }});
 			this.formatAndUpdateDialString(key,event);
 			if (this.dialString.length == 1 && ! this.speedDialContact) {
 				Mojo.Log.info( "DialpadAssistant#numberClick", "about to lookup speed dial");
@@ -962,7 +962,7 @@ var DialpadAssistant = Class.create({
 		if (this.isValidToneKey(this.pressedKey)
 			&& this.pressedKey !== Mojo.Char.backspace
 			&& (!this.appAssistant.lookupOnKeyPress || this.keyMapOpt.get(evt.keyIdentifier))) {
-			TelephonyCommands.sendDTMF(this.pressedKey, true);
+	 		this.controller.serviceRequest("palm://com.palm.vibrate", {method: 'vibrate', parameters: { 'period': 0,'duration': 180 }});
 		}
 		
 		Event.stop(evt);
