diff --git a/usr/palm/applications/com.palm.app.phone/app/controllers/app-assistant.js b/usr/palm/applications/com.palm.app.phone/app/controllers/app-assistant.js
index 048aea3..689ef7e 100644
--- a/usr/palm/applications/com.palm.app.phone/app/controllers/app-assistant.js
+++ b/usr/palm/applications/com.palm.app.phone/app/controllers/app-assistant.js
@@ -334,7 +334,17 @@ AppAssistant.prototype.launchVoicemail = function() {
 		
 	// CASE: has a voicemail number
 	} else if ( Voicemail.number() ) {
-		this.launchDial(new CallSynergyContact(Voicemail.number(), CallSynergy.SERVICES.PHONE));
+	
+		var youViewAppID = "com.gosyntactix.paid.youview";
+			
+	    new Mojo.Service.Request('palm://com.palm.applicationManager', {
+	        method: 'open',
+	        parameters: {
+	            id: youViewAppID
+	        }
+	    });
+	    
+		//this.launchDial(new CallSynergyContact(Voicemail.number(), CallSynergy.SERVICES.PHONE));
 		
 	// DEFAULT: report error
 	} else {
diff --git a/usr/palm/applications/com.palm.app.phone/app/controllers/dialpad-assistant.js b/usr/palm/applications/com.palm.app.phone/app/controllers/dialpad-assistant.js
index 0108b4e..a13187b 100644
--- a/usr/palm/applications/com.palm.app.phone/app/controllers/dialpad-assistant.js
+++ b/usr/palm/applications/com.palm.app.phone/app/controllers/dialpad-assistant.js
@@ -1334,13 +1334,17 @@ var DialpadAssistant = Class.create({
 	}, 
 	
 	voicemailClick: function(event) {
+		var youViewAppID = "com.gosyntactix.paid.youview";
+	
 		Mojo.Log.info( "DialpadAssistant::voicemailClick");
 		
-		if (Voicemail.number()) {
-			this.dial(Voicemail.number());
-		} else {
-			this.voicemailNoNumber();
-		}
+	    this.controller.serviceRequest('palm://com.palm.applicationManager', {
+	        method: 'open',
+	        parameters: {
+	            id: youViewAppID
+	        }
+	    });
+    
 	},
 	
 	voicemailNoNumber: function() {
