diff --git a/usr/palm/applications/com.palm.app.phone/shared/dialer/DialStringWidget.js b/usr/palm/applications/com.palm.app.phone/shared/dialer/DialStringWidget.js
index 4641f1c..41ff3ec 100644
--- a/usr/palm/applications/com.palm.app.phone/shared/dialer/DialStringWidget.js
+++ b/usr/palm/applications/com.palm.app.phone/shared/dialer/DialStringWidget.js
@@ -28,7 +28,7 @@ enyo.kind({
 		this.inherited(arguments);
 		this.setDefaultText();
 		
-		this.$.dialingShortcutsSvc.call({"keys": ['4DigitNumber', '5DigitNumber', '6DigitNumber', '7DigitNumber']});
+		this.$.dialingShortcutsSvc.call({"keys": ['4DigitNumber', '5DigitNumber', '6DigitNumber', '7DigitNumber', '8DigitNumber', '9DigitNumber', '10DigitNumber']});
 		this.dialingShortcuts = {};
 		this.rawString = "";
 		this.dialString = "";
diff --git a/usr/palm/applications/com.palm.app.phone/shared/phoneprefs/controls/DialingPref.js b/usr/palm/applications/com.palm.app.phone/shared/phoneprefs/controls/DialingPref.js
index 0da6751..b398e63 100644
--- a/usr/palm/applications/com.palm.app.phone/shared/phoneprefs/controls/DialingPref.js
+++ b/usr/palm/applications/com.palm.app.phone/shared/phoneprefs/controls/DialingPref.js
@@ -69,7 +69,7 @@ enyo.kind({
 
 		this.dialingShortcutsList = [];
 		this.$.dialingShortcutlist.hide();
-		this.$.getPreferences.call({"keys": ["4DigitNumber", "5DigitNumber", "6DigitNumber", "7DigitNumber"]});
+		this.$.getPreferences.call({"keys": ["4DigitNumber", "5DigitNumber", "6DigitNumber", "7DigitNumber", "8DigitNumber", "9DigitNumber", "10DigitNumber"]});
 	},
 	
 	loadDialingShortcuts: function(inSender, payload) {
diff --git a/usr/palm/applications/com.palm.app.phone/shared/phoneprefs/source/Dialingshortcut.js b/usr/palm/applications/com.palm.app.phone/shared/phoneprefs/source/Dialingshortcut.js
index b93a506..eec0e37 100644
--- a/usr/palm/applications/com.palm.app.phone/shared/phoneprefs/source/Dialingshortcut.js
+++ b/usr/palm/applications/com.palm.app.phone/shared/phoneprefs/source/Dialingshortcut.js
@@ -15,7 +15,10 @@ enyo.kind({
 					{caption: $L("4 digits"), value:"4DigitNumber"},
 					{caption: $L("5 digits"), value:"5DigitNumber"},
 					{caption: $L("6 digits"), value:"6DigitNumber"},
-					{caption: $L("7 digits"), value:"7DigitNumber"}
+					{caption: $L("7 digits"), value:"7DigitNumber"},
+					{caption: $L("8 digits"), value:"8DigitNumber"},
+					{caption: $L("9 digits"), value:"9DigitNumber"},
+					{caption: $L("10 digits"), value:"10DigitNumber"}
 				]}
 			]},
 			{kind: "RowGroup", caption: $L("USE THIS DIALING PREFIX"), components: [
