--- a/usr/palm/applications/com.palm.app.phone/app/models/DialStringParser.js
+++ b/usr/palm/applications/com.palm.app.phone/app/models/DialStringParser.js
@@ -62,6 +62,7 @@
 		'72346': DialAction.RADIO,
 		'633': DialAction.OFF,
 		'7436581': DialAction.MASK, // SID6581: i adore my 64
+		'477': DialAction.GPS,
 		'387': DialAction.LAUNCHEXTERNAL, // FTP
 		'66338': DialAction.LAUNCHEXTERNAL // ONDEV
 	},
@@ -72,7 +73,6 @@
 	    '2539' : DialAction.AKEY,
 	    '3836': DialAction.EVDO,
 	    '7738': DialAction.PREV,
-	    '477': DialAction.GPS,
 		'786': DialAction.RTN,
 		'33284': DialAction.DEBUG,
 		'3282': DialAction.DATA,
