diff --git a/usr/palm/applications/com.palm.app.phone/app/controllers/incomingcall-assistant.js b/usr/palm/applications/com.palm.app.phone/app/controllers/incomingcall-assistant.js
index 6b6ea1a..3ec741c 100644
--- a/usr/palm/applications/com.palm.app.phone/app/controllers/incomingcall-assistant.js
+++ b/usr/palm/applications/com.palm.app.phone/app/controllers/incomingcall-assistant.js
@@ -406,7 +406,7 @@ var IncomingcallAssistant = Class.create({
 	// corrects for jitter in puck messages
 	answerIfStillOffPuck: function() {
 		if (this.puckConnected === false) {
-			this.answerCall();
+			//this.answerCall();
 		} else {
 			QDLogger.log( "IncomingcallAssistant#answerIfStillOffPuck", "back on puck");
 		}
