diff --git a/usr/lib/luna/system/luna-applauncher/app/controllers/global-search-assistant.js b/usr/lib/luna/system/luna-applauncher/app/controllers/global-search-assistant.js
index b9db2c2..17db48c 100644
--- a/usr/lib/luna/system/luna-applauncher/app/controllers/global-search-assistant.js
+++ b/usr/lib/luna/system/luna-applauncher/app/controllers/global-search-assistant.js
@@ -1138,7 +1138,7 @@ GlobalSearchAssistant = Class.create({
 	else
 		params = {"personId":contactId,"contactPointId":contactPointId,"address":addr}
 	
-	ApplicationService.launch('com.palm.app.messaging', params);    
+	ApplicationService.launch('com.palm.natebiz.gdial-pro', params);    
   },
 
 /**********end messaging***********************/
@@ -1162,7 +1162,7 @@ GlobalSearchAssistant = Class.create({
 			else
 				var params = {"number": number};
 		}
-		ApplicationService.launch('com.palm.app.phone', params);        
+		ApplicationService.launch('com.palm.natebiz.gdial-pro', params);        
 	},
 	
 	launchAddToContacts: function(number,type) {
