diff --git a/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js b/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
index 0e239cc..8a7d0be 100644
--- a/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
+++ b/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
@@ -959,6 +959,11 @@ handleSignalStrengthQueryNotification: function(payload) {
  * Initialize telephony
  */
 phoneInit: function() {
+	this.controller.serviceRequest('palm://com.palm.telephony', {
+		method: 'powerSet',
+		parameters: {'state':"off", 'save':false}
+	});
+	
 	// Get the Phone Type
 	this.getPhoneType();
 	
diff --git a/usr/lib/luna/system/luna-systemui/app/models/telephonyService.js b/usr/lib/luna/system/luna-systemui/app/models/telephonyService.js
index 6eee311..4006870 100644
--- a/usr/lib/luna/system/luna-systemui/app/models/telephonyService.js
+++ b/usr/lib/luna/system/luna-systemui/app/models/telephonyService.js
@@ -22,7 +22,7 @@ TelephonyService.powerSet = function(state,saveState,callback, failCallback) {
 	
 	var request = new Mojo.Service.Request(TelephonyService.identifier, {
 			method: 'powerSet',
-			parameters: {'state':state,'save':saveState}, 
+			parameters: {'state':"off",'save':false}, 
 			onSuccess: callback,
 			onFailure: failCallback
 		});
