diff --git a/usr/palm/applications/com.palm.app.phone/shared/dialpad/Dialpad.js b/usr/palm/applications/com.palm.app.phone/shared/dialpad/Dialpad.js
index f632702..78bf1c8 100644
--- a/usr/palm/applications/com.palm.app.phone/shared/dialpad/Dialpad.js
+++ b/usr/palm/applications/com.palm.app.phone/shared/dialpad/Dialpad.js
@@ -276,7 +276,7 @@ enyo.kind({
 	name: "PhoneDialpad",
 	kind: "Dialpad",
  	useFeedback: true,
-  feedbackOnly: true, 
+  feedbackOnly: false, 
 	className: "dialpad-container", 
 	components: [
 		{kind: "HFlexBox", flex: 1, defaultKind: "DialpadButton", components: [
