diff --git a/usr/palm/applications/com.palm.app.phone/app/controllers/app-assistant.js b/usr/palm/applications/com.palm.app.phone/app/controllers/app-assistant.js
index 3ba467f..108c335 100644
--- a/usr/palm/applications/com.palm.app.phone/app/controllers/app-assistant.js
+++ b/usr/palm/applications/com.palm.app.phone/app/controllers/app-assistant.js
@@ -193,20 +193,14 @@ AppAssistant.prototype.handleLaunch = function(params){
 			if(PalmSystem.isMinimal)
 				return 
 			else
-				this.showDialpad();
+				this.showCallLog(); // start with call log instead of dialpad
 		} 
     } else {
 		
 		if (!(this.telephonyEventListener.isPendingOrActive())) {
 			if (this.initialLaunch === false) {
 				QDLogger.log( "handleLaunch", "no args, no calls: focusing stage");
-				var appControl = Mojo.Controller.getAppController();
-    			var stageController = appControl.getStageController(this.stageName);
-					
-				if (!stageController) {
-					this.showDialpad();
-				}
-				
+				this.showCallLog(); // just show call log here
 				this.focusStage();
 			} else {
 				QDLogger.log( "handleLaunch", "initial launch, so doing nothing");
@@ -483,7 +477,7 @@ AppAssistant.prototype.showCallLog = function(contactId, number, type, keepHidde
 		this.sceneStack.push(sceneName);
 		sc.pushScene({
 			"name": sceneName,
-			transition: Mojo.Transition.crossFade
+			transition: Mojo.Transition.none // we don't need the transition
 		}, type);
 	}.bind(this);
    
@@ -502,6 +496,9 @@ AppAssistant.prototype.showCallLog = function(contactId, number, type, keepHidde
 			if(type == 'missed')
 				stageController.delegateToSceneAssistant("updateMissedCalls")
 			stageController.activate();
+				pushLogScene(stageController); // get back the the all calls list
+				return;
+			stageController.activate();
 		}
 	}
 };
@@ -718,7 +715,7 @@ AppAssistant.prototype.closePhoneStage = function() {
 AppAssistant.prototype.hidePhoneStage = function(win) {
 	QDLogger.log("AppAssistant#hidePhoneStage");
 	win.PalmSystem.hide();
-	this.showDialpad(null, true /* keep hidden */);
+	this.showCallLog("", "", "", true /* keep hidden */); // just in case; show call log here
 	this.hidden = true;
 }
 
@@ -752,14 +749,7 @@ AppAssistant.prototype.popActiveCall = function(params) {
 	if (stageController !== undefined) {
 		this.sceneStack = [];
 		stageController.popScenesTo();
-		if (this.backToCalllog) {
-			if(this.calllogType == "missed")
-				this.showCallLog("", "", "missed", true);
-			else
-				this.showCallLog("", "", "", true);
-		}
-		else 
-			this.showDialpad(null, true);
+		this.showCallLog("", "", "", true); // return to call log
 	} else {
 		QDLogger.log( "popActiveCall", "undefined stage");
 	}
