diff --git a/usr/palm/applications/com.palm.app.phone/app/controllers/calllog-assistant.js b/usr/palm/applications/com.palm.app.phone/app/controllers/calllog-assistant.js
index 2cb234a..3574694 100644
--- a/usr/palm/applications/com.palm.app.phone/app/controllers/calllog-assistant.js
+++ b/usr/palm/applications/com.palm.app.phone/app/controllers/calllog-assistant.js
@@ -444,7 +444,7 @@ var CalllogAssistant = Class.create({
 			}
 			
 			var type = this.type != "missed" ? "" : this.type;			
-			this.appAssistant.launchDial(number, type, contact);
+			this.appAssistant.showDialpad({"action": "fill", "number": TelephonyCommands.normalizeNumber(number)});		
 		}
 	},
 	
