diff --git a/usr/palm/applications/com.palm.app.phone/app/controllers/activecall-assistant.js b/usr/palm/applications/com.palm.app.phone/app/controllers/activecall-assistant.js
index 148d8ba..e98daf9 100644
--- a/usr/palm/applications/com.palm.app.phone/app/controllers/activecall-assistant.js
+++ b/usr/palm/applications/com.palm.app.phone/app/controllers/activecall-assistant.js
@@ -24,6 +24,17 @@ var ActivecallAssistant = Class.create({
 	},
 	
 	initialize: function(params){
+//BEGIN Slider Patch block 1 of 3
+	  	var request = new Mojo.Service.Request('palm://com.palm.keys/switches', {
+                method: 'status',
+                parameters: {"get" : "slider"},
+				onFailure: function() {
+						//QDLogger.log("Slider: Failed to get slider status");
+					},
+				onSuccess: this.getSliderPos.bind(this)
+        		});
+		this.sliderSubscriptionx = TelephonyCommands.sliderSubscribe(this.onSliderEventDuringCall.bind(this));
+//END Slider Patch block 1 of 3
 	    this.appAssistant = Mojo.Controller.getAppController().assistant;
 		this.eventListener = this.appAssistant.telephonyEventListener;
 		this.announcer = this.appAssistant.announcer;
@@ -462,6 +473,41 @@ break;
 		}
 
     },
+
+//BEGIN Slider Patch block 2 of 3
+    getSliderPos:  function(response) {
+		if (response.key === "slider") {
+			this.sliderPos = response.state;
+		}
+    },
+	onSliderEventDuringCall: function(response) {
+		if (response && response.key && response.state) {
+		} else {
+			return;
+		}
+		
+		var scenarios = this.appAssistant.telephonyEventListener.getAvailableAudioScenarios();
+		var frontSpeaker = false;
+		
+        if(this.audioActiveProfile == "phone_front_speaker") frontSpeaker = true;
+        
+		if (response.key == "slider"&&frontSpeaker) {
+			if (this.sliderPos=='up'&&response.state == "down") {
+				//First, "tap" each Disconnect Button.
+				for (i = 0; i < 3; i++) {
+					if (this.controller.get("disc_button_" + i)) {
+						Mojo.Event.send(this.controller.get("disc_button_" + i), Mojo.Event.tap);
+					}
+				}
+				//In case the Disconnect Buttons don't work, or there are calls not disconnected, just call the function to disconnect all calls.
+				//The reason we don't just do this one function is that the screen redraw is not pretty.
+				this.eventListener.disconnectAllCalls();
+			}
+			this.sliderPos = response.state;
+		} 
+	},
+//END Slider Patch block 2 of 3
+    
 	
 	// sets up pause/wait digit handling 
 	handlePauseWaitDigits: function(callStateMessage) {
@@ -1302,6 +1348,9 @@ break;
 		this.controller.document.removeEventListener("mouseup", this.boundEndContinuousDTMF);
         this.controller.document.removeEventListener('keydown', this.boundStartDTMFHardKey);
         this.controller.document.removeEventListener('keyup', this.boundEndDTMFHardKey);
+//BEGIN Slider Patch block 3 of 3
+		this.sliderSubscriptionx.cancel();	//remove slider listener
+//END Slider Patch block 3 of 3
 		
 		// drop puck subscription
 		this.puckSubscription.cancel();
