diff --git a/usr/palm/applications/com.palm.app.phone/shared/dialpad/Dialpad.js b/usr/palm/applications/com.palm.app.phone/shared/dialpad/Dialpad.js
index f632702..87d82f1 100644
--- a/usr/palm/applications/com.palm.app.phone/shared/dialpad/Dialpad.js
+++ b/usr/palm/applications/com.palm.app.phone/shared/dialpad/Dialpad.js
@@ -36,6 +36,7 @@ enyo.kind({
 		]},
 		{name: "client", flex: 1, layoutKind: "VFlexLayout"},
 		
+		{name: "hapticFeedback", kind: enyo.PalmService, service: "palm://com.palm.vibrate/", method: "vibrate", onSuccess: "", onFailure:""},
 		{name: "playDialTone", kind: enyo.PalmService, service: "palm://com.palm.audio/dtmf/", method: "playDTMF", onSuccess: "", onFailure:""},
 		{name: "stopDialTone", kind: enyo.PalmService, service: "palm://com.palm.audio/dtmf/", method: "stopDTMF", onSuccess: "", onFailure:""}
 	],
@@ -211,6 +212,10 @@ enyo.kind({
 				 		"oneshot": true,
 				 		"feedbackOnly": this.feedbackOnly
 		                        });//sendDTMF
+				this.$.hapticFeedback.call({
+						"period": 0,
+						"duration": 75
+					});//vibrate
                         } else {
 	                        var held = false;
 				enyo.application.CallSynergizer.callDtmf(tone, held);
@@ -229,6 +234,10 @@ enyo.kind({
 				 		"oneshot": false,
 				 		"feedbackOnly": this.feedbackOnly
 		                        });
+				this.$.hapticFeedback.call({
+						"period": 0,
+						"duration": 75
+					});//vibrate
 			} else {
 				var held = true;
 				enyo.application.CallSynergizer.callDtmf(tone, held);
