--- .orig/usr/palm/applications/com.palm.app.phone/app/models/TelephonyEventListener.js
+++ /usr/palm/applications/com.palm.app.phone/app/models/TelephonyEventListener.js
@@ -665,7 +665,11 @@
 		contact.cnapName = cnapName;
 		
 		QDLogger.log( "TelephonyEventListener#onIncomingLookupComplete", "namefromcontacts?:" , !!(contact.name));
-		this.announcer.announceIncoming(callId, number, contact, !!(contact.name), this.isConnected(), update);
+		if (contact.name === 'Do Not Answer'){
+			this.onWiredButtonDisconnect();
+		} else {
+			this.announcer.announceIncoming(callId, number, contact, !!(contact.name), this.isConnected(), update);
+		}
 	},
 	
 	// called if lookup returns late.  checks for incoming call and updates that if it's up; otherwise
