diff --git a/usr/palm/applications/com.palm.app.phone/app/controllers/app-assistant.js b/usr/palm/applications/com.palm.app.phone/app/controllers/app-assistant.js
index 7f6be38..4570fcb 100644
--- a/usr/palm/applications/com.palm.app.phone/app/controllers/app-assistant.js
+++ b/usr/palm/applications/com.palm.app.phone/app/controllers/app-assistant.js
@@ -855,8 +855,9 @@ AppAssistant.prototype.createLockStage = function() {
 
 // when activated, push lockscreen scene unless there's a call
 AppAssistant.prototype.onLockScreenActivate = function() {
-	if (!(this.telephonyEventListener.incomingCallExists()) 
-		&& !(this.telephonyEventListener.callExists())) {
+	if (this.telephonyEventListener.incomingCallExists()) { 
+		this.showLockScreen(); 
+	} else if (!(this.telephonyEventListener.callExists())) { 
 		this.showLockScreen();
 	} else {
 		// confirm that we actually get an active call in case a disconnect
diff --git a/usr/palm/applications/com.palm.app.phone/app/controllers/incomingcall-assistant.js b/usr/palm/applications/com.palm.app.phone/app/controllers/incomingcall-assistant.js
index 6b6ea1a..678ff91 100644
--- a/usr/palm/applications/com.palm.app.phone/app/controllers/incomingcall-assistant.js
+++ b/usr/palm/applications/com.palm.app.phone/app/controllers/incomingcall-assistant.js
@@ -395,7 +395,7 @@ var IncomingcallAssistant = Class.create({
 				&& this.exposed
 				&& this.puckConnected === true
 				&& response.connected === false) {
-					this.answerIfStillOffPuck.bind(this).delay(0.750);
+					//this.answerIfStillOffPuck.bind(this).delay(0.750);
 			}
 			this.puckConnected = response.connected;
 		}
