diff --git a/usr/palm/applications/com.palm.app.phone/app/views/incomingcall/details-stamp.html b/usr/palm/applications/com.palm.app.phone/app/views/incomingcall/details-stamp.html
index 3a99fcd..a786a60 100644
--- a/usr/palm/applications/com.palm.app.phone/app/views/incomingcall/details-stamp.html
+++ b/usr/palm/applications/com.palm.app.phone/app/views/incomingcall/details-stamp.html
@@ -2,7 +2,7 @@
   <div class="incoming-background-wrapper">
     <div class="incoming-picture-container" id="contact_picture_frame">
       <img id="picture" class="incoming-picture-square" src="images/contacts-unknown-icon-large.png" /> 
-      <img class="incoming-picture-overlay" src="images/incoming-photo-overlay.png"></div>
+      <img class="incoming-picture-overlay" src="images/incoming-photo-overlay.png">
     </div>
     <div class="incoming-call-text stamp-size" x-mojo-tap-highlight="immediate">
       <div class="incoming-container" id="contact_info">
@@ -18,4 +18,9 @@
       </div>
     </div>
   </div>
+ </div>
+ <div style="width:320px;height:30px;margin:0px;overflow:hidden;position:relative;left:1px;top:-115px;">
+    <div style="font-size:30pt;color:yellow;line-height:30px;white-space:nowrap;text-overflow:clip;text-align:center;">
+       #{displayName}
+    </div>
 </div>
\ No newline at end of file
diff --git a/usr/palm/applications/com.palm.app.phone/app/views/incomingcall/details-unknown-with-location.html.rej b/usr/palm/applications/com.palm.app.phone/app/views/incomingcall/details-unknown-with-location.html.rej
new file mode 100644
index 0000000..ff9a42a
--- /dev/null
+++ b/usr/palm/applications/com.palm.app.phone/app/views/incomingcall/details-unknown-with-location.html.rej
@@ -0,0 +1,31 @@
+--- usr/palm/applications/com.palm.app.phone/app/views/incomingcall/details-unknown-with-location.html
++++ usr/palm/applications/com.palm.app.phone/app/views/incomingcall/details-unknown-with-location.html
+@@ -4,7 +4,6 @@
+       <div class="incoming-container" id="contact_info">
+ 		<!-- TODO NOV-100619 -->
+         <div id="unknownlabel" class="incoming-number" x-mojo-loc=''>
+-          #{-displayNumber}
+         </div>
+         <div id="contact_number" class="incoming-name truncating-text">
+           #{-locationFormatted}
+@@ -7,7 +6,6 @@
+           #{-displayNumber}
+         </div>
+         <div id="contact_number" class="incoming-name truncating-text">
+-          #{-locationFormatted}
+         </div>
+       </div>
+     </div>
+@@ -12,4 +10,12 @@
+       </div>
+     </div>
+   </div>
++       <div style="width:320px;height:75px;margin:0px;overflow:hidden;position:relative;left:-35px;top:10px;">
++          <div style="font-size:30pt;color:yellow;line-height:35px;white-space:nowrap;text-overflow:clip;text-align:center;">
++               #{-displayNumber}
++          </div>
++          <div style="font-size:26pt;color:yellow;line-height:35px;white-space:nowrap;text-overflow:clip;text-align:center;">
++              #{-locationFormatted}
++          </div>
++       </div>
+ </div>
\ No newline at end of file
diff --git a/usr/palm/applications/com.palm.app.phone/app/views/incomingcall/details-unknown-with-location.html.~1~ b/usr/palm/applications/com.palm.app.phone/app/views/incomingcall/details-unknown-with-location.html.~1~
new file mode 100644
index 0000000..e69de29
diff --git a/usr/palm/applications/com.palm.app.phone/app/views/incomingcall/details-unknown-with-name.html.rej b/usr/palm/applications/com.palm.app.phone/app/views/incomingcall/details-unknown-with-name.html.rej
new file mode 100644
index 0000000..5b79970
--- /dev/null
+++ b/usr/palm/applications/com.palm.app.phone/app/views/incomingcall/details-unknown-with-name.html.rej
@@ -0,0 +1,20 @@
+--- usr/palm/applications/com.palm.app.phone/app/views/incomingcall/details-unknown-with-name.html
++++ usr/palm/applications/com.palm.app.phone/app/views/incomingcall/details-unknown-with-name.html
+@@ -4,7 +4,6 @@
+       <div class="incoming-container" id="contact_info">
+ 		<!-- TODO NOV-100619 -->
+         <div id="unknownlabel" class="incoming-number" x-mojo-loc=''>
+-          #{-displayName}
+         </div>
+         <div id="contact_number" class="incoming-name truncating-text">
+           #{-displayNumber}
+@@ -12,4 +11,9 @@
+       </div>
+     </div>
+   </div>
++       <div style="width:320px;height:75px;margin:0px;overflow:hidden;position:relative;left:-35px;top:10px;">
++          <div style="font-size:30pt;color:yellow;line-height:35px;white-space:nowrap;text-overflow:clip;text-align:center;">
++          #{-displayName}
++          </div>
++       </div>
+ </div>
\ No newline at end of file
diff --git a/usr/palm/applications/com.palm.app.phone/app/views/incomingcall/details-unknown-with-name.html.~1~ b/usr/palm/applications/com.palm.app.phone/app/views/incomingcall/details-unknown-with-name.html.~1~
new file mode 100644
index 0000000..e69de29
