diff --git a/usr/palm/applications/com.palm.app.phone/phoneApp/source/AppMenu.js b/usr/palm/applications/com.palm.app.phone/phoneApp/source/AppMenu.js
index c106b35..d83185b 100644
--- a/usr/palm/applications/com.palm.app.phone/phoneApp/source/AppMenu.js
+++ b/usr/palm/applications/com.palm.app.phone/phoneApp/source/AppMenu.js
@@ -10,6 +10,11 @@ enyo.kind({
 	},
 	components: [
 		{name: "editMenu", kind: enyo.EditMenu, selectAllDisabled: true, className: 'edit-label', onCopy: "doCopy", onPaste: "doPaste", showShortcuts:true},
+		{name: "exportCallHistory", caption: $L("Export Call History"), className: 'item-label', onclick: "exportCallHistory", showing: false},
+		{name: "exportCallHistoryPrompt", kind: "DialogPrompt", 
+			message: $L("History exported to 'calllog.csv'."),
+			acceptButtonCaption: $L("OK"),
+			cancelButtonCaption: null},
 		{name: "clearCallHistory", caption: $L("Clear Call History"), className: 'item-label', onclick: "clearCallHistory", showing: false},
 		{name: "checkSkypeCredit", caption: $L("Check Skype Credit"), className: 'item-label', onclick: "checkSkypeCredit", showing: false},
 		{name: "voicemailGreeting", caption: $L("Voicemail Greeting"), className: 'item-label', onclick: "voicemailGreeting", showing: false},
@@ -31,6 +36,8 @@ enyo.kind({
 			onAccept: "clearCallHistoryConfirm",
 		},
 		{name:"purgeCallHistory", kind:"PalmService", service: enyo.palmServices.database, method: "del"},
+		{name:"queryCallHistory", kind:"PalmService", service: enyo.palmServices.database, method: "find"},
+		{name:"filemgr", kind:"PalmService", service: "palm://ca.canucksoftware.filemgr", method: "write"},
 		{name:"noVoicemailNumberPrompt", kind: "NoVoicemailNumberPrompt",},
 		{name:"noVoicemailGreetingNumberPrompt", kind: "DialogPrompt", 
 			message: $L("Unable to find voicemail greeting number."),
@@ -104,6 +111,63 @@ enyo.kind({
 			"query":{"from":"com.palm.phonecall:1"},"purge":false
 		});
 	},
+	exportCallHistory: function() {
+		var line = "Type\tDate/Time\tDuration [s]\tFrom\tService\t\To\tName\tService\n";
+		this.$.filemgr.call({"file":"/media/internal/calllog.csv","str":line,"append":false},{onSuccess:"_onExportCallHistorySuccess"});
+
+		this.$.queryCallHistory.call({
+			"query":{"from":"com.palm.phonecall:1"},
+			"watch":false,
+			"count":true
+		}, {
+			onSuccess: "_onGetCallHistorySuccess"
+		});
+	},
+	_onGetCallHistorySuccess: function(inSender, inResponse) {
+		var line = "";
+
+		var results = inResponse.results;
+		for (var i=0; i<results.length; i++) {
+			enyo.error("response " + enyo.json.stringify(results[i]));
+			var secs = Math.round(results[i].duration / 1000);
+			line += results[i].type + "\t";
+			line += new enyo.g11n.DateFmt({format: "medium"}).format(new Date(results[i].timestamp)) + "\t";
+			line += this.formatNumber(Math.floor(secs / 60)) + ":" + this.formatNumber(secs % 60) + "\t";
+			line += results[i].from.addr + "\t";
+			line += results[i].from.service + "\t";
+			for (var j=0; j<results[i].to.length; j++) {
+				line += results[i].to[j].addr + "\t";
+				line += results[i].to[j].name + "\t";
+				line += results[i].to[j].service + "\t";
+			}
+			line += "\n";
+			//this.$.filemgr.call({"file":"/media/internal/agendaz.html","str":line,"append":true});
+		}
+
+		this.$.filemgr.call({"file":"/media/internal/calllog.csv","str":line,"append":true},{onSuccess:"_onExportCallHistorySuccess"});
+
+		enyo.error("inResponse.next " + inResponse.next);
+		if (inResponse.next != undefined) {
+			this.$.queryCallHistory.call({
+				"query":{"from":"com.palm.phonecall:1"},
+				"watch":false,
+				"count":true,
+				"page":inResponse.next,
+			}, {
+				onSuccess: "_onGetCallHistorySuccess"
+			});
+		} else {
+			this.$.exportCallHistoryPrompt.open();
+		}
+	},
+	_onExportCallHistorySuccess: function(inSender, inResponse) {
+	},
+	formatNumber: function(number) {
+		if (("" + number).length < 2) {
+			return "0" + number;
+		}
+		return number;
+	},
 	checkSkypeCredit: function() {
         if(this.accounts == undefined) {
     		this.accounts = this.createComponent({name: "accounts", kind: "DBModel.Accounts",}, {"owner": this});
@@ -112,6 +176,7 @@ enyo.kind({
 	},
 	openMenu : function(bShowCallHistory, isVoicemail) {		
 		this.$.clearCallHistory.setShowing(bShowCallHistory);
+		this.$.exportCallHistory.setShowing(bShowCallHistory);
 		this.$.checkSkypeCredit.setShowing(enyo.application.Cache.hasSkypeAcct === true);
 		if (isVoicemail) {
 			var carrierName = enyo.application.VoicemailService.getCarrierName();
