diff --git a/usr/palm/applications/com.palm.app.phone/app/controllers/calllog-assistant.js b/usr/palm/applications/com.palm.app.phone/app/controllers/calllog-assistant.js
index a7d2920..014a5e6 100644
--- a/usr/palm/applications/com.palm.app.phone/app/controllers/calllog-assistant.js
+++ b/usr/palm/applications/com.palm.app.phone/app/controllers/calllog-assistant.js
@@ -311,7 +311,8 @@ var CalllogAssistant = Class.create({
 					if ( isSMS ) {
 						TelephonyCommands.messagingLaunch(target.getAttribute('addr'), target.getAttribute('service'), itemModel.recentcall_address.personId);
 					} else {
-						this.appAssistant.launchDial(new CallSynergyContact(target.getAttribute('addr'), target.getAttribute('service'), itemModel.recentcall_address.personId));
+						//this.appAssistant.launchDial(new CallSynergyContact(target.getAttribute('addr'), target.getAttribute('service'), itemModel.recentcall_address.personId));
+						UIStateMachine.enter('dialpad_card',{"action": "fill", "number": TelephonyCommands.normalizeNumber(target.getAttribute('addr'))}, false, true);
 					}
 					return;
 					
@@ -529,7 +530,8 @@ var CalllogAssistant = Class.create({
 		}
 		// CASE: make call
 		else if ( CallSynergyContact.canBeCalled(model.recentcall_address.service, model.recentcall_address.addr) ) {
-			this.appAssistant.launchDial(new CallSynergyContact(model.recentcall_address.addr, model.recentcall_address.service, model.recentcall_address.personId, undefined, true));
+			//this.appAssistant.launchDial(new CallSynergyContact(model.recentcall_address.addr, model.recentcall_address.service, model.recentcall_address.personId, undefined, true));
+			UIStateMachine.enter('dialpad_card',{"action": "fill", "number": TelephonyCommands.normalizeNumber(model.recentcall_address.addr)}, false, true);
 		}
     }
 });
\ No newline at end of file
