// enable ondemand cpu powersaving option
Index: /etc/event.d/powersave-cpu-scaling
===================================================================
--- /dev/null
+++ /etc/event.d/powersave-cpu-scaling
@@ -0,0 +1,13 @@
+# Enables cpu scaling
+
+start on stopped finish
+stop on runlevel [!2]
+
+console none
+
+script
+    echo "500000" > /sys/devices/system/cpu/cpu0/cpufreq/scaling_max_freq
+    echo "250000" > /sys/devices/system/cpu/cpu0/cpufreq/scaling_min_freq
+    echo "ondemand" > /sys/devices/system/cpu/cpu0/cpufreq/scaling_governor
+    echo "30" > /sys/devices/system/cpu/cpu0/cpufreq/ondemand/up_threshold
+end script
