// enable SmartReflex powersaving option
Index: /etc/event.d/powersave-smartreflex
===================================================================
--- /dev/null
+++ /etc/event.d/powersave-smartreflex
@@ -0,0 +1,23 @@
+# -*- mode: shell-script; -*-
+description "SmartReflex"
+author "Alex Markson"
+version 1.0
+
+start on stopped finish
+stop on runlevel [!2]
+
+console none
+
+script
+
+# SmartReflex
+# "SmartReflex™ driver allows for auto-compensation of VDD1 and
+# VDD2 voltages (around the voltages specified by current OPP)
+# by analyzing the silicon characteristics, temperature, voltage etc"
+#
+# Enable SmartReflex
+echo -n 1 > /sys/power/sr_vdd1_autocomp
+echo -n 1 > /sys/power/sr_vdd2_autocomp
+
+end script
+
