diff --git a/etc/event.d/cpu-scaling b/etc/event.d/cpu-scaling
new file mode 100644
index 0000000..e012ed5
--- /dev/null
+++ b/etc/event.d/cpu-scaling
@@ -0,0 +1,13 @@
+# Enables cpu scaling, This version is just a "turbo" mode. No batt savings.
+
+start on stopped finish
+stop on runlevel [!2]
+
+console none
+
+script
+    echo "600000" > /sys/devices/system/cpu/cpu0/cpufreq/scaling_max_freq
+    echo "500000" > /sys/devices/system/cpu/cpu0/cpufreq/scaling_min_freq
+    echo "ondemand" > /sys/devices/system/cpu/cpu0/cpufreq/scaling_governor
+    echo "30" > /sys/devices/system/cpu/cpu0/cpufreq/ondemand/up_threshold
+end script
