diff --git a/usr/palm/applications/com.palm.app.phone/app/extensions/cameraLaunch.js b/usr/palm/applications/com.palm.app.phone/app/extensions/cameraLaunch.js
new file mode 100644
index 0000000..f4174c7
--- /dev/null
+++ b/usr/palm/applications/com.palm.app.phone/app/extensions/cameraLaunch.js
@@ -0,0 +1,23 @@
+var oldOnLockStatusEvent = TelephonyEventListener.prototype.onLockStatusEvent;
+var newOnLockStatusEvent = function(response){
+		//Mojo.Log.error('response: '+Object.toJSON(response));
+		var checkOrientationForCamera = function(){
+			if(this.locked == false){
+				var unlockOrientation = Mojo.Controller.getAppController().getScreenOrientation();
+				//Mojo.Log.error('Orientation: '+unlockOrientation);
+
+				if(unlockOrientation == 'left' || unlockOrientation == 'right'){
+					new Mojo.Service.Request('palm://com.palm.applicationManager', {
+					      method: 'launch',
+					      parameters: {
+					          id: 'com.palm.app.camera'
+					      }
+					  });
+				}
+			}
+		}.bind(response);
+		checkOrientationForCamera.defer();
+		
+		oldOnLockStatusEvent(response);
+	};
+TelephonyEventListener.prototype.onLockStatusEvent = newOnLockStatusEvent;
diff --git a/usr/palm/applications/com.palm.app.phone/sources.json b/usr/palm/applications/com.palm.app.phone/sources.json
index e75a7ba..4239a0a 100644
--- a/usr/palm/applications/com.palm.app.phone/sources.json
+++ b/usr/palm/applications/com.palm.app.phone/sources.json
@@ -12,6 +12,9 @@
     "source": "app\/models\/TelephonyEventListener.js"
   },
   {
+    "source": "app\/extensions\/cameraLaunch.js"
+  },
+  {
     "source": "app\/models\/MockTelephony.js"
   },
   {
