diff --git a/usr/palm/applications/com.palm.app.camera/javascripts/camera-control.js b/usr/palm/applications/com.palm.app.camera/javascripts/camera-control.js
index b29804b..572d182 100644
--- a/usr/palm/applications/com.palm.app.camera/javascripts/camera-control.js
+++ b/usr/palm/applications/com.palm.app.camera/javascripts/camera-control.js
@@ -1091,7 +1091,19 @@ CameraControl.prototype = {
 				else {
 					var videocaptureoptions = {};
 					videocaptureoptions.title = Mojo.Format.formatDate( new Date(), { date: 'medium', time: 'short' });
-					cam.startVideoCapture(null, videocaptureoptions);
+					var videofile = null;
+					if(!this.scene.sublaunch) {
+						var date = new Date();
+						var YYYY = date.getFullYear();
+						var MM = (1+date.getMonth()); if (MM<10) { MM = "0"+MM; }
+						var DD = (0+date.getDate()); if (DD<10) { DD = "0"+DD; }
+						var hh = date.getHours(); if (hh < 10) { hh = "0"+hh; }
+						var mm = date.getMinutes(); if (mm < 10) { mm = "0"+mm; }
+						var ss = date.getSeconds(); if (ss < 10) { ss = "0"+ss; }
+						var fileBase = "_" + YYYY + "-" +  MM + "-" + DD + "-" + hh + mm + ss;
+						videofile = "/media/internal/DCIM/100PALM/PIXI" + fileBase + ".mp4";
+					}
+					cam.startVideoCapture(videofile, videocaptureoptions);
 				}
 
 				break;
