diff --git a/usr/palm/applications/com.palm.app.camera/app/controllers/capture-assistant.js b/usr/palm/applications/com.palm.app.camera/app/controllers/capture-assistant.js
index 498f869..2a45448 100644
--- a/usr/palm/applications/com.palm.app.camera/app/controllers/capture-assistant.js
+++ b/usr/palm/applications/com.palm.app.camera/app/controllers/capture-assistant.js
@@ -354,7 +354,7 @@ CaptureAssistant = Class.create({
 //		llog(evtStr);
 
 		// Space bar also takes a picture.
-		if (Mojo.Char.spaceBar == event.originalEvent.keyCode) {
+		if (Mojo.Char.spaceBar|| Mojo.Char.metaKey|| Mojo.Char.home == event.originalEvent.keyCode) {
 			if (event.type === Mojo.Event.keydown) {
 				if (!this.captureBurst) {
 					this.captureBurst = true;
