diff --git a/usr/palm/applications/com.palm.app.camera/app/controllers/capture-assistant.js b/usr/palm/applications/com.palm.app.camera/app/controllers/capture-assistant.js
index 5d8756b..7a80431 100644
--- a/usr/palm/applications/com.palm.app.camera/app/controllers/capture-assistant.js
+++ b/usr/palm/applications/com.palm.app.camera/app/controllers/capture-assistant.js
@@ -102,13 +102,13 @@ CaptureAssistant.prototype = {
 			}
 
 			// Always start in still capture mode.  
-			this._handleCaptureToggle(mode);
+			// this._handleCaptureToggle(mode);
 
 			// Last capture mode is stored in prefs.  
 			// The following line would restore the last capture mode from preferences. 
 			// Some people might prefer to remember what the last capture mode was.  
 			// Enough clues?!
-			//this._handleCaptureToggle(this.cameraControl.prefs[CameraControl.PREFS.CAPTURE_MODE], true/*isFromPrefs*/);
+			this._handleCaptureToggle(this.cameraControl.prefs[CameraControl.PREFS.CAPTURE_MODE], true/*isFromPrefs*/);
 
 		}
 
