diff --git a/usr/palm/applications/com.palm.app.camera/javascripts/camera-control.js b/usr/palm/applications/com.palm.app.camera/javascripts/camera-control.js
index f9d2140..c2f557f 100644
--- a/usr/palm/applications/com.palm.app.camera/javascripts/camera-control.js
+++ b/usr/palm/applications/com.palm.app.camera/javascripts/camera-control.js
@@ -935,7 +935,7 @@ CameraControl.prototype = {
 				}
 				else {
 					var videocaptureoptions = {};
-					videocaptureoptions.title = Mojo.Format.formatDate( new Date(), { date: 'medium', time: 'short' });
+					videocaptureoptions.title = Mojo.Format.formatDate( new Date(), { date: 'short', time: 'short' });
 					cam.startVideoCapture(null, videocaptureoptions);
 				}
 
