diff --git a/media/cryptofs/apps/usr/palm/applications/com.palm.app.camera/app.js b/media/cryptofs/apps/usr/palm/applications/com.palm.app.camera/app.js
index f64c01b..96e48aa 100644
--- a/media/cryptofs/apps/usr/palm/applications/com.palm.app.camera/app.js
+++ b/media/cryptofs/apps/usr/palm/applications/com.palm.app.camera/app.js
@@ -2,6 +2,51 @@
 	CameraApplication: 2
 	Camera application controller, contains UI components, Media capture object, State Machine, DbService and UI Manager
 */
+
+
+// Hack: can't figure out how to add new files to unified diff. Prepending to app.js. UDC may be broken.
+/*
+	KeysService:
+	Provides volume button camera triggers
+*/
+enyo.kind({
+	name: "KeysService",
+	description: "Key Listener",
+	kind: "enyo.Component",
+	components: [
+		{ name: "volButtonStatus", kind: "PalmService", service: "palm://com.palm.keys/audio/", method: "status", onSuccess: "handleVolButtonPress", onFailure: "serviceError", subscribe: true }
+	],
+	create:function()
+	{
+		this.inherited(arguments);
+		this.$.volButtonStatus.call();
+	},
+	destroy:function()
+	{
+		this.$.volButtonStatus.destroy();
+	},
+	handleVolButtonPress:function(inSender, inResponse)
+	{
+		if (inResponse && inResponse.key && inResponse.state && inResponse.state == "down")
+		{
+			if (inResponse.key == "volume_up")
+			{
+				this.owner.DisplayCameraMode();
+				this.owner._handleImageCaptureOnClickEvent();
+			}
+			else if (inResponse.key == "volume_down")
+			{
+				this.owner.DisplayVideoMode();
+				this.owner._handleImageCaptureOnClickEvent();
+			}
+		}
+	},
+	serviceError:function(inSender, inResponse)
+	{
+		console.error("KeysService: Failed to subscribe keys: " + enyo.json.stringify(inResponse));
+	}	
+});
+
 enyo.kind({
 	name:"enyo.CameraApplication",
 	kind:enyo.VFlexBox,
@@ -136,6 +181,9 @@ enyo.kind({
 		/* Cross app support; needed to marshal the filename back to the callee*/
 		{ 
 			name:"crossAppResult", kind: "CrossAppResult" 
+		},
+		{
+			name:"keysService", kind: "KeysService"
 		}
 	],	
 	/**************************************************************
@@ -1013,6 +1061,16 @@ enyo.kind({
 
 		this.$.UIManager._setDefaultLaunchView();
 	},	
+	DisplayCameraMode:function()
+	{
+		this.$.cameraInstance.currentCameraMode = this.$.config.cameraMode.camera;
+		this.$.UIManager._setDefaultLaunchView();
+	},
+	DisplayVideoMode:function()
+	{
+		this.$.cameraInstance.currentCameraMode = this.$.config.cameraMode.video;
+		this.$.UIManager._setDefaultLaunchView();
+	},
 	_initializeMediaCaptureInstance:function()
 	{
 		/* Media Capture object - use to drive camera actions */
