diff --git a/usr/palm/applications/com.palm.app.camera/app/controllers/capture-assistant.js b/usr/palm/applications/com.palm.app.camera/app/controllers/capture-assistant.js
index 5d8756b..38530a2 100644
--- a/usr/palm/applications/com.palm.app.camera/app/controllers/capture-assistant.js
+++ b/usr/palm/applications/com.palm.app.camera/app/controllers/capture-assistant.js
@@ -1421,9 +1421,9 @@ CaptureAssistant.prototype = {
 		
 		// Set a scrim right now, don't bother waiting for the timeout.
 		this.controller.get('camera-icon').className = "camera-icon";
-		this.controller.sceneElement.addClassName('preview-off');
-		this.controller.sceneElement.style.background = "#404040";
-		this.setScrimImage();
+		//this.controller.sceneElement.addClassName('preview-off');
+		this.controller.sceneElement.style.background = "transparent";
+		//this.setScrimImage();
 		
 		//Let fileNotifyd rest
 		this.stopFileTimer();
@@ -1434,9 +1434,10 @@ CaptureAssistant.prototype = {
 
 			try{
 				if (!this.isFocused()){
-					this.cameraControl.freezePreview(this.vstabAvailable);
+					//this.cameraControl.freezePreview(this.vstabAvailable);
 	
-					GeoControl.stopTracking();
+					//GeoControl.stopTracking();
+					this.cameraControl.resumePreview(this.vstabAvailable);
 				}
 			}
 			catch(e){
@@ -1547,8 +1548,8 @@ CaptureAssistant.prototype = {
 		this.stopFileTimer();
 		
 		try {
-			this.cameraControl.freezePreview(this.vstabAvailable);
-			GeoControl.stopTracking();
+			//this.cameraControl.freezePreview(this.vstabAvailable);
+			//GeoControl.stopTracking();
 			
 		} catch (e1) {
 			llog("Exception thrown: "+Object.toJSON(e1));
