diff --git a/usr/palm/applications/com.palm.app.camera/app/controllers/capture-assistant.js b/usr/palm/applications/com.palm.app.camera/app/controllers/capture-assistant.js
index d38dff5..654ab69 100644
--- a/usr/palm/applications/com.palm.app.camera/app/controllers/capture-assistant.js
+++ b/usr/palm/applications/com.palm.app.camera/app/controllers/capture-assistant.js
@@ -984,6 +984,7 @@ CaptureAssistant.prototype = {
 
 		this._idleCheckTimer = setTimeout(this._triggerIdleHandler,  this.cameraControl.prefs[CameraControl.PREFS.IDLEMAX]);
 		this._powerService.stopScreenDim();
+		this.sysVolMute(true);
 	},
 	/**
 	 * Stop checking for capture activity and re-enable the screen dim.
@@ -999,6 +1000,7 @@ CaptureAssistant.prototype = {
 			clearTimeout(this._idleCheckTimer);
 			delete this._idleCheckTimer;
 		}
+		this.sysVolMute(false);
 	},
 	/**
 	 * Indicate that an event has occured and reset the screen-off check time.
@@ -1009,6 +1011,7 @@ CaptureAssistant.prototype = {
 			clearTimeout(this._idleCheckTimer);
 			this._idleCheckTimer = setTimeout(this._triggerIdleHandler,  this.cameraControl.prefs[CameraControl.PREFS.IDLEMAX]);
 
+		this.sysVolMute(true);
 		}
 		else{
 			this.startIdleCheck();
@@ -1026,6 +1029,13 @@ CaptureAssistant.prototype = {
 
 		this._powerService.allowScreenDim();
 	},
+
+	sysVolMute: function(mute){
+		this.controller.serviceRequest('palm://com.palm.audio/system', {
+			method:'setMuted',
+			parameters:{muted:mute}
+		});
+	},
 			
 
 	/**
