diff --git a/usr/palm/applications/com.palm.app.camera/app/controllers/capture-assistant.js b/usr/palm/applications/com.palm.app.camera/app/controllers/capture-assistant.js
index 498f869..050fdc6 100644
--- a/usr/palm/applications/com.palm.app.camera/app/controllers/capture-assistant.js
+++ b/usr/palm/applications/com.palm.app.camera/app/controllers/capture-assistant.js
@@ -872,6 +872,7 @@ CaptureAssistant = Class.create({
 
 		this._idleCheckTimer = setTimeout(this._triggerIdleHandler,  this.cameraControl.prefs[CameraControl.PREFS.IDLEMAX]);
 		this._powerService.stopScreenDim();
+		this.sysVolMute(true);
 	},
 	/**
 	 * Stop checking for capture activity and re-enable the screen dim.
@@ -887,6 +888,7 @@ CaptureAssistant = Class.create({
 			clearTimeout(this._idleCheckTimer);
 			delete this._idleCheckTimer;
 		}
+		this.sysVolMute(false);
 	},
 	/**
 	 * Indicate that an event has occured and reset the screen-off check time.
@@ -897,6 +899,7 @@ CaptureAssistant = Class.create({
 			clearTimeout(this._idleCheckTimer);
 			this._idleCheckTimer = setTimeout(this._triggerIdleHandler,  this.cameraControl.prefs[CameraControl.PREFS.IDLEMAX]);
 
+		this.sysVolMute(true);
 		}
 		else{
 			this.startIdleCheck();
@@ -915,6 +918,12 @@ CaptureAssistant = Class.create({
 		this._powerService.allowScreenDim();
 	},
 			
+	sysVolMute: function(mute){
+		this.controller.serviceRequest('palm://com.palm.audio/system', {
+			method:'setMuted',
+			parameters:{muted:mute}
+		});
+	},
 
 	/**
 	 * Click handler for flashmode button.
