diff --git a/usr/palm/applications/com.palm.app.camera/app/controllers/capture-assistant.js b/usr/palm/applications/com.palm.app.camera/app/controllers/capture-assistant.js
index 2952de4..f11abf1 100644
--- a/usr/palm/applications/com.palm.app.camera/app/controllers/capture-assistant.js
+++ b/usr/palm/applications/com.palm.app.camera/app/controllers/capture-assistant.js
@@ -79,6 +79,10 @@ CaptureAssistant = Class.create({
 			this._handleFlashButton();
 		}.bind(this));
 
+		this.controller.get('shutterSoundButtonState').observe(Mojo.Event.tap, function(){
+			this._handleShutterSoundButton();
+		}.bind(this));
+
 		this.onKeyPressHandler = this.onKeyPress.bindAsEventListener(this);
 		Mojo.listen(this.controller.sceneElement, Mojo.Event.keydown, this.onKeyPressHandler);
 		Mojo.listen(this.controller.sceneElement, Mojo.Event.keyup, this.onKeyPressHandler);
@@ -140,6 +144,11 @@ CaptureAssistant = Class.create({
 			this.setFlashState(prefFlashState);
 		}		
 
+		if (undefined === this.shutterSoundState){
+			var prefShutterSoundState = this.cameraControl.prefs[CameraControl.PREFS.SOUNDS];
+			this.setShutterSoundState(prefShutterSoundState);
+		}	
+
 		try {
 			var initialOrientation = PalmSystem.screenOrientation;
 			this.orientationChanged(initialOrientation);
@@ -629,6 +638,45 @@ CaptureAssistant = Class.create({
 	},
 
 	/**
+	 * Click handler for the shutter sound button.
+	 *  
+	 * @param {Event} event  The click event
+	 */
+	_handleShutterSoundButton: function(event){
+		if (this.shutterSoundState == 'enabled'){
+			this.shutterSoundState = 'disabled';
+		}
+		else {
+			this.shutterSoundState = 'enabled';
+		}
+
+		this.setShutterSoundState(this.shutterSoundState);
+	},
+		
+	setShutterSoundState: function(state){
+		if ((state != 'disabled') && (state != 'enabled')) {
+			/*
+			var elemC = this.controller.get('sagar_console');
+			if (elemC) {
+				elemC.innerHTML = state;
+			}
+			*/
+			llog("Requested shutter sound state is out of range ("+state+")");
+			return;
+		}
+
+		this.shutterSoundState = state;
+		CameraPrefs.updatePref(this.cameraControl.prefs, CameraControl.PREFS.SOUNDS, state);
+	
+		if (this.shutterSoundState == 'enabled') {
+			this.controller.get('shutterSoundButtonState').className = "shutter-sound-button shutter-sound-on";
+		}
+		else if (this.shutterSoundState == 'disabled') {
+			this.controller.get('shutterSoundButtonState').className = "shutter-sound-button shutter-sound-off";
+		}
+	},
+
+	/**
 	 * Handler for click of photoroll button.
 	 * 
 	 * The photoroll scene will be pushed on.
diff --git a/usr/palm/applications/com.palm.app.camera/app/views/capture/capture-scene.html b/usr/palm/applications/com.palm.app.camera/app/views/capture/capture-scene.html
index a586aba..e8243d2 100644
--- a/usr/palm/applications/com.palm.app.camera/app/views/capture/capture-scene.html
+++ b/usr/palm/applications/com.palm.app.camera/app/views/capture/capture-scene.html
@@ -14,6 +14,8 @@
 		<div id="captureSpinner" class="capture-spinner" x-mojo-element="Spinner"></div>		
 		<div class="capture-button" id="captureButton">
 		</div>
+		<div class="shutter-sound-button shutter-sound-off" id="shutterSoundButtonState">
+		</div>
 		<div class="flash-button flash-off" id="flashButtonState">
 		</div>
 	</div>
@@ -23,7 +25,7 @@
 			
 		</div>
 	</div>
-	<!-- <div id="sagar_console" style="z-index: 100; position: fixed; top: 10px; left: 10px; width: 90px; height: 20px; background-color: #000000; color: #8080ff; font-family: sans-serif; font-size: 12px; display: none;" >Hello!</div> -->
+	<!--<div id="sagar_console" style="z-index: 100; position: fixed; top: 10px; left: 10px; width: 90px; height: 20px; background-color: #000000; color: #8080ff; font-family: sans-serif; font-size: 12px; display: none;" >Hello!</div> -->
 	<div id="gpsMeter" style="z-index: 100; position: fixed; top: 10px; right: 10px; width: 90px; height: 20px; background-color: #a0a0a0; display: none" ></div> -->
 	
 	<div id="palmScrim" class="palm-scrim" style="z-index: 1001; display: none;">
diff --git a/usr/palm/applications/com.palm.app.camera/stylesheets/camera.css b/usr/palm/applications/com.palm.app.camera/stylesheets/camera.css
index 9480ffa..f0e242d 100644
--- a/usr/palm/applications/com.palm.app.camera/stylesheets/camera.css
+++ b/usr/palm/applications/com.palm.app.camera/stylesheets/camera.css
@@ -119,6 +119,28 @@ body.palm-default {
 		z-index: 20;
 }
 
+.shutter-sound-button,
+.shutter-sound-button.shutter-sound-off {
+		width: 50px;
+		height: 50px;
+		background: url(../images/menu-shutter-sound.png) 0 0 no-repeat;
+		position: absolute;
+		left: 188px;
+		top: 15px;
+}
+
+.shutter-sound-button.shutter-sound-on:active {
+		background-position: 0px -50px;
+}
+
+.shutter-sound-button.shutter-sound-on {
+		background-position: 0px -100px;
+}
+
+.shutter-sound-button.shutter-sound-on:active {
+		background-position: 0px -150px;
+}
+
 /* Photo Roll */
 
 .photo-roll {
