diff --git a/usr/palm/applications/com.palm.app.notes/app/controllers/grid-assistant.js b/usr/palm/applications/com.palm.app.notes/app/controllers/grid-assistant.js
index 215c0b1..dbf9f0b 100644
--- a/usr/palm/applications/com.palm.app.notes/app/controllers/grid-assistant.js
+++ b/usr/palm/applications/com.palm.app.notes/app/controllers/grid-assistant.js
@@ -2,7 +2,7 @@
 /*global Globalization, Class: false, Mojo: false, $L: false, Template: false, Event: false, Form: false, Weave: false, GridList: false, Note: false*/
 
 /* Copyright 2009 Palm, Inc.  All rights reserved. */
-
+var count = 1;
 var GridAssistant = Class.create(
 {
 	$widget_noteGrid: null,
@@ -56,7 +56,11 @@ var GridAssistant = Class.create(
 	{
 		//this.controller.listen(this.controller.get("filterField"), Mojo.Event.filter, this.filter.bind(this), true);
 		this.controller.listen(this.controller.get("filterField"), Mojo.Event.filterImmediate, this.$filter_filterField.bind(this), true);
-		if (newMemoText){
+		if (count == 1){
+			this._handleNewNote(newMemoText);	
+                        count++;			
+		}
+                if (newMemoText){
 			this._handleNewNote(newMemoText);				
 		}
 	},
