diff --git a/usr/palm/applications/com.palm.app.notes/app/controllers/note-assistant.js b/usr/palm/applications/com.palm.app.notes/app/controllers/note-assistant.js
index fd50861..6346ca2 100644
--- a/usr/palm/applications/com.palm.app.notes/app/controllers/note-assistant.js
+++ b/usr/palm/applications/com.palm.app.notes/app/controllers/note-assistant.js
@@ -239,6 +239,52 @@ var NoteAssistant = Class.create(
         this.data.note.color = newColor;
     },
     
+    $menu_timestamp_memo_cmd: function(event)
+	{
+	     var myTimeStamp = new Date ();
+	            
+				var nowyear = myTimeStamp.getFullYear();
+				var nowmonth = myTimeStamp.getMonth();
+				var nowday = myTimeStamp.getDate();
+				var nowhour = myTimeStamp.getHours();
+				var nowminutes = myTimeStamp.getMinutes();
+				
+				 nowmonth = (nowmonth +1);
+				 if (nowminutes <= 9) 
+				 {
+   				 nowminutes = "0" + nowminutes;
+   					 }
+                if (nowhour > 12) 
+                {
+     			 nowhour = nowhour - 12;
+    			  var ending = " p.m.";
+   				 } 
+   				 else {
+    			  nowhour = nowhour;
+   				   var ending = " a.m.";
+  				  }
+   			    if (nowhour == 12) {
+   			    var ending = " p.m.";
+   				}
+ 			   if (nowhour == 00) {
+  		         nowhour = "12";
+  			  }
+	
+				
+				
+				  var val = this.controller.get('noteText').mojo.getValue(); 
+				   this.clipB=  val + nowmonth + "/"+ nowday + "/"+ nowyear +" @ "+ nowhour+ ":"+ nowminutes + ending;
+						this.controller.stageController.setClipboard(this.clipB);
+				 Mojo.Controller.getAppController().showBanner(" TimeStamp Added ", {source: 'notification'});
+	        
+	            this.controller.get('noteText').mojo.focus();
+				
+				
+	             PalmSystem.paste();
+	       
+	    
+	},
+	
     _onNoteLoaded: function(note, isnew)
     {
 		Mojo.Log.info("onNoteLoaded: %j", note, isnew);
diff --git a/usr/palm/applications/com.palm.app.notes/app/views/note/note-scene.html b/usr/palm/applications/com.palm.app.notes/app/views/note/note-scene.html
index 38257c8..5510c33 100644
--- a/usr/palm/applications/com.palm.app.notes/app/views/note/note-scene.html
+++ b/usr/palm/applications/com.palm.app.notes/app/views/note/note-scene.html
@@ -8,6 +8,7 @@
 		</div>
 		<div x-mojo-loc="" x-mojo-action="email-memo-cmd">Email memo...</div>
 		<div x-mojo-loc="" x-mojo-action="delete-memo-cmd">Delete</div>
+		<div x-mojo-loc="" x-mojo-action="timestamp-memo-cmd">TimeStamp</div>
 		<div x-mojo-loc="" x-mojo-action="palm-help-cmd">Help</div>
 	</div>
   	<div class="memos-note-backdrop"></div>
