diff --git a/usr/palm/applications/com.palm.app.notes/app/controllers/note-assistant.js b/usr/palm/applications/com.palm.app.notes/app/controllers/note-assistant.js
index e6a5050..bd63035 100644
--- a/usr/palm/applications/com.palm.app.notes/app/controllers/note-assistant.js
+++ b/usr/palm/applications/com.palm.app.notes/app/controllers/note-assistant.js
@@ -134,6 +134,7 @@ var NoteAssistant = Class.create(
 	
 	activate: function()
 	{
+             
 		this.controller.document.addEventListener(Mojo.Event.stageDeactivate, this._saveNote.bind(this), false);
 		this._setupAutoSave(false);
 	},
@@ -276,6 +277,60 @@ var NoteAssistant = Class.create(
         
         this.data.note.color = newColor;
     },
+
+    $menu_timestamp_memo_cmd: function(event)
+	{
+	      var myTimeStamp = new Date ();
+	            
+				var nowyear = myTimeStamp.getFullYear();
+				var nowmonth = myTimeStamp.getMonth();
+				var nowday = myTimeStamp.getDate();
+				var nowhour = myTimeStamp.getHours();
+				var nowminutes = myTimeStamp.getMinutes();
+				
+                                nowmonth = (nowmonth +1);
+
+                                 if (nowminutes <= 9) 
+				 {
+   				 nowminutes = "0" + nowminutes;
+   					 }
+                if (nowhour > 12) 
+                {
+     			 nowhour = nowhour - 12;
+    			  var ending = " p.m.";
+   				 } 
+   				 else {
+    			  nowhour = nowhour;
+   				   var ending = " a.m.";
+  				  }
+   			    if (nowhour == 12) {
+   			    var ending = " p.m.";
+   				}
+ 			   if (nowhour == 00) {
+  		         nowhour = "12";
+  			  }
+	
+				
+				
+				
+				//var myTime = nowmonth + "/"+ nowday + "/"+ nowyear +" @ "+ nowhour+ ":"+ nowminutes + ending;
+			        var val = this.controller.get('noteText').mojo.getValue(); 
+                            //    var val =  this._getHTMLNote()
+                                //  var val = this.$widget_noteText.getValue();
+                                 this.clipB=  val + nowmonth + "/"+ nowday + "/"+ nowyear +" @ "+ nowhour+ ":"+ nowminutes + ending;
+						this.controller.stageController.setClipboard(this.clipB);
+				 Mojo.Controller.getAppController().showBanner(" TimeStamp Added ", {source: 'notification'});
+	          
+              
+
+               this.controller.get('noteText').mojo.focus();
+
+              
+             
+                    PalmSystem.paste();
+				 
+	    
+	},
     
     _onNoteLoaded: function(note, isnew)
     {
diff --git a/usr/palm/applications/com.palm.app.notes/app/views/note/note-scene.html b/usr/palm/applications/com.palm.app.notes/app/views/note/note-scene.html
index 38257c8..5510c33 100644
--- a/usr/palm/applications/com.palm.app.notes/app/views/note/note-scene.html
+++ b/usr/palm/applications/com.palm.app.notes/app/views/note/note-scene.html
@@ -8,6 +8,7 @@
 		</div>
 		<div x-mojo-loc="" x-mojo-action="email-memo-cmd">Email memo...</div>
 		<div x-mojo-loc="" x-mojo-action="delete-memo-cmd">Delete</div>
+		<div x-mojo-loc="" x-mojo-action="timestamp-memo-cmd">TimeStamp</div>
 		<div x-mojo-loc="" x-mojo-action="palm-help-cmd">Help</div>
 	</div>
   	<div class="memos-note-backdrop"></div>
