diff --git a/etc/palm/db/permissions/com.palm.note b/etc/palm/db/permissions/com.palm.note
index 03dfdf4..5c0271c 100644
--- a/etc/palm/db/permissions/com.palm.note
+++ b/etc/palm/db/permissions/com.palm.note
@@ -9,6 +9,13 @@
      },{
          "type": "db.kind",
          "object": "com.palm.note:1",
+         "caller": "info.mobo.*",
+         "operations": {
+             "read": "allow"
+         }
+     },{
+         "type": "db.kind",
+         "object": "com.palm.note:1",
          "caller": "com.palm.app.notes",
          "operations": {
              "create": "allow",
