diff --git a/usr/palm/applications/com.palm.app.notes/app/controllers/grid-assistant.js b/usr/palm/applications/com.palm.app.notes/app/controllers/grid-assistant.js
index d470fb2..82978a4 100644
--- a/usr/palm/applications/com.palm.app.notes/app/controllers/grid-assistant.js
+++ b/usr/palm/applications/com.palm.app.notes/app/controllers/grid-assistant.js
@@ -1,5 +1,5 @@
 /* Copyright 2009 Palm, Inc.  All rights reserved. */
-
+var count =1;
 var GridAssistant = Class.create(
 {
 	$widget_noteGrid: null,
@@ -121,6 +121,14 @@ var GridAssistant = Class.create(
 			text: this.filterString || ""
         });
     },
+
+    activate: function()
+    {
+        if(count ==1 ){
+          this._handleNewNote();
+          count++;
+        }
+    }, 
     
     _handleEditNote: function(id, color)
     {
