disable-charging-alert-sound.patch

Index: /usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
===================================================================
--- .orig/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
+++ /usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
@@ -1264,7 +1264,7 @@ handlePowerNotifications: function(paylo
 				this.controller.showBanner({
 						messageText: batteryalert,
 						icon: "/usr/lib/luna/system/luna-systemui/images/notification-small-charging.png",
-						soundClass: "alerts"
+						soundClass: ""
 					},null, 'chargingAlert'); 				
 			}
 			this.chargingBannerShown = true;			
