diff --git a/usr/lib/luna/system/luna-applauncher/app/controllers/global-search-assistant.js b/usr/lib/luna/system/luna-applauncher/app/controllers/global-search-assistant.js
index 7d94506..1c90d5e 100644
--- a/usr/lib/luna/system/luna-applauncher/app/controllers/global-search-assistant.js
+++ b/usr/lib/luna/system/luna-applauncher/app/controllers/global-search-assistant.js
@@ -48,6 +48,7 @@ GlobalSearchAssistant = Class.create({
 		this.offset=0;
 		this.limit=40;
 		this.URLs = {
+			'keytoss':$L("http://pre.keytoss.com/parse.php5?term="),
 			'google':$L("www.google.com/m/search?client=ms-palm-webOS&channel=iss&q="),
 			'wikipedia':$L("http://en.m.wikipedia.org/wiki/Special:Search?search="),
 			'twitter': $L("http://search.twitter.com/search?q=")
@@ -1656,6 +1657,7 @@ GlobalSearchAssistant = Class.create({
 		this.expandedSearchDrawer.showExpanded = false;
 		this.controller.modelChanged(this.expandedSearchDrawer);
 		this.searchApps.clear();
+		$('keytoss').removeClassName('selected');
 		$('google').removeClassName('selected');
 		$('map').removeClassName('selected');
 		$('wikipedia').removeClassName('selected');
@@ -1761,6 +1763,7 @@ GlobalSearchAssistant = Class.create({
 	}
 	//clear search':
 	$('google').removeClassName('palm-focus');
+	$('keytoss').removeClassName('palm-focus');
 	$('web').removeClassName('palm-focus');
   },
   highlightSelection: function() {
@@ -1819,19 +1822,23 @@ GlobalSearchAssistant = Class.create({
 			break;
 		case 'search':
 			if (this.webDrawer.showWeb == true) {
-				$('google').removeClassName('palm-focus');
+				//$('google').removeClassName('palm-focus');
+				$('keytoss').removeClassName('palm-focus');
 				$('web').addClassName('palm-focus');
 				this.highlightTarget = $('web');
 			}
 			else if(this.expandedSearchDrawer.showExpanded == true){
-				$('google').addClassName('palm-focus');
+				//$('google').addClassName('palm-focus');
+				$('keytoss').addClassName('palm-focus');
 				$('web').removeClassName('palm-focus');
-				this.highlightTarget = $('google');
+				//this.highlightTarget = $('google');
+				this.highlightTarget = $('keytoss');
 			}
 			break;		
 	}
 	if (this.enterKeyActionItem != "search") {
 		$('google').removeClassName('palm-focus');
+		$('keytoss').removeClassName('palm-focus');
 		$('web').removeClassName('palm-focus');
 	}
   },
@@ -1923,7 +1930,8 @@ GlobalSearchAssistant = Class.create({
 			if (this.webDrawer.showWeb == true)
 				this.launchBrowser(this.currentFilter);
 			else {
-				this.launchBrowser(this.URLs['google'] + encodeURIComponent(this.currentFilter));
+				//this.launchBrowser(this.URLs['google'] + encodeURIComponent(this.currentFilter));
+				this.launchBrowser(this.URLs['keytoss'] + encodeURIComponent(this.currentFilter));
 			}
 			break;
 	}
diff --git a/usr/lib/luna/system/luna-applauncher/app/views/global-search/expanded-searches-div.html b/usr/lib/luna/system/luna-applauncher/app/views/global-search/expanded-searches-div.html
index 7bfc200..00cc1cc 100644
--- a/usr/lib/luna/system/luna-applauncher/app/views/global-search/expanded-searches-div.html
+++ b/usr/lib/luna/system/luna-applauncher/app/views/global-search/expanded-searches-div.html
@@ -9,6 +9,11 @@
 </div>
 
 <div id='expanded_searches_drawer' x-mojo-element="Drawer">
+	<div class="palm-row" id="keytoss" name="search-identifier" x-mojo-tap-highlight="persistent">
+		<div class="palm-row-wrapper">
+			<div class="search-keytoss"></div>
+		</div>
+	</div>
 	<div class="palm-row" id="google" name="search-identifier" x-mojo-tap-highlight="persistent">
 		<div class="palm-row-wrapper">
 			<div class="search-google"></div>
diff --git a/usr/lib/luna/system/luna-applauncher/stylesheets/global-search.css b/usr/lib/luna/system/luna-applauncher/stylesheets/global-search.css
index 1f10209..5887f9c 100644
--- a/usr/lib/luna/system/luna-applauncher/stylesheets/global-search.css
+++ b/usr/lib/luna/system/luna-applauncher/stylesheets/global-search.css
@@ -119,6 +119,12 @@ Copyright 2009 Palm, Inc.  All rights reserved.
 		background: url(../images/search-add.png) center center no-repeat;
 }
 
+.palm-group.search .search-keytoss {   
+		width: 100%;
+		height: 52px;       
+		background: url(../images/search-keytoss.png) center center no-repeat;
+}
+
 .palm-group.search .search-google {   
 		width: 100%;
 		height: 52px;       
diff --git a/usr/palm/applications/com.palm.app.browser/app/controllers/urlsearch-controller.js b/usr/palm/applications/com.palm.app.browser/app/controllers/urlsearch-controller.js
index 986f0bb..61d4d0c 100644
--- a/usr/palm/applications/com.palm.app.browser/app/controllers/urlsearch-controller.js
+++ b/usr/palm/applications/com.palm.app.browser/app/controllers/urlsearch-controller.js
@@ -58,6 +58,12 @@ function UrlSearchController(controller){
 
 // Our search providers
 UrlSearchController.SearchProviders = {
+	keytoss: {
+		searchTitle: $L('Search KeyToss'),
+		searchTemplate: $L('KeyToss "#{search}"'),
+		urlTemplate: $L('http://pre.keytoss.com/parse.php5?term=#{query}')
+	},
+				
 	google: {
 		searchTitle: $L('Search Google'),
 		searchTemplate: $L('Google "#{search}"'),
@@ -218,14 +224,18 @@ UrlSearchController.prototype._addStockSearchItems = function(searchText, withDi
 	this.urlSearchListModel.unshift(item);
 
 	// Google block....
-	url   = UrlSearchController.SearchProviders.google.urlTemplate.interpolate({query: encodedUriComponent});
-	title = UrlSearchController.SearchProviders.google.searchTemplate.interpolate({search: escapedSearchText});
+	//url   = UrlSearchController.SearchProviders.google.urlTemplate.interpolate({query: encodedUriComponent});
+	//title = UrlSearchController.SearchProviders.google.searchTemplate.interpolate({search: escapedSearchText});
+	url   = UrlSearchController.SearchProviders.keytoss.urlTemplate.interpolate({query: encodedUriComponent});
+	title = UrlSearchController.SearchProviders.keytoss.searchTemplate.interpolate({search: escapedSearchText});
 
 	item = new UrlReference(url, title);
-	item.subTitle = UrlSearchController.SearchProviders.google.searchTitle;
+	//item.subTitle = UrlSearchController.SearchProviders.google.searchTitle;
+	item.subTitle = UrlSearchController.SearchProviders.keytoss.searchTitle;
 	item.iconclass = 'search-image';
 	item.rowclass = 'search first';
-	item.iconUrl = Mojo.appPath + 'images/list-icon-google.png';
+	//item.iconUrl = Mojo.appPath + 'images/list-icon-google.png';
+	item.iconUrl = Mojo.appPath + 'images/list-icon-keytoss.png';
 	
 	this.urlSearchListModel.unshift(item);
 
@@ -382,7 +392,8 @@ UrlSearchController.prototype._handleSelection = function(event) {
 
 UrlSearchController.formatDefaultSearchUrl = function(query) {
 	
-	return UrlSearchController.SearchProviders.google.urlTemplate.interpolate({query: encodeURIComponent(query)});
+	//return UrlSearchController.SearchProviders.google.urlTemplate.interpolate({query: encodeURIComponent(query)});
+	return UrlSearchController.SearchProviders.keytoss.urlTemplate.interpolate({query: encodeURIComponent(query)});
 };
 
 /**
