diff --git a/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js b/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
index f343ce2..c7f389b 100644
--- a/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
+++ b/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
@@ -1479,7 +1479,7 @@ handlePowerNotifications: function(payload) {
 				this.controller.showBanner({
 						messageText: batteryalert,
 						icon: "/usr/lib/luna/system/luna-systemui/images/notification-small-charging.png",
-						soundClass: soundClassName
+						soundClass: "vibrate"
 					},null, 'chargingAlert'); 				
 			}
 			this.chargingBannerShown = true;			
