
--- .orig/usr/palm/applications/com.palm.app.musicplayer/scripts/MediaKeyControlHandler.js
+++ /usr/palm/applications/com.palm.app.musicplayer/scripts/MediaKeyControlHandler.js
@@ -30,6 +30,9 @@

				// start playing automatically when the song is switched
				// this is needed to work around NOV-79034
+				this.serviceRequest = new Mojo.Service.Request("palm://com.palm.vibrate", {
+					method: 'vibrate', parameters: { 'period': 0,'duration': 750 }
+				});
				this.musicPlayer.setPlayingState(true);
				break;

