diff --git a/usr/palm/applications/com.palm.app.musicplayer/app/controllers/localnowplaying-assistant.js b/usr/palm/applications/com.palm.app.musicplayer/app/controllers/localnowplaying-assistant.js
index 9dc2a75..9c52dce 100644
--- a/usr/palm/applications/com.palm.app.musicplayer/app/controllers/localnowplaying-assistant.js
+++ b/usr/palm/applications/com.palm.app.musicplayer/app/controllers/localnowplaying-assistant.js
@@ -163,6 +163,7 @@ var LocalnowplayingAssistant = Class.create(
 					onSuccess: callback
 		    });			
 			this.controller.get('progress-info').hide();
+                      
 		},
 		
 		progressBarDragEnd: function(){
@@ -802,7 +803,7 @@ var LocalnowplayingAssistant = Class.create(
 				setTimeout(this.controller.stageController.activate, 0);
 				this.callActivate = false;
 			}
-			
+			this.toggleViews();
 		},
 		
 		
@@ -2345,4 +2346,3 @@ var LocalnowplayingAssistant = Class.create(
 		PROGRESS_INFO_DISPLAY_DURATION: 5000,	
 
 });
-
