diff --git a/usr/palm/applications/com.palm.app.musicplayer/scripts/MediaKeyControlHandler.js b/usr/palm/applications/com.palm.app.musicplayer/scripts/MediaKeyControlHandler.js
index a7f7843..264c645 100644
--- a/usr/palm/applications/com.palm.app.musicplayer/scripts/MediaKeyControlHandler.js
+++ b/usr/palm/applications/com.palm.app.musicplayer/scripts/MediaKeyControlHandler.js
@@ -28,6 +28,9 @@ var MediaKeyControlHandler = Class.create(
 			case "nextAndPlay":
 				// start playing automatically when the song is switched
 				// this is needed to work around NOV-79034
+				this.serviceRequest = new Mojo.Service.Request("palm://com.palm.vibrate", {
+					method: 'vibrate', parameters: { 'period': 0,'duration': 750 }
+				});
 				this.musicPlayer.setPlayingState(true);
 
 				this.musicPlayer.nextSong();
