diff --git a/usr/palm/applications/com.palm.app.musicplayer/app/controllers/mainmenu-assistant.js b/usr/palm/applications/com.palm.app.musicplayer/app/controllers/mainmenu-assistant.js
index 1d13e53..cfc861d 100644
--- a/usr/palm/applications/com.palm.app.musicplayer/app/controllers/mainmenu-assistant.js
+++ b/usr/palm/applications/com.palm.app.musicplayer/app/controllers/mainmenu-assistant.js
@@ -3,7 +3,7 @@
 /*jslint browser:true*/
 /*globals Mojo, Util, AppAssistant, $L*/
 
-
+var count = 1;
 var MainmenuAssistant = function(forceRefresh) {
 	// true if the app is reset for some reason (e.g. enter MSM mode, media server crash)
 	this.forceRefresh = forceRefresh;
@@ -318,6 +318,10 @@ MainmenuAssistant.prototype = {
 	
 		// so that if we then go to artists or albums, the generic art and the frames show up right away
 		Util.preloadGenericIcons();
+                if (count==1){
+                count=2;
+                this.handleShuffleAllSelection();
+                }
 	},
 
 
