diff --git a/usr/palm/applications/com.palm.app.musicplayer/app/controllers/localnowplaying-assistant.js b/usr/palm/applications/com.palm.app.musicplayer/app/controllers/localnowplaying-assistant.js
index ee3c9f5..266d2e8 100644
--- a/usr/palm/applications/com.palm.app.musicplayer/app/controllers/localnowplaying-assistant.js
+++ b/usr/palm/applications/com.palm.app.musicplayer/app/controllers/localnowplaying-assistant.js
@@ -867,7 +867,7 @@ LocalnowplayingAssistant.prototype = {
 			setTimeout(this.controller.stageController.activate, 0);
 			this.callActivate = false;
 		}
-		
+		this.toggleViews();
 	},
 	
 	
