diff --git a/usr/palm/applications/com.palm.app.pdfviewer/app/controllers/document-assistant.js b/usr/palm/applications/com.palm.app.pdfviewer/app/controllers/document-assistant.js
index 708a648..18d609f 100644
--- a/usr/palm/applications/com.palm.app.pdfviewer/app/controllers/document-assistant.js
+++ b/usr/palm/applications/com.palm.app.pdfviewer/app/controllers/document-assistant.js
@@ -220,8 +220,6 @@ var DocumentAssistant = Class.create({
 
     orientationChanged: function(orientation) {
 
-       return;
-/*
        if (this.currentOrientation === orientation)
        {
            return;
@@ -238,7 +236,6 @@ var DocumentAssistant = Class.create({
 
        // Re-create the adapter
        this.dosetup();
-*/
     },
 
     NeedPassword: function(docid) {
