diff --git a/usr/palm/applications/com.palm.app.pdfviewer/app/controllers/app-assistant.js b/usr/palm/applications/com.palm.app.pdfviewer/app/controllers/app-assistant.js
index 2ae0e7f..72afed5 100644
--- a/usr/palm/applications/com.palm.app.pdfviewer/app/controllers/app-assistant.js
+++ b/usr/palm/applications/com.palm.app.pdfviewer/app/controllers/app-assistant.js
@@ -2,6 +2,12 @@
 /*global Mojo, Class, $L, Template, WordAssistant, ExcelAssistant, PowerPointAssistant */
 /*jslint rhino:true, undef:true */
 
+PDFVIEW = {};
+PDFVIEW.prefs = {
+    openLastDoc: false,
+    autoRotate: false
+};
+
 function AppAssistant(appController)
 {
 }
@@ -21,6 +27,13 @@ AppAssistant.prototype.handleLaunch = function(params) {
 
         if (Mojo.Controller.appInfo.noWindow) 
         {
+	    var cookieModel = new Mojo.Model.Cookie('preferences'); 
+	    var cookie = cookieModel.get(); 
+	    if (cookie) { 
+		PDFVIEW.prefs.openLastDoc = cookie.openLastDoc; 
+		PDFVIEW.prefs.autoRotate = cookie.autoRotate; 
+	    }
+
             var f = function(stageController) {
 
                 if (launchParams.target) {
diff --git a/usr/palm/applications/com.palm.app.pdfviewer/app/controllers/document-assistant.js b/usr/palm/applications/com.palm.app.pdfviewer/app/controllers/document-assistant.js
index 708a648..9eee05a 100644
--- a/usr/palm/applications/com.palm.app.pdfviewer/app/controllers/document-assistant.js
+++ b/usr/palm/applications/com.palm.app.pdfviewer/app/controllers/document-assistant.js
@@ -9,6 +9,7 @@ var DocumentAssistant = Class.create({
         items: [
         { label:$L('Open'), command:'file-open-cmd' },
         { label:$L('Save As...'), command:'file-saveas-cmd' },
+        { label:$L('Preferences'), command:'prefs-cmd' },
         { label:$L('Share'), command:'share-cmd' },
         { label:$L('Find'), command:'find-cmd', chosen:false },
         { label:$L('Bookmarks'), command:'pdf-outline', chosen:false },
@@ -16,13 +17,19 @@ var DocumentAssistant = Class.create({
     ]},
 
 	cmdMenuModel: { visible: false, items: [
+		{ items: [
+			{ icon:'sync', command:'rotate-cmd' }
+		]},
 		{},
 		{ items: [
 			{ icon:'back', command:'prev-cmd' },
 			{ template: 'document/command_bar', command: 'edit-cmd' },
 			{ icon:'forward', command:'next-cmd' }
 		]},
-		{}
+		{},
+		{ items: [
+			{ icon:'refresh', command:'last-cmd' }
+		]}
 	]},
 
 	numericMap: {
@@ -43,6 +50,13 @@ var DocumentAssistant = Class.create({
 
     deactivate: function() {
         this.findField.mojo.blur();
+	  var time = new Date();
+	  time.setMonth(time.getMonth()+1);
+	  this.cookieModel.put({ 
+	    currentpage: this.currentpage,
+	    lastjumpsource: this.lastjumpsource,
+	    currentOrientation: this.currentOrientation
+	  }, time);
     },
 
 	initialize: function(parameters) {
@@ -52,6 +66,7 @@ var DocumentAssistant = Class.create({
 		this.document_noext = undefined;
 		this.saveShowing = false;
 		this.currentpage = 1;
+		this.lastjumpsource = 1;
 		this.wrongpasscount = -1;
 		this.jumptimeout = undefined;
 		this.outline_depth = 1;
@@ -70,6 +85,8 @@ var DocumentAssistant = Class.create({
 		this._loadingSpinnerTimeout = null;
 		this._spinnerShowTime = 0;
         this.documentPassword = null;
+		this.currentOrientation = "up";
+		this.autoRotate = PDFVIEW.prefs.autoRotate;
 
         // Variables need for find / search
         this.findShowing = false;
@@ -97,6 +114,23 @@ var DocumentAssistant = Class.create({
             this.filename = this.filename.substr(7);
         }
 
+	    var lastDocModel = new Mojo.Model.Cookie('lastdoc');
+	    lastDocModel.put({
+		lastdoc: this.filename
+	    });
+
+		this.cookieModel = new Mojo.Model.Cookie(this.filename); 
+		var cookie = this.cookieModel.get(); 
+		if (cookie) { 
+		  this.currentpage = cookie.currentpage; 
+		  this.lastjumpsource = cookie.lastjumpsource;
+		  if (!this.autoRotate) {
+		    this.currentOrientation = cookie.currentOrientation;
+		  }
+		}
+
+	    this.cmdMenuModel.items[0].items[0].disabled = this.autoRotate;
+
         Mojo.Log.info("Filename = '%s'", this.filename);
     },
 
@@ -105,13 +139,16 @@ var DocumentAssistant = Class.create({
 
         try {
 
+            // Restore old view.
+            this.controller.stageController.setWindowOrientation(this.currentOrientation);
+
             this.spinnerRunning = false;
 
 			// Set up the view buttons (header)
-			var viewMenuModel = { visible: false, items: [ { template: 'document/header_bar' } ] };
+			/*var viewMenuModel = { visible: false, items: [ { template: 'document/header_bar' } ] };
             this.controller.setupWidget(Mojo.Menu.viewMenu,
                                         {menuClass:'no-fade'},
-                                        viewMenuModel);
+                                        viewMenuModel);*/
 
 			// Set up the command buttons (footer)
             this.controller.setupWidget(Mojo.Menu.commandMenu,
@@ -220,9 +257,7 @@ var DocumentAssistant = Class.create({
 
     orientationChanged: function(orientation) {
 
-       return;
-/*
-       if (this.currentOrientation === orientation)
+       if (!this.autoRotate || this.currentOrientation === orientation)
        {
            return;
        }
@@ -238,7 +273,27 @@ var DocumentAssistant = Class.create({
 
        // Re-create the adapter
        this.dosetup();
-*/
+    },
+
+    rotateView: function() {
+       var orientation;
+       if (this.currentOrientation === "left") {
+           orientation = "up";
+       } else {
+           orientation = "left";
+       }
+
+       // Replace the adapter with an empty div during the rotate
+       var viewerObject = this.controller.get('viewerObject');
+       var emptyDiv = this.controller.document.createElement('div');
+       emptyDiv.id = 'viewerObject'; 
+       viewerObject.parentNode.replaceChild(emptyDiv, viewerObject); 
+
+       this.currentOrientation = orientation;
+       this.controller.window.PalmSystem.setWindowOrientation(orientation);
+
+       // Re-create the adapter
+       this.dosetup();
     },
 
     NeedPassword: function(docid) {
@@ -332,8 +387,8 @@ var DocumentAssistant = Class.create({
 
 		if (this.currentpage !== undefined) {
 
-			this.cmdMenuModel.items[1].items[0].disabled = (this.currentpage == 1);
-			this.cmdMenuModel.items[1].items[2].disabled = (this.currentpage == this.pagecount);
+			this.cmdMenuModel.items[2].items[0].disabled = (this.currentpage == 1);
+			this.cmdMenuModel.items[2].items[2].disabled = (this.currentpage == this.pagecount);
 			this.controller.modelChanged(this.cmdMenuModel);
 
 			// rebind key event
@@ -375,6 +430,13 @@ var DocumentAssistant = Class.create({
         this.JumpToPage( this.currentpage );
 	},
 
+	goToLastJumpSource: function () {
+		var target = this.lastjumpsource;
+		this.lastjumpsource = this.currentpage;
+		this.currentpage = target;
+		this.JumpToPage( this.currentpage );
+	},
+
     forceFootChange: function () {
 
         var goto_page = parseInt(this.controller.get('page_foot_cur_ent').value, 10);
@@ -462,7 +524,7 @@ var DocumentAssistant = Class.create({
 
 		// Set up the command buttons (footer)
 		var page_len = (this.pagecount+'').length;
-		this.cmdMenuModel.items[1].items[1].width = 60 + (page_len * 15);
+		this.cmdMenuModel.items[2].items[1].width = 60 + (page_len * 15);
 		this.controller.modelChanged(this.cmdMenuModel);
 
 		this.controller.setMenuVisible(Mojo.Menu.commandMenu, true);
@@ -568,6 +630,7 @@ var DocumentAssistant = Class.create({
 
 		if (e.keyCode == Mojo.Char.enter) {
 			var goto_page = parseInt(this.controller.get('page_foot_cur_ent').value, 10);
+			this.lastjumpsource = this.currentpage;
             this.JumpToPage(goto_page);
 			this.showFootPageOf();
 		}
@@ -783,7 +846,7 @@ var DocumentAssistant = Class.create({
         var splittext = this.filename.split('/');
         this.document_name = splittext[splittext.length-1];
         this.document_noext = this.document_name.substr(0, this.document_name.indexOf('.pdf'));
-        this.controller.get('documentname').innerHTML = this.document_noext;
+        //this.controller.get('documentname').innerHTML = this.document_noext;
         new_obj.eventListener = this;
 
         this.adapter = new_obj;
@@ -809,9 +872,10 @@ var DocumentAssistant = Class.create({
     openFilePicker: function () {
 
         // Right now lets just open a new window and send it to the recent scene
+	var openParams = { disableOpenLast: true };
         this.controller.serviceRequest('palm://com.palm.applicationManager', {
             method: 'open',
-            parameters: { 'id': 'com.palm.app.pdfviewer' }
+            parameters: { 'id': 'com.palm.app.pdfviewer', params: openParams }
         });
     },
 
@@ -835,6 +899,10 @@ var DocumentAssistant = Class.create({
 						this.saveAsDialog();
 						break;
 
+					case 'prefs-cmd':
+						this.controller.stageController.pushScene('preferences', this);
+						break;
+
 					case 'share-cmd':
 						this.shareDocument();
 						break;
@@ -860,6 +928,14 @@ var DocumentAssistant = Class.create({
 						this.showFootPageOf();
 						break;
 
+					case 'last-cmd':					
+						this.goToLastJumpSource();
+						break;
+						
+					case 'rotate-cmd':					
+						this.rotateView();
+						break;
+						
 					case 'pdf-outline':
 
 						// Create the PDF outline/bookmarks
@@ -919,7 +995,7 @@ var DocumentAssistant = Class.create({
 			var menuAssistant = this.controller._menu.assistant;
 			menuAssistant.viewSpacerHeight = 0;
 
-            this.controller.setMenuVisible(Mojo.Menu.viewMenu, visible);
+            //this.controller.setMenuVisible(Mojo.Menu.viewMenu, visible);
         }
     },
 
@@ -1281,7 +1357,7 @@ SaveAsDialogAssistant = Class.create({
                         this.documentAssistant.document_name = save_input;
                         this.documentAssistant.document_noext = this.documentAssistant.document_name.substr(0, this.documentAssistant.document_name.indexOf('.pdf'));
 
-                        this.documentAssistant.controller.get('documentname').innerHTML = this.documentAssistant.document_noext;
+                        //this.documentAssistant.controller.get('documentname').innerHTML = this.documentAssistant.document_noext;
 
                     }
                     catch(e)
diff --git a/usr/palm/applications/com.palm.app.pdfviewer/app/controllers/preferences-assistant.js b/usr/palm/applications/com.palm.app.pdfviewer/app/controllers/preferences-assistant.js
new file mode 100644
index 0000000..e8fa320
--- /dev/null
+++ b/usr/palm/applications/com.palm.app.pdfviewer/app/controllers/preferences-assistant.js
@@ -0,0 +1,44 @@
+function PreferencesAssistant() {
+}
+
+PreferencesAssistant.prototype.setup = function() {
+    this.controller.setupWidget('lastDoc',
+				this.lastDocAttributes = {
+				    trueLabel: $L('On'),
+				    falseLabel: $L('Off')
+				},
+				this.lastDocModel = {
+				    value: PDFVIEW.prefs.openLastDoc,
+				    disabled: false			
+				});
+
+    this.controller.setupWidget('autoRotate',
+				this.autoRotateAttributes = {
+				    trueLabel: $L('On'),
+				    falseLabel: $L('Off')
+				},
+				this.autoRotateModel = {
+				    value: PDFVIEW.prefs.autoRotate,
+				    disabled: false			
+				});
+
+    this.prefsappMenuModel = {
+	visible: true,
+	items: [
+	    Mojo.Menu.editItem,
+	    { label: $L("Help") + "...", command: Mojo.Menu.helpCmd, disabled: true }
+	]
+    };
+    this.controller.setupWidget(Mojo.Menu.appMenu, {omitDefaultItems: true}, this.prefsappMenuModel);
+    this.controller.setInitialFocusedElement(null);
+}
+
+PreferencesAssistant.prototype.deactivate = function(event) {
+    this.cookieModel = new Mojo.Model.Cookie('preferences');
+    PDFVIEW.prefs.openLastDoc = this.lastDocModel.value;
+    PDFVIEW.prefs.autoRotate = this.autoRotateModel.value;
+    this.cookieModel.put({ 
+	openLastDoc: PDFVIEW.prefs.openLastDoc,
+	autoRotate: PDFVIEW.prefs.autoRotate
+    });
+}
diff --git a/usr/palm/applications/com.palm.app.pdfviewer/app/controllers/recent-assistant.js b/usr/palm/applications/com.palm.app.pdfviewer/app/controllers/recent-assistant.js
index ad54e64..d1ae625 100644
--- a/usr/palm/applications/com.palm.app.pdfviewer/app/controllers/recent-assistant.js
+++ b/usr/palm/applications/com.palm.app.pdfviewer/app/controllers/recent-assistant.js
@@ -56,6 +56,17 @@ var RecentAssistant = Class.create({
 	{
         //Mojo.Log.info("RecentAssistant: activate");
 
+	    if (!this.disableOpenLast && PDFVIEW.prefs.openLastDoc && this.lastdoc != null) {
+		this.disableOpenLast = true;
+                // Push the new document scene
+                var args = { name: 'document', disableSceneScroller: true };
+                var params = { hidemode: 'none', filename: this.lastdoc };
+		
+                this.controller.stageController.pushScene(args, params);
+	    } else {
+
+	    this.disableOpenLast = true;;
+
         var params = {
             // Only show files
             kind: 'file',
@@ -78,10 +89,21 @@ var RecentAssistant = Class.create({
 
         Mojo.FilePicker.pickFile(params, this.controller.stageController);
 
+	    }
     },
 
-    initialize: function() 
+    initialize: function(parameters) 
 	{
+	    this.disableOpenLast = false;
+	    this.lastdoc = null;
+	    this.cookieModel = new Mojo.Model.Cookie('lastdoc');
+	    var cookie = this.cookieModel.get();
+	    if (cookie) {
+		this.lastdoc = cookie.lastdoc;
+	    }
+	    if (parameters.launchParams) {
+		this.disableOpenLast = parameters.launchParams.disableOpenLast;
+	    }
         //Mojo.Log.info("RecentAssistant: initialize");
     },
 
diff --git a/usr/palm/applications/com.palm.app.pdfviewer/app/views/document/command_bar.html b/usr/palm/applications/com.palm.app.pdfviewer/app/views/document/command_bar.html
index 8fe00ff..96851f0 100644
--- a/usr/palm/applications/com.palm.app.pdfviewer/app/views/document/command_bar.html
+++ b/usr/palm/applications/com.palm.app.pdfviewer/app/views/document/command_bar.html
@@ -6,7 +6,7 @@
     </div>
     <div id="page_foot_ent" class="foot-page" style="display: none;">
         <input id="page_foot_cur_ent" class="foot-page-ent" />
-		<span class="foot-page-cnt">&nbsp;/&nbsp;</span>
-        <span id="page_foot_cnt_ent" class="foot-page-cnt"></span>
+		<span class="foot-page-cnt-inp">&nbsp;/&nbsp;</span>
+        <span id="page_foot_cnt_ent" class="foot-page-cnt-inp"></span>
     </div>
 </div>
diff --git a/usr/palm/applications/com.palm.app.pdfviewer/app/views/document/document-scene.html b/usr/palm/applications/com.palm.app.pdfviewer/app/views/document/document-scene.html
index 3e95f8b..25d9402 100644
--- a/usr/palm/applications/com.palm.app.pdfviewer/app/views/document/document-scene.html
+++ b/usr/palm/applications/com.palm.app.pdfviewer/app/views/document/document-scene.html
@@ -1,4 +1,4 @@
-<div id="header" style="height:56px;">
+<div id="header" style="height:0px;">
   <div id="findField" name="findField" x-mojo-element="Find"> </div>
 </div>
 
diff --git a/usr/palm/applications/com.palm.app.pdfviewer/app/views/document/header_bar.html b/usr/palm/applications/com.palm.app.pdfviewer/app/views/document/header_bar.html
index cdb2766..0a5480f 100644
--- a/usr/palm/applications/com.palm.app.pdfviewer/app/views/document/header_bar.html
+++ b/usr/palm/applications/com.palm.app.pdfviewer/app/views/document/header_bar.html
@@ -1,4 +1,3 @@
 <div id="topHead" class="palm-header left">
     <div id="my_spinner" x-mojo-element="Spinner"> </div>
-    <div id="documentname" x-mojo-loc="" class="truncating-text"></div>
 </div>
diff --git a/usr/palm/applications/com.palm.app.pdfviewer/app/views/preferences/preferences-scene.html b/usr/palm/applications/com.palm.app.pdfviewer/app/views/preferences/preferences-scene.html
new file mode 100644
index 0000000..caa5024
--- /dev/null
+++ b/usr/palm/applications/com.palm.app.pdfviewer/app/views/preferences/preferences-scene.html
@@ -0,0 +1,41 @@
+<div id="prefs_view" class="prefs-main">
+  <div class="palm-page-header">
+    <div class="header-icon pdfview"></div>
+    <div class="header-text">
+		  <span x-mojo-loc=''>Preferences</span>
+    </div>
+  </div>
+
+	 <div class="palm-group">
+		<div class="palm-group-title">
+			<span x-mojo-loc=''>Startup</span>
+		</div>
+		<div class="palm-list">
+			<div class="palm-row single">
+				<div class="palm-row-wrapper">
+					<div x-mojo-element="ToggleButton" id="lastDoc"></div>
+					<div class="title capitalize" x-mojo-loc='' id='lastDocLabel'>Load last document on startup</div>
+				</div>
+			</div>
+		</div>
+	</div>
+
+	<div class="palm-group">
+		<div class="palm-group-title">
+			<span x-mojo-loc=''>Rotation</span>
+		</div>
+		<div class="palm-list">
+			<div class="palm-row single">
+				<div class="palm-row-wrapper">
+					<div x-mojo-element="ToggleButton" id="autoRotate"></div>
+					<div class="title capitalize" x-mojo-loc=''>Use accelerometer for rotation</div>
+				</div>
+			</div>
+		</div>
+	</div>
+
+	<div class="palm-info-text single" x-mojo-loc=''>
+		Changes will take effect next time a document is opened.
+	</div>
+
+</div>
diff --git a/usr/palm/applications/com.palm.app.pdfviewer/sources.json b/usr/palm/applications/com.palm.app.pdfviewer/sources.json
index ef04afe..b213095 100644
--- a/usr/palm/applications/com.palm.app.pdfviewer/sources.json
+++ b/usr/palm/applications/com.palm.app.pdfviewer/sources.json
@@ -6,5 +6,6 @@
   { "source": "app\/controllers\/error-assistant.js" },
   { "source": "app\/controllers\/pagetext-assistant.js" },
   { "source": "app\/controllers\/password-assistant.js" },
+  { "source": "app\/controllers\/preferences-assistant.js" },
   { "source": "javascripts\/widget_find.js" }
 ]
diff --git a/usr/palm/applications/com.palm.app.pdfviewer/stylesheets/pdf-viewer.css b/usr/palm/applications/com.palm.app.pdfviewer/stylesheets/pdf-viewer.css
index 572f564..76dabb2 100644
--- a/usr/palm/applications/com.palm.app.pdfviewer/stylesheets/pdf-viewer.css
+++ b/usr/palm/applications/com.palm.app.pdfviewer/stylesheets/pdf-viewer.css
@@ -166,7 +166,7 @@ div.foot-page
 
 span.foot-page-cur
 {
-    color: white;
+    color: transparent;
     font-size: 14px;
     font-style: normal;
     font-weight: bold;
@@ -193,6 +193,16 @@ input.foot-page-ent
 
 span.foot-page-cnt
 {
+    color: transparent;
+    font-size: 14px;
+    font-style: normal;
+    font-weight: bold;
+    text-transform: none;
+    text-decoration: none;
+}
+
+span.foot-page-cnt-inp
+{
     color: white;
     font-size: 14px;
     font-style: normal;
@@ -200,3 +210,52 @@ span.foot-page-cnt
     text-transform: none;
     text-decoration: none;
 }
+
+/* Grouped buttons */
+
+.palm-menu-group .palm-menu-button,
+.palm-menu-group .palm-menu-label,
+.palm-menu-group .palm-menu-text {
+  top: 0px;
+  -webkit-border-image: url(../images/textfield-noshadow.png) 0 20 0 20 stretch stretch;
+}
+
+.palm-menu-group .palm-menu-button:first-child,
+.palm-menu-group .palm-menu-label:first-child,
+.palm-menu-group .palm-menu-text:first-child {
+  -webkit-border-image: url(../images/textfield-noshadow.png) 0 20 0 20 stretch stretch;
+  padding-left: 10px;
+}
+
+.palm-menu-group .palm-menu-button:last-child,
+.palm-menu-group .palm-menu-label:last-child,
+.palm-menu-group .palm-menu-text:last-child {
+  -webkit-border-image: url(../images/textfield-noshadow.png) 0 20 0 20 stretch stretch;
+  padding-right: 10px;
+}
+
+/* Disabled states for menu system buttons. */
+
+.palm-menu-group .palm-menu-text.palm-disabled,
+.palm-menu-group .palm-menu-label.palm-disabled,
+.palm-menu-group .palm-menu-button.palm-disabled {
+  -webkit-border-image: url(../images/textfield-noshadow.png) 0 20 0 20 stretch stretch;
+}
+
+.palm-menu-group .palm-menu-text.palm-disabled:first-child,
+.palm-menu-group .palm-menu-label.palm-disabled:first-child,
+.palm-menu-group .palm-menu-button.palm-disabled:first-child {
+  -webkit-border-image: url(../images/textfield-noshadow.png) 0 20 0 20 stretch stretch;
+}
+
+.palm-menu-group .palm-menu-text.palm-disabled:last-child,
+.palm-menu-group .palm-menu-label.palm-disabled:last-child,
+.palm-menu-group .palm-menu-button.palm-disabled:last-child {
+  -webkit-border-image: url(../images/textfield-noshadow.png) 0 20 0 20 stretch stretch;
+}
+
+.palm-page-header .header-icon.pdfview {
+		width: 32px;
+		height: 32px;
+		background: url(../images/pdfsmall.png) center center no-repeat;
+}
