diff --git a/usr/palm/applications/com.palm.app.pdfviewer/app/controllers/document-assistant.js b/usr/palm/applications/com.palm.app.pdfviewer/app/controllers/document-assistant.js
index 666d24e..4120555 100644
--- a/usr/palm/applications/com.palm.app.pdfviewer/app/controllers/document-assistant.js
+++ b/usr/palm/applications/com.palm.app.pdfviewer/app/controllers/document-assistant.js
@@ -220,7 +220,6 @@ var DocumentAssistant = Class.create({
 
     orientationChanged: function(orientation) {
 
-       return;
        if (this.currentOrientation === orientation) 
        {
            return;
