diff --git a/usr/palm/applications/com.palm.app.youtube/app/controllers/video-assistant.js b/usr/palm/applications/com.palm.app.youtube/app/controllers/video-assistant.js
index 99701ce..aa98479 100644
--- a/usr/palm/applications/com.palm.app.youtube/app/controllers/video-assistant.js
+++ b/usr/palm/applications/com.palm.app.youtube/app/controllers/video-assistant.js
@@ -77,6 +77,10 @@ var VideoAssistant = Class.create({
 						}})
 				event.stop(event);
 			}	
+			else if (event.command == 'clipboard' && this.video.shareLink && this.video.shareLink != "") {
+				this.controller.stageController.setClipboard(this.video.shareLink);
+				event.stop(event);
+			}
 			else {
 				AppAssistant.common.handleCommand(event);	
 			}
@@ -92,7 +96,8 @@ var VideoAssistant = Class.create({
 	shareMenuModel: {label: $L('Share'), 
 					items: [
 						{label: $L('Email'), command:'email'}, 
-						{label: $L('Text Message'), command:'msg' }
+						{label: $L('Text Message'), command:'msg'},
+						{label: $L('Copy URL to Clipboard'), command:'clipboard' }
 					]},
 	
 });
