diff --git a/usr/palm/applications/com.palm.app.youtube/app/controllers/video-assistant.js b/usr/palm/applications/com.palm.app.youtube/app/controllers/video-assistant.js
index 99701ce..d0ae94e 100644
--- a/usr/palm/applications/com.palm.app.youtube/app/controllers/video-assistant.js
+++ b/usr/palm/applications/com.palm.app.youtube/app/controllers/video-assistant.js
@@ -68,6 +68,16 @@ var VideoAssistant = Class.create({
 						}});
 				event.stop(event);
 			}
+			else if (event.command == 'share-fb' && this.video.shareLink && this.video.shareLink != "") {
+				var share_url = this.video.shareLink.split('&')[0];
+				this.controller.serviceRequest('palm://com.palm.applicationManager', {
+					method: 'open',
+					parameters: {
+						id: "com.palm.app.facebook",
+						params: {status: this.video.title + ' - ' + share_url}
+					}});
+			event.stop(event);
+			}
 			else if (event.command == 'msg' && this.video.shareLink && this.video.shareLink != "") {
 				this.controller.serviceRequest('luna://com.palm.applicationManager', {	
 						method: 'open',
@@ -92,7 +102,8 @@ var VideoAssistant = Class.create({
 	shareMenuModel: {label: $L('Share'), 
 					items: [
 						{label: $L('Email'), command:'email'}, 
-						{label: $L('Text Message'), command:'msg' }
+						{label: $L('Text Message'), command:'msg' },
+						{label: $L('Share via Facebook'), command: 'share-fb'}
 					]},
 	
 });
