diff --git a/usr/palm/applications/com.palm.app.youtube/app/models/yt-api.js b/usr/palm/applications/com.palm.app.youtube/app/models/yt-api.js
index 7c49f85..2854bf2 100644
--- a/usr/palm/applications/com.palm.app.youtube/app/models/yt-api.js
+++ b/usr/palm/applications/com.palm.app.youtube/app/models/yt-api.js
@@ -96,7 +96,7 @@ var YoutubeAPI = Class.create ({
 	// two: once we obtain the working device key, we store it and never need to register again.
 	testDeviceKeyIsValid: function(authToken) { 
 		Mojo.Log.info("Youtube: API testDeviceKeyIsValid");
-		var url = 'http://gdata.youtube.com/feeds/api/standardfeeds/most_popular?v=2&max-results=#{maxResults}&start-index=#{startindex}&time=today&alt=json&format=#{format}&safeSearch=strict'.
+		var url = 'https://gdata.youtube.com/feeds/api/standardfeeds/most_popular?v=2&max-results=#{maxResults}&start-index=#{startindex}&time=today&alt=json&format=#{format}&safeSearch=none'.
 		interpolate({
 			maxResults: 1,
 			startindex: 1,
@@ -200,7 +200,7 @@ var YoutubeAPI = Class.create ({
 			return;
 		}
 					
-		var url = 'http://gdata.youtube.com/feeds/api/standardfeeds/most_popular?v=2&max-results=#{maxResults}&start-index=#{startindex}&time=this_week&alt=json&format=#{format}&safeSearch=strict&client=#{clientId}&orderby=published'.
+		var url = 'https://gdata.youtube.com/feeds/api/standardfeeds/most_popular?v=2&max-results=#{maxResults}&start-index=#{startindex}&time=this_week&alt=json&format=#{format}&safeSearch=none&client=#{clientId}&orderby=published'.
 			interpolate({
 			maxResults: numResults,
 			startindex: startIndex,
@@ -222,7 +222,7 @@ var YoutubeAPI = Class.create ({
 			return;
 		}
 		
-		var url = 'http://gdata.youtube.com/feeds/api/standardfeeds/most_viewed?v=2&max-results=#{maxResults}&start-index=#{startindex}&time=this_week&alt=json&format=#{format}&safeSearch=strict&client=#{clientId}&orderby=viewCount'.
+		var url = 'https://gdata.youtube.com/feeds/api/standardfeeds/most_viewed?v=2&max-results=#{maxResults}&start-index=#{startindex}&time=this_week&alt=json&format=#{format}&safeSearch=none&client=#{clientId}&orderby=viewCount'.
 			interpolate({
 				maxResults: numResults,
 				startindex: startIndex,
@@ -246,7 +246,7 @@ var YoutubeAPI = Class.create ({
 		
 		query = encodeURIComponent(query);
 		
-		var url = 'http://gdata.youtube.com/feeds/api/videos?v=2&q=#{vq}&orderby=relevance&max-results=#{maxResults}&start-index=#{startindex}&alt=json&format=#{format}&safeSearch=strict&client=#{clientId}'.
+		var url = 'https://gdata.youtube.com/feeds/api/videos?v=2&q=#{vq}&orderby=relevance&max-results=#{maxResults}&start-index=#{startindex}&alt=json&format=#{format}&safeSearch=none&client=#{clientId}'.
 			interpolate({
 				vq: query,
 				maxResults: numResults,
@@ -271,7 +271,7 @@ var YoutubeAPI = Class.create ({
 		
 		query = encodeURIComponent(query);
 		
-		var url = 'http://gdata.youtube.com/feeds/api/videos?v=2&author=#{vq}&orderby=relevance&max-results=#{maxResults}&start-index=#{startindex}&alt=json&format=#{format}&safeSearch=strict&client=#{clientId}'.
+		var url = 'https://gdata.youtube.com/feeds/api/videos?v=2&author=#{vq}&orderby=relevance&max-results=#{maxResults}&start-index=#{startindex}&alt=json&format=#{format}&safeSearch=none&client=#{clientId}'.
 			interpolate({
 			    vq: query,
 			    maxResults: numResults,
@@ -295,7 +295,7 @@ var YoutubeAPI = Class.create ({
 			return;
 		}
 		
-		var url = '#{base}&max-results=#{maxResults}&start-index=#{startindex}&alt=json&format=#{format}&safeSearch=strict'.
+		var url = '#{base}&max-results=#{maxResults}&start-index=#{startindex}&alt=json&format=#{format}&safeSearch=none'.
 			interpolate({
 			    //base: video.feed_path,
 				base: feedPath,
@@ -318,7 +318,7 @@ var YoutubeAPI = Class.create ({
 			return;
 		}
 		
-		var url = 'http://gdata.youtube.com/feeds/api/videos/#{videoId}?v=2&alt=json&format=#{format}&client=#{clientId}&safeSearch=strict'.
+		var url = 'https://gdata.youtube.com/feeds/api/videos/#{videoId}?v=2&alt=json&format=#{format}&client=#{clientId}&safeSearch=none'.
 			interpolate({
 				videoId: id,
 			    format: this.videoFormat,
@@ -484,7 +484,7 @@ var YoutubeAPI = Class.create ({
 	// TODO, do I ever need format 2?
 	// 2 - 3gp over http
 	// 3 - mp4 over http (H.264)
-	videoFormat : 3,
+	videoFormat : 6,
 	
 	// URLs
 	registerDeviceURL: "http://www.google.com/youtube/accounts/registerDevice"
