diff --git a/usr/palm/applications/com.palm.app.youtube/app/controllers/home-assistant.js b/usr/palm/applications/com.palm.app.youtube/app/controllers/home-assistant.js
index 934d77e..c316837 100644
--- a/usr/palm/applications/com.palm.app.youtube/app/controllers/home-assistant.js
+++ b/usr/palm/applications/com.palm.app.youtube/app/controllers/home-assistant.js
@@ -13,6 +13,9 @@ var HomeAssistant = Class.create({
 		// setup application menu
 		this.controller.setupWidget(Mojo.Menu.appMenu, {omitDefaultItems: true}, AppAssistant.appMenuModel);
 		
+		// enable free orientation
+    	this.controller.window.PalmSystem.setWindowOrientation("free");	
+		
 		// setup bottom menus		
 		this.controller.setupWidget('videos-menu', undefined, this.videosMenuModel);				
 		this.controller.setupWidget(Mojo.Menu.commandMenu, {menuClass: "no-fade"}, this.cmdMenuModel);
@@ -53,6 +56,14 @@ var HomeAssistant = Class.create({
 			});
     },
 	
+	orientationChanged: function(orientation) {
+		if (orientation === "left" || orientation === "right") {
+			this.controller.sceneElement.addClassName('landscape');
+		} else {
+			this.controller.sceneElement.removeClassName('landscape');
+		}
+	},
+	
 	activate: function() {
 		$('search_string').mojo.focus();
 		this._checkHistoryCount();
diff --git a/usr/palm/applications/com.palm.app.youtube/app/controllers/search-assistant.js b/usr/palm/applications/com.palm.app.youtube/app/controllers/search-assistant.js
index f99f18d..d429959 100644
--- a/usr/palm/applications/com.palm.app.youtube/app/controllers/search-assistant.js
+++ b/usr/palm/applications/com.palm.app.youtube/app/controllers/search-assistant.js
@@ -21,6 +21,9 @@ var SearchAssistant = Class.create({
 		// setup application menu
 		this.controller.setupWidget(Mojo.Menu.appMenu, {omitDefaultItems: true}, AppAssistant.appMenuModel);
 		
+		// enable free orientation
+    	this.controller.window.PalmSystem.setWindowOrientation("free");			
+		
 		// setup search field
 		this.searchFieldModel.attributes = {
 			hintText: $L('Search All Videos...'),
@@ -67,6 +70,14 @@ var SearchAssistant = Class.create({
 			});		
     },
 	
+	orientationChanged: function(orientation) {
+		if (orientation === "left" || orientation === "right") {
+			this.controller.sceneElement.addClassName('landscape');
+		} else {
+			this.controller.sceneElement.removeClassName('landscape');
+		}
+	},
+	
 	activate: function() {
 		// in case of cross app launch we want to focus the app only after 
 		// search scene is ready to go. This is so that user won't see a
diff --git a/usr/palm/applications/com.palm.app.youtube/app/controllers/video-assistant.js b/usr/palm/applications/com.palm.app.youtube/app/controllers/video-assistant.js
index 5c3be25..db5e966 100644
--- a/usr/palm/applications/com.palm.app.youtube/app/controllers/video-assistant.js
+++ b/usr/palm/applications/com.palm.app.youtube/app/controllers/video-assistant.js
@@ -12,6 +12,9 @@ var VideoAssistant = Class.create({
     setup: function() {
 		// setup application menu
 		this.controller.setupWidget(Mojo.Menu.appMenu, {omitDefaultItems: true}, AppAssistant.appMenuModel);
+	    
+		// enable free orientation
+    	this.controller.window.PalmSystem.setWindowOrientation("free");		
 		
 		// setup bottom menus
 		this.cmdMenuModel = {items: [{label: $L('More'), submenu:'more-menu'}, 
@@ -30,7 +33,15 @@ var VideoAssistant = Class.create({
 				AppAssistant.common.playVideo(this.video);
 			}.bind(this));
     },
-	
+
+	orientationChanged: function(orientation) {
+		if (orientation === "left" || orientation === "right") {
+			this.controller.sceneElement.addClassName('landscape');
+		} else {
+			this.controller.sceneElement.removeClassName('landscape');
+		}
+	},
+
 	activate: function() {
 		// in case of cross app launch we want to focus the app only after 
 		// search scene is ready to go. This is so that user won't see a
