diff --git a/usr/palm/applications/com.palm.app.youtube/app/models/yt-api.js b/usr/palm/applications/com.palm.app.youtube/app/models/yt-api.js
index 7c49f85..7d6a6ce 100644
--- a/usr/palm/applications/com.palm.app.youtube/app/models/yt-api.js
+++ b/usr/palm/applications/com.palm.app.youtube/app/models/yt-api.js
@@ -200,7 +200,7 @@ var YoutubeAPI = Class.create ({
 			return;
 		}
 					
-		var url = 'http://gdata.youtube.com/feeds/api/standardfeeds/most_popular?v=2&max-results=#{maxResults}&start-index=#{startindex}&time=this_week&alt=json&format=#{format}&safeSearch=strict&client=#{clientId}&orderby=published'.
+		var url = 'http://gdata.youtube.com/feeds/api/standardfeeds/BR/most_popular?v=2&max-results=#{maxResults}&start-index=#{startindex}&time=this_week&alt=json&format=#{format}&safeSearch=strict&client=#{clientId}&orderby=published'.
 			interpolate({
 			maxResults: numResults,
 			startindex: startIndex,
@@ -222,7 +222,7 @@ var YoutubeAPI = Class.create ({
 			return;
 		}
 		
-		var url = 'http://gdata.youtube.com/feeds/api/standardfeeds/most_viewed?v=2&max-results=#{maxResults}&start-index=#{startindex}&time=this_week&alt=json&format=#{format}&safeSearch=strict&client=#{clientId}&orderby=viewCount'.
+		var url = 'http://gdata.youtube.com/feeds/api/standardfeeds/BR/most_viewed?v=2&max-results=#{maxResults}&start-index=#{startindex}&time=this_week&alt=json&format=#{format}&safeSearch=strict&client=#{clientId}&orderby=viewCount'.
 			interpolate({
 				maxResults: numResults,
 				startindex: startIndex,
