diff --git a/usr/palm/services/com.palm.service.videos.youtube/assistant.js b/usr/palm/services/com.palm.service.videos.youtube/assistant.js
index c6008d7..fcdd13d 100644
--- a/usr/palm/services/com.palm.service.videos.youtube/assistant.js
+++ b/usr/palm/services/com.palm.service.videos.youtube/assistant.js
@@ -107,7 +107,7 @@ var YouTube = (function() {
 				$t : (vid.tags? strip(vid.tags) : "..")
 			};
 			//this makes the video private
-			//metadata["media:group"]["yt:private"] = {};
+			metadata["media:group"]["yt:private"] = {};
 
 			return IMPORTS["foundations.xml"].json2xml(
 				{version: "1.0", entry: metadata}, 
