diff --git a/usr/palm/services/com.palm.service.videos.youtube/assistant.js b/usr/palm/services/com.palm.service.videos.youtube/assistant.js
index c6008d7..87fb26b 100644
--- a/usr/palm/services/com.palm.service.videos.youtube/assistant.js
+++ b/usr/palm/services/com.palm.service.videos.youtube/assistant.js
@@ -236,6 +236,7 @@ var YouTube = (function() {
 		uploadDurationLimit : 600, //10 minutes
 
 		uploadCategories : [
+			{label:('People & Blogs'), value:"People"},
 			{label:('Autos & Vehicles'), value:"Autos"},
 			{label:('Comedy'), value:"Comedy"},
 			{label:('Education'), value:"Education"},
@@ -245,7 +246,6 @@ var YouTube = (function() {
 			{label:('Sports'), value:"Sports"},
 			{label:('Travel & Events'), value:"Travel"},
 			{label:('Gaming'), value:"Games"},
-			{label:('People & Blogs'), value:"People"},
 			{label:('News'), value:"News & Politics"},
 			{label:('Entertainment'), value:"Entertainment"},
 			{label:('Howto & Style'), value:"Howto"},
