diff --git a/usr/palm/services/com.palm.service.videos.youtube/assistant.js b/usr/palm/services/com.palm.service.videos.youtube/assistant.js
index c6008d7..d0c5483 100644
--- a/usr/palm/services/com.palm.service.videos.youtube/assistant.js
+++ b/usr/palm/services/com.palm.service.videos.youtube/assistant.js
@@ -108,7 +108,11 @@ var YouTube = (function() {
 			};
 			//this makes the video private
 			//metadata["media:group"]["yt:private"] = {};
-
+			metadata["yt:accessControl"] = {
+				action : "list",
+				permission : "denied"
+			};
+			
 			return IMPORTS["foundations.xml"].json2xml(
 				{version: "1.0", entry: metadata}, 
 				{includeHeader: true});
