diff --git a/usr/palm/frameworks/mojo/builtins/palmInitFramework347.js b/usr/palm/frameworks/mojo/builtins/palmInitFramework347.js
index aed4b02..679c5fa 100644
--- a/usr/palm/frameworks/mojo/builtins/palmInitFramework347.js
+++ b/usr/palm/frameworks/mojo/builtins/palmInitFramework347.js
@@ -11195,6 +11195,10 @@ widgetAssistant.sendChanges(triggeringEvent);
 
 
 sendTap:function(triggeringEvent){
+		this.serviceRequest = new Mojo.Service.Request					
+		("palm://com.palm.vibrate", {
+		method: 'vibrate', parameters: {'period': 0,'duration': 75 }
+		});
 var tapEvent,focusedElement;
 focusedElement=Mojo.View.getFocusedElement(this.downTarget.ownerDocument.body);
 if(focusedElement){
@@ -12466,6 +12470,10 @@ Mojo.Log.error("WARNING: Caught exception in dragndrop container.dragLeave(): "+
 
 
 tapEvent:function(event){
+		this.serviceRequest = new Mojo.Service.Request					
+		("palm://com.palm.vibrate", {
+		method: 'vibrate', parameters: {'period': 0,'duration': 75 }
+		});
 this.dragEnd(event);
 },
 
