diff --git a/etc/palm/lunaAnimations.conf b/etc/palm/lunaAnimations.conf
index 972ec3f..686de16 100644
--- a/etc/palm/lunaAnimations.conf
+++ b/etc/palm/lunaAnimations.conf
@@ -1,25 +1,25 @@
 [Equations]
-easeInStrength=3
-easeOutStrength=3
-easeInOutStrength=3
+easeInStrength=5
+easeOutStrength=5
+easeInOutStrength=5
 
 [FPS]
 normalFPS=35
 slowFPS=20
 
 [Cards]
-cardLaunchDuration=400
-cardLaunchStrength=40
+cardLaunchDuration=100
+cardLaunchStrength=50
 cardSlideDuration=300
 cardSlideStrength=30
-cardMaximizeDuration=300
-cardMaximizeStrength=30
-cardMinimizeDuration=350
-cardMinimizeStrength=40
-cardDeleteDuration=300
-cardDeleteStrength=30
-cardScootAwayOnLaunchDuration=400
-cardScootAwayOnLaunchStrength=30
+cardMaximizeDuration=100
+cardMaximizeStrength=40
+cardMinimizeDuration=100
+cardMinimizeStrength=50
+cardDeleteDuration=150
+cardDeleteStrength=40
+cardScootAwayOnLaunchDuration=100
+cardScootAwayOnLaunchStrength=40
 cardMoveNormalDuration=100
 cardMoveNormalStrength=30
 cardMoveOverviewDuration=100
@@ -28,17 +28,17 @@ cardShuffleReorderDuration=200
 cardShuffleReorderStrength=30
 cardSwitchReachEndMaximizedDuration=200
 cardSwitchReachEndMaximizedStrength=30
-cardSwitchMaximizedDuration=350
+cardSwitchMaximizedDuration=150
 cardSwitchMaximizedStrength=30
-cardPrepareAddDuration=150
+cardPrepareAddDuration=50
 cardBeforeAddDelay=0
-cardAddMaxDuration=750
-cardLoadingPulseDuration=900
+cardAddMaxDuration=250
+cardLoadingPulseDuration=200
 cardLoadingPulseStrength=20
 cardLoadingCrossFadeDuration=300
 cardLoadingCrossFadeStrength=10
-cardLoadingTimeBeforeShowingPulsing=900
-cardTransitionDuration=300
+cardLoadingTimeBeforeShowingPulsing=150
+cardTransitionDuration=100
 cardTransitionStrength=20
 
 [Spaces]
@@ -46,9 +46,9 @@ positiveSpaceChangeDuration=400
 positiveSpaceChangeStrength=30
 
 [Launcher]
-quickLaunchDuration=350
+quickLaunchDuration=150
 quickLaunchStrength=30
-quickLaunchSlideToStacheDuration=150
+quickLaunchSlideToStacheDuration=90
 quickLaunchSlideToStacheStrength=30
-launcherDuration=350
+launcherDuration=150
 launcherStrength=30
