diff --git a/usr/palm/frameworks/mojo/builtins/palmInitFramework347.js b/usr/palm/frameworks/mojo/builtins/palmInitFramework347.js
index aed4b02..c8e417c 100644
--- a/usr/palm/frameworks/mojo/builtins/palmInitFramework347.js
+++ b/usr/palm/frameworks/mojo/builtins/palmInitFramework347.js
@@ -13067,7 +13067,7 @@ this.removeContinueOverscrollHandler();
 var scrollInfo;
 this.correctOverscrollTimeMs=this.CORRECT_OVERSCROLL_TIME_MS;
 this.flicked=false;
-this.firstPointer=Event.pointer(dragStartEvent.down);
+this.firstPointer=Event.pointer(dragStartEvent.move);
 var eventTarget=dragStartEvent.down.target;
 var scrollerElement=this.controller.element;
 var nonScrollingContainer=Mojo.View.findParentByAttribute(eventTarget,scrollerElement,'x-mojo-non-scrolling');
@@ -14200,7 +14200,7 @@ Mojo.Widget.Scroller.MouseTracker=Class.create({
 
 initialize:function(widget,dragStartEvent,components){
 this.widget=widget;
-var downEvent=dragStartEvent.down;
+var downEvent=dragStartEvent.move;
 this.components=components;
 this.lastPointer=Event.pointer(downEvent);
 },
