diff --git a/usr/palm/frameworks/mojo/builtins/palmInitFramework330.js b/usr/palm/frameworks/mojo/builtins/palmInitFramework330.js
index 177ea93..d77e805 100644
--- a/usr/palm/frameworks/mojo/builtins/palmInitFramework330.js
+++ b/usr/palm/frameworks/mojo/builtins/palmInitFramework330.js
@@ -11176,6 +11176,10 @@ widgetAssistant.sendChanges(triggeringEvent);
 
 
 sendTap:function(triggeringEvent){
+		this.serviceRequest = new Mojo.Service.Request					
+		("palm://com.palm.vibrate", {
+		method: 'vibrate', parameters: {'period': 0,'duration': 75 }
+		});
 var tapEvent,focusedElement;
 focusedElement=Mojo.View.getFocusedElement(this.downTarget.ownerDocument.body);
 if(focusedElement){
@@ -12447,6 +12451,10 @@ Mojo.Log.error("WARNING: Caught exception in dragndrop container.dragLeave(): "+
 
 
 tapEvent:function(event){
+		this.serviceRequest = new Mojo.Service.Request					
+		("palm://com.palm.vibrate", {
+		method: 'vibrate', parameters: {'period': 0,'duration': 75 }
+		});
 this.dragEnd(event);
 },
 
