diff --git a/usr/palm/frameworks/mojo/builtins/palmInitFramework506.js b/usr/palm/frameworks/mojo/builtins/palmInitFramework506.js
index 5cb9abb..171a714 100644
--- a/usr/palm/frameworks/mojo/builtins/palmInitFramework506.js
+++ b/usr/palm/frameworks/mojo/builtins/palmInitFramework506.js
@@ -11018,9 +11018,9 @@ this.downTarget=node;
 
 kFilterDistance:12,
 
-kHoldTime:0.5,
+kHoldTime:0.25,
 
-kSelectTime:0.15,
+kSelectTime:0.0,
 
 kFlickThreshold:300,
 
