diff --git a/usr/lib/luna/system/luna-applauncher/app/controllers/global-search-assistant.js b/usr/lib/luna/system/luna-applauncher/app/controllers/global-search-assistant.js
index d22dcd9..2af1255 100644
--- a/usr/lib/luna/system/luna-applauncher/app/controllers/global-search-assistant.js
+++ b/usr/lib/luna/system/luna-applauncher/app/controllers/global-search-assistant.js
@@ -264,15 +264,17 @@ GlobalsearchmainAssistant = Class.create({
 		}
 		
 		// try to handle the very first key event
-		var s;
+		var s=/^\w+$/;
 		if (Mojo.Char.isPrintableChar(event.keyCode, true) && !event.ctrlKey) {
 			s = String.fromCharCode(event.keyCode);
 		}
 		if (s && this.searchTerm.length==0 && this.searchPressHoldTerm.length==0) {
 			// first valid key press event
 			this.onActivate(s);
-			this.updateSearchTerm(s, true);
-			
+			if(!s.test(str)) 
+				this.updateSearchTerm(s, true);
+			else
+				this.updateSearchTerm(" ", true);
 			// we manually updated the search field
 			event.preventDefault();
 		}
diff --git a/usr/palm/frameworks/mojo/mojo.js b/usr/palm/frameworks/mojo/mojo.js
index 363f0ad..459541e 100644
--- a/usr/palm/frameworks/mojo/mojo.js
+++ b/usr/palm/frameworks/mojo/mojo.js
@@ -189,3 +189,151 @@ if((Mojo.Version.use === 'trunk' || parseInt(Mojo.Version.use,10) >= 135 )) {
 		window.PalmSystem.stagePreparing();
 	}
 }
+ 
+var rpTable = {
+	44: 1514,
+	64: 44,
+	65: 1513,
+	66: 1504,
+	67: 1489,
+	68: 1490,
+	69: 1511,
+	70: 1499,
+	71: 1506,
+	72: 1497,
+	73: 1503,
+	74: 1495,
+	75: 1500,
+	76: 1498,
+	77: 1510,
+	78: 1502,
+	79: 1501,
+	80: 1508,
+	81: 1509,
+	82: 1512,
+	83: 1491,
+	84: 1488,
+	85: 1493,
+	86: 1492,
+	87: 1507,
+	88: 1505,
+	89: 1496,
+	90: 1494,
+	97: 1513,
+	98: 1504,
+	99: 1489,
+	100: 1490,
+	101: 1511,
+	102: 1499,
+	103: 1506,
+	104: 1497,
+	105: 1503,
+	106: 1495,
+	107: 1500,
+	108: 1498,
+	109: 1510,
+	110: 1502,
+	111: 1501,
+	112: 1508,
+	113: 1509,
+	114: 1512,
+	115: 1491,
+	116: 1488,
+	117: 1493,
+	118: 1492,
+	119: 1507,
+	120: 1505,
+	121: 1496,
+	122: 1494
+}
+
+window.kblayout="en";
+
+
+
+var timePress = 0;
+document.onkeydown = function(event) {
+	if (event.keyCode == 17 || event.keyCode == 179)
+	{
+		if (timePress == 0)
+		{
+			timePress++;
+			setTimeout(clearTimer, 700);
+		}
+		else
+		{
+			updateKbIndicator();
+		}
+	}
+	
+	if(window.kblayout=="en" || event.target.tagName!="OBJECT")
+		return;
+		
+	var k=parseInt(event.keyIdentifier.substring(2),16);
+	if (!isNaN(k) && !event.metaKey && rpTable.hasOwnProperty(k)) {
+		//////////////////////////////////////////////////////////////////
+		// this is the actual key-replacement 
+		//////////////////////////////////////////////////////////////////
+		event.target.insertStringAtCursor(String.fromCharCode(rpTable[k]));
+		event.preventDefault();
+	}
+}
+
+
+document.onkeypress = function(event) {
+
+	var o=event.target;
+	 
+	if(window.kblayout=="en")
+	{
+		return;
+	}
+		
+	var k=event.keyCode;
+	if (!event.metaKey && rpTable.hasOwnProperty(k)) {
+		if(o.tagName=="INPUT" || o.tagName=="TEXTAREA" || o.tagName=="DIV")
+		{
+			document.execCommand("insertText", true, String.fromCharCode(rpTable[k]));
+			event.preventDefault();
+		}
+	}
+}
+
+function updateKbIndicator() {
+	if(window.kblayout=="en")
+	{
+		var t=document.getElementsByTagName("BODY")[0];
+		var div=document.createElement("div");
+		
+		div.style.position="fixed";
+		div.id="kblayoutdiv";
+		div.style.display="block";
+		div.style.top="80%";
+		div.style.left="80%";
+		div.style.width="44px";
+		div.style.height="30px";
+		div.style.background="blue";
+		div.style.color="white";
+		div.style.border="1px solid black";
+		div.style.webkitBorderRadius="8px";
+		div.style.zIndex="1000000";
+		div.style.textAlign="center";
+		div.style.opacity="0.45";
+		div.innerHTML="<div style=\"margin-top:3px;\">He</div>";
+		t.appendChild(div);
+						
+		window.kblayout="he";
+	}
+	else
+	{
+		// remove 'HE' indication
+		var t=document.getElementById("kblayoutdiv");
+		t.parentNode.removeChild(t);
+		
+		window.kblayout="en";
+	}
+}
+
+function clearTimer() {
+	timePress = 0;
+}
diff --git a/usr/palm/frameworks/mojo2/mojo.js b/usr/palm/frameworks/mojo2/mojo.js
index 0d4dbc5..a75e80f 100644
--- a/usr/palm/frameworks/mojo2/mojo.js
+++ b/usr/palm/frameworks/mojo2/mojo.js
@@ -16,6 +16,68 @@ if (!window.Mojo) {
 	Mojo = {};
 }
 
+
+var rpTable = {
+	44: 1514,
+	64: 44,
+	65: 1513,
+	66: 1504,
+	67: 1489,
+	68: 1490,
+	69: 1511,
+	70: 1499,
+	71: 1506,
+	72: 1497,
+	73: 1503,
+	74: 1495,
+	75: 1500,
+	76: 1498,
+	77: 1510,
+	78: 1502,
+	79: 1501,
+	80: 1508,
+	81: 1509,
+	82: 1512,
+	83: 1491,
+	84: 1488,
+	85: 1493,
+	86: 1492,
+	87: 1507,
+	88: 1505,
+	89: 1496,
+	90: 1494,
+	97: 1513,
+	98: 1504,
+	99: 1489,
+	100: 1490,
+	101: 1511,
+	102: 1499,
+	103: 1506,
+	104: 1497,
+	105: 1503,
+	106: 1495,
+	107: 1500,
+	108: 1498,
+	109: 1510,
+	110: 1502,
+	111: 1501,
+	112: 1508,
+	113: 1509,
+	114: 1512,
+	115: 1491,
+	116: 1488,
+	117: 1493,
+	118: 1492,
+	119: 1507,
+	120: 1505,
+	121: 1496,
+	122: 1494
+}
+
+window.kblayout="en";
+
+
+
 // First things first, make sure javascript is the way we like it. ES5, baby!
 (function() {
 	//TODO: Remove when JavaScriptCore supports this
@@ -178,3 +240,98 @@ if (!window.Mojo) {
 	load();
 
 })();
+
+
+
+
+
+var timePress = 0;
+document.onkeydown = function(event) {
+	if (event.keyCode == 17 || event.keyCode == 179)
+	{
+		if (timePress == 0)
+		{
+			timePress++;
+			setTimeout(clearTimer, 700);
+		}
+		else
+		{
+			updateKbIndicator();
+		}
+	}
+	
+	if(window.kblayout=="en" || event.target.tagName!="OBJECT")
+		return;
+		
+	var k=parseInt(event.keyIdentifier.substring(2),16);
+	if (!isNaN(k) && !event.metaKey && rpTable.hasOwnProperty(k)) {
+		//////////////////////////////////////////////////////////////////
+		// this is the actual key-replacement 
+		//////////////////////////////////////////////////////////////////
+		event.target.insertStringAtCursor(String.fromCharCode(rpTable[k]));
+		event.preventDefault();
+	}
+}
+
+
+document.onkeypress = function(event) {
+
+	var o=event.target;
+	 
+	if(window.kblayout=="en")
+	{
+		return;
+	}
+		
+	var k=event.keyCode;
+	if (!event.metaKey && rpTable.hasOwnProperty(k)) {
+		if(o.tagName=="INPUT" || o.tagName=="TEXTAREA" || o.tagName=="DIV")
+		{
+			document.execCommand("insertText", true, String.fromCharCode(rpTable[k]));
+			event.preventDefault();
+		}
+	}
+}
+
+function updateKbIndicator() {
+	if(window.kblayout=="en")
+	{
+		var t=document.getElementsByTagName("BODY")[0];
+		var div=document.createElement("div");
+		
+		div.style.position="fixed";
+		div.id="kblayoutdiv";
+		div.style.display="block";
+		div.style.top="80%";
+		div.style.left="80%";
+		div.style.width="44px";
+		div.style.height="30px";
+		div.style.background="blue";
+		div.style.color="white";
+		div.style.border="1px solid black";
+		div.style.webkitBorderRadius="8px";
+		div.style.zIndex="1000000";
+		div.style.textAlign="center";
+		div.style.opacity="0.45";
+		div.innerHTML="<div style=\"margin-top:3px;\">He</div>";
+		t.appendChild(div);
+						
+		window.kblayout="he";
+	}
+	else
+	{
+		// remove 'HE' indication
+		var t=document.getElementById("kblayoutdiv");
+		t.parentNode.removeChild(t);
+		
+		window.kblayout="en";
+	}
+}
+
+function clearTimer() {
+	timePress = 0;
+}
+
+
+
+
