diff --git a/usr/palm/frameworks/mojo/builtins/palmInitFramework506.js b/usr/palm/frameworks/mojo/builtins/palmInitFramework506.js
index 5cb9abb..eadeea0 100644
--- a/usr/palm/frameworks/mojo/builtins/palmInitFramework506.js
+++ b/usr/palm/frameworks/mojo/builtins/palmInitFramework506.js
@@ -14159,7 +14159,7 @@ FADE_ELEMENT_ATTRIBUTE:'x-mojo-scroll-fade',
 
 kMinimumSizeDifferenceForScrolling:3,
 DELTA_DISTANCE_TO_PREVENT_TAP:25,
-kFlickRatio:0.5
+kFlickRatio:0.3
 
 });
 
