diff --git a/usr/palm/frameworks/mojo/builtins/palmInitFramework363.js b/usr/palm/frameworks/mojo/builtins/palmInitFramework363.js
index 06f0f6b..2d42bd2 100644
--- a/usr/palm/frameworks/mojo/builtins/palmInitFramework363.js
+++ b/usr/palm/frameworks/mojo/builtins/palmInitFramework363.js
@@ -2930,15 +2930,7 @@ bannerParams={messageText:bannerParams};
 }
 defaultsCopy=Object.extend({},this.kDefaultBannerParams);
 bannerParams=Object.extend(defaultsCopy,bannerParams);
-this.removeBanner(category);
-try{
-bannerId=PalmSystem.addBannerMessage(bannerParams.messageText,Object.toJSON(launchArguments),
-bannerParams.icon,bannerParams.soundClass,bannerParams.soundFile,bannerParams.soundDuration);
-bannerKey=category||'banner';
-this.banners.set(bannerKey,bannerId);
-}catch(addBannerMessageException){
-Mojo.Log.error(addBannerMessageException.toString());
-}
+this.playSoundNotification(bannerParams.soundClass,bannerParams.soundFile,bannerParams.soundDuration);
 },
 
 
