diff --git a/usr/palm/frameworks/mojo/builtins/palmInitFramework347.js b/usr/palm/frameworks/mojo/builtins/palmInitFramework347.js
index aed4b02..4b6efce 100644
--- a/usr/palm/frameworks/mojo/builtins/palmInitFramework347.js
+++ b/usr/palm/frameworks/mojo/builtins/palmInitFramework347.js
@@ -14200,7 +14200,7 @@ Mojo.Widget.Scroller.MouseTracker=Class.create({
 
 initialize:function(widget,dragStartEvent,components){
 this.widget=widget;
-var downEvent=dragStartEvent.down;
+var downEvent=dragStartEvent.move;
 this.components=components;
 this.lastPointer=Event.pointer(downEvent);
 },
