diff --git a/usr/palm/frameworks/mojo/builtins/palmInitFramework347.js b/usr/palm/frameworks/mojo/builtins/palmInitFramework347.js
index aed4b02..24954fe 100644
--- a/usr/palm/frameworks/mojo/builtins/palmInitFramework347.js
+++ b/usr/palm/frameworks/mojo/builtins/palmInitFramework347.js
@@ -14109,7 +14109,7 @@ FADE_ELEMENT_ATTRIBUTE:'x-mojo-scroll-fade',
 
 kMinimumSizeDifferenceForScrolling:3,
 DELTA_DISTANCE_TO_PREVENT_TAP:25,
-kFlickRatio:0.5
+kFlickRatio:0.3
 
 });
 
