diff --git a/usr/lib/luna/system/luna-systemui/app/controllers/storagealert-assistant.js b/usr/lib/luna/system/luna-systemui/app/controllers/storagealert-assistant.js
index 558171d..dec3019 100644
--- a/usr/lib/luna/system/luna-systemui/app/controllers/storagealert-assistant.js
+++ b/usr/lib/luna/system/luna-systemui/app/controllers/storagealert-assistant.js
@@ -5,7 +5,7 @@ var StoragealertAssistant = Class.create({
 	initialize: function(showMSMWarning,showMediaWarning,disableImasq) {
 		this.showMSMWarning = showMSMWarning;		
 		this.showMediaWarning = showMediaWarning;
-		this.disableImasq = disableImasq;
+		this.disableImasq = true;
 		this.isTapOnButton = false;			
     },
     
diff --git a/usr/lib/luna/system/luna-systemui/app/controllers/usbdashboard-assistant.js b/usr/lib/luna/system/luna-systemui/app/controllers/usbdashboard-assistant.js
index 338f6d3..a652d6f 100644
--- a/usr/lib/luna/system/luna-systemui/app/controllers/usbdashboard-assistant.js
+++ b/usr/lib/luna/system/luna-systemui/app/controllers/usbdashboard-assistant.js
@@ -5,7 +5,7 @@ UsbdashboardAssistant = Class.create({
 	initialize: function(showMSMWarning, showMediaWarning,disableImasq) {
 		this.showMSMWarning = showMSMWarning;
 		this.showMediaWarning = showMediaWarning;
-		this.disableImasq = disableImasq;
+		this.disableImasq = true;
 	},
 	
 	setup: function() {
