diff --git a/usr/palm/frameworks/mojo/builtins/palmInitFramework298.js b/usr/palm/frameworks/mojo/builtins/palmInitFramework298.js
index f657e03..a067063 100644
--- a/usr/palm/frameworks/mojo/builtins/palmInitFramework298.js
+++ b/usr/palm/frameworks/mojo/builtins/palmInitFramework298.js
@@ -553,7 +553,7 @@ currentCoordinate);delete this.targetCoordinate}}},finishScroll:function(){var c
 newVal){this.targetCoordinate[component]=newVal},getComponentTarget:function(component){return this.targetCoordinate[component]},setComponentCurrent:function(component,newVal){this.currentCoordinate[component]=newVal},getComponentCurrent:function(component){return this.currentCoordinate[component]},setComponentOriginal:function(component,newVal){this.originalCoordinate[component]=newVal},getComponentOriginal:function(component){return this.originalCoordinate[component]},adjustTarget:function(deltaCoordinates){var components=
 this.components;var animateComponent;var motionForCoordinate,component,targetForComponent;var limits=this.strategy.getPermittedAxes();var limit;for(var i=components.length-1;i>=0;i--){component=components[i];animateComponent=this.animateComponent[component];limit=limits[component];targetForComponent=this.targetCoordinate[component];if(targetForComponent>limit.maxLimit||targetForComponent<limit.minLimit)motionForCoordinate=0.5*deltaCoordinates[component];else motionForCoordinate=deltaCoordinates[component];
 targetForComponent+=motionForCoordinate;if(motionForCoordinate!==0){targetForComponent=Math.max(limit.minOverLimit,targetForComponent);targetForComponent=Math.min(limit.maxOverLimit,targetForComponent);this.targetCoordinate[component]=targetForComponent}}},stopAnimating:function(){if(this.animating){this.animating=false;this.getAnimationQueue().remove(this)}},startAnimating:function(){var elementPos;if(!this.animating&&this.targetCoordinate){if(typeof this.currentCoordinate.x==="undefined"&&typeof this.currentCoordinate.y===
-"undefined"){Mojo.Log.warn("currentCoordinate not set up before call to startAnimating");elementPos=this.getScrollPosition();this.currentCoordinate.y=elementPos.top;this.currentCoordinate.x=elementPos.left}this.getAnimationQueue().add(this);this.animating=true}},FLICK_RATIO:0.5,FLICK_SPEED:0.06,OVERSCROLL_SPEED:0.3,ANIMATE_SNAP_SPEED:0.3,NONFLICK_SPEED:0.6,FADE_ELEMENT_ATTRIBUTE:"x-mojo-scroll-fade"};
+"undefined"){Mojo.Log.warn("currentCoordinate not set up before call to startAnimating");elementPos=this.getScrollPosition();this.currentCoordinate.y=elementPos.top;this.currentCoordinate.x=elementPos.left}this.getAnimationQueue().add(this);this.animating=true}},FLICK_RATIO:0.3,FLICK_SPEED:0.06,OVERSCROLL_SPEED:0.3,ANIMATE_SNAP_SPEED:0.3,NONFLICK_SPEED:0.6,FADE_ELEMENT_ATTRIBUTE:"x-mojo-scroll-fade"};
 (function(){var proto=Mojo.Widget.Scroller.Animator.prototype;proto.FADE_ELEMENT_SELECTOR="*["+proto.FADE_ELEMENT_ATTRIBUTE+"]"})();Mojo.Widget.Scroller.ComponentAnimator=function(animator,scroller,component){this.minMoveDelta=1;this.animator=animator;this.scroller=scroller;this.roundTowardZero=animator.roundTowardZero;this.component=component;this.bindHandlers();this.animateAlgorithm=this.dragAnimate;this.currentLocation="bordering";this.motionMap={flick:this.flickStateHandler,drag:this.dragStateHandler};this.motionSM=new Mojo.Widget.Scroller.StateMachine("drag",this.motionMap,{stateSwitcher:this.motionStateSwitcher});this.overscroller=
 new Mojo.Widget.Scroller.Overscroller(scroller,this,scroller.controller.window,this.scroller.controller.element);this.getTime=Date.now};
 Mojo.Widget.Scroller.ComponentAnimator.prototype={cleanup:function(){this.overscroller.cleanup()},bindHandlers:function(){this.flickStateHandler=this.flickStateHandler.bind(this);this.dragStateHandler=this.dragStateHandler.bind(this)},dragAnimate:function(deltaDist){this.done=true;return deltaDist},_zenoCalculate:function(fdr,elapsedFrames){var approachFDR;var unitDistanceRemaining;var unitDistanceTraveled;approachFDR=1-fdr;unitDistanceRemaining=Math.pow(approachFDR,elapsedFrames);unitDistanceTraveled=
diff --git a/usr/palm/frameworks/mojo/builtins/palmInitFramework367.js b/usr/palm/frameworks/mojo/builtins/palmInitFramework367.js
index b062507..ac6d3a3 100644
--- a/usr/palm/frameworks/mojo/builtins/palmInitFramework367.js
+++ b/usr/palm/frameworks/mojo/builtins/palmInitFramework367.js
@@ -14138,7 +14138,7 @@ FADE_ELEMENT_ATTRIBUTE:'x-mojo-scroll-fade',
 
 kMinimumSizeDifferenceForScrolling:3,
 DELTA_DISTANCE_TO_PREVENT_TAP:25,
-kFlickRatio:0.5
+kFlickRatio:0.3
 
 });
 
