diff --git a/usr/palm/frameworks/mojo/builtins/palmInitFramework367.js b/usr/palm/frameworks/mojo/builtins/palmInitFramework367.js
index b062507..ac6d3a3 100644
--- a/usr/palm/frameworks/mojo/builtins/palmInitFramework367.js
+++ b/usr/palm/frameworks/mojo/builtins/palmInitFramework367.js
@@ -14138,7 +14138,7 @@ FADE_ELEMENT_ATTRIBUTE:'x-mojo-scroll-fade',
 
 kMinimumSizeDifferenceForScrolling:3,
 DELTA_DISTANCE_TO_PREVENT_TAP:25,
-kFlickRatio:0.5
+kFlickRatio:0.3
 
 });
 
