diff --git a/usr/palm/frameworks/mojo.core/submission/32/javascript/animation.js b/usr/palm/frameworks/mojo.core/submission/32/javascript/animation.js
index a98a8e8..ee15df7 100644
--- a/usr/palm/frameworks/mojo.core/submission/32/javascript/animation.js
+++ b/usr/palm/frameworks/mojo.core/submission/32/javascript/animation.js
@@ -10,10 +10,10 @@
  **/
 
 var Animation = function() {
-	var targetFPS = 40;
-	var stepRate = (1/targetFPS) * 1000;
+	var targetFPS = 60;
+	var stepRate = targetFPS;
 	var showFPSUpdate = 1000;
-	var maxExtraFrames = 1; // never run more than N frames in one step.
+	var maxExtraFrames = 6; // never run more than N frames in one step.
 	var fpsBoxId = 'mojo-fps-display-box';
 
 	function createFpsEl(doc) {
diff --git a/usr/palm/frameworks/mojo.core/submission/32/javascript/scroll.js b/usr/palm/frameworks/mojo.core/submission/32/javascript/scroll.js
index e7a7f68..b37421c 100644
--- a/usr/palm/frameworks/mojo.core/submission/32/javascript/scroll.js
+++ b/usr/palm/frameworks/mojo.core/submission/32/javascript/scroll.js
@@ -10,15 +10,15 @@ var Scroll = function() {
 
 	// Private constants
 	var STEP_RATE = Animation.stepRate * 1.2,
-		MAX_TIME_SKIP = STEP_RATE * 3,
+		MAX_TIME_SKIP = STEP_RATE,
 		FLICK_SPEED = 0.06,
 		FLICK_RATIO = 0.5,
 		OVERSCROLL_TARGET_SPEED = 0.25,
 		CORRECT_OVERSCROLL_SPEED = 0.3,
 		DELAYED_STOP_MS = 150,
-		MIN_SCROLL_DELTA_SIZE = 3,
+		MIN_SCROLL_DELTA_SIZE = 1,
 		MIN_MOVE_DELTA = 1,
-		SCROLL_EVENT_RATE = 10;
+		SCROLL_EVENT_RATE = 1;
 
 	var PROPERTY_MAP = {
 		x: 'width',
