diff --git a/etc/palm/autoreplace/en_us/text-edit-autoreplace b/etc/palm/autoreplace/en_us/text-edit-autoreplace
index 64939a3..0977849 100644
--- a/etc/palm/autoreplace/en_us/text-edit-autoreplace
+++ b/etc/palm/autoreplace/en_us/text-edit-autoreplace
@@ -12,6 +12,7 @@ accademic|academic
 accademy|academy
 acccused|accused
 acceptence|acceptance
+accepteble|acceptable
 acceptible|acceptable
 accessable|accessible
 accidentaly|accidentally
@@ -52,6 +53,8 @@ acquaintence|acquaintance
 acquaintences|acquaintances
 acquiantence|acquaintance
 acquiantences|acquaintances
+aquire|acquire
+aquit|acquit
 activites|activities
 activly|actively
 actualy|actually
@@ -271,6 +274,7 @@ assualted|assaulted
 asteriod|asteroid
 asume|assume
 atempting|attempting
+athiest|atheist
 atorney|attorney
 atributed|attributed
 attemp|attempt
@@ -345,6 +349,7 @@ beleiving|believing
 belive|believe
 belived|believed
 belives|beliefs
+bellweather|bellwether
 beng|being
 benificial|beneficial
 benifit|benefit
@@ -377,6 +382,7 @@ cacuses|caucuses
 cahracters|characters
 calaber|caliber
 calander|calendar
+calender|calendar
 caluclate|calculate
 caluclated|calculated
 caluculate|calculate
@@ -424,6 +430,7 @@ chalenging|challenging
 challange|challenge
 challanged|challenged
 challege|challenge
+changable|changeable
 Champange|Champagne
 charachter|character
 charachters|characters
@@ -457,8 +464,10 @@ coctail|cocktail
 colateral|collateral
 colelctive|collective
 collecton|collection
+collectable|collectible
 collegue|colleague
 collegues|colleagues
+colum|column
 comander|commander
 comany|company
 comapany|company
@@ -526,6 +535,9 @@ confortable|comfortable
 congradulations|congratulations
 congresional|congressional
 conscent|consent
+conchience|conscience
+conchiensious|conscientious
+conscous|conscious
 consciouness|consciousness
 consdider|consider
 consdidered|considered
@@ -622,6 +634,7 @@ curch|church
 curcuit|circuit
 currenly|currently
 dael|deal
+daquiri|daiquiri
 damenor|demeanor
 decendant|descendant
 decendent|descendant
@@ -728,6 +741,7 @@ discribes|describes
 discribing|describing
 disctinction|distinction
 disctinctive|distinctive
+disipline|discipline
 disiplined|disciplined
 disolved|dissolved
 disover|discover
@@ -773,6 +787,9 @@ dramtic|dramatic
 dreasm|dreams
 driectly|directly
 drnik|drink
+drnuk|drunk
+drunkeness|drunkenness
+dumbell|dumbbell
 durig|during
 durring|during
 duting|during
@@ -856,6 +873,7 @@ epsiode|episode
 equialent|equivalent
 equiped|equipped
 equippment|equipment
+equiptment|equipment
 equivelant|equivalent
 equivelent|equivalent
 equivilant|equivalent
@@ -895,6 +913,7 @@ excange|exchange
 excecute|execute
 excecuted|executed
 excecution|execution
+excede|exceed
 excedded|exceeded
 excelent|excellent
 excellant|excellent
@@ -913,6 +932,10 @@ exerciese|exercises
 exersize|exercise
 exerternal|external
 exhibtion|exhibition
+excilirate|exhilarate
+excilarate|exhilarate
+excilirating|exhilarating
+excilarating|exhilarating
 exibition|exhibition
 exicting|exciting
 existance|existence
@@ -953,6 +976,7 @@ fidn|find
 fiel|feel
 fiels|feels
 fiercly|fiercely
+firey|fiery
 fightings|fighting
 fimilies|families
 finacial|financial
@@ -1055,6 +1079,7 @@ hapening|happening
 happend|happened
 happended|happened
 happenned|happened
+harrass|harass
 harasment|harassment
 harassement|harassment
 harrasment|harassment
@@ -1076,6 +1101,9 @@ hertiage|heritage
 hes|he's
 hesistant|hesitant
 hieght|height
+heighth|height
+hieghth|height
+heirarchy|hierarchy
 higer|higher
 higest|highest
 higway|highway
@@ -1114,6 +1142,7 @@ idaeidae|idea
 idaes|ideas
 identicial|identical
 idesa|ideas
+ignorence|ignorance
 ill|I'll
 illess|illness
 illiegal|illegal
@@ -1125,6 +1154,8 @@ imagin|imagine
 imaginery|imaginary
 imanent|imminent
 imcomplete|incomplete
+imediate|immediate
+immedeate|immediate
 imediately|immediately
 imense|immense
 imigrant|immigrant
@@ -1169,6 +1200,7 @@ indipendent|independent
 indipendently|independently
 indpendent|independent
 indpendently|independently
+indispensible|indispensable
 indulgue|indulge
 indutrial|industrial
 indviduals|individuals
@@ -1192,6 +1224,7 @@ inlcuding|including
 inmigrant|immigrant
 inmigrants|immigrants
 inocence|innocence
+innoculate|inoculate
 inofficial|unofficial
 inot|into
 inproving|improving
@@ -1250,12 +1283,15 @@ ive|I've
 iwll|will
 iwth|with
 jeapardy|jeopardy
+jewelery|jewelry
 jouney|journey
 jstu|just
 jsut|just
 judical|judicial
 juducial|judicial
+judgement|judgment
 juristiction|jurisdiction
+kernal|kernel
 kindergarden|kindergarten
 knifes|knives
 knive|knife
@@ -1279,6 +1315,8 @@ layed|laid
 leage|league
 leanr|lean
 leathal|lethal
+liason|liaison
+liesure|leisure
 lefted|left
 legitamate|legitimate
 legitmate|legitimate
@@ -1294,9 +1332,11 @@ liekd|liked
 lieuenant|lieutenant
 lieved|lived
 liftime|lifetime
+lightening|lightning
 likelyhood|likelihood
 liscense|license
 lisence|license
+licence|license
 lisense|license
 litature|literature
 literture|literature
@@ -1345,7 +1385,11 @@ meaninng|meaning
 mear|mere
 mechandise|merchandise
 medacine|medicine
+midieval|medieval
+midevil|medieval
+medeval|medieval
 memeber|member
+momento|memento
 menally|mentally
 messanger|messenger
 messenging|messaging
@@ -1355,6 +1399,9 @@ mileu|milieu
 miliary|military
 milion|million
 miliraty|military
+milennium|millennium
+millenium|millennium
+milenium|millennium
 millioniare|millionaire
 millitary|military
 millon|million
@@ -1363,6 +1410,7 @@ minature|miniature
 ministery|ministry
 minstry|ministry
 minumum|minimum
+miniscule|minuscule
 mirrorred|mirrored
 mischeivous|mischievous
 mischevious|mischievous
@@ -1371,6 +1419,7 @@ misdameanor|misdemeanor
 misdemenor|misdemeanor
 misile|missile
 missle|missile
+mispell|misspell
 misterious|mysterious
 mistery|mystery
 misteryous|mysterious
@@ -1446,6 +1495,8 @@ noteably|notably
 noth|north
 nothern|northern
 noticable|noticeable
+noticeible|noticeable
+noticible|noticeable
 noticably|noticeably
 noticeing|noticing
 noticible|noticeable
@@ -1481,6 +1532,8 @@ occassions|occasions
 occationally|occasionally
 occour|occur
 occurance|occurrence
+ocurance|occurrence
+ocurrance|occurrence
 occured|occurred
 occurence|occurrence
 occuring|occurring
@@ -1561,6 +1614,7 @@ particulary|particularly
 pary|party
 pased|passed
 pasengers|passengers
+passtime|pastime
 paticular|particular
 payed|paid
 peageant|pageant
@@ -1587,6 +1641,7 @@ permenantly|permanently
 peronal|personal
 perosnality|personality
 perphas|perhaps
+perseverence|perseverance
 persistant|persistent
 personel|personal
 personell|personnel
@@ -1602,12 +1657,16 @@ phenomonenon|phenomenon
 phenomonon|phenomenon
 philisophy|philosophy
 philosphy|philosophy
+pheonix|Phoenix
+phoenix|Phoenix
 physicaly|physically
 pich|pitch
 pinapple|pineapple
 pinnaple|pineapple
 planed|planned
 plausable|plausible
+playright|playwright
+playwrite|playwright
 pleasent|pleasant
 plesant|pleasant
 poenis|penis
@@ -1656,6 +1715,8 @@ practially|practically
 practicaly|practically
 practicly|practically
 pratice|practice
+preccede|precede
+preceede|precede
 precedessor|predecessor
 preceeded|preceded
 precentage|percentage
@@ -1721,6 +1782,8 @@ prominant|prominent
 promotted|promoted
 pronouced|pronounced
 pronounched|pronounced
+pronounciation|pronunciation
+pronunsiation|pronunciation
 proove|prove
 prooved|proved
 prophacy|prophecy
@@ -1749,6 +1812,9 @@ pwoer|power
 pyscic|psychic
 qtuie|quiet
 questonable|questionable
+questionaire|questionnaire
+questionnair|questionnaire
+questionair|questionnaire
 quicklyu|quickly
 quitted|quit
 qutie|quiet
@@ -1765,6 +1831,7 @@ reasearch|research
 rebiulding|rebuilding
 rebounce|rebound
 reccomend|recommend
+recomend|recommend
 reccomendations|recommendations
 reccomended|recommended
 reccommend|recommend
@@ -1774,6 +1841,7 @@ reched|reached
 recided|resided
 recident|resident
 recidents|residents
+reciept|receipt
 recieve|receive
 recieved|received
 reciever|receiver
@@ -1795,6 +1863,8 @@ recuiting|recruiting
 rediculous|ridiculous
 refect|reflect
 refered|referred
+refferred|referred
+reffered|referred
 referiang|referring
 refering|referring
 refernces|references
@@ -1824,6 +1894,7 @@ releived|relieved
 releiver|reliever
 releses|releases
 relevent|relevant
+relavent|relevant
 religeous|religious
 religous|religious
 relitavely|relatively
@@ -1850,6 +1921,8 @@ repsectively|respectively
 requirment|requirement
 requred|required
 resaurant|restaurant
+restarant|restaurant
+restuarant|restaurant
 resembelance|resemblance
 resembes|resembles
 resemblence|resemblance
@@ -1889,7 +1962,9 @@ returnd|returned
 revolutionar|revolutionary
 rewriet|rewrite
 rhythem|rhythm
+rhythum|rhythm
 rhythim|rhythm
+rime|rhyme
 rininging|ringing
 rised|rose
 rocord|record
@@ -1917,6 +1992,7 @@ sattelites|satellites
 saught|sought
 saveing|saving
 schedual|schedule
+scedule|schedule
 scholarhip|scholarship
 scientfic|scientific
 scientifc|scientific
@@ -1948,6 +2024,8 @@ seperating|separating
 seperation|separation
 seperatist|separatist
 sergent|sergeant
+sergant|sergeant
+sergaent|sergeant
 settelement|settlement
 settlment|settlement
 severeal|several
@@ -2123,6 +2201,8 @@ sufficent|sufficient
 sumary|summary
 sunglases|sunglasses
 suop|soup
+supercede|supersede
+superceed|supersede
 suphisticated|sophisticated
 supose|suppose
 suposed|supposed
@@ -2292,8 +2372,11 @@ troups|troops
 truely|truly
 turnk|trunk
 tust|trust
+twelth|twelfth
+twelveth|twelfth
 twon|town
 twpo|two
+tyrany|tyranny
 tyhat|that
 tyhe|they
 typcial|typical
@@ -2403,9 +2486,11 @@ wasnt|wasn't
 wass|was
 watn|want
 weaponary|weaponry
+wether|weather
 weas|was
 wehn|when
 weild|wield
+wierd|weird
 wereabouts|whereabouts
 werent|weren't
 weve|we've
