diff --git a/usr/palm/frameworks/mojo/builtins/palmInitFramework2205.js b/usr/palm/frameworks/mojo/builtins/palmInitFramework2205.js
index 125d606..1816fec 100644
--- a/usr/palm/frameworks/mojo/builtins/palmInitFramework2205.js
+++ b/usr/palm/frameworks/mojo/builtins/palmInitFramework2205.js
@@ -453,7 +453,7 @@ this.kHoldTime*1E3);this.downEvent=_.extend({},event);this.velocityHistory=[];th
 if(downHighlightTarget){this.downHighlightVersion=Mojo.Gesture._newSelectionFeedback;downHighlightMode=downHighlightTarget.getAttribute(Mojo.Gesture.selectionHighlightFeedbackAttribute);this.downMode=downHighlightMode;if(downHighlightMode===Mojo.Gesture.immediateFeedback||downHighlightMode===Mojo.Gesture.spontaneousFeedback||downHighlightMode===Mojo.Gesture.immediatePersistentFeedback)this.applySelectHighlight(downHighlightTarget);applySelectHighlightTimerHandler=this.applySelectHighlightFromTimer.bind(this);
 this.selectTimer=_.delay(applySelectHighlightTimerHandler,this.kSelectTime*1E3);this.clearSelected=this.clearSelected.bind(this)}else{downHighlightTarget=Mojo.View.findParentByAttribute(this.downTarget,this.document,Mojo.Gesture.selectionHighlightAttribute);if(downHighlightTarget){this.downHighlightVersion=Mojo.Gesture._oldSelectionFeedback;downHighlightMode=downHighlightTarget.getAttribute(Mojo.Gesture.selectionHighlightAttribute);if(downHighlightMode===Mojo.Gesture.immediateSelection)this.applySelectHighlight(downHighlightTarget)}applySelectHighlightTimerHandler=
 this.applySelectHighlightFromTimer.bind(this);this.selectTimer=_.delay(applySelectHighlightTimerHandler,this.kSelectTime*1E3);this.clearSelected=this.clearSelected.bind(this)}};
-Mojo.Gesture.Recognizer.prototype={setDownTarget:function(node){this.downTarget=node},kFilterDistance:12,kHoldTime:0.5,kSelectTime:0.15,kFlickThreshold:300,filterMousePosition:function(event,currentPointer){var dist=Mojo.Gesture.calculateAbsDistance(this.originalPointer,currentPointer);if(this.simulateFlick){if(!this.filter.x||dist.x>=this.kFilterDistance){event.filteredX=currentPointer.x;this.filter.x=false}else event.filteredX=this.originalPointer.x;if(!this.filter.y||dist.y>=this.kFilterDistance){this.filter.y=
+Mojo.Gesture.Recognizer.prototype={setDownTarget:function(node){this.downTarget=node},kFilterDistance:12,kHoldTime:0.25,kSelectTime:0.0,kFlickThreshold:300,filterMousePosition:function(event,currentPointer){var dist=Mojo.Gesture.calculateAbsDistance(this.originalPointer,currentPointer);if(this.simulateFlick){if(!this.filter.x||dist.x>=this.kFilterDistance){event.filteredX=currentPointer.x;this.filter.x=false}else event.filteredX=this.originalPointer.x;if(!this.filter.y||dist.y>=this.kFilterDistance){this.filter.y=
 false;event.filteredY=currentPointer.y}else event.filteredY=this.originalPointer.y;event.filteredPointer={x:event.filteredX,y:event.filteredY}}return dist},calculateVelocity:function(event,currentPointer){var deltaT,currentVelocity,aveX,aveY;if(this.lastPointer){this.lastDelta=Mojo.Gesture.calculateDistance(currentPointer,this.lastPointer);deltaT=event.timeStamp-this.lastTimeStamp;if(deltaT>0)currentVelocity={x:Math.round(1E3*this.lastDelta.x/deltaT),y:Math.round(1E3*this.lastDelta.y/deltaT)};else currentVelocity=
 {x:0,y:0};if(this.velocityHistory.length>=2)this.velocityHistory.shift();this.velocityHistory.push(currentVelocity);if(this.velocityHistory.length>=2){aveX=Math.round(this.velocityHistory[1].x*0.7+this.velocityHistory[0].x*0.3);aveY=Math.round(this.velocityHistory[1].y*0.7+this.velocityHistory[0].y*0.3);this.velocity={x:aveX,y:aveY}}else this.velocity=currentVelocity;event.distance=this.lastDelta}this.lastPointer=currentPointer;this.lastTimeStamp=event.timeStamp},mouseDown:function(event){},stopSelectTimer:function(){if(this.selectTimer){window.clearTimeout(this.selectTimer);
 delete this.selectTimer}},applySelectHighlight:function(hitTarget){var currentTarget=Mojo.View.findParentByAttribute(hitTarget,this.document,Mojo.Gesture.selectionHighlightFeedbackAttribute);if(currentTarget){this.downHighlightVersion=Mojo.Gesture._newSelectionFeedback;if(currentTarget&&currentTarget!==document&&!this.preventTap){this.highlightedElement=currentTarget;this.highlightTargetTime=Date.now();Mojo.Gesture.highlightTarget=currentTarget;Mojo.Gesture.highlightTargetTime=this.highlightTargetTime;
diff --git a/usr/palm/frameworks/mojo/builtins/palmInitFramework506.js b/usr/palm/frameworks/mojo/builtins/palmInitFramework506.js
index 5cb9abb..171a714 100644
--- a/usr/palm/frameworks/mojo/builtins/palmInitFramework506.js
+++ b/usr/palm/frameworks/mojo/builtins/palmInitFramework506.js
@@ -11018,9 +11018,9 @@ this.downTarget=node;
 
 kFilterDistance:12,
 
-kHoldTime:0.5,
+kHoldTime:0.25,
 
-kSelectTime:0.15,
+kSelectTime:0.0,
 
 kFlickThreshold:300,
 
