diff --git a/usr/palm/frameworks/mojo/builtins/palmInitFramework347.js b/usr/palm/frameworks/mojo/builtins/palmInitFramework347.js
index aed4b02..8bc67d0 100644
--- a/usr/palm/frameworks/mojo/builtins/palmInitFramework347.js
+++ b/usr/palm/frameworks/mojo/builtins/palmInitFramework347.js
@@ -5834,7 +5834,7 @@ Mojo.Animation.kCSSAppMenuAnimationTimingFunction='ease-out';
 Mojo.Animation.kCSSScrimAnimationDuration=0.1;
 Mojo.Animation.kCSSScrimAnimationTimingFunction=undefined;
 
-Mojo.Animation.targetFPS=40;
+Mojo.Animation.targetFPS=100;
 Mojo.Animation.stepRate=(1/Mojo.Animation.targetFPS)*1000;
 Mojo.Animation.showFPS=false;
 Mojo.Animation.showFPSUpdate=1000;
