diff --git a/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js b/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
index 0e239cc..64d5103 100644
--- a/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
+++ b/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
@@ -338,7 +338,7 @@ handleAppName:function(payload) {
 		this.controller.window.clearTimeout(this.delayedUpdateAppMenuTitle);
 		this.delayedUpdateAppMenuTitle = undefined;
 	}
-	this.delayedUpdateAppMenuTitle = this.controller.window.setTimeout(this.updateAppMenuTitle.bind(this, payload), 500);
+	this.delayedUpdateAppMenuTitle = this.controller.window.setTimeout(this.updateAppMenuTitle.bind(this, payload), 0);
 },
 
 updateAppMenuTitle: function(payload) {
