--- .orig/usr/lib/luna/system/luna-systemui/app/controllers/storagealert-assistant.js
+++ /usr/lib/luna/system/luna-systemui/app/controllers/storagealert-assistant.js
@@ -5,7 +5,7 @@
 	initialize: function(showMSMWarning,showMediaWarning,disableImasq) {
 		this.showMSMWarning = showMSMWarning;		
 		this.showMediaWarning = showMediaWarning;
-		this.disableImasq = disableImasq;
+		this.disableImasq = true;
 		this.isTapOnButton = false;			
     },
     
--- .orig/usr/lib/luna/system/luna-systemui/app/controllers/usbdashboard-assistant.js
+++ /usr/lib/luna/system/luna-systemui/app/controllers/usbdashboard-assistant.js
@@ -5,7 +5,7 @@
 	initialize: function(showMSMWarning, showMediaWarning,disableImasq) {
 		this.showMSMWarning = showMSMWarning;
 		this.showMediaWarning = showMediaWarning;
-		this.disableImasq = disableImasq;
+		this.disableImasq = true;
 	},
 	
 	setup: function() {

