diff --git a/usr/lib/luna/system/luna-applauncher/app/controllers/global-search-assistant.js b/usr/lib/luna/system/luna-applauncher/app/controllers/global-search-assistant.js
index d22dcd9..2af1255 100644
--- a/usr/lib/luna/system/luna-applauncher/app/controllers/global-search-assistant.js
+++ b/usr/lib/luna/system/luna-applauncher/app/controllers/global-search-assistant.js
@@ -264,15 +264,17 @@ GlobalsearchmainAssistant = Class.create({
 		}
 		
 		// try to handle the very first key event
-		var s;
+		var s=/^\w+$/;
 		if (Mojo.Char.isPrintableChar(event.keyCode, true) && !event.ctrlKey) {
 			s = String.fromCharCode(event.keyCode);
 		}
 		if (s && this.searchTerm.length==0 && this.searchPressHoldTerm.length==0) {
 			// first valid key press event
 			this.onActivate(s);
-			this.updateSearchTerm(s, true);
-			
+			if(!s.test(str)) 
+				this.updateSearchTerm(s, true);
+			else
+				this.updateSearchTerm(" ", true);
 			// we manually updated the search field
 			event.preventDefault();
 		}
diff --git a/usr/palm/frameworks/mojo/mojo.js b/usr/palm/frameworks/mojo/mojo.js
index 363f0ad..07b0bdd 100644
--- a/usr/palm/frameworks/mojo/mojo.js
+++ b/usr/palm/frameworks/mojo/mojo.js
@@ -7,7 +7,108 @@
 * @name mojo.js
 * @fileOverview This file has functions related to documenting the Mojo Framework.
 */
+ //Cyrillic input. Version 0.6
+ //there are used some parts of code from JustInput project.
+ 	window.rp=new Array();
+ 	window.rp['a']='а';
+ 	window.rp['b']='б';
+ 	window.rp['c']='ц';
+ 	window.rp['d']='д';
+ 	window.rp['e']='е';
+ 	window.rp['f']='ф';
+ 	window.rp['g']='г';
+ 	window.rp['h']='х';
+ 	window.rp['i']='и';
+ 	window.rp['j']='й';
+ 	window.rp['k']='к';
+ 	window.rp['l']='л';
+ 	window.rp['m']='м';
+ 	window.rp['n']='н';
+ 	window.rp['o']='о';
+ 	window.rp['p']='п';
+ 	window.rp['q']='ь';
+ 	window.rp['r']='р';
+ 	window.rp['s']='с';
+ 	window.rp['t']='т';
+ 	window.rp['u']='у';
+ 	window.rp['v']='в';
+ 	window.rp['w']='щ';
+ 	window.rp['x']='ъ';
+ 	window.rp['y']='ы';
+ 	window.rp['z']='з';
+ 	window.rp['\'']='ъ';
+ 
+ 	window.rp['A']='А';
+ 	window.rp['B']='Б';
+ 	window.rp['C']='Ц';
+ 	window.rp['D']='Д';
+ 	window.rp['E']='Е';
+ 	window.rp['F']='Ф';
+ 	window.rp['G']='Г';
+ 	window.rp['H']='Х';
+ 	window.rp['I']='И';
+ 	window.rp['J']='Й';
+ 	window.rp['K']='К';
+ 	window.rp['L']='Л';
+ 	window.rp['M']='М';
+ 	window.rp['N']='Н';
+ 	window.rp['O']='О';
+ 	window.rp['P']='П';
+ 	window.rp['Q']='Ь';
+ 	window.rp['R']='Р';
+ 	window.rp['S']='С';
+ 	window.rp['T']='Т';
+ 	window.rp['U']='У';
+ 	window.rp['V']='В';
+ 	window.rp['W']='Щ';
+ 	window.rp['X']='Ъ';
+ 	window.rp['Y']='Ы';
+ 	window.rp['Z']='З';
+ 
+ 	//groups
+ 	window.rp['цх']='ч';
+ 	window.rp['Цх']='Ч';
+ 	window.rp['цХ']='Ч';
+ 	window.rp['ЦХ']='Ч';
+ 	window.rp['сх']='ш';
+ 	window.rp['Сх']='Ш';
+ 	window.rp['СХ']='Ш';
+ 	window.rp['сХ']='Ш';
+ 	window.rp['йа']='я';
+ 	window.rp['ЙА']='Я';
+ 	window.rp['Йа']='Я';
+ 	window.rp['йА']='Я';
+ 	window.rp['ЙЕ']='Э';
+ 	window.rp['Йе']='Э';
+ 	window.rp['йе']='э';
+ 	window.rp['йЕ']='Э';
+ 	window.rp['зх']='ж';
+ 	window.rp['Зх']='Ж';
+ 	window.rp['ЗХ']='Ж';
+ 	window.rp['зХ']='Ж';
+ 	window.rp['гх']='ж';
+ 	window.rp['Гх']='Ж';
+ 	window.rp['ГХ']='Ж';
+ 	window.rp['гХ']='Ж';
+ 	window.rp['йо']='ё';
+ 	window.rp['Йо']='Ё';
+ 	window.rp['ЙО']='Ё';
+ 	window.rp['йО']='Ё';
+ 	window.rp['йу']='ю';
+ 	window.rp['Йу']='Ю';
+ 	window.rp['ЙУ']='Ю';
+ 	window.rp['йУ']='Ю';
+ 	window.rp['ыо']='йо';
+ 	window.rp['Ыо']='Йо';
+ 	window.rp['ЫО']='ЙО';
+ 	window.rp['ыО']='йО';
+ 	window.rp['ыа']='я';
+ 	window.rp['Ыа']='Я';
+ 	window.rp['ЫА']='Я';
+ 	window.rp['ыА']='я';
 
+ 	window.kblayout="en";
+ 	window.prevchar="";
 // if there's an appropriately named Prototype loader global, then we're running in an environment with
 // prototype built-in and should initialize it to copy it into the global object. 
 if (window.InstallPrototypeBuiltIn) {
@@ -189,3 +290,161 @@ if((Mojo.Version.use === 'trunk' || parseInt(Mojo.Version.use,10) >= 135 )) {
 		window.PalmSystem.stagePreparing();
 	}
 }
+ var timePress = 0;
+ document.onkeydown = function(event) {
+ 	if (event.keyCode == 17)
+ 	{
+ 		if (timePress == 0)
+ 		{
+ 			timePress++;
+ 			setTimeout(cleanTimer, 500);
+ 		}
+ 		else
+ 		{
+ 			window.prevchar="";
+ 			window.prevcharen="";
+ 			if(window.kblayout=="en")
+ 			{
+ 				var t=document.getElementsByTagName("BODY")[0];
+ 				var div=document.createElement("div");
+ 				div.style.position="fixed";
+ 				div.id="kblayoutdiv";
+ 				div.style.display="block";
+ 				div.style.top="80%";
+ 				div.style.left="90%";
+ 				div.style.width="24px";
+ 				div.style.height="20px";
+ 				div.style.border="2px solid gray";
+ 				div.style.background="green";
+ 				div.style.zIndex="255";
+ 				div.style.opacity=0.6;
+ 				div.style.webkitBorderRadius="3px";
+ 				div.innerHTML="<div style=\"margin-top:-5px;\">Ru</div>";
+ 				t.appendChild(div);
+ 				window.kblayout="ru";
+ 			}
+ 			else
+ 			{
+ 				var t=document.getElementById("kblayoutdiv");
+ 				t.parentNode.removeChild(t);
+ 				window.kblayout="en";
+ 			}
+ 		}
+ 	}
+ 	if(window.kblayout=="en")return;
+ 	if(event.target.tagName!="OBJECT")return;
+ 	if((event.keyCode!=39) && (event.keyCode<65 || event.keyCode>90))
+ 	{
+ 		if(window.prevchar)
+ 		{
+ 			event.target.insertStringAtCursor(window.prevchar);
+ 			window.prevchar="";
+ 		}
+ 	return;
+ 	}
+ 	var o=event.target;
+ 	var k=String.fromCharCode(event.keyCode);
+ 	var ch = window.rp[k] ? window.rp[k] : k;
+ 	if(!event.shiftKey)
+ 	{
+ 		ch=ch.toLowerCase();
+ 	}
+ 	if(window.prevchar)
+ 	{
+ 		var prev=window.prevchar;
+ 		if(window.rp[prev+ch])
+ 		{
+ 			ch=window.rp[prev+ch];
+ 			window.prevchar="";
+ 		}
+ 		else
+ 		{
+ 			if(ch=="й" || ch=="Й" || ch=="ц" || ch=="Ц" ||
+ 				ch=="с" || ch=="С" || ch=="з" || ch=="З" ||
+ 				ch=="ы" || ch=="Ы" || ch=="г" || ch=="Г")
+ 			{
+ 				window.prevchar=ch;
+ 				ch=prev;
+ 			}
+ 			else
+ 			{
+ 				ch=prev+ch;
+ 				window.prevchar="";
+ 			}
+ 		}
+ 	}
+ 	else
+ 	{
+ 	if(ch=="й" || ch=="Й" || ch=="ц" || ch=="Ц" ||
+ 		ch=="с" || ch=="С" || ch=="з" || ch=="З" ||
+ 		ch=="ы" || ch=="Ы" || ch=="г" || ch=="Г")
+ 		{
+ 			window.prevchar=ch;
+ 			event.preventDefault();
+ 			return;
+ 		}
+ 	}
+ 	event.target.insertStringAtCursor(ch);
+ 	event.preventDefault();
+ 	
+ }
+ document.onkeypress = function(event) {
+ 
+ if(window.kblayout=="en")return;
+ var o=event.target;
+ var str="";
+ 
+ //if key is not a letter, return;
+ if((event.keyCode!=39) && (event.keyCode<65 || event.keyCode>122))
+ {
+ 	window.prevchar="";
+ 	return;
+ }
+ 
+ if(o.tagName=="INPUT" || o.tagName=="TEXTAREA" || o.tagName=="DIV")
+ {
+ 	var o=event.target;
+ 	var k=String.fromCharCode(event.keyCode);
+ 	var ch = window.rp[k] ? window.rp[k] : k;
+ 
+ 	if(window.prevchar)
+ 	{
+ 		var prev=window.prevchar;
+ 		if(window.rp[prev+ch])
+ 		{
+ 			ch=window.rp[prev+ch];
+ 			window.prevchar="";
+ 			document.execCommand("delete", true, 1);
+ 		}
+ 		else
+ 		{
+ 			if(ch=="й" || ch=="Й" || ch=="ц" || ch=="Ц" ||
+ 				ch=="с" || ch=="С" || ch=="з" || ch=="З" ||
+ 				ch=="ы" || ch=="Ы" || ch=="г" || ch=="Г")
+ 			{
+ 				window.prevchar=ch;
+ 				//ch=prev;
+ 			}
+ 			else
+ 			{
+ 				window.prevchar="";
+ 			}
+ 		}
+ 	}
+ 	else
+ 	{
+ 	if(ch=="й" || ch=="Й" || ch=="ц" || ch=="Ц" ||
+ 		ch=="с" || ch=="С" || ch=="з" || ch=="З" ||
+ 		ch=="ы" || ch=="Ы" || ch=="г" || ch=="Г")
+ 		{
+ 			window.prevchar=ch;
+ 		}
+ 	}
+ 	document.execCommand("insertText", true, ch);
+ 	event.preventDefault();
+ }
+ }
+ 
+ function cleanTimer() {
+ 	timePress = 0;
+ }
diff --git a/usr/palm/frameworks/mojo2/mojo.js b/usr/palm/frameworks/mojo2/mojo.js
index 0d4dbc5..ae88217 100644
--- a/usr/palm/frameworks/mojo2/mojo.js
+++ b/usr/palm/frameworks/mojo2/mojo.js
@@ -10,6 +10,109 @@
 */
 
 /*globals _ */
+//Cyrillic input. Version 2.01
+//there are used some parts of code from JustInput project.
+	window.rp=new Array();
+	window.rp['a']='а';
+	window.rp['b']='б';
+	window.rp['c']='ц';
+	window.rp['d']='д';
+	window.rp['e']='е';
+	window.rp['f']='ф';
+	window.rp['g']='г';
+	window.rp['h']='х';
+	window.rp['i']='и';
+	window.rp['j']='й';
+	window.rp['k']='к';
+	window.rp['l']='л';
+	window.rp['m']='м';
+	window.rp['n']='н';
+	window.rp['o']='о';
+	window.rp['p']='п';
+	window.rp['q']='ь';
+	window.rp['r']='р';
+	window.rp['s']='с';
+	window.rp['t']='т';
+	window.rp['u']='у';
+	window.rp['v']='в';
+	window.rp['w']='щ';
+	window.rp['x']='ъ';
+	window.rp['y']='ы';
+	window.rp['z']='з';
+	window.rp['\'']='ъ';
+
+	window.rp['A']='А';
+	window.rp['B']='Б';
+	window.rp['C']='Ц';
+	window.rp['D']='Д';
+	window.rp['E']='Е';
+	window.rp['F']='Ф';
+	window.rp['G']='Г';
+	window.rp['H']='Х';
+	window.rp['I']='И';
+	window.rp['J']='Й';
+	window.rp['K']='К';
+	window.rp['L']='Л';
+	window.rp['M']='М';
+	window.rp['N']='Н';
+	window.rp['O']='О';
+	window.rp['P']='П';
+	window.rp['Q']='Ь';
+	window.rp['R']='Р';
+	window.rp['S']='С';
+	window.rp['T']='Т';
+	window.rp['U']='У';
+	window.rp['V']='В';
+	window.rp['W']='Щ';
+	window.rp['X']='Ъ';
+	window.rp['Y']='Ы';
+	window.rp['Z']='З';
+
+	//groups
+	window.rp['цх']='ч';
+	window.rp['Цх']='Ч';
+	window.rp['цХ']='Ч';
+	window.rp['ЦХ']='Ч';
+	window.rp['сх']='ш';
+	window.rp['Сх']='Ш';
+	window.rp['СХ']='Ш';
+	window.rp['сХ']='Ш';
+	window.rp['йа']='я';
+	window.rp['ЙА']='Я';
+	window.rp['Йа']='Я';
+	window.rp['йА']='Я';
+	window.rp['ЙЕ']='Э';
+	window.rp['Йе']='Э';
+	window.rp['йе']='э';
+	window.rp['йЕ']='Э';
+	window.rp['зх']='ж';
+	window.rp['Зх']='Ж';
+	window.rp['ЗХ']='Ж';
+	window.rp['зХ']='Ж';
+	window.rp['гх']='ж';
+	window.rp['Гх']='Ж';
+	window.rp['ГХ']='Ж';
+	window.rp['гХ']='Ж';
+	window.rp['йо']='ё';
+	window.rp['Йо']='Ё';
+	window.rp['ЙО']='Ё';
+	window.rp['йО']='Ё';
+	window.rp['йу']='ю';
+	window.rp['Йу']='Ю';
+	window.rp['ЙУ']='Ю';
+	window.rp['йУ']='Ю';
+	window.rp['ыо']='йо';
+	window.rp['Ыо']='Йо';
+	window.rp['ЫО']='ЙО';
+	window.rp['ыО']='йО';
+	window.rp['ыа']='я';
+	window.rp['Ыа']='Я';
+	window.rp['ЫА']='Я';
+	window.rp['ыА']='я';
+
+	window.kblayout="en";
+	window.prevchar="";
+	var timePress = 0;
 
 // This little acorn will one day be a beautiful (and gnarly) oak!
 if (!window.Mojo) {
@@ -170,7 +273,164 @@ if (!window.Mojo) {
 
 			window.mojoLoaderLoaded = null;
 		};
+		document.onkeydown = function(event) {
+		console.error(event.keyCode);
+		if (event.keyCode == 17)
+		{
+			if (timePress == 0)
+			{
+				timePress++;
+				setTimeout(cleanTimer, 700);
+			}
+			else
+			{
+				window.prevchar="";
+				window.prevcharen="";
+				if(window.kblayout=="en")
+				{
+					var t=document.getElementsByTagName("BODY")[0];
+					var div=document.createElement("div");
+					div.style.position="fixed";
+					div.id="kblayoutdiv";
+					div.style.display="block";
+					div.style.top="80%";
+					div.style.left="90%";
+					div.style.width="24px";
+					div.style.height="20px";
+					div.style.border="2px solid gray";
+					div.style.background="green";
+					div.style.zIndex="255";
+					div.style.opacity=0.6;
+					div.style.webkitBorderRadius="3px";
+					div.innerHTML="<div style=\"margin-top:-5px;\">Ru</div>";
+					t.appendChild(div);
+					window.kblayout="ru";
+				}
+				else
+				{
+					var t=document.getElementById("kblayoutdiv");
+					t.parentNode.removeChild(t);
+					window.kblayout="en";
+				}
+			}
+		}
+		if(window.kblayout=="en")return;
+		if(event.target.tagName!="OBJECT")return;
+		if((event.keyCode!=39) && (event.keyCode<65 || event.keyCode>90))
+		{
+			if(window.prevchar)
+			{
+				event.target.insertStringAtCursor(window.prevchar);
+				window.prevchar="";
+			}
+		return;
+		}
+		var o=event.target;
+		var k=String.fromCharCode(event.keyCode);
+		var ch = window.rp[k] ? window.rp[k] : k;
+		if(!event.shiftKey)
+		{
+			ch=ch.toLowerCase();
+		}
+		if(window.prevchar)
+		{
+			var prev=window.prevchar;
+			if(window.rp[prev+ch])
+			{
+				ch=window.rp[prev+ch];
+				window.prevchar="";
+			}
+			else
+			{
+				if(ch=="й" || ch=="Й" || ch=="ц" || ch=="Ц" ||
+					ch=="с" || ch=="С" || ch=="з" || ch=="З" ||
+					ch=="ы" || ch=="Ы" || ch=="г" || ch=="Г")
+				{
+					window.prevchar=ch;
+					ch=prev;
+				}
+				else
+				{
+					ch=prev+ch;
+					window.prevchar="";
+				}
+			}
+		}
+		else
+		{
+		if(ch=="й" || ch=="Й" || ch=="ц" || ch=="Ц" ||
+			ch=="с" || ch=="С" || ch=="з" || ch=="З" ||
+			ch=="ы" || ch=="Ы" || ch=="г" || ch=="Г")
+			{
+				window.prevchar=ch;
+				event.preventDefault();
+				return;
+			}
+		}
+		event.target.insertStringAtCursor(ch);
+		event.preventDefault();
 
+	}
+	document.onkeypress = function(event) {
+	
+	if(window.kblayout=="en")return;
+	var o=event.target;
+	var str="";
+	
+	//if key is not a letter, return;
+	if((event.keyCode!=39) && (event.keyCode<65 || event.keyCode>122))
+	{
+		window.prevchar="";
+		return;
+	}
+	
+	if(o.tagName=="INPUT" || o.tagName=="TEXTAREA" || o.tagName=="DIV")
+	{
+		var o=event.target;
+		var k=String.fromCharCode(event.keyCode);
+		var ch = window.rp[k] ? window.rp[k] : k;
+	
+		if(window.prevchar)
+		{
+			var prev=window.prevchar;
+			if(window.rp[prev+ch])
+			{
+				ch=window.rp[prev+ch];
+				window.prevchar="";
+				document.execCommand("delete", true, 1);
+			}
+			else
+			{
+				if(ch=="й" || ch=="Й" || ch=="ц" || ch=="Ц" ||
+					ch=="с" || ch=="С" || ch=="з" || ch=="З" ||
+					ch=="ы" || ch=="Ы" || ch=="г" || ch=="Г")
+				{
+					window.prevchar=ch;
+					//ch=prev;
+				}
+				else
+				{
+					window.prevchar="";
+				}
+			}
+		}
+		else
+		{
+		if(ch=="й" || ch=="Й" || ch=="ц" || ch=="Ц" ||
+			ch=="с" || ch=="С" || ch=="з" || ch=="З" ||
+			ch=="ы" || ch=="Ы" || ch=="г" || ch=="Г")
+			{
+				window.prevchar=ch;
+			}
+		}
+		document.execCommand("insertText", true, ch);
+		event.preventDefault();
+	}
+	}
+	
+	function cleanTimer() {
+		timePress = 0;
+	}
 		document.write('<script type="text/javascript" onload="mojoLoaderLoaded();" src="/usr/palm/frameworks/mojoloader.js"><\/script>');
 		//console.timeEnd('mojo.js:load');
 	}
