diff --git a/usr/palm/frameworks/mojo/builtins/palmInitFramework347.js b/usr/palm/frameworks/mojo/builtins/palmInitFramework347.js
index aed4b02..d5e8b70 100644
--- a/usr/palm/frameworks/mojo/builtins/palmInitFramework347.js
+++ b/usr/palm/frameworks/mojo/builtins/palmInitFramework347.js
@@ -2890,15 +2890,7 @@ bannerParams={messageText:bannerParams};
 }
 defaultsCopy=Object.extend({},this.kDefaultBannerParams);
 bannerParams=Object.extend(defaultsCopy,bannerParams);
-this.removeBanner(category);
-try{
-bannerId=PalmSystem.addBannerMessage(bannerParams.messageText,Object.toJSON(launchArguments),
-bannerParams.icon,bannerParams.soundClass,bannerParams.soundFile,bannerParams.soundDuration);
-bannerKey=category||'banner';
-this.banners.set(bannerKey,bannerId);
-}catch(addBannerMessageException){
-Mojo.Log.error(addBannerMessageException.toString());
-}
+this.playSoundNotification(bannerParams.soundClass,bannerParams.soundFile,bannerParams.soundDuration);
 },
 
 
