diff --git a/usr/palm/frameworks/mojo/builtins/palmInitFramework200_72.js b/usr/palm/frameworks/mojo/builtins/palmInitFramework200_72.js
index ac3129f..2690732 100644
--- a/usr/palm/frameworks/mojo/builtins/palmInitFramework200_72.js
+++ b/usr/palm/frameworks/mojo/builtins/palmInitFramework200_72.js
@@ -31,6 +31,8 @@ Mojo.Config.ACCOUNT_IMAGES_HOME=Mojo.Config.IMAGES_HOME+'/accounts';
 Mojo.Config.CSS_HOME=Mojo.Config.MOJO_HOME+Mojo.generateFrameworkHome()+'/stylesheets';
 Mojo.Config.JS_HOME=Mojo.Config.MOJO_HOME+Mojo.generateFrameworkHome()+'/javascripts';
 
+Mojo.Config.THEME_HOME="/media/internal/org.webosinternals.thmapi/.theme";
+
 Mojo.Config.debuggingEnabled=false;
 Mojo.Config.loadStylesWithLink=true;
 
@@ -399,6 +401,18 @@ Mojo.generateFrameworkComponentPaths(cssFileName,"stylesheet",localized,cssVersi
 Mojo.loadStylesheet(targetDocument,Mojo.hostingPrefix+path);
 });
 });
+var themeConfig = Mojo.loadJSONFile(Mojo.Config.THEME_HOME+'/theme_config.json');
+var appId = Mojo.appInfo.id;
+
+if (themeConfig) {
+  if (themeConfig.global && themeConfig.global.cssFileName) {
+    Mojo.loadStylesheet(targetDocument, Mojo.hostingPrefix+Mojo.Config.THEME_HOME+'/'+themeConfig.global.cssFileName);
+  }
+
+  if (themeConfig.app && themeConfig.app[appId] && themeConfig.app[appId].cssFileName) {
+    Mojo.loadStylesheet(targetDocument, Mojo.hostingPrefix+Mojo.Config.THEME_HOME+'/'+themeConfig.app[appId].cssFileName);
+  }
+}
 };
 
 
