--- .orig/media/cryptofs/apps/usr/palm/applications/com.palm.pandora/app/core/device_service.js
+++ /media/cryptofs/apps/usr/palm/applications/com.palm.pandora/app/core/device_service.js
@@ -51,7 +51,7 @@
 	},
 
 	registerForMediaEvents: function() {
-		this.mediaServiceRef = new Mojo.Service.Request('palm://com.palm.keys/media', {
+/*		this.mediaServiceRef = new Mojo.Service.Request('palm://com.palm.keys/media', {
 			method: 'status',
 			onSuccess: this.mediaCallback,
 			parameters: {subscribe: true}
@@ -55,6 +55,11 @@
 			method: 'status',
 			onSuccess: this.mediaCallback,
 			parameters: {subscribe: true}
+		});*/
+		this.mediaServiceRef = new Mojo.Service.Request('palm://com.palm.mediaevents', {
+			method: 'mediaEvents',
+			onSuccess: this.mediaCallback,
+			parameters: {appName: AppConstants.appId, subscribe: true}
 		});
 	},
 
@@ -118,7 +123,7 @@
 			return;
 		}
 
-        if (event.key == "pause" && event.state == "down") {
+/*        if (event.key == "pause" && event.state == "down") {
             AppGlobals.station.pause();
         } else if (event.key == "play" && event.state == "down") {
             AppGlobals.station.pause();
@@ -129,6 +134,6 @@
         } else if (event.key == "stop" && event.state == "down") {
             AppGlobals.station.pauseIfPlaying();
         }
-	}
+	}*/
 
 });
@@ -131,4 +136,34 @@
         }
 	}
 
+		Util.log("media event: " + event.key);
+		switch (event.key) {
+			case "next":
+				AppGlobals.station.skip();
+				break;
+
+			case "nextAndPlay":
+				this.serviceRequest = new Mojo.Service.Request("palm://com.palm.vibrate", {
+					method: 'vibrate', parameters: { 'period': 0,'duration': 750 }
+				});
+				AppGlobals.station.skip();
+				break;
+
+			case "pause":
+				AppGlobals.station.pause();
+				break;
+
+			case "play":
+				AppGlobals.station.pause();
+				break;
+
+			case "togglePausePlay":
+				AppGlobals.station.pause();
+				break;
+
+			case "stop":
+				AppGlobals.station.pauseIfPlaying();
+				break;
+		}
+	}
 });
\ No newline at end of file
