--- .orig/media/cryptofs/apps/usr/palm/applications/com.palm.pandora/app/core/device_service.js
+++ /media/cryptofs/apps/usr/palm/applications/com.palm.pandora/app/core/device_service.js
@@ -51,8 +51,8 @@
 	},
 
 	registerForMediaEvents: function() {
-		this.mediaServiceRef = new Mojo.Service.Request('palm://com.palm.keys/media', {
-			method: 'status',
+		this.mediaServiceRef = new Mojo.Service.Request('palm://com.palm.mediaevents', {
+			method: 'mediaEvents',
 			onSuccess: this.mediaCallback,
 			parameters: {subscribe: true}
 		});
@@ -54,7 +54,7 @@
 		this.mediaServiceRef = new Mojo.Service.Request('palm://com.palm.keys/media', {
 			method: 'status',
 			onSuccess: this.mediaCallback,
-			parameters: {subscribe: true}
+			parameters: {appName: AppConstants.appId, subscribe: true}
 		});
 	},
 
@@ -118,17 +118,28 @@
 			return;
 		}
 
-        if (event.key == "pause" && event.state == "down") {
-            AppGlobals.station.pause();
-        } else if (event.key == "play" && event.state == "down") {
-            AppGlobals.station.pause();
-        } else if (event.key == "next" && event.state == "down") {
-            AppGlobals.station.skip();
-        } else if (event.key == "togglePausePlay" && event.state == "down") {
-            AppGlobals.station.pause();
-        } else if (event.key == "stop" && event.state == "down") {
-            AppGlobals.station.pauseIfPlaying();
-        }
+		Util.log("media event: " + event.key);
+		switch (event.key) {
+			case "next":
+				AppGlobals.station.skip();
+				break;
+
+			case "pause":
+				AppGlobals.station.pause();
+				break;
+
+			case "play":
+				AppGlobals.station.pause();
+				break;
+
+			case "togglePausePlay":
+				AppGlobals.station.pause();
+				break;
+
+			case "stop":
+				AppGlobals.station.pauseIfPlaying();
+				break;
+		}
 	}
 
 });
\ No newline at end of file
