--- .orig/var/usr/palm/applications/com.palm.pandora/app/core/device_service.js
+++ /var/usr/palm/applications/com.palm.pandora/app/core/device_service.js
@@ -121,6 +121,9 @@
 		Util.log("media event: " + event.key);
 		switch (event.key) {
 			case "next":
+				this.serviceRequest = new Mojo.Service.Request("palm://com.palm.vibrate", {
+					method: 'vibrate', parameters: { 'period': 0,'duration': 750 }
+				});
 				AppGlobals.station.skip();
 				break;
